<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 29 Gene annotation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 29 Gene annotation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 29 Gene annotation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dsi-allCells-allSetsTop.html"/>

<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>12</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="12.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>12.1</b> Load data</a></li>
<li class="chapter" data-level="12.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>12.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="12.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>12.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="12.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-8"><i class="fa fa-check"></i><b>12.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>13</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="13.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>13.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="13.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>13.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="13.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>13.4</b> Batch correction</a><ul>
<li class="chapter" data-level="13.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>13.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>13.5</b> SCTransform</a><ul>
<li class="chapter" data-level="13.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>13.5.1</b> Batch only</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>13.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="13.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>13.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="13.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>13.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>13.7</b> fastMNN</a></li>
<li class="chapter" data-level="13.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>13.8</b> Harmony</a></li>
<li class="chapter" data-level="13.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-9"><i class="fa fa-check"></i><b>13.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>14.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-10"><i class="fa fa-check"></i><b>14.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.3</b> Loading the data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.5</b> Linear regression</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>15.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-3"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-4"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data"><i class="fa fa-check"></i><b>18.3</b> Load the data</a></li>
<li class="chapter" data-level="18.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.5</b> Linear regression</a></li>
<li class="chapter" data-level="18.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>18.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="18.6.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.6.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>18.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.7.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.7.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.7.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-1"><i class="fa fa-check"></i><b>19.2</b> Load the data</a></li>
<li class="chapter" data-level="19.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>19.5</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.6</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>19.6.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="19.6.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-4"><i class="fa fa-check"></i><b>19.6.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="19.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="19.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>19.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>19.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-5"><i class="fa fa-check"></i><b>19.7.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="19.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>19.7.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>19.8</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="19.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-5"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>20.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="20.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>20.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="20.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>20.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="20.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>20.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="20.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-6"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="21.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-17"><i class="fa fa-check"></i><b>21.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="22.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-18"><i class="fa fa-check"></i><b>22.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-19"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>24</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="24.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>24.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>24.0.2</b> Data integration</a></li>
<li class="chapter" data-level="24.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>24.0.4</b> Clustering</a></li>
<li class="chapter" data-level="24.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>24.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>24.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>24.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="24.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>24.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="24.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>24.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="24.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>24.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="24.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>24.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="24.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>24.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>24.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-20"><i class="fa fa-check"></i><b>24.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>25</b> Doublet detection</a><ul>
<li class="chapter" data-level="25.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>25.1</b> Learning objectives</a></li>
<li class="chapter" data-level="25.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>25.2</b> Set up analysis</a></li>
<li class="chapter" data-level="25.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>25.4</b> Overview</a></li>
<li class="chapter" data-level="25.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>25.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="25.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>25.6</b> Further comments</a></li>
<li class="chapter" data-level="25.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-21"><i class="fa fa-check"></i><b>25.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>26</b> Annex</a></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-6"><i class="fa fa-check"></i><b>27.1</b> Abbreviations</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-8"><i class="fa fa-check"></i><b>27.2</b> Motivation</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-2"><i class="fa fa-check"></i><b>27.3</b> Load the data</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>27.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-6"><i class="fa fa-check"></i><b>27.5</b> Linear regression</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>27.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.6.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-6"><i class="fa fa-check"></i><b>27.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.6.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-6"><i class="fa fa-check"></i><b>27.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>27.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>27.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>27.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.7.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>27.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.7.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>27.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.7.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-6"><i class="fa fa-check"></i><b>27.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>27.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>27.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-22"><i class="fa fa-check"></i><b>27.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>28</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-9"><i class="fa fa-check"></i><b>28.1</b> Motivation</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-3"><i class="fa fa-check"></i><b>28.2</b> Load the data</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>28.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-7"><i class="fa fa-check"></i><b>28.4</b> Linear regression</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>28.5</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.5.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>28.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.5.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>28.5.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>28.6</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>28.6.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.6.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>28.6.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="28.6.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>28.6.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.6.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-7"><i class="fa fa-check"></i><b>28.6.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="28.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>28.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.6.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>28.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>28.7</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>28.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="28.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-8"><i class="fa fa-check"></i><b>28.7.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>28.7.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>28.8</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-23"><i class="fa fa-check"></i><b>28.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>29</b> Gene annotation</a><ul>
<li class="chapter" data-level="29.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-5"><i class="fa fa-check"></i><b>29.1</b> Introduction</a></li>
<li class="chapter" data-level="29.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>29.2</b> Sample sheet</a></li>
<li class="chapter" data-level="29.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>29.3</b> SRR9264343</a></li>
<li class="chapter" data-level="29.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-24"><i class="fa fa-check"></i><b>29.4</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="GeneAnnotTop" class="section level1">
<h1><span class="header-section-number">Chapter 29</span> Gene annotation</h1>
<div id="introduction-5" class="section level2">
<h2><span class="header-section-number">29.1</span> Introduction</h2>
<p>The 10X cellranger pipeline describes gene with Ensembl IDs and gene symbols.
We will need chromosomal location to identify mitochondrial genes to use in quality control.</p>
<p>The online resource used to annotate genes is not always available (see biomaRt package).
We will therefore annotate genes now in a separate chapter that can be run separately.
The annotation is written to a file that is used in following chapters.</p>
<p>As cellranger generates count matrices with a fixed gene list,
we will use the outcome for a single sample, eg SRR9264343.</p>
<div class="sourceCode" id="cb2364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2364-1"><a href="GeneAnnotTop.html#cb2364-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb2364-2"><a href="GeneAnnotTop.html#cb2364-2"></a>wrkDir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/Data/CaronBourque2020/grch38300&quot;</span>, projDir) </span>
<span id="cb2364-3"><a href="GeneAnnotTop.html#cb2364-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb2364-4"><a href="GeneAnnotTop.html#cb2364-4"></a>biomartBool &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># biomaRt sometimes fails, do it once, write to file and use that copy.</span></span></code></pre></div>
<div class="sourceCode" id="cb2365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2365-1"><a href="GeneAnnotTop.html#cb2365-1"></a><span class="kw">dir.create</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects&quot;</span>,</span>
<span id="cb2365-2"><a href="GeneAnnotTop.html#cb2365-2"></a>           projDir, outDirBit),</span>
<span id="cb2365-3"><a href="GeneAnnotTop.html#cb2365-3"></a>       <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div id="sample-sheet-3" class="section level2">
<h2><span class="header-section-number">29.2</span> Sample sheet</h2>
<p>We will load both the Caron and Hca data sets.</p>
<div class="sourceCode" id="cb2366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2366-1"><a href="GeneAnnotTop.html#cb2366-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb2366-2"><a href="GeneAnnotTop.html#cb2366-2"></a>cb_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/CaronBourque2020/SraRunTable.txt&quot;</span>)</span>
<span id="cb2366-3"><a href="GeneAnnotTop.html#cb2366-3"></a><span class="co"># Human Cell Atlas</span></span>
<span id="cb2366-4"><a href="GeneAnnotTop.html#cb2366-4"></a>hca_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/Hca/accList_Hca.txt&quot;</span>)</span>
<span id="cb2366-5"><a href="GeneAnnotTop.html#cb2366-5"></a></span>
<span id="cb2366-6"><a href="GeneAnnotTop.html#cb2366-6"></a><span class="co"># read sample sheet in:</span></span>
<span id="cb2366-7"><a href="GeneAnnotTop.html#cb2366-7"></a>splShtColToKeep &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Run&quot;</span>, <span class="st">&quot;Sample.Name&quot;</span>, <span class="st">&quot;source_name&quot;</span>)</span>
<span id="cb2366-8"><a href="GeneAnnotTop.html#cb2366-8"></a></span>
<span id="cb2366-9"><a href="GeneAnnotTop.html#cb2366-9"></a>cb_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(cb_sampleSheetFn, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb2366-10"><a href="GeneAnnotTop.html#cb2366-10"></a>hca_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(hca_sampleSheetFn, <span class="dt">header=</span>F, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb2366-11"><a href="GeneAnnotTop.html#cb2366-11"></a><span class="kw">colnames</span>(hca_sampleSheet) &lt;-<span class="st"> &quot;Sample.Name&quot;</span></span>
<span id="cb2366-12"><a href="GeneAnnotTop.html#cb2366-12"></a>hca_sampleSheet<span class="op">$</span>Run &lt;-<span class="st"> </span>hca_sampleSheet<span class="op">$</span>Sample.Name</span>
<span id="cb2366-13"><a href="GeneAnnotTop.html#cb2366-13"></a>hca_sampleSheet<span class="op">$</span>source_name &lt;-<span class="st"> &quot;ABMMC&quot;</span> <span class="co"># adult BMMC</span></span>
<span id="cb2366-14"><a href="GeneAnnotTop.html#cb2366-14"></a></span>
<span id="cb2366-15"><a href="GeneAnnotTop.html#cb2366-15"></a>sampleSheet &lt;-<span class="st"> </span><span class="kw">rbind</span>(cb_sampleSheet[,splShtColToKeep], hca_sampleSheet[,splShtColToKeep])</span>
<span id="cb2366-16"><a href="GeneAnnotTop.html#cb2366-16"></a></span>
<span id="cb2366-17"><a href="GeneAnnotTop.html#cb2366-17"></a>sampleSheet <span class="op">%&gt;%</span></span>
<span id="cb2366-18"><a href="GeneAnnotTop.html#cb2366-18"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2366-19"><a href="GeneAnnotTop.html#cb2366-19"></a><span class="st">    </span>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-f82431af846525a222fa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f82431af846525a222fa">{"x":{"filter":"none","data":[["SRR9264343","SRR9264344","SRR9264345","SRR9264346","SRR9264347","SRR9264348","SRR9264349","SRR9264350","SRR9264351","SRR9264352","SRR9264353","SRR9264354","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],["GSM3872434","GSM3872435","GSM3872436","GSM3872437","GSM3872438","GSM3872439","GSM3872440","GSM3872441","GSM3872442","GSM3872442","GSM3872443","GSM3872444","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],["ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","HHD","HHD","PRE-T","PRE-T","PBMMC","PBMMC","PBMMC","PBMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Run<\/th>\n      <th>Sample.Name<\/th>\n      <th>source_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="srr9264343" class="section level2">
<h2><span class="header-section-number">29.3</span> SRR9264343</h2>
<p>We will load the data for the first sample in the sample sheet: SRR9264343.</p>
<div class="sourceCode" id="cb2367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2367-1"><a href="GeneAnnotTop.html#cb2367-1"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb2367-2"><a href="GeneAnnotTop.html#cb2367-2"></a>sample.path &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/outs/raw_feature_bc_matrix/&quot;</span>,</span>
<span id="cb2367-3"><a href="GeneAnnotTop.html#cb2367-3"></a>                       wrkDir,</span>
<span id="cb2367-4"><a href="GeneAnnotTop.html#cb2367-4"></a>                       sampleSheet[i,<span class="st">&quot;Run&quot;</span>],</span>
<span id="cb2367-5"><a href="GeneAnnotTop.html#cb2367-5"></a>                       sampleSheet[i,<span class="st">&quot;Run&quot;</span>])</span>
<span id="cb2367-6"><a href="GeneAnnotTop.html#cb2367-6"></a>sce.raw &lt;-<span class="st"> </span><span class="kw">read10xCounts</span>(sample.path, <span class="dt">col.names=</span><span class="ot">TRUE</span>)</span>
<span id="cb2367-7"><a href="GeneAnnotTop.html#cb2367-7"></a>sce.raw</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 33538 737280 
## metadata(1): Samples
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(3): ID Symbol Type
## colnames(737280): AAACCTGAGAAACCAT-1 AAACCTGAGAAACCGC-1 ...
##   TTTGTCATCTTTAGTC-1 TTTGTCATCTTTCCTC-1
## colData names(2): Sample Barcode
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<!-- Gene annotation -->
<!-- Should be in separate script to run once only -->
<!-- SRR9264343 -->
<!-- raw or filtered: same genes -->
<div class="sourceCode" id="cb2369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2369-1"><a href="GeneAnnotTop.html#cb2369-1"></a><span class="co"># retrieve the feature information</span></span>
<span id="cb2369-2"><a href="GeneAnnotTop.html#cb2369-2"></a>gene.info &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce.raw)</span>
<span id="cb2369-3"><a href="GeneAnnotTop.html#cb2369-3"></a></span>
<span id="cb2369-4"><a href="GeneAnnotTop.html#cb2369-4"></a><span class="co"># setup the biomaRt connection to Ensembl using the correct species genome (hsapiens_gene_ensembl)</span></span>
<span id="cb2369-5"><a href="GeneAnnotTop.html#cb2369-5"></a>ensembl &lt;-<span class="st"> </span><span class="kw">useEnsembl</span>(<span class="dt">biomart=</span><span class="st">&#39;ensembl&#39;</span>,</span>
<span id="cb2369-6"><a href="GeneAnnotTop.html#cb2369-6"></a>                      <span class="dt">dataset=</span><span class="st">&#39;hsapiens_gene_ensembl&#39;</span>,</span>
<span id="cb2369-7"><a href="GeneAnnotTop.html#cb2369-7"></a>                      <span class="dt">mirror =</span> <span class="st">&quot;www&quot;</span>)</span>
<span id="cb2369-8"><a href="GeneAnnotTop.html#cb2369-8"></a></span>
<span id="cb2369-9"><a href="GeneAnnotTop.html#cb2369-9"></a><span class="co">#ensembl = useMart(biomart=&quot;ensembl&quot;,</span></span>
<span id="cb2369-10"><a href="GeneAnnotTop.html#cb2369-10"></a><span class="co">#         dataset=&#39;hsapiens_gene_ensembl&#39;,</span></span>
<span id="cb2369-11"><a href="GeneAnnotTop.html#cb2369-11"></a><span class="co">#         host = &quot;www.ensembl.org&quot;) # ensemblRedirect = FALSE</span></span>
<span id="cb2369-12"><a href="GeneAnnotTop.html#cb2369-12"></a></span>
<span id="cb2369-13"><a href="GeneAnnotTop.html#cb2369-13"></a><span class="co"># retrieve the attributes of interest from biomaRt using the Ensembl gene ID as the key</span></span>
<span id="cb2369-14"><a href="GeneAnnotTop.html#cb2369-14"></a><span class="co"># beware that this will only retrieve information for matching IDs</span></span>
<span id="cb2369-15"><a href="GeneAnnotTop.html#cb2369-15"></a>gene_symbol &lt;-<span class="st"> </span><span class="kw">getBM</span>(<span class="dt">attributes=</span><span class="kw">c</span>(<span class="st">&#39;ensembl_gene_id&#39;</span>,</span>
<span id="cb2369-16"><a href="GeneAnnotTop.html#cb2369-16"></a>                                  <span class="st">&#39;external_gene_name&#39;</span>,</span>
<span id="cb2369-17"><a href="GeneAnnotTop.html#cb2369-17"></a>                                  <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb2369-18"><a href="GeneAnnotTop.html#cb2369-18"></a>                                  <span class="st">&#39;start_position&#39;</span>,</span>
<span id="cb2369-19"><a href="GeneAnnotTop.html#cb2369-19"></a>                                  <span class="st">&#39;end_position&#39;</span>,</span>
<span id="cb2369-20"><a href="GeneAnnotTop.html#cb2369-20"></a>                                  <span class="st">&#39;strand&#39;</span>),</span>
<span id="cb2369-21"><a href="GeneAnnotTop.html#cb2369-21"></a>                     <span class="dt">filters=</span><span class="st">&#39;ensembl_gene_id&#39;</span>,</span>
<span id="cb2369-22"><a href="GeneAnnotTop.html#cb2369-22"></a>                     <span class="dt">mart=</span>ensembl,</span>
<span id="cb2369-23"><a href="GeneAnnotTop.html#cb2369-23"></a>                     <span class="dt">values=</span>gene.info[, <span class="dv">1</span>])</span>
<span id="cb2369-24"><a href="GeneAnnotTop.html#cb2369-24"></a></span>
<span id="cb2369-25"><a href="GeneAnnotTop.html#cb2369-25"></a><span class="co"># create a new data frame of the feature information</span></span>
<span id="cb2369-26"><a href="GeneAnnotTop.html#cb2369-26"></a>gene.merge &lt;-<span class="st"> </span><span class="kw">merge</span>(gene_symbol, gene.info, <span class="dt">by.x=</span><span class="kw">c</span>(<span class="st">&#39;ensembl_gene_id&#39;</span>), <span class="dt">by.y=</span><span class="kw">c</span>(<span class="st">&#39;ID&#39;</span>), <span class="dt">all.y=</span><span class="ot">TRUE</span>)</span>
<span id="cb2369-27"><a href="GeneAnnotTop.html#cb2369-27"></a><span class="kw">rownames</span>(gene.merge) &lt;-<span class="st"> </span>gene.merge<span class="op">$</span>ensembl_gene_id</span>
<span id="cb2369-28"><a href="GeneAnnotTop.html#cb2369-28"></a></span>
<span id="cb2369-29"><a href="GeneAnnotTop.html#cb2369-29"></a><span class="co"># set the order for the same as the original gene information</span></span>
<span id="cb2369-30"><a href="GeneAnnotTop.html#cb2369-30"></a>gene.merge &lt;-<span class="st"> </span>gene.merge[gene.info[, <span class="dv">1</span>], ]</span>
<span id="cb2369-31"><a href="GeneAnnotTop.html#cb2369-31"></a></span>
<span id="cb2369-32"><a href="GeneAnnotTop.html#cb2369-32"></a><span class="co"># reset the rowdata on the SCE object to contain all of this information</span></span>
<span id="cb2369-33"><a href="GeneAnnotTop.html#cb2369-33"></a><span class="kw">rowData</span>(sce.raw) &lt;-<span class="st"> </span>gene.merge</span></code></pre></div>
<p>Top 10 rows of the gene annotation table:</p>
<div class="sourceCode" id="cb2370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2370-1"><a href="GeneAnnotTop.html#cb2370-1"></a><span class="kw">rowData</span>(sce.raw)[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,] <span class="op">%&gt;%</span></span>
<span id="cb2370-2"><a href="GeneAnnotTop.html#cb2370-2"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2370-3"><a href="GeneAnnotTop.html#cb2370-3"></a><span class="st">    </span>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-f82431af846525a222fa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f82431af846525a222fa">{"x":{"filter":"none","data":[["ENSG00000243485","ENSG00000237613","ENSG00000186092","ENSG00000238009","ENSG00000239945","ENSG00000239906","ENSG00000241599","ENSG00000236601","ENSG00000284733","ENSG00000235146"],["MIR1302-2HG","FAM138A","OR4F5","","","","","","OR4F29",""],["1","1","1","1","1","1","1","1","1","1"],[29554,34554,65419,89295,89551,139790,160446,358857,450740,587629],[31109,36081,71585,133723,91105,140339,161525,366052,451678,594768],[1,-1,1,-1,-1,-1,1,1,-1,1],["MIR1302-2HG","FAM138A","OR4F5","AL627309.1","AL627309.3","AL627309.2","AL627309.4","AL732372.1","OR4F29","AC114498.1"],["Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ensembl_gene_id<\/th>\n      <th>external_gene_name<\/th>\n      <th>chromosome_name<\/th>\n      <th>start_position<\/th>\n      <th>end_position<\/th>\n      <th>strand<\/th>\n      <th>Symbol<\/th>\n      <th>Type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb2371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2371-1"><a href="GeneAnnotTop.html#cb2371-1"></a><span class="co"># slow</span></span>
<span id="cb2371-2"><a href="GeneAnnotTop.html#cb2371-2"></a><span class="co"># Write object to file</span></span>
<span id="cb2371-3"><a href="GeneAnnotTop.html#cb2371-3"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sceRaw_postBiomart.Rds&quot;</span>, projDir, outDirBit)</span>
<span id="cb2371-4"><a href="GeneAnnotTop.html#cb2371-4"></a><span class="kw">saveRDS</span>(sce.raw, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb2372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2372-1"><a href="GeneAnnotTop.html#cb2372-1"></a><span class="co"># Read object in:</span></span>
<span id="cb2372-2"><a href="GeneAnnotTop.html#cb2372-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sceRaw_postBiomart.Rds&quot;</span>, projDir, outDirBit)</span>
<span id="cb2372-3"><a href="GeneAnnotTop.html#cb2372-3"></a>sce.raw &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
</div>
<div id="session-information-24" class="section level2">
<h2><span class="header-section-number">29.4</span> Session information</h2>
<div class="sourceCode" id="cb2373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2373-1"><a href="GeneAnnotTop.html#cb2373-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] dplyr_1.0.6                 biomaRt_2.46.3             
##  [3] DropletUtils_1.10.3         SingleCellExperiment_1.12.0
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
##  [9] IRanges_2.24.1              S4Vectors_0.28.1           
## [11] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [13] matrixStats_0.58.0          knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7              bit64_4.0.5              
##  [3] progress_1.2.2            httr_1.4.2               
##  [5] tools_4.0.3               bslib_0.2.5              
##  [7] utf8_1.2.1                R6_2.5.0                 
##  [9] DT_0.18                   HDF5Array_1.18.1         
## [11] DBI_1.1.1                 rhdf5filters_1.2.0       
## [13] withr_2.4.2               tidyselect_1.1.1         
## [15] prettyunits_1.1.1         bit_4.0.4                
## [17] curl_4.3.1                compiler_4.0.3           
## [19] xml2_1.3.2                DelayedArray_0.16.3      
## [21] bookdown_0.22             sass_0.4.0               
## [23] askpass_1.1               rappdirs_0.3.3           
## [25] stringr_1.4.0             digest_0.6.27            
## [27] rmarkdown_2.8             R.utils_2.10.1           
## [29] XVector_0.30.0            pkgconfig_2.0.3          
## [31] htmltools_0.5.1.1         sparseMatrixStats_1.2.1  
## [33] dbplyr_2.1.1              fastmap_1.1.0            
## [35] limma_3.46.0              htmlwidgets_1.5.3        
## [37] rlang_0.4.11              RSQLite_2.2.7            
## [39] DelayedMatrixStats_1.12.3 jquerylib_0.1.4          
## [41] generics_0.1.0            jsonlite_1.7.2           
## [43] crosstalk_1.1.1           BiocParallel_1.24.1      
## [45] R.oo_1.24.0               RCurl_1.98-1.3           
## [47] magrittr_2.0.1            GenomeInfoDbData_1.2.4   
## [49] scuttle_1.0.4             Matrix_1.3-3             
## [51] Rcpp_1.0.6                Rhdf5lib_1.12.1          
## [53] fansi_0.4.2               lifecycle_1.0.0          
## [55] R.methodsS3_1.8.1         stringi_1.6.1            
## [57] yaml_2.2.1                edgeR_3.32.1             
## [59] zlibbioc_1.36.0           rhdf5_2.34.0             
## [61] BiocFileCache_1.14.0      grid_4.0.3               
## [63] blob_1.2.1                dqrng_0.3.0              
## [65] crayon_1.4.1              lattice_0.20-44          
## [67] beachmat_2.6.4            hms_1.0.0                
## [69] locfit_1.5-9.4            pillar_1.6.1             
## [71] codetools_0.2-18          XML_3.99-0.6             
## [73] glue_1.4.2                evaluate_0.14            
## [75] vctrs_0.3.8               openssl_1.4.4            
## [77] purrr_0.3.4               assertthat_0.2.1         
## [79] cachem_1.0.5              xfun_0.23                
## [81] tibble_3.1.2              AnnotationDbi_1.52.0     
## [83] memoise_2.0.0             ellipsis_0.3.2</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dsi-allCells-allSetsTop.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/geneAnnot.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC.pdf", "KmWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
