<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Quality Control - with 500 cells per sample | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Quality Control - with 500 cells per sample | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Quality Control - with 500 cells per sample | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="PreProcAllCellsTop.html"/>
<link rel="next" href="NormalisationAllCellsTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>12</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="12.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>12.1</b> Load data</a></li>
<li class="chapter" data-level="12.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>12.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="12.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>12.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="12.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-8"><i class="fa fa-check"></i><b>12.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>13</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="13.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>13.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="13.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>13.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="13.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>13.4</b> Batch correction</a><ul>
<li class="chapter" data-level="13.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>13.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>13.5</b> SCTransform</a><ul>
<li class="chapter" data-level="13.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>13.5.1</b> Batch only</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>13.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="13.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>13.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="13.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>13.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>13.7</b> fastMNN</a></li>
<li class="chapter" data-level="13.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>13.8</b> Harmony</a></li>
<li class="chapter" data-level="13.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-9"><i class="fa fa-check"></i><b>13.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>14.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-10"><i class="fa fa-check"></i><b>14.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.3</b> Loading the data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.5</b> Linear regression</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>15.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-3"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-4"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data"><i class="fa fa-check"></i><b>18.3</b> Load the data</a></li>
<li class="chapter" data-level="18.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.5</b> Linear regression</a></li>
<li class="chapter" data-level="18.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>18.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="18.6.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.6.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>18.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.7.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.7.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.7.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-1"><i class="fa fa-check"></i><b>19.2</b> Load the data</a></li>
<li class="chapter" data-level="19.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>19.5</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.6</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>19.6.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="19.6.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-4"><i class="fa fa-check"></i><b>19.6.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="19.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="19.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>19.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>19.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-5"><i class="fa fa-check"></i><b>19.7.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="19.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>19.7.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>19.8</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="19.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-5"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>20.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="20.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>20.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="20.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>20.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="20.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>20.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="20.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-6"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="21.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-17"><i class="fa fa-check"></i><b>21.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="22.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-18"><i class="fa fa-check"></i><b>22.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-19"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>24</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="24.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>24.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>24.0.2</b> Data integration</a></li>
<li class="chapter" data-level="24.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>24.0.4</b> Clustering</a></li>
<li class="chapter" data-level="24.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>24.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>24.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>24.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="24.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>24.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="24.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>24.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="24.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>24.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="24.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>24.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="24.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>24.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>24.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-20"><i class="fa fa-check"></i><b>24.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>25</b> Doublet detection</a><ul>
<li class="chapter" data-level="25.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>25.1</b> Learning objectives</a></li>
<li class="chapter" data-level="25.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>25.2</b> Set up analysis</a></li>
<li class="chapter" data-level="25.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>25.4</b> Overview</a></li>
<li class="chapter" data-level="25.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>25.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="25.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>25.6</b> Further comments</a></li>
<li class="chapter" data-level="25.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-21"><i class="fa fa-check"></i><b>25.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>26</b> Annex</a></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-6"><i class="fa fa-check"></i><b>27.1</b> Abbreviations</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-8"><i class="fa fa-check"></i><b>27.2</b> Motivation</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-2"><i class="fa fa-check"></i><b>27.3</b> Load the data</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>27.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-6"><i class="fa fa-check"></i><b>27.5</b> Linear regression</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>27.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.6.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-6"><i class="fa fa-check"></i><b>27.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.6.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-6"><i class="fa fa-check"></i><b>27.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>27.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>27.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>27.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.7.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>27.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.7.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>27.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.7.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-6"><i class="fa fa-check"></i><b>27.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>27.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>27.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-22"><i class="fa fa-check"></i><b>27.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>28</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-9"><i class="fa fa-check"></i><b>28.1</b> Motivation</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-3"><i class="fa fa-check"></i><b>28.2</b> Load the data</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>28.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-7"><i class="fa fa-check"></i><b>28.4</b> Linear regression</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>28.5</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.5.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>28.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.5.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>28.5.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>28.6</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>28.6.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.6.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>28.6.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="28.6.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>28.6.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.6.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-7"><i class="fa fa-check"></i><b>28.6.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="28.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>28.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.6.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>28.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>28.7</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>28.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="28.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-8"><i class="fa fa-check"></i><b>28.7.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>28.7.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>28.8</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-23"><i class="fa fa-check"></i><b>28.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>29</b> Gene annotation</a><ul>
<li class="chapter" data-level="29.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-5"><i class="fa fa-check"></i><b>29.1</b> Introduction</a></li>
<li class="chapter" data-level="29.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>29.2</b> Sample sheet</a></li>
<li class="chapter" data-level="29.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>29.3</b> SRR9264343</a></li>
<li class="chapter" data-level="29.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-24"><i class="fa fa-check"></i><b>29.4</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="PreProcTop" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Quality Control - with 500 cells per sample</h1>
<div id="introduction-3" class="section level2">
<h2><span class="header-section-number">5.1</span> Introduction</h2>
<!--
OSCA chapter 6 Quality Control
-->
<p>We will use two sets of Bone Marrow Mononuclear Cells (BMMC):</p>
<ul>
<li>‘CaronBourque2020’: pediatric samples</li>
<li>‘Hca’: HCA Census of Immune Cells for adult BMMCs</li>
</ul>
<p>Fastq files were retrieved from publicly available archive (SRA and HCA).</p>
<p>Sequencing quality was assessed and visualised using fastQC and MultiQC.</p>
<p>Reads were aligned against GRCh38 and features counted using cellranger (v3.1.0).</p>
<p>We will now check the quality of the data further:</p>
<ul>
<li>mapping quality and amplification rate</li>
<li>cell counts</li>
<li>distribution of keys quality metrics</li>
</ul>
<p>We will then:</p>
<ul>
<li>filter genes with very low expression</li>
<li>identify low-quality cells</li>
<li>filter and/or mark low quality cells</li>
</ul>
<!-- TODO:
- add references and/or links
- add code used to fetch the data or point to script or have chapter for retrieval+QC and for cellranger
-->
</div>
<div id="load-packages-1" class="section level2">
<h2><span class="header-section-number">5.2</span> Load packages</h2>
<ul>
<li>SingleCellExperiment - to store the data</li>
<li>Matrix - to deal with sparse and/or large matrices</li>
<li>DropletUtils - utilities for the analysis of droplet-based, inc. cell counting</li>
<li>scater - QC</li>
<li>scran - normalisation</li>
<li>igraph - graphs</li>
<li>biomaRt - for gene annotation</li>
<li>ggplot2 - for plotting</li>
<li>irlba - for faster PCA</li>
</ul>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="PreProcTop.html#cb262-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb262-2"><a href="PreProcTop.html#cb262-2"></a>wrkDir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/CaronBourque2020/grch38300&quot;</span>, projDir) </span>
<span id="cb262-3"><a href="PreProcTop.html#cb262-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit <span class="co"># &quot;AnaWiSeurat/Attempt1&quot;</span></span>
<span id="cb262-4"><a href="PreProcTop.html#cb262-4"></a>qcPlotDirBit &lt;-<span class="st"> &quot;Plots/Qc&quot;</span></span>
<span id="cb262-5"><a href="PreProcTop.html#cb262-5"></a>poolBool &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># FALSE # whether to read each sample in and pool them and write object to file, or just load that file.</span></span>
<span id="cb262-6"><a href="PreProcTop.html#cb262-6"></a>biomartBool &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># FALSE # biomaRt sometimes fails, do it once, write to file and use that copy. See geneAnnot.Rmd</span></span>
<span id="cb262-7"><a href="PreProcTop.html#cb262-7"></a>addQcBool &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb262-8"><a href="PreProcTop.html#cb262-8"></a>runAll &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb262-9"><a href="PreProcTop.html#cb262-9"></a>saveRds &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># overwrite existing Rds files</span></span>
<span id="cb262-10"><a href="PreProcTop.html#cb262-10"></a><span class="kw">dir.create</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s&quot;</span>, projDir, outDirBit, qcPlotDirBit),</span>
<span id="cb262-11"><a href="PreProcTop.html#cb262-11"></a>           <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>,</span>
<span id="cb262-12"><a href="PreProcTop.html#cb262-12"></a>           <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</span>
<span id="cb262-13"><a href="PreProcTop.html#cb262-13"></a><span class="kw">dir.create</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects&quot;</span>, projDir, outDirBit),</span>
<span id="cb262-14"><a href="PreProcTop.html#cb262-14"></a>       <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div id="sample-sheet-2" class="section level2">
<h2><span class="header-section-number">5.3</span> Sample sheet</h2>
<p>We will load both the Caron and Hca data sets.</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="PreProcTop.html#cb263-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb263-2"><a href="PreProcTop.html#cb263-2"></a>cb_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/CaronBourque2020/SraRunTable.txt&quot;</span>)</span>
<span id="cb263-3"><a href="PreProcTop.html#cb263-3"></a><span class="co"># Human Cell Atlas</span></span>
<span id="cb263-4"><a href="PreProcTop.html#cb263-4"></a>hca_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/Hca/accList_Hca.txt&quot;</span>)</span>
<span id="cb263-5"><a href="PreProcTop.html#cb263-5"></a></span>
<span id="cb263-6"><a href="PreProcTop.html#cb263-6"></a><span class="co"># read sample sheet in:</span></span>
<span id="cb263-7"><a href="PreProcTop.html#cb263-7"></a>splShtColToKeep &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Run&quot;</span>, <span class="st">&quot;Sample.Name&quot;</span>, <span class="st">&quot;source_name&quot;</span>)</span>
<span id="cb263-8"><a href="PreProcTop.html#cb263-8"></a></span>
<span id="cb263-9"><a href="PreProcTop.html#cb263-9"></a>cb_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(cb_sampleSheetFn, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb263-10"><a href="PreProcTop.html#cb263-10"></a>hca_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(hca_sampleSheetFn, <span class="dt">header=</span>F, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb263-11"><a href="PreProcTop.html#cb263-11"></a><span class="kw">colnames</span>(hca_sampleSheet) &lt;-<span class="st"> &quot;Sample.Name&quot;</span></span>
<span id="cb263-12"><a href="PreProcTop.html#cb263-12"></a>hca_sampleSheet<span class="op">$</span>Run &lt;-<span class="st"> </span>hca_sampleSheet<span class="op">$</span>Sample.Name</span>
<span id="cb263-13"><a href="PreProcTop.html#cb263-13"></a>hca_sampleSheet<span class="op">$</span>source_name &lt;-<span class="st"> &quot;ABMMC&quot;</span> <span class="co"># adult BMMC</span></span>
<span id="cb263-14"><a href="PreProcTop.html#cb263-14"></a></span>
<span id="cb263-15"><a href="PreProcTop.html#cb263-15"></a>sampleSheet &lt;-<span class="st"> </span><span class="kw">rbind</span>(cb_sampleSheet[,splShtColToKeep], hca_sampleSheet[,splShtColToKeep])</span>
<span id="cb263-16"><a href="PreProcTop.html#cb263-16"></a></span>
<span id="cb263-17"><a href="PreProcTop.html#cb263-17"></a>sampleSheet <span class="op">%&gt;%</span></span>
<span id="cb263-18"><a href="PreProcTop.html#cb263-18"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb263-19"><a href="PreProcTop.html#cb263-19"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-9eceb20dc2a931752ae4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9eceb20dc2a931752ae4">{"x":{"filter":"none","data":[["SRR9264343","SRR9264344","SRR9264345","SRR9264346","SRR9264347","SRR9264348","SRR9264349","SRR9264350","SRR9264351","SRR9264352","SRR9264353","SRR9264354","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],["GSM3872434","GSM3872435","GSM3872436","GSM3872437","GSM3872438","GSM3872439","GSM3872440","GSM3872441","GSM3872442","GSM3872442","GSM3872443","GSM3872444","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],["ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","HHD","HHD","PRE-T","PRE-T","PBMMC","PBMMC","PBMMC","PBMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC","ABMMC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Run<\/th>\n      <th>Sample.Name<\/th>\n      <th>source_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="data-representation-1" class="section level2">
<h2><span class="header-section-number">5.4</span> Data representation</h2>
<p>We will use a <a href="https://bioconductor.org/packages/SingleCellExperiment/">SingleCellExperiment</a> object that is described <a href="https://www.nature.com/articles/s41592-019-0654-x">here</a> and stores various data types:</p>
<ul>
<li>the count matrix</li>
<li>feature (gene) annotation</li>
<li>droplet annotation</li>
<li>outcome of downstream analysis such as dimensionality reduction</li>
</ul>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="PreProcTop.html#cb264-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/Images/tenxLibStructureV3.png&quot;</span>, <span class="st">&quot;..&quot;</span>)</span>
<span id="cb264-2"><a href="PreProcTop.html#cb264-2"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Images/tenxLibStructureV3.png" width="80%" /></p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="PreProcTop.html#cb265-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
</div>
<div id="example-1" class="section level2">
<h2><span class="header-section-number">5.5</span> Example</h2>
<p>We will load the data for the first sample in the sample sheet: SRR9264343.</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="PreProcTop.html#cb266-1"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb266-2"><a href="PreProcTop.html#cb266-2"></a>sample.path &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/outs/raw_feature_bc_matrix/&quot;</span>,</span>
<span id="cb266-3"><a href="PreProcTop.html#cb266-3"></a>                       wrkDir,</span>
<span id="cb266-4"><a href="PreProcTop.html#cb266-4"></a>                       sampleSheet[i,<span class="st">&quot;Run&quot;</span>],</span>
<span id="cb266-5"><a href="PreProcTop.html#cb266-5"></a>                       sampleSheet[i,<span class="st">&quot;Run&quot;</span>])</span>
<span id="cb266-6"><a href="PreProcTop.html#cb266-6"></a>sce.raw &lt;-<span class="st"> </span><span class="kw">read10xCounts</span>(sample.path, <span class="dt">col.names=</span><span class="ot">TRUE</span>)</span>
<span id="cb266-7"><a href="PreProcTop.html#cb266-7"></a>sce.raw</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 33538 737280 
## metadata(1): Samples
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(3): ID Symbol Type
## colnames(737280): AAACCTGAGAAACCAT-1 AAACCTGAGAAACCGC-1 ...
##   TTTGTCATCTTTAGTC-1 TTTGTCATCTTTCCTC-1
## colData names(2): Sample Barcode
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>We can access these different types of data with various functions.</p>
<p>Number of genes and droplets in the count matrix:</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="PreProcTop.html#cb268-1"></a><span class="kw">dim</span>(<span class="kw">counts</span>(sce.raw))</span></code></pre></div>
<pre><code>## [1]  33538 737280</code></pre>
<p>Features, with rowData():</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="PreProcTop.html#cb270-1"></a><span class="kw">head</span>(<span class="kw">rowData</span>(sce.raw))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 3 columns
##                              ID      Symbol            Type
##                     &lt;character&gt; &lt;character&gt;     &lt;character&gt;
## ENSG00000243485 ENSG00000243485 MIR1302-2HG Gene Expression
## ENSG00000237613 ENSG00000237613     FAM138A Gene Expression
## ENSG00000186092 ENSG00000186092       OR4F5 Gene Expression
## ENSG00000238009 ENSG00000238009  AL627309.1 Gene Expression
## ENSG00000239945 ENSG00000239945  AL627309.3 Gene Expression
## ENSG00000239906 ENSG00000239906  AL627309.2 Gene Expression</code></pre>
<p>Samples, with colData():</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="PreProcTop.html#cb272-1"></a><span class="kw">head</span>(<span class="kw">colData</span>(sce.raw))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 2 columns
##                                    Sample            Barcode
##                               &lt;character&gt;        &lt;character&gt;
## AAACCTGAGAAACCAT-1 /ssd/personal/baller.. AAACCTGAGAAACCAT-1
## AAACCTGAGAAACCGC-1 /ssd/personal/baller.. AAACCTGAGAAACCGC-1
## AAACCTGAGAAACCTA-1 /ssd/personal/baller.. AAACCTGAGAAACCTA-1
## AAACCTGAGAAACGAG-1 /ssd/personal/baller.. AAACCTGAGAAACGAG-1
## AAACCTGAGAAACGCC-1 /ssd/personal/baller.. AAACCTGAGAAACGCC-1
## AAACCTGAGAAAGTGG-1 /ssd/personal/baller.. AAACCTGAGAAAGTGG-1</code></pre>
<p>Single-cell RNA-seq data compared to bulk RNA-seq is sparse, especially with droplet-based methods such as 10X, mostly because:</p>
<ul>
<li>a given cell does not express each gene</li>
<li>the library preparation does not capture all transcript the cell does express</li>
<li>the sequencing depth per cell is far lower</li>
</ul>
<p>Counts, with counts(). Given the large number of droplets in a sample, count matrices can be large. They are however very sparse and can be stored in a ‘sparse matrix’ that only stores non-zero values, for example a ‘dgCMatrix’ object (‘DelayedArray’ class).</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="PreProcTop.html#cb274-1"></a><span class="kw">counts</span>(sce.raw) &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">counts</span>(sce.raw), <span class="st">&quot;dgCMatrix&quot;</span>)</span>
<span id="cb274-2"><a href="PreProcTop.html#cb274-2"></a><span class="kw">counts</span>(sce.raw)[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>## 10 x 10 sparse Matrix of class &quot;dgCMatrix&quot;
##                                    
## ENSG00000243485 . . . . . . . . . .
## ENSG00000237613 . . . . . . . . . .
## ENSG00000186092 . . . . . . . . . .
## ENSG00000238009 . . . . . . . . . .
## ENSG00000239945 . . . . . . . . . .
## ENSG00000239906 . . . . . . . . . .
## ENSG00000241599 . . . . . . . . . .
## ENSG00000236601 . . . . . . . . . .
## ENSG00000284733 . . . . . . . . . .
## ENSG00000235146 . . . . . . . . . .</code></pre>
</div>
<div id="mapping-qc-1" class="section level2">
<h2><span class="header-section-number">5.6</span> Mapping QC</h2>
<div id="gene-body-coverage-1" class="section level3">
<h3><span class="header-section-number">5.6.1</span> Gene body coverage</h3>
<p>The plot below show the average coverage (y-axis) along the body of genes (x-axis).</p>
<!-- In /ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaKmWiC/ScriptsKmWiC: -->
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="PreProcTop.html#cb276-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/Images/1_AAACCTGAGACTTTCG-1.rseqcGeneBodyCovCheck.txt.geneBodyCoverage.curves.png&quot;</span>, <span class="st">&quot;..&quot;</span>)</span>
<span id="cb276-2"><a href="PreProcTop.html#cb276-2"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Images/1_AAACCTGAGACTTTCG-1.rseqcGeneBodyCovCheck.txt.geneBodyCoverage.curves.png" width="80%" /></p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="PreProcTop.html#cb277-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
</div>
<div id="amplification-rate-1" class="section level3">
<h3><span class="header-section-number">5.6.2</span> Amplification rate</h3>
<p>We will use the information stored in the ‘molecule info’ file to count the number of UMI and reads for each gene in each cell.</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="PreProcTop.html#cb278-1"></a><span class="co">##mol.info.file &lt;- sprintf(&quot;%s/%s/%s/outs/molecule_info.h5&quot;, wrkDir, sampleSheet[i,&quot;Run&quot;], sampleSheet[i,&quot;Run&quot;])</span></span>
<span id="cb278-2"><a href="PreProcTop.html#cb278-2"></a><span class="co">##mol.info &lt;- read10xMolInfo(mol.info.file)</span></span>
<span id="cb278-3"><a href="PreProcTop.html#cb278-3"></a><span class="co"># or mol.info object if issue with H5Fopen</span></span>
<span id="cb278-4"><a href="PreProcTop.html#cb278-4"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb278-5"><a href="PreProcTop.html#cb278-5"></a>mol.info.file &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/outs/molecule_info_h5.Rds&quot;</span>,</span>
<span id="cb278-6"><a href="PreProcTop.html#cb278-6"></a>             wrkDir,</span>
<span id="cb278-7"><a href="PreProcTop.html#cb278-7"></a>             sampleSheet[i,<span class="st">&quot;Run&quot;</span>],</span>
<span id="cb278-8"><a href="PreProcTop.html#cb278-8"></a>             sampleSheet[i,<span class="st">&quot;Run&quot;</span>])</span>
<span id="cb278-9"><a href="PreProcTop.html#cb278-9"></a>mol.info &lt;-<span class="st"> </span><span class="kw">readRDS</span>(mol.info.file)</span>
<span id="cb278-10"><a href="PreProcTop.html#cb278-10"></a><span class="kw">rm</span>(mol.info.file)</span></code></pre></div>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="PreProcTop.html#cb279-1"></a><span class="co"># slow</span></span>
<span id="cb279-2"><a href="PreProcTop.html#cb279-2"></a></span>
<span id="cb279-3"><a href="PreProcTop.html#cb279-3"></a><span class="co"># &#39;data&#39; slot:</span></span>
<span id="cb279-4"><a href="PreProcTop.html#cb279-4"></a>mol.info<span class="op">$</span>data</span></code></pre></div>
<pre><code>## DataFrame with 18544666 rows and 5 columns
##                      cell       umi gem_group      gene     reads
##               &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 1        AAACCTGAGAAACCTA    467082         1      3287         1
## 2        AAACCTGAGAAACCTA    205888         1      3446         1
## 3        AAACCTGAGAAACCTA    866252         1      3896         3
## 4        AAACCTGAGAAACCTA    796027         1      3969         1
## 5        AAACCTGAGAAACCTA    542561         1      5008         1
## ...                   ...       ...       ...       ...       ...
## 18544662 TTTGTCATCTTTAGTC    927060         1     23634         1
## 18544663 TTTGTCATCTTTAGTC    975865         1     27143         1
## 18544664 TTTGTCATCTTTAGTC    364964         1     27467         4
## 18544665 TTTGTCATCTTTAGTC    152570         1     30125         7
## 18544666 TTTGTCATCTTTAGTC    383230         1     30283         5</code></pre>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="PreProcTop.html#cb281-1"></a><span class="co"># &#39;genes&#39; slot</span></span>
<span id="cb281-2"><a href="PreProcTop.html#cb281-2"></a><span class="kw">head</span>(mol.info<span class="op">$</span>genes)</span></code></pre></div>
<pre><code>## [1] &quot;ENSG00000243485&quot; &quot;ENSG00000237613&quot; &quot;ENSG00000186092&quot; &quot;ENSG00000238009&quot;
## [5] &quot;ENSG00000239945&quot; &quot;ENSG00000239906&quot;</code></pre>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="PreProcTop.html#cb283-1"></a><span class="co"># for each cell and gene, count UMIs</span></span>
<span id="cb283-2"><a href="PreProcTop.html#cb283-2"></a><span class="co"># slow, but needs running, at least once</span></span>
<span id="cb283-3"><a href="PreProcTop.html#cb283-3"></a><span class="co"># so write it to file to load later if need be.</span></span>
<span id="cb283-4"><a href="PreProcTop.html#cb283-4"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_preProc_ampDf1.Rds&quot;</span>, projDir, outDirBit)</span>
<span id="cb283-5"><a href="PreProcTop.html#cb283-5"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb283-6"><a href="PreProcTop.html#cb283-6"></a>{</span>
<span id="cb283-7"><a href="PreProcTop.html#cb283-7"></a>  ampDf &lt;-<span class="st"> </span>mol.info<span class="op">$</span>data <span class="op">%&gt;%</span></span>
<span id="cb283-8"><a href="PreProcTop.html#cb283-8"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb283-9"><a href="PreProcTop.html#cb283-9"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">umi =</span> <span class="kw">as.character</span>(umi)) <span class="op">%&gt;%</span></span>
<span id="cb283-10"><a href="PreProcTop.html#cb283-10"></a><span class="st">    </span><span class="kw">group_by</span>(cell, gene) <span class="op">%&gt;%</span></span>
<span id="cb283-11"><a href="PreProcTop.html#cb283-11"></a><span class="st">        </span><span class="kw">summarise</span>(<span class="dt">nUmis =</span> <span class="kw">n</span>(),</span>
<span id="cb283-12"><a href="PreProcTop.html#cb283-12"></a>                  <span class="dt">totReads=</span><span class="kw">sum</span>(reads)) <span class="op">%&gt;%</span></span>
<span id="cb283-13"><a href="PreProcTop.html#cb283-13"></a><span class="st">        </span><span class="kw">data.frame</span>()</span>
<span id="cb283-14"><a href="PreProcTop.html#cb283-14"></a></span>
<span id="cb283-15"><a href="PreProcTop.html#cb283-15"></a>    <span class="co"># Write object to file</span></span>
<span id="cb283-16"><a href="PreProcTop.html#cb283-16"></a>    <span class="kw">saveRDS</span>(ampDf, tmpFn)</span>
<span id="cb283-17"><a href="PreProcTop.html#cb283-17"></a>} <span class="cf">else</span> {</span>
<span id="cb283-18"><a href="PreProcTop.html#cb283-18"></a>    ampDf &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb283-19"><a href="PreProcTop.html#cb283-19"></a>}</span>
<span id="cb283-20"><a href="PreProcTop.html#cb283-20"></a><span class="kw">rm</span>(tmpFn)</span>
<span id="cb283-21"><a href="PreProcTop.html#cb283-21"></a></span>
<span id="cb283-22"><a href="PreProcTop.html#cb283-22"></a><span class="co"># distribution of totReads</span></span>
<span id="cb283-23"><a href="PreProcTop.html#cb283-23"></a><span class="kw">summary</span>(ampDf<span class="op">$</span>totReads)</span></code></pre></div>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
##     1.00     2.00     6.00    15.23    12.00 79275.00</code></pre>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="PreProcTop.html#cb285-1"></a><span class="co"># distribution of nUmis</span></span>
<span id="cb285-2"><a href="PreProcTop.html#cb285-2"></a><span class="kw">summary</span>(ampDf<span class="op">$</span>nUmis)</span></code></pre></div>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
##    1.000    1.000    1.000    2.377    1.000 7137.000</code></pre>
<p>We now plot the amplification rate.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="PreProcTop.html#cb287-1"></a>sp2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(ampDf, <span class="kw">aes</span>(<span class="dt">x=</span>nUmis, <span class="dt">y=</span>totReads)) <span class="op">+</span></span>
<span id="cb287-2"><a href="PreProcTop.html#cb287-2"></a><span class="st">  </span><span class="kw">geom_bin2d</span>(<span class="dt">bins =</span> <span class="dv">50</span>) <span class="op">+</span></span>
<span id="cb287-3"><a href="PreProcTop.html#cb287-3"></a><span class="st">  </span><span class="kw">scale_fill_continuous</span>(<span class="dt">type =</span> <span class="st">&quot;viridis&quot;</span>) <span class="op">+</span></span>
<span id="cb287-4"><a href="PreProcTop.html#cb287-4"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans=</span><span class="st">&#39;log10&#39;</span>) <span class="op">+</span></span>
<span id="cb287-5"><a href="PreProcTop.html#cb287-5"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans=</span><span class="st">&#39;log10&#39;</span>) <span class="op">+</span></span>
<span id="cb287-6"><a href="PreProcTop.html#cb287-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;totReads vs nUmis&quot;</span>) <span class="op">+</span></span>
<span id="cb287-7"><a href="PreProcTop.html#cb287-7"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span>
<span id="cb287-8"><a href="PreProcTop.html#cb287-8"></a></span>
<span id="cb287-9"><a href="PreProcTop.html#cb287-9"></a>sp2</span></code></pre></div>
<p><img src="preProc_files/figure-html/ampDf_bin2d-1.png" width="672" /></p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="PreProcTop.html#cb288-1"></a><span class="kw">gc</span>(<span class="dt">verbose=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>##             used  (Mb) gc trigger   (Mb)  max used   (Mb)
## Ncells   8621710 460.5   15837858  845.9  11041355  589.7
## Vcells 106269424 810.8  335474351 2559.5 335474351 2559.5</code></pre>
</div>
</div>
<div id="cell-calling-for-droplet-data-1" class="section level2">
<h2><span class="header-section-number">5.7</span> Cell calling for droplet data</h2>
<p>For a given sample, amongst the tens of thousands of droplets used in the assay, some will contain a cell while many others will not. The presence of RNA in a droplet will show with non-zero UMI count. This is however not sufficient to infer that the droplet does contain a cell. Indeed, after sample preparation, some cell debris including RNA will still float in the mix. This ambient RNA is unwillingly captured during library preparation and sequenced.</p>
<p>Cellranger generates a count matrix that includes all droplets analysed in the assay. We will now load this ‘raw matrix’ for one sample and draw the distribution of UMI counts.</p>
<p>Distribution of UMI counts:</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="PreProcTop.html#cb290-1"></a>libSizeDf &lt;-<span class="st"> </span>mol.info<span class="op">$</span>data <span class="op">%&gt;%</span></span>
<span id="cb290-2"><a href="PreProcTop.html#cb290-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb290-3"><a href="PreProcTop.html#cb290-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">umi =</span> <span class="kw">as.character</span>(umi)) <span class="op">%&gt;%</span></span>
<span id="cb290-4"><a href="PreProcTop.html#cb290-4"></a><span class="st">  </span><span class="kw">group_by</span>(cell) <span class="op">%&gt;%</span></span>
<span id="cb290-5"><a href="PreProcTop.html#cb290-5"></a><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">nUmis =</span> <span class="kw">n</span>(), <span class="dt">totReads=</span><span class="kw">sum</span>(reads)) <span class="op">%&gt;%</span></span>
<span id="cb290-6"><a href="PreProcTop.html#cb290-6"></a><span class="st">    </span><span class="kw">data.frame</span>()</span>
<span id="cb290-7"><a href="PreProcTop.html#cb290-7"></a><span class="kw">rm</span>(mol.info)</span></code></pre></div>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="PreProcTop.html#cb291-1"></a><span class="kw">ggplot</span>(libSizeDf, <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">log10</span>(nUmis))) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">50</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/nUmis_hist-1.png" width="672" /></p>
<p>Library size varies widely, both amongst empty droplets and droplets carrying cells, mostly due to:</p>
<ul>
<li>variation in droplet size,</li>
<li>amplification efficiency,</li>
<li>sequencing</li>
</ul>
<p>Most cell counting methods try to identify the library size that best distinguishes empty from cell-carrying droplets.</p>
<div id="mixture-model-1" class="section level3">
<h3><span class="header-section-number">5.7.1</span> Mixture model</h3>
<p>This method by default fits a mixture of two normal distributions to the logged library sizes:</p>
<ul>
<li>one with a small mean for empty droplets</li>
<li>the other with a higher mean for cell-carrying droplets</li>
</ul>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="PreProcTop.html#cb292-1"></a><span class="kw">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb292-2"><a href="PreProcTop.html#cb292-2"></a><span class="co"># get package</span></span>
<span id="cb292-3"><a href="PreProcTop.html#cb292-3"></a><span class="kw">library</span>(<span class="st">&quot;mixtools&quot;</span>)</span>
<span id="cb292-4"><a href="PreProcTop.html#cb292-4"></a><span class="co"># have library sizes on a log10 scale</span></span>
<span id="cb292-5"><a href="PreProcTop.html#cb292-5"></a>log10_lib_size &lt;-<span class="st"> </span><span class="kw">log10</span>(libSizeDf<span class="op">$</span>nUmis)</span>
<span id="cb292-6"><a href="PreProcTop.html#cb292-6"></a><span class="co"># fit mixture</span></span>
<span id="cb292-7"><a href="PreProcTop.html#cb292-7"></a>mix &lt;-<span class="st"> </span><span class="kw">normalmixEM</span>(log10_lib_size,</span>
<span id="cb292-8"><a href="PreProcTop.html#cb292-8"></a>           <span class="dt">mu=</span><span class="kw">c</span>(<span class="kw">log10</span>(<span class="dv">100</span>), <span class="kw">log10</span>(<span class="dv">1000</span>)),</span>
<span id="cb292-9"><a href="PreProcTop.html#cb292-9"></a>           <span class="dt">maxrestarts=</span><span class="dv">50</span>,</span>
<span id="cb292-10"><a href="PreProcTop.html#cb292-10"></a>           <span class="dt">epsilon =</span> <span class="fl">1e-03</span>)</span></code></pre></div>
<pre><code>## number of iterations= 29</code></pre>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="PreProcTop.html#cb294-1"></a><span class="co"># plot</span></span>
<span id="cb294-2"><a href="PreProcTop.html#cb294-2"></a><span class="kw">plot</span>(mix, <span class="dt">which=</span><span class="dv">2</span>, <span class="dt">xlab2=</span><span class="st">&quot;log(mol per cell)&quot;</span>)</span>
<span id="cb294-3"><a href="PreProcTop.html#cb294-3"></a><span class="co"># get density for each distribution:</span></span>
<span id="cb294-4"><a href="PreProcTop.html#cb294-4"></a>p1 &lt;-<span class="st"> </span><span class="kw">dnorm</span>(log10_lib_size, <span class="dt">mean=</span>mix<span class="op">$</span>mu[<span class="dv">1</span>], <span class="dt">sd=</span>mix<span class="op">$</span>sigma[<span class="dv">1</span>])</span>
<span id="cb294-5"><a href="PreProcTop.html#cb294-5"></a>p2 &lt;-<span class="st"> </span><span class="kw">dnorm</span>(log10_lib_size, <span class="dt">mean=</span>mix<span class="op">$</span>mu[<span class="dv">2</span>], <span class="dt">sd=</span>mix<span class="op">$</span>sigma[<span class="dv">2</span>])</span>
<span id="cb294-6"><a href="PreProcTop.html#cb294-6"></a><span class="co"># find intersection:</span></span>
<span id="cb294-7"><a href="PreProcTop.html#cb294-7"></a><span class="cf">if</span> (mix<span class="op">$</span>mu[<span class="dv">1</span>] <span class="op">&lt;</span><span class="st"> </span>mix<span class="op">$</span>mu[<span class="dv">2</span>]) {</span>
<span id="cb294-8"><a href="PreProcTop.html#cb294-8"></a>    split &lt;-<span class="st"> </span><span class="kw">min</span>(log10_lib_size[p2 <span class="op">&gt;</span><span class="st"> </span>p1])</span>
<span id="cb294-9"><a href="PreProcTop.html#cb294-9"></a>} <span class="cf">else</span> {</span>
<span id="cb294-10"><a href="PreProcTop.html#cb294-10"></a>    split &lt;-<span class="st"> </span><span class="kw">min</span>(log10_lib_size[p1 <span class="op">&gt;</span><span class="st"> </span>p2])</span>
<span id="cb294-11"><a href="PreProcTop.html#cb294-11"></a>}</span>
<span id="cb294-12"><a href="PreProcTop.html#cb294-12"></a><span class="co"># show split on plot:</span></span>
<span id="cb294-13"><a href="PreProcTop.html#cb294-13"></a><span class="kw">abline</span>(<span class="dt">v=</span>split, <span class="dt">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_mixtools-1.png" width="672" /></p>
</div>
<div id="barcode-rank-plot-1" class="section level3">
<h3><span class="header-section-number">5.7.2</span> Barcode rank plot</h3>
<p>The barcode rank plot shows the library sizes against their rank in decreasing order,
for the first 10000 droplets only.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="PreProcTop.html#cb295-1"></a>barcode_rank &lt;-<span class="st"> </span><span class="kw">rank</span>(<span class="op">-</span>libSizeDf<span class="op">$</span>nUmis)</span>
<span id="cb295-2"><a href="PreProcTop.html#cb295-2"></a><span class="kw">plot</span>(barcode_rank, libSizeDf<span class="op">$</span>nUmis, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>), <span class="dt">ylab=</span><span class="st">&quot;library size&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_barcodeRankPlot_initScale-1.png" width="672" /></p>
<p>Given the exponential shape of the curve above, library sizes can be shown on the log10 scale:</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="PreProcTop.html#cb296-1"></a><span class="kw">plot</span>(barcode_rank, log10_lib_size, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_barcodeRankPlot_logScale-1.png" width="672" /></p>
<p>The plot above shows that the majority of droplets have fewer than 100 UMIs, e.g. droplets with rank greater than 4000. We will redraw the plot to focus on droplets with lower ranks, by using the log10 scale for the x-axis.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="PreProcTop.html#cb297-1"></a><span class="kw">plot</span>(<span class="kw">log10</span>(barcode_rank),</span>
<span id="cb297-2"><a href="PreProcTop.html#cb297-2"></a>     log10_lib_size,</span>
<span id="cb297-3"><a href="PreProcTop.html#cb297-3"></a>     <span class="dt">xlim=</span><span class="kw">log10</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)))</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_barcodeRankPlot_logScale2-1.png" width="672" /></p>
<p>The point on the curve where it drops sharply may be used as the split point. Before that point library sizes are high, because droplets carry a cell. After that point, library sizes are far smaller because droplets do not carry a cell, only ambient RNA (… or do they?).</p>
<p>Here, we could ‘visually’ approximate the number of cells to 2500. There are however more robust and convenient methods.</p>
</div>
<div id="inflection-point-1" class="section level3">
<h3><span class="header-section-number">5.7.3</span> Inflection point</h3>
<p>We could also compute the inflection point of the curve.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="PreProcTop.html#cb298-1"></a>o &lt;-<span class="st"> </span><span class="kw">order</span>(barcode_rank)</span>
<span id="cb298-2"><a href="PreProcTop.html#cb298-2"></a>log10_lib_size &lt;-<span class="st"> </span>log10_lib_size[o]</span>
<span id="cb298-3"><a href="PreProcTop.html#cb298-3"></a>barcode_rank &lt;-<span class="st"> </span>barcode_rank[o]</span>
<span id="cb298-4"><a href="PreProcTop.html#cb298-4"></a></span>
<span id="cb298-5"><a href="PreProcTop.html#cb298-5"></a>rawdiff &lt;-<span class="st"> </span><span class="kw">diff</span>(log10_lib_size)<span class="op">/</span><span class="kw">diff</span>(barcode_rank)</span>
<span id="cb298-6"><a href="PreProcTop.html#cb298-6"></a>inflection &lt;-<span class="st"> </span><span class="kw">which</span>(rawdiff <span class="op">==</span><span class="st"> </span><span class="kw">min</span>(rawdiff[<span class="dv">100</span><span class="op">:</span><span class="kw">length</span>(rawdiff)], <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</span>
<span id="cb298-7"><a href="PreProcTop.html#cb298-7"></a></span>
<span id="cb298-8"><a href="PreProcTop.html#cb298-8"></a><span class="kw">plot</span>(<span class="dt">x=</span><span class="kw">log10</span>(barcode_rank),</span>
<span id="cb298-9"><a href="PreProcTop.html#cb298-9"></a>     <span class="dt">y=</span>log10_lib_size,</span>
<span id="cb298-10"><a href="PreProcTop.html#cb298-10"></a>     <span class="dt">xlim=</span><span class="kw">log10</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)))</span>
<span id="cb298-11"><a href="PreProcTop.html#cb298-11"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="kw">log10</span>(inflection), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_inflex-1.png" width="672" /></p>
<p>The inflection is at 3279 UMIs (3.5157414 on the log10 scale).
2306 droplets have at least these many UMIs and would thus contain one cell (or more).</p>
</div>
<div id="cellranger-v1-and-v2-1" class="section level3">
<h3><span class="header-section-number">5.7.4</span> Cellranger v1 and v2</h3>
<p>Given an expected number of cells, cellranger used to assume a ~10-fold range of library sizes for real cells and estimate this range (cellranger v1 and v2). The threshold was defined as the 99th quantile of the library size, divided by 10.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="PreProcTop.html#cb299-1"></a><span class="co"># approximate number of cells expected:</span></span>
<span id="cb299-2"><a href="PreProcTop.html#cb299-2"></a>n_cells &lt;-<span class="st"> </span><span class="dv">2500</span></span>
<span id="cb299-3"><a href="PreProcTop.html#cb299-3"></a><span class="co"># CellRanger</span></span>
<span id="cb299-4"><a href="PreProcTop.html#cb299-4"></a>totals &lt;-<span class="st"> </span><span class="kw">sort</span>(libSizeDf<span class="op">$</span>nUmis, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</span>
<span id="cb299-5"><a href="PreProcTop.html#cb299-5"></a><span class="co"># 99th percentile of top n_cells divided by 10</span></span>
<span id="cb299-6"><a href="PreProcTop.html#cb299-6"></a>thresh =<span class="st"> </span>totals[<span class="kw">round</span>(<span class="fl">0.01</span><span class="op">*</span>n_cells)]<span class="op">/</span><span class="dv">10</span></span>
<span id="cb299-7"><a href="PreProcTop.html#cb299-7"></a><span class="co">#plot(log10(totals), xlim=c(1,10000))</span></span>
<span id="cb299-8"><a href="PreProcTop.html#cb299-8"></a><span class="kw">plot</span>(<span class="dt">x=</span><span class="kw">log10</span>(<span class="kw">seq</span>(<span class="dv">1</span>,<span class="dv">10000</span>)),</span>
<span id="cb299-9"><a href="PreProcTop.html#cb299-9"></a>     <span class="dt">y=</span><span class="kw">log10</span>(totals)[<span class="dv">1</span><span class="op">:</span><span class="dv">10000</span>]</span>
<span id="cb299-10"><a href="PreProcTop.html#cb299-10"></a>     )</span>
<span id="cb299-11"><a href="PreProcTop.html#cb299-11"></a><span class="kw">abline</span>(<span class="dt">h=</span><span class="kw">log10</span>(thresh), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_cellrangerV1n2-1.png" width="672" /></p>
<p>The threshold is at 1452 UMIs and 2773 cells are detected.</p>
</div>
<div id="dropletutils-and-emptydrops-1" class="section level3">
<h3><span class="header-section-number">5.7.5</span> DropletUtils and EmptyDrops</h3>
<p>The <code>DropletUtils</code> package offers utilities to analyse droplet-based data, including cell counting using the library size as seen above. These simple approaches may exclude droplets with small or quiet cells with low RNA content. The <code>emptyDrops</code> method calls cells by first computing the expression profile for droplets with RNA content so low they almost certainly do not contain any cell: the ‘background’ or ‘ambient’ profile. The method then tests each non-background droplet for significant difference in expression profile.</p>
<p>Let’s first check the knee and inflection methods.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="PreProcTop.html#cb300-1"></a>br.out &lt;-<span class="st"> </span><span class="kw">barcodeRanks</span>(<span class="kw">counts</span>(sce.raw))</span>
<span id="cb300-2"><a href="PreProcTop.html#cb300-2"></a><span class="kw">plot</span>(br.out<span class="op">$</span>rank, br.out<span class="op">$</span>total, <span class="dt">log=</span><span class="st">&quot;xy&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Rank&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Total UMI count&quot;</span>)</span>
<span id="cb300-3"><a href="PreProcTop.html#cb300-3"></a>o &lt;-<span class="st"> </span><span class="kw">order</span>(br.out<span class="op">$</span>rank)</span>
<span id="cb300-4"><a href="PreProcTop.html#cb300-4"></a><span class="kw">lines</span>(br.out<span class="op">$</span>rank[o], br.out<span class="op">$</span>fitted[o], <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb300-5"><a href="PreProcTop.html#cb300-5"></a></span>
<span id="cb300-6"><a href="PreProcTop.html#cb300-6"></a><span class="kw">abline</span>(<span class="dt">h=</span><span class="kw">metadata</span>(br.out)<span class="op">$</span>knee, <span class="dt">col=</span><span class="st">&quot;dodgerblue&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>)</span>
<span id="cb300-7"><a href="PreProcTop.html#cb300-7"></a><span class="kw">abline</span>(<span class="dt">h=</span><span class="kw">metadata</span>(br.out)<span class="op">$</span>inflection, <span class="dt">col=</span><span class="st">&quot;forestgreen&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>)</span>
<span id="cb300-8"><a href="PreProcTop.html#cb300-8"></a><span class="kw">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>), </span>
<span id="cb300-9"><a href="PreProcTop.html#cb300-9"></a>    <span class="dt">legend=</span><span class="kw">c</span>(<span class="st">&quot;knee&quot;</span>, <span class="st">&quot;inflection&quot;</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_dropletUtils-1.png" width="672" /></p>
<p>Testing for empty droplets.</p>
<p>We will call cells with a false discovery rate (FDR) of 0.1% so that at most 1 in 1000 droplets called may be empty.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="PreProcTop.html#cb301-1"></a><span class="co"># a bit slow</span></span>
<span id="cb301-2"><a href="PreProcTop.html#cb301-2"></a><span class="co"># significance is computed by simulation so we set a seed for reproducibility</span></span>
<span id="cb301-3"><a href="PreProcTop.html#cb301-3"></a><span class="kw">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb301-4"><a href="PreProcTop.html#cb301-4"></a><span class="co"># run analysis:</span></span>
<span id="cb301-5"><a href="PreProcTop.html#cb301-5"></a>e.out &lt;-<span class="st"> </span><span class="kw">emptyDrops</span>(<span class="kw">counts</span>(sce.raw))</span>
<span id="cb301-6"><a href="PreProcTop.html#cb301-6"></a>e.out</span></code></pre></div>
<pre><code>## DataFrame with 737280 rows and 5 columns
##                        Total   LogProb    PValue   Limited       FDR
##                    &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;logical&gt; &lt;numeric&gt;
## AAACCTGAGAAACCAT-1         0        NA        NA        NA        NA
## AAACCTGAGAAACCGC-1         0        NA        NA        NA        NA
## AAACCTGAGAAACCTA-1        31        NA        NA        NA        NA
## AAACCTGAGAAACGAG-1         0        NA        NA        NA        NA
## AAACCTGAGAAACGCC-1         0        NA        NA        NA        NA
## ...                      ...       ...       ...       ...       ...
## TTTGTCATCTTTACAC-1         0        NA        NA        NA        NA
## TTTGTCATCTTTACGT-1         1        NA        NA        NA        NA
## TTTGTCATCTTTAGGG-1         0        NA        NA        NA        NA
## TTTGTCATCTTTAGTC-1        26        NA        NA        NA        NA
## TTTGTCATCTTTCCTC-1         0        NA        NA        NA        NA</code></pre>
<p>NAs are assigned to droplets used to compute the ambient profile.</p>
<p>Get numbers of droplets in each class defined by FDR and the cut-off used:</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="PreProcTop.html#cb303-1"></a><span class="kw">summary</span>(e.out<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>)</span></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE    NA&#39;s 
## logical     487    3075  733718</code></pre>
<p>The test significance is computed by permutation. For each droplet tested, the number of permutations may limit the value of the p-value. This information is available in the ‘Limited’ column. If ‘Limited’ is ‘TRUE’ for any non-significant droplet, the number of permutations was too low, should be increased and the analysis re-run.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="PreProcTop.html#cb305-1"></a><span class="kw">table</span>(<span class="dt">Sig=</span>e.out<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>, <span class="dt">Limited=</span>e.out<span class="op">$</span>Limited)</span></code></pre></div>
<pre><code>##        Limited
## Sig     FALSE TRUE
##   FALSE   487    0
##   TRUE     76 2999</code></pre>
<p>Let’s check that the background comprises only empty droplets. If the droplets used to define the background profile are indeed empty, testing them should result in a flat distribution of p-values. Let’s test the ‘ambient’ droplets and draw the p-value distribution.</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="PreProcTop.html#cb307-1"></a><span class="kw">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb307-2"><a href="PreProcTop.html#cb307-2"></a>limit &lt;-<span class="st"> </span><span class="dv">100</span>   </span>
<span id="cb307-3"><a href="PreProcTop.html#cb307-3"></a>all.out &lt;-<span class="st"> </span><span class="kw">emptyDrops</span>(<span class="kw">counts</span>(sce.raw), <span class="dt">lower=</span>limit, <span class="dt">test.ambient=</span><span class="ot">TRUE</span>)</span>
<span id="cb307-4"><a href="PreProcTop.html#cb307-4"></a><span class="kw">hist</span>(all.out<span class="op">$</span>PValue[all.out<span class="op">$</span>Total <span class="op">&lt;=</span><span class="st"> </span>limit <span class="op">&amp;</span><span class="st"> </span>all.out<span class="op">$</span>Total <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>],</span>
<span id="cb307-5"><a href="PreProcTop.html#cb307-5"></a>    <span class="dt">xlab=</span><span class="st">&quot;P-value&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>) </span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_emptyDrops_sigEmpty-1.png" width="672" /></p>
<p>The distribution of p-values looks uniform with no large peak for small values: no cell in these droplets.</p>
<p>To evaluate the outcome of the analysis, we will plot the strength of the evidence against library size.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="PreProcTop.html#cb308-1"></a>is.cell &lt;-<span class="st"> </span>e.out<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span></span></code></pre></div>
<p>Number of cells detected: 3075.</p>
<p>The plot plot shows the strength of the evidence against the library size.
Each point is a droplet coloured:</p>
<ul>
<li>in black if without cell,</li>
<li>in red if with a cell (or more)</li>
<li>in green if with a cell (or more) as defined with <code>emptyDrops</code> but not the inflection method.</li>
</ul>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="PreProcTop.html#cb309-1"></a><span class="co"># colour:</span></span>
<span id="cb309-2"><a href="PreProcTop.html#cb309-2"></a>cellColour &lt;-<span class="st"> </span><span class="kw">ifelse</span>(is.cell, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;black&quot;</span>) <span class="co"># rep(&quot;black&quot;, nrow((e.out)))</span></span>
<span id="cb309-3"><a href="PreProcTop.html#cb309-3"></a><span class="co"># boolean for presence of cells as defined by the inflection method</span></span>
<span id="cb309-4"><a href="PreProcTop.html#cb309-4"></a>tmpBoolInflex &lt;-<span class="st"> </span>e.out<span class="op">$</span>Total <span class="op">&gt;</span><span class="st"> </span><span class="kw">metadata</span>(br.out)<span class="op">$</span>inflection</span>
<span id="cb309-5"><a href="PreProcTop.html#cb309-5"></a><span class="co"># boolean for presence of cells as defined by the emptyDrops method</span></span>
<span id="cb309-6"><a href="PreProcTop.html#cb309-6"></a>tmpBoolSmall &lt;-<span class="st"> </span>e.out<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span></span>
<span id="cb309-7"><a href="PreProcTop.html#cb309-7"></a>tmpBoolRecov &lt;-<span class="st"> </span><span class="op">!</span>tmpBoolInflex <span class="op">&amp;</span><span class="st"> </span>tmpBoolSmall</span>
<span id="cb309-8"><a href="PreProcTop.html#cb309-8"></a>cellColour[tmpBoolRecov] &lt;-<span class="st"> &quot;green&quot;</span> <span class="co"># &#39;recovered&#39; cells</span></span>
<span id="cb309-9"><a href="PreProcTop.html#cb309-9"></a></span>
<span id="cb309-10"><a href="PreProcTop.html#cb309-10"></a><span class="co"># plot strength of significance vs library size</span></span>
<span id="cb309-11"><a href="PreProcTop.html#cb309-11"></a><span class="kw">plot</span>(<span class="kw">log10</span>(e.out<span class="op">$</span>Total),</span>
<span id="cb309-12"><a href="PreProcTop.html#cb309-12"></a>     <span class="op">-</span>e.out<span class="op">$</span>LogProb,</span>
<span id="cb309-13"><a href="PreProcTop.html#cb309-13"></a>     <span class="dt">col=</span>cellColour,</span>
<span id="cb309-14"><a href="PreProcTop.html#cb309-14"></a>     <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="kw">max</span>(<span class="kw">log10</span>(e.out<span class="op">$</span>Total))),</span>
<span id="cb309-15"><a href="PreProcTop.html#cb309-15"></a>     <span class="dt">xlab=</span><span class="st">&quot;Total UMI count&quot;</span>,</span>
<span id="cb309-16"><a href="PreProcTop.html#cb309-16"></a>     <span class="dt">ylab=</span><span class="st">&quot;-Log Probability&quot;</span>)</span>
<span id="cb309-17"><a href="PreProcTop.html#cb309-17"></a></span>
<span id="cb309-18"><a href="PreProcTop.html#cb309-18"></a><span class="co"># add point to show &#39;recovered&#39; cell on top</span></span>
<span id="cb309-19"><a href="PreProcTop.html#cb309-19"></a><span class="kw">points</span>(<span class="kw">log10</span>(e.out<span class="op">$</span>Total)[tmpBoolRecov],</span>
<span id="cb309-20"><a href="PreProcTop.html#cb309-20"></a>       <span class="op">-</span>e.out<span class="op">$</span>LogProb[tmpBoolRecov],</span>
<span id="cb309-21"><a href="PreProcTop.html#cb309-21"></a>       <span class="dt">pch=</span><span class="dv">16</span>,</span>
<span id="cb309-22"><a href="PreProcTop.html#cb309-22"></a>       <span class="dt">col=</span><span class="st">&quot;green&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/cellCall_emptyDrops_diagPlot-1.png" width="672" /></p>
<p>Let’s filter out empty droplets.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="PreProcTop.html#cb310-1"></a>sce.ed &lt;-<span class="st"> </span>sce.raw[,<span class="kw">which</span>(e.out<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>)] <span class="co"># &#39;ed&#39; for empty droplet</span></span>
<span id="cb310-2"><a href="PreProcTop.html#cb310-2"></a><span class="kw">rm</span>(sce.raw);</span>
<span id="cb310-3"><a href="PreProcTop.html#cb310-3"></a><span class="kw">gc</span>();</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger   (Mb)  max used   (Mb)
## Ncells  8977270 479.5   15837858  845.9  15837858  845.9
## Vcells 58782587 448.5  268379481 2047.6 335474351 2559.5</code></pre>
<p>And check the new SCE object:</p>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="PreProcTop.html#cb312-1"></a>sce.ed</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 33538 3075 
## metadata(1): Samples
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(3): ID Symbol Type
## colnames(3075): AAACCTGAGACTTTCG-1 AAACCTGGTCTTCAAG-1 ...
##   TTTGTCACAGGCTCAC-1 TTTGTCAGTTCGGCAC-1
## colData names(2): Sample Barcode
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="PreProcTop.html#cb314-1"></a><span class="kw">rm</span>(sce.ed)</span></code></pre></div>
<p>Cell calling in cellranger v3 uses a method similar to emptyDrops()
and a ‘filtered matrix’ is generated that only keeps droplets deemed to contain
a cell (or more). We will load these filtered matrices now.</p>
</div>
</div>
<div id="load-filtered-matrices-1" class="section level2">
<h2><span class="header-section-number">5.8</span> Load filtered matrices</h2>
<p>Each sample was analysed with cellranger separately.
We load filtered matrices one sample at a time,
showing for each the name and number of features and cells.</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="PreProcTop.html#cb315-1"></a><span class="co"># load data:</span></span>
<span id="cb315-2"><a href="PreProcTop.html#cb315-2"></a><span class="co"># a bit slow</span></span>
<span id="cb315-3"><a href="PreProcTop.html#cb315-3"></a><span class="co"># use &#39;cache.lazy = FALSE&#39; to avoid &#39;long vectors not supported yet&#39; when &#39;cache=TRUE&#39;</span></span>
<span id="cb315-4"><a href="PreProcTop.html#cb315-4"></a></span>
<span id="cb315-5"><a href="PreProcTop.html#cb315-5"></a>sce.list &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">nrow</span>(sampleSheet))</span>
<span id="cb315-6"><a href="PreProcTop.html#cb315-6"></a></span>
<span id="cb315-7"><a href="PreProcTop.html#cb315-7"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(sampleSheet))</span>
<span id="cb315-8"><a href="PreProcTop.html#cb315-8"></a>{</span>
<span id="cb315-9"><a href="PreProcTop.html#cb315-9"></a>    <span class="kw">print</span>(<span class="kw">sprintf</span>(<span class="st">&quot;&#39;Run&#39; %s, &#39;Sample.Name&#39; %s&quot;</span>, sampleSheet[i,<span class="st">&quot;Run&quot;</span>], sampleSheet[i,<span class="st">&quot;Sample.Name&quot;</span>]))</span>
<span id="cb315-10"><a href="PreProcTop.html#cb315-10"></a>    sample.path &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/outs/filtered_feature_bc_matrix/&quot;</span>,</span>
<span id="cb315-11"><a href="PreProcTop.html#cb315-11"></a>                  <span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/grch38300&quot;</span>,</span>
<span id="cb315-12"><a href="PreProcTop.html#cb315-12"></a>                      projDir,</span>
<span id="cb315-13"><a href="PreProcTop.html#cb315-13"></a>                      <span class="kw">ifelse</span>(sampleSheet[i,<span class="st">&quot;source_name&quot;</span>] <span class="op">==</span><span class="st"> &quot;ABMMC&quot;</span>,</span>
<span id="cb315-14"><a href="PreProcTop.html#cb315-14"></a>                         <span class="st">&quot;Hca&quot;</span>,</span>
<span id="cb315-15"><a href="PreProcTop.html#cb315-15"></a>                         <span class="st">&quot;CaronBourque2020&quot;</span>)),</span>
<span id="cb315-16"><a href="PreProcTop.html#cb315-16"></a>                  sampleSheet[i,<span class="st">&quot;Run&quot;</span>],</span>
<span id="cb315-17"><a href="PreProcTop.html#cb315-17"></a>                  sampleSheet[i,<span class="st">&quot;Run&quot;</span>])</span>
<span id="cb315-18"><a href="PreProcTop.html#cb315-18"></a>    sce.list[[i]] &lt;-<span class="st"> </span><span class="kw">read10xCounts</span>(sample.path)</span>
<span id="cb315-19"><a href="PreProcTop.html#cb315-19"></a>    <span class="kw">print</span>(<span class="kw">dim</span>(sce.list[[i]]))</span>
<span id="cb315-20"><a href="PreProcTop.html#cb315-20"></a>}</span></code></pre></div>
<pre><code>## [1] &quot;&#39;Run&#39; SRR9264343, &#39;Sample.Name&#39; GSM3872434&quot;
## [1] 33538  3088
## [1] &quot;&#39;Run&#39; SRR9264344, &#39;Sample.Name&#39; GSM3872435&quot;
## [1] 33538  6678
## [1] &quot;&#39;Run&#39; SRR9264345, &#39;Sample.Name&#39; GSM3872436&quot;
## [1] 33538  5054
## [1] &quot;&#39;Run&#39; SRR9264346, &#39;Sample.Name&#39; GSM3872437&quot;
## [1] 33538  6096
## [1] &quot;&#39;Run&#39; SRR9264347, &#39;Sample.Name&#39; GSM3872438&quot;
## [1] 33538  5442
## [1] &quot;&#39;Run&#39; SRR9264348, &#39;Sample.Name&#39; GSM3872439&quot;
## [1] 33538  5502
## [1] &quot;&#39;Run&#39; SRR9264349, &#39;Sample.Name&#39; GSM3872440&quot;
## [1] 33538  4126
## [1] &quot;&#39;Run&#39; SRR9264350, &#39;Sample.Name&#39; GSM3872441&quot;
## [1] 33538  3741
## [1] &quot;&#39;Run&#39; SRR9264351, &#39;Sample.Name&#39; GSM3872442&quot;
## [1] 33538   978
## [1] &quot;&#39;Run&#39; SRR9264352, &#39;Sample.Name&#39; GSM3872442&quot;
## [1] 33538  1150
## [1] &quot;&#39;Run&#39; SRR9264353, &#39;Sample.Name&#39; GSM3872443&quot;
## [1] 33538  4964
## [1] &quot;&#39;Run&#39; SRR9264354, &#39;Sample.Name&#39; GSM3872444&quot;
## [1] 33538  4255
## [1] &quot;&#39;Run&#39; MantonBM1, &#39;Sample.Name&#39; MantonBM1&quot;
## [1] 33538 23283
## [1] &quot;&#39;Run&#39; MantonBM2, &#39;Sample.Name&#39; MantonBM2&quot;
## [1] 33538 25055
## [1] &quot;&#39;Run&#39; MantonBM3, &#39;Sample.Name&#39; MantonBM3&quot;
## [1] 33538 24548
## [1] &quot;&#39;Run&#39; MantonBM4, &#39;Sample.Name&#39; MantonBM4&quot;
## [1] 33538 26478
## [1] &quot;&#39;Run&#39; MantonBM5, &#39;Sample.Name&#39; MantonBM5&quot;
## [1] 33538 26383
## [1] &quot;&#39;Run&#39; MantonBM6, &#39;Sample.Name&#39; MantonBM6&quot;
## [1] 33538 22801
## [1] &quot;&#39;Run&#39; MantonBM7, &#39;Sample.Name&#39; MantonBM7&quot;
## [1] 33538 24372
## [1] &quot;&#39;Run&#39; MantonBM8, &#39;Sample.Name&#39; MantonBM8&quot;
## [1] 33538 24860</code></pre>
<p>Let’s combine all 20 samples into a single object.</p>
<p>We first check the feature lists are identical.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="PreProcTop.html#cb317-1"></a><span class="co"># check row names are the same</span></span>
<span id="cb317-2"><a href="PreProcTop.html#cb317-2"></a><span class="co"># compare to that for the first sample</span></span>
<span id="cb317-3"><a href="PreProcTop.html#cb317-3"></a></span>
<span id="cb317-4"><a href="PreProcTop.html#cb317-4"></a>rowNames1 &lt;-<span class="st"> </span><span class="kw">rownames</span>(sce.list[[<span class="dv">1</span>]])</span>
<span id="cb317-5"><a href="PreProcTop.html#cb317-5"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(sampleSheet))</span>
<span id="cb317-6"><a href="PreProcTop.html#cb317-6"></a>{</span>
<span id="cb317-7"><a href="PreProcTop.html#cb317-7"></a>    <span class="kw">print</span>(<span class="kw">identical</span>(rowNames1, <span class="kw">rownames</span>(sce.list[[i]])))</span>
<span id="cb317-8"><a href="PreProcTop.html#cb317-8"></a>}</span></code></pre></div>
<pre><code>## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE
## [1] TRUE</code></pre>
<p>A cell barcode comprises the actual sequence and a ‘group ID’, e.g. AAACCTGAGAAACCAT-1.
The latter helps distinguish cells with identical barcode sequence but come from
different samples. As each sample was analysed separately, the group ID is set
to 1 in all data sets. To pool these data sets we first need to change group IDs
so cell barcodes are unique across all samples. We will use the position of the
sample in the sample sheet. We also downsample to 2000 cells maximum per sample
for faster computation during the course.</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="PreProcTop.html#cb319-1"></a><span class="co"># subset 2000 cells:</span></span>
<span id="cb319-2"><a href="PreProcTop.html#cb319-2"></a><span class="co"># mind that is not mentioned in Rds file names ...</span></span>
<span id="cb319-3"><a href="PreProcTop.html#cb319-3"></a><span class="cf">if</span>(<span class="kw">ncol</span>(sce.list[[<span class="dv">1</span>]]) <span class="op">&lt;</span><span class="st"> </span><span class="dv">2000</span>)</span>
<span id="cb319-4"><a href="PreProcTop.html#cb319-4"></a>{</span>
<span id="cb319-5"><a href="PreProcTop.html#cb319-5"></a>  sce &lt;-<span class="st"> </span>sce.list[[<span class="dv">1</span>]]</span>
<span id="cb319-6"><a href="PreProcTop.html#cb319-6"></a>} <span class="cf">else</span> {</span>
<span id="cb319-7"><a href="PreProcTop.html#cb319-7"></a>  sce &lt;-<span class="st"> </span>sce.list[[<span class="dv">1</span>]][, <span class="kw">sample</span>(<span class="kw">ncol</span>(sce.list[[<span class="dv">1</span>]]), <span class="dv">2000</span>)]</span>
<span id="cb319-8"><a href="PreProcTop.html#cb319-8"></a>}</span>
<span id="cb319-9"><a href="PreProcTop.html#cb319-9"></a></span>
<span id="cb319-10"><a href="PreProcTop.html#cb319-10"></a><span class="kw">colData</span>(sce)<span class="op">$</span>Barcode &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;([0-9])$&quot;</span>, <span class="dv">1</span>, <span class="kw">colData</span>(sce)<span class="op">$</span>Barcode)</span>
<span id="cb319-11"><a href="PreProcTop.html#cb319-11"></a><span class="kw">print</span>(<span class="kw">head</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode))</span></code></pre></div>
<pre><code>## [1] &quot;GATGCTAGTATCAGTC-1&quot; &quot;GTCCTCAGTCCAAGTT-1&quot; &quot;CGCGTTTAGGCTAGCA-1&quot;
## [4] &quot;GGATTACGTGCGAAAC-1&quot; &quot;CGTTGGGCAGCCAATT-1&quot; &quot;GTGCGGTGTCATTAGC-1&quot;</code></pre>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="PreProcTop.html#cb321-1"></a><span class="kw">print</span>(<span class="kw">tail</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode))</span></code></pre></div>
<pre><code>## [1] &quot;CGTCCATAGATTACCC-1&quot; &quot;CAGCCGACATCCCATC-1&quot; &quot;GCAGCCACACTTCTGC-1&quot;
## [4] &quot;CAAGATCGTCGGCACT-1&quot; &quot;GTAACGTTCCCATTAT-1&quot; &quot;TTCGGTCAGGGTTCCC-1&quot;</code></pre>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="PreProcTop.html#cb323-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(sampleSheet))</span>
<span id="cb323-2"><a href="PreProcTop.html#cb323-2"></a>{</span>
<span id="cb323-3"><a href="PreProcTop.html#cb323-3"></a>  <span class="cf">if</span>(<span class="kw">ncol</span>(sce.list[[i]]) <span class="op">&lt;</span><span class="st"> </span><span class="dv">2000</span>)</span>
<span id="cb323-4"><a href="PreProcTop.html#cb323-4"></a>  {</span>
<span id="cb323-5"><a href="PreProcTop.html#cb323-5"></a>      sce.tmp &lt;-<span class="st"> </span>sce.list[[i]]</span>
<span id="cb323-6"><a href="PreProcTop.html#cb323-6"></a>  } <span class="cf">else</span> {</span>
<span id="cb323-7"><a href="PreProcTop.html#cb323-7"></a>    sce.tmp &lt;-<span class="st"> </span>sce.list[[i]][, <span class="kw">sample</span>(<span class="kw">ncol</span>(sce.list[[i]]), <span class="dv">2000</span>)]</span>
<span id="cb323-8"><a href="PreProcTop.html#cb323-8"></a>  }</span>
<span id="cb323-9"><a href="PreProcTop.html#cb323-9"></a>    <span class="kw">colData</span>(sce.tmp)<span class="op">$</span>Barcode &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;([0-9])$&quot;</span>, i, <span class="kw">colData</span>(sce.tmp)<span class="op">$</span>Barcode)</span>
<span id="cb323-10"><a href="PreProcTop.html#cb323-10"></a>    sce &lt;-<span class="st"> </span><span class="kw">cbind</span>(sce, sce.tmp)</span>
<span id="cb323-11"><a href="PreProcTop.html#cb323-11"></a>    <span class="kw">print</span>(<span class="kw">tail</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode, <span class="dv">2</span>))</span>
<span id="cb323-12"><a href="PreProcTop.html#cb323-12"></a>}</span></code></pre></div>
<pre><code>## [1] &quot;AGAGTGGCACCAGTTA-2&quot; &quot;AACACGTGTACCTACA-2&quot;
## [1] &quot;CGGCTAGAGCTGAACG-3&quot; &quot;TAGCCGGAGTTAGCGG-3&quot;
## [1] &quot;ACACCAAGTCAGAAGC-4&quot; &quot;AAGGTTCAGGCCGAAT-4&quot;
## [1] &quot;AGGGTGAAGAGTCGGT-5&quot; &quot;CTGCGGAGTGAGTATA-5&quot;
## [1] &quot;GGCGACTGTGTTTGGT-6&quot; &quot;TTGGAACAGTGTACTC-6&quot;
## [1] &quot;TGTGGTACACTTACGA-7&quot; &quot;ACTGAGTTCTATCGCC-7&quot;
## [1] &quot;GTCACAACAATCGGTT-8&quot; &quot;CTACATTGTCTCTCGT-8&quot;
## [1] &quot;TTTGGTTGTGCATCTA-9&quot; &quot;TTTGTCACAGCTCGAC-9&quot;
## [1] &quot;TTTGTCAGTAAATGTG-10&quot; &quot;TTTGTCAGTACAAGTA-10&quot;
## [1] &quot;TGCGGGTAGCGTAGTG-11&quot; &quot;CTAGAGTCAAAGTGCG-11&quot;
## [1] &quot;GCAATCAAGGCTATCT-12&quot; &quot;GGGTCTGCATAAGACA-12&quot;
## [1] &quot;TATCTCAGTCATATCG-13&quot; &quot;TCGTACCTCTGGTATG-13&quot;
## [1] &quot;CGTTCTGCAGCTGTTA-14&quot; &quot;TCGCGAGTCATCTGCC-14&quot;
## [1] &quot;ACTTACTCAGGTTTCA-15&quot; &quot;ACTGCTCTCGGTTAAC-15&quot;
## [1] &quot;AGAGTGGCAAAGGCGT-16&quot; &quot;ACACCCTAGATCCCGC-16&quot;
## [1] &quot;GCTGGGTAGGCTCATT-17&quot; &quot;TGTTCCGCACAGACAG-17&quot;
## [1] &quot;CTACATTGTAGCGATG-18&quot; &quot;TGGGCGTCACACCGCA-18&quot;
## [1] &quot;TTCCCAGCAGGGTACA-19&quot; &quot;GTTCTCGGTCCTCCAT-19&quot;
## [1] &quot;GCTGGGTCATAACCTG-20&quot; &quot;TATGCCCCAGCGTCCA-20&quot;</code></pre>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="PreProcTop.html#cb325-1"></a><span class="kw">rm</span>(sce.list)</span></code></pre></div>
<p>We now add the sample sheet information to the object metadata.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="PreProcTop.html#cb326-1"></a>colDataOrig &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)</span>
<span id="cb326-2"><a href="PreProcTop.html#cb326-2"></a></span>
<span id="cb326-3"><a href="PreProcTop.html#cb326-3"></a><span class="co"># split path:</span></span>
<span id="cb326-4"><a href="PreProcTop.html#cb326-4"></a>tmpList &lt;-<span class="st"> </span><span class="kw">strsplit</span>(colDataOrig<span class="op">$</span>Sample, <span class="dt">split=</span><span class="st">&quot;/&quot;</span>)</span>
<span id="cb326-5"><a href="PreProcTop.html#cb326-5"></a><span class="co"># get Run ID, to use to match sample in the meta data and sample sheet objects:</span></span>
<span id="cb326-6"><a href="PreProcTop.html#cb326-6"></a>tmpVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(tmpList, <span class="cf">function</span>(x){x[<span class="dv">9</span>]}))</span>
<span id="cb326-7"><a href="PreProcTop.html#cb326-7"></a><span class="kw">colData</span>(sce)<span class="op">$</span>Run &lt;-<span class="st"> </span>tmpVec</span>
<span id="cb326-8"><a href="PreProcTop.html#cb326-8"></a><span class="co"># remove path to filtered matrix files by sample name in &#39;Sample&#39;:</span></span>
<span id="cb326-9"><a href="PreProcTop.html#cb326-9"></a><span class="kw">colData</span>(sce)<span class="op">$</span>Sample &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb326-10"><a href="PreProcTop.html#cb326-10"></a><span class="co"># merge:</span></span>
<span id="cb326-11"><a href="PreProcTop.html#cb326-11"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb326-12"><a href="PreProcTop.html#cb326-12"></a><span class="st">  </span>data.frame <span class="op">%&gt;%</span></span>
<span id="cb326-13"><a href="PreProcTop.html#cb326-13"></a><span class="st">  </span><span class="kw">left_join</span>(sampleSheet[,splShtColToKeep], <span class="st">&quot;Run&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb326-14"><a href="PreProcTop.html#cb326-14"></a><span class="st">  </span><span class="kw">relocate</span>() <span class="op">%&gt;%</span></span>
<span id="cb326-15"><a href="PreProcTop.html#cb326-15"></a><span class="st">  </span>DataFrame</span>
<span id="cb326-16"><a href="PreProcTop.html#cb326-16"></a><span class="kw">rm</span>(tmpVec)</span></code></pre></div>
<p>Let’s save the object for future reference.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="PreProcTop.html#cb327-1"></a><span class="co"># Write object to file</span></span>
<span id="cb327-2"><a href="PreProcTop.html#cb327-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postPool.Rds&quot;</span>,</span>
<span id="cb327-3"><a href="PreProcTop.html#cb327-3"></a>         projDir, outDirBit)</span>
<span id="cb327-4"><a href="PreProcTop.html#cb327-4"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span>
<span id="cb327-5"><a href="PreProcTop.html#cb327-5"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="PreProcTop.html#cb328-1"></a><span class="co"># Read object in:</span></span>
<span id="cb328-2"><a href="PreProcTop.html#cb328-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postPool.Rds&quot;</span>,</span>
<span id="cb328-3"><a href="PreProcTop.html#cb328-3"></a>         projDir, outDirBit)</span>
<span id="cb328-4"><a href="PreProcTop.html#cb328-4"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb328-5"><a href="PreProcTop.html#cb328-5"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
</div>
<div id="properties-of-scrna-seq-data-1" class="section level2">
<h2><span class="header-section-number">5.9</span> Properties of scRNA-seq data</h2>
<!-- TODO add text -->
<p>The number and identity of genes detected in a cell vary across cells: the total number of genes detected across all cells is far larger than the number of genes detected in each cell.</p>
<p>Total number of genes detected across cells:</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="PreProcTop.html#cb329-1"></a><span class="co"># for each gene, compute total number of UMIs across all cells,</span></span>
<span id="cb329-2"><a href="PreProcTop.html#cb329-2"></a><span class="co"># then counts genes with at least one UMI:</span></span>
<span id="cb329-3"><a href="PreProcTop.html#cb329-3"></a><span class="kw">sum</span>(<span class="kw">rowSums</span>(<span class="kw">counts</span>(sce)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<pre><code>## [1] 25116</code></pre>
<p>Summary of the distribution of the number of genes detected per cell:</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="PreProcTop.html#cb331-1"></a><span class="co"># for each cell count number of genes with at least 1 UMI</span></span>
<span id="cb331-2"><a href="PreProcTop.html#cb331-2"></a><span class="co"># then compute distribution moments:</span></span>
<span id="cb331-3"><a href="PreProcTop.html#cb331-3"></a><span class="kw">summary</span>(<span class="kw">colSums</span>(<span class="kw">counts</span>(sce) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##      26     717     940    1228    1521    6586</code></pre>
<p>Now let’s plot for each gene the total number of UMIs and the proportion of cells that express it. Lowly expressed genes tend to be detected in a large proportion of cells. The higher the overall expression the lower the proportion of cells.</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="PreProcTop.html#cb333-1"></a><span class="co"># randomly subset 1000 cells.</span></span>
<span id="cb333-2"><a href="PreProcTop.html#cb333-2"></a>tmpCounts &lt;-<span class="st"> </span><span class="kw">counts</span>(sce)[,<span class="kw">sample</span>(<span class="dv">1000</span>)]</span>
<span id="cb333-3"><a href="PreProcTop.html#cb333-3"></a></span>
<span id="cb333-4"><a href="PreProcTop.html#cb333-4"></a><span class="kw">plot</span>(<span class="kw">rowSums</span>(tmpCounts),</span>
<span id="cb333-5"><a href="PreProcTop.html#cb333-5"></a>     <span class="kw">rowMeans</span>(tmpCounts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>),</span>
<span id="cb333-6"><a href="PreProcTop.html#cb333-6"></a>     <span class="dt">log =</span> <span class="st">&quot;x&quot;</span>,</span>
<span id="cb333-7"><a href="PreProcTop.html#cb333-7"></a>     <span class="dt">xlab=</span><span class="st">&quot;total number of UMIs&quot;</span>,</span>
<span id="cb333-8"><a href="PreProcTop.html#cb333-8"></a>     <span class="dt">ylab=</span><span class="st">&quot;proportion of cells expressing the gene&quot;</span></span>
<span id="cb333-9"><a href="PreProcTop.html#cb333-9"></a>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p>Count genes that are not ‘expressed’ (detected):</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="PreProcTop.html#cb334-1"></a>not.expressed &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(sce)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb334-2"><a href="PreProcTop.html#cb334-2"></a><span class="kw">table</span>(not.expressed)</span></code></pre></div>
<pre><code>## not.expressed
## FALSE  TRUE 
## 25116  8422</code></pre>
<p>Plot the percentage of counts per gene and show genes with the highest UMI counts:</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="PreProcTop.html#cb336-1"></a><span class="co">#Compute the relative expression of each gene per cell</span></span>
<span id="cb336-2"><a href="PreProcTop.html#cb336-2"></a>rel_expression &lt;-<span class="st"> </span><span class="kw">t</span>( <span class="kw">t</span>(<span class="kw">counts</span>(sce)) <span class="op">/</span><span class="st"> </span>Matrix<span class="op">::</span><span class="kw">colSums</span>(<span class="kw">counts</span>(sce))) <span class="op">*</span><span class="st"> </span><span class="dv">100</span></span>
<span id="cb336-3"><a href="PreProcTop.html#cb336-3"></a><span class="kw">rownames</span>(rel_expression) &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol</span>
<span id="cb336-4"><a href="PreProcTop.html#cb336-4"></a>most_expressed &lt;-<span class="st"> </span><span class="kw">sort</span>(Matrix<span class="op">::</span><span class="kw">rowSums</span>( rel_expression ),T)[<span class="dv">20</span><span class="op">:</span><span class="dv">1</span>] <span class="op">/</span><span class="st"> </span><span class="kw">ncol</span>(sce)</span>
<span id="cb336-5"><a href="PreProcTop.html#cb336-5"></a></span>
<span id="cb336-6"><a href="PreProcTop.html#cb336-6"></a><span class="kw">boxplot</span>( <span class="kw">as.matrix</span>(<span class="kw">t</span>(rel_expression[<span class="kw">names</span>(most_expressed),])),</span>
<span id="cb336-7"><a href="PreProcTop.html#cb336-7"></a>    <span class="dt">cex=</span>.<span class="dv">1</span>,</span>
<span id="cb336-8"><a href="PreProcTop.html#cb336-8"></a>    <span class="dt">las=</span><span class="dv">1</span>,</span>
<span id="cb336-9"><a href="PreProcTop.html#cb336-9"></a>    <span class="dt">xlab=</span><span class="st">&quot;% total count per cell&quot;</span>,</span>
<span id="cb336-10"><a href="PreProcTop.html#cb336-10"></a>    <span class="dt">col=</span>scales<span class="op">::</span><span class="kw">hue_pal</span>()(<span class="dv">20</span>)[<span class="dv">20</span><span class="op">:</span><span class="dv">1</span>],</span>
<span id="cb336-11"><a href="PreProcTop.html#cb336-11"></a>    <span class="dt">horizontal=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="PreProcTop.html#cb337-1"></a><span class="kw">rm</span>(rel_expression, most_expressed)</span></code></pre></div>
<p>Mind that we have combined two data sets here. It may be interesting to count non-expressed genes in each set separately.</p>
</div>
<div id="quality-control-1" class="section level2">
<h2><span class="header-section-number">5.10</span> Quality control</h2>
<!-- https://osca.bioconductor.org/quality-control.html -->
<p>Cell calling performed above does not inform on the quality of the library in each of the droplets kept. Poor-quality cells, or rather droplets, may be caused by cell damage during dissociation or failed library preparation. They usually have low UMI counts, few genes detected and/or high mitochondrial content. The presence may affect normalisation, assessment of cell population heterogeneity, clustering and trajectory:</p>
<ul>
<li>Normalisation: Contaminating genes, ‘the ambient RNA’, are detected at low levels in all libraires. In low quality libraries with low RNA content, scaling will increase counts for these genes more than for better-quality cells, resulting in their apparent upregulation in these cells and increased variance overall.</li>
<li>Cell population heterogeneity: variance estimation and dimensionality reduction with PCA where the first principal component will be correlated with library size, rather than biology.</li>
<li>Clustering and trajectory: higher mitochondrial and/or nuclear RNA content may cause low-quality cells to cluster separately or form states or trajectories between distinct cell types.</li>
</ul>
<p>We will now exclude lowly expressed features and identify low-quality cells using the following metrics mostly:</p>
<ul>
<li>library size, i.e. the total number of UMIs per cell</li>
<li>number of features detected per cell</li>
<li>mitochondrial content, i.e. the proportion of UMIs that map to mitochondrial genes, with higher values consistent with leakage from the cytoplasm of RNA, but not mitochondria</li>
</ul>
<p>We will first annotate genes, to know which ones are mitochondrial, then use <a href="https://bioconductor.org/packages/3.11/bioc/html/scater.html">scater</a>’s <code>addPerCellQC()</code> to compute various metrics.</p>
<p>Annotate genes with biomaRt.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="PreProcTop.html#cb338-1"></a><span class="co"># retrieve the feature information</span></span>
<span id="cb338-2"><a href="PreProcTop.html#cb338-2"></a>gene.info &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)</span>
<span id="cb338-3"><a href="PreProcTop.html#cb338-3"></a></span>
<span id="cb338-4"><a href="PreProcTop.html#cb338-4"></a><span class="co"># setup the biomaRt connection to Ensembl using the correct species genome (hsapiens_gene_ensembl)</span></span>
<span id="cb338-5"><a href="PreProcTop.html#cb338-5"></a>ensembl &lt;-<span class="st"> </span><span class="kw">useEnsembl</span>(<span class="dt">biomart=</span><span class="st">&#39;ensembl&#39;</span>,</span>
<span id="cb338-6"><a href="PreProcTop.html#cb338-6"></a>                      <span class="dt">dataset=</span><span class="st">&#39;hsapiens_gene_ensembl&#39;</span>,</span>
<span id="cb338-7"><a href="PreProcTop.html#cb338-7"></a>                      <span class="dt">mirror =</span> <span class="st">&quot;www&quot;</span>)</span>
<span id="cb338-8"><a href="PreProcTop.html#cb338-8"></a></span>
<span id="cb338-9"><a href="PreProcTop.html#cb338-9"></a><span class="co">#ensembl = useMart(biomart=&quot;ensembl&quot;,</span></span>
<span id="cb338-10"><a href="PreProcTop.html#cb338-10"></a><span class="co">#         dataset=&#39;hsapiens_gene_ensembl&#39;,</span></span>
<span id="cb338-11"><a href="PreProcTop.html#cb338-11"></a><span class="co">#         host = &quot;www.ensembl.org&quot;) # ensemblRedirect = FALSE</span></span>
<span id="cb338-12"><a href="PreProcTop.html#cb338-12"></a></span>
<span id="cb338-13"><a href="PreProcTop.html#cb338-13"></a><span class="co"># retrieve the attributes of interest from biomaRt using the Ensembl gene ID as the key</span></span>
<span id="cb338-14"><a href="PreProcTop.html#cb338-14"></a><span class="co"># beware that this will only retrieve information for matching IDs</span></span>
<span id="cb338-15"><a href="PreProcTop.html#cb338-15"></a>gene_symbol &lt;-<span class="st"> </span><span class="kw">getBM</span>(<span class="dt">attributes=</span><span class="kw">c</span>(<span class="st">&#39;ensembl_gene_id&#39;</span>,</span>
<span id="cb338-16"><a href="PreProcTop.html#cb338-16"></a>                                  <span class="st">&#39;external_gene_name&#39;</span>,</span>
<span id="cb338-17"><a href="PreProcTop.html#cb338-17"></a>                                  <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb338-18"><a href="PreProcTop.html#cb338-18"></a>                                  <span class="st">&#39;start_position&#39;</span>,</span>
<span id="cb338-19"><a href="PreProcTop.html#cb338-19"></a>                                  <span class="st">&#39;end_position&#39;</span>,</span>
<span id="cb338-20"><a href="PreProcTop.html#cb338-20"></a>                                  <span class="st">&#39;strand&#39;</span>),</span>
<span id="cb338-21"><a href="PreProcTop.html#cb338-21"></a>                     <span class="dt">filters=</span><span class="st">&#39;ensembl_gene_id&#39;</span>,</span>
<span id="cb338-22"><a href="PreProcTop.html#cb338-22"></a>                     <span class="dt">mart=</span>ensembl,</span>
<span id="cb338-23"><a href="PreProcTop.html#cb338-23"></a>                     <span class="dt">values=</span>gene.info[, <span class="dv">1</span>])</span>
<span id="cb338-24"><a href="PreProcTop.html#cb338-24"></a></span>
<span id="cb338-25"><a href="PreProcTop.html#cb338-25"></a><span class="co"># create a new data frame of the feature information</span></span>
<span id="cb338-26"><a href="PreProcTop.html#cb338-26"></a>gene.merge &lt;-<span class="st"> </span><span class="kw">merge</span>(gene_symbol,</span>
<span id="cb338-27"><a href="PreProcTop.html#cb338-27"></a>            gene.info,</span>
<span id="cb338-28"><a href="PreProcTop.html#cb338-28"></a>            <span class="dt">by.x=</span><span class="kw">c</span>(<span class="st">&#39;ensembl_gene_id&#39;</span>),</span>
<span id="cb338-29"><a href="PreProcTop.html#cb338-29"></a>            <span class="dt">by.y=</span><span class="kw">c</span>(<span class="st">&#39;ID&#39;</span>),</span>
<span id="cb338-30"><a href="PreProcTop.html#cb338-30"></a>            <span class="dt">all.y=</span><span class="ot">TRUE</span>)</span>
<span id="cb338-31"><a href="PreProcTop.html#cb338-31"></a><span class="kw">rownames</span>(gene.merge) &lt;-<span class="st"> </span>gene.merge<span class="op">$</span>ensembl_gene_id</span>
<span id="cb338-32"><a href="PreProcTop.html#cb338-32"></a></span>
<span id="cb338-33"><a href="PreProcTop.html#cb338-33"></a><span class="co"># set the order for the same as the original gene information</span></span>
<span id="cb338-34"><a href="PreProcTop.html#cb338-34"></a>gene.merge &lt;-<span class="st"> </span>gene.merge[gene.info[, <span class="dv">1</span>], ]</span>
<span id="cb338-35"><a href="PreProcTop.html#cb338-35"></a></span>
<span id="cb338-36"><a href="PreProcTop.html#cb338-36"></a><span class="co"># reset the rowdata on the SCE object to contain all of this information</span></span>
<span id="cb338-37"><a href="PreProcTop.html#cb338-37"></a><span class="kw">rowData</span>(sce) &lt;-<span class="st"> </span>gene.merge</span>
<span id="cb338-38"><a href="PreProcTop.html#cb338-38"></a></span>
<span id="cb338-39"><a href="PreProcTop.html#cb338-39"></a><span class="kw">rm</span>(gene_symbol, gene.merge)</span></code></pre></div>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="PreProcTop.html#cb339-1"></a><span class="co"># slow</span></span>
<span id="cb339-2"><a href="PreProcTop.html#cb339-2"></a><span class="co"># Write object to file</span></span>
<span id="cb339-3"><a href="PreProcTop.html#cb339-3"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postBiomart.Rds&quot;</span>,</span>
<span id="cb339-4"><a href="PreProcTop.html#cb339-4"></a>         projDir, outDirBit)</span>
<span id="cb339-5"><a href="PreProcTop.html#cb339-5"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span>
<span id="cb339-6"><a href="PreProcTop.html#cb339-6"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="PreProcTop.html#cb340-1"></a><span class="co"># Read object in:</span></span>
<span id="cb340-2"><a href="PreProcTop.html#cb340-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postBiomart.Rds&quot;</span>,</span>
<span id="cb340-3"><a href="PreProcTop.html#cb340-3"></a>         projDir, outDirBit)</span>
<span id="cb340-4"><a href="PreProcTop.html#cb340-4"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb340-5"><a href="PreProcTop.html#cb340-5"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>Number of genes per chromosome, inc. 13 on the mitochondrial genome:</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="PreProcTop.html#cb341-1"></a><span class="co"># number of genes per chromosome</span></span>
<span id="cb341-2"><a href="PreProcTop.html#cb341-2"></a><span class="kw">table</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>chromosome_name) <span class="op">%&gt;%</span></span>
<span id="cb341-3"><a href="PreProcTop.html#cb341-3"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb341-4"><a href="PreProcTop.html#cb341-4"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Chromosome=</span>Var1, <span class="dt">NbGenes=</span>Freq) <span class="op">%&gt;%</span></span>
<span id="cb341-5"><a href="PreProcTop.html#cb341-5"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-eccd69a070e79a2bd7cb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-eccd69a070e79a2bd7cb">{"x":{"filter":"none","data":[["1","10","11","12","13","14","15","16","17","18","19","2","20","21","22","3","4","5","6","7","8","9","GL000009.2","GL000194.1","GL000195.1","GL000205.2","GL000213.1","GL000218.1","GL000219.1","KI270711.1","KI270713.1","KI270721.1","KI270726.1","KI270727.1","KI270728.1","KI270731.1","KI270734.1","MT","X","Y"],[3126,1220,1910,1748,661,1380,1121,1515,1880,654,1989,2261,888,499,831,1704,1344,1646,1616,1526,1302,1210,1,2,2,1,1,1,1,1,2,1,2,4,6,1,3,13,1064,97]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Chromosome<\/th>\n      <th>NbGenes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="PreProcTop.html#cb342-1"></a><span class="co"># mitochondrial genes</span></span>
<span id="cb342-2"><a href="PreProcTop.html#cb342-2"></a>is.mito &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>chromosome_name<span class="op">==</span><span class="st">&quot;MT&quot;</span>)</span></code></pre></div>
<p>Calculate and store QC metrics for genes with addPerFeatureQC() and for cells
with addPerCellQC().</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="PreProcTop.html#cb343-1"></a><span class="co"># long</span></span>
<span id="cb343-2"><a href="PreProcTop.html#cb343-2"></a><span class="co"># for genes</span></span>
<span id="cb343-3"><a href="PreProcTop.html#cb343-3"></a>sce &lt;-<span class="st"> </span><span class="kw">addPerFeatureQC</span>(sce)</span>
<span id="cb343-4"><a href="PreProcTop.html#cb343-4"></a><span class="kw">head</span>(<span class="kw">rowData</span>(sce)) <span class="op">%&gt;%</span></span>
<span id="cb343-5"><a href="PreProcTop.html#cb343-5"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb343-6"><a href="PreProcTop.html#cb343-6"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>) <span class="co"># ENS ID</span></span></code></pre></div>
<div id="htmlwidget-3cc9a2415ae30da5cda5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3cc9a2415ae30da5cda5">{"x":{"filter":"none","data":[["ENSG00000243485","ENSG00000237613","ENSG00000186092","ENSG00000238009","ENSG00000239945","ENSG00000239906"],["MIR1302-2HG","FAM138A","OR4F5","","",""],["1","1","1","1","1","1"],[29554,34554,65419,89295,89551,139790],[31109,36081,71585,133723,91105,140339],[1,-1,1,-1,-1,-1],["MIR1302-2HG","FAM138A","OR4F5","AL627309.1","AL627309.3","AL627309.2"],["Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression","Gene Expression"],[0,0,0,0.00086550566512799,5.24548887956358e-05,0],[0,0,0,0.086550566512799,0.00524548887956358,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ensembl_gene_id<\/th>\n      <th>external_gene_name<\/th>\n      <th>chromosome_name<\/th>\n      <th>start_position<\/th>\n      <th>end_position<\/th>\n      <th>strand<\/th>\n      <th>Symbol<\/th>\n      <th>Type<\/th>\n      <th>mean<\/th>\n      <th>detected<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,8,9]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Three columns of interest for cells:</p>
<ul>
<li>‘sum’: total UMI count</li>
<li>‘detected’: number of features (genes) detected</li>
<li>‘subsets_Mito_percent’: percentage of reads mapped to mitochondrial transcripts</li>
</ul>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="PreProcTop.html#cb344-1"></a><span class="co"># for cells</span></span>
<span id="cb344-2"><a href="PreProcTop.html#cb344-2"></a>sce &lt;-<span class="st"> </span><span class="kw">addPerCellQC</span>(sce, <span class="dt">subsets=</span><span class="kw">list</span>(<span class="dt">Mito=</span>is.mito))</span>
<span id="cb344-3"><a href="PreProcTop.html#cb344-3"></a><span class="kw">head</span>(<span class="kw">colData</span>(sce)) <span class="op">%&gt;%</span></span>
<span id="cb344-4"><a href="PreProcTop.html#cb344-4"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb344-5"><a href="PreProcTop.html#cb344-5"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-7aa50ed2f9803d60bbab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7aa50ed2f9803d60bbab">{"x":{"filter":"none","data":[["GATGCTAGTATCAGTC-1","GTCCTCAGTCCAAGTT-1","CGCGTTTAGGCTAGCA-1","GGATTACGTGCGAAAC-1","CGTTGGGCAGCCAATT-1","GTGCGGTGTCATTAGC-1"],["SRR9264343","SRR9264343","SRR9264343","SRR9264343","SRR9264343","SRR9264343"],["GSM3872434","GSM3872434","GSM3872434","GSM3872434","GSM3872434","GSM3872434"],["ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1","ETV6-RUNX1"],[4301,573,4379,4989,3900,4108],[1550,331,1675,1867,1528,1579],[265,22,225,244,263,288],[11,7,12,11,12,12],[6.16135782376192,3.83944153577661,5.13815939712263,4.89075967127681,6.74358974358974,7.01071080817916],[4301,573,4379,4989,3900,4108]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Barcode<\/th>\n      <th>Run<\/th>\n      <th>Sample.Name<\/th>\n      <th>source_name<\/th>\n      <th>sum<\/th>\n      <th>detected<\/th>\n      <th>subsets_Mito_sum<\/th>\n      <th>subsets_Mito_detected<\/th>\n      <th>subsets_Mito_percent<\/th>\n      <th>total<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="PreProcTop.html#cb345-1"></a><span class="co"># Write object to file</span></span>
<span id="cb345-2"><a href="PreProcTop.html#cb345-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postAddQc.Rds&quot;</span>,</span>
<span id="cb345-3"><a href="PreProcTop.html#cb345-3"></a>         projDir, outDirBit)</span>
<span id="cb345-4"><a href="PreProcTop.html#cb345-4"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span>
<span id="cb345-5"><a href="PreProcTop.html#cb345-5"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="PreProcTop.html#cb346-1"></a><span class="co"># Read object in:</span></span>
<span id="cb346-2"><a href="PreProcTop.html#cb346-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postAddQc.Rds&quot;</span>,</span>
<span id="cb346-3"><a href="PreProcTop.html#cb346-3"></a>         projDir, outDirBit)</span>
<span id="cb346-4"><a href="PreProcTop.html#cb346-4"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb346-5"><a href="PreProcTop.html#cb346-5"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div id="qc-metric-distribution-1" class="section level3">
<h3><span class="header-section-number">5.10.1</span> QC metric distribution</h3>
<p>Overall:</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="PreProcTop.html#cb347-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb347-2"><a href="PreProcTop.html#cb347-2"></a><span class="kw">hist</span>(<span class="kw">log10</span>(sce<span class="op">$</span>sum),</span>
<span id="cb347-3"><a href="PreProcTop.html#cb347-3"></a>     <span class="dt">breaks=</span><span class="dv">20</span>,</span>
<span id="cb347-4"><a href="PreProcTop.html#cb347-4"></a>     <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>,</span>
<span id="cb347-5"><a href="PreProcTop.html#cb347-5"></a>     <span class="dt">xlab=</span><span class="st">&quot;Log-total UMI count&quot;</span>,</span>
<span id="cb347-6"><a href="PreProcTop.html#cb347-6"></a>     <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb347-7"><a href="PreProcTop.html#cb347-7"></a><span class="kw">hist</span>(sce<span class="op">$</span>subsets_Mito_percent,</span>
<span id="cb347-8"><a href="PreProcTop.html#cb347-8"></a>     <span class="dt">breaks=</span><span class="dv">20</span>,</span>
<span id="cb347-9"><a href="PreProcTop.html#cb347-9"></a>     <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>,</span>
<span id="cb347-10"><a href="PreProcTop.html#cb347-10"></a>     <span class="dt">xlab=</span><span class="st">&quot;Proportion of reads in mitochondrial genes&quot;</span>,</span>
<span id="cb347-11"><a href="PreProcTop.html#cb347-11"></a>     <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb347-12"><a href="PreProcTop.html#cb347-12"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">20</span>, <span class="dt">lty=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">&#39;purple&#39;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/qc_metricDistrib_twoHist-1.png" width="672" /></p>
<p>Per sample group:</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="PreProcTop.html#cb348-1"></a>sce<span class="op">$</span>source_name &lt;-<span class="st"> </span><span class="kw">factor</span>(sce<span class="op">$</span>source_name)</span>
<span id="cb348-2"><a href="PreProcTop.html#cb348-2"></a>sce<span class="op">$</span>block &lt;-<span class="st"> </span>sce<span class="op">$</span>source_name</span>
<span id="cb348-3"><a href="PreProcTop.html#cb348-3"></a>sce<span class="op">$</span>setName &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;ABMMC&quot;</span>, sce<span class="op">$</span>source_name), <span class="st">&quot;Hca&quot;</span>, <span class="st">&quot;Caron&quot;</span>)</span></code></pre></div>
<p>Library size (‘Total count’):</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="PreProcTop.html#cb349-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;sum&quot;</span>,</span>
<span id="cb349-2"><a href="PreProcTop.html#cb349-2"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#facet_wrap(~setName) + </span></span>
<span id="cb349-3"><a href="PreProcTop.html#cb349-3"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Total count&quot;</span>)</span>
<span id="cb349-4"><a href="PreProcTop.html#cb349-4"></a>p</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>Number of genes (‘detected’):</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="PreProcTop.html#cb350-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>, </span>
<span id="cb350-2"><a href="PreProcTop.html#cb350-2"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#facet_wrap(~setName) + </span></span>
<span id="cb350-3"><a href="PreProcTop.html#cb350-3"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detected features&quot;</span>)</span>
<span id="cb350-4"><a href="PreProcTop.html#cb350-4"></a>p</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Mitochondrial content (‘subsets_Mito_percent’):</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="PreProcTop.html#cb351-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>, </span>
<span id="cb351-2"><a href="PreProcTop.html#cb351-2"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># facet_wrap(~setName) +</span></span>
<span id="cb351-3"><a href="PreProcTop.html#cb351-3"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="st">&quot;Mito percent&quot;</span>)</span>
<span id="cb351-4"><a href="PreProcTop.html#cb351-4"></a>p</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="PreProcTop.html#cb352-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/qc_metricDistrib_plotColData2.png&quot;</span>, projDir, outDirBit, qcPlotDirBit)</span>
<span id="cb352-2"><a href="PreProcTop.html#cb352-2"></a><span class="kw">CairoPNG</span>(tmpFn)</span>
<span id="cb352-3"><a href="PreProcTop.html#cb352-3"></a><span class="co"># violin plots</span></span>
<span id="cb352-4"><a href="PreProcTop.html#cb352-4"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(</span>
<span id="cb352-5"><a href="PreProcTop.html#cb352-5"></a>    <span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;sum&quot;</span>,</span>
<span id="cb352-6"><a href="PreProcTop.html#cb352-6"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#facet_wrap(~setName) + </span></span>
<span id="cb352-7"><a href="PreProcTop.html#cb352-7"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Total count&quot;</span>),</span>
<span id="cb352-8"><a href="PreProcTop.html#cb352-8"></a>    <span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>, </span>
<span id="cb352-9"><a href="PreProcTop.html#cb352-9"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#facet_wrap(~setName) + </span></span>
<span id="cb352-10"><a href="PreProcTop.html#cb352-10"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detected features&quot;</span>),</span>
<span id="cb352-11"><a href="PreProcTop.html#cb352-11"></a>    <span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>, </span>
<span id="cb352-12"><a href="PreProcTop.html#cb352-12"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># facet_wrap(~setName) +</span></span>
<span id="cb352-13"><a href="PreProcTop.html#cb352-13"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="st">&quot;Mito percent&quot;</span>),</span>
<span id="cb352-14"><a href="PreProcTop.html#cb352-14"></a>    <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb352-15"><a href="PreProcTop.html#cb352-15"></a>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/qc_metricDistrib_plotColData_write-1.png" width="672" /></p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="PreProcTop.html#cb353-1"></a><span class="kw">dev.off</span>()</span></code></pre></div>
<pre><code>## pdf 
##   2</code></pre>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="PreProcTop.html#cb355-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;../%s/qc_metricDistrib_plotColData2.png&quot;</span>, qcPlotDirBit)</span>
<span id="cb355-2"><a href="PreProcTop.html#cb355-2"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Qc/qc_metricDistrib_plotColData2.png" width="80%" /></p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="PreProcTop.html#cb356-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>Correlation between the number of genes detected and library size (‘detected’ against ‘sum’):</p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="PreProcTop.html#cb357-1"></a>sp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="kw">colData</span>(sce)),</span>
<span id="cb357-2"><a href="PreProcTop.html#cb357-2"></a>             <span class="kw">aes</span>(<span class="dt">x=</span>sum,</span>
<span id="cb357-3"><a href="PreProcTop.html#cb357-3"></a>                 <span class="dt">y=</span>detected,</span>
<span id="cb357-4"><a href="PreProcTop.html#cb357-4"></a>                 <span class="dt">col=</span>subsets_Mito_percent)) <span class="op">+</span></span>
<span id="cb357-5"><a href="PreProcTop.html#cb357-5"></a><span class="st">  </span><span class="kw">geom_point</span>()</span>
<span id="cb357-6"><a href="PreProcTop.html#cb357-6"></a>sp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>source_name)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
</div>
<div id="identification-of-low-quality-cells-with-adaptive-thresholds-1" class="section level2">
<h2><span class="header-section-number">5.11</span> Identification of low-quality cells with adaptive thresholds</h2>
<p>One can use hard threshold for the library size, number of genes detected and mitochondrial content. These will however vary across runs. It may therefore be preferable to rely on outlier detection to identify cells that markerdly differ from most cells.</p>
<p>We saw above that the distribution of the QC metrics is close to Normal. Hence, we can detect outlier using the median and the median absolute deviation (MAD) from the median (not the mean and the standard deviation that both are sensitive to outliers).</p>
<p>For a given metric, an outlier value is one that lies over some number of MADs away from the median. A cell will be excluded if it is an outlier in the part of the range to avoid, for example low gene counts, or high mitochondrial content. For a normal distribution, a threshold defined with a distance of 3 MADs from the median retains about 99% of values.</p>
<div id="library-size-1" class="section level3">
<h3><span class="header-section-number">5.11.1</span> Library size</h3>
<p>For the library size we use the log scale to avoid negative values for lower part of the distribution.</p>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="PreProcTop.html#cb358-1"></a>qc.lib2 &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>sum, <span class="dt">log=</span><span class="ot">TRUE</span>, <span class="dt">type=</span><span class="st">&quot;lower&quot;</span>)</span>
<span id="cb358-2"><a href="PreProcTop.html#cb358-2"></a><span class="kw">table</span>(qc.lib2)</span></code></pre></div>
<pre><code>## qc.lib2
## FALSE  TRUE 
## 37795   333</code></pre>
<p>Threshold values:</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="PreProcTop.html#cb360-1"></a><span class="kw">attr</span>(qc.lib2, <span class="st">&quot;thresholds&quot;</span>)</span></code></pre></div>
<pre><code>##    lower   higher 
## 576.4435      Inf</code></pre>
</div>
<div id="number-of-genes-1" class="section level3">
<h3><span class="header-section-number">5.11.2</span> Number of genes</h3>
<p>For the number of genes detected we also use the log scale to avoid negative values for lower part of the distribution.</p>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="PreProcTop.html#cb362-1"></a>qc.nexprs2 &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>detected, <span class="dt">log=</span><span class="ot">TRUE</span>, <span class="dt">type=</span><span class="st">&quot;lower&quot;</span>)</span>
<span id="cb362-2"><a href="PreProcTop.html#cb362-2"></a><span class="kw">table</span>(qc.nexprs2)</span></code></pre></div>
<pre><code>## qc.nexprs2
## FALSE  TRUE 
## 37723   405</code></pre>
<p>Threshold values:</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="PreProcTop.html#cb364-1"></a><span class="kw">attr</span>(qc.nexprs2, <span class="st">&quot;thresholds&quot;</span>)</span></code></pre></div>
<pre><code>##    lower   higher 
## 204.1804      Inf</code></pre>
</div>
<div id="mitochondrial-content-2" class="section level3">
<h3><span class="header-section-number">5.11.3</span> Mitochondrial content</h3>
<p>For the mitochondrial content the exclusion zone is in the higher part of the distribution.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="PreProcTop.html#cb366-1"></a>qc.mito2 &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>subsets_Mito_percent, <span class="dt">type=</span><span class="st">&quot;higher&quot;</span>)</span>
<span id="cb366-2"><a href="PreProcTop.html#cb366-2"></a><span class="kw">table</span>(qc.mito2)</span></code></pre></div>
<pre><code>## qc.mito2
## FALSE  TRUE 
## 35755  2373</code></pre>
<p>Threshold values:</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="PreProcTop.html#cb368-1"></a><span class="kw">attr</span>(qc.mito2, <span class="st">&quot;thresholds&quot;</span>)</span></code></pre></div>
<pre><code>##    lower   higher 
##     -Inf 8.715114</code></pre>
</div>
<div id="summary-1" class="section level3">
<h3><span class="header-section-number">5.11.4</span> Summary</h3>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="PreProcTop.html#cb370-1"></a>discard2 &lt;-<span class="st"> </span>qc.lib2 <span class="op">|</span><span class="st"> </span>qc.nexprs2 <span class="op">|</span><span class="st"> </span>qc.mito2</span>
<span id="cb370-2"><a href="PreProcTop.html#cb370-2"></a></span>
<span id="cb370-3"><a href="PreProcTop.html#cb370-3"></a><span class="co"># Summarize the number of cells removed for each reason.</span></span>
<span id="cb370-4"><a href="PreProcTop.html#cb370-4"></a><span class="kw">DataFrame</span>(<span class="dt">LibSize=</span><span class="kw">sum</span>(qc.lib2),</span>
<span id="cb370-5"><a href="PreProcTop.html#cb370-5"></a>      <span class="dt">NExprs=</span><span class="kw">sum</span>(qc.nexprs2),</span>
<span id="cb370-6"><a href="PreProcTop.html#cb370-6"></a>      <span class="dt">MitoProp=</span><span class="kw">sum</span>(qc.mito2),</span>
<span id="cb370-7"><a href="PreProcTop.html#cb370-7"></a>      <span class="dt">Total=</span><span class="kw">sum</span>(discard2))</span></code></pre></div>
<pre><code>## DataFrame with 1 row and 4 columns
##     LibSize    NExprs  MitoProp     Total
##   &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 1       333       405      2373      2900</code></pre>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="PreProcTop.html#cb372-1"></a><span class="kw">rm</span>(discard2, qc.lib2, qc.nexprs2, qc.mito2)</span></code></pre></div>
</div>
<div id="all-steps-at-once-1" class="section level3">
<h3><span class="header-section-number">5.11.5</span> All steps at once</h3>
<p>The steps above may be run at once with quickPerCellQC():</p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="PreProcTop.html#cb373-1"></a>reasons &lt;-<span class="st"> </span><span class="kw">quickPerCellQC</span>(<span class="kw">colData</span>(sce),</span>
<span id="cb373-2"><a href="PreProcTop.html#cb373-2"></a>              <span class="dt">percent_subsets=</span><span class="kw">c</span>(<span class="st">&quot;subsets_Mito_percent&quot;</span>))</span>
<span id="cb373-3"><a href="PreProcTop.html#cb373-3"></a><span class="kw">colSums</span>(<span class="kw">as.matrix</span>(reasons)) <span class="op">%&gt;%</span></span>
<span id="cb373-4"><a href="PreProcTop.html#cb373-4"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb373-5"><a href="PreProcTop.html#cb373-5"></a><span class="st">  </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div id="htmlwidget-662f83bad4a8a30bd873" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-662f83bad4a8a30bd873">{"x":{"filter":"none","data":[["low_lib_size","low_n_features","high_subsets_Mito_percent","discard"],[333,405,2373,2900]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="assumptions-1" class="section level3">
<h3><span class="header-section-number">5.11.6</span> Assumptions</h3>
<p>Data quality depends on the tissue analysed, some being difficult to dissociate,
e.g. brain, so that one level of QC stringency will not fit all data sets.</p>
<p>Filtering based on QC metrics as done here assumes that these QC metrics are not
correlated with biology. This may not necessarily be true in highly heterogenous
data sets where some cell types represented by good-quality cells may have low
RNA content or high mitochondrial content.</p>
</div>
</div>
<div id="experimental-factors-1" class="section level2">
<h2><span class="header-section-number">5.12</span> Experimental factors</h2>
<p>The two data sets analysed here may have been obtained in experiments with
different settings, such as cell preparation or sequencing depth. Such
differences between these two batches would affect the adaptive thesholds
discussed above. We will now perform QC in each batch separately.</p>
<p>We will use the quickPerCellQC() ‘batch’ option.</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="PreProcTop.html#cb374-1"></a>batch.reasons &lt;-<span class="st"> </span><span class="kw">quickPerCellQC</span>(<span class="kw">colData</span>(sce),</span>
<span id="cb374-2"><a href="PreProcTop.html#cb374-2"></a>                                <span class="dt">percent_subsets=</span><span class="kw">c</span>(<span class="st">&quot;subsets_Mito_percent&quot;</span>),</span>
<span id="cb374-3"><a href="PreProcTop.html#cb374-3"></a>                <span class="dt">batch=</span>sce<span class="op">$</span>setName)</span>
<span id="cb374-4"><a href="PreProcTop.html#cb374-4"></a><span class="kw">colSums</span>(<span class="kw">as.matrix</span>(batch.reasons)) <span class="op">%&gt;%</span></span>
<span id="cb374-5"><a href="PreProcTop.html#cb374-5"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb374-6"><a href="PreProcTop.html#cb374-6"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div id="htmlwidget-b80324c37caecc5c2650" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b80324c37caecc5c2650">{"x":{"filter":"none","data":[["low_lib_size","low_n_features","high_subsets_Mito_percent","discard"],[0,218,2249,2443]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="PreProcTop.html#cb375-1"></a>sce<span class="op">$</span>discard &lt;-<span class="st"> </span>batch.reasons<span class="op">$</span>discard</span></code></pre></div>
<p>Fewer cells are discarded, in particular because of small library size and low gene number.</p>
<p>But the differences are deeper as the two sets only partially overlap:</p>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb376-1"><a href="PreProcTop.html#cb376-1"></a><span class="kw">table</span>(reasons<span class="op">$</span>discard, batch.reasons<span class="op">$</span>discard)</span></code></pre></div>
<pre><code>##        
##         FALSE  TRUE
##   FALSE 34880   348
##   TRUE    805  2095</code></pre>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="PreProcTop.html#cb378-1"></a><span class="kw">rm</span>(reasons, batch.reasons)</span></code></pre></div>
<p>Library size:</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="PreProcTop.html#cb379-1"></a><span class="kw">plotColData</span>(sce,</span>
<span id="cb379-2"><a href="PreProcTop.html#cb379-2"></a>        <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;sum&quot;</span>,</span>
<span id="cb379-3"><a href="PreProcTop.html#cb379-3"></a>        <span class="dt">colour_by=</span><span class="st">&quot;discard&quot;</span>,</span>
<span id="cb379-4"><a href="PreProcTop.html#cb379-4"></a>        <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span></span>
<span id="cb379-5"><a href="PreProcTop.html#cb379-5"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Total count&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Number of genes detected:</p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="PreProcTop.html#cb380-1"></a><span class="kw">plotColData</span>(sce,</span>
<span id="cb380-2"><a href="PreProcTop.html#cb380-2"></a>            <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>,</span>
<span id="cb380-3"><a href="PreProcTop.html#cb380-3"></a>            <span class="dt">colour_by=</span><span class="st">&quot;discard&quot;</span>, </span>
<span id="cb380-4"><a href="PreProcTop.html#cb380-4"></a>            <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span></span>
<span id="cb380-5"><a href="PreProcTop.html#cb380-5"></a><span class="st">  </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detected features&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="PreProcTop.html#cb381-1"></a><span class="kw">plotColData</span>(sce,</span>
<span id="cb381-2"><a href="PreProcTop.html#cb381-2"></a>            <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>,</span>
<span id="cb381-3"><a href="PreProcTop.html#cb381-3"></a>            <span class="dt">colour_by=</span><span class="st">&quot;discard&quot;</span>,</span>
<span id="cb381-4"><a href="PreProcTop.html#cb381-4"></a>            <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span></span>
<span id="cb381-5"><a href="PreProcTop.html#cb381-5"></a><span class="st">          </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by) <span class="op">+</span><span class="st"> </span></span>
<span id="cb381-6"><a href="PreProcTop.html#cb381-6"></a><span class="st">        </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detected features&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Mitochondrial content:</p>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="PreProcTop.html#cb382-1"></a><span class="kw">plotColData</span>(sce,</span>
<span id="cb382-2"><a href="PreProcTop.html#cb382-2"></a>            <span class="dt">x=</span><span class="st">&quot;block&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>,</span>
<span id="cb382-3"><a href="PreProcTop.html#cb382-3"></a>            <span class="dt">colour_by=</span><span class="st">&quot;discard&quot;</span>,</span>
<span id="cb382-4"><a href="PreProcTop.html#cb382-4"></a>            <span class="dt">other_fields=</span><span class="st">&quot;setName&quot;</span>) <span class="op">+</span></span>
<span id="cb382-5"><a href="PreProcTop.html#cb382-5"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Mito percent&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<div id="identify-poor-quality-batches-1" class="section level3">
<h3><span class="header-section-number">5.12.1</span> Identify poor-quality batches</h3>
<p>We will now consider the ‘sample’ batch to illustrate how to identify batches
with overall low quality or different from other batches. Let’s compare
thresholds across sample groups.</p>
<div id="number-of-genes-detected-1" class="section level4">
<h4><span class="header-section-number">5.12.1.1</span> Number of genes detected</h4>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="PreProcTop.html#cb383-1"></a><span class="co"># compute</span></span>
<span id="cb383-2"><a href="PreProcTop.html#cb383-2"></a>discard.nexprs &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>detected,</span>
<span id="cb383-3"><a href="PreProcTop.html#cb383-3"></a>                <span class="dt">log=</span><span class="ot">TRUE</span>,</span>
<span id="cb383-4"><a href="PreProcTop.html#cb383-4"></a>                <span class="dt">type=</span><span class="st">&quot;lower&quot;</span>,</span>
<span id="cb383-5"><a href="PreProcTop.html#cb383-5"></a>                <span class="dt">batch=</span>sce<span class="op">$</span>Sample.Name)</span>
<span id="cb383-6"><a href="PreProcTop.html#cb383-6"></a>nexprs.thresholds &lt;-<span class="st"> </span><span class="kw">attr</span>(discard.nexprs, <span class="st">&quot;thresholds&quot;</span>)[<span class="st">&quot;lower&quot;</span>,]</span>
<span id="cb383-7"><a href="PreProcTop.html#cb383-7"></a>nexprs.thresholds <span class="op">%&gt;%</span></span>
<span id="cb383-8"><a href="PreProcTop.html#cb383-8"></a><span class="st">  </span><span class="kw">round</span>(<span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb383-9"><a href="PreProcTop.html#cb383-9"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb383-10"><a href="PreProcTop.html#cb383-10"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div id="htmlwidget-f8a66a8ab38f2df19fe2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f8a66a8ab38f2df19fe2">{"x":{"filter":"none","data":[["GSM3872434","GSM3872435","GSM3872436","GSM3872437","GSM3872438","GSM3872439","GSM3872440","GSM3872441","GSM3872442","GSM3872443","GSM3872444","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],[603,334,326,127,174,571,191,275,71,132,238,430,402,399,315,244,464,372,357]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Without block:</p>
<div class="sourceCode" id="cb384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb384-1"><a href="PreProcTop.html#cb384-1"></a><span class="co"># plots - without blocking</span></span>
<span id="cb384-2"><a href="PreProcTop.html#cb384-2"></a>discard.nexprs.woBlock &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>detected,</span>
<span id="cb384-3"><a href="PreProcTop.html#cb384-3"></a>                    <span class="dt">log=</span><span class="ot">TRUE</span>,</span>
<span id="cb384-4"><a href="PreProcTop.html#cb384-4"></a>                    <span class="dt">type=</span><span class="st">&quot;lower&quot;</span>)</span>
<span id="cb384-5"><a href="PreProcTop.html#cb384-5"></a>without.blocking &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;Sample.Name&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>,</span>
<span id="cb384-6"><a href="PreProcTop.html#cb384-6"></a>    <span class="dt">colour_by=</span><span class="kw">I</span>(discard.nexprs.woBlock))</span>
<span id="cb384-7"><a href="PreProcTop.html#cb384-7"></a></span>
<span id="cb384-8"><a href="PreProcTop.html#cb384-8"></a>without.blocking <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>With block:</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="PreProcTop.html#cb385-1"></a><span class="co"># plots - with blocking</span></span>
<span id="cb385-2"><a href="PreProcTop.html#cb385-2"></a>with.blocking &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;Sample.Name&quot;</span>, <span class="dt">y=</span><span class="st">&quot;detected&quot;</span>,</span>
<span id="cb385-3"><a href="PreProcTop.html#cb385-3"></a>    <span class="dt">colour_by=</span><span class="kw">I</span>(discard.nexprs))</span>
<span id="cb385-4"><a href="PreProcTop.html#cb385-4"></a>with.blocking <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div id="mitochondrial-content-3" class="section level5">
<h5><span class="header-section-number">5.12.1.1.1</span> Mitochondrial content</h5>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="PreProcTop.html#cb386-1"></a>discard.mito &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>subsets_Mito_percent,</span>
<span id="cb386-2"><a href="PreProcTop.html#cb386-2"></a>              <span class="dt">type=</span><span class="st">&quot;higher&quot;</span>,</span>
<span id="cb386-3"><a href="PreProcTop.html#cb386-3"></a>              <span class="dt">batch=</span>sce<span class="op">$</span>Sample.Name)</span>
<span id="cb386-4"><a href="PreProcTop.html#cb386-4"></a>mito.thresholds &lt;-<span class="st"> </span><span class="kw">attr</span>(discard.mito, <span class="st">&quot;thresholds&quot;</span>)[<span class="st">&quot;higher&quot;</span>,]</span>
<span id="cb386-5"><a href="PreProcTop.html#cb386-5"></a>mito.thresholds <span class="op">%&gt;%</span></span>
<span id="cb386-6"><a href="PreProcTop.html#cb386-6"></a><span class="st">  </span><span class="kw">round</span>(<span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb386-7"><a href="PreProcTop.html#cb386-7"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb386-8"><a href="PreProcTop.html#cb386-8"></a><span class="st">    </span><span class="kw">datatable</span>(<span class="dt">rownames =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div id="htmlwidget-420172f2502c8475403d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-420172f2502c8475403d">{"x":{"filter":"none","data":[["GSM3872434","GSM3872435","GSM3872436","GSM3872437","GSM3872438","GSM3872439","GSM3872440","GSM3872441","GSM3872442","GSM3872443","GSM3872444","MantonBM1","MantonBM2","MantonBM3","MantonBM4","MantonBM5","MantonBM6","MantonBM7","MantonBM8"],[13,9,7,14,17,7,9,8,8,13,9,8,7,6,8,7,7,6,7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Without block:</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="PreProcTop.html#cb387-1"></a><span class="co"># plots - without blocking</span></span>
<span id="cb387-2"><a href="PreProcTop.html#cb387-2"></a>discard.mito.woBlock &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(sce<span class="op">$</span>subsets_Mito_percent, <span class="dt">type=</span><span class="st">&quot;higher&quot;</span>)</span>
<span id="cb387-3"><a href="PreProcTop.html#cb387-3"></a>without.blocking &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;Sample.Name&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>,</span>
<span id="cb387-4"><a href="PreProcTop.html#cb387-4"></a>    <span class="dt">colour_by=</span><span class="kw">I</span>(discard.mito.woBlock))</span>
<span id="cb387-5"><a href="PreProcTop.html#cb387-5"></a>without.blocking <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>With block:</p>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="PreProcTop.html#cb388-1"></a><span class="co"># plots - with blocking</span></span>
<span id="cb388-2"><a href="PreProcTop.html#cb388-2"></a>with.blocking &lt;-<span class="st"> </span><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;Sample.Name&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>,</span>
<span id="cb388-3"><a href="PreProcTop.html#cb388-3"></a>    <span class="dt">colour_by=</span><span class="kw">I</span>(discard.mito))</span>
<span id="cb388-4"><a href="PreProcTop.html#cb388-4"></a>with.blocking <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
</div>
<div id="samples-to-check-1" class="section level4">
<h4><span class="header-section-number">5.12.1.2</span> Samples to check</h4>
<p>Names of samples with a ‘low’ threshold for the number of genes detected:</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="PreProcTop.html#cb389-1"></a><span class="co"># names</span></span>
<span id="cb389-2"><a href="PreProcTop.html#cb389-2"></a><span class="kw">names</span>(nexprs.thresholds)[<span class="kw">isOutlier</span>(nexprs.thresholds, <span class="dt">type=</span><span class="st">&quot;lower&quot;</span>)]</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>Names of samples with a ‘high’ threshold for mitocondrial content:</p>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="PreProcTop.html#cb391-1"></a><span class="co"># names</span></span>
<span id="cb391-2"><a href="PreProcTop.html#cb391-2"></a><span class="kw">names</span>(mito.thresholds)[<span class="kw">isOutlier</span>(mito.thresholds, <span class="dt">type=</span><span class="st">&quot;higher&quot;</span>)]</span></code></pre></div>
<pre><code>## [1] &quot;GSM3872434&quot; &quot;GSM3872437&quot; &quot;GSM3872438&quot; &quot;GSM3872443&quot;</code></pre>
</div>
</div>
<div id="qc-metrics-space-1" class="section level3">
<h3><span class="header-section-number">5.12.2</span> QC metrics space</h3>
<p>A similar approach exists to identify outliers using a set of metrics together. We will the same QC metrics as above.</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="PreProcTop.html#cb393-1"></a><span class="co"># slow</span></span>
<span id="cb393-2"><a href="PreProcTop.html#cb393-2"></a>stats &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">log10</span>(sce<span class="op">$</span>sum),</span>
<span id="cb393-3"><a href="PreProcTop.html#cb393-3"></a>           <span class="kw">log10</span>(sce<span class="op">$</span>detected),</span>
<span id="cb393-4"><a href="PreProcTop.html#cb393-4"></a>           sce<span class="op">$</span>subsets_Mito_percent)</span>
<span id="cb393-5"><a href="PreProcTop.html#cb393-5"></a></span>
<span id="cb393-6"><a href="PreProcTop.html#cb393-6"></a><span class="kw">library</span>(robustbase)</span>
<span id="cb393-7"><a href="PreProcTop.html#cb393-7"></a>outlying &lt;-<span class="st"> </span><span class="kw">adjOutlyingness</span>(stats, <span class="dt">only.outlyingness =</span> <span class="ot">TRUE</span>)</span>
<span id="cb393-8"><a href="PreProcTop.html#cb393-8"></a>multi.outlier &lt;-<span class="st"> </span><span class="kw">isOutlier</span>(outlying, <span class="dt">type =</span> <span class="st">&quot;higher&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="PreProcTop.html#cb394-1"></a><span class="kw">summary</span>(multi.outlier)</span></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE 
## logical   34254    3874</code></pre>
<p>Compare with previous filtering:</p>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="PreProcTop.html#cb396-1"></a><span class="kw">table</span>(sce<span class="op">$</span>discard, multi.outlier)</span></code></pre></div>
<pre><code>##        multi.outlier
##         FALSE  TRUE
##   FALSE 33416  2269
##   TRUE    838  1605</code></pre>
</div>
<div id="qc-pca-1" class="section level3">
<h3><span class="header-section-number">5.12.3</span> QC PCA</h3>
<p>One can also perform a principal components analysis (PCA) on cells, based on the column metadata in a SingleCellExperiment object. Here we will only use the library size, the number of genes detected (which is correlated with library size) and the mitochondrial content.</p>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="PreProcTop.html#cb398-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runColDataPCA</span>(sce, <span class="dt">variables=</span><span class="kw">list</span>(</span>
<span id="cb398-2"><a href="PreProcTop.html#cb398-2"></a>    <span class="st">&quot;sum&quot;</span>, <span class="st">&quot;detected&quot;</span>, <span class="st">&quot;subsets_Mito_percent&quot;</span>),</span>
<span id="cb398-3"><a href="PreProcTop.html#cb398-3"></a>    <span class="dt">outliers=</span><span class="ot">TRUE</span>)</span>
<span id="cb398-4"><a href="PreProcTop.html#cb398-4"></a></span>
<span id="cb398-5"><a href="PreProcTop.html#cb398-5"></a><span class="co">#reducedDimNames(sce)</span></span>
<span id="cb398-6"><a href="PreProcTop.html#cb398-6"></a><span class="co">#head(reducedDim(sce))</span></span>
<span id="cb398-7"><a href="PreProcTop.html#cb398-7"></a><span class="co">#head(colData(sce))</span></span>
<span id="cb398-8"><a href="PreProcTop.html#cb398-8"></a></span>
<span id="cb398-9"><a href="PreProcTop.html#cb398-9"></a><span class="co">#p &lt;- plotReducedDim(sce, dimred=&quot;PCA_coldata&quot;, colour_by = &quot;Sample.Name&quot;)</span></span>
<span id="cb398-10"><a href="PreProcTop.html#cb398-10"></a>p &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_coldata&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;outlier&quot;</span>)</span>
<span id="cb398-11"><a href="PreProcTop.html#cb398-11"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>sce<span class="op">$</span>discard)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>Compare with previous filtering:</p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="PreProcTop.html#cb399-1"></a><span class="kw">table</span>(sce<span class="op">$</span>discard, sce<span class="op">$</span>outlier)</span></code></pre></div>
<pre><code>##        
##         FALSE  TRUE
##   FALSE 34997   688
##   TRUE    768  1675</code></pre>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="PreProcTop.html#cb401-1"></a><span class="kw">table</span>(multi.outlier, sce<span class="op">$</span>outlier)</span></code></pre></div>
<pre><code>##              
## multi.outlier FALSE  TRUE
##         FALSE 33631   623
##         TRUE   2134  1740</code></pre>
</div>
<div id="other-diagnostic-plots-1" class="section level3">
<h3><span class="header-section-number">5.12.4</span> Other diagnostic plots</h3>
<p>Mitochondrial content against library size:</p>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="PreProcTop.html#cb403-1"></a><span class="kw">plotColData</span>(sce, <span class="dt">x=</span><span class="st">&quot;sum&quot;</span>, <span class="dt">y=</span><span class="st">&quot;subsets_Mito_percent&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;discard&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="PreProcTop.html#cb404-1"></a>sp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="kw">colData</span>(sce)),</span>
<span id="cb404-2"><a href="PreProcTop.html#cb404-2"></a>         <span class="kw">aes</span>(<span class="dt">x=</span>sum, <span class="dt">y=</span>subsets_Mito_percent, <span class="dt">col=</span>discard)) <span class="op">+</span></span>
<span id="cb404-3"><a href="PreProcTop.html#cb404-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.05</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></span>
<span id="cb404-4"><a href="PreProcTop.html#cb404-4"></a><span class="st">  </span><span class="kw">geom_density_2d</span>(<span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">colour =</span> <span class="st">&quot;blue&quot;</span>) <span class="op">+</span></span>
<span id="cb404-5"><a href="PreProcTop.html#cb404-5"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="dv">1</span>))) <span class="op">+</span></span>
<span id="cb404-6"><a href="PreProcTop.html#cb404-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb404-7"><a href="PreProcTop.html#cb404-7"></a><span class="co">#sp</span></span>
<span id="cb404-8"><a href="PreProcTop.html#cb404-8"></a>ggExtra<span class="op">::</span><span class="kw">ggMarginal</span>(sp)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Mind distributions:</p>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="PreProcTop.html#cb405-1"></a>sp <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>source_name)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="filter-low-quality-cells-out-1" class="section level3">
<h3><span class="header-section-number">5.12.5</span> Filter low-quality cells out</h3>
<p>We will now exclude poor-quality cells.</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="PreProcTop.html#cb406-1"></a>scePreQc &lt;-<span class="st"> </span>sce</span>
<span id="cb406-2"><a href="PreProcTop.html#cb406-2"></a>sce &lt;-<span class="st"> </span>scePreQc[,<span class="op">!</span>scePreQc<span class="op">$</span>discard]</span>
<span id="cb406-3"><a href="PreProcTop.html#cb406-3"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 33538 35685 
## metadata(20): Samples Samples ... Samples Samples
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(10): ensembl_gene_id external_gene_name ... mean detected
## colnames: NULL
## colData names(14): Barcode Run ... discard outlier
## reducedDimNames(1): PCA_coldata
## altExpNames(0):</code></pre>
<p>We also write the R object to file to use later if need be.</p>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="PreProcTop.html#cb408-1"></a><span class="co"># Write object to file</span></span>
<span id="cb408-2"><a href="PreProcTop.html#cb408-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postQc.Rds&quot;</span>, projDir, outDirBit)</span>
<span id="cb408-3"><a href="PreProcTop.html#cb408-3"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb409-1"><a href="PreProcTop.html#cb409-1"></a><span class="co"># Read object in:</span></span>
<span id="cb409-2"><a href="PreProcTop.html#cb409-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_postQc.Rds&quot;</span>,</span>
<span id="cb409-3"><a href="PreProcTop.html#cb409-3"></a>         projDir, outDirBit)</span>
<span id="cb409-4"><a href="PreProcTop.html#cb409-4"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
</div>
</div>
<div id="novelty-1" class="section level2">
<h2><span class="header-section-number">5.13</span> Novelty</h2>
<p>The number of genes per UMI for each cell informs on the level of sequencing saturation achieved ( <a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionIV/lessons/SC_quality_control_analysis.html">hbctraining</a>). For a given cell, as sequencing depth increases each extra UMI is less likely to correspnf to a gene not already detected in that cell. Cells with small library size tend to have higher overall ‘novelty’ i.e. they have not reached saturation for any given gene. Outlier cell may have a library with low complexity. This may suggest the some cell types, e.g. red blood cells. The expected novelty is about 0.8.</p>
<p>Here we see that some PBMMCs have low novelty, ie overall fewer genes were detected for an equivalent number of UMIs in these cells than in others.</p>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="PreProcTop.html#cb410-1"></a>p &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb410-2"><a href="PreProcTop.html#cb410-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb410-3"><a href="PreProcTop.html#cb410-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>sum, <span class="dt">y=</span>detected, <span class="dt">color=</span>subsets_Mito_percent)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb410-4"><a href="PreProcTop.html#cb410-4"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb410-5"><a href="PreProcTop.html#cb410-5"></a><span class="st">  </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span>lm) <span class="op">+</span></span>
<span id="cb410-6"><a href="PreProcTop.html#cb410-6"></a><span class="st">  </span><span class="kw">scale_x_log10</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb410-7"><a href="PreProcTop.html#cb410-7"></a><span class="st">  </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb410-8"><a href="PreProcTop.html#cb410-8"></a><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="dv">800</span>) <span class="op">+</span></span>
<span id="cb410-9"><a href="PreProcTop.html#cb410-9"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>source_name)</span>
<span id="cb410-10"><a href="PreProcTop.html#cb410-10"></a>p</span></code></pre></div>
<p><img src="preProc_files/figure-html/novelty_scat-1.png" width="672" /></p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="PreProcTop.html#cb411-1"></a><span class="co"># write plot to file</span></span>
<span id="cb411-2"><a href="PreProcTop.html#cb411-2"></a></span>
<span id="cb411-3"><a href="PreProcTop.html#cb411-3"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/novelty_scat.png&quot;</span>,</span>
<span id="cb411-4"><a href="PreProcTop.html#cb411-4"></a>         projDir, outDirBit, qcPlotDirBit)</span>
<span id="cb411-5"><a href="PreProcTop.html#cb411-5"></a><span class="kw">ggsave</span>(<span class="dt">plot=</span>p, <span class="dt">file=</span>tmpFn, <span class="dt">type =</span> <span class="st">&quot;cairo-png&quot;</span>)</span>
<span id="cb411-6"><a href="PreProcTop.html#cb411-6"></a></span>
<span id="cb411-7"><a href="PreProcTop.html#cb411-7"></a><span class="co"># Novelty</span></span>
<span id="cb411-8"><a href="PreProcTop.html#cb411-8"></a><span class="co"># the number of genes per UMI for each cell,</span></span>
<span id="cb411-9"><a href="PreProcTop.html#cb411-9"></a><span class="co"># https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionIV/lessons/SC_quality_control_analysis.html</span></span>
<span id="cb411-10"><a href="PreProcTop.html#cb411-10"></a></span>
<span id="cb411-11"><a href="PreProcTop.html#cb411-11"></a><span class="co"># Add number of UMIs per gene for each cell to metadata</span></span>
<span id="cb411-12"><a href="PreProcTop.html#cb411-12"></a><span class="kw">colData</span>(sce)<span class="op">$</span>log10GenesPerUMI &lt;-<span class="st"> </span><span class="kw">log10</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>detected) <span class="op">/</span><span class="st"> </span><span class="kw">log10</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>sum)</span></code></pre></div>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="PreProcTop.html#cb412-1"></a><span class="co"># Visualize the overall novelty of the gene expression by visualizing the genes detected per UMI</span></span>
<span id="cb412-2"><a href="PreProcTop.html#cb412-2"></a>p &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb412-3"><a href="PreProcTop.html#cb412-3"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb412-4"><a href="PreProcTop.html#cb412-4"></a><span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>log10GenesPerUMI, <span class="dt">color =</span> source_name, <span class="dt">fill =</span> source_name)) <span class="op">+</span></span>
<span id="cb412-5"><a href="PreProcTop.html#cb412-5"></a><span class="st">        </span><span class="kw">geom_density</span>()</span>
<span id="cb412-6"><a href="PreProcTop.html#cb412-6"></a>p</span></code></pre></div>
<p><img src="preProc_files/figure-html/novelty_dens-1.png" width="672" /></p>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="PreProcTop.html#cb413-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/novelty_dens.png&quot;</span>,</span>
<span id="cb413-2"><a href="PreProcTop.html#cb413-2"></a>         projDir, outDirBit, qcPlotDirBit)</span>
<span id="cb413-3"><a href="PreProcTop.html#cb413-3"></a><span class="kw">ggsave</span>(<span class="dt">plot=</span>p, <span class="dt">file=</span>tmpFn, <span class="dt">type=</span><span class="st">&#39;cairo-png&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="PreProcTop.html#cb414-1"></a><span class="kw">rm</span>(sce)</span></code></pre></div>
</div>
<div id="qc-based-on-sparsity-1" class="section level2">
<h2><span class="header-section-number">5.14</span> QC based on sparsity</h2>
<p>The approach above identified poor-quality using thresholds on the number of genes detected and mitochondrial content. We will here specifically look at the sparsity of the data, both at the gene and cell levels.</p>
<div id="remove-genes-that-are-not-expressed-at-all-1" class="section level3">
<h3><span class="header-section-number">5.14.1</span> Remove genes that are not expressed at all</h3>
<p>Genes that are not expressed at all are not informative, so we remove them:</p>
<div class="sourceCode" id="cb415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb415-1"><a href="PreProcTop.html#cb415-1"></a>not.expressed &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(scePreQc)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb415-2"><a href="PreProcTop.html#cb415-2"></a></span>
<span id="cb415-3"><a href="PreProcTop.html#cb415-3"></a><span class="co"># store the cell-wise information</span></span>
<span id="cb415-4"><a href="PreProcTop.html#cb415-4"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(scePreQc)</span>
<span id="cb415-5"><a href="PreProcTop.html#cb415-5"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(scePreQc)</span>
<span id="cb415-6"><a href="PreProcTop.html#cb415-6"></a></span>
<span id="cb415-7"><a href="PreProcTop.html#cb415-7"></a>nz.counts &lt;-<span class="st"> </span><span class="kw">counts</span>(scePreQc)[<span class="op">!</span>not.expressed, ]</span>
<span id="cb415-8"><a href="PreProcTop.html#cb415-8"></a>sce.nz &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span>nz.counts))</span>
<span id="cb415-9"><a href="PreProcTop.html#cb415-9"></a></span>
<span id="cb415-10"><a href="PreProcTop.html#cb415-10"></a><span class="co"># reset the column data on the new object</span></span>
<span id="cb415-11"><a href="PreProcTop.html#cb415-11"></a><span class="kw">colData</span>(sce.nz) &lt;-<span class="st"> </span>cols.meta</span>
<span id="cb415-12"><a href="PreProcTop.html#cb415-12"></a><span class="kw">rowData</span>(sce.nz) &lt;-<span class="st"> </span>rows.meta[<span class="op">!</span>not.expressed, ]</span>
<span id="cb415-13"><a href="PreProcTop.html#cb415-13"></a></span>
<span id="cb415-14"><a href="PreProcTop.html#cb415-14"></a>sce.nz</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 25116 38128 
## metadata(0):
## assays(1): counts
## rownames(25116): ENSG00000238009 ENSG00000239945 ... ENSG00000275063
##   ENSG00000271254
## rowData names(10): ensembl_gene_id external_gene_name ... mean detected
## colnames: NULL
## colData names(14): Barcode Run ... discard outlier
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="PreProcTop.html#cb417-1"></a><span class="co"># Write object to file</span></span>
<span id="cb417-2"><a href="PreProcTop.html#cb417-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz.Rds&quot;</span>,</span>
<span id="cb417-3"><a href="PreProcTop.html#cb417-3"></a>         projDir, outDirBit)</span>
<span id="cb417-4"><a href="PreProcTop.html#cb417-4"></a><span class="kw">saveRDS</span>(sce.nz, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="PreProcTop.html#cb418-1"></a><span class="co"># Write object to file</span></span>
<span id="cb418-2"><a href="PreProcTop.html#cb418-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz.Rds&quot;</span>,</span>
<span id="cb418-3"><a href="PreProcTop.html#cb418-3"></a>         projDir, outDirBit)</span>
<span id="cb418-4"><a href="PreProcTop.html#cb418-4"></a>sce.nz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
</div>
<div id="sparsity-plots-1" class="section level3">
<h3><span class="header-section-number">5.14.2</span> Sparsity plots</h3>
<p>We will compute:</p>
<ul>
<li>the cell sparsity: for each cell, the proportion of genes that are not detected</li>
<li>the gene sparsity: for each gene, the proportion of cells in which it is not detected</li>
</ul>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="PreProcTop.html#cb419-1"></a><span class="co"># compute - SLOW</span></span>
<span id="cb419-2"><a href="PreProcTop.html#cb419-2"></a>cell_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.nz) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">2</span>, sum)<span class="op">/</span><span class="kw">nrow</span>(<span class="kw">counts</span>(sce.nz))</span>
<span id="cb419-3"><a href="PreProcTop.html#cb419-3"></a>gene_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.nz) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">1</span>, sum)<span class="op">/</span><span class="kw">ncol</span>(<span class="kw">counts</span>(sce.nz))</span>
<span id="cb419-4"><a href="PreProcTop.html#cb419-4"></a></span>
<span id="cb419-5"><a href="PreProcTop.html#cb419-5"></a><span class="kw">colData</span>(sce.nz)<span class="op">$</span>cell_sparsity &lt;-<span class="st"> </span>cell_sparsity</span>
<span id="cb419-6"><a href="PreProcTop.html#cb419-6"></a><span class="kw">rowData</span>(sce.nz)<span class="op">$</span>gene_sparsity &lt;-<span class="st"> </span>gene_sparsity</span>
<span id="cb419-7"><a href="PreProcTop.html#cb419-7"></a></span>
<span id="cb419-8"><a href="PreProcTop.html#cb419-8"></a><span class="co"># write outcome to file for later use</span></span>
<span id="cb419-9"><a href="PreProcTop.html#cb419-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb419-10"><a href="PreProcTop.html#cb419-10"></a>         projDir, outDirBit)</span>
<span id="cb419-11"><a href="PreProcTop.html#cb419-11"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;colData&quot;</span> =<span class="st"> </span><span class="kw">colData</span>(sce.nz),</span>
<span id="cb419-12"><a href="PreProcTop.html#cb419-12"></a>         <span class="st">&quot;rowData&quot;</span> =<span class="st"> </span><span class="kw">rowData</span>(sce.nz)),</span>
<span id="cb419-13"><a href="PreProcTop.html#cb419-13"></a>    tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="PreProcTop.html#cb420-1"></a><span class="co"># Read object in:</span></span>
<span id="cb420-2"><a href="PreProcTop.html#cb420-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb420-3"><a href="PreProcTop.html#cb420-3"></a>         projDir, outDirBit)</span>
<span id="cb420-4"><a href="PreProcTop.html#cb420-4"></a>tmpList &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb420-5"><a href="PreProcTop.html#cb420-5"></a>cell_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>colData<span class="op">$</span>cell_sparsity</span>
<span id="cb420-6"><a href="PreProcTop.html#cb420-6"></a>gene_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>rowData<span class="op">$</span>gene_sparsity</span></code></pre></div>
<p>We now plot the distribution of these two metrics.</p>
<p>The cell sparsity plot shows that cells have between 85% and 99% 0’s, which is typical.</p>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="PreProcTop.html#cb421-1"></a><span class="kw">hist</span>(cell_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Cell sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>The gene sparsity plot shows that a large number of genes are almost never detected, which is also regularly observed.</p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="PreProcTop.html#cb422-1"></a><span class="kw">hist</span>(gene_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Gene sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
<div id="filters-1" class="section level3">
<h3><span class="header-section-number">5.14.3</span> Filters</h3>
<p>We also remove cells with sparsity higher than 0.99, and/or mitochondrial content higher than 20%.</p>
<p>Genes detected in a few cells only are unlikely to be informative and would hinder normalisation. We will remove genes that are expressed in fewer than 20 cells.</p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="PreProcTop.html#cb423-1"></a><span class="co"># filter</span></span>
<span id="cb423-2"><a href="PreProcTop.html#cb423-2"></a>sparse.cells &lt;-<span class="st"> </span>cell_sparsity <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.99</span></span>
<span id="cb423-3"><a href="PreProcTop.html#cb423-3"></a>mito.cells &lt;-<span class="st"> </span>sce.nz<span class="op">$</span>subsets_Mito_percent <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span></span>
<span id="cb423-4"><a href="PreProcTop.html#cb423-4"></a></span>
<span id="cb423-5"><a href="PreProcTop.html#cb423-5"></a>min.cells &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>(<span class="dv">20</span><span class="op">/</span><span class="kw">length</span>(cell_sparsity))</span>
<span id="cb423-6"><a href="PreProcTop.html#cb423-6"></a>sparse.genes &lt;-<span class="st"> </span>gene_sparsity <span class="op">&gt;</span><span class="st"> </span>min.cells</span></code></pre></div>
<p>Number of genes removed:</p>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="PreProcTop.html#cb424-1"></a><span class="kw">table</span>(sparse.genes)</span></code></pre></div>
<pre><code>## sparse.genes
## FALSE  TRUE 
## 17854  7262</code></pre>
<p>Number of cells removed:</p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="PreProcTop.html#cb426-1"></a><span class="kw">table</span>(sparse.cells, mito.cells)</span></code></pre></div>
<pre><code>##             mito.cells
## sparse.cells FALSE  TRUE
##        FALSE 36871   671
##        TRUE    525    61</code></pre>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="PreProcTop.html#cb428-1"></a><span class="co"># remove cells from the SCE object that are poor quality</span></span>
<span id="cb428-2"><a href="PreProcTop.html#cb428-2"></a><span class="co"># remove the sparse genes, then re-set the counts and row data accordingly</span></span>
<span id="cb428-3"><a href="PreProcTop.html#cb428-3"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.nz)</span>
<span id="cb428-4"><a href="PreProcTop.html#cb428-4"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce.nz)</span>
<span id="cb428-5"><a href="PreProcTop.html#cb428-5"></a></span>
<span id="cb428-6"><a href="PreProcTop.html#cb428-6"></a>counts.nz &lt;-<span class="st"> </span><span class="kw">counts</span>(sce.nz)[<span class="op">!</span>sparse.genes, <span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells)]</span>
<span id="cb428-7"><a href="PreProcTop.html#cb428-7"></a>sce.nz &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="dt">assays=</span><span class="kw">list</span>(<span class="dt">counts=</span>counts.nz))</span>
<span id="cb428-8"><a href="PreProcTop.html#cb428-8"></a><span class="kw">colData</span>(sce.nz) &lt;-<span class="st"> </span>cols.meta[<span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells),]</span>
<span id="cb428-9"><a href="PreProcTop.html#cb428-9"></a><span class="kw">rowData</span>(sce.nz) &lt;-<span class="st"> </span>rows.meta[<span class="op">!</span>sparse.genes, ]</span>
<span id="cb428-10"><a href="PreProcTop.html#cb428-10"></a>sce.nz</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 17854 36871 
## metadata(0):
## assays(1): counts
## rownames(17854): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(15): Barcode Run ... outlier cell_sparsity
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="PreProcTop.html#cb430-1"></a><span class="co"># Write object to file</span></span>
<span id="cb430-2"><a href="PreProcTop.html#cb430-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz_postQc.Rds&quot;</span>,</span>
<span id="cb430-3"><a href="PreProcTop.html#cb430-3"></a>         projDir, outDirBit)</span>
<span id="cb430-4"><a href="PreProcTop.html#cb430-4"></a><span class="kw">saveRDS</span>(sce.nz, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="PreProcTop.html#cb431-1"></a><span class="co"># Read object in:</span></span>
<span id="cb431-2"><a href="PreProcTop.html#cb431-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/sce_nz_postQc.Rds&quot;</span>,</span>
<span id="cb431-3"><a href="PreProcTop.html#cb431-3"></a>         projDir, outDirBit)</span>
<span id="cb431-4"><a href="PreProcTop.html#cb431-4"></a>sce.nz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
<p>Compare with filter above (mind that the comparison is not fair because we used a less stringent, hard filtering on mitochondrial content):</p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="PreProcTop.html#cb432-1"></a><span class="kw">table</span>(scePreQc<span class="op">$</span>discard, (sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells))</span></code></pre></div>
<pre><code>##        
##         FALSE  TRUE
##   FALSE 35351   334
##   TRUE   1520   923</code></pre>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="PreProcTop.html#cb434-1"></a><span class="kw">rm</span>(scePreQc)</span></code></pre></div>
</div>
<div id="separate-caron-and-hca-batches-1" class="section level3">
<h3><span class="header-section-number">5.14.4</span> Separate Caron and Hca batches</h3>
<p>We will now check sparsity for each batch separately.</p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="PreProcTop.html#cb435-1"></a>sce.nz.caron &lt;-<span class="st"> </span>sce.nz[,sce.nz<span class="op">$</span>setName<span class="op">==</span><span class="st">&quot;Caron&quot;</span>]</span>
<span id="cb435-2"><a href="PreProcTop.html#cb435-2"></a>sce.nz.hca &lt;-<span class="st"> </span>sce.nz[,sce.nz<span class="op">$</span>setName<span class="op">==</span><span class="st">&quot;Hca&quot;</span>]</span></code></pre></div>
</div>
<div id="caron-only-1" class="section level3">
<h3><span class="header-section-number">5.14.5</span> Caron only</h3>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="PreProcTop.html#cb436-1"></a>setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb436-2"><a href="PreProcTop.html#cb436-2"></a>sce.x &lt;-<span class="st"> </span>sce.nz.caron</span>
<span id="cb436-3"><a href="PreProcTop.html#cb436-3"></a><span class="kw">rm</span>(sce.nz.caron)</span></code></pre></div>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="PreProcTop.html#cb437-1"></a><span class="co"># compute - SLOW</span></span>
<span id="cb437-2"><a href="PreProcTop.html#cb437-2"></a>cell_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.x) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">2</span>, sum)<span class="op">/</span><span class="kw">nrow</span>(<span class="kw">counts</span>(sce.x))</span>
<span id="cb437-3"><a href="PreProcTop.html#cb437-3"></a>gene_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.x) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">1</span>, sum)<span class="op">/</span><span class="kw">ncol</span>(<span class="kw">counts</span>(sce.x))</span>
<span id="cb437-4"><a href="PreProcTop.html#cb437-4"></a></span>
<span id="cb437-5"><a href="PreProcTop.html#cb437-5"></a><span class="kw">colData</span>(sce.x)<span class="op">$</span>cell_sparsity &lt;-<span class="st"> </span>cell_sparsity</span>
<span id="cb437-6"><a href="PreProcTop.html#cb437-6"></a><span class="kw">rowData</span>(sce.x)<span class="op">$</span>gene_sparsity &lt;-<span class="st"> </span>gene_sparsity</span>
<span id="cb437-7"><a href="PreProcTop.html#cb437-7"></a></span>
<span id="cb437-8"><a href="PreProcTop.html#cb437-8"></a><span class="co"># write outcome to file for later use</span></span>
<span id="cb437-9"><a href="PreProcTop.html#cb437-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb437-10"><a href="PreProcTop.html#cb437-10"></a>         projDir, outDirBit, setName)</span>
<span id="cb437-11"><a href="PreProcTop.html#cb437-11"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;colData&quot;</span> =<span class="st"> </span><span class="kw">colData</span>(sce.x),</span>
<span id="cb437-12"><a href="PreProcTop.html#cb437-12"></a>         <span class="st">&quot;rowData&quot;</span> =<span class="st"> </span><span class="kw">rowData</span>(sce.x)),</span>
<span id="cb437-13"><a href="PreProcTop.html#cb437-13"></a>    tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="PreProcTop.html#cb438-1"></a><span class="co"># Read object in:</span></span>
<span id="cb438-2"><a href="PreProcTop.html#cb438-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb438-3"><a href="PreProcTop.html#cb438-3"></a>         projDir, outDirBit, setName)</span>
<span id="cb438-4"><a href="PreProcTop.html#cb438-4"></a>tmpList &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb438-5"><a href="PreProcTop.html#cb438-5"></a>cell_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>colData<span class="op">$</span>cell_sparsity</span>
<span id="cb438-6"><a href="PreProcTop.html#cb438-6"></a>gene_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>rowData<span class="op">$</span>gene_sparsity</span></code></pre></div>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="PreProcTop.html#cb439-1"></a><span class="co"># plot</span></span>
<span id="cb439-2"><a href="PreProcTop.html#cb439-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sparsity.png&quot;</span>,</span>
<span id="cb439-3"><a href="PreProcTop.html#cb439-3"></a>         projDir, outDirBit, qcPlotDirBit, setName)</span>
<span id="cb439-4"><a href="PreProcTop.html#cb439-4"></a><span class="kw">CairoPNG</span>(tmpFn)</span>
<span id="cb439-5"><a href="PreProcTop.html#cb439-5"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb439-6"><a href="PreProcTop.html#cb439-6"></a><span class="kw">hist</span>(cell_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Cell sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb439-7"><a href="PreProcTop.html#cb439-7"></a><span class="kw">hist</span>(gene_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Gene sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb439-8"><a href="PreProcTop.html#cb439-8"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">40</span>, <span class="dt">lty=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">&#39;purple&#39;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/Caron_sparsity_plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="PreProcTop.html#cb440-1"></a><span class="kw">dev.off</span>()</span></code></pre></div>
<pre><code>## pdf 
##   2</code></pre>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="PreProcTop.html#cb442-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;../%s/%s_sparsity.png&quot;</span>,</span>
<span id="cb442-2"><a href="PreProcTop.html#cb442-2"></a>         qcPlotDirBit, setName)</span>
<span id="cb442-3"><a href="PreProcTop.html#cb442-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Qc/caron_sparsity.png" width="80%" /></p>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="PreProcTop.html#cb443-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="PreProcTop.html#cb444-1"></a><span class="co"># filter</span></span>
<span id="cb444-2"><a href="PreProcTop.html#cb444-2"></a>sparse.cells &lt;-<span class="st"> </span>cell_sparsity <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.99</span></span>
<span id="cb444-3"><a href="PreProcTop.html#cb444-3"></a>mito.cells &lt;-<span class="st"> </span>sce.x<span class="op">$</span>subsets_Mito_percent <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span></span>
<span id="cb444-4"><a href="PreProcTop.html#cb444-4"></a></span>
<span id="cb444-5"><a href="PreProcTop.html#cb444-5"></a>min.cells &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>(<span class="dv">20</span><span class="op">/</span><span class="kw">length</span>(cell_sparsity))</span>
<span id="cb444-6"><a href="PreProcTop.html#cb444-6"></a>sparse.genes &lt;-<span class="st"> </span>gene_sparsity <span class="op">&gt;</span><span class="st"> </span>min.cells</span></code></pre></div>
<p>We write the R object to ‘caron_sce_nz_postQc.Rds’.</p>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="PreProcTop.html#cb445-1"></a><span class="co"># remove cells from the SCE object that are poor quality</span></span>
<span id="cb445-2"><a href="PreProcTop.html#cb445-2"></a><span class="co"># remove the sparse genes, then re-set the counts and row data accordingly</span></span>
<span id="cb445-3"><a href="PreProcTop.html#cb445-3"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.x)</span>
<span id="cb445-4"><a href="PreProcTop.html#cb445-4"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce.x)</span>
<span id="cb445-5"><a href="PreProcTop.html#cb445-5"></a></span>
<span id="cb445-6"><a href="PreProcTop.html#cb445-6"></a>counts.x &lt;-<span class="st"> </span><span class="kw">counts</span>(sce.x)[<span class="op">!</span>sparse.genes, <span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells)]</span>
<span id="cb445-7"><a href="PreProcTop.html#cb445-7"></a>sce.x &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="dt">assays=</span><span class="kw">list</span>(<span class="dt">counts=</span>counts.x))</span>
<span id="cb445-8"><a href="PreProcTop.html#cb445-8"></a><span class="kw">colData</span>(sce.x) &lt;-<span class="st"> </span>cols.meta[<span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells),]</span>
<span id="cb445-9"><a href="PreProcTop.html#cb445-9"></a><span class="kw">rowData</span>(sce.x) &lt;-<span class="st"> </span>rows.meta[<span class="op">!</span>sparse.genes, ]</span>
<span id="cb445-10"><a href="PreProcTop.html#cb445-10"></a>sce.x</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 16629 20908 
## metadata(0):
## assays(1): counts
## rownames(16629): ENSG00000237491 ENSG00000225880 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(15): Barcode Run ... outlier cell_sparsity
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="PreProcTop.html#cb447-1"></a><span class="co"># Write object to file</span></span>
<span id="cb447-2"><a href="PreProcTop.html#cb447-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc.Rds&quot;</span>, projDir, outDirBit, setName)</span>
<span id="cb447-3"><a href="PreProcTop.html#cb447-3"></a><span class="kw">saveRDS</span>(sce.x, tmpFn)</span>
<span id="cb447-4"><a href="PreProcTop.html#cb447-4"></a></span>
<span id="cb447-5"><a href="PreProcTop.html#cb447-5"></a><span class="co"># update sce.nz.caron</span></span>
<span id="cb447-6"><a href="PreProcTop.html#cb447-6"></a>sce.nz.caron &lt;-<span class="st"> </span>sce.x  <span class="co"># TODO not used</span></span>
<span id="cb447-7"><a href="PreProcTop.html#cb447-7"></a><span class="kw">rm</span>(sce.x)</span></code></pre></div>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="PreProcTop.html#cb448-1"></a><span class="co"># Read object in:</span></span>
<span id="cb448-2"><a href="PreProcTop.html#cb448-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc.Rds&quot;</span>,</span>
<span id="cb448-3"><a href="PreProcTop.html#cb448-3"></a>         dirRel, outDirBit, setName)</span>
<span id="cb448-4"><a href="PreProcTop.html#cb448-4"></a>sce.nz.caron &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
</div>
<div id="hca-only-1" class="section level3">
<h3><span class="header-section-number">5.14.6</span> Hca only</h3>
<div class="sourceCode" id="cb449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb449-1"><a href="PreProcTop.html#cb449-1"></a>setName &lt;-<span class="st"> &quot;hca&quot;</span></span>
<span id="cb449-2"><a href="PreProcTop.html#cb449-2"></a>sce.x &lt;-<span class="st"> </span>sce.nz.hca</span>
<span id="cb449-3"><a href="PreProcTop.html#cb449-3"></a><span class="kw">rm</span>(sce.nz.hca)</span></code></pre></div>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="PreProcTop.html#cb450-1"></a><span class="co"># compute - SLOW</span></span>
<span id="cb450-2"><a href="PreProcTop.html#cb450-2"></a>cell_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.x) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">2</span>, sum)<span class="op">/</span><span class="kw">nrow</span>(<span class="kw">counts</span>(sce.x))</span>
<span id="cb450-3"><a href="PreProcTop.html#cb450-3"></a>gene_sparsity &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(sce.x) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">1</span>, sum)<span class="op">/</span><span class="kw">ncol</span>(<span class="kw">counts</span>(sce.x))</span>
<span id="cb450-4"><a href="PreProcTop.html#cb450-4"></a></span>
<span id="cb450-5"><a href="PreProcTop.html#cb450-5"></a><span class="kw">colData</span>(sce.x)<span class="op">$</span>cell_sparsity &lt;-<span class="st"> </span>cell_sparsity</span>
<span id="cb450-6"><a href="PreProcTop.html#cb450-6"></a><span class="kw">rowData</span>(sce.x)<span class="op">$</span>gene_sparsity &lt;-<span class="st"> </span>gene_sparsity</span>
<span id="cb450-7"><a href="PreProcTop.html#cb450-7"></a></span>
<span id="cb450-8"><a href="PreProcTop.html#cb450-8"></a><span class="co"># write outcome to file for later use</span></span>
<span id="cb450-9"><a href="PreProcTop.html#cb450-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb450-10"><a href="PreProcTop.html#cb450-10"></a>         projDir, outDirBit, setName)</span>
<span id="cb450-11"><a href="PreProcTop.html#cb450-11"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;colData&quot;</span> =<span class="st"> </span><span class="kw">colData</span>(sce.x),</span>
<span id="cb450-12"><a href="PreProcTop.html#cb450-12"></a>         <span class="st">&quot;rowData&quot;</span> =<span class="st"> </span><span class="kw">rowData</span>(sce.x)),</span>
<span id="cb450-13"><a href="PreProcTop.html#cb450-13"></a>    tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="PreProcTop.html#cb451-1"></a><span class="co"># Read object in:</span></span>
<span id="cb451-2"><a href="PreProcTop.html#cb451-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_sparsityCellGene.Rds&quot;</span>,</span>
<span id="cb451-3"><a href="PreProcTop.html#cb451-3"></a>         projDir, outDirBit, setName)</span>
<span id="cb451-4"><a href="PreProcTop.html#cb451-4"></a>tmpList &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb451-5"><a href="PreProcTop.html#cb451-5"></a>cell_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>colData<span class="op">$</span>cell_sparsity</span>
<span id="cb451-6"><a href="PreProcTop.html#cb451-6"></a>gene_sparsity &lt;-<span class="st"> </span>tmpList<span class="op">$</span>rowData<span class="op">$</span>gene_sparsity</span></code></pre></div>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="PreProcTop.html#cb452-1"></a><span class="co"># plot</span></span>
<span id="cb452-2"><a href="PreProcTop.html#cb452-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sparsity.png&quot;</span>,</span>
<span id="cb452-3"><a href="PreProcTop.html#cb452-3"></a>         projDir, outDirBit, qcPlotDirBit, setName)</span>
<span id="cb452-4"><a href="PreProcTop.html#cb452-4"></a><span class="kw">CairoPNG</span>(tmpFn)</span>
<span id="cb452-5"><a href="PreProcTop.html#cb452-5"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb452-6"><a href="PreProcTop.html#cb452-6"></a><span class="kw">hist</span>(cell_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Cell sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb452-7"><a href="PreProcTop.html#cb452-7"></a><span class="kw">hist</span>(gene_sparsity, <span class="dt">breaks=</span><span class="dv">50</span>, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Gene sparsity&quot;</span>, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb452-8"><a href="PreProcTop.html#cb452-8"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">40</span>, <span class="dt">lty=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">&#39;purple&#39;</span>)</span></code></pre></div>
<p><img src="preProc_files/figure-html/Hca_sparsity_plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="PreProcTop.html#cb453-1"></a><span class="kw">dev.off</span>()</span></code></pre></div>
<pre><code>## pdf 
##   2</code></pre>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="PreProcTop.html#cb455-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;../%s/%s_sparsity.png&quot;</span>,</span>
<span id="cb455-2"><a href="PreProcTop.html#cb455-2"></a>         qcPlotDirBit, setName)</span>
<span id="cb455-3"><a href="PreProcTop.html#cb455-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Qc/hca_sparsity.png" width="80%" /></p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="PreProcTop.html#cb456-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="PreProcTop.html#cb457-1"></a><span class="co"># filter</span></span>
<span id="cb457-2"><a href="PreProcTop.html#cb457-2"></a>sparse.cells &lt;-<span class="st"> </span>cell_sparsity <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.99</span></span>
<span id="cb457-3"><a href="PreProcTop.html#cb457-3"></a>mito.cells &lt;-<span class="st"> </span>sce.x<span class="op">$</span>subsets_Mito_percent <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span></span>
<span id="cb457-4"><a href="PreProcTop.html#cb457-4"></a></span>
<span id="cb457-5"><a href="PreProcTop.html#cb457-5"></a>min.cells &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>(<span class="dv">20</span><span class="op">/</span><span class="kw">length</span>(cell_sparsity))</span>
<span id="cb457-6"><a href="PreProcTop.html#cb457-6"></a>sparse.genes &lt;-<span class="st"> </span>gene_sparsity <span class="op">&gt;</span><span class="st"> </span>min.cells</span></code></pre></div>
<p>We write the R object to ‘hca_sce_nz_postQc.Rds’.</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="PreProcTop.html#cb458-1"></a><span class="co"># remove cells from the SCE object that are poor quality</span></span>
<span id="cb458-2"><a href="PreProcTop.html#cb458-2"></a><span class="co"># remove the sparse genes, then re-set the counts and row data accordingly</span></span>
<span id="cb458-3"><a href="PreProcTop.html#cb458-3"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.x)</span>
<span id="cb458-4"><a href="PreProcTop.html#cb458-4"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce.x)</span>
<span id="cb458-5"><a href="PreProcTop.html#cb458-5"></a></span>
<span id="cb458-6"><a href="PreProcTop.html#cb458-6"></a>counts.x &lt;-<span class="st"> </span><span class="kw">counts</span>(sce.x)[<span class="op">!</span>sparse.genes, <span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells)]</span>
<span id="cb458-7"><a href="PreProcTop.html#cb458-7"></a>sce.x &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="dt">assays=</span><span class="kw">list</span>(<span class="dt">counts=</span>counts.x))</span>
<span id="cb458-8"><a href="PreProcTop.html#cb458-8"></a><span class="kw">colData</span>(sce.x) &lt;-<span class="st"> </span>cols.meta[<span class="op">!</span>(sparse.cells <span class="op">|</span><span class="st"> </span>mito.cells),]</span>
<span id="cb458-9"><a href="PreProcTop.html#cb458-9"></a><span class="kw">rowData</span>(sce.x) &lt;-<span class="st"> </span>rows.meta[<span class="op">!</span>sparse.genes, ]</span>
<span id="cb458-10"><a href="PreProcTop.html#cb458-10"></a>sce.x</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 14994 15963 
## metadata(0):
## assays(1): counts
## rownames(14994): ENSG00000237491 ENSG00000225880 ... ENSG00000276345
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(15): Barcode Run ... outlier cell_sparsity
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="PreProcTop.html#cb460-1"></a><span class="co"># Write object to file</span></span>
<span id="cb460-2"><a href="PreProcTop.html#cb460-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc.Rds&quot;</span>,</span>
<span id="cb460-3"><a href="PreProcTop.html#cb460-3"></a>         projDir, outDirBit, setName)</span>
<span id="cb460-4"><a href="PreProcTop.html#cb460-4"></a><span class="kw">saveRDS</span>(sce.x, tmpFn)</span>
<span id="cb460-5"><a href="PreProcTop.html#cb460-5"></a></span>
<span id="cb460-6"><a href="PreProcTop.html#cb460-6"></a><span class="co"># update sce.nz.hca</span></span>
<span id="cb460-7"><a href="PreProcTop.html#cb460-7"></a>sce.nz.hca &lt;-<span class="st"> </span>sce.x  <span class="co"># TODO not used</span></span>
<span id="cb460-8"><a href="PreProcTop.html#cb460-8"></a><span class="kw">rm</span>(sce.x)</span></code></pre></div>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="PreProcTop.html#cb461-1"></a><span class="co"># Read object in:</span></span>
<span id="cb461-2"><a href="PreProcTop.html#cb461-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc.Rds&quot;</span>,</span>
<span id="cb461-3"><a href="PreProcTop.html#cb461-3"></a>         projDir, outDirBit, setName)</span>
<span id="cb461-4"><a href="PreProcTop.html#cb461-4"></a>sce.nz.hca &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn) <span class="co"># TODO not used</span></span></code></pre></div>
</div>
</div>
<div id="session-information-1" class="section level2">
<h2><span class="header-section-number">5.15</span> Session information</h2>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="PreProcTop.html#cb462-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] robustbase_0.93-7           mixtools_1.2.0             
##  [3] Cairo_1.5-12.2              dplyr_1.0.5                
##  [5] DT_0.18                     irlba_2.3.3                
##  [7] biomaRt_2.46.3              Matrix_1.3-2               
##  [9] igraph_1.2.6                DropletUtils_1.10.3        
## [11] scater_1.18.6               ggplot2_3.3.3              
## [13] scran_1.18.7                SingleCellExperiment_1.12.0
## [15] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [17] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [19] IRanges_2.24.1              S4Vectors_0.28.1           
## [21] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [23] matrixStats_0.58.0          knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] BiocFileCache_1.14.0      splines_4.0.3            
##   [3] BiocParallel_1.24.1       crosstalk_1.1.1          
##   [5] digest_0.6.27             htmltools_0.5.1.1        
##   [7] viridis_0.6.0             fansi_0.4.2              
##   [9] magrittr_2.0.1            memoise_2.0.0            
##  [11] limma_3.46.0              R.utils_2.10.1           
##  [13] askpass_1.1               prettyunits_1.1.1        
##  [15] colorspace_2.0-0          blob_1.2.1               
##  [17] rappdirs_0.3.3            xfun_0.22                
##  [19] crayon_1.4.1              RCurl_1.98-1.3           
##  [21] jsonlite_1.7.2            survival_3.2-11          
##  [23] glue_1.4.2                gtable_0.3.0             
##  [25] zlibbioc_1.36.0           XVector_0.30.0           
##  [27] DelayedArray_0.16.3       BiocSingular_1.6.0       
##  [29] kernlab_0.9-29            Rhdf5lib_1.12.1          
##  [31] DEoptimR_1.0-8            HDF5Array_1.18.1         
##  [33] scales_1.1.1              DBI_1.1.1                
##  [35] edgeR_3.32.1              miniUI_0.1.1.1           
##  [37] Rcpp_1.0.6                isoband_0.2.4            
##  [39] viridisLite_0.4.0         xtable_1.8-4             
##  [41] progress_1.2.2            dqrng_0.3.0              
##  [43] bit_4.0.4                 rsvd_1.0.5               
##  [45] htmlwidgets_1.5.3         httr_1.4.2               
##  [47] ellipsis_0.3.2            pkgconfig_2.0.3          
##  [49] XML_3.99-0.6              R.methodsS3_1.8.1        
##  [51] farver_2.1.0              scuttle_1.0.4            
##  [53] sass_0.3.1                dbplyr_2.1.1             
##  [55] locfit_1.5-9.4            utf8_1.2.1               
##  [57] labeling_0.4.2            tidyselect_1.1.1         
##  [59] rlang_0.4.10              later_1.2.0              
##  [61] AnnotationDbi_1.52.0      munsell_0.5.0            
##  [63] tools_4.0.3               cachem_1.0.4             
##  [65] generics_0.1.0            RSQLite_2.2.6            
##  [67] evaluate_0.14             stringr_1.4.0            
##  [69] fastmap_1.1.0             yaml_2.2.1               
##  [71] bit64_4.0.5               purrr_0.3.4              
##  [73] nlme_3.1-152              sparseMatrixStats_1.2.1  
##  [75] mime_0.10                 R.oo_1.24.0              
##  [77] ggExtra_0.9               xml2_1.3.2               
##  [79] compiler_4.0.3            beeswarm_0.3.1           
##  [81] curl_4.3.1                tibble_3.1.1             
##  [83] statmod_1.4.35            bslib_0.2.4              
##  [85] stringi_1.5.3             highr_0.9                
##  [87] lattice_0.20-44           bluster_1.0.0            
##  [89] vctrs_0.3.7               pillar_1.6.0             
##  [91] lifecycle_1.0.0           rhdf5filters_1.2.0       
##  [93] jquerylib_0.1.3           BiocNeighbors_1.8.2      
##  [95] cowplot_1.1.1             bitops_1.0-7             
##  [97] httpuv_1.5.5              R6_2.5.0                 
##  [99] bookdown_0.22             promises_1.2.0.1         
## [101] gridExtra_2.3             vipor_0.4.5              
## [103] codetools_0.2-18          MASS_7.3-54              
## [105] assertthat_0.2.1          rhdf5_2.34.0             
## [107] openssl_1.4.4             withr_2.4.2              
## [109] GenomeInfoDbData_1.2.4    mgcv_1.8-35              
## [111] hms_1.0.0                 grid_4.0.3               
## [113] beachmat_2.6.4            rmarkdown_2.7            
## [115] DelayedMatrixStats_1.12.3 segmented_1.3-3          
## [117] shiny_1.6.0               ggbeeswarm_0.6.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="PreProcAllCellsTop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="NormalisationAllCellsTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/preProc.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC.pdf", "KmWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
