<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 22 Cell cycle assignment | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 22 Cell cycle assignment | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 22 Cell cycle assignment | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cluster-marker-genes.html"/>
<link rel="next" href="differential-expression-and-abundance-between-conditions.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-11"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-12"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-13"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html"><i class="fa fa-check"></i><b>24</b> Extract T-cells from HCA Dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#variance-modeling"><i class="fa fa-check"></i><b>24.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.2" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#data-integration"><i class="fa fa-check"></i><b>24.2</b> Data integration</a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#clustering-1"><i class="fa fa-check"></i><b>24.4</b> Clustering</a></li>
<li class="chapter" data-level="24.5" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#cell-type-classification"><i class="fa fa-check"></i><b>24.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.6" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#extract-t-cells"><i class="fa fa-check"></i><b>24.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.7" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#ackowledgements"><i class="fa fa-check"></i><b>24.7</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.8" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#session-information-14"><i class="fa fa-check"></i><b>24.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html"><i class="fa fa-check"></i><b>25</b> Setting up the data</a><ul>
<li class="chapter" data-level="25.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>25.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="25.2" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>25.2</b> Find temporally expressed genes</a><ul>
<li class="chapter" data-level="25.2.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>25.2.1</b> Pseudotime analysis for another HCA sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="exercise-1.html"><a href="exercise-1.html"><i class="fa fa-check"></i><b>26</b> Exercise 1</a><ul>
<li class="chapter" data-level="26.1" data-path="exercise-1.html"><a href="exercise-1.html#ackowledgements-1"><i class="fa fa-check"></i><b>26.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="26.2" data-path="exercise-1.html"><a href="exercise-1.html#session-information-15"><i class="fa fa-check"></i><b>26.2</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cellCyclePhasesTop" class="section level1">
<h1><span class="header-section-number">Chapter 22</span> Cell cycle assignment</h1>
<p>Source: <a href="https://osca.bioconductor.org/cell-cycle-assignment.html">Cell cycle assignment</a> of the OSCA book.</p>
<div id="load-data-10" class="section level2">
<h2><span class="header-section-number">22.1</span> Load data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts for 500 cells per sample.</p>
<div class="sourceCode" id="cb1801"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1801-1"><a href="cellCyclePhasesTop.html#cb1801-1"></a><span class="co"># read uncorrected counts</span></span>
<span id="cb1801-2"><a href="cellCyclePhasesTop.html#cb1801-2"></a>fn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_uncorr.Rds&quot;</span>,</span>
<span id="cb1801-3"><a href="cellCyclePhasesTop.html#cb1801-3"></a>              projDir,</span>
<span id="cb1801-4"><a href="cellCyclePhasesTop.html#cb1801-4"></a>              outDirBit,</span>
<span id="cb1801-5"><a href="cellCyclePhasesTop.html#cb1801-5"></a>              setName,</span>
<span id="cb1801-6"><a href="cellCyclePhasesTop.html#cb1801-6"></a>              setSuf,</span>
<span id="cb1801-7"><a href="cellCyclePhasesTop.html#cb1801-7"></a>              dsiSuf, <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1801-8"><a href="cellCyclePhasesTop.html#cb1801-8"></a>              splSetToGet2) </span>
<span id="cb1801-9"><a href="cellCyclePhasesTop.html#cb1801-9"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>fn)</span>
<span id="cb1801-10"><a href="cellCyclePhasesTop.html#cb1801-10"></a></span>
<span id="cb1801-11"><a href="cellCyclePhasesTop.html#cb1801-11"></a><span class="co"># Read object in:</span></span>
<span id="cb1801-12"><a href="cellCyclePhasesTop.html#cb1801-12"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_clust.Rds&quot;</span>,</span>
<span id="cb1801-13"><a href="cellCyclePhasesTop.html#cb1801-13"></a>                 projDir, outDirBit, setName, setSuf)</span>
<span id="cb1801-14"><a href="cellCyclePhasesTop.html#cb1801-14"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_clust.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1803"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1803-1"><a href="cellCyclePhasesTop.html#cb1803-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb1803-2"><a href="cellCyclePhasesTop.html#cb1803-2"></a>{</span>
<span id="cb1803-3"><a href="cellCyclePhasesTop.html#cb1803-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb1803-4"><a href="cellCyclePhasesTop.html#cb1803-4"></a>}</span>
<span id="cb1803-5"><a href="cellCyclePhasesTop.html#cb1803-5"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1803-6"><a href="cellCyclePhasesTop.html#cb1803-6"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1803-7"><a href="cellCyclePhasesTop.html#cb1803-7"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 7371 3500 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(7371): TSPAN6 FGR ... POLR2J3_ENSG00000285437 AC003043.2
## rowData names(12): rotation ensembl_gene_id ... detected gene_sparsity
## colnames: NULL
## colData names(25): Barcode Run ... cluster.out.10 cluster.out.10.col
## reducedDimNames(3): corrected TSNE UMAP
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1805"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1805-1"><a href="cellCyclePhasesTop.html#cb1805-1"></a><span class="co"># copy clustering output to uncorrected SCE</span></span>
<span id="cb1805-2"><a href="cellCyclePhasesTop.html#cb1805-2"></a>sce &lt;-<span class="st"> </span>uncorrected</span>
<span id="cb1805-3"><a href="cellCyclePhasesTop.html#cb1805-3"></a>x &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out)[,<span class="kw">c</span>(<span class="st">&quot;Barcode&quot;</span>,<span class="st">&quot;leiden&quot;</span>)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb1805-4"><a href="cellCyclePhasesTop.html#cb1805-4"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span><span class="kw">colData</span>(uncorrected) <span class="op">%&gt;%</span></span>
<span id="cb1805-5"><a href="cellCyclePhasesTop.html#cb1805-5"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1805-6"><a href="cellCyclePhasesTop.html#cb1805-6"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(x, <span class="dt">by=</span><span class="st">&quot;Barcode&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1805-7"><a href="cellCyclePhasesTop.html#cb1805-7"></a><span class="st">  </span>DataFrame</span>
<span id="cb1805-8"><a href="cellCyclePhasesTop.html#cb1805-8"></a><span class="kw">colData</span>(sce)<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>leiden</span>
<span id="cb1805-9"><a href="cellCyclePhasesTop.html#cb1805-9"></a>sce<span class="op">$</span>clusterStg &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, sce<span class="op">$</span>cluster),</span>
<span id="cb1805-10"><a href="cellCyclePhasesTop.html#cb1805-10"></a>             <span class="dt">levels =</span> <span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, <span class="kw">levels</span>(sce<span class="op">$</span>cluster)))</span>
<span id="cb1805-11"><a href="cellCyclePhasesTop.html#cb1805-11"></a><span class="kw">rm</span>(uncorrected)</span></code></pre></div>
</div>
<div id="motivation-6" class="section level2">
<h2><span class="header-section-number">22.2</span> Motivation</h2>
<p>On occasion, it can be desirable to determine cell cycle activity from scRNA-seq data. In and of itself, the distribution of cells across phases of the cell cycle is not usually informative, but we can use this to determine if there are differences in proliferation between subpopulations or across treatment conditions. Many of the key events in the cell cycle (e.g., passage through checkpoints) are post-translational and thus not directly visible in transcriptomic data; nonetheless, there are enough changes in expression that can be exploited to determine cell cycle phase. We demonstrate using the 416B dataset, which is known to contain actively cycling cells after oncogene induction.</p>
</div>
<div id="using-the-cyclins" class="section level2">
<h2><span class="header-section-number">22.3</span> Using the cyclins</h2>
<p>The cyclins control progression through the cell cycle and have well-characterized patterns of expression across cell cycle phases. Cyclin D is expressed throughout but peaks at G1; cyclin E is expressed highest in the G1/S transition; cyclin A is expressed across S and G2; and cyclin B is expressed highest in late G2 and mitosis. Inspection of the relative expression of cyclins across the population can often be sufficient to determine the relative cell cycle activity in each cluster. For example, cluster 1 is likely to be in G1 while the other clusters are scattered across the later phases.</p>
<div class="sourceCode" id="cb1806"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1806-1"><a href="cellCyclePhasesTop.html#cb1806-1"></a><span class="co">#cyclin.genes &lt;- grep(&quot;^CCN[ABDE][0-9]$&quot;, rowData(sce)$SYMBOL)</span></span>
<span id="cb1806-2"><a href="cellCyclePhasesTop.html#cb1806-2"></a>cyclin.genes &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;^CCN[ABDE]+&quot;</span>, <span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol)</span>
<span id="cb1806-3"><a href="cellCyclePhasesTop.html#cb1806-3"></a>cyclin.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(sce)[cyclin.genes]</span>
<span id="cb1806-4"><a href="cellCyclePhasesTop.html#cb1806-4"></a><span class="kw">rowData</span>(sce)[cyclin.genes,]</span></code></pre></div>
<pre><code>## DataFrame with 11 rows and 11 columns
##                 ensembl_gene_id external_gene_name chromosome_name
##                     &lt;character&gt;        &lt;character&gt;     &lt;character&gt;
## ENSG00000100814 ENSG00000100814           CCNB1IP1              14
## ENSG00000105173 ENSG00000105173              CCNE1              19
## ENSG00000110092 ENSG00000110092              CCND1              11
## ENSG00000112576 ENSG00000112576              CCND3               6
## ENSG00000118971 ENSG00000118971              CCND2              12
## ENSG00000133101 ENSG00000133101              CCNA1              13
## ENSG00000134057 ENSG00000134057              CCNB1               5
## ENSG00000145386 ENSG00000145386              CCNA2               4
## ENSG00000157456 ENSG00000157456              CCNB2              15
## ENSG00000166946 ENSG00000166946            CCNDBP1              15
## ENSG00000175305 ENSG00000175305              CCNE2               8
##                 start_position end_position strandNum      Symbol
##                      &lt;integer&gt;    &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## ENSG00000100814       20311368     20333312        -1    CCNB1IP1
## ENSG00000105173       29811991     29824312         1       CCNE1
## ENSG00000110092       69641156     69654474         1       CCND1
## ENSG00000112576       41934934     42050357        -1       CCND3
## ENSG00000118971        4269771      4305353         1       CCND2
## ENSG00000133101       36431520     36442870         1       CCNA1
## ENSG00000134057       69167135     69178245         1       CCNB1
## ENSG00000145386      121816444    121823883        -1       CCNA2
## ENSG00000157456       59105126     59125045         1       CCNB2
## ENSG00000166946       43185118     43197177         1     CCNDBP1
## ENSG00000175305       94879770     94896678        -1       CCNE2
##                            Type       mean  detected gene_sparsity
##                     &lt;character&gt;  &lt;numeric&gt; &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000100814 Gene Expression 0.16142992 12.683592      0.847762
## ENSG00000105173 Gene Expression 0.06231641  3.435795      0.975033
## ENSG00000110092 Gene Expression 0.01138271  0.907470      0.989239
## ENSG00000112576 Gene Expression 1.23806651 47.508393      0.441745
## ENSG00000118971 Gene Expression 0.15739089 11.222723      0.890903
## ENSG00000133101 Gene Expression 0.00388166  0.330466      0.994500
## ENSG00000134057 Gene Expression 0.21152434  6.569975      0.926583
## ENSG00000145386 Gene Expression 0.17656316  7.217793      0.916204
## ENSG00000157456 Gene Expression 0.24488565  8.248531      0.905969
## ENSG00000166946 Gene Expression 0.39052665 25.314729      0.722259
## ENSG00000175305 Gene Expression 0.03039761  2.384075      0.974555</code></pre>
<div class="sourceCode" id="cb1808"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1808-1"><a href="cellCyclePhasesTop.html#cb1808-1"></a><span class="co"># only use the 10 most variable cyclins:</span></span>
<span id="cb1808-2"><a href="cellCyclePhasesTop.html#cb1808-2"></a><span class="kw">require</span>(DelayedArray)</span>
<span id="cb1808-3"><a href="cellCyclePhasesTop.html#cb1808-3"></a>tmpVar &lt;-<span class="st"> </span>DelayedMatrixStats<span class="op">::</span><span class="kw">rowVars</span>( <span class="kw">DelayedArray</span>(<span class="kw">assay</span>(sce[cyclin.genes,], <span class="st">&quot;logcounts&quot;</span>)))</span>
<span id="cb1808-4"><a href="cellCyclePhasesTop.html#cb1808-4"></a><span class="kw">names</span>(tmpVar) &lt;-<span class="st"> </span>cyclin.genes</span>
<span id="cb1808-5"><a href="cellCyclePhasesTop.html#cb1808-5"></a>cyclin.genes.sub &lt;-<span class="st"> </span><span class="kw">names</span>(tmpVar[<span class="kw">order</span>(tmpVar, <span class="dt">decreasing=</span>T)])[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span>
<span id="cb1808-6"><a href="cellCyclePhasesTop.html#cb1808-6"></a><span class="kw">rowData</span>(sce)[cyclin.genes.sub,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;Symbol&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 2 columns
##                 ensembl_gene_id      Symbol
##                     &lt;character&gt; &lt;character&gt;
## ENSG00000166946 ENSG00000166946     CCNDBP1
## ENSG00000112576 ENSG00000112576       CCND3
## ENSG00000134057 ENSG00000134057       CCNB1
## ENSG00000118971 ENSG00000118971       CCND2
## ENSG00000157456 ENSG00000157456       CCNB2
## ENSG00000100814 ENSG00000100814    CCNB1IP1
## ENSG00000145386 ENSG00000145386       CCNA2
## ENSG00000105173 ENSG00000105173       CCNE1
## ENSG00000175305 ENSG00000175305       CCNE2
## ENSG00000110092 ENSG00000110092       CCND1</code></pre>
<div class="sourceCode" id="cb1810"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1810-1"><a href="cellCyclePhasesTop.html#cb1810-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb1810-2"><a href="cellCyclePhasesTop.html#cb1810-2"></a><span class="kw">plotHeatmap</span>(sce,</span>
<span id="cb1810-3"><a href="cellCyclePhasesTop.html#cb1810-3"></a>            <span class="dt">order_columns_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1810-4"><a href="cellCyclePhasesTop.html#cb1810-4"></a>            <span class="dt">cluster_rows=</span><span class="ot">TRUE</span>,</span>
<span id="cb1810-5"><a href="cellCyclePhasesTop.html#cb1810-5"></a>            <span class="dt">features=</span><span class="kw">sort</span>(cyclin.genes.sub))</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotHeatmap_cellCyclePhases-1.png" width="672" /></p>
<p>With the gene expression levels reconstructed after batch correction with fastMNN:</p>
<div class="sourceCode" id="cb1811"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1811-1"><a href="cellCyclePhasesTop.html#cb1811-1"></a>top.markers &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1811-2"><a href="cellCyclePhasesTop.html#cb1811-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1811-3"><a href="cellCyclePhasesTop.html#cb1811-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>cyclin.genes.sub) <span class="op">%&gt;%</span></span>
<span id="cb1811-4"><a href="cellCyclePhasesTop.html#cb1811-4"></a><span class="st">  </span><span class="kw">pull</span>(Symbol)</span>
<span id="cb1811-5"><a href="cellCyclePhasesTop.html#cb1811-5"></a></span>
<span id="cb1811-6"><a href="cellCyclePhasesTop.html#cb1811-6"></a><span class="co"># have matrix to annotate sample with cluster and sample:</span></span>
<span id="cb1811-7"><a href="cellCyclePhasesTop.html#cb1811-7"></a><span class="co">#tmpData &lt;- logcounts(sce)[top.markers,]</span></span>
<span id="cb1811-8"><a href="cellCyclePhasesTop.html#cb1811-8"></a>tmpData3 &lt;-<span class="st"> </span><span class="kw">assay</span>(mnn.out) <span class="op">%&gt;%</span></span>
<span id="cb1811-9"><a href="cellCyclePhasesTop.html#cb1811-9"></a><span class="st">  </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span></span>
<span id="cb1811-10"><a href="cellCyclePhasesTop.html#cb1811-10"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1811-11"><a href="cellCyclePhasesTop.html#cb1811-11"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;Symbol&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1811-12"><a href="cellCyclePhasesTop.html#cb1811-12"></a><span class="st">  </span><span class="kw">filter</span>(Symbol <span class="op">%in%</span><span class="st"> </span>top.markers) <span class="op">%&gt;%</span></span>
<span id="cb1811-13"><a href="cellCyclePhasesTop.html#cb1811-13"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="st">&quot;Symbol&quot;</span>)</span>
<span id="cb1811-14"><a href="cellCyclePhasesTop.html#cb1811-14"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out)<span class="op">$</span>Barcode</span>
<span id="cb1811-15"><a href="cellCyclePhasesTop.html#cb1811-15"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(mnn.out))</span>
<span id="cb1811-16"><a href="cellCyclePhasesTop.html#cb1811-16"></a></span>
<span id="cb1811-17"><a href="cellCyclePhasesTop.html#cb1811-17"></a><span class="co"># concat sample and barcode names to make unique name across the whole data set</span></span>
<span id="cb1811-18"><a href="cellCyclePhasesTop.html#cb1811-18"></a>tmpCellNames &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">colData</span>(mnn.out)<span class="op">$</span>Sample.Name,</span>
<span id="cb1811-19"><a href="cellCyclePhasesTop.html#cb1811-19"></a>                      <span class="kw">colData</span>(mnn.out)<span class="op">$</span>Barcode, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1811-20"><a href="cellCyclePhasesTop.html#cb1811-20"></a><span class="co"># use these to name columns of matrix the show as heatmap:</span></span>
<span id="cb1811-21"><a href="cellCyclePhasesTop.html#cb1811-21"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode                    </span></span>
<span id="cb1811-22"><a href="cellCyclePhasesTop.html#cb1811-22"></a></span>
<span id="cb1811-23"><a href="cellCyclePhasesTop.html#cb1811-23"></a><span class="co"># columns annotation with cell name:</span></span>
<span id="cb1811-24"><a href="cellCyclePhasesTop.html#cb1811-24"></a>mat_col &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">cluster =</span> sce<span class="op">$</span>cluster,</span>
<span id="cb1811-25"><a href="cellCyclePhasesTop.html#cb1811-25"></a>              <span class="dt">sample =</span> sce<span class="op">$</span>Sample.Name,</span>
<span id="cb1811-26"><a href="cellCyclePhasesTop.html#cb1811-26"></a>              <span class="dt">type =</span> sce<span class="op">$</span>source_name</span>
<span id="cb1811-27"><a href="cellCyclePhasesTop.html#cb1811-27"></a>        )</span>
<span id="cb1811-28"><a href="cellCyclePhasesTop.html#cb1811-28"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span><span class="kw">colnames</span>(tmpData3)</span>
<span id="cb1811-29"><a href="cellCyclePhasesTop.html#cb1811-29"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode</span></span>
<span id="cb1811-30"><a href="cellCyclePhasesTop.html#cb1811-30"></a></span>
<span id="cb1811-31"><a href="cellCyclePhasesTop.html#cb1811-31"></a><span class="co"># Prepare colours for clusters:</span></span>
<span id="cb1811-32"><a href="cellCyclePhasesTop.html#cb1811-32"></a>colourCount =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(sce<span class="op">$</span>cluster))</span>
<span id="cb1811-33"><a href="cellCyclePhasesTop.html#cb1811-33"></a>getPalette =<span class="st"> </span><span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb1811-34"><a href="cellCyclePhasesTop.html#cb1811-34"></a></span>
<span id="cb1811-35"><a href="cellCyclePhasesTop.html#cb1811-35"></a>mat_colors &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">group =</span> <span class="kw">getPalette</span>(colourCount))</span>
<span id="cb1811-36"><a href="cellCyclePhasesTop.html#cb1811-36"></a><span class="kw">names</span>(mat_colors<span class="op">$</span>group) &lt;-<span class="st"> </span><span class="kw">unique</span>(sce<span class="op">$</span>cluster)</span>
<span id="cb1811-37"><a href="cellCyclePhasesTop.html#cb1811-37"></a></span>
<span id="cb1811-38"><a href="cellCyclePhasesTop.html#cb1811-38"></a><span class="co"># plot heatmap:</span></span>
<span id="cb1811-39"><a href="cellCyclePhasesTop.html#cb1811-39"></a><span class="co">#pheatmap(tmpData3,</span></span>
<span id="cb1811-40"><a href="cellCyclePhasesTop.html#cb1811-40"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tmpData3[,<span class="kw">order</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>clusterStg)],</span>
<span id="cb1811-41"><a href="cellCyclePhasesTop.html#cb1811-41"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1811-42"><a href="cellCyclePhasesTop.html#cb1811-42"></a>           <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1811-43"><a href="cellCyclePhasesTop.html#cb1811-43"></a>           <span class="co">#show_rownames     = FALSE,</span></span>
<span id="cb1811-44"><a href="cellCyclePhasesTop.html#cb1811-44"></a>           <span class="dt">show_rownames     =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1811-45"><a href="cellCyclePhasesTop.html#cb1811-45"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1811-46"><a href="cellCyclePhasesTop.html#cb1811-46"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1811-47"><a href="cellCyclePhasesTop.html#cb1811-47"></a>           <span class="dt">annotation_colors =</span> mat_colors,</span>
<span id="cb1811-48"><a href="cellCyclePhasesTop.html#cb1811-48"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1811-49"><a href="cellCyclePhasesTop.html#cb1811-49"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pheatmap_topMarkers_cellCyclePhases-1.png" width="672" /></p>
<p>For example, we can use standard DE methods (Chapter 11) to look for upregulation of each cyclin, which would imply that a subpopulation contains more cells in the corresponding cell cycle phase. The same logic applies to comparisons between treatment conditions, as described in Chapter 14.</p>
<div class="sourceCode" id="cb1812"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1812-1"><a href="cellCyclePhasesTop.html#cb1812-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1812-2"><a href="cellCyclePhasesTop.html#cb1812-2"></a>markers &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1812-3"><a href="cellCyclePhasesTop.html#cb1812-3"></a>                       <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1812-4"><a href="cellCyclePhasesTop.html#cb1812-4"></a>                       <span class="dt">subset.row=</span>cyclin.genes,</span>
<span id="cb1812-5"><a href="cellCyclePhasesTop.html#cb1812-5"></a>                       <span class="dt">test.type=</span><span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb1812-6"><a href="cellCyclePhasesTop.html#cb1812-6"></a>                       <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1812-7"><a href="cellCyclePhasesTop.html#cb1812-7"></a>                       <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1812-8"><a href="cellCyclePhasesTop.html#cb1812-8"></a></span>
<span id="cb1812-9"><a href="cellCyclePhasesTop.html#cb1812-9"></a><span class="co"># We can infer that cluster 4 or 6 has more cells in G2/M than the other clusters,</span></span>
<span id="cb1812-10"><a href="cellCyclePhasesTop.html#cb1812-10"></a><span class="co"># based on higher expression of the cyclin B&#39;s.</span></span>
<span id="cb1812-11"><a href="cellCyclePhasesTop.html#cb1812-11"></a>markers[[<span class="dv">4</span>]] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1812-12"><a href="cellCyclePhasesTop.html#cb1812-12"></a><span class="co">#markers[[6]] %&gt;% data.frame() %&gt;%</span></span>
<span id="cb1812-13"><a href="cellCyclePhasesTop.html#cb1812-13"></a><span class="st">    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1812-14"><a href="cellCyclePhasesTop.html#cb1812-14"></a><span class="st">    </span><span class="kw">left_join</span>(<span class="kw">data.frame</span>(<span class="kw">rowData</span>(sce))[, <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;Symbol&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1812-15"><a href="cellCyclePhasesTop.html#cb1812-15"></a><span class="st">    </span><span class="kw">select</span>(Symbol, Top, p.value, FDR, summary.AUC)</span></code></pre></div>
<pre><code>##      Symbol Top      p.value          FDR summary.AUC
## 1     CCND3   1 3.456680e-14 9.505869e-14   0.8133940
## 2     CCNA2   1 2.519496e-17 1.385723e-16   0.6902023
## 3     CCNB2   1 8.745314e-24 9.619845e-23   0.7356370
## 4     CCNB1   2 6.859532e-15 2.515162e-14   0.6846747
## 5     CCNE2   2 3.130980e-04 5.740130e-04   0.5442160
## 6  CCNB1IP1   3 2.634628e-02 3.622614e-02   0.5586490
## 7     CCNE1   4 6.661062e-05 1.465434e-04   0.5519253
## 8     CCND2   7 1.000000e+00 1.000000e+00   0.5088315
## 9   CCNDBP1   7 1.096260e-02 1.722694e-02   0.5643214
## 10    CCND1   9 1.000000e+00 1.000000e+00   0.4969816
## 11    CCNA1  11 1.000000e+00 1.000000e+00   0.5000000</code></pre>
<p>Direct examination of cyclin expression is easily to understand, interpret and experimentally validate. However, it is best suited for statements about relative cell cycle activity; for example, we would find it difficult to assign cell cycle phase without the presence of clusters spanning all phases to provide benchmarks for “high” and “low” expression of each cyclin. We also assume that cyclin expression is not affected by biological processes other than the cell cycle, which may be a strong assumption in some cases, e.g., malignant cells. This strategy is strongly dependent on having good sequencing coverage of the cyclins, which is less of an issue for the whole-of-transcriptome methods described below that use information from more genes.</p>
</div>
<div id="using-the-cyclone-classifier" class="section level2">
<h2><span class="header-section-number">22.4</span> Using the cyclone() classifier</h2>
<p>The prediction method described by Scialdone et al. (2015) is another approach for classifying cells into cell cycle phases. Using a reference dataset, we first compute the sign of the difference in expression between each pair of genes. Pairs with changes in the sign across cell cycle phases are chosen as markers. Cells in a test dataset can then be classified into the appropriate phase, based on whether the observed sign for each marker pair is consistent with one phase or another. This approach is implemented in the cyclone() function from the scran package, which also contains pre-trained set of marker pairs for mouse and human data.</p>
<div class="sourceCode" id="cb1814"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1814-1"><a href="cellCyclePhasesTop.html#cb1814-1"></a><span class="kw">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb1814-2"><a href="cellCyclePhasesTop.html#cb1814-2"></a><span class="kw">library</span>(scran)</span>
<span id="cb1814-3"><a href="cellCyclePhasesTop.html#cb1814-3"></a>hs.pairs &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">&quot;exdata&quot;</span>, <span class="st">&quot;human_cycle_markers.rds&quot;</span>, </span>
<span id="cb1814-4"><a href="cellCyclePhasesTop.html#cb1814-4"></a>    <span class="dt">package=</span><span class="st">&quot;scran&quot;</span>))</span>
<span id="cb1814-5"><a href="cellCyclePhasesTop.html#cb1814-5"></a></span>
<span id="cb1814-6"><a href="cellCyclePhasesTop.html#cb1814-6"></a><span class="co"># Using Ensembl IDs to match up with the annotation in &#39;mm.pairs&#39;.</span></span>
<span id="cb1814-7"><a href="cellCyclePhasesTop.html#cb1814-7"></a>assignments &lt;-<span class="st"> </span><span class="kw">cyclone</span>(sce,</span>
<span id="cb1814-8"><a href="cellCyclePhasesTop.html#cb1814-8"></a>                       hs.pairs,</span>
<span id="cb1814-9"><a href="cellCyclePhasesTop.html#cb1814-9"></a>                       <span class="dt">gene.names=</span><span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id) <span class="co"># SLOW, even with 3500 cells only</span></span></code></pre></div>
<p>Write assignments to file.</p>
<div class="sourceCode" id="cb1815"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1815-1"><a href="cellCyclePhasesTop.html#cb1815-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_cyclone.Rds&quot;</span>,</span>
<span id="cb1815-2"><a href="cellCyclePhasesTop.html#cb1815-2"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1815-3"><a href="cellCyclePhasesTop.html#cb1815-3"></a><span class="kw">print</span>(tmpFn)</span>
<span id="cb1815-4"><a href="cellCyclePhasesTop.html#cb1815-4"></a><span class="kw">saveRDS</span>(assignments, <span class="dt">file=</span>tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb1816"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1816-1"><a href="cellCyclePhasesTop.html#cb1816-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_cyclone.Rds&quot;</span>,</span>
<span id="cb1816-2"><a href="cellCyclePhasesTop.html#cb1816-2"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1816-3"><a href="cellCyclePhasesTop.html#cb1816-3"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1_cyclone.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1818"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1818-1"><a href="cellCyclePhasesTop.html#cb1818-1"></a>assignments &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>tmpFn)</span></code></pre></div>
<p>Copy cell cycle assignments to SCE object:</p>
<div class="sourceCode" id="cb1819"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1819-1"><a href="cellCyclePhasesTop.html#cb1819-1"></a><span class="kw">colData</span>(sce)<span class="op">$</span>phases &lt;-<span class="st"> </span>assignments<span class="op">$</span>phases</span></code></pre></div>
<p>For each cell, a higher score for a phase corresponds to a higher probability that the cell is in that phase. We focus on the G1 and G2/M scores as these are the most informative for classification.</p>
<p>The plot below show the cell cycle phase scores obtained by applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases.</p>
<div class="sourceCode" id="cb1820"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1820-1"><a href="cellCyclePhasesTop.html#cb1820-1"></a><span class="co">#plot(assignments$score$G1, assignments$score$G2M,</span></span>
<span id="cb1820-2"><a href="cellCyclePhasesTop.html#cb1820-2"></a><span class="co">#    xlab=&quot;G1 score&quot;, ylab=&quot;G2/M score&quot;, pch=16)</span></span>
<span id="cb1820-3"><a href="cellCyclePhasesTop.html#cb1820-3"></a></span>
<span id="cb1820-4"><a href="cellCyclePhasesTop.html#cb1820-4"></a><span class="kw">ggplot</span>(assignments<span class="op">$</span>score, <span class="kw">aes</span>(<span class="dt">x=</span>G1, <span class="dt">y=</span>G2M, <span class="dt">col=</span>S)) <span class="op">+</span></span>
<span id="cb1820-5"><a href="cellCyclePhasesTop.html#cb1820-5"></a><span class="st">  </span><span class="kw">geom_point</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_plot_cellCyclePhases-1.png" width="672" /></p>
<p>Cells are classified as being in G1 phase if the G1 score is above 0.5 and greater than the G2/M score; in G2/M phase if the G2/M score is above 0.5 and greater than the G1 score; and in S phase if neither score is above 0.5.</p>
<div class="sourceCode" id="cb1821"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1821-1"><a href="cellCyclePhasesTop.html#cb1821-1"></a><span class="kw">colLabels</span>(sce) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>source_name</span>
<span id="cb1821-2"><a href="cellCyclePhasesTop.html#cb1821-2"></a><span class="kw">table</span>(assignments<span class="op">$</span>phases, <span class="kw">colLabels</span>(sce))</span></code></pre></div>
<pre><code>##      
##       ABMMC ETV6-RUNX1  HHD PBMMC PRE-T
##   G1      0       1644    0  1089     0
##   G2M     0         93    0   229     0
##   S       0        263    0   182     0</code></pre>
<div class="sourceCode" id="cb1823"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1823-1"><a href="cellCyclePhasesTop.html#cb1823-1"></a><span class="co"># prop per phase</span></span>
<span id="cb1823-2"><a href="cellCyclePhasesTop.html#cb1823-2"></a><span class="kw">round</span>(<span class="kw">prop.table</span>(<span class="kw">table</span>(assignments<span class="op">$</span>phases, <span class="kw">colLabels</span>(sce)),<span class="dv">1</span>),<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##      
##       ABMMC ETV6-RUNX1  HHD PBMMC PRE-T
##   G1   0.00       0.60 0.00  0.40  0.00
##   G2M  0.00       0.29 0.00  0.71  0.00
##   S    0.00       0.59 0.00  0.41  0.00</code></pre>
<div class="sourceCode" id="cb1825"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1825-1"><a href="cellCyclePhasesTop.html#cb1825-1"></a><span class="co"># prop per type</span></span>
<span id="cb1825-2"><a href="cellCyclePhasesTop.html#cb1825-2"></a><span class="kw">round</span>(<span class="kw">prop.table</span>(<span class="kw">table</span>(assignments<span class="op">$</span>phases, <span class="kw">colLabels</span>(sce)),<span class="dv">2</span>),<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##      
##       ABMMC ETV6-RUNX1 HHD PBMMC PRE-T
##   G1              0.82      0.73      
##   G2M             0.05      0.15      
##   S               0.13      0.12</code></pre>
<div class="sourceCode" id="cb1827"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1827-1"><a href="cellCyclePhasesTop.html#cb1827-1"></a>tsne1 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;phases&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1827-2"><a href="cellCyclePhasesTop.html#cb1827-2"></a>tsne1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_tsne_cellCyclePhases-1.png" width="672" /></p>
<div class="sourceCode" id="cb1828"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1828-1"><a href="cellCyclePhasesTop.html#cb1828-1"></a>tsne1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>phases)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_tsne_cellCyclePhases-2.png" width="672" /></p>
<div class="sourceCode" id="cb1829"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1829-1"><a href="cellCyclePhasesTop.html#cb1829-1"></a>umap1 &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;phases&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1829-2"><a href="cellCyclePhasesTop.html#cb1829-2"></a>umap1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_umap_cellCyclePhases-1.png" width="672" /></p>
<div class="sourceCode" id="cb1830"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1830-1"><a href="cellCyclePhasesTop.html#cb1830-1"></a>umap1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>phases)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_umap_cellCyclePhases-2.png" width="672" /></p>
<div class="sourceCode" id="cb1831"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1831-1"><a href="cellCyclePhasesTop.html#cb1831-1"></a>tmpPhases &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>phases</span>
<span id="cb1831-2"><a href="cellCyclePhasesTop.html#cb1831-2"></a><span class="kw">names</span>(tmpPhases) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Barcode</span>
<span id="cb1831-3"><a href="cellCyclePhasesTop.html#cb1831-3"></a><span class="kw">colData</span>(mnn.out)<span class="op">$</span>phases &lt;-<span class="st"> </span>tmpPhases[<span class="kw">colData</span>(mnn.out)<span class="op">$</span>Barcode]</span>
<span id="cb1831-4"><a href="cellCyclePhasesTop.html#cb1831-4"></a>tsne1b &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb1831-5"><a href="cellCyclePhasesTop.html#cb1831-5"></a>           <span class="dt">colour_by=</span><span class="st">&quot;phases&quot;</span>,</span>
<span id="cb1831-6"><a href="cellCyclePhasesTop.html#cb1831-6"></a>           <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>) <span class="op">+</span></span>
<span id="cb1831-7"><a href="cellCyclePhasesTop.html#cb1831-7"></a>fontsize</span>
<span id="cb1831-8"><a href="cellCyclePhasesTop.html#cb1831-8"></a>umap1b &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(mnn.out,</span>
<span id="cb1831-9"><a href="cellCyclePhasesTop.html#cb1831-9"></a>           <span class="dt">colour_by=</span><span class="st">&quot;phases&quot;</span>,</span>
<span id="cb1831-10"><a href="cellCyclePhasesTop.html#cb1831-10"></a>           <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>) <span class="op">+</span></span>
<span id="cb1831-11"><a href="cellCyclePhasesTop.html#cb1831-11"></a>fontsize</span>
<span id="cb1831-12"><a href="cellCyclePhasesTop.html#cb1831-12"></a></span>
<span id="cb1831-13"><a href="cellCyclePhasesTop.html#cb1831-13"></a>tsne1b <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>mnn.out<span class="op">$</span>phases)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_dimRedPlot_cellCyclePhases-1.png" width="672" /></p>
<div class="sourceCode" id="cb1832"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1832-1"><a href="cellCyclePhasesTop.html#cb1832-1"></a>umap1b <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>mnn.out<span class="op">$</span>phases)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/score_dimRedPlot_cellCyclePhases-2.png" width="672" /></p>
<pre class="tidy_dimredplot_cellcyclephases"><code>rm(tsne1, tsne1b, umap1, umap1b, mnn.out)</code></pre>
<p>Compare to assignments made in each sample separately</p>
<div class="sourceCode" id="cb1834"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1834-1"><a href="cellCyclePhasesTop.html#cb1834-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb1834-2"><a href="cellCyclePhasesTop.html#cb1834-2"></a><span class="co">#cb_sampleSheetFn &lt;- file.path(projDir, &quot;Data/CaronBourque2020/SraRunTable.txt&quot;)</span></span>
<span id="cb1834-3"><a href="cellCyclePhasesTop.html#cb1834-3"></a><span class="co">#cb_sampleSheet &lt;- read.table(cb_sampleSheetFn, header=T, sep=&quot;,&quot;)</span></span>
<span id="cb1834-4"><a href="cellCyclePhasesTop.html#cb1834-4"></a><span class="co">#splVec &lt;- cb_sampleSheet %&gt;% filter(source_name == splSetToGet) %&gt;%</span></span>
<span id="cb1834-5"><a href="cellCyclePhasesTop.html#cb1834-5"></a><span class="co">#        pull(Sample.Name) %&gt;% unique</span></span>
<span id="cb1834-6"><a href="cellCyclePhasesTop.html#cb1834-6"></a></span>
<span id="cb1834-7"><a href="cellCyclePhasesTop.html#cb1834-7"></a>splVec &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name)</span>
<span id="cb1834-8"><a href="cellCyclePhasesTop.html#cb1834-8"></a></span>
<span id="cb1834-9"><a href="cellCyclePhasesTop.html#cb1834-9"></a>all.sce &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1834-10"><a href="cellCyclePhasesTop.html#cb1834-10"></a><span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb1834-11"><a href="cellCyclePhasesTop.html#cb1834-11"></a>{</span>
<span id="cb1834-12"><a href="cellCyclePhasesTop.html#cb1834-12"></a>        vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1834-13"><a href="cellCyclePhasesTop.html#cb1834-13"></a><span class="st">                </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1834-14"><a href="cellCyclePhasesTop.html#cb1834-14"></a><span class="st">                </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb1834-15"><a href="cellCyclePhasesTop.html#cb1834-15"></a><span class="st">                </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1834-16"><a href="cellCyclePhasesTop.html#cb1834-16"></a>        tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1834-17"><a href="cellCyclePhasesTop.html#cb1834-17"></a></span>
<span id="cb1834-18"><a href="cellCyclePhasesTop.html#cb1834-18"></a>        all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb1834-19"><a href="cellCyclePhasesTop.html#cb1834-19"></a>}</span>
<span id="cb1834-20"><a href="cellCyclePhasesTop.html#cb1834-20"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb1834-21"><a href="cellCyclePhasesTop.html#cb1834-21"></a><span class="co"># use logNormCounts()</span></span>
<span id="cb1834-22"><a href="cellCyclePhasesTop.html#cb1834-22"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, logNormCounts)</span>
<span id="cb1834-23"><a href="cellCyclePhasesTop.html#cb1834-23"></a></span>
<span id="cb1834-24"><a href="cellCyclePhasesTop.html#cb1834-24"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb1834-25"><a href="cellCyclePhasesTop.html#cb1834-25"></a><span class="co"># model varaince with modelGeneVar()</span></span>
<span id="cb1834-26"><a href="cellCyclePhasesTop.html#cb1834-26"></a><span class="co"># find highly variable genes (HVGs) with getTopHVGs()</span></span>
<span id="cb1834-27"><a href="cellCyclePhasesTop.html#cb1834-27"></a>all.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, modelGeneVar)</span>
<span id="cb1834-28"><a href="cellCyclePhasesTop.html#cb1834-28"></a>all.hvgs &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, getTopHVGs, <span class="dt">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb1834-29"><a href="cellCyclePhasesTop.html#cb1834-29"></a></span>
<span id="cb1834-30"><a href="cellCyclePhasesTop.html#cb1834-30"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb1834-31"><a href="cellCyclePhasesTop.html#cb1834-31"></a><span class="co"># use runPCA()</span></span>
<span id="cb1834-32"><a href="cellCyclePhasesTop.html#cb1834-32"></a><span class="co"># then compute embeddings with runTSNE() and runUMAP()</span></span>
<span id="cb1834-33"><a href="cellCyclePhasesTop.html#cb1834-33"></a><span class="kw">library</span>(BiocSingular)</span>
<span id="cb1834-34"><a href="cellCyclePhasesTop.html#cb1834-34"></a><span class="kw">set.seed</span>(<span class="dv">10000</span>)</span>
<span id="cb1834-35"><a href="cellCyclePhasesTop.html#cb1834-35"></a>all.sce &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="dt">FUN=</span>runPCA, <span class="dt">x=</span>all.sce, <span class="dt">subset_row=</span>all.hvgs,</span>
<span id="cb1834-36"><a href="cellCyclePhasesTop.html#cb1834-36"></a>    <span class="dt">MoreArgs=</span><span class="kw">list</span>(<span class="dt">ncomponents=</span><span class="dv">25</span>, <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>()),</span>
<span id="cb1834-37"><a href="cellCyclePhasesTop.html#cb1834-37"></a>    <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb1834-38"><a href="cellCyclePhasesTop.html#cb1834-38"></a></span>
<span id="cb1834-39"><a href="cellCyclePhasesTop.html#cb1834-39"></a><span class="kw">set.seed</span>(<span class="dv">100000</span>)</span>
<span id="cb1834-40"><a href="cellCyclePhasesTop.html#cb1834-40"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, runTSNE, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1834-41"><a href="cellCyclePhasesTop.html#cb1834-41"></a></span>
<span id="cb1834-42"><a href="cellCyclePhasesTop.html#cb1834-42"></a><span class="kw">set.seed</span>(<span class="dv">1000000</span>)</span>
<span id="cb1834-43"><a href="cellCyclePhasesTop.html#cb1834-43"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, runUMAP, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1834-44"><a href="cellCyclePhasesTop.html#cb1834-44"></a></span>
<span id="cb1834-45"><a href="cellCyclePhasesTop.html#cb1834-45"></a><span class="co">#--- clustering ---#</span></span>
<span id="cb1834-46"><a href="cellCyclePhasesTop.html#cb1834-46"></a><span class="co"># cluster each sample separately</span></span>
<span id="cb1834-47"><a href="cellCyclePhasesTop.html#cb1834-47"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="kw">names</span>(all.sce)) {</span>
<span id="cb1834-48"><a href="cellCyclePhasesTop.html#cb1834-48"></a>    g &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(all.sce[[n]], <span class="dt">k=</span><span class="dv">10</span>, <span class="dt">use.dimred=</span><span class="st">&#39;PCA&#39;</span>)</span>
<span id="cb1834-49"><a href="cellCyclePhasesTop.html#cb1834-49"></a>    clust &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(g)<span class="op">$</span>membership</span>
<span id="cb1834-50"><a href="cellCyclePhasesTop.html#cb1834-50"></a>    <span class="co">#colLabels(all.sce[[n]])  &lt;- factor(clust)</span></span>
<span id="cb1834-51"><a href="cellCyclePhasesTop.html#cb1834-51"></a>    all.sce[[n]]<span class="op">$</span>label  &lt;-<span class="st"> </span><span class="kw">factor</span>(clust)</span>
<span id="cb1834-52"><a href="cellCyclePhasesTop.html#cb1834-52"></a>}</span>
<span id="cb1834-53"><a href="cellCyclePhasesTop.html#cb1834-53"></a></span>
<span id="cb1834-54"><a href="cellCyclePhasesTop.html#cb1834-54"></a><span class="co">#--- cyclone ---#</span></span>
<span id="cb1834-55"><a href="cellCyclePhasesTop.html#cb1834-55"></a>hs.pairs &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">&quot;exdata&quot;</span>, <span class="st">&quot;human_cycle_markers.rds&quot;</span>, </span>
<span id="cb1834-56"><a href="cellCyclePhasesTop.html#cb1834-56"></a>    <span class="dt">package=</span><span class="st">&quot;scran&quot;</span>))</span>
<span id="cb1834-57"><a href="cellCyclePhasesTop.html#cb1834-57"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="kw">names</span>(all.sce)) {</span>
<span id="cb1834-58"><a href="cellCyclePhasesTop.html#cb1834-58"></a>    assignments &lt;-<span class="st"> </span><span class="kw">cyclone</span>(all.sce[[n]],</span>
<span id="cb1834-59"><a href="cellCyclePhasesTop.html#cb1834-59"></a>                       hs.pairs,</span>
<span id="cb1834-60"><a href="cellCyclePhasesTop.html#cb1834-60"></a>                       <span class="dt">gene.names=</span><span class="kw">rowData</span>(all.sce[[n]])<span class="op">$</span>ensembl_gene_id)</span>
<span id="cb1834-61"><a href="cellCyclePhasesTop.html#cb1834-61"></a>    all.sce[[n]]<span class="op">$</span>phases  &lt;-<span class="st"> </span>assignments<span class="op">$</span>phases</span>
<span id="cb1834-62"><a href="cellCyclePhasesTop.html#cb1834-62"></a>}</span>
<span id="cb1834-63"><a href="cellCyclePhasesTop.html#cb1834-63"></a><span class="co">#lapply(all.sce, function(x){table(colData(x)$phases)})</span></span>
<span id="cb1834-64"><a href="cellCyclePhasesTop.html#cb1834-64"></a>tmpList &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce,</span>
<span id="cb1834-65"><a href="cellCyclePhasesTop.html#cb1834-65"></a>       <span class="cf">function</span>(x){</span>
<span id="cb1834-66"><a href="cellCyclePhasesTop.html#cb1834-66"></a>         <span class="co">#colData(x)[,c(&quot;Barcode&quot;, &quot;phases&quot;)]</span></span>
<span id="cb1834-67"><a href="cellCyclePhasesTop.html#cb1834-67"></a>         <span class="kw">data.frame</span>(</span>
<span id="cb1834-68"><a href="cellCyclePhasesTop.html#cb1834-68"></a>           <span class="st">&quot;Barcode&quot;</span> =<span class="st"> </span><span class="kw">colData</span>(x)<span class="op">$</span>Barcode,</span>
<span id="cb1834-69"><a href="cellCyclePhasesTop.html#cb1834-69"></a>           <span class="st">&quot;phasesBySpl&quot;</span> =<span class="st"> </span><span class="kw">colData</span>(x)<span class="op">$</span>phases</span>
<span id="cb1834-70"><a href="cellCyclePhasesTop.html#cb1834-70"></a>         )</span>
<span id="cb1834-71"><a href="cellCyclePhasesTop.html#cb1834-71"></a>         }</span>
<span id="cb1834-72"><a href="cellCyclePhasesTop.html#cb1834-72"></a>       )</span>
<span id="cb1834-73"><a href="cellCyclePhasesTop.html#cb1834-73"></a>phaseBySpl &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, tmpList)</span>
<span id="cb1834-74"><a href="cellCyclePhasesTop.html#cb1834-74"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1834-75"><a href="cellCyclePhasesTop.html#cb1834-75"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1834-76"><a href="cellCyclePhasesTop.html#cb1834-76"></a><span class="st">  </span><span class="kw">left_join</span>(phaseBySpl, <span class="dt">by=</span><span class="st">&quot;Barcode&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1834-77"><a href="cellCyclePhasesTop.html#cb1834-77"></a><span class="st">  </span>DataFrame</span>
<span id="cb1834-78"><a href="cellCyclePhasesTop.html#cb1834-78"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>phases,</span>
<span id="cb1834-79"><a href="cellCyclePhasesTop.html#cb1834-79"></a>      <span class="kw">colData</span>(sce)<span class="op">$</span>phasesBySpl)</span></code></pre></div>
<pre><code>##      
##         G1  G2M    S
##   G1  2155  228  350
##   G2M  221   59   42
##   S    333   47   65</code></pre>
</div>
<div id="regressing-out-cell-cycle-phase" class="section level2">
<h2><span class="header-section-number">22.5</span> Regressing out cell cycle phase</h2>
<p>For some time, it was popular to regress out the cell cycle phase prior to downstream analyses. The aim was to remove uninteresting variation due to cell cycle, thus improving resolution of other biological processes of interest. We could implement this by performing cell cycle phase assignment as described above, treating each phase as a separate batch and applying any of the batch correction strategies. The most common approach is to use a linear model to simply regress out the phase effect, e.g., via regressBatches().</p>
<div class="sourceCode" id="cb1836"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1836-1"><a href="cellCyclePhasesTop.html#cb1836-1"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1836-2"><a href="cellCyclePhasesTop.html#cb1836-2"></a>sce.nocycle &lt;-<span class="st"> </span><span class="kw">regressBatches</span>(sce, <span class="dt">batch=</span>sce<span class="op">$</span>phases)</span>
<span id="cb1836-3"><a href="cellCyclePhasesTop.html#cb1836-3"></a></span>
<span id="cb1836-4"><a href="cellCyclePhasesTop.html#cb1836-4"></a><span class="co"># PCA</span></span>
<span id="cb1836-5"><a href="cellCyclePhasesTop.html#cb1836-5"></a><span class="co">#plotPCA(sce, colour_by = &quot;Sample.Name&quot;)</span></span>
<span id="cb1836-6"><a href="cellCyclePhasesTop.html#cb1836-6"></a>sce.nocycle &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb1836-7"><a href="cellCyclePhasesTop.html#cb1836-7"></a>  sce.nocycle,</span>
<span id="cb1836-8"><a href="cellCyclePhasesTop.html#cb1836-8"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;corrected&quot;</span></span>
<span id="cb1836-9"><a href="cellCyclePhasesTop.html#cb1836-9"></a>)</span>
<span id="cb1836-10"><a href="cellCyclePhasesTop.html#cb1836-10"></a>p &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(</span>
<span id="cb1836-11"><a href="cellCyclePhasesTop.html#cb1836-11"></a>    sce.nocycle,</span>
<span id="cb1836-12"><a href="cellCyclePhasesTop.html#cb1836-12"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span></span>
<span id="cb1836-13"><a href="cellCyclePhasesTop.html#cb1836-13"></a>)</span>
<span id="cb1836-14"><a href="cellCyclePhasesTop.html#cb1836-14"></a>p</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/regress_phases_cellCyclePhases-1.png" width="672" /></p>
<div class="sourceCode" id="cb1837"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1837-1"><a href="cellCyclePhasesTop.html#cb1837-1"></a><span class="co"># TSNE</span></span>
<span id="cb1837-2"><a href="cellCyclePhasesTop.html#cb1837-2"></a>sce.nocycle &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(</span>
<span id="cb1837-3"><a href="cellCyclePhasesTop.html#cb1837-3"></a>  sce.nocycle,</span>
<span id="cb1837-4"><a href="cellCyclePhasesTop.html#cb1837-4"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1837-5"><a href="cellCyclePhasesTop.html#cb1837-5"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(</span>
<span id="cb1837-6"><a href="cellCyclePhasesTop.html#cb1837-6"></a>    sce.nocycle,</span>
<span id="cb1837-7"><a href="cellCyclePhasesTop.html#cb1837-7"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span></span>
<span id="cb1837-8"><a href="cellCyclePhasesTop.html#cb1837-8"></a>)</span>
<span id="cb1837-9"><a href="cellCyclePhasesTop.html#cb1837-9"></a>p</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/regress_phases_cellCyclePhases-2.png" width="672" /></p>
<p>That said, we do not consider adjusting for cell cycle to be a necessary step in routine scRNA-seq analyses. In most applications, the cell cycle is a minor factor of variation, secondary to differences between cell types. Any attempt at removal would also need to assume that the cell cycle effect is orthogonal to other biological processes. For example, regression would potentially remove interesting signal if cell cycle activity varied across clusters or conditions, with a prime example being the increased proliferation of activated T cells (Richard et al. 2018). We suggest only performing cell cycle adjustment on an as-needed basis in populations with clear cell cycle effects.</p>
<p><strong>Exercise</strong> Remove the cell cycle genes listed in the ‘cell cycle’ GO term, perform PCA and plot t-SNE.</p>
<div class="sourceCode" id="cb1838"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1838-1"><a href="cellCyclePhasesTop.html#cb1838-1"></a><span class="co"># 2021-04-14: issue with org.Hs.eg.db, see https://support.bioconductor.org/p/9136239/</span></span>
<span id="cb1838-2"><a href="cellCyclePhasesTop.html#cb1838-2"></a><span class="kw">library</span>(org.Hs.eg.db)</span>
<span id="cb1838-3"><a href="cellCyclePhasesTop.html#cb1838-3"></a>cc.genes &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys=</span><span class="st">&quot;GO:0007049&quot;</span>, <span class="dt">keytype=</span><span class="st">&quot;GOALL&quot;</span>, <span class="dt">column=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb1838-4"><a href="cellCyclePhasesTop.html#cb1838-4"></a><span class="kw">length</span>(cc.genes)</span>
<span id="cb1838-5"><a href="cellCyclePhasesTop.html#cb1838-5"></a>sce.uncycle &lt;-<span class="st"> </span>sce[<span class="op">!</span><span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>cc.genes<span class="op">$</span>ENSEMBL,]</span>
<span id="cb1838-6"><a href="cellCyclePhasesTop.html#cb1838-6"></a></span>
<span id="cb1838-7"><a href="cellCyclePhasesTop.html#cb1838-7"></a><span class="co"># PCA</span></span>
<span id="cb1838-8"><a href="cellCyclePhasesTop.html#cb1838-8"></a>sce.uncycle &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb1838-9"><a href="cellCyclePhasesTop.html#cb1838-9"></a>  sce.uncycle,</span>
<span id="cb1838-10"><a href="cellCyclePhasesTop.html#cb1838-10"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span></span>
<span id="cb1838-11"><a href="cellCyclePhasesTop.html#cb1838-11"></a>)</span>
<span id="cb1838-12"><a href="cellCyclePhasesTop.html#cb1838-12"></a>p &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(</span>
<span id="cb1838-13"><a href="cellCyclePhasesTop.html#cb1838-13"></a>    sce.uncycle,</span>
<span id="cb1838-14"><a href="cellCyclePhasesTop.html#cb1838-14"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;phases&quot;</span>,</span>
<span id="cb1838-15"><a href="cellCyclePhasesTop.html#cb1838-15"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb1838-16"><a href="cellCyclePhasesTop.html#cb1838-16"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb1838-17"><a href="cellCyclePhasesTop.html#cb1838-17"></a>) </span>
<span id="cb1838-18"><a href="cellCyclePhasesTop.html#cb1838-18"></a>p</span>
<span id="cb1838-19"><a href="cellCyclePhasesTop.html#cb1838-19"></a></span>
<span id="cb1838-20"><a href="cellCyclePhasesTop.html#cb1838-20"></a><span class="co"># TSNE</span></span>
<span id="cb1838-21"><a href="cellCyclePhasesTop.html#cb1838-21"></a>sce.uncycle &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce.uncycle, <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb1838-22"><a href="cellCyclePhasesTop.html#cb1838-22"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(</span>
<span id="cb1838-23"><a href="cellCyclePhasesTop.html#cb1838-23"></a>    sce.uncycle,</span>
<span id="cb1838-24"><a href="cellCyclePhasesTop.html#cb1838-24"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;phases&quot;</span>,</span>
<span id="cb1838-25"><a href="cellCyclePhasesTop.html#cb1838-25"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb1838-26"><a href="cellCyclePhasesTop.html#cb1838-26"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb1838-27"><a href="cellCyclePhasesTop.html#cb1838-27"></a>)</span>
<span id="cb1838-28"><a href="cellCyclePhasesTop.html#cb1838-28"></a>p</span>
<span id="cb1838-29"><a href="cellCyclePhasesTop.html#cb1838-29"></a></span>
<span id="cb1838-30"><a href="cellCyclePhasesTop.html#cb1838-30"></a><span class="co"># UMAP</span></span>
<span id="cb1838-31"><a href="cellCyclePhasesTop.html#cb1838-31"></a>sce.uncycle &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce.uncycle, <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb1838-32"><a href="cellCyclePhasesTop.html#cb1838-32"></a>p &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(</span>
<span id="cb1838-33"><a href="cellCyclePhasesTop.html#cb1838-33"></a>    sce.uncycle,</span>
<span id="cb1838-34"><a href="cellCyclePhasesTop.html#cb1838-34"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;phases&quot;</span>,</span>
<span id="cb1838-35"><a href="cellCyclePhasesTop.html#cb1838-35"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb1838-36"><a href="cellCyclePhasesTop.html#cb1838-36"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb1838-37"><a href="cellCyclePhasesTop.html#cb1838-37"></a>)</span>
<span id="cb1838-38"><a href="cellCyclePhasesTop.html#cb1838-38"></a>p</span></code></pre></div>
<p>Write SCE object to file.</p>
<div class="sourceCode" id="cb1839"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1839-1"><a href="cellCyclePhasesTop.html#cb1839-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_cellCycle.Rds&quot;</span>,</span>
<span id="cb1839-2"><a href="cellCyclePhasesTop.html#cb1839-2"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1839-3"><a href="cellCyclePhasesTop.html#cb1839-3"></a><span class="kw">print</span>(tmpFn)</span>
<span id="cb1839-4"><a href="cellCyclePhasesTop.html#cb1839-4"></a><span class="kw">saveRDS</span>(sce, <span class="dt">file=</span>tmpFn)</span></code></pre></div>

<!--
  setSuf: "_5hCellPerSpl"
-->
<div class="sourceCode" id="cb1840"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1840-1"><a href="cellCyclePhasesTop.html#cb1840-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb1840-2"><a href="cellCyclePhasesTop.html#cb1840-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb1840-3"><a href="cellCyclePhasesTop.html#cb1840-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb1840-4"><a href="cellCyclePhasesTop.html#cb1840-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb1840-5"><a href="cellCyclePhasesTop.html#cb1840-5"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb1840-6"><a href="cellCyclePhasesTop.html#cb1840-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb1840-7"><a href="cellCyclePhasesTop.html#cb1840-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb1840-8"><a href="cellCyclePhasesTop.html#cb1840-8"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1840-9"><a href="cellCyclePhasesTop.html#cb1840-9"></a></span>
<span id="cb1840-10"><a href="cellCyclePhasesTop.html#cb1840-10"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1840-11"><a href="cellCyclePhasesTop.html#cb1840-11"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1840-12"><a href="cellCyclePhasesTop.html#cb1840-12"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span></span>
<span id="cb1840-13"><a href="cellCyclePhasesTop.html#cb1840-13"></a>    setSuf &lt;-<span class="st"> &quot;_5hCps&quot;</span></span>
<span id="cb1840-14"><a href="cellCyclePhasesTop.html#cb1840-14"></a>}</span>
<span id="cb1840-15"><a href="cellCyclePhasesTop.html#cb1840-15"></a></span>
<span id="cb1840-16"><a href="cellCyclePhasesTop.html#cb1840-16"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1840-17"><a href="cellCyclePhasesTop.html#cb1840-17"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1840-18"><a href="cellCyclePhasesTop.html#cb1840-18"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1840-19"><a href="cellCyclePhasesTop.html#cb1840-19"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cluster-marker-genes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-and-abundance-between-conditions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/cellCyclePhases.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
