<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 19 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 19 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 19 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"/>
<link rel="next" href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-11"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-12"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-13"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html"><i class="fa fa-check"></i><b>24</b> Extract T-cells from HCA Dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#variance-modeling"><i class="fa fa-check"></i><b>24.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.2" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#data-integration"><i class="fa fa-check"></i><b>24.2</b> Data integration</a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#clustering-1"><i class="fa fa-check"></i><b>24.4</b> Clustering</a></li>
<li class="chapter" data-level="24.5" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#cell-type-classification"><i class="fa fa-check"></i><b>24.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.6" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#extract-t-cells"><i class="fa fa-check"></i><b>24.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.7" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#ackowledgements"><i class="fa fa-check"></i><b>24.7</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.8" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#session-information-14"><i class="fa fa-check"></i><b>24.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html"><i class="fa fa-check"></i><b>25</b> Setting up the data</a><ul>
<li class="chapter" data-level="25.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>25.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="25.2" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>25.2</b> Find temporally expressed genes</a><ul>
<li class="chapter" data-level="25.2.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>25.2.1</b> Pseudotime analysis for another HCA sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="exercise-1.html"><a href="exercise-1.html"><i class="fa fa-check"></i><b>26</b> Exercise 1</a><ul>
<li class="chapter" data-level="26.1" data-path="exercise-1.html"><a href="exercise-1.html#ackowledgements-1"><i class="fa fa-check"></i><b>26.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="26.2" data-path="exercise-1.html"><a href="exercise-1.html#session-information-15"><i class="fa fa-check"></i><b>26.2</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dataSetIntegration_allSetsTop" class="section level1">
<h1><span class="header-section-number">Chapter 19</span> Data integration - all Caron sample types</h1>
<p><strong>CODE NEEDS UPDATING</strong>
<strong>as were other daya set integration chapters with fewr sample types</strong></p>
<div class="sourceCode" id="cb1483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1483-1"><a href="dataSetIntegration-allSetsTop.html#cb1483-1"></a>projDir &lt;-<span class="st"> &quot;/mnt/scratcha/bioinformatics/baller01/20200511_FernandesM_ME_crukBiSs2020&quot;</span></span>
<span id="cb1483-2"><a href="dataSetIntegration-allSetsTop.html#cb1483-2"></a>outDirBit &lt;-<span class="st"> &quot;AnaWiSce/Attempt1&quot;</span></span>
<span id="cb1483-3"><a href="dataSetIntegration-allSetsTop.html#cb1483-3"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span></code></pre></div>
<div class="sourceCode" id="cb1484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1484-1"><a href="dataSetIntegration-allSetsTop.html#cb1484-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb1484-2"><a href="dataSetIntegration-allSetsTop.html#cb1484-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb1484-3"><a href="dataSetIntegration-allSetsTop.html#cb1484-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb1484-4"><a href="dataSetIntegration-allSetsTop.html#cb1484-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb1484-5"><a href="dataSetIntegration-allSetsTop.html#cb1484-5"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb1484-6"><a href="dataSetIntegration-allSetsTop.html#cb1484-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb1484-7"><a href="dataSetIntegration-allSetsTop.html#cb1484-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb1484-8"><a href="dataSetIntegration-allSetsTop.html#cb1484-8"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1484-9"><a href="dataSetIntegration-allSetsTop.html#cb1484-9"></a></span>
<span id="cb1484-10"><a href="dataSetIntegration-allSetsTop.html#cb1484-10"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1484-11"><a href="dataSetIntegration-allSetsTop.html#cb1484-11"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1484-12"><a href="dataSetIntegration-allSetsTop.html#cb1484-12"></a>    splSetToGet &lt;-<span class="st"> &quot;dummy&quot;</span></span>
<span id="cb1484-13"><a href="dataSetIntegration-allSetsTop.html#cb1484-13"></a>    setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb1484-14"><a href="dataSetIntegration-allSetsTop.html#cb1484-14"></a>}</span>
<span id="cb1484-15"><a href="dataSetIntegration-allSetsTop.html#cb1484-15"></a></span>
<span id="cb1484-16"><a href="dataSetIntegration-allSetsTop.html#cb1484-16"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1484-17"><a href="dataSetIntegration-allSetsTop.html#cb1484-17"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1484-18"><a href="dataSetIntegration-allSetsTop.html#cb1484-18"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1484-19"><a href="dataSetIntegration-allSetsTop.html#cb1484-19"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
<p>Source: <a href="https://osca.bioconductor.org/integrating-datasets.html">Integrating Datasets</a> of the OSCA book and the <a href="https://rdrr.io/github/LTLA/batchelor/man/fastMNN.html">fastMNN manual</a>.</p>
<div id="motivation-5" class="section level2">
<h2><span class="header-section-number">19.1</span> Motivation</h2>
<p>Large single-cell RNA sequencing (scRNA-seq) projects usually need to generate data across multiple batches due to logistical constraints. However, the processing of different batches is often subject to uncontrollable differences, e.g., changes in operator, differences in reagent quality. This results in systematic differences in the observed expression in cells from different batches, which we refer to as “batch effects”. Batch effects are problematic as they can be major drivers of heterogeneity in the data, masking the relevant biological differences and complicating interpretation of the results.</p>
<p>Computational correction of these effects is critical for eliminating batch-to-batch variation, allowing data across multiple batches to be combined for common downstream analysis. However, existing methods based on linear models (Ritchie et al. 2015; Leek et al. 2012) assume that the composition of cell populations are either known or the same across batches. To overcome these limitations, bespoke methods have been developed for batch correction of single-cell data (Haghverdi et al. 2018; Butler et al. 2018; Lin et al. 2019) that do not require a priori knowledge about the composition of the population. This allows them to be used in workflows for exploratory analyses of scRNA-seq data where such knowledge is usually unavailable.</p>
</div>
<div id="load-data-7" class="section level2">
<h2><span class="header-section-number">19.2</span> Load data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts.</p>
<div class="sourceCode" id="cb1485"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1485-1"><a href="dataSetIntegration-allSetsTop.html#cb1485-1"></a><span class="co">#setName &lt;- &quot;caron&quot;</span></span>
<span id="cb1485-2"><a href="dataSetIntegration-allSetsTop.html#cb1485-2"></a><span class="co"># Read object in:</span></span>
<span id="cb1485-3"><a href="dataSetIntegration-allSetsTop.html#cb1485-3"></a><span class="co">#setSuf &lt;- &quot;&quot;</span></span>
<span id="cb1485-4"><a href="dataSetIntegration-allSetsTop.html#cb1485-4"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1485-5"><a href="dataSetIntegration-allSetsTop.html#cb1485-5"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_allCells.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1487"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1487-1"><a href="dataSetIntegration-allSetsTop.html#cb1487-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb1487-2"><a href="dataSetIntegration-allSetsTop.html#cb1487-2"></a>{</span>
<span id="cb1487-3"><a href="dataSetIntegration-allSetsTop.html#cb1487-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb1487-4"><a href="dataSetIntegration-allSetsTop.html#cb1487-4"></a>}</span>
<span id="cb1487-5"><a href="dataSetIntegration-allSetsTop.html#cb1487-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1487-6"><a href="dataSetIntegration-allSetsTop.html#cb1487-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1489-1"><a href="dataSetIntegration-allSetsTop.html#cb1489-1"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span> <span class="co"># to avoid error later</span></span>
<span id="cb1489-2"><a href="dataSetIntegration-allSetsTop.html#cb1489-2"></a></span>
<span id="cb1489-3"><a href="dataSetIntegration-allSetsTop.html#cb1489-3"></a><span class="co">#head(rowData(sce))</span></span>
<span id="cb1489-4"><a href="dataSetIntegration-allSetsTop.html#cb1489-4"></a><span class="co">#head(colData(sce))</span></span>
<span id="cb1489-5"><a href="dataSetIntegration-allSetsTop.html#cb1489-5"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>block)</span></code></pre></div>
<pre><code>## 
##      ABMMC ETV6-RUNX1        HHD      PBMMC      PRE-T 
##          0      19488       9988      10885       7469</code></pre>
<div class="sourceCode" id="cb1491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1491-1"><a href="dataSetIntegration-allSetsTop.html#cb1491-1"></a><span class="co">#assayNames(sce)</span></span>
<span id="cb1491-2"><a href="dataSetIntegration-allSetsTop.html#cb1491-2"></a><span class="co">#reducedDimNames(sce)</span></span></code></pre></div>
<p>Read in the sample sheet:</p>
<div class="sourceCode" id="cb1492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1492-1"><a href="dataSetIntegration-allSetsTop.html#cb1492-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb1492-2"><a href="dataSetIntegration-allSetsTop.html#cb1492-2"></a>cb_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/CaronBourque2020/SraRunTable.txt&quot;</span>)</span>
<span id="cb1492-3"><a href="dataSetIntegration-allSetsTop.html#cb1492-3"></a>cb_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(cb_sampleSheetFn, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb1492-4"><a href="dataSetIntegration-allSetsTop.html#cb1492-4"></a>cb_sampleSheet &lt;-<span class="st">  </span>cb_sampleSheet <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>Run <span class="op">==</span><span class="st"> &quot;SRR9264351&quot;</span>)</span>
<span id="cb1492-5"><a href="dataSetIntegration-allSetsTop.html#cb1492-5"></a>cb_sampleSheet</span></code></pre></div>
<pre><code>##           Run Assay.Type AvgSpotLen       Bases  BioProject    BioSample
## 1  SRR9264343    RNA-Seq        132 27850288884 PRJNA548203 SAMN12011162
## 2  SRR9264344    RNA-Seq        132 43613421192 PRJNA548203 SAMN12011172
## 3  SRR9264345    RNA-Seq        132 43838527392 PRJNA548203 SAMN12011171
## 4  SRR9264346    RNA-Seq        132 39752529300 PRJNA548203 SAMN12011170
## 5  SRR9264347    RNA-Seq        132 41035092252 PRJNA548203 SAMN12011169
## 6  SRR9264348    RNA-Seq        132 42840756288 PRJNA548203 SAMN12011168
## 7  SRR9264349    RNA-Seq        132 42953865372 PRJNA548203 SAMN12011167
## 8  SRR9264350    RNA-Seq        132 42822420960 PRJNA548203 SAMN12011166
## 9  SRR9264352    RNA-Seq        132 36199482528 PRJNA548203 SAMN12011165
## 10 SRR9264353    RNA-Seq        132 41446760124 PRJNA548203 SAMN12011164
## 11 SRR9264354    RNA-Seq        132 42802129128 PRJNA548203 SAMN12011163
##          Bytes
## 1  18644549905
## 2  27638885644
## 3  28054431102
## 4  25564104997
## 5  24777477094
## 6  27432674292
## 7  27523442193
## 8  27282064655
## 9  22143300246
## 10 26850120365
## 11 27774281557
##                                                            Cell_type
## 1     Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
## 2     Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
## 3     Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
## 4     Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
## 5  Pre-B High hyper diploid [HHD] acute lymphoblastic leukemia cells
## 6  Pre-B High hyper diploid [HHD] acute lymphoblastic leukemia cells
## 7                           Pre-T acute lymphoblastic leukemia cells
## 8                           Pre-T acute lymphoblastic leukemia cells
## 9                    Healthy pediatric bone marrow mononuclear cells
## 10                   Healthy pediatric bone marrow mononuclear cells
## 11                   Healthy pediatric bone marrow mononuclear cells
##    Center.Name Consent DATASTORE.filetype DATASTORE.provider
## 1          GEO  public          fastq,sra         gs,ncbi,s3
## 2          GEO  public          fastq,sra         gs,ncbi,s3
## 3          GEO  public          fastq,sra         gs,ncbi,s3
## 4          GEO  public          fastq,sra         gs,ncbi,s3
## 5          GEO  public          fastq,sra         gs,ncbi,s3
## 6          GEO  public          fastq,sra         gs,ncbi,s3
## 7          GEO  public          fastq,sra         gs,ncbi,s3
## 8          GEO  public          fastq,sra         gs,ncbi,s3
## 9          GEO  public          fastq,sra         gs,ncbi,s3
## 10         GEO  public          fastq,sra         gs,ncbi,s3
## 11         GEO  public          fastq,sra         gs,ncbi,s3
##                  DATASTORE.region                          disease_state
## 1  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 2  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 3  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 4  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 5  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 6  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 7  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 8  gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 9  gs.US,ncbi.public,s3.us-east-1              Healthy pediatric control
## 10 gs.US,ncbi.public,s3.us-east-1              Healthy pediatric control
## 11 gs.US,ncbi.public,s3.us-east-1              Healthy pediatric control
##    Experiment GEO_Accession..exp.          Instrument LibraryLayout
## 1  SRX6034681          GSM3872434 Illumina HiSeq 4000        PAIRED
## 2  SRX6034682          GSM3872435 Illumina HiSeq 4000        PAIRED
## 3  SRX6034683          GSM3872436 Illumina HiSeq 4000        PAIRED
## 4  SRX6034684          GSM3872437 Illumina HiSeq 4000        PAIRED
## 5  SRX6034685          GSM3872438 Illumina HiSeq 4000        PAIRED
## 6  SRX6034686          GSM3872439 Illumina HiSeq 4000        PAIRED
## 7  SRX6034687          GSM3872440 Illumina HiSeq 4000        PAIRED
## 8  SRX6034688          GSM3872441 Illumina HiSeq 4000        PAIRED
## 9  SRX6034689          GSM3872442 Illumina HiSeq 4000        PAIRED
## 10 SRX6034690          GSM3872443 Illumina HiSeq 4000        PAIRED
## 11 SRX6034691          GSM3872444 Illumina HiSeq 4000        PAIRED
##    LibrarySelection  LibrarySource     Organism Platform          ReleaseDate
## 1              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 2              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 3              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 4              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 5              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 6              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 7              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 8              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 9              cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 10             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 11             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
##    Sample.Name source_name SRA.Study
## 1   GSM3872434  ETV6-RUNX1 SRP201012
## 2   GSM3872435  ETV6-RUNX1 SRP201012
## 3   GSM3872436  ETV6-RUNX1 SRP201012
## 4   GSM3872437  ETV6-RUNX1 SRP201012
## 5   GSM3872438         HHD SRP201012
## 6   GSM3872439         HHD SRP201012
## 7   GSM3872440       PRE-T SRP201012
## 8   GSM3872441       PRE-T SRP201012
## 9   GSM3872442       PBMMC SRP201012
## 10  GSM3872443       PBMMC SRP201012
## 11  GSM3872444       PBMMC SRP201012</code></pre>
<p>Have sample names:</p>
<div class="sourceCode" id="cb1494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1494-1"><a href="dataSetIntegration-allSetsTop.html#cb1494-1"></a>cb_sampleSheet<span class="op">$</span>Sample.Name2 &lt;-<span class="st"> &quot;NA&quot;</span></span>
<span id="cb1494-2"><a href="dataSetIntegration-allSetsTop.html#cb1494-2"></a>typeVec &lt;-<span class="st"> </span><span class="kw">unique</span>(cb_sampleSheet<span class="op">$</span>source_name)</span>
<span id="cb1494-3"><a href="dataSetIntegration-allSetsTop.html#cb1494-3"></a><span class="cf">for</span> (tx <span class="cf">in</span> typeVec)</span>
<span id="cb1494-4"><a href="dataSetIntegration-allSetsTop.html#cb1494-4"></a>{</span>
<span id="cb1494-5"><a href="dataSetIntegration-allSetsTop.html#cb1494-5"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(cb_sampleSheet<span class="op">$</span>source_name <span class="op">==</span><span class="st"> </span>tx)</span>
<span id="cb1494-6"><a href="dataSetIntegration-allSetsTop.html#cb1494-6"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(tmpInd))</span>
<span id="cb1494-7"><a href="dataSetIntegration-allSetsTop.html#cb1494-7"></a>    {</span>
<span id="cb1494-8"><a href="dataSetIntegration-allSetsTop.html#cb1494-8"></a>        cb_sampleSheet<span class="op">$</span>Sample.Name2[tmpInd[i]] &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s_%s&quot;</span>, tx, i)</span>
<span id="cb1494-9"><a href="dataSetIntegration-allSetsTop.html#cb1494-9"></a>    }</span>
<span id="cb1494-10"><a href="dataSetIntegration-allSetsTop.html#cb1494-10"></a>}</span>
<span id="cb1494-11"><a href="dataSetIntegration-allSetsTop.html#cb1494-11"></a></span>
<span id="cb1494-12"><a href="dataSetIntegration-allSetsTop.html#cb1494-12"></a><span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name2 &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1494-13"><a href="dataSetIntegration-allSetsTop.html#cb1494-13"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1494-14"><a href="dataSetIntegration-allSetsTop.html#cb1494-14"></a><span class="st">    </span><span class="kw">left_join</span>( cb_sampleSheet, <span class="dt">by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1494-15"><a href="dataSetIntegration-allSetsTop.html#cb1494-15"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name2)</span>
<span id="cb1494-16"><a href="dataSetIntegration-allSetsTop.html#cb1494-16"></a></span>
<span id="cb1494-17"><a href="dataSetIntegration-allSetsTop.html#cb1494-17"></a>splVec &lt;-<span class="st"> </span>cb_sampleSheet <span class="op">%&gt;%</span></span>
<span id="cb1494-18"><a href="dataSetIntegration-allSetsTop.html#cb1494-18"></a><span class="st">    </span><span class="co">#filter(source_name == &quot;ETV6-RUNX1&quot;) %&gt;%</span></span>
<span id="cb1494-19"><a href="dataSetIntegration-allSetsTop.html#cb1494-19"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name2)</span>
<span id="cb1494-20"><a href="dataSetIntegration-allSetsTop.html#cb1494-20"></a>splVec</span></code></pre></div>
<pre><code>##  [1] &quot;ETV6-RUNX1_1&quot; &quot;ETV6-RUNX1_2&quot; &quot;ETV6-RUNX1_3&quot; &quot;ETV6-RUNX1_4&quot; &quot;HHD_1&quot;       
##  [6] &quot;HHD_2&quot;        &quot;PRE-T_1&quot;      &quot;PRE-T_2&quot;      &quot;PBMMC_1&quot;      &quot;PBMMC_2&quot;     
## [11] &quot;PBMMC_3&quot;</code></pre>
<!--
Subset cells (e.g 500 or 1000 per sample for a faster analysis):
-->
<div class="sourceCode" id="cb1496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1496-1"><a href="dataSetIntegration-allSetsTop.html#cb1496-1"></a><span class="co"># mind we now have a downsampled set to use all along</span></span>
<span id="cb1496-2"><a href="dataSetIntegration-allSetsTop.html#cb1496-2"></a><span class="co"># so avoid doing it again</span></span>
<span id="cb1496-3"><a href="dataSetIntegration-allSetsTop.html#cb1496-3"></a><span class="co"># also, changes setsuf to shorter form eg _5hCps, which we could use from the start </span><span class="al">TODO</span></span>
<span id="cb1496-4"><a href="dataSetIntegration-allSetsTop.html#cb1496-4"></a></span>
<span id="cb1496-5"><a href="dataSetIntegration-allSetsTop.html#cb1496-5"></a>all.sce &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1496-6"><a href="dataSetIntegration-allSetsTop.html#cb1496-6"></a></span>
<span id="cb1496-7"><a href="dataSetIntegration-allSetsTop.html#cb1496-7"></a><span class="cf">if</span>(setSuf <span class="op">==</span><span class="st"> &quot;_allCells&quot;</span> <span class="op">|</span><span class="st"> </span>setSuf <span class="op">==</span><span class="st"> &quot;_5hCellPerSpl&quot;</span>)</span>
<span id="cb1496-8"><a href="dataSetIntegration-allSetsTop.html#cb1496-8"></a>{</span>
<span id="cb1496-9"><a href="dataSetIntegration-allSetsTop.html#cb1496-9"></a>  <span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb1496-10"><a href="dataSetIntegration-allSetsTop.html#cb1496-10"></a>  {</span>
<span id="cb1496-11"><a href="dataSetIntegration-allSetsTop.html#cb1496-11"></a>    vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1496-12"><a href="dataSetIntegration-allSetsTop.html#cb1496-12"></a><span class="st">        </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1496-13"><a href="dataSetIntegration-allSetsTop.html#cb1496-13"></a><span class="st">        </span><span class="kw">filter</span>(Sample.Name2 <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb1496-14"><a href="dataSetIntegration-allSetsTop.html#cb1496-14"></a><span class="st">        </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1496-15"><a href="dataSetIntegration-allSetsTop.html#cb1496-15"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1496-16"><a href="dataSetIntegration-allSetsTop.html#cb1496-16"></a>    all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb1496-17"><a href="dataSetIntegration-allSetsTop.html#cb1496-17"></a>  }</span>
<span id="cb1496-18"><a href="dataSetIntegration-allSetsTop.html#cb1496-18"></a>} <span class="cf">else</span> {  </span>
<span id="cb1496-19"><a href="dataSetIntegration-allSetsTop.html#cb1496-19"></a>  nbCells &lt;-<span class="st"> </span><span class="dv">500</span></span>
<span id="cb1496-20"><a href="dataSetIntegration-allSetsTop.html#cb1496-20"></a>  setSuf &lt;-<span class="st"> &quot;_5hCps&quot;</span> <span class="co"># &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb1496-21"><a href="dataSetIntegration-allSetsTop.html#cb1496-21"></a></span>
<span id="cb1496-22"><a href="dataSetIntegration-allSetsTop.html#cb1496-22"></a>  <span class="co">#nbCells &lt;- 1000</span></span>
<span id="cb1496-23"><a href="dataSetIntegration-allSetsTop.html#cb1496-23"></a>  <span class="co">#setSuf &lt;- &quot;_1kCps&quot; # &quot;_1kCellPerSpl&quot;</span></span>
<span id="cb1496-24"><a href="dataSetIntegration-allSetsTop.html#cb1496-24"></a></span>
<span id="cb1496-25"><a href="dataSetIntegration-allSetsTop.html#cb1496-25"></a>  <span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb1496-26"><a href="dataSetIntegration-allSetsTop.html#cb1496-26"></a>  {</span>
<span id="cb1496-27"><a href="dataSetIntegration-allSetsTop.html#cb1496-27"></a>    vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1496-28"><a href="dataSetIntegration-allSetsTop.html#cb1496-28"></a><span class="st">        </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1496-29"><a href="dataSetIntegration-allSetsTop.html#cb1496-29"></a><span class="st">        </span><span class="kw">filter</span>(Sample.Name2 <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb1496-30"><a href="dataSetIntegration-allSetsTop.html#cb1496-30"></a><span class="st">        </span><span class="kw">slice_sample</span>(<span class="dt">n=</span>nbCells) <span class="op">%&gt;%</span></span>
<span id="cb1496-31"><a href="dataSetIntegration-allSetsTop.html#cb1496-31"></a><span class="st">        </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1496-32"><a href="dataSetIntegration-allSetsTop.html#cb1496-32"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1496-33"><a href="dataSetIntegration-allSetsTop.html#cb1496-33"></a>    all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb1496-34"><a href="dataSetIntegration-allSetsTop.html#cb1496-34"></a>  }</span>
<span id="cb1496-35"><a href="dataSetIntegration-allSetsTop.html#cb1496-35"></a>}</span>
<span id="cb1496-36"><a href="dataSetIntegration-allSetsTop.html#cb1496-36"></a></span>
<span id="cb1496-37"><a href="dataSetIntegration-allSetsTop.html#cb1496-37"></a><span class="co"># show size of sets:</span></span>
<span id="cb1496-38"><a href="dataSetIntegration-allSetsTop.html#cb1496-38"></a><span class="kw">lapply</span>(all.sce, dim)</span></code></pre></div>
<pre><code>## $`ETV6-RUNX1_1`
## [1] 18431  2853
## 
## $`ETV6-RUNX1_2`
## [1] 18431  6615
## 
## $`ETV6-RUNX1_3`
## [1] 18431  4727
## 
## $`ETV6-RUNX1_4`
## [1] 18431  5293
## 
## $HHD_1
## [1] 18431  4551
## 
## $HHD_2
## [1] 18431  5437
## 
## $`PRE-T_1`
## [1] 18431  3841
## 
## $`PRE-T_2`
## [1] 18431  3628
## 
## $PBMMC_1
## [1] 18431  2084
## 
## $PBMMC_2
## [1] 18431  4658
## 
## $PBMMC_3
## [1] 18431  4143</code></pre>
<div class="sourceCode" id="cb1498"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1498-1"><a href="dataSetIntegration-allSetsTop.html#cb1498-1"></a><span class="co"># sizeFactors(all.sce[[1]])</span></span></code></pre></div>
<p>We will analyse each sample separately, namely:</p>
<ul>
<li>normalise counts</li>
<li>model gene expression variance</li>
<li>identify highly variable genes</li>
<li>perform dimensionality reduction</li>
<li>cluster cells</li>
</ul>
<div class="sourceCode" id="cb1499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1499-1"><a href="dataSetIntegration-allSetsTop.html#cb1499-1"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb1499-2"><a href="dataSetIntegration-allSetsTop.html#cb1499-2"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, logNormCounts)</span>
<span id="cb1499-3"><a href="dataSetIntegration-allSetsTop.html#cb1499-3"></a></span>
<span id="cb1499-4"><a href="dataSetIntegration-allSetsTop.html#cb1499-4"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb1499-5"><a href="dataSetIntegration-allSetsTop.html#cb1499-5"></a><span class="kw">library</span>(scran)</span>
<span id="cb1499-6"><a href="dataSetIntegration-allSetsTop.html#cb1499-6"></a>all.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, modelGeneVar)</span>
<span id="cb1499-7"><a href="dataSetIntegration-allSetsTop.html#cb1499-7"></a>all.hvgs &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, getTopHVGs, <span class="dt">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb1499-8"><a href="dataSetIntegration-allSetsTop.html#cb1499-8"></a></span>
<span id="cb1499-9"><a href="dataSetIntegration-allSetsTop.html#cb1499-9"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb1499-10"><a href="dataSetIntegration-allSetsTop.html#cb1499-10"></a><span class="kw">library</span>(BiocSingular)</span>
<span id="cb1499-11"><a href="dataSetIntegration-allSetsTop.html#cb1499-11"></a><span class="kw">set.seed</span>(<span class="dv">10000</span>)</span>
<span id="cb1499-12"><a href="dataSetIntegration-allSetsTop.html#cb1499-12"></a>all.sce &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="dt">FUN=</span>runPCA, <span class="dt">x=</span>all.sce, <span class="dt">subset_row=</span>all.hvgs, </span>
<span id="cb1499-13"><a href="dataSetIntegration-allSetsTop.html#cb1499-13"></a>    <span class="dt">MoreArgs=</span><span class="kw">list</span>(<span class="dt">ncomponents=</span><span class="dv">25</span>, <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>()), </span>
<span id="cb1499-14"><a href="dataSetIntegration-allSetsTop.html#cb1499-14"></a>    <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb1499-15"><a href="dataSetIntegration-allSetsTop.html#cb1499-15"></a></span>
<span id="cb1499-16"><a href="dataSetIntegration-allSetsTop.html#cb1499-16"></a><span class="co"># TSNE</span></span>
<span id="cb1499-17"><a href="dataSetIntegration-allSetsTop.html#cb1499-17"></a><span class="co">#set.seed(100000)</span></span>
<span id="cb1499-18"><a href="dataSetIntegration-allSetsTop.html#cb1499-18"></a><span class="co">#all.sce &lt;- lapply(all.sce, runTSNE, dimred=&quot;PCA&quot;)</span></span>
<span id="cb1499-19"><a href="dataSetIntegration-allSetsTop.html#cb1499-19"></a></span>
<span id="cb1499-20"><a href="dataSetIntegration-allSetsTop.html#cb1499-20"></a><span class="co"># UMAP</span></span>
<span id="cb1499-21"><a href="dataSetIntegration-allSetsTop.html#cb1499-21"></a><span class="co">#set.seed(1000000)</span></span>
<span id="cb1499-22"><a href="dataSetIntegration-allSetsTop.html#cb1499-22"></a><span class="co">#all.sce &lt;- lapply(all.sce, runUMAP, dimred=&quot;PCA&quot;)</span></span>
<span id="cb1499-23"><a href="dataSetIntegration-allSetsTop.html#cb1499-23"></a></span>
<span id="cb1499-24"><a href="dataSetIntegration-allSetsTop.html#cb1499-24"></a><span class="co">#--- clustering ---#</span></span>
<span id="cb1499-25"><a href="dataSetIntegration-allSetsTop.html#cb1499-25"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="kw">names</span>(all.sce)) {</span>
<span id="cb1499-26"><a href="dataSetIntegration-allSetsTop.html#cb1499-26"></a>    g &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(all.sce[[n]], <span class="dt">k=</span><span class="dv">10</span>, <span class="dt">use.dimred=</span><span class="st">&#39;PCA&#39;</span>)</span>
<span id="cb1499-27"><a href="dataSetIntegration-allSetsTop.html#cb1499-27"></a>    clust &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(g)<span class="op">$</span>membership</span>
<span id="cb1499-28"><a href="dataSetIntegration-allSetsTop.html#cb1499-28"></a>    <span class="kw">colLabels</span>(all.sce[[n]])  &lt;-<span class="st"> </span><span class="kw">factor</span>(clust)</span>
<span id="cb1499-29"><a href="dataSetIntegration-allSetsTop.html#cb1499-29"></a>}</span></code></pre></div>
<p>To prepare for the batch correction, we need to:</p>
<ul>
<li>subset all batches to the common “universe” of features</li>
<li>rescale each batch to adjust for differences in sequencing depth between batches</li>
<li>perform feature selection by averaging the variance components across all batches</li>
</ul>
<p>We subset all batches to the common “universe” of features. In this case, it is straightforward as both batches use Ensembl gene annotation.</p>
<div class="sourceCode" id="cb1500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1500-1"><a href="dataSetIntegration-allSetsTop.html#cb1500-1"></a>allNames &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){<span class="kw">rownames</span>(x)}))</span>
<span id="cb1500-2"><a href="dataSetIntegration-allSetsTop.html#cb1500-2"></a>allNamesNb &lt;-<span class="st"> </span><span class="kw">table</span>(allNames)</span>
<span id="cb1500-3"><a href="dataSetIntegration-allSetsTop.html#cb1500-3"></a>universe &lt;-<span class="st"> </span><span class="kw">names</span>(allNamesNb)[allNamesNb<span class="op">==</span><span class="kw">length</span>(splVec)] </span>
<span id="cb1500-4"><a href="dataSetIntegration-allSetsTop.html#cb1500-4"></a><span class="kw">length</span>(universe)</span></code></pre></div>
<pre><code>## [1] 18431</code></pre>
<div class="sourceCode" id="cb1502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1502-1"><a href="dataSetIntegration-allSetsTop.html#cb1502-1"></a><span class="co"># Subsetting the SingleCellExperiment object.</span></span>
<span id="cb1502-2"><a href="dataSetIntegration-allSetsTop.html#cb1502-2"></a>uni.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){x[universe,]})</span>
<span id="cb1502-3"><a href="dataSetIntegration-allSetsTop.html#cb1502-3"></a><span class="co"># Also subsetting the variance modelling results, for convenience.</span></span>
<span id="cb1502-4"><a href="dataSetIntegration-allSetsTop.html#cb1502-4"></a>uni.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, <span class="cf">function</span>(x){x[universe,]})</span></code></pre></div>
<p>We rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.</p>
<div class="sourceCode" id="cb1503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1503-1"><a href="dataSetIntegration-allSetsTop.html#cb1503-1"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1503-2"><a href="dataSetIntegration-allSetsTop.html#cb1503-2"></a>rescaled.mbn &lt;-<span class="st"> </span><span class="kw">multiBatchNorm</span>(uni.sce, <span class="dt">batch =</span> <span class="st">&quot;Sample.Name2&quot;</span>)</span></code></pre></div>
<p>We perform feature selection by averaging the variance components across all batches with the combineVar() function. We compute the average as it is responsive to batch-specific HVGs while still preserving the within-batch ranking of genes. This allows us to use the same strategies described in Section 8.3 to select genes of interest. In contrast, approaches based on taking the intersection or union of HVGs across batches become increasingly conservative or liberal, respectively, with an increasing number of batches.</p>
<div class="sourceCode" id="cb1504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1504-1"><a href="dataSetIntegration-allSetsTop.html#cb1504-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1504-2"><a href="dataSetIntegration-allSetsTop.html#cb1504-2"></a>combined.dec &lt;-<span class="st"> </span><span class="kw">combineVar</span>(</span>
<span id="cb1504-3"><a href="dataSetIntegration-allSetsTop.html#cb1504-3"></a>  uni.dec[[<span class="dv">1</span>]], uni.dec[[<span class="dv">2</span>]], uni.dec[[<span class="dv">3</span>]], uni.dec[[<span class="dv">4</span>]],</span>
<span id="cb1504-4"><a href="dataSetIntegration-allSetsTop.html#cb1504-4"></a>  uni.dec[[<span class="dv">5</span>]], uni.dec[[<span class="dv">6</span>]],</span>
<span id="cb1504-5"><a href="dataSetIntegration-allSetsTop.html#cb1504-5"></a>  uni.dec[[<span class="dv">7</span>]], uni.dec[[<span class="dv">8</span>]],</span>
<span id="cb1504-6"><a href="dataSetIntegration-allSetsTop.html#cb1504-6"></a>  uni.dec[[<span class="dv">9</span>]], uni.dec[[<span class="dv">10</span>]], uni.dec[[<span class="dv">11</span>]]</span>
<span id="cb1504-7"><a href="dataSetIntegration-allSetsTop.html#cb1504-7"></a>  )</span>
<span id="cb1504-8"><a href="dataSetIntegration-allSetsTop.html#cb1504-8"></a>chosen.hvgs &lt;-<span class="st"> </span>combined.dec<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb1504-9"><a href="dataSetIntegration-allSetsTop.html#cb1504-9"></a><span class="kw">sum</span>(chosen.hvgs)</span></code></pre></div>
<pre><code>## [1] 12466</code></pre>
<p>When integrating datasets of variable composition, it is generally safer to err on the side of including more genes than are used in a single dataset analysis, to ensure that markers are retained for any dataset-specific subpopulations that might be present. For a top X selection, this means using a larger X (say, ~5000), or in this case, we simply take all genes above the trend. That said, many of the signal-to-noise considerations described in Section 8.3 still apply here, so some experimentation may be necessary for best results.</p>
<p>Alternatively, a more forceful approach to feature selection can be used based on marker genes from within-batch comparisons.</p>
</div>
<div id="diagnosing-batch-effects-5" class="section level2">
<h2><span class="header-section-number">19.3</span> Diagnosing batch effects</h2>
<p>Before we actually perform any correction, it is worth examining whether there is any batch effect in this dataset. We combine the SingleCellExperiments and perform a PCA on the log-expression values for all genes with positive (average) biological components.</p>
<div class="sourceCode" id="cb1506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1506-1"><a href="dataSetIntegration-allSetsTop.html#cb1506-1"></a><span class="co"># Synchronizing the metadata for cbind()ing.</span></span>
<span id="cb1506-2"><a href="dataSetIntegration-allSetsTop.html#cb1506-2"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[2]]))</span></span>
<span id="cb1506-3"><a href="dataSetIntegration-allSetsTop.html#cb1506-3"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[3]]))</span></span>
<span id="cb1506-4"><a href="dataSetIntegration-allSetsTop.html#cb1506-4"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[4]]))</span></span>
<span id="cb1506-5"><a href="dataSetIntegration-allSetsTop.html#cb1506-5"></a></span>
<span id="cb1506-6"><a href="dataSetIntegration-allSetsTop.html#cb1506-6"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(rescaled.mbn, <span class="cf">function</span>(x){x<span class="op">$</span>batch &lt;-<span class="st"> </span>x<span class="op">$</span>Sample.Name2; x})</span>
<span id="cb1506-7"><a href="dataSetIntegration-allSetsTop.html#cb1506-7"></a>rescaled.mbn &lt;-<span class="st"> </span>rescaled2</span>
<span id="cb1506-8"><a href="dataSetIntegration-allSetsTop.html#cb1506-8"></a><span class="kw">rm</span>(rescaled2)</span>
<span id="cb1506-9"><a href="dataSetIntegration-allSetsTop.html#cb1506-9"></a></span>
<span id="cb1506-10"><a href="dataSetIntegration-allSetsTop.html#cb1506-10"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">do.call</span>(cbind, rescaled.mbn)</span>
<span id="cb1506-11"><a href="dataSetIntegration-allSetsTop.html#cb1506-11"></a></span>
<span id="cb1506-12"><a href="dataSetIntegration-allSetsTop.html#cb1506-12"></a><span class="co"># Using RandomParam() as it is more efficient for file-backed matrices.</span></span>
<span id="cb1506-13"><a href="dataSetIntegration-allSetsTop.html#cb1506-13"></a><span class="kw">library</span>(scater)</span>
<span id="cb1506-14"><a href="dataSetIntegration-allSetsTop.html#cb1506-14"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb1506-15"><a href="dataSetIntegration-allSetsTop.html#cb1506-15"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb1506-16"><a href="dataSetIntegration-allSetsTop.html#cb1506-16"></a>  uncorrected,</span>
<span id="cb1506-17"><a href="dataSetIntegration-allSetsTop.html#cb1506-17"></a>  <span class="dt">subset_row=</span>chosen.hvgs,</span>
<span id="cb1506-18"><a href="dataSetIntegration-allSetsTop.html#cb1506-18"></a>  <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>())</span></code></pre></div>
<p>We use graph-based clustering on the components to obtain a summary of the population structure. As our each sample group is represented by at least two replicates, each cluster should ideally consist of cells from several batches. This is the case for some but not all clusters. Some clusters comprise of cells from a single sample. This may indicate that cells of the same type are artificially separated due to technical differences between batches. They may also be cancer cell population private to samples.</p>
<div class="sourceCode" id="cb1507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1507-1"><a href="dataSetIntegration-allSetsTop.html#cb1507-1"></a>nSteps &lt;-<span class="st"> </span><span class="dv">4</span> <span class="co"># waltrap default</span></span>
<span id="cb1507-2"><a href="dataSetIntegration-allSetsTop.html#cb1507-2"></a><span class="cf">if</span>(<span class="kw">ncol</span>(uncorrected)<span class="op">&gt;</span><span class="dv">10000</span>){nSteps &lt;-<span class="st"> </span><span class="dv">20</span>} <span class="co"># or else get 63 clusters with caron allCells</span></span>
<span id="cb1507-3"><a href="dataSetIntegration-allSetsTop.html#cb1507-3"></a><span class="kw">library</span>(scran)</span>
<span id="cb1507-4"><a href="dataSetIntegration-allSetsTop.html#cb1507-4"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(uncorrected, <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1507-5"><a href="dataSetIntegration-allSetsTop.html#cb1507-5"></a>clusters &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr, <span class="dt">steps=</span>nSteps)<span class="op">$</span>membership</span>
<span id="cb1507-6"><a href="dataSetIntegration-allSetsTop.html#cb1507-6"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters, <span class="dt">Batch=</span>uncorrected<span class="op">$</span>batch)</span>
<span id="cb1507-7"><a href="dataSetIntegration-allSetsTop.html#cb1507-7"></a>tab</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##      1             2           11          143          458     6    13       7
##      2             0            0            0            0     0     0       0
##      3             2            0           18           16     0     3      46
##      4          2698            3            0            0     0     0       0
##      5             0            0            0            0     0     0       0
##      6             0            1            0            0     1     0     408
##      7            25          220         1697          333   358   162     278
##      8             0            0            0            0     1  5085       0
##      9             1            3           74          140     1    11      12
##      10            0            0            0            0  3858    94      10
##      11            0            1            0            0     0     0       0
##      12            4            3           54          334     0     6       0
##      13          112         6226           76           54    30     3      13
##      14            0            0            0            0     0     0       0
##      15            0            0            0            1     0     0       0
##      16            0            0            0         3651     0     0       0
##      17            1            3            5            7     4     4     248
##      18            0            0           37           98     0     1       0
##      19            0            4           19           35    48     8       2
##      20            1            0            0            0     0     1     357
##      21            1            6            7            5    11     2     209
##      22            0            0            0            0     0     0       3
##      23            6          102          460          141   231    38      60
##      24            0            0            0            3     0     0       0
##      25            0           11         2126           10     0     0       0
##      26            0            0            0            0     0     0       0
##      27            0            0            0            0     0     0       0
##      28            0           21           10            5     2     6      15
##      29            0            0            1            2     0     0       0
##      30            0            0            0            0     0     0     416
##      31            0            0            0            0     0     0       0
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##      1      445      51       3       6
##      2      244       0       0       0
##      3      265      49       1      45
##      4        0       0       0       0
##      5      601       2       0       0
##      6       55     800       2       6
##      7       13    1273      53     602
##      8        0       0       0       0
##      9      309      75       0      45
##      10       0       2       1       0
##      11       0       1       6    2468
##      12     626      11       1      14
##      13      24      37       3      12
##      14       0       0    2298       0
##      15     367       1       0       0
##      16       0       0       0       0
##      17      41     696       3     228
##      18      36       6       1       1
##      19       3      39       3       9
##      20       2     340       1       0
##      21      15     207       3      44
##      22       0       0     995       9
##      23       6     519      24     114
##      24     352       0       0       0
##      25       0       0       0       0
##      26      38       0       0       0
##      27       0       0     442       0
##      28      30      33       1      24
##      29    1162       0       0       0
##      30       0       1       0       1
##      31      24       0       0       0</code></pre>
<div class="sourceCode" id="cb1509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1509-1"><a href="dataSetIntegration-allSetsTop.html#cb1509-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab,</span>
<span id="cb1509-2"><a href="dataSetIntegration-allSetsTop.html#cb1509-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1509-3"><a href="dataSetIntegration-allSetsTop.html#cb1509-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1509-4"><a href="dataSetIntegration-allSetsTop.html#cb1509-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1509-5"><a href="dataSetIntegration-allSetsTop.html#cb1509-5"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-492-1.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The strong separation between cells from different batches is consistent with the clustering results.</p>
<div class="sourceCode" id="cb1510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1510-1"><a href="dataSetIntegration-allSetsTop.html#cb1510-1"></a><span class="kw">set.seed</span>(<span class="dv">1111001</span>)</span>
<span id="cb1510-2"><a href="dataSetIntegration-allSetsTop.html#cb1510-2"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(uncorrected, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1510-3"><a href="dataSetIntegration-allSetsTop.html#cb1510-3"></a><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-493-1.png" width="672" /></p>
<div class="sourceCode" id="cb1511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1511-1"><a href="dataSetIntegration-allSetsTop.html#cb1511-1"></a><span class="kw">set.seed</span>(<span class="dv">1111001</span>)</span>
<span id="cb1511-2"><a href="dataSetIntegration-allSetsTop.html#cb1511-2"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(uncorrected, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1511-3"><a href="dataSetIntegration-allSetsTop.html#cb1511-3"></a><span class="kw">plotUMAP</span>(uncorrected, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-494-1.png" width="672" /></p>
<p>Of course, the other explanation for batch-specific clusters is that there are cell types that are unique to each batch. The degree of intermingling of cells from different batches is not an effective diagnostic when the batches involved might actually contain unique cell subpopulations. If a cluster only contains cells from a single batch, one can always debate whether that is caused by a failure of the correction method or if there is truly a batch-specific subpopulation. For example, do batch-specific metabolic or differentiation states represent distinct subpopulations? Or should they be merged together? We will not attempt to answer this here, only noting that each batch correction algorithm will make different (and possibly inappropriate) decisions on what constitutes “shared” and “unique” populations.</p>
</div>
<div id="linear-regression-5" class="section level2">
<h2><span class="header-section-number">19.4</span> Linear regression</h2>
<p>Batch effects in bulk RNA sequencing studies are commonly removed with linear regression. This involves fitting a linear model to each gene’s expression profile, setting the undesirable batch term to zero and recomputing the observations sans the batch effect, yielding a set of corrected expression values for downstream analyses. Linear modelling is the basis of the removeBatchEffect() function from the limma package (Ritchie et al. 2015) as well the comBat() function from the sva package (Leek et al. 2012).</p>
<p>To use this approach in a scRNA-seq context, we assume that the composition of cell subpopulations is the same across batches. We also assume that the batch effect is additive, i.e., any batch-induced fold-change in expression is the same across different cell subpopulations for any given gene. These are strong assumptions as batches derived from different individuals will naturally exhibit variation in cell type abundances and expression. Nonetheless, they may be acceptable when dealing with batches that are technical replicates generated from the same population of cells. (In fact, when its assumptions hold, linear regression is the most statistically efficient as it uses information from all cells to compute the common batch vector.) Linear modelling can also accommodate situations where the composition is known a priori by including the cell type as a factor in the linear model, but this situation is even less common.</p>
<p>We use the rescaleBatches() function from the batchelor package to remove the batch effect. This is roughly equivalent to applying a linear regression to the log-expression values per gene, with some adjustments to improve performance and efficiency. For each gene, the mean expression in each batch is scaled down until it is equal to the lowest mean across all batches. We deliberately choose to scale all expression values down as this mitigates differences in variance when batches lie at different positions on the mean-variance trend. (Specifically, the shrinkage effect of the pseudo-count is greater for smaller counts, suppressing any differences in variance across batches.) An additional feature of rescaleBatches() is that it will preserve sparsity in the input matrix for greater efficiency, whereas other methods like removeBatchEffect() will always return a dense matrix.</p>
<div class="sourceCode" id="cb1512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1512-1"><a href="dataSetIntegration-allSetsTop.html#cb1512-1"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1512-2"><a href="dataSetIntegration-allSetsTop.html#cb1512-2"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">rescaleBatches</span>(rescaled.mbn)</span>
<span id="cb1512-3"><a href="dataSetIntegration-allSetsTop.html#cb1512-3"></a>rescaled.rb</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(1): corrected
## rownames(18431): ENSG00000000003 ENSG00000000419 ... ENSG00000285486
##   ENSG00000285492
## rowData names(0):
## colnames: NULL
## colData names(1): batch
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>After clustering, we observe fewer clusters and these consist of mixtures of cells from the several replicates, consistent with the removal of the batch effect. This conclusion is supported by the apparent mixing of cells from different batches on the TSNE plot below. However, at least one batch-specific cluster is still present, indicating that the correction is not entirely complete. This is attributable to violation of one of the aforementioned assumptions, even in this simple case involving replicated batches.</p>
<div class="sourceCode" id="cb1514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1514-1"><a href="dataSetIntegration-allSetsTop.html#cb1514-1"></a><span class="kw">set.seed</span>(<span class="dv">1010101010</span>) <span class="co"># To ensure reproducibility of IRLBA.</span></span>
<span id="cb1514-2"><a href="dataSetIntegration-allSetsTop.html#cb1514-2"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">runPCA</span>(rescaled.rb, <span class="dt">subset_row=</span>chosen.hvgs, <span class="dt">exprs_values=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1514-3"><a href="dataSetIntegration-allSetsTop.html#cb1514-3"></a></span>
<span id="cb1514-4"><a href="dataSetIntegration-allSetsTop.html#cb1514-4"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(rescaled.rb, <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1514-5"><a href="dataSetIntegration-allSetsTop.html#cb1514-5"></a>clusters.resc &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1514-6"><a href="dataSetIntegration-allSetsTop.html#cb1514-6"></a>rescaled.rb<span class="op">$</span>clusters.resc &lt;-<span class="st"> </span>clusters.resc</span>
<span id="cb1514-7"><a href="dataSetIntegration-allSetsTop.html#cb1514-7"></a>tab.resc &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.resc, <span class="dt">Batch=</span>rescaled.rb<span class="op">$</span>batch)</span>
<span id="cb1514-8"><a href="dataSetIntegration-allSetsTop.html#cb1514-8"></a>tab.resc</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##      1             2            2            4            3     2     2     135
##      2           239          499           55          223   700   225     116
##      3             0            0            0            2     0     0       0
##      4             1            0            9            2     0     2      31
##      5             0            1            2           11     0     0       0
##      6             0            0            0            0     0     0       0
##      7             1           25           22            1   139    25     106
##      8             0            0           19           96     0     0       0
##      9             0            0            0            0     0     0       0
##      10            0            0          117            0     0     0       7
##      11           16          210         1197          285   182    87     313
##      12            3           34          209           82   167    34      30
##      13            0            0           33           67     0     0       0
##      14            0            2            6            1     0     0      93
##      15            1            0           48            0     0     9       9
##      16            0            0            1          268     0     0       0
##      17            1            3            5            8     4     2     153
##      18          205         1490          542          292   976  1279     388
##      19            0           15            9            5     2     6      12
##      20          167          351          137          294    82   303      47
##      21          100          295          112           57   447   110      26
##      22          131          187           29          178    89    85      42
##      23          243          373          172          659   300   633      29
##      24            0            0           55            0     0     0       0
##      25            0            0            5           69     0     0       0
##      26            0            0            1           63     0     0       1
##      27            1            3            0            0     0    20       0
##      28           23            1            5            3    30     0       2
##      29            9           48          161           66   166    67      75
##      30            0            0            0            1     0     0       0
##      31         1624         2984         1001         2049  1139  2500     117
##      32            1            0            4            0     0     2      12
##      33           12           20            2           12    28     3       8
##      34           68           51          701           14    43    34      23
##      35            0            0            5            0     0     1       5
##      36            0            0            0            0     0     0       0
##      37            0            0            1            1     0     0       0
##      38            0            4           17           11    47     8       2
##      39            0            0            0            0     0     0       0
##      40            0            0            1          360     0     0       0
##      41            0            1            1            0     0     0       0
##      42            0            0           23            0     0     0       1
##      43            0            0            0            0     0     0       0
##      44            0            0            0            0     0     0       0
##      45            0            0            0            0     0     0       0
##      46            0            2           12            8     0     0       0
##      47            0            0            2            1     0     0       0
##      48            0            0            0           72     0     0       0
##      49            0            0            0            0     0     0       0
##      50            5           14            0            0     7     0       0
##      51            0            0            0            0     0     0       0
##      52            0            0            0            0     0     0       0
##      53            0            0            0            0     0     0       0
##      54            0            0            0            4     0     0       0
##      55            0            0            0            0     0     0       0
##      56            0            0            0            0     0     0      43
##      57            0            0            2            1     1     0     258
##      58            0            0            0           24     0     0       0
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##      1       53     245       1      46
##      2       13      63    1339     638
##      3      100       0       0       0
##      4        0      19       1      25
##      5       60      16       0       1
##      6       95       0       0       0
##      7       24     118       0     143
##      8      112       0       0       0
##      9       72       0       0       0
##      10       0      50       0       5
##      11     183     881      26     314
##      12     255      81      21     101
##      13      30      13       0       1
##      14       2     164       0       0
##      15       0      40       0      19
##      16      13       0       0       0
##      17      83     522       1     188
##      18     122     800     185     202
##      19       9      29       0      18
##      20       5      27      98      41
##      21      25      29     810     203
##      22       1      30      47      13
##      23       8      24     304      60
##      24       0       8       0       1
##      25       8       3       0       0
##      26      20       1       0       0
##      27       0       0       0       0
##      28      17       9       0       3
##      29     295     265      28     265
##      30      64       0       0       0
##      31     333     535     599    1272
##      32       0      20       1      27
##      33       0       1     336       0
##      34       4      56      34       1
##      35      53      23       0       4
##      36      59       0       0       0
##      37     115       0       0       0
##      38       7      18       2       0
##      39     179       0       0       0
##      40      16       0       0       0
##      41      14       4       1       3
##      42       0      17       0      17
##      43      98       0       0       0
##      44     125       0       0       0
##      45      20       0       0       0
##      46     863       3       0       0
##      47      11       1       0       3
##      48     610       0       0       0
##      49      41       0       0       0
##      50       0       0       4       0
##      51      59       0       0       0
##      52      19       0       0       0
##      53      38       0       0       0
##      54     321       0       0       0
##      55       0       1       0      13
##      56       0      23       0       0
##      57       4       4       0       1
##      58       0       0       3       0</code></pre>
<div class="sourceCode" id="cb1516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1516-1"><a href="dataSetIntegration-allSetsTop.html#cb1516-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.resc,</span>
<span id="cb1516-2"><a href="dataSetIntegration-allSetsTop.html#cb1516-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1516-3"><a href="dataSetIntegration-allSetsTop.html#cb1516-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1516-4"><a href="dataSetIntegration-allSetsTop.html#cb1516-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1516-5"><a href="dataSetIntegration-allSetsTop.html#cb1516-5"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-497-1.png" width="672" /></p>
<!-- and/or barplots -->
<div class="sourceCode" id="cb1517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1517-1"><a href="dataSetIntegration-allSetsTop.html#cb1517-1"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(rescaled.rb, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1517-2"><a href="dataSetIntegration-allSetsTop.html#cb1517-2"></a>rescaled.rb<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(rescaled.rb<span class="op">$</span>batch)</span>
<span id="cb1517-3"><a href="dataSetIntegration-allSetsTop.html#cb1517-3"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(rescaled.rb, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span>
<span id="cb1517-4"><a href="dataSetIntegration-allSetsTop.html#cb1517-4"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(rescaled.rb, <span class="dt">colour_by=</span><span class="st">&quot;clusters.resc&quot;</span>)</span>
<span id="cb1517-5"><a href="dataSetIntegration-allSetsTop.html#cb1517-5"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p.clu, p.batch, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-498-1.png" width="806.4" /></p>
</div>
<div id="performing-mnn-correction-5" class="section level2">
<h2><span class="header-section-number">19.5</span> Performing MNN correction</h2>
<div id="algorithm-overview-5" class="section level3">
<h3><span class="header-section-number">19.5.1</span> Algorithm overview</h3>
<p>Consider a cell a in batch A, and identify the cells in batch B that are nearest neighbors to a in the expression space defined by the selected features. Repeat this for a cell b in batch B, identifying its nearest neighbors in A. Mutual nearest neighbors (MNNs) are pairs of cells from different batches that belong in each other’s set of nearest neighbors. The reasoning is that MNN pairs represent cells from the same biological state prior to the application of a batch effect - see Haghverdi et al. (2018) for full theoretical details. Thus, the difference between cells in MNN pairs can be used as an estimate of the batch effect, the subtraction of which yields batch-corrected values.</p>
<p>Compared to linear regression, MNN correction does not assume that the population composition is the same or known beforehand. This is because it learns the shared population structure via identification of MNN pairs and uses this information to obtain an appropriate estimate of the batch effect. Instead, the key assumption of MNN-based approaches is that the batch effect is orthogonal to the biology in high-dimensional expression space. Violations reduce the effectiveness and accuracy of the correction, with the most common case arising from variations in the direction of the batch effect between clusters. Nonetheless, the assumption is usually reasonable as a random vector is very likely to be orthogonal in high-dimensional space.</p>
</div>
<div id="application-to-the-data-5" class="section level3">
<h3><span class="header-section-number">19.5.2</span> Application to the data</h3>
<p>The batchelor package provides an implementation of the MNN approach via the fastMNN() function. Unlike the MNN method originally described by Haghverdi et al. (2018), the fastMNN() function performs PCA to reduce the dimensions beforehand and speed up the downstream neighbor detection steps.</p>
<p>We apply it to our two PBMC batches to remove the batch effect across the highly variable genes in chosen.hvgs. To reduce computational work and technical noise, all cells in all batches are projected into the low-dimensional space defined by the top d principal components. Identification of MNNs and calculation of correction vectors are then performed in this low-dimensional space.</p>
<div class="sourceCode" id="cb1518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1518-1"><a href="dataSetIntegration-allSetsTop.html#cb1518-1"></a><span class="co"># Using randomized SVD here, as this is faster than </span></span>
<span id="cb1518-2"><a href="dataSetIntegration-allSetsTop.html#cb1518-2"></a><span class="co"># irlba for file-backed matrices.</span></span>
<span id="cb1518-3"><a href="dataSetIntegration-allSetsTop.html#cb1518-3"></a><span class="kw">set.seed</span>(<span class="dv">1000101001</span>)</span>
<span id="cb1518-4"><a href="dataSetIntegration-allSetsTop.html#cb1518-4"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(</span>
<span id="cb1518-5"><a href="dataSetIntegration-allSetsTop.html#cb1518-5"></a>  rescaled.mbn,</span>
<span id="cb1518-6"><a href="dataSetIntegration-allSetsTop.html#cb1518-6"></a>  <span class="dt">auto.merge=</span><span class="ot">TRUE</span>,</span>
<span id="cb1518-7"><a href="dataSetIntegration-allSetsTop.html#cb1518-7"></a>  <span class="dt">d=</span><span class="dv">50</span>,</span>
<span id="cb1518-8"><a href="dataSetIntegration-allSetsTop.html#cb1518-8"></a>  <span class="dt">k=</span><span class="dv">20</span>,</span>
<span id="cb1518-9"><a href="dataSetIntegration-allSetsTop.html#cb1518-9"></a>  <span class="dt">subset.row=</span>chosen.hvgs,</span>
<span id="cb1518-10"><a href="dataSetIntegration-allSetsTop.html#cb1518-10"></a>  <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>)</span>
<span id="cb1518-11"><a href="dataSetIntegration-allSetsTop.html#cb1518-11"></a>  )</span>
<span id="cb1518-12"><a href="dataSetIntegration-allSetsTop.html#cb1518-12"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 12466 47830 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(12466): ENSG00000000003 ENSG00000000457 ... ENSG00000285476
##   ENSG00000285492
## rowData names(1): rotation
## colnames: NULL
## colData names(1): batch
## reducedDimNames(1): corrected
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1520-1"><a href="dataSetIntegration-allSetsTop.html#cb1520-1"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb1520-2"><a href="dataSetIntegration-allSetsTop.html#cb1520-2"></a>mnn.out<span class="op">$</span>type &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch)</span></code></pre></div>
<p>The function returns a SingleCellExperiment object containing corrected values for downstream analyses like clustering or visualization. Each column of mnn.out corresponds to a cell in one of the batches, while each row corresponds to an input gene in chosen.hvgs. The batch field in the column metadata contains a vector specifying the batch of origin of each cell.</p>
<div class="sourceCode" id="cb1521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1521-1"><a href="dataSetIntegration-allSetsTop.html#cb1521-1"></a><span class="kw">head</span>(mnn.out<span class="op">$</span>batch) </span></code></pre></div>
<pre><code>## [1] ETV6-RUNX1_1 ETV6-RUNX1_1 ETV6-RUNX1_1 ETV6-RUNX1_1 ETV6-RUNX1_1
## [6] ETV6-RUNX1_1
## 11 Levels: ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 ... PRE-T_2</code></pre>
<p>The corrected matrix in the reducedDims() contains the low-dimensional corrected coordinates for all cells, which we will use in place of the PCs in our downstream analyses.</p>
<div class="sourceCode" id="cb1523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1523-1"><a href="dataSetIntegration-allSetsTop.html#cb1523-1"></a><span class="kw">dim</span>(<span class="kw">reducedDim</span>(mnn.out, <span class="st">&quot;corrected&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 47830    50</code></pre>
<p>A reconstructed matrix in the assays() contains the corrected expression values for each gene in each cell, obtained by projecting the low-dimensional coordinates in corrected back into gene expression space. We do not recommend using this for anything other than visualization.</p>
<div class="sourceCode" id="cb1525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1525-1"><a href="dataSetIntegration-allSetsTop.html#cb1525-1"></a><span class="kw">dim</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 12466 47830</code></pre>
<div class="sourceCode" id="cb1527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1527-1"><a href="dataSetIntegration-allSetsTop.html#cb1527-1"></a><span class="kw">print</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</span></code></pre></div>
<pre><code>## &lt;5 x 3&gt; matrix of class LowRankMatrix and type &quot;double&quot;:
##                          [,1]          [,2]          [,3]
## ENSG00000000003 -8.970458e-05 -6.411674e-05 -6.896130e-05
## ENSG00000000457  6.004816e-05  5.014359e-04  2.004637e-04
## ENSG00000000938 -3.567196e-04  3.084119e-04  4.221600e-04
## ENSG00000001167  2.328682e-04  7.270576e-04  6.971589e-04
## ENSG00000001461 -8.717612e-04 -6.734327e-04 -7.530539e-04</code></pre>
<p>The most relevant parameter for tuning fastMNN() is k, which specifies the number of nearest neighbors to consider when defining MNN pairs. This can be interpreted as the minimum anticipated frequency of any shared cell type or state in each batch. Increasing k will generally result in more aggressive merging as the algorithm is more generous in matching subpopulations across batches. It can occasionally be desirable to increase k if one clearly sees that the same cell types are not being adequately merged across batches.</p>
<!--
See Chapter 32 for an example of a more complex fastMNN() merge involving several human pancreas datasets generated by different authors on different patients with different technologies.
-->
</div>
<div id="correction-diagnostics-5" class="section level3">
<h3><span class="header-section-number">19.5.3</span> Correction diagnostics</h3>
<p>We cluster on the low-dimensional corrected coordinates to obtain a partitioning of the cells that serves as a proxy for the population structure. If the batch effect is successfully corrected, clusters corresponding to shared cell types or states should contain cells from multiple batches. We see that all clusters contain contributions from each batch after correction, consistent with our expectation that the two batches are replicates of each other.</p>
<div class="sourceCode" id="cb1529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1529-1"><a href="dataSetIntegration-allSetsTop.html#cb1529-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1529-2"><a href="dataSetIntegration-allSetsTop.html#cb1529-2"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(mnn.out, <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1529-3"><a href="dataSetIntegration-allSetsTop.html#cb1529-3"></a>clusters.mnn &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1529-4"><a href="dataSetIntegration-allSetsTop.html#cb1529-4"></a>mnn.out<span class="op">$</span>clusters.mnn &lt;-<span class="st"> </span>clusters.mnn</span>
<span id="cb1529-5"><a href="dataSetIntegration-allSetsTop.html#cb1529-5"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.mnn, <span class="dt">Batch=</span>mnn.out<span class="op">$</span>batch)</span>
<span id="cb1529-6"><a href="dataSetIntegration-allSetsTop.html#cb1529-6"></a>tab.mnn</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##      1           187          265           46          277    91   133      19
##      2             0            0            2           11     0     1      27
##      3             3            1           40          279     0     6       0
##      4             0            0            0            0    13     2       5
##      5            14          177         1546          268   248    98     614
##      6             0            0            0            0     1     0      71
##      7             0            0            6            1     5    12       4
##      8             8            1            6            4    94    80      73
##      9             0            3            4            7     4     2     126
##      10            2          138            3            3   722    41       0
##      11            0            1           25           46     1     3       5
##      12            0            2            3            1    10     1     339
##      13            0            1            3            1     9     0     118
##      14         1675         1800          409         1479  1532  2588     109
##      15            0            0           19           44     0     6       3
##      16           70          186           61           62   165    71      14
##      17            6           32          108           35    88    45      69
##      18            0            0            0            0     0     1      36
##      19            0            0           30           61     0     1       0
##      20            0            1           20           32     0     1       0
##      21            2            0           12           10     0     1      15
##      22            0            1            1            0     6     2       1
##      23            1            1            5            2   217     3      85
##      24            1            1           12           22     0     0       0
##      25            0            8           32          210     6     5       2
##      26          398         3055         1460         1011   665  1360      38
##      27          267          485          144          584   211   550      24
##      28            0           13            2            4     1     5       4
##      29            3           42          236           67    86    16      22
##      30          181          311           63          295   130   342      38
##      31            0            1            1            1     0     0      11
##      32            0            0            1            0     7     0      53
##      33            1            2            2            1     2     0       2
##      34            4           58          215           73   142    21      35
##      35           21            0            4            3     3     1       1
##      36            1            2           28           95     0     8       5
##      37            0            8            8            1     1     1      11
##      38            1            0            5            3     1     1       2
##      39            0            0            2            2     0     1       1
##      40            1            0            2            1     1     0       0
##      41            4           10           39           33    36    17       5
##      42            1            2           16           60     0     0       0
##      43            0            0            5           27     0     0       0
##      44            1            1           36           74     0     0       0
##      45            0            0           19           30     0     0       0
##      46            0            0            0            0     0     1      14
##      47            0            0            7            0     0     1       6
##      48            0            4           18           35    46     8       2
##      49            0            0           19           37     0     0       1
##      50            0            0            0            0     0     0      23
##      51            0            2            2            1     7     0      32
##      52            0            0            0            0     0     1      19
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##      1        4      22      47       3
##      2       79      17       1      11
##      3      543       8       1      15
##      4        3       3    1244     788
##      5     1148     965      25     367
##      6       57      81       0      26
##      7        5       4     320    1054
##      8       15      27     722     270
##      9      250     530       3     188
##      10       1       6       1       0
##      11     117      30       0      12
##      12     201     753       1       6
##      13      48     133       2       0
##      14      44     121      30      49
##      15     118      18       0      17
##      16      22       7     846     209
##      17     336     220      22     183
##      18      49      44       0       6
##      19      35       9       0       0
##      20      78      13       0      15
##      21     103      15       0      25
##      22       8       6     208      78
##      23      48      69       2      13
##      24       1       7       0       1
##      25     234      15       1       2
##      26      34     148       7       1
##      27      21      37       4      11
##      28      11      18       1      13
##      29     158     160      13      47
##      30       8      29      24       1
##      31      29      33       0      11
##      32      23      77       0       0
##      33       2       0     296      14
##      34     249     196       8      67
##      35      18      18       0       2
##      36     117       9       0       3
##      37      22      15       0      12
##      38      18       5       1       4
##      39      14       3       0       5
##      40      18       2       0       3
##      41      34      81       5      64
##      42       0       2       0       0
##      43       1       0       1       1
##      44     168      22       1       0
##      45       2       0       0       0
##      46      33      39       0       5
##      47      54      16       0       4
##      48      12      34       3       9
##      49       0       3       1       0
##      50      14      17       0       7
##      51      17      28       0       4
##      52      34      28       0       2</code></pre>
<div class="sourceCode" id="cb1531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1531-1"><a href="dataSetIntegration-allSetsTop.html#cb1531-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.mnn,</span>
<span id="cb1531-2"><a href="dataSetIntegration-allSetsTop.html#cb1531-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1531-3"><a href="dataSetIntegration-allSetsTop.html#cb1531-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1531-4"><a href="dataSetIntegration-allSetsTop.html#cb1531-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1531-5"><a href="dataSetIntegration-allSetsTop.html#cb1531-5"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-504-1.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The presence of visual clusters containing cells from both batches provides a comforting illusion that the correction was successful.</p>
<div class="sourceCode" id="cb1532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1532-1"><a href="dataSetIntegration-allSetsTop.html#cb1532-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb1532-2"><a href="dataSetIntegration-allSetsTop.html#cb1532-2"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb1532-3"><a href="dataSetIntegration-allSetsTop.html#cb1532-3"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1532-4"><a href="dataSetIntegration-allSetsTop.html#cb1532-4"></a><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-505-1.png" width="806.4" /></p>
<div class="sourceCode" id="cb1533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1533-1"><a href="dataSetIntegration-allSetsTop.html#cb1533-1"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span>
<span id="cb1533-2"><a href="dataSetIntegration-allSetsTop.html#cb1533-2"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;clusters.mnn&quot;</span>)</span>
<span id="cb1533-3"><a href="dataSetIntegration-allSetsTop.html#cb1533-3"></a><span class="co">#gridExtra::grid.arrange(p.clu, p.batch, ncol=2)</span></span>
<span id="cb1533-4"><a href="dataSetIntegration-allSetsTop.html#cb1533-4"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p.clu, p.batch<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>type), <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-505-2.png" width="806.4" /></p>
<p>For fastMNN(), one useful diagnostic is the proportion of variance within each batch that is lost during MNN correction. Specifically, this refers to the within-batch variance that is removed during orthogonalization with respect to the average correction vector at each merge step. This is returned via the lost.var field in the metadata of mnn.out, which contains a matrix of the variance lost in each batch (column) at each merge step (row).</p>
<div class="sourceCode" id="cb1534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1534-1"><a href="dataSetIntegration-allSetsTop.html#cb1534-1"></a><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span></code></pre></div>
<pre><code>##       ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4       HHD_1
##  [1,]  0.000000000  0.000000000  0.000000000  0.000000000 0.000000000
##  [2,]  0.000000000  0.000000000  0.084463640  0.000000000 0.000000000
##  [3,]  0.000000000  0.000000000  0.008280342  0.095017041 0.000000000
##  [4,]  0.000000000  0.064389261  0.023261243  0.041177341 0.000000000
##  [5,]  0.000000000  0.012352895  0.043870823  0.067557228 0.048516656
##  [6,]  0.000000000  0.002779509  0.002084513  0.002706268 0.002204974
##  [7,]  0.000000000  0.011868285  0.020881803  0.030617693 0.015255640
##  [8,]  0.000000000  0.005304089  0.008755495  0.003861718 0.005637460
##  [9,]  0.082254851  0.003885598  0.009924919  0.011583848 0.002969818
## [10,]  0.007424601  0.006740011  0.012982525  0.011087357 0.008901993
##             HHD_2   PRE-T_1     PRE-T_2     PBMMC_1     PBMMC_2     PBMMC_3
##  [1,] 0.000000000 0.0000000 0.000000000 0.000000000 0.070331394 0.035235202
##  [2,] 0.000000000 0.0000000 0.000000000 0.000000000 0.003311880 0.008051414
##  [3,] 0.000000000 0.0000000 0.000000000 0.000000000 0.009652129 0.004678349
##  [4,] 0.000000000 0.0000000 0.000000000 0.000000000 0.039974155 0.011177234
##  [5,] 0.000000000 0.0000000 0.000000000 0.000000000 0.080780029 0.021861199
##  [6,] 0.074470370 0.0000000 0.000000000 0.000000000 0.002955670 0.002106351
##  [7,] 0.017966721 0.0000000 0.000000000 0.076763560 0.036944237 0.016965467
##  [8,] 0.005220124 0.0000000 0.112508397 0.006247915 0.003233413 0.006395939
##  [9,] 0.003663400 0.0000000 0.009135704 0.004539476 0.015052363 0.005845391
## [10,] 0.006854063 0.0659627 0.010755769 0.008591078 0.015418877 0.008891157</code></pre>
<div class="sourceCode" id="cb1536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1536-1"><a href="dataSetIntegration-allSetsTop.html#cb1536-1"></a>tmpData &lt;-<span class="st"> </span><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span>
<span id="cb1536-2"><a href="dataSetIntegration-allSetsTop.html#cb1536-2"></a></span>
<span id="cb1536-3"><a href="dataSetIntegration-allSetsTop.html#cb1536-3"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb1536-4"><a href="dataSetIntegration-allSetsTop.html#cb1536-4"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1536-5"><a href="dataSetIntegration-allSetsTop.html#cb1536-5"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1536-6"><a href="dataSetIntegration-allSetsTop.html#cb1536-6"></a>           <span class="dt">cluster_rows      =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1536-7"><a href="dataSetIntegration-allSetsTop.html#cb1536-7"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1536-8"><a href="dataSetIntegration-allSetsTop.html#cb1536-8"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-507-1.png" width="672" /></p>
<p>Large proportions of lost variance (&gt;10%) suggest that correction is removing genuine biological heterogeneity. This would occur due to violations of the assumption of orthogonality between the batch effect and the biological subspace (Haghverdi et al. 2018). In this case, the proportion of lost variance is small, indicating that non-orthogonality is not a major concern.</p>
</div>
</div>
<div id="preserving-biological-heterogeneity-5" class="section level2">
<h2><span class="header-section-number">19.6</span> Preserving biological heterogeneity</h2>
<div id="comparison-to-within-batch-clusters-1" class="section level3">
<h3><span class="header-section-number">19.6.1</span> Comparison to within-batch clusters</h3>
<p>Another useful diagnostic check is to compare the clustering within each batch to the clustering of the merged data. Accurate data integration should preserve variance within each batch as there should be nothing to remove between cells in the same batch. This check complements the previously mentioned diagnostics that only focus on the removal of differences between batches. Specifically, it protects us against cases where the correction method simply aggregates all cells together, which would achieve perfect mixing but also discard the biological heterogeneity of interest.</p>
<p>Ideally, we should see a many-to-1 mapping where the across-batch clustering is nested inside the within-batch clusterings. This indicates that any within-batch structure was preserved after correction while acknowledging that greater resolution is possible with more cells. In practice, more discrepancies can be expected even when the correction is perfect, due to the existence of closely related clusters that were arbitrarily separated in the within-batch clustering. As a general rule, we can be satisfied with the correction if the vast majority of entries in Figure 13.4 are zero, though this may depend on whether specific clusters of interest are gained or lost.</p>
<p>One heatmap is generated for each of the PBMC 3K and 4K datasets, where each entry is colored according to the number of cells with each pair of labels (before and after correction).</p>
<div class="sourceCode" id="cb1537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1537-1"><a href="dataSetIntegration-allSetsTop.html#cb1537-1"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1537-2"><a href="dataSetIntegration-allSetsTop.html#cb1537-2"></a></span>
<span id="cb1537-3"><a href="dataSetIntegration-allSetsTop.html#cb1537-3"></a><span class="co"># For the first batch (adding +10 for a smoother color transition</span></span>
<span id="cb1537-4"><a href="dataSetIntegration-allSetsTop.html#cb1537-4"></a><span class="co"># from zero to non-zero counts for any given matrix entry).</span></span>
<span id="cb1537-5"><a href="dataSetIntegration-allSetsTop.html#cb1537-5"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1537-6"><a href="dataSetIntegration-allSetsTop.html#cb1537-6"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1537-7"><a href="dataSetIntegration-allSetsTop.html#cb1537-7"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb1537-8"><a href="dataSetIntegration-allSetsTop.html#cb1537-8"></a>             <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1537-9"><a href="dataSetIntegration-allSetsTop.html#cb1537-9"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>), <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1537-10"><a href="dataSetIntegration-allSetsTop.html#cb1537-10"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]), <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1537-11"><a href="dataSetIntegration-allSetsTop.html#cb1537-11"></a></span>
<span id="cb1537-12"><a href="dataSetIntegration-allSetsTop.html#cb1537-12"></a><span class="co"># For the second batch.</span></span>
<span id="cb1537-13"><a href="dataSetIntegration-allSetsTop.html#cb1537-13"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1537-14"><a href="dataSetIntegration-allSetsTop.html#cb1537-14"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1537-15"><a href="dataSetIntegration-allSetsTop.html#cb1537-15"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb1537-16"><a href="dataSetIntegration-allSetsTop.html#cb1537-16"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1537-17"><a href="dataSetIntegration-allSetsTop.html#cb1537-17"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>), <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1537-18"><a href="dataSetIntegration-allSetsTop.html#cb1537-18"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]), <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1537-19"><a href="dataSetIntegration-allSetsTop.html#cb1537-19"></a></span>
<span id="cb1537-20"><a href="dataSetIntegration-allSetsTop.html#cb1537-20"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-508-1.png" width="672" /></p>
<p>Another evaluation approach is to compute the coassignment probabilities (Section 10.6), i.e., the probability that cells from two within-batch clusters are clustered together in the across-batch clustering. High probabilities off the diagonal in Figure 13.5 indicate that within-batch clusters are merged in the across-batch analysis. We would generally expect low off-diagonal probabilities for most pairs of clusters, though this may not be reasonably possible if the within-batch clusters were poorly separated in the first place.</p>
<p>Coassignment probabilities for the within-batch clusters, based on coassignment of cells in the across-batch clusters obtained after MNN correction. One heatmap is generated for each of the GSM3872434 and GSM3872435, where each entry is colored according to the coassignment probability between each pair of within-batch clusters:</p>
<div class="sourceCode" id="cb1538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1538-1"><a href="dataSetIntegration-allSetsTop.html#cb1538-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb1538-2"><a href="dataSetIntegration-allSetsTop.html#cb1538-2"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1538-3"><a href="dataSetIntegration-allSetsTop.html#cb1538-3"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1538-4"><a href="dataSetIntegration-allSetsTop.html#cb1538-4"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb1538-5"><a href="dataSetIntegration-allSetsTop.html#cb1538-5"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1538-6"><a href="dataSetIntegration-allSetsTop.html#cb1538-6"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)), <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]), <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1538-7"><a href="dataSetIntegration-allSetsTop.html#cb1538-7"></a></span>
<span id="cb1538-8"><a href="dataSetIntegration-allSetsTop.html#cb1538-8"></a><span class="co"># For the second batch.</span></span>
<span id="cb1538-9"><a href="dataSetIntegration-allSetsTop.html#cb1538-9"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1538-10"><a href="dataSetIntegration-allSetsTop.html#cb1538-10"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1538-11"><a href="dataSetIntegration-allSetsTop.html#cb1538-11"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb1538-12"><a href="dataSetIntegration-allSetsTop.html#cb1538-12"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1538-13"><a href="dataSetIntegration-allSetsTop.html#cb1538-13"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)), <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]), <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1538-14"><a href="dataSetIntegration-allSetsTop.html#cb1538-14"></a></span>
<span id="cb1538-15"><a href="dataSetIntegration-allSetsTop.html#cb1538-15"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-509-1.png" width="672" /></p>
<p>Finally, we can summarize the agreement between clusterings by computing the Rand index. This provides a simple metric that we can use to assess the preservation of variation by different correction methods. Larger rand indices (i.e., closer to 1) are more desirable, though this must be balanced against the ability of each method to actually remove the batch effect.</p>
<div class="sourceCode" id="cb1539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1539-1"><a href="dataSetIntegration-allSetsTop.html#cb1539-1"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(fossil))</span>
<span id="cb1539-2"><a href="dataSetIntegration-allSetsTop.html#cb1539-2"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1539-3"><a href="dataSetIntegration-allSetsTop.html#cb1539-3"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1539-4"><a href="dataSetIntegration-allSetsTop.html#cb1539-4"></a>ri1 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb1539-5"><a href="dataSetIntegration-allSetsTop.html#cb1539-5"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1539-6"><a href="dataSetIntegration-allSetsTop.html#cb1539-6"></a>ri1</span></code></pre></div>
<pre><code>## [1] 0.6663171</code></pre>
<div class="sourceCode" id="cb1541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1541-1"><a href="dataSetIntegration-allSetsTop.html#cb1541-1"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1541-2"><a href="dataSetIntegration-allSetsTop.html#cb1541-2"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1541-3"><a href="dataSetIntegration-allSetsTop.html#cb1541-3"></a>ri2 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb1541-4"><a href="dataSetIntegration-allSetsTop.html#cb1541-4"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1541-5"><a href="dataSetIntegration-allSetsTop.html#cb1541-5"></a>ri2</span></code></pre></div>
<pre><code>## [1] 0.7142638</code></pre>
</div>
<div id="encouraging-consistency-with-marker-genes-5" class="section level3">
<h3><span class="header-section-number">19.6.2</span> Encouraging consistency with marker genes</h3>
<p>In some situations, we will already have performed within-batch analyses to characterize salient aspects of population heterogeneity. This is not uncommon when merging datasets from different sources where each dataset has already been analyzed, annotated and interpreted separately. It is subsequently desirable for the integration procedure to retain these “known interesting” aspects of each dataset in the merged dataset. We can encourage this outcome by using the marker genes within each dataset as our selected feature set for fastMNN() and related methods. This focuses on the relevant heterogeneity and represents a semi-supervised approach that is a natural extension of the strategy presented in th feature selection section.</p>
<!-- To illustrate, we apply this strategy to our PBMC datasets. -->
<p>We identify the top marker genes from pairwise Wilcoxon ranked sum tests between every pair of clusters within each batch, analogous to the method used by SingleR (Chapter 12). In this case, we use the top 10 marker genes but any value can be used depending on the acceptable trade-off between signal and noise (and speed). We then take the union across all comparisons in all batches and use that in place of our HVG set in fastMNN().</p>
<div class="sourceCode" id="cb1543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1543-1"><a href="dataSetIntegration-allSetsTop.html#cb1543-1"></a><span class="co"># Recall that groups for marker detection</span></span>
<span id="cb1543-2"><a href="dataSetIntegration-allSetsTop.html#cb1543-2"></a><span class="co"># are automatically defined from &#39;colLabels()&#39;. </span></span>
<span id="cb1543-3"><a href="dataSetIntegration-allSetsTop.html#cb1543-3"></a>stats1 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">1</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1543-4"><a href="dataSetIntegration-allSetsTop.html#cb1543-4"></a>markers1 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats1[[<span class="dv">1</span>]], stats1[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1543-5"><a href="dataSetIntegration-allSetsTop.html#cb1543-5"></a></span>
<span id="cb1543-6"><a href="dataSetIntegration-allSetsTop.html#cb1543-6"></a>stats2 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">2</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1543-7"><a href="dataSetIntegration-allSetsTop.html#cb1543-7"></a>markers2 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats2[[<span class="dv">1</span>]], stats2[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1543-8"><a href="dataSetIntegration-allSetsTop.html#cb1543-8"></a></span>
<span id="cb1543-9"><a href="dataSetIntegration-allSetsTop.html#cb1543-9"></a>stats3 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">3</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1543-10"><a href="dataSetIntegration-allSetsTop.html#cb1543-10"></a>markers3 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats3[[<span class="dv">1</span>]], stats3[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1543-11"><a href="dataSetIntegration-allSetsTop.html#cb1543-11"></a></span>
<span id="cb1543-12"><a href="dataSetIntegration-allSetsTop.html#cb1543-12"></a>stats4 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">4</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1543-13"><a href="dataSetIntegration-allSetsTop.html#cb1543-13"></a>markers4 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats4[[<span class="dv">1</span>]], stats4[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1543-14"><a href="dataSetIntegration-allSetsTop.html#cb1543-14"></a></span>
<span id="cb1543-15"><a href="dataSetIntegration-allSetsTop.html#cb1543-15"></a>marker.set &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">unlist</span>(<span class="kw">c</span>(<span class="kw">unlist</span>(markers1), <span class="kw">unlist</span>(markers2), <span class="kw">unlist</span>(markers3), <span class="kw">unlist</span>(markers4))))</span>
<span id="cb1543-16"><a href="dataSetIntegration-allSetsTop.html#cb1543-16"></a><span class="kw">length</span>(marker.set) <span class="co"># getting the total number of genes selected in this manner.</span></span></code></pre></div>
<pre><code>## [1] 740</code></pre>
<!-- MIND marker.set is from thr RUN etc and used for all 11 samples -->
<div class="sourceCode" id="cb1545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1545-1"><a href="dataSetIntegration-allSetsTop.html#cb1545-1"></a><span class="kw">set.seed</span>(<span class="dv">1000110</span>)</span>
<span id="cb1545-2"><a href="dataSetIntegration-allSetsTop.html#cb1545-2"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(</span>
<span id="cb1545-3"><a href="dataSetIntegration-allSetsTop.html#cb1545-3"></a>  rescaled.mbn[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],</span>
<span id="cb1545-4"><a href="dataSetIntegration-allSetsTop.html#cb1545-4"></a>  <span class="dt">subset.row=</span>marker.set,</span>
<span id="cb1545-5"><a href="dataSetIntegration-allSetsTop.html#cb1545-5"></a>  <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>))</span></code></pre></div>
<p>A quick inspection indicates that the original within-batch structure is indeed preserved in the corrected data. This highlights the utility of a marker-based feature set for integrating datasets that have already been characterized separately in a manner that preserves existing interpretations of each dataset. We note that some within-batch clusters have merged, most likely due to the lack of robust separation in the first place, though this may also be treated as a diagnostic on the appropriateness of the integration depending on the context.</p>
<div class="sourceCode" id="cb1546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1546-1"><a href="dataSetIntegration-allSetsTop.html#cb1546-1"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out2, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1546-2"><a href="dataSetIntegration-allSetsTop.html#cb1546-2"></a>batchVec &lt;-<span class="st"> </span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb1546-3"><a href="dataSetIntegration-allSetsTop.html#cb1546-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(</span>
<span id="cb1546-4"><a href="dataSetIntegration-allSetsTop.html#cb1546-4"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">1</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">1</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">1</span>]),</span>
<span id="cb1546-5"><a href="dataSetIntegration-allSetsTop.html#cb1546-5"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">2</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">2</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">2</span>]),</span>
<span id="cb1546-6"><a href="dataSetIntegration-allSetsTop.html#cb1546-6"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">3</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">3</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">3</span>]),</span>
<span id="cb1546-7"><a href="dataSetIntegration-allSetsTop.html#cb1546-7"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">4</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">4</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">4</span>]),</span>
<span id="cb1546-8"><a href="dataSetIntegration-allSetsTop.html#cb1546-8"></a>    <span class="dt">ncol=</span><span class="dv">2</span></span>
<span id="cb1546-9"><a href="dataSetIntegration-allSetsTop.html#cb1546-9"></a>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-513-1.png" width="672" /></p>
</div>
<div id="using-the-corrected-values" class="section level3">
<h3><span class="header-section-number">19.6.3</span> Using the corrected values</h3>
<p>it is preferable to perform DE analyses using the uncorrected expression values with blocking on the batch. This strategy is based on the expectation that any genuine DE between clusters should still be present in a within-batch comparison where batch effects are absent. It penalizes genes that exhibit inconsistent DE across batches, thus protecting against misleading conclusions when a population in one batch is aligned to a similar-but-not-identical population in another batch. We demonstrate this approach below using a blocked t-test to detect markers in the PBMC dataset, where the presence of the same pattern across clusters within each batch is reassuring. If integration is performed across multiple conditions, it is even more important to use the uncorrected expression values for downstream analyses - see Section 14.5.2 for a discussion.</p>
<div class="sourceCode" id="cb1547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1547-1"><a href="dataSetIntegration-allSetsTop.html#cb1547-1"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(</span>
<span id="cb1547-2"><a href="dataSetIntegration-allSetsTop.html#cb1547-2"></a>  uncorrected,</span>
<span id="cb1547-3"><a href="dataSetIntegration-allSetsTop.html#cb1547-3"></a>  clusters.mnn,</span>
<span id="cb1547-4"><a href="dataSetIntegration-allSetsTop.html#cb1547-4"></a>  <span class="dt">block=</span>uncorrected<span class="op">$</span>batch, <span class="co"># </span><span class="al">TODO</span><span class="co"> batch or type?</span></span>
<span id="cb1547-5"><a href="dataSetIntegration-allSetsTop.html#cb1547-5"></a>  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1547-6"><a href="dataSetIntegration-allSetsTop.html#cb1547-6"></a>  <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb1547-7"><a href="dataSetIntegration-allSetsTop.html#cb1547-7"></a>  <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,</span>
<span id="cb1547-8"><a href="dataSetIntegration-allSetsTop.html#cb1547-8"></a>                                <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),</span>
<span id="cb1547-9"><a href="dataSetIntegration-allSetsTop.html#cb1547-9"></a>                                <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb1548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1548-1"><a href="dataSetIntegration-allSetsTop.html#cb1548-1"></a>demo &lt;-<span class="st"> </span>m.out[[<span class="st">&quot;1&quot;</span>]]</span>
<span id="cb1548-2"><a href="dataSetIntegration-allSetsTop.html#cb1548-2"></a><span class="kw">as.data.frame</span>(demo[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,<span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]) </span></code></pre></div>
<pre><code>##                  Symbol Top       p.value           FDR
## ENSG00000034510  TMSB10   1 4.822073e-221 5.227979e-218
## ENSG00000107447    DNTT   1  7.308739e-98  1.603659e-95
## ENSG00000117632   STMN1   1  0.000000e+00  0.000000e+00
## ENSG00000124766    SOX4   1  5.664909e-46  7.970224e-44
## ENSG00000164104   HMGB2   1  0.000000e+00  0.000000e+00
## ENSG00000170540 ARL6IP1   1  4.344847e-76  7.850968e-74
## ENSG00000189403   HMGB1   1  0.000000e+00  0.000000e+00
## ENSG00000196230    TUBB   1 2.552552e-246 3.920507e-243
## ENSG00000198034   RPS4X   1 7.992635e-172 5.079733e-169
## ENSG00000198830   HMGN2   1  0.000000e+00  0.000000e+00
## ENSG00000204287 HLA-DRA   1  0.000000e+00  0.000000e+00
## ENSG00000205542  TMSB4X   1 9.852209e-283 2.017623e-279
## ENSG00000205581   HMGN1   1 2.026161e-155 8.684690e-153
## ENSG00000251562  MALAT1   1 1.588821e-234 2.091683e-231
## ENSG00000272398    CD24   1 6.506038e-136 2.032420e-133
## ENSG00000019582    CD74   2 9.509121e-314 2.921043e-310
## ENSG00000111348 ARHGDIB   2  1.072067e-66  1.864081e-64
## ENSG00000117724   CENPF   2  6.841406e-07  6.150925e-05
## ENSG00000131747   TOP2A   2  3.810227e-24  4.256139e-22
## ENSG00000164611   PTTG1   2  5.292404e-57  8.337119e-55</code></pre>
<div class="sourceCode" id="cb1550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1550-1"><a href="dataSetIntegration-allSetsTop.html#cb1550-1"></a><span class="co">#as.data.frame(demo[1:20,c(&quot;external_gene_name&quot;, &quot;Top&quot;, &quot;p.value&quot;, &quot;FDR&quot;)]) </span></span></code></pre></div>
<p>Expression level for the top gene, ENSG00000034510:</p>
<div class="sourceCode" id="cb1551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1551-1"><a href="dataSetIntegration-allSetsTop.html#cb1551-1"></a>geneEnsId &lt;-<span class="st"> </span><span class="kw">rownames</span>(demo)[<span class="dv">1</span>]</span>
<span id="cb1551-2"><a href="dataSetIntegration-allSetsTop.html#cb1551-2"></a><span class="kw">plotExpression</span>(uncorrected, <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)), </span>
<span id="cb1551-3"><a href="dataSetIntegration-allSetsTop.html#cb1551-3"></a>    <span class="dt">features=</span>geneEnsId, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-516-1.png" width="672" /></p>
</div>
</div>
<div id="challenge-same-but-with-an-ordered-merging" class="section level2">
<h2><span class="header-section-number">19.7</span> <strong>Challenge</strong> Same but with an ordered merging</h2>
<p>We will first merge replicates in each sample group separately, then sample groups, starting with the group with the larger number of ‘cell types’.</p>
<p>Hint: use the merge.order option in fastMNN ( … maybe with “list( list(1,2,3,4), list(9,10,11), list(5,6), list(7,8) )” )</p>
<div class="sourceCode" id="cb1552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1552-1"><a href="dataSetIntegration-allSetsTop.html#cb1552-1"></a><span class="co"># Using randomized SVD here, as this is faster than </span></span>
<span id="cb1552-2"><a href="dataSetIntegration-allSetsTop.html#cb1552-2"></a><span class="co"># irlba for file-backed matrices.</span></span>
<span id="cb1552-3"><a href="dataSetIntegration-allSetsTop.html#cb1552-3"></a><span class="kw">set.seed</span>(<span class="dv">1000101001</span>)</span>
<span id="cb1552-4"><a href="dataSetIntegration-allSetsTop.html#cb1552-4"></a><span class="kw">rm</span>(mnn.out)</span>
<span id="cb1552-5"><a href="dataSetIntegration-allSetsTop.html#cb1552-5"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(rescaled.mbn,</span>
<span id="cb1552-6"><a href="dataSetIntegration-allSetsTop.html#cb1552-6"></a>           <span class="dt">merge.order=</span><span class="kw">list</span>( <span class="kw">list</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="kw">list</span>(<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">11</span>), <span class="kw">list</span>(<span class="dv">5</span>,<span class="dv">6</span>), <span class="kw">list</span>(<span class="dv">7</span>,<span class="dv">8</span>) ),</span>
<span id="cb1552-7"><a href="dataSetIntegration-allSetsTop.html#cb1552-7"></a>           <span class="dt">d=</span><span class="dv">50</span>, <span class="dt">k=</span><span class="dv">20</span>, <span class="dt">subset.row=</span>chosen.hvgs,</span>
<span id="cb1552-8"><a href="dataSetIntegration-allSetsTop.html#cb1552-8"></a>    <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>))</span>
<span id="cb1552-9"><a href="dataSetIntegration-allSetsTop.html#cb1552-9"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 12466 47830 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(12466): ENSG00000000003 ENSG00000000457 ... ENSG00000285476
##   ENSG00000285492
## rowData names(1): rotation
## colnames: NULL
## colData names(1): batch
## reducedDimNames(1): corrected
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1554-1"><a href="dataSetIntegration-allSetsTop.html#cb1554-1"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb1554-2"><a href="dataSetIntegration-allSetsTop.html#cb1554-2"></a>mnn.out<span class="op">$</span>type &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch)</span>
<span id="cb1554-3"><a href="dataSetIntegration-allSetsTop.html#cb1554-3"></a>mnn.out<span class="op">$</span>type &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>type)</span></code></pre></div>
<div class="sourceCode" id="cb1555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1555-1"><a href="dataSetIntegration-allSetsTop.html#cb1555-1"></a><span class="co">#class(mnn.out$batch)</span></span>
<span id="cb1555-2"><a href="dataSetIntegration-allSetsTop.html#cb1555-2"></a><span class="co">#head(mnn.out$batch)</span></span>
<span id="cb1555-3"><a href="dataSetIntegration-allSetsTop.html#cb1555-3"></a><span class="co">#dim(reducedDim(mnn.out, &quot;corrected&quot;))</span></span>
<span id="cb1555-4"><a href="dataSetIntegration-allSetsTop.html#cb1555-4"></a><span class="co">#assay(mnn.out, &quot;reconstructed&quot;)</span></span>
<span id="cb1555-5"><a href="dataSetIntegration-allSetsTop.html#cb1555-5"></a><span class="kw">print</span>(<span class="kw">dim</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)))</span></code></pre></div>
<pre><code>## [1] 12466 47830</code></pre>
<div class="sourceCode" id="cb1557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1557-1"><a href="dataSetIntegration-allSetsTop.html#cb1557-1"></a><span class="kw">print</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</span></code></pre></div>
<pre><code>## &lt;5 x 3&gt; matrix of class LowRankMatrix and type &quot;double&quot;:
##                          [,1]          [,2]          [,3]
## ENSG00000000003 -1.272426e-04 -6.787507e-05 -5.702688e-05
## ENSG00000000457 -3.356405e-05  4.087098e-04  1.054453e-04
## ENSG00000000938 -2.351897e-03 -4.617906e-04 -3.513377e-04
## ENSG00000001167 -6.807745e-05  2.264625e-04  2.274302e-04
## ENSG00000001461 -2.447150e-04 -4.125638e-05 -2.362878e-04</code></pre>
<p>Diagnostic table and plots:</p>
<div class="sourceCode" id="cb1559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1559-1"><a href="dataSetIntegration-allSetsTop.html#cb1559-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1559-2"><a href="dataSetIntegration-allSetsTop.html#cb1559-2"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(mnn.out, <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1559-3"><a href="dataSetIntegration-allSetsTop.html#cb1559-3"></a>clusters.mnn &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1559-4"><a href="dataSetIntegration-allSetsTop.html#cb1559-4"></a>mnn.out<span class="op">$</span>clusters.mnn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;c%s&quot;</span>, clusters.mnn)</span>
<span id="cb1559-5"><a href="dataSetIntegration-allSetsTop.html#cb1559-5"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>mnn.out<span class="op">$</span>clusters.mnn, <span class="dt">Batch=</span>mnn.out<span class="op">$</span>batch)</span>
<span id="cb1559-6"><a href="dataSetIntegration-allSetsTop.html#cb1559-6"></a>tab.mnn</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##     c1             0            0            2            1     0     2       1
##     c10          400          582          203          925   375   644      19
##     c11            1            3           46          162     3     4       0
##     c12            0            0           21           67     0     1       1
##     c13           98          153           14          120    31    61      41
##     c14            0            0           20           69     0     2       0
##     c15            2            0           13           22     0     1      16
##     c16            1            0           21           37     1     3       5
##     c17            1            2           37          124     0     3       3
##     c18           12          136         1367          230   206    89     253
##     c19          132          251           48          218   110   308       8
##     c2          1526         1787          418         1541  1030  1922      39
##     c20            9           33           63           20    26    55     148
##     c21            0            0            3            3     0     1       7
##     c22            1            5           16            8    14     4      33
##     c23            0            1           15           22     0     0       0
##     c24           93          165           31          186    97   109      11
##     c25          213            8            4            6   332   134      34
##     c26            0            0            1            0     1     1      22
##     c27            0            1            0            1    15     1      39
##     c28            0            2            0            3     2     2     101
##     c29            0            8            7            1     1     1      11
##     c3             5            0            8            5    36    22      45
##     c30            0            2            1            0     1     0      48
##     c31            0            1           19           21     0     0       1
##     c32            0            0            3            1     0     0      10
##     c33            0            1           20           38     0     4       2
##     c34            0            0            0            0     0     0      35
##     c35            1            0            2            1     1     0       0
##     c36          129         2690          860          571   520  1617      20
##     c37            0            4           23          137     0     4       0
##     c38            4           11           37           31    34    18       6
##     c39            4           57          229           73   143    24      38
##     c4           105          480          106          107   428   147      23
##     c40            0           12            3            4     1     5       4
##     c41            0            0            0            0     0     1       2
##     c42           56           29          437           10    25    18       0
##     c43            1            0            5            4     1     1       3
##     c44            1            0            0            0    10     5      13
##     c45            2           43          218           66    83    13      19
##     c46            0            0            0            1     2     0      27
##     c47            0            1            1            1     1     0      13
##     c48            0            0            3            2     0     3       1
##     c49            2            1            0            0     3     1       4
##     c5             0            0            3           10     0     0      29
##     c50            0            0            6          111     3     6       3
##     c51            3            3           25          130     0     1       0
##     c52            0            0            1            0     1     1      16
##     c53           21            0            3            3     1     0       3
##     c54            1            2           15           31     0     0       0
##     c55            0            0            2            0     0     0       5
##     c56            0            0           12            8     0     0       0
##     c57            0            4           17           34    46     8       2
##     c58            1            2            3            1    13     6      36
##     c59            6            1            0            1    15     1      23
##     c6             7           66          275           66   119    48      36
##     c60            0            0            2            0     0     1       1
##     c61            0            0           15           25     0     0       0
##     c62            0            0            1            1     0     0     364
##     c63            0            1            0            0     0     1      19
##     c64            0            0            0            0     0     0       7
##     c65            0            0            4           22     0     0       0
##     c66            0            0            8            0     1     0       1
##     c7             9           19            5            7    36    15     195
##     c8             6           48            5            2   782   118     173
##     c9             0            0            0            2     1     0      65
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##     c1        3       2      25     879
##     c10      26      57      18       3
##     c11     213      22       2      15
##     c12       0       5       0       0
##     c13      22      49      15       2
##     c14      12       8       0       1
##     c15     111      16       0      21
##     c16     109      24       0      14
##     c17     114       8       0       6
##     c18    1001     915      25     347
##     c19      20      37      27       3
##     c2       46     105       3       9
##     c20      36     226     220     328
##     c21      40       8       1      16
##     c22     227      58       7      42
##     c23       1       6       0       0
##     c24       4      27      33       1
##     c25      47      60       3      19
##     c26       0       1     795     577
##     c27      17      29     286      62
##     c28     140     250       3      86
##     c29      22      15       0      12
##     c3       13      18     534     226
##     c30      56     166       0       2
##     c31      58      11       0       0
##     c32      61     128       0      25
##     c33     115      23       0       0
##     c34      25      20       0      15
##     c35      18       2       0       3
##     c36      31     152       3       3
##     c37     386       5       0       9
##     c38      27      67       3      25
##     c39     262     229       9      66
##     c4       25      15     819     179
##     c40      11      18       1      13
##     c41       3       2     440     114
##     c42       0       3       0       0
##     c43      19       4       1       3
##     c44      10       7     222      38
##     c45     140     168      14      48
##     c46      21      26       0       5
##     c47      27     100       0      52
##     c48      14       3       0       3
##     c49       5      11      69      31
##     c5       72      26       0       9
##     c50     152      16       1      14
##     c51     214       3       1       0
##     c52      37      46       0       3
##     c53       7      14       3       1
##     c54       0       0       0       0
##     c55      35       8       0       1
##     c56      23       3       0       0
##     c57      25      76       3      30
##     c58      16      26       1       5
##     c59       8      14     172      12
##     c6      198     223      20     203
##     c60      14       8       0       3
##     c61       1       0       0       0
##     c62      68       7       0       2
##     c63      34      27       1       3
##     c64       9      10       0       0
##     c65       0       0       1       0
##     c66       6      23       0       0
##     c7       23     134      60       7
##     c8      115     291       0       4
##     c9       63      82       0      28</code></pre>
<div class="sourceCode" id="cb1561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1561-1"><a href="dataSetIntegration-allSetsTop.html#cb1561-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.mnn,</span>
<span id="cb1561-2"><a href="dataSetIntegration-allSetsTop.html#cb1561-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1561-3"><a href="dataSetIntegration-allSetsTop.html#cb1561-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1561-4"><a href="dataSetIntegration-allSetsTop.html#cb1561-4"></a>           <span class="co">#cluster_rows      = FALSE,</span></span>
<span id="cb1561-5"><a href="dataSetIntegration-allSetsTop.html#cb1561-5"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1561-6"><a href="dataSetIntegration-allSetsTop.html#cb1561-6"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-520-1.png" width="672" /></p>
<div class="sourceCode" id="cb1562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1562-1"><a href="dataSetIntegration-allSetsTop.html#cb1562-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb1562-2"><a href="dataSetIntegration-allSetsTop.html#cb1562-2"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb1562-3"><a href="dataSetIntegration-allSetsTop.html#cb1562-3"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1562-4"><a href="dataSetIntegration-allSetsTop.html#cb1562-4"></a></span>
<span id="cb1562-5"><a href="dataSetIntegration-allSetsTop.html#cb1562-5"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span>
<span id="cb1562-6"><a href="dataSetIntegration-allSetsTop.html#cb1562-6"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;clusters.mnn&quot;</span>)</span>
<span id="cb1562-7"><a href="dataSetIntegration-allSetsTop.html#cb1562-7"></a><span class="co">#gridExtra::grid.arrange(p.clu, p.batch, ncol=2)</span></span>
<span id="cb1562-8"><a href="dataSetIntegration-allSetsTop.html#cb1562-8"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p.clu, p.batch<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>type), <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-521-1.png" width="806.4" /></p>
<p>Write mnn.out object to file</p>
<div class="sourceCode" id="cb1563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1563-1"><a href="dataSetIntegration-allSetsTop.html#cb1563-1"></a><span class="kw">colData</span>(mnn.out) &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">colData</span>(uncorrected),<span class="kw">colData</span>(mnn.out)[,<span class="kw">c</span>(<span class="st">&quot;type&quot;</span>, <span class="st">&quot;clusters.mnn&quot;</span>)])</span>
<span id="cb1563-2"><a href="dataSetIntegration-allSetsTop.html#cb1563-2"></a><span class="co"># Write object to file</span></span>
<span id="cb1563-3"><a href="dataSetIntegration-allSetsTop.html#cb1563-3"></a><span class="co"># fastMnnWholeByList -&gt; Fmwbl</span></span>
<span id="cb1563-4"><a href="dataSetIntegration-allSetsTop.html#cb1563-4"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_Fmwbl.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1563-5"><a href="dataSetIntegration-allSetsTop.html#cb1563-5"></a><span class="kw">saveRDS</span>(mnn.out, tmpFn)</span>
<span id="cb1563-6"><a href="dataSetIntegration-allSetsTop.html#cb1563-6"></a></span>
<span id="cb1563-7"><a href="dataSetIntegration-allSetsTop.html#cb1563-7"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_Fmwbl2.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1563-8"><a href="dataSetIntegration-allSetsTop.html#cb1563-8"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;chosen.hvgs&quot;</span>=chosen.hvgs, <span class="st">&quot;uncorrected&quot;</span>=uncorrected,<span class="st">&quot;rescaled.mbn&quot;</span>=rescaled.mbn), tmpFn)</span></code></pre></div>
<p>Proportions of lost variance</p>
<div class="sourceCode" id="cb1564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1564-1"><a href="dataSetIntegration-allSetsTop.html#cb1564-1"></a><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span></code></pre></div>
<pre><code>##       ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4      HHD_1      HHD_2
##  [1,]  0.000000000  0.000000000   0.00000000   0.00000000 0.00000000 0.00000000
##  [2,]  0.000000000  0.000000000   0.00000000   0.00000000 0.01079507 0.03060563
##  [3,]  0.000000000  0.000000000   0.00000000   0.00000000 0.00000000 0.00000000
##  [4,]  0.000000000  0.000000000   0.00000000   0.00000000 0.00000000 0.00000000
##  [5,]  0.017172859  0.023902496   0.00000000   0.00000000 0.00000000 0.00000000
##  [6,]  0.008978858  0.007648388   0.03794260   0.00000000 0.00000000 0.00000000
##  [7,]  0.001911633  0.002032952   0.00601280   0.05552391 0.00000000 0.00000000
##  [8,]  0.037686543  0.036119614   0.11166554   0.15752586 0.00000000 0.00000000
##  [9,]  0.013632906  0.017484058   0.03405170   0.03724725 0.06993428 0.08494841
## [10,]  0.018112512  0.018197771   0.02578305   0.02708306 0.02106184 0.02089790
##          PRE-T_1    PRE-T_2     PBMMC_1     PBMMC_2    PBMMC_3
##  [1,] 0.01767670 0.01496369 0.000000000 0.000000000 0.00000000
##  [2,] 0.00000000 0.00000000 0.000000000 0.000000000 0.00000000
##  [3,] 0.00000000 0.00000000 0.044327324 0.177401113 0.00000000
##  [4,] 0.00000000 0.00000000 0.004530961 0.004332593 0.07142984
##  [5,] 0.00000000 0.00000000 0.000000000 0.000000000 0.00000000
##  [6,] 0.00000000 0.00000000 0.000000000 0.000000000 0.00000000
##  [7,] 0.00000000 0.00000000 0.000000000 0.000000000 0.00000000
##  [8,] 0.00000000 0.00000000 0.018866504 0.019922131 0.01752755
##  [9,] 0.00000000 0.00000000 0.015351173 0.048171864 0.01937957
## [10,] 0.06526886 0.11916262 0.020463688 0.023654810 0.01798412</code></pre>
<div class="sourceCode" id="cb1566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1566-1"><a href="dataSetIntegration-allSetsTop.html#cb1566-1"></a>tmpData &lt;-<span class="st"> </span><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span>
<span id="cb1566-2"><a href="dataSetIntegration-allSetsTop.html#cb1566-2"></a></span>
<span id="cb1566-3"><a href="dataSetIntegration-allSetsTop.html#cb1566-3"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb1566-4"><a href="dataSetIntegration-allSetsTop.html#cb1566-4"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1566-5"><a href="dataSetIntegration-allSetsTop.html#cb1566-5"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1566-6"><a href="dataSetIntegration-allSetsTop.html#cb1566-6"></a>           <span class="dt">cluster_rows      =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1566-7"><a href="dataSetIntegration-allSetsTop.html#cb1566-7"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb1566-8"><a href="dataSetIntegration-allSetsTop.html#cb1566-8"></a>           )</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-523-1.png" width="672" /></p>
<p>Comparison to within-batch clusters</p>
<div class="sourceCode" id="cb1567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1567-1"><a href="dataSetIntegration-allSetsTop.html#cb1567-1"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1567-2"><a href="dataSetIntegration-allSetsTop.html#cb1567-2"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch) <span class="co"># somehow need to re-factor batch</span></span>
<span id="cb1567-3"><a href="dataSetIntegration-allSetsTop.html#cb1567-3"></a><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)</span></code></pre></div>
<pre><code>##  [1] &quot;ETV6-RUNX1_1&quot; &quot;ETV6-RUNX1_2&quot; &quot;ETV6-RUNX1_3&quot; &quot;ETV6-RUNX1_4&quot; &quot;HHD_1&quot;       
##  [6] &quot;HHD_2&quot;        &quot;PBMMC_1&quot;      &quot;PBMMC_2&quot;      &quot;PBMMC_3&quot;      &quot;PRE-T_1&quot;     
## [11] &quot;PRE-T_2&quot;</code></pre>
<div class="sourceCode" id="cb1569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1569-1"><a href="dataSetIntegration-allSetsTop.html#cb1569-1"></a><span class="co"># For the first batch (adding +10 for a smoother color transition</span></span>
<span id="cb1569-2"><a href="dataSetIntegration-allSetsTop.html#cb1569-2"></a><span class="co"># from zero to non-zero counts for any given matrix entry).</span></span>
<span id="cb1569-3"><a href="dataSetIntegration-allSetsTop.html#cb1569-3"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1569-4"><a href="dataSetIntegration-allSetsTop.html#cb1569-4"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1569-5"><a href="dataSetIntegration-allSetsTop.html#cb1569-5"></a><span class="kw">length</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]))</span></code></pre></div>
<pre><code>## [1] 2853</code></pre>
<div class="sourceCode" id="cb1571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1571-1"><a href="dataSetIntegration-allSetsTop.html#cb1571-1"></a>rescaled.mbn[[batchPlace]]</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 2853 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): ENSG00000000003 ENSG00000000419 ... ENSG00000285486
##   ENSG00000285492
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(20): Sample Barcode ... label batch
## reducedDimNames(1): PCA
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1573-1"><a href="dataSetIntegration-allSetsTop.html#cb1573-1"></a>rescaled.mbn[[batchPlace]] <span class="op">%&gt;%</span><span class="st"> </span>colData <span class="op">%&gt;%</span><span class="st"> </span>head</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 20 columns
##                   Sample            Barcode         Run Sample.Name source_name
##              &lt;character&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;factor&gt;
## 1 /ssd/personal/baller.. AAACCTGAGACTTTCG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 2 /ssd/personal/baller.. AAACCTGGTCTTCAAG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 3 /ssd/personal/baller.. AAACCTGGTGTTGAGG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 4 /ssd/personal/baller.. AAACCTGTCCCAAGTA-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 5 /ssd/personal/baller.. AAACCTGTCGAATGCT-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 6 /ssd/personal/baller.. AAACGGGCACCATCCT-1  SRR9264343  GSM3872434  ETV6-RUNX1
##         sum  detected subsets_Mito_sum subsets_Mito_detected
##   &lt;numeric&gt; &lt;integer&gt;        &lt;numeric&gt;             &lt;integer&gt;
## 1      6462      1996              290                    12
## 2     11706      3079              567                    12
## 3      7981      2511              423                    12
## 4      8354      2307              519                    13
## 5      1373       701               91                    11
## 6      2142       985              102                    10
##   subsets_Mito_percent     total      block     setName   discard   outlier
##              &lt;numeric&gt; &lt;numeric&gt;   &lt;factor&gt; &lt;character&gt; &lt;logical&gt; &lt;logical&gt;
## 1              4.48777      6462 ETV6-RUNX1       Caron     FALSE     FALSE
## 2              4.84367     11706 ETV6-RUNX1       Caron     FALSE     FALSE
## 3              5.30009      7981 ETV6-RUNX1       Caron     FALSE     FALSE
## 4              6.21259      8354 ETV6-RUNX1       Caron     FALSE     FALSE
## 5              6.62782      1373 ETV6-RUNX1       Caron     FALSE     FALSE
## 6              4.76190      2142 ETV6-RUNX1       Caron     FALSE     FALSE
##   cell_sparsity sizeFactor Sample.Name2    label        batch
##       &lt;numeric&gt;  &lt;numeric&gt;  &lt;character&gt; &lt;factor&gt;  &lt;character&gt;
## 1      0.907503   2.645503 ETV6-RUNX1_1       11 ETV6-RUNX1_1
## 2      0.857361   4.791955 ETV6-RUNX1_1       9  ETV6-RUNX1_1
## 3      0.883637   3.267372 ETV6-RUNX1_1       2  ETV6-RUNX1_1
## 4      0.893091   3.420076 ETV6-RUNX1_1       13 ETV6-RUNX1_1
## 5      0.967515   0.562098 ETV6-RUNX1_1       6  ETV6-RUNX1_1
## 6      0.954400   0.876512 ETV6-RUNX1_1       7  ETV6-RUNX1_1</code></pre>
<div class="sourceCode" id="cb1575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1575-1"><a href="dataSetIntegration-allSetsTop.html#cb1575-1"></a><span class="kw">length</span>(<span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span></code></pre></div>
<pre><code>## [1] 2853</code></pre>
<div class="sourceCode" id="cb1577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1577-1"><a href="dataSetIntegration-allSetsTop.html#cb1577-1"></a><span class="co">#save.image(&quot;dataSetIntegrationWhole.debug.Rdata&quot;)</span></span>
<span id="cb1577-2"><a href="dataSetIntegration-allSetsTop.html#cb1577-2"></a></span>
<span id="cb1577-3"><a href="dataSetIntegration-allSetsTop.html#cb1577-3"></a><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]))</span></code></pre></div>
<pre><code>## 
## after 10 after 11 after 13 after 15 after 16 after 17 after 18 after 19 
##      400        1       98        2        1        1       12      132 
##  after 2 after 20 after 22 after 24 after 25  after 3 after 35 after 36 
##     1526        9        1       93      213        5        1      129 
## after 38 after 39  after 4 after 42 after 43 after 44 after 45 after 49 
##        4        4      105       56        1        1        2        2 
## after 51 after 53 after 54 after 58 after 59  after 6  after 7  after 8 
##        3       21        1        1        6        7        9        6</code></pre>
<div class="sourceCode" id="cb1579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1579-1"><a href="dataSetIntegration-allSetsTop.html#cb1579-1"></a><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span></code></pre></div>
<pre><code>## 
##  before 1 before 10 before 11 before 12 before 13 before 14  before 2  before 3 
##       406        93        59        61       466         7        65        31 
##  before 4  before 5  before 6  before 7  before 8  before 9 
##        84        32       664       340       483        62</code></pre>
<div class="sourceCode" id="cb1581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1581-1"><a href="dataSetIntegration-allSetsTop.html#cb1581-1"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb1581-2"><a href="dataSetIntegration-allSetsTop.html#cb1581-2"></a>             <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1581-3"><a href="dataSetIntegration-allSetsTop.html#cb1581-3"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb1581-4"><a href="dataSetIntegration-allSetsTop.html#cb1581-4"></a>  <span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb1581-5"><a href="dataSetIntegration-allSetsTop.html#cb1581-5"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb1581-6"><a href="dataSetIntegration-allSetsTop.html#cb1581-6"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1581-7"><a href="dataSetIntegration-allSetsTop.html#cb1581-7"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]),</span>
<span id="cb1581-8"><a href="dataSetIntegration-allSetsTop.html#cb1581-8"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1581-9"><a href="dataSetIntegration-allSetsTop.html#cb1581-9"></a></span>
<span id="cb1581-10"><a href="dataSetIntegration-allSetsTop.html#cb1581-10"></a><span class="co"># For the second batch.</span></span>
<span id="cb1581-11"><a href="dataSetIntegration-allSetsTop.html#cb1581-11"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1581-12"><a href="dataSetIntegration-allSetsTop.html#cb1581-12"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1581-13"><a href="dataSetIntegration-allSetsTop.html#cb1581-13"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb1581-14"><a href="dataSetIntegration-allSetsTop.html#cb1581-14"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1581-15"><a href="dataSetIntegration-allSetsTop.html#cb1581-15"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb1581-16"><a href="dataSetIntegration-allSetsTop.html#cb1581-16"></a>  <span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb1581-17"><a href="dataSetIntegration-allSetsTop.html#cb1581-17"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb1581-18"><a href="dataSetIntegration-allSetsTop.html#cb1581-18"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1581-19"><a href="dataSetIntegration-allSetsTop.html#cb1581-19"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]),</span>
<span id="cb1581-20"><a href="dataSetIntegration-allSetsTop.html#cb1581-20"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1581-21"><a href="dataSetIntegration-allSetsTop.html#cb1581-21"></a></span>
<span id="cb1581-22"><a href="dataSetIntegration-allSetsTop.html#cb1581-22"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-525-1.png" width="672" /></p>
<p>Co-assignment probabilities</p>
<div class="sourceCode" id="cb1582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1582-1"><a href="dataSetIntegration-allSetsTop.html#cb1582-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb1582-2"><a href="dataSetIntegration-allSetsTop.html#cb1582-2"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1582-3"><a href="dataSetIntegration-allSetsTop.html#cb1582-3"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1582-4"><a href="dataSetIntegration-allSetsTop.html#cb1582-4"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb1582-5"><a href="dataSetIntegration-allSetsTop.html#cb1582-5"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb1582-6"><a href="dataSetIntegration-allSetsTop.html#cb1582-6"></a>  tab,</span>
<span id="cb1582-7"><a href="dataSetIntegration-allSetsTop.html#cb1582-7"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb1582-8"><a href="dataSetIntegration-allSetsTop.html#cb1582-8"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1582-9"><a href="dataSetIntegration-allSetsTop.html#cb1582-9"></a>  <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1582-10"><a href="dataSetIntegration-allSetsTop.html#cb1582-10"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]),</span>
<span id="cb1582-11"><a href="dataSetIntegration-allSetsTop.html#cb1582-11"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1582-12"><a href="dataSetIntegration-allSetsTop.html#cb1582-12"></a></span>
<span id="cb1582-13"><a href="dataSetIntegration-allSetsTop.html#cb1582-13"></a><span class="co"># For the second batch.</span></span>
<span id="cb1582-14"><a href="dataSetIntegration-allSetsTop.html#cb1582-14"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1582-15"><a href="dataSetIntegration-allSetsTop.html#cb1582-15"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1582-16"><a href="dataSetIntegration-allSetsTop.html#cb1582-16"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb1582-17"><a href="dataSetIntegration-allSetsTop.html#cb1582-17"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb1582-18"><a href="dataSetIntegration-allSetsTop.html#cb1582-18"></a>  tab,</span>
<span id="cb1582-19"><a href="dataSetIntegration-allSetsTop.html#cb1582-19"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb1582-20"><a href="dataSetIntegration-allSetsTop.html#cb1582-20"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1582-21"><a href="dataSetIntegration-allSetsTop.html#cb1582-21"></a>  <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1582-22"><a href="dataSetIntegration-allSetsTop.html#cb1582-22"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]),</span>
<span id="cb1582-23"><a href="dataSetIntegration-allSetsTop.html#cb1582-23"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1582-24"><a href="dataSetIntegration-allSetsTop.html#cb1582-24"></a></span>
<span id="cb1582-25"><a href="dataSetIntegration-allSetsTop.html#cb1582-25"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-526-1.png" width="672" /></p>
<p>Rand index:</p>
<div class="sourceCode" id="cb1583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1583-1"><a href="dataSetIntegration-allSetsTop.html#cb1583-1"></a><span class="kw">library</span>(fossil)</span>
<span id="cb1583-2"><a href="dataSetIntegration-allSetsTop.html#cb1583-2"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1583-3"><a href="dataSetIntegration-allSetsTop.html#cb1583-3"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1583-4"><a href="dataSetIntegration-allSetsTop.html#cb1583-4"></a>ri1 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb1583-5"><a href="dataSetIntegration-allSetsTop.html#cb1583-5"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1583-6"><a href="dataSetIntegration-allSetsTop.html#cb1583-6"></a>ri1</span></code></pre></div>
<pre><code>## [1] 0.7285867</code></pre>
<div class="sourceCode" id="cb1585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1585-1"><a href="dataSetIntegration-allSetsTop.html#cb1585-1"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1585-2"><a href="dataSetIntegration-allSetsTop.html#cb1585-2"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb1585-3"><a href="dataSetIntegration-allSetsTop.html#cb1585-3"></a>ri2 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb1585-4"><a href="dataSetIntegration-allSetsTop.html#cb1585-4"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb1585-5"><a href="dataSetIntegration-allSetsTop.html#cb1585-5"></a>ri2</span></code></pre></div>
<pre><code>## [1] 0.7373159</code></pre>
<p>Cluster markers:</p>
<div class="sourceCode" id="cb1587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1587-1"><a href="dataSetIntegration-allSetsTop.html#cb1587-1"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(</span>
<span id="cb1587-2"><a href="dataSetIntegration-allSetsTop.html#cb1587-2"></a>  uncorrected,</span>
<span id="cb1587-3"><a href="dataSetIntegration-allSetsTop.html#cb1587-3"></a>  clusters.mnn,</span>
<span id="cb1587-4"><a href="dataSetIntegration-allSetsTop.html#cb1587-4"></a>  <span class="dt">block=</span>uncorrected<span class="op">$</span>batch,</span>
<span id="cb1587-5"><a href="dataSetIntegration-allSetsTop.html#cb1587-5"></a>  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1587-6"><a href="dataSetIntegration-allSetsTop.html#cb1587-6"></a>  <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb1587-7"><a href="dataSetIntegration-allSetsTop.html#cb1587-7"></a>  <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,</span>
<span id="cb1587-8"><a href="dataSetIntegration-allSetsTop.html#cb1587-8"></a>                                <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),</span>
<span id="cb1587-9"><a href="dataSetIntegration-allSetsTop.html#cb1587-9"></a>                                <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb1588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1588-1"><a href="dataSetIntegration-allSetsTop.html#cb1588-1"></a>demo &lt;-<span class="st"> </span>m.out[[<span class="st">&quot;1&quot;</span>]]</span>
<span id="cb1588-2"><a href="dataSetIntegration-allSetsTop.html#cb1588-2"></a><span class="kw">as.data.frame</span>(demo[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,<span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]) </span></code></pre></div>
<pre><code>##                  Symbol Top       p.value           FDR
## ENSG00000026025     VIM   1 3.261422e-169 4.623943e-166
## ENSG00000034510  TMSB10   1  3.417467e-25  9.401096e-23
## ENSG00000087086     FTL   1  1.000000e+00  1.000000e+00
## ENSG00000106123   EPHB6   1  2.120372e-95  1.563223e-92
## ENSG00000110848    CD69   1  4.005738e-09  5.273554e-07
## ENSG00000111348 ARHGDIB   1  2.204008e-47  9.027125e-45
## ENSG00000115085   ZAP70   1  6.211844e-02  1.000000e+00
## ENSG00000124766    SOX4   1 1.675786e-218 2.807856e-215
## ENSG00000136161  RCBTB2   1 1.185587e-124 1.092578e-121
## ENSG00000137731   FXYD2   1  1.220771e-62  5.769237e-60
## ENSG00000138326   RPS24   1  5.909752e-11  9.153163e-09
## ENSG00000161970   RPL26   1  0.000000e+00  0.000000e+00
## ENSG00000163682    RPL9   1  1.024456e-16  2.145653e-14
## ENSG00000167286    CD3D   1 1.863341e-238 5.723874e-235
## ENSG00000167526   RPL13   1  4.376031e-37  1.521786e-34
## ENSG00000169442    CD52   1  1.004943e-80  5.974871e-78
## ENSG00000173762     CD7   1  2.642735e-29  7.856169e-27
## ENSG00000184009   ACTG1   1 4.823072e-150 5.926270e-147
## ENSG00000198712  MT-CO2   1  3.039516e-05  3.314871e-03
## ENSG00000198755  RPL10A   1  3.001145e-20  7.375213e-18</code></pre>
<p>Expression level fot the top gene, VIM, (ENSG00000026025):</p>
<div class="sourceCode" id="cb1590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1590-1"><a href="dataSetIntegration-allSetsTop.html#cb1590-1"></a>geneEnsId &lt;-<span class="st"> </span><span class="kw">rownames</span>(demo)[<span class="dv">1</span>]</span>
<span id="cb1590-2"><a href="dataSetIntegration-allSetsTop.html#cb1590-2"></a></span>
<span id="cb1590-3"><a href="dataSetIntegration-allSetsTop.html#cb1590-3"></a><span class="kw">plotExpression</span>(</span>
<span id="cb1590-4"><a href="dataSetIntegration-allSetsTop.html#cb1590-4"></a>  uncorrected,</span>
<span id="cb1590-5"><a href="dataSetIntegration-allSetsTop.html#cb1590-5"></a>  <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)),</span>
<span id="cb1590-6"><a href="dataSetIntegration-allSetsTop.html#cb1590-6"></a>  <span class="dt">features=</span>geneEnsId,</span>
<span id="cb1590-7"><a href="dataSetIntegration-allSetsTop.html#cb1590-7"></a>  <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span></span>
<span id="cb1590-8"><a href="dataSetIntegration-allSetsTop.html#cb1590-8"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-530-1.png" width="672" /></p>
</div>
<div id="session-information-11" class="section level2">
<h2><span class="header-section-number">19.8</span> Session information</h2>
<div class="sourceCode" id="cb1591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1591-1"><a href="dataSetIntegration-allSetsTop.html#cb1591-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] edgeR_3.32.1                DelayedArray_0.16.3        
##  [3] RColorBrewer_1.1-2          reticulate_1.18            
##  [5] purrr_0.3.4                 tidyr_1.1.3                
##  [7] tibble_3.1.1                broom_0.7.6                
##  [9] leiden_0.3.7                cluster_2.1.0              
## [11] dynamicTreeCut_1.63-1       bluster_1.0.0              
## [13] fossil_0.4.0                shapefiles_0.7             
## [15] foreign_0.8-80              maps_3.3.0                 
## [17] sp_1.4-5                    clustree_0.4.3             
## [19] ggraph_2.0.5                pheatmap_1.0.12            
## [21] harmony_1.0                 Rcpp_1.0.6                 
## [23] batchelor_1.6.3             SeuratObject_4.0.0         
## [25] Seurat_4.0.1                limma_3.46.0               
## [27] ggfortify_0.4.11            BiocSingular_1.6.0         
## [29] scuttle_1.0.4               robustbase_0.93-7          
## [31] mixtools_1.2.0              Cairo_1.5-12.2             
## [33] dplyr_1.0.5                 DT_0.18                    
## [35] irlba_2.3.3                 biomaRt_2.46.3             
## [37] Matrix_1.2-18               igraph_1.2.6               
## [39] DropletUtils_1.10.3         scater_1.18.6              
## [41] ggplot2_3.3.3               scran_1.18.7               
## [43] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [45] Biobase_2.50.0              GenomicRanges_1.42.0       
## [47] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [49] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [51] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [53] knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.1            scattermore_0.7          
##   [3] R.methodsS3_1.8.1         bit64_4.0.5              
##   [5] R.utils_2.10.1            data.table_1.14.0        
##   [7] rpart_4.1-15              RCurl_1.98-1.3           
##   [9] generics_0.1.0            cowplot_1.1.1            
##  [11] RSQLite_2.2.6             RANN_2.6.1               
##  [13] future_1.21.0             bit_4.0.4                
##  [15] spatstat.data_2.1-0       xml2_1.3.2               
##  [17] httpuv_1.5.5              assertthat_0.2.1         
##  [19] viridis_0.5.1             xfun_0.22                
##  [21] hms_1.0.0                 jquerylib_0.1.3          
##  [23] evaluate_0.14             promises_1.1.1           
##  [25] DEoptimR_1.0-8            fansi_0.4.2              
##  [27] progress_1.2.2            dbplyr_2.0.0             
##  [29] DBI_1.1.1                 htmlwidgets_1.5.3        
##  [31] spatstat.geom_2.1-0       ellipsis_0.3.1           
##  [33] RSpectra_0.16-0           backports_1.2.1          
##  [35] bookdown_0.21             deldir_0.2-10            
##  [37] sparseMatrixStats_1.2.1   vctrs_0.3.7              
##  [39] ROCR_1.0-11               abind_1.4-5              
##  [41] cachem_1.0.4              withr_2.4.2              
##  [43] ggforce_0.3.3             checkmate_2.0.0          
##  [45] sctransform_0.3.2.9005    prettyunits_1.1.1        
##  [47] goftest_1.2-2             segmented_1.3-3          
##  [49] lazyeval_0.2.2            crayon_1.4.1             
##  [51] labeling_0.4.2            pkgconfig_2.0.3          
##  [53] tweenr_1.0.2              nlme_3.1-149             
##  [55] vipor_0.4.5               rlang_0.4.10             
##  [57] globals_0.14.0            lifecycle_1.0.0          
##  [59] miniUI_0.1.1.1            BiocFileCache_1.14.0     
##  [61] rsvd_1.0.3                polyclip_1.10-0          
##  [63] lmtest_0.9-38             Rhdf5lib_1.12.1          
##  [65] zoo_1.8-9                 beeswarm_0.2.3           
##  [67] ggridges_0.5.3            png_0.1-7                
##  [69] viridisLite_0.3.0         bitops_1.0-6             
##  [71] R.oo_1.24.0               KernSmooth_2.23-17       
##  [73] rhdf5filters_1.2.0        blob_1.2.1               
##  [75] DelayedMatrixStats_1.12.3 stringr_1.4.0            
##  [77] parallelly_1.24.0         readr_1.4.0              
##  [79] beachmat_2.6.4            scales_1.1.1             
##  [81] memoise_2.0.0             magrittr_2.0.1           
##  [83] plyr_1.8.6                ica_1.0-2                
##  [85] zlibbioc_1.36.0           compiler_4.0.3           
##  [87] dqrng_0.2.1               fitdistrplus_1.1-3       
##  [89] XVector_0.30.0            listenv_0.8.0            
##  [91] patchwork_1.1.1           pbapply_1.4-3            
##  [93] MASS_7.3-53               mgcv_1.8-33              
##  [95] tidyselect_1.1.0          stringi_1.5.3            
##  [97] highr_0.8                 yaml_2.2.1               
##  [99] askpass_1.1               locfit_1.5-9.4           
## [101] ggrepel_0.9.1             grid_4.0.3               
## [103] sass_0.3.1                tools_4.0.3              
## [105] future.apply_1.7.0        rstudioapi_0.13          
## [107] gridExtra_2.3             farver_2.1.0             
## [109] Rtsne_0.15                digest_0.6.27            
## [111] FNN_1.1.3                 shiny_1.5.0              
## [113] later_1.1.0.1             RcppAnnoy_0.0.18         
## [115] httr_1.4.2                AnnotationDbi_1.52.0     
## [117] kernlab_0.9-29            colorspace_2.0-0         
## [119] XML_3.99-0.5              tensor_1.5               
## [121] splines_4.0.3             uwot_0.1.10              
## [123] statmod_1.4.35            spatstat.utils_2.1-0     
## [125] graphlayouts_0.7.1        plotly_4.9.3             
## [127] xtable_1.8-4              jsonlite_1.7.2           
## [129] tidygraph_1.2.0           R6_2.5.0                 
## [131] pillar_1.6.0              htmltools_0.5.1.1        
## [133] mime_0.10                 glue_1.4.2               
## [135] fastmap_1.1.0             BiocParallel_1.24.1      
## [137] BiocNeighbors_1.8.2       codetools_0.2-16         
## [139] utf8_1.2.1                lattice_0.20-41          
## [141] bslib_0.2.4               spatstat.sparse_2.0-0    
## [143] ResidualMatrix_1.0.0      curl_4.3                 
## [145] ggbeeswarm_0.6.0          openssl_1.4.3            
## [147] survival_3.2-7            rmarkdown_2.7            
## [149] munsell_0.5.0             rhdf5_2.34.0             
## [151] GenomeInfoDbData_1.2.4    HDF5Array_1.18.1         
## [153] reshape2_1.4.4            gtable_0.3.0             
## [155] spatstat.core_2.1-2</code></pre>

<!--
  setSuf: '_allCells'
source /home/baller01/.local/share/r-miniconda/bin/activate r-reticulate
source /home/baller01/.local/share/r-miniconda/bin/deactivate r-reticulate
-->
<div class="sourceCode" id="cb1593"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1593-1"><a href="dataSetIntegration-allSetsTop.html#cb1593-1"></a><span class="co"># OLD</span></span>
<span id="cb1593-2"><a href="dataSetIntegration-allSetsTop.html#cb1593-2"></a><span class="co">#projDir &lt;- &quot;/mnt/scratcha/bioinformatics/baller01/20200511_FernandesM_ME_crukBiSs2020&quot;</span></span>
<span id="cb1593-3"><a href="dataSetIntegration-allSetsTop.html#cb1593-3"></a>projDir &lt;-<span class="st"> &quot;/data/personal/baller01/20200511_FernandesM_ME_crukBiSs2020&quot;</span></span>
<span id="cb1593-4"><a href="dataSetIntegration-allSetsTop.html#cb1593-4"></a></span>
<span id="cb1593-5"><a href="dataSetIntegration-allSetsTop.html#cb1593-5"></a>outDirBit &lt;-<span class="st"> &quot;AnaWiSce/Attempt3&quot;</span></span>
<span id="cb1593-6"><a href="dataSetIntegration-allSetsTop.html#cb1593-6"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1593-7"><a href="dataSetIntegration-allSetsTop.html#cb1593-7"></a>splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span> <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1593-8"><a href="dataSetIntegration-allSetsTop.html#cb1593-8"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting</span></span></code></pre></div>
<div class="sourceCode" id="cb1594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1594-1"><a href="dataSetIntegration-allSetsTop.html#cb1594-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb1594-2"><a href="dataSetIntegration-allSetsTop.html#cb1594-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb1594-3"><a href="dataSetIntegration-allSetsTop.html#cb1594-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb1594-4"><a href="dataSetIntegration-allSetsTop.html#cb1594-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb1594-5"><a href="dataSetIntegration-allSetsTop.html#cb1594-5"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb1594-6"><a href="dataSetIntegration-allSetsTop.html#cb1594-6"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb1594-7"><a href="dataSetIntegration-allSetsTop.html#cb1594-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb1594-8"><a href="dataSetIntegration-allSetsTop.html#cb1594-8"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1594-9"><a href="dataSetIntegration-allSetsTop.html#cb1594-9"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1594-10"><a href="dataSetIntegration-allSetsTop.html#cb1594-10"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1594-11"><a href="dataSetIntegration-allSetsTop.html#cb1594-11"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span></span>
<span id="cb1594-12"><a href="dataSetIntegration-allSetsTop.html#cb1594-12"></a>    setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb1594-13"><a href="dataSetIntegration-allSetsTop.html#cb1594-13"></a>    dsiSuf &lt;-<span class="st"> &#39;_dsi&#39;</span></span>
<span id="cb1594-14"><a href="dataSetIntegration-allSetsTop.html#cb1594-14"></a>}</span>
<span id="cb1594-15"><a href="dataSetIntegration-allSetsTop.html#cb1594-15"></a></span>
<span id="cb1594-16"><a href="dataSetIntegration-allSetsTop.html#cb1594-16"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1594-17"><a href="dataSetIntegration-allSetsTop.html#cb1594-17"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1594-18"><a href="dataSetIntegration-allSetsTop.html#cb1594-18"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1594-19"><a href="dataSetIntegration-allSetsTop.html#cb1594-19"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span>
<span id="cb1594-20"><a href="dataSetIntegration-allSetsTop.html#cb1594-20"></a></span>
<span id="cb1594-21"><a href="dataSetIntegration-allSetsTop.html#cb1594-21"></a>anaStg &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;&#39;%s_%s&#39;&quot;</span>, splSetToGet2, setSuf)</span>
<span id="cb1594-22"><a href="dataSetIntegration-allSetsTop.html#cb1594-22"></a></span>
<span id="cb1594-23"><a href="dataSetIntegration-allSetsTop.html#cb1594-23"></a><span class="co">#displayToUse &lt;- Sys.getenv(&quot;DISPLAY&quot;)</span></span>
<span id="cb1594-24"><a href="dataSetIntegration-allSetsTop.html#cb1594-24"></a><span class="co">#Cairo::CairoX11(display=displayToUse);</span></span></code></pre></div>
<div class="sourceCode" id="cb1595"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1595-1"><a href="dataSetIntegration-allSetsTop.html#cb1595-1"></a><span class="co">#```{r setup, include=FALSE, echo=FALSE}</span></span>
<span id="cb1595-2"><a href="dataSetIntegration-allSetsTop.html#cb1595-2"></a>    <span class="co"># removing chunk name for now because would need to make it dynamic, eg {{anaStg}} and expand</span></span>
<span id="cb1595-3"><a href="dataSetIntegration-allSetsTop.html#cb1595-3"></a>    <span class="co"># TODO do that later</span></span>
<span id="cb1595-4"><a href="dataSetIntegration-allSetsTop.html#cb1595-4"></a></span>
<span id="cb1595-5"><a href="dataSetIntegration-allSetsTop.html#cb1595-5"></a><span class="co"># First, set some variables:</span></span>
<span id="cb1595-6"><a href="dataSetIntegration-allSetsTop.html#cb1595-6"></a><span class="kw">require</span>(knitr)</span>
<span id="cb1595-7"><a href="dataSetIntegration-allSetsTop.html#cb1595-7"></a>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">error=</span><span class="ot">FALSE</span>, <span class="dt">message=</span><span class="ot">FALSE</span>, <span class="dt">warning=</span><span class="ot">FALSE</span>, <span class="dt">cache=</span>cacheBool)</span>
<span id="cb1595-8"><a href="dataSetIntegration-allSetsTop.html#cb1595-8"></a>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1595-9"><a href="dataSetIntegration-allSetsTop.html#cb1595-9"></a>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">eval =</span> <span class="ot">TRUE</span>) </span>
<span id="cb1595-10"><a href="dataSetIntegration-allSetsTop.html#cb1595-10"></a>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">dev=</span><span class="st">&quot;CairoPNG&quot;</span>)</span>
<span id="cb1595-11"><a href="dataSetIntegration-allSetsTop.html#cb1595-11"></a>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">fig.width=</span><span class="dv">7</span>, <span class="dt">fig.height=</span><span class="dv">7</span>)</span>
<span id="cb1595-12"><a href="dataSetIntegration-allSetsTop.html#cb1595-12"></a><span class="co">#opts_chunk$set(dev = c(&quot;png&quot;))</span></span>
<span id="cb1595-13"><a href="dataSetIntegration-allSetsTop.html#cb1595-13"></a><span class="co">#options(bitmapType=&#39;cairo&#39;)</span></span>
<span id="cb1595-14"><a href="dataSetIntegration-allSetsTop.html#cb1595-14"></a><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1595-15"><a href="dataSetIntegration-allSetsTop.html#cb1595-15"></a><span class="kw">set.seed</span>(<span class="dv">123</span>) <span class="co"># for reproducibility</span></span></code></pre></div>
<div class="sourceCode" id="cb1596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1596-1"><a href="dataSetIntegration-allSetsTop.html#cb1596-1"></a><span class="kw">capabilities</span>(<span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## cairo 
##  TRUE</code></pre>
<div class="sourceCode" id="cb1598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1598-1"><a href="dataSetIntegration-allSetsTop.html#cb1598-1"></a><span class="kw">getOption</span>(<span class="st">&#39;bitmapType&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Xlib&quot;</code></pre>
<div class="sourceCode" id="cb1600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1600-1"><a href="dataSetIntegration-allSetsTop.html#cb1600-1"></a><span class="kw">options</span>(<span class="dt">bitmapType=</span><span class="st">&#39;cairo&#39;</span>)</span>
<span id="cb1600-2"><a href="dataSetIntegration-allSetsTop.html#cb1600-2"></a><span class="kw">getOption</span>(<span class="st">&#39;bitmapType&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;cairo&quot;</code></pre>
<div class="sourceCode" id="cb1602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1602-1"><a href="dataSetIntegration-allSetsTop.html#cb1602-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb1602-2"><a href="dataSetIntegration-allSetsTop.html#cb1602-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb1602-3"><a href="dataSetIntegration-allSetsTop.html#cb1602-3"></a><span class="kw">library</span>(scran)</span>
<span id="cb1602-4"><a href="dataSetIntegration-allSetsTop.html#cb1602-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb1602-5"><a href="dataSetIntegration-allSetsTop.html#cb1602-5"></a><span class="kw">library</span>(dynamicTreeCut)</span>
<span id="cb1602-6"><a href="dataSetIntegration-allSetsTop.html#cb1602-6"></a><span class="kw">library</span>(cluster) <span class="co"># for silhouette</span></span>
<span id="cb1602-7"><a href="dataSetIntegration-allSetsTop.html#cb1602-7"></a><span class="kw">library</span>(igraph) <span class="co"># for graph-based clustering and plotting networks </span></span>
<span id="cb1602-8"><a href="dataSetIntegration-allSetsTop.html#cb1602-8"></a><span class="kw">library</span>(leiden) <span class="co"># for community detection</span></span>
<span id="cb1602-9"><a href="dataSetIntegration-allSetsTop.html#cb1602-9"></a></span>
<span id="cb1602-10"><a href="dataSetIntegration-allSetsTop.html#cb1602-10"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>), <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/dataSetIntegration_allSets.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
