<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Dimensionality reduction for analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Dimensionality reduction for analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Dimensionality reduction for analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="batch-correction-gsm3872442-set.html"/>
<link rel="next" href="DataIntegrationPbmmcTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.10" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.11" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-17"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html"><i class="fa fa-check"></i><b>24</b> Extract T-cells from HCA Dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#variance-modeling"><i class="fa fa-check"></i><b>24.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.2" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#data-integration"><i class="fa fa-check"></i><b>24.2</b> Data integration</a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#clustering-1"><i class="fa fa-check"></i><b>24.4</b> Clustering</a></li>
<li class="chapter" data-level="24.5" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#cell-type-classification"><i class="fa fa-check"></i><b>24.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.6" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#extract-t-cells"><i class="fa fa-check"></i><b>24.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.7" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#ackowledgements"><i class="fa fa-check"></i><b>24.7</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.8" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#session-information-18"><i class="fa fa-check"></i><b>24.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html"><i class="fa fa-check"></i><b>25</b> Setting up the data</a><ul>
<li class="chapter" data-level="25.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>25.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="25.2" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>25.2</b> Find temporally expressed genes</a><ul>
<li class="chapter" data-level="25.2.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>25.2.1</b> Pseudotime analysis for another HCA sample</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#exercise-1"><i class="fa fa-check"></i><b>25.3</b> Exercise 1</a></li>
<li class="chapter" data-level="25.4" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#ackowledgements-1"><i class="fa fa-check"></i><b>25.4</b> Ackowledgements</a></li>
<li class="chapter" data-level="25.5" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#session-information-19"><i class="fa fa-check"></i><b>25.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html"><i class="fa fa-check"></i><b>26</b> <strong>Pseudotime Alignment</strong></a><ul>
<li class="chapter" data-level="26.1" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#ackowledgements-2"><i class="fa fa-check"></i><b>26.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="26.2" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#session-information-20"><i class="fa fa-check"></i><b>26.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>27</b> Doublet detection</a><ul>
<li class="chapter" data-level="27.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>27.1</b> Learning objectives</a></li>
<li class="chapter" data-level="27.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>27.2</b> Set up analysis</a></li>
<li class="chapter" data-level="27.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-11"><i class="fa fa-check"></i><b>27.3</b> Load data</a></li>
<li class="chapter" data-level="27.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>27.4</b> Overview</a></li>
<li class="chapter" data-level="27.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>27.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="27.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>27.6</b> Further comments</a></li>
<li class="chapter" data-level="27.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-21"><i class="fa fa-check"></i><b>27.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dimensionality-reduction-for-analysis" class="section level1">
<h1><span class="header-section-number">Chapter 13</span> Dimensionality reduction for analysis</h1>
<p><strong>TODO</strong> add text and fix fig size.</p>
<div class="sourceCode" id="cb992"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb992-1"><a href="dimensionality-reduction-for-analysis.html#cb992-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb992-2"><a href="dimensionality-reduction-for-analysis.html#cb992-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb992-3"><a href="dimensionality-reduction-for-analysis.html#cb992-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb992-4"><a href="dimensionality-reduction-for-analysis.html#cb992-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb992-5"><a href="dimensionality-reduction-for-analysis.html#cb992-5"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb992-6"><a href="dimensionality-reduction-for-analysis.html#cb992-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb992-7"><a href="dimensionality-reduction-for-analysis.html#cb992-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb992-8"><a href="dimensionality-reduction-for-analysis.html#cb992-8"></a></span>
<span id="cb992-9"><a href="dimensionality-reduction-for-analysis.html#cb992-9"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>) {</span>
<span id="cb992-10"><a href="dimensionality-reduction-for-analysis.html#cb992-10"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb992-11"><a href="dimensionality-reduction-for-analysis.html#cb992-11"></a>    setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb992-12"><a href="dimensionality-reduction-for-analysis.html#cb992-12"></a>    dirRel &lt;-<span class="st"> &quot;..&quot;</span></span>
<span id="cb992-13"><a href="dimensionality-reduction-for-analysis.html#cb992-13"></a>}</span>
<span id="cb992-14"><a href="dimensionality-reduction-for-analysis.html#cb992-14"></a></span>
<span id="cb992-15"><a href="dimensionality-reduction-for-analysis.html#cb992-15"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span></code></pre></div>
<div class="sourceCode" id="cb993"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb993-1"><a href="dimensionality-reduction-for-analysis.html#cb993-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb993-2"><a href="dimensionality-reduction-for-analysis.html#cb993-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb993-3"><a href="dimensionality-reduction-for-analysis.html#cb993-3"></a><span class="kw">library</span>(scran)</span>
<span id="cb993-4"><a href="dimensionality-reduction-for-analysis.html#cb993-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb993-5"><a href="dimensionality-reduction-for-analysis.html#cb993-5"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>),</span>
<span id="cb993-6"><a href="dimensionality-reduction-for-analysis.html#cb993-6"></a>          <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
<div id="load-data-2" class="section level2">
<h2><span class="header-section-number">13.1</span> Load data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts for 500 cells per sample.</p>
<div class="sourceCode" id="cb994"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb994-1"><a href="dimensionality-reduction-for-analysis.html#cb994-1"></a><span class="co"># Read object in:</span></span>
<span id="cb994-2"><a href="dimensionality-reduction-for-analysis.html#cb994-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dimRed.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb994-3"><a href="dimensionality-reduction-for-analysis.html#cb994-3"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dimRed.Rds&quot;</code></pre>
<div class="sourceCode" id="cb996"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb996-1"><a href="dimensionality-reduction-for-analysis.html#cb996-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb996-2"><a href="dimensionality-reduction-for-analysis.html#cb996-2"></a>{</span>
<span id="cb996-3"><a href="dimensionality-reduction-for-analysis.html#cb996-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb996-4"><a href="dimensionality-reduction-for-analysis.html#cb996-4"></a>}</span>
<span id="cb996-5"><a href="dimensionality-reduction-for-analysis.html#cb996-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb996-6"><a href="dimensionality-reduction-for-analysis.html#cb996-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 16629 5500 
## metadata(0):
## assays(2): counts logcounts
## rownames(16629): ENSG00000237491 ENSG00000225880 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(16): Barcode Run ... cell_sparsity sizeFactor
## reducedDimNames(3): PCA TSNE UMAP
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb998"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb998-1"><a href="dimensionality-reduction-for-analysis.html#cb998-1"></a><span class="co">#any(duplicated(rowData(sce)$ensembl_gene_id))</span></span>
<span id="cb998-2"><a href="dimensionality-reduction-for-analysis.html#cb998-2"></a><span class="co"># some function(s) used below complain about &#39;strand&#39; already being used in row data,</span></span>
<span id="cb998-3"><a href="dimensionality-reduction-for-analysis.html#cb998-3"></a><span class="co"># so rename that column now:</span></span>
<span id="cb998-4"><a href="dimensionality-reduction-for-analysis.html#cb998-4"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span></span>
<span id="cb998-5"><a href="dimensionality-reduction-for-analysis.html#cb998-5"></a></span>
<span id="cb998-6"><a href="dimensionality-reduction-for-analysis.html#cb998-6"></a><span class="co"># also get feature selection outcome</span></span>
<span id="cb998-7"><a href="dimensionality-reduction-for-analysis.html#cb998-7"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_featSel.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb998-8"><a href="dimensionality-reduction-for-analysis.html#cb998-8"></a>tmpList &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb998-9"><a href="dimensionality-reduction-for-analysis.html#cb998-9"></a>dec.sce &lt;-<span class="st"> </span>tmpList<span class="op">$</span>dec.sce</span>
<span id="cb998-10"><a href="dimensionality-reduction-for-analysis.html#cb998-10"></a>hvg.index &lt;-<span class="st"> </span>tmpList<span class="op">$</span>hvg.index</span>
<span id="cb998-11"><a href="dimensionality-reduction-for-analysis.html#cb998-11"></a><span class="kw">rm</span>(tmpList)</span></code></pre></div>
</div>
<div id="denoising-expression-values-using-pca" class="section level2">
<h2><span class="header-section-number">13.2</span> Denoising expression values using PCA</h2>
<p>Aim: use the trend fitted above to identify PCs linked to biology.</p>
<p>Assumption: biology drives most of the variance hence should be captured by the first PCs, while technical noise affects each gene independently, hence is captured by later PCs.</p>
<p>Logic: Compute the sum of the technical component across genes used in the PCA, use it as the amount of variance not related to biology and that we should therefore remove. Later PCs are excluded until the amount of variance they account for matches that corresponding to the technical component.</p>
<div class="sourceCode" id="cb999"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb999-1"><a href="dimensionality-reduction-for-analysis.html#cb999-1"></a>var.fit &lt;-<span class="st"> </span><span class="kw">metadata</span>(dec.sce)</span>
<span id="cb999-2"><a href="dimensionality-reduction-for-analysis.html#cb999-2"></a></span>
<span id="cb999-3"><a href="dimensionality-reduction-for-analysis.html#cb999-3"></a><span class="co"># remove uninteresting PCs:</span></span>
<span id="cb999-4"><a href="dimensionality-reduction-for-analysis.html#cb999-4"></a><span class="co">#options(BiocSingularParam.default=IrlbaParam())</span></span>
<span id="cb999-5"><a href="dimensionality-reduction-for-analysis.html#cb999-5"></a>sce &lt;-<span class="st"> </span><span class="kw">denoisePCA</span>(sce, <span class="dt">technical=</span>var.fit<span class="op">$</span>trend, <span class="dt">assay.type=</span><span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb999-6"><a href="dimensionality-reduction-for-analysis.html#cb999-6"></a><span class="kw">rm</span>(var.fit)</span>
<span id="cb999-7"><a href="dimensionality-reduction-for-analysis.html#cb999-7"></a></span>
<span id="cb999-8"><a href="dimensionality-reduction-for-analysis.html#cb999-8"></a><span class="co"># check assay names, should see &#39;PCA&#39;:</span></span>
<span id="cb999-9"><a href="dimensionality-reduction-for-analysis.html#cb999-9"></a><span class="co">##assayNames(sce)</span></span>
<span id="cb999-10"><a href="dimensionality-reduction-for-analysis.html#cb999-10"></a><span class="co"># check dimension of the PC table:</span></span>
<span id="cb999-11"><a href="dimensionality-reduction-for-analysis.html#cb999-11"></a><span class="co">##dim(reducedDim(sce, &quot;PCA&quot;)) </span></span>
<span id="cb999-12"><a href="dimensionality-reduction-for-analysis.html#cb999-12"></a></span>
<span id="cb999-13"><a href="dimensionality-reduction-for-analysis.html#cb999-13"></a>sce.pca &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA&quot;</span>) <span class="co"># get copy of PCA matrix</span></span>
<span id="cb999-14"><a href="dimensionality-reduction-for-analysis.html#cb999-14"></a>tmpCol &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;grey&quot;</span>, nbPcToComp) <span class="co"># set colours to show selected PCs in green</span></span>
<span id="cb999-15"><a href="dimensionality-reduction-for-analysis.html#cb999-15"></a>tmpCol[<span class="dv">1</span><span class="op">:</span><span class="kw">dim</span>(sce.pca)[<span class="dv">2</span>]] &lt;-<span class="st"> &quot;green&quot;</span></span>
<span id="cb999-16"><a href="dimensionality-reduction-for-analysis.html#cb999-16"></a><span class="kw">barplot</span>(<span class="kw">attributes</span>(sce.pca)<span class="op">$</span>percentVar[<span class="dv">1</span><span class="op">:</span>nbPcToComp],</span>
<span id="cb999-17"><a href="dimensionality-reduction-for-analysis.html#cb999-17"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;Scree plot for the %s first PCs&quot;</span>, nbPcToComp),</span>
<span id="cb999-18"><a href="dimensionality-reduction-for-analysis.html#cb999-18"></a>        <span class="dt">names.arg=</span><span class="dv">1</span><span class="op">:</span>nbPcToComp,</span>
<span id="cb999-19"><a href="dimensionality-reduction-for-analysis.html#cb999-19"></a>        <span class="dt">col=</span>tmpCol,</span>
<span id="cb999-20"><a href="dimensionality-reduction-for-analysis.html#cb999-20"></a>        <span class="dt">cex.names =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/comp_denoisePCA-1.png" width="672" /></p>
<div class="sourceCode" id="cb1000"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1000-1"><a href="dimensionality-reduction-for-analysis.html#cb1000-1"></a><span class="co"># cumulative proportion of variance explained by selected PCs</span></span>
<span id="cb1000-2"><a href="dimensionality-reduction-for-analysis.html#cb1000-2"></a><span class="kw">cumsum</span>(<span class="kw">attributes</span>(sce.pca)<span class="op">$</span>percentVar)[<span class="dv">1</span><span class="op">:</span><span class="kw">dim</span>(sce.pca)[<span class="dv">2</span>]]</span></code></pre></div>
<pre><code>## [1]  9.871634 14.209766 16.592818 18.511972 19.903829 21.048455 21.995189</code></pre>
<div class="sourceCode" id="cb1002"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1002-1"><a href="dimensionality-reduction-for-analysis.html#cb1002-1"></a><span class="co"># plot on PC1 and PC2 plane:</span></span>
<span id="cb1002-2"><a href="dimensionality-reduction-for-analysis.html#cb1002-2"></a><span class="kw">plotPCA</span>(sce, <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/comp_denoisePCA-2.png" width="672" /></p>
<div class="sourceCode" id="cb1003"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1003-1"><a href="dimensionality-reduction-for-analysis.html#cb1003-1"></a><span class="co">#require(knitr); knit_exit()</span></span>
<span id="cb1003-2"><a href="dimensionality-reduction-for-analysis.html#cb1003-2"></a><span class="kw">rm</span>(sce.pca, tmpCol)</span></code></pre></div>
<p>Show cells on plane for PC1 and PC2:</p>
<div class="sourceCode" id="cb1004"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1004-1"><a href="dimensionality-reduction-for-analysis.html#cb1004-1"></a><span class="kw">plotReducedDim</span>(sce,</span>
<span id="cb1004-2"><a href="dimensionality-reduction-for-analysis.html#cb1004-2"></a>           <span class="dt">dimred =</span> <span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb1004-3"><a href="dimensionality-reduction-for-analysis.html#cb1004-3"></a>           <span class="dt">ncomponents =</span> <span class="dv">3</span>,</span>
<span id="cb1004-4"><a href="dimensionality-reduction-for-analysis.html#cb1004-4"></a>           <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span></span>
<span id="cb1004-5"><a href="dimensionality-reduction-for-analysis.html#cb1004-5"></a>fontsize</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_denoisePCA-1.png" width="672" /></p>
</div>
<div id="visualise-expression-patterns-of-some-hvgs" class="section level2">
<h2><span class="header-section-number">13.3</span> Visualise expression patterns of some HVGs</h2>
<div class="sourceCode" id="cb1005"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1005-1"><a href="dimensionality-reduction-for-analysis.html#cb1005-1"></a>o &lt;-<span class="st"> </span><span class="kw">order</span>(dec.sce<span class="op">$</span>bio, <span class="dt">decreasing=</span><span class="ot">TRUE</span>)</span>
<span id="cb1005-2"><a href="dimensionality-reduction-for-analysis.html#cb1005-2"></a>chosen.genes.index &lt;-<span class="st"> </span>o[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]</span>
<span id="cb1005-3"><a href="dimensionality-reduction-for-analysis.html#cb1005-3"></a>dec.sce <span class="op">%&gt;%</span></span>
<span id="cb1005-4"><a href="dimensionality-reduction-for-analysis.html#cb1005-4"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1005-5"><a href="dimensionality-reduction-for-analysis.html#cb1005-5"></a><span class="st">    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1005-6"><a href="dimensionality-reduction-for-analysis.html#cb1005-6"></a><span class="st">    </span><span class="kw">right_join</span>(<span class="kw">data.frame</span>(<span class="kw">rowData</span>(sce)[chosen.genes.index,]), <span class="dt">by=</span><span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1005-7"><a href="dimensionality-reduction-for-analysis.html#cb1005-7"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(ensembl_gene_id, Symbol, bio, FDR) <span class="op">%&gt;%</span></span>
<span id="cb1005-8"><a href="dimensionality-reduction-for-analysis.html#cb1005-8"></a><span class="st">    </span><span class="kw">arrange</span>(<span class="op">-</span>bio)</span></code></pre></div>
<pre><code>##    ensembl_gene_id   Symbol       bio           FDR
## 1  ENSG00000244734      HBB 10.641550  0.000000e+00
## 2  ENSG00000188536     HBA2  8.405716  0.000000e+00
## 3  ENSG00000206172     HBA1  7.626859  0.000000e+00
## 4  ENSG00000223609      HBD  3.545440 5.867441e-223
## 5  ENSG00000019582     CD74  2.879184  1.017888e-43
## 6  ENSG00000206177      HBM  2.609258 4.928194e-200
## 7  ENSG00000169877     AHSP  2.550093 3.711101e-179
## 8  ENSG00000204287  HLA-DRA  2.168673  1.190475e-24
## 9  ENSG00000133742      CA1  1.917362 6.557357e-133
## 10 ENSG00000158578    ALAS2  1.599942 2.922058e-145
## 11 ENSG00000147454 SLC25A37  1.599805  8.332453e-74
## 12 ENSG00000165949    IFI27  1.452955 5.306226e-148
## 13 ENSG00000100721    TCL1A  1.374755  5.414268e-14
## 14 ENSG00000090013    BLVRB  1.347283  1.732791e-76
## 15 ENSG00000170345      FOS  1.257980  8.310386e-09
## 16 ENSG00000145335     SNCA  1.163172  5.667289e-86
## 17 ENSG00000211592     IGKC  1.106157  4.275977e-30
## 18 ENSG00000143546   S100A8  1.100766  1.972941e-99
## 19 ENSG00000090382      LYZ  1.055593 2.611221e-102
## 20 ENSG00000170180     GYPA  1.040273  5.596387e-96</code></pre>
<div class="sourceCode" id="cb1007"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1007-1"><a href="dimensionality-reduction-for-analysis.html#cb1007-1"></a><span class="kw">rm</span>(dec.sce)</span></code></pre></div>
<p>On PCA plot:</p>
<div class="sourceCode" id="cb1008"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1008-1"><a href="dimensionality-reduction-for-analysis.html#cb1008-1"></a><span class="co"># make and store PCA plot for top HVG 1:</span></span>
<span id="cb1008-2"><a href="dimensionality-reduction-for-analysis.html#cb1008-2"></a>pca1 &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(sce,</span>
<span id="cb1008-3"><a href="dimensionality-reduction-for-analysis.html#cb1008-3"></a>               <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb1008-4"><a href="dimensionality-reduction-for-analysis.html#cb1008-4"></a>               <span class="dt">colour_by=</span><span class="kw">rowData</span>(sce)[chosen.genes.index[<span class="dv">1</span>],<span class="st">&quot;ensembl_gene_id&quot;</span>]) <span class="op">+</span></span>
<span id="cb1008-5"><a href="dimensionality-reduction-for-analysis.html#cb1008-5"></a>fontsize  <span class="co"># + coord_fixed()</span></span>
<span id="cb1008-6"><a href="dimensionality-reduction-for-analysis.html#cb1008-6"></a><span class="co"># make and store PCA plot for top HVG 2:</span></span>
<span id="cb1008-7"><a href="dimensionality-reduction-for-analysis.html#cb1008-7"></a>pca2 &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(sce,</span>
<span id="cb1008-8"><a href="dimensionality-reduction-for-analysis.html#cb1008-8"></a>               <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb1008-9"><a href="dimensionality-reduction-for-analysis.html#cb1008-9"></a>               <span class="dt">colour_by=</span><span class="kw">rowData</span>(sce)[chosen.genes.index[<span class="dv">2</span>],<span class="st">&quot;ensembl_gene_id&quot;</span>]) <span class="op">+</span></span>
<span id="cb1008-10"><a href="dimensionality-reduction-for-analysis.html#cb1008-10"></a>fontsize <span class="co"># + coord_fixed()</span></span>
<span id="cb1008-11"><a href="dimensionality-reduction-for-analysis.html#cb1008-11"></a></span>
<span id="cb1008-12"><a href="dimensionality-reduction-for-analysis.html#cb1008-12"></a>pca1</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_pca_HVGtop2-1.png" width="672" /></p>
<div class="sourceCode" id="cb1009"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1009-1"><a href="dimensionality-reduction-for-analysis.html#cb1009-1"></a>pca2</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_pca_HVGtop2-2.png" width="672" /></p>
<div class="sourceCode" id="cb1010"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1010-1"><a href="dimensionality-reduction-for-analysis.html#cb1010-1"></a><span class="co"># display plots next to each other:</span></span>
<span id="cb1010-2"><a href="dimensionality-reduction-for-analysis.html#cb1010-2"></a><span class="co"># multiplot(pca1, pca2, cols=2)</span></span>
<span id="cb1010-3"><a href="dimensionality-reduction-for-analysis.html#cb1010-3"></a></span>
<span id="cb1010-4"><a href="dimensionality-reduction-for-analysis.html#cb1010-4"></a>pca1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name) <span class="op">+</span><span class="st"> </span><span class="kw">coord_fixed</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_pca_HVGtop2_facet-1.png" width="1152" /></p>
<div class="sourceCode" id="cb1011"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1011-1"><a href="dimensionality-reduction-for-analysis.html#cb1011-1"></a>pca2 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name) <span class="op">+</span><span class="st"> </span><span class="kw">coord_fixed</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_pca_HVGtop2_facet-2.png" width="1152" /></p>
<div class="sourceCode" id="cb1012"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1012-1"><a href="dimensionality-reduction-for-analysis.html#cb1012-1"></a><span class="kw">rm</span>(pca1, pca2)</span></code></pre></div>
<p>On t-SNE plot:</p>
<div class="sourceCode" id="cb1013"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1013-1"><a href="dimensionality-reduction-for-analysis.html#cb1013-1"></a><span class="co"># plot TSNE, accessing counts for the gene of interest with the ID used to name rows in the count matrix:</span></span>
<span id="cb1013-2"><a href="dimensionality-reduction-for-analysis.html#cb1013-2"></a><span class="co"># make and store TSNE plot for top HVG 1:</span></span>
<span id="cb1013-3"><a href="dimensionality-reduction-for-analysis.html#cb1013-3"></a>tsne1 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="kw">rowData</span>(sce)[chosen.genes.index[<span class="dv">1</span>],<span class="st">&quot;ensembl_gene_id&quot;</span>]) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1013-4"><a href="dimensionality-reduction-for-analysis.html#cb1013-4"></a><span class="co"># make and store TSNE plot for top HVG 2:</span></span>
<span id="cb1013-5"><a href="dimensionality-reduction-for-analysis.html#cb1013-5"></a>tsne2 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="kw">rowData</span>(sce)[chosen.genes.index[<span class="dv">2</span>],<span class="st">&quot;ensembl_gene_id&quot;</span>]) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1013-6"><a href="dimensionality-reduction-for-analysis.html#cb1013-6"></a></span>
<span id="cb1013-7"><a href="dimensionality-reduction-for-analysis.html#cb1013-7"></a>tsne1</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_tsne_HVGtop2-1.png" width="672" /></p>
<div class="sourceCode" id="cb1014"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1014-1"><a href="dimensionality-reduction-for-analysis.html#cb1014-1"></a>tsne2</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_tsne_HVGtop2-2.png" width="672" /></p>
<div class="sourceCode" id="cb1015"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1015-1"><a href="dimensionality-reduction-for-analysis.html#cb1015-1"></a><span class="co"># display plots next to each other:</span></span>
<span id="cb1015-2"><a href="dimensionality-reduction-for-analysis.html#cb1015-2"></a><span class="co">#multiplot(tsne1, tsne2, cols=2)</span></span>
<span id="cb1015-3"><a href="dimensionality-reduction-for-analysis.html#cb1015-3"></a></span>
<span id="cb1015-4"><a href="dimensionality-reduction-for-analysis.html#cb1015-4"></a>tsne1 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_tsne_HVGtop2_facet-1.png" width="1152" /></p>
<div class="sourceCode" id="cb1016"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1016-1"><a href="dimensionality-reduction-for-analysis.html#cb1016-1"></a>tsne2 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_count_tsne_HVGtop2_facet-2.png" width="1152" /></p>
<div class="sourceCode" id="cb1017"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1017-1"><a href="dimensionality-reduction-for-analysis.html#cb1017-1"></a><span class="co"># display plots next to each other, splitting each by sample:</span></span>
<span id="cb1017-2"><a href="dimensionality-reduction-for-analysis.html#cb1017-2"></a><span class="co">#multiplot(tsne1 + facet_grid(. ~ sce$Sample2), tsne2 + facet_grid(. ~ sce$Sample2), cols=2)</span></span>
<span id="cb1017-3"><a href="dimensionality-reduction-for-analysis.html#cb1017-3"></a></span>
<span id="cb1017-4"><a href="dimensionality-reduction-for-analysis.html#cb1017-4"></a><span class="kw">rm</span>(tsne1, tsne2)</span></code></pre></div>
<p>Write R object to file</p>
<div class="sourceCode" id="cb1018"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1018-1"><a href="dimensionality-reduction-for-analysis.html#cb1018-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_denoised.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1018-2"><a href="dimensionality-reduction-for-analysis.html#cb1018-2"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_denoised.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1020"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1020-1"><a href="dimensionality-reduction-for-analysis.html#cb1020-1"></a><span class="kw">saveRDS</span>(sce, <span class="dt">file=</span>tmpFn)</span></code></pre></div>
<p>Genes correlated with PC1:</p>
<!-- http://bioinformatics.age.mpg.de/presentations-tutorials/presentations/modules/single-cell//bioconductor_tutorial.html -->
<div class="sourceCode" id="cb1021"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1021-1"><a href="dimensionality-reduction-for-analysis.html#cb1021-1"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1021-2"><a href="dimensionality-reduction-for-analysis.html#cb1021-2"></a><span class="kw">rownames</span>(sce) &lt;-<span class="st"> </span><span class="kw">uniquifyFeatureNames</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id,</span>
<span id="cb1021-3"><a href="dimensionality-reduction-for-analysis.html#cb1021-3"></a>                      <span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol)</span>
<span id="cb1021-4"><a href="dimensionality-reduction-for-analysis.html#cb1021-4"></a>pc1 &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA&quot;</span>)[,<span class="dv">1</span>]</span>
<span id="cb1021-5"><a href="dimensionality-reduction-for-analysis.html#cb1021-5"></a>design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>pc1)</span>
<span id="cb1021-6"><a href="dimensionality-reduction-for-analysis.html#cb1021-6"></a><span class="kw">library</span>(limma)</span>
<span id="cb1021-7"><a href="dimensionality-reduction-for-analysis.html#cb1021-7"></a>fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(<span class="kw">logcounts</span>(sce), design)</span>
<span id="cb1021-8"><a href="dimensionality-reduction-for-analysis.html#cb1021-8"></a>fit &lt;-<span class="st"> </span><span class="kw">eBayes</span>(fit, <span class="dt">trend=</span><span class="ot">TRUE</span>, <span class="dt">robust=</span><span class="ot">TRUE</span>)</span>
<span id="cb1021-9"><a href="dimensionality-reduction-for-analysis.html#cb1021-9"></a>topTab &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit)</span>
<span id="cb1021-10"><a href="dimensionality-reduction-for-analysis.html#cb1021-10"></a><span class="kw">head</span>(topTab)</span></code></pre></div>
<pre><code>##           logFC   AveExpr        t P.Value adj.P.Val        B
## ALAS2 0.1310205 0.4024173 253.4874       0         0 6982.458
## HBD   0.1897173 0.7859679 236.4959       0         0 6632.414
## HBM   0.1623695 0.5804766 218.6957       0         0 6242.859
## AHSP  0.1609771 0.6023857 213.2943       0         0 6119.639
## HBA1  0.2684543 1.6248769 208.5281       0         0 6008.861
## HBA2  0.2777881 1.8690151 189.7412       0         0 5552.362</code></pre>
<div class="sourceCode" id="cb1023"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1023-1"><a href="dimensionality-reduction-for-analysis.html#cb1023-1"></a><span class="kw">rm</span>(topTab)</span></code></pre></div>
<div class="sourceCode" id="cb1024"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1024-1"><a href="dimensionality-reduction-for-analysis.html#cb1024-1"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1024-2"><a href="dimensionality-reduction-for-analysis.html#cb1024-2"></a>de.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(<span class="kw">topTable</span>(fit, <span class="dt">coef=</span><span class="dv">2</span>, <span class="dt">n=</span><span class="dv">50</span>))</span>
<span id="cb1024-3"><a href="dimensionality-reduction-for-analysis.html#cb1024-3"></a>heat.vals &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce)[de.genes,]</span>
<span id="cb1024-4"><a href="dimensionality-reduction-for-analysis.html#cb1024-4"></a>heat.vals &lt;-<span class="st"> </span>heat.vals <span class="op">-</span><span class="st"> </span><span class="kw">rowMeans</span>(heat.vals)</span>
<span id="cb1024-5"><a href="dimensionality-reduction-for-analysis.html#cb1024-5"></a>heat.vals[heat.vals <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1024-6"><a href="dimensionality-reduction-for-analysis.html#cb1024-6"></a>heat.vals[heat.vals <span class="op">&lt;</span><span class="st"> </span><span class="dv">-2</span>] &lt;-<span class="st"> </span><span class="dv">-2</span></span>
<span id="cb1024-7"><a href="dimensionality-reduction-for-analysis.html#cb1024-7"></a>p &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(heat.vals[,<span class="kw">order</span>(pc1)], <span class="dt">cluster_cols=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/heatmap_dimRedForAna-1.png" width="672" /></p>
<div class="sourceCode" id="cb1025"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1025-1"><a href="dimensionality-reduction-for-analysis.html#cb1025-1"></a><span class="co">#plot(p$gtable)</span></span>
<span id="cb1025-2"><a href="dimensionality-reduction-for-analysis.html#cb1025-2"></a><span class="kw">rm</span>(pc1, design, fit, de.genes, heat.vals)</span>
<span id="cb1025-3"><a href="dimensionality-reduction-for-analysis.html#cb1025-3"></a><span class="kw">rm</span>(sce)</span></code></pre></div>
</div>
<div id="session-information-9" class="section level2">
<h2><span class="header-section-number">13.4</span> Session information</h2>
<div class="sourceCode" id="cb1026"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1026-1"><a href="dimensionality-reduction-for-analysis.html#cb1026-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] edgeR_3.32.1                DelayedArray_0.16.3        
##  [3] RColorBrewer_1.1-2          reticulate_1.18            
##  [5] purrr_0.3.4                 tidyr_1.1.3                
##  [7] tibble_3.1.1                broom_0.7.6                
##  [9] leiden_0.3.7                cluster_2.1.0              
## [11] dynamicTreeCut_1.63-1       bluster_1.0.0              
## [13] fossil_0.4.0                shapefiles_0.7             
## [15] foreign_0.8-80              maps_3.3.0                 
## [17] sp_1.4-5                    clustree_0.4.3             
## [19] ggraph_2.0.5                pheatmap_1.0.12            
## [21] harmony_1.0                 Rcpp_1.0.6                 
## [23] batchelor_1.6.3             SeuratObject_4.0.0         
## [25] Seurat_4.0.1                limma_3.46.0               
## [27] ggfortify_0.4.11            BiocSingular_1.6.0         
## [29] scuttle_1.0.4               robustbase_0.93-7          
## [31] mixtools_1.2.0              Cairo_1.5-12.2             
## [33] dplyr_1.0.5                 DT_0.18                    
## [35] irlba_2.3.3                 biomaRt_2.46.3             
## [37] Matrix_1.2-18               igraph_1.2.6               
## [39] DropletUtils_1.10.3         scater_1.18.6              
## [41] ggplot2_3.3.3               scran_1.18.7               
## [43] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [45] Biobase_2.50.0              GenomicRanges_1.42.0       
## [47] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [49] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [51] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [53] knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.1            scattermore_0.7          
##   [3] R.methodsS3_1.8.1         bit64_4.0.5              
##   [5] R.utils_2.10.1            data.table_1.14.0        
##   [7] rpart_4.1-15              RCurl_1.98-1.3           
##   [9] generics_0.1.0            cowplot_1.1.1            
##  [11] RSQLite_2.2.6             RANN_2.6.1               
##  [13] future_1.21.0             bit_4.0.4                
##  [15] spatstat.data_2.1-0       xml2_1.3.2               
##  [17] httpuv_1.5.5              assertthat_0.2.1         
##  [19] viridis_0.5.1             xfun_0.22                
##  [21] hms_1.0.0                 jquerylib_0.1.3          
##  [23] evaluate_0.14             promises_1.1.1           
##  [25] DEoptimR_1.0-8            fansi_0.4.2              
##  [27] progress_1.2.2            dbplyr_2.0.0             
##  [29] DBI_1.1.1                 htmlwidgets_1.5.3        
##  [31] spatstat.geom_2.1-0       ellipsis_0.3.1           
##  [33] backports_1.2.1           bookdown_0.21            
##  [35] deldir_0.2-10             sparseMatrixStats_1.2.1  
##  [37] vctrs_0.3.7               ROCR_1.0-11              
##  [39] abind_1.4-5               cachem_1.0.4             
##  [41] withr_2.4.2               ggforce_0.3.3            
##  [43] sctransform_0.3.2.9005    prettyunits_1.1.1        
##  [45] goftest_1.2-2             segmented_1.3-3          
##  [47] lazyeval_0.2.2            crayon_1.4.1             
##  [49] labeling_0.4.2            pkgconfig_2.0.3          
##  [51] tweenr_1.0.2              nlme_3.1-149             
##  [53] vipor_0.4.5               rlang_0.4.10             
##  [55] globals_0.14.0            lifecycle_1.0.0          
##  [57] miniUI_0.1.1.1            BiocFileCache_1.14.0     
##  [59] rsvd_1.0.3                polyclip_1.10-0          
##  [61] lmtest_0.9-38             Rhdf5lib_1.12.1          
##  [63] zoo_1.8-9                 beeswarm_0.2.3           
##  [65] ggridges_0.5.3            png_0.1-7                
##  [67] viridisLite_0.3.0         bitops_1.0-6             
##  [69] R.oo_1.24.0               KernSmooth_2.23-17       
##  [71] rhdf5filters_1.2.0        blob_1.2.1               
##  [73] DelayedMatrixStats_1.12.3 stringr_1.4.0            
##  [75] parallelly_1.24.0         readr_1.4.0              
##  [77] beachmat_2.6.4            scales_1.1.1             
##  [79] memoise_2.0.0             magrittr_2.0.1           
##  [81] plyr_1.8.6                ica_1.0-2                
##  [83] zlibbioc_1.36.0           compiler_4.0.3           
##  [85] dqrng_0.2.1               fitdistrplus_1.1-3       
##  [87] XVector_0.30.0            listenv_0.8.0            
##  [89] patchwork_1.1.1           pbapply_1.4-3            
##  [91] MASS_7.3-53               mgcv_1.8-33              
##  [93] tidyselect_1.1.0          stringi_1.5.3            
##  [95] highr_0.8                 yaml_2.2.1               
##  [97] askpass_1.1               locfit_1.5-9.4           
##  [99] ggrepel_0.9.1             grid_4.0.3               
## [101] sass_0.3.1                tools_4.0.3              
## [103] future.apply_1.7.0        rstudioapi_0.13          
## [105] gridExtra_2.3             farver_2.1.0             
## [107] Rtsne_0.15                digest_0.6.27            
## [109] shiny_1.5.0               later_1.1.0.1            
## [111] RcppAnnoy_0.0.18          httr_1.4.2               
## [113] AnnotationDbi_1.52.0      kernlab_0.9-29           
## [115] colorspace_2.0-0          XML_3.99-0.5             
## [117] tensor_1.5                splines_4.0.3            
## [119] uwot_0.1.10               statmod_1.4.35           
## [121] spatstat.utils_2.1-0      graphlayouts_0.7.1       
## [123] plotly_4.9.3              xtable_1.8-4             
## [125] jsonlite_1.7.2            tidygraph_1.2.0          
## [127] R6_2.5.0                  pillar_1.6.0             
## [129] htmltools_0.5.1.1         mime_0.10                
## [131] glue_1.4.2                fastmap_1.1.0            
## [133] BiocParallel_1.24.1       BiocNeighbors_1.8.2      
## [135] codetools_0.2-16          utf8_1.2.1               
## [137] lattice_0.20-41           bslib_0.2.4              
## [139] spatstat.sparse_2.0-0     ResidualMatrix_1.0.0     
## [141] curl_4.3                  ggbeeswarm_0.6.0         
## [143] openssl_1.4.3             survival_3.2-7           
## [145] rmarkdown_2.7             munsell_0.5.0            
## [147] rhdf5_2.34.0              GenomeInfoDbData_1.2.4   
## [149] HDF5Array_1.18.1          reshape2_1.4.4           
## [151] gtable_0.3.0              spatstat.core_2.1-2</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="batch-correction-gsm3872442-set.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="DataIntegrationPbmmcTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/dimRedForAna.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
