<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Dimensionality reduction for visualisation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Dimensionality reduction for visualisation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Dimensionality reduction for visualisation | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="NormalisationGSM3872434Top.html"/>
<link rel="next" href="identifying-confounding-factors-caron-set.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.10" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.11" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-17"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html"><i class="fa fa-check"></i><b>24</b> Extract T-cells from HCA Dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#variance-modeling"><i class="fa fa-check"></i><b>24.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.2" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#data-integration"><i class="fa fa-check"></i><b>24.2</b> Data integration</a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#clustering-1"><i class="fa fa-check"></i><b>24.4</b> Clustering</a></li>
<li class="chapter" data-level="24.5" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#cell-type-classification"><i class="fa fa-check"></i><b>24.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.6" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#extract-t-cells"><i class="fa fa-check"></i><b>24.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.7" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#ackowledgements"><i class="fa fa-check"></i><b>24.7</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.8" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#session-information-18"><i class="fa fa-check"></i><b>24.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html"><i class="fa fa-check"></i><b>25</b> Setting up the data</a><ul>
<li class="chapter" data-level="25.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>25.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="25.2" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>25.2</b> Find temporally expressed genes</a><ul>
<li class="chapter" data-level="25.2.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>25.2.1</b> Pseudotime analysis for another HCA sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="exercise-1.html"><a href="exercise-1.html"><i class="fa fa-check"></i><b>26</b> Exercise 1</a><ul>
<li class="chapter" data-level="26.1" data-path="exercise-1.html"><a href="exercise-1.html#ackowledgements-1"><i class="fa fa-check"></i><b>26.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="26.2" data-path="exercise-1.html"><a href="exercise-1.html#session-information-19"><i class="fa fa-check"></i><b>26.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html"><i class="fa fa-check"></i><b>27</b> <strong>Pseudotime Alignment</strong></a><ul>
<li class="chapter" data-level="27.1" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#ackowledgements-2"><i class="fa fa-check"></i><b>27.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="27.2" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#session-information-20"><i class="fa fa-check"></i><b>27.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>28</b> Doublet detection</a><ul>
<li class="chapter" data-level="28.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>28.2</b> Set up analysis</a></li>
<li class="chapter" data-level="28.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-11"><i class="fa fa-check"></i><b>28.3</b> Load data</a></li>
<li class="chapter" data-level="28.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>28.4</b> Overview</a></li>
<li class="chapter" data-level="28.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>28.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="28.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>28.6</b> Further comments</a></li>
<li class="chapter" data-level="28.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-21"><i class="fa fa-check"></i><b>28.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dimRedForVizTop" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Dimensionality reduction for visualisation</h1>
<!--
Should have Caron and HCA separately
-->
<div class="sourceCode" id="cb785"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb785-1"><a href="dimRedForVizTop.html#cb785-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb785-2"><a href="dimRedForVizTop.html#cb785-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb785-3"><a href="dimRedForVizTop.html#cb785-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span></code></pre></div>
<p>In part 1 we gathered the data, aligned reads, checked quality, and normalised read counts. We will now identify genes to focus on, use visualisation to explore the data, collapse the data set, cluster cells by their expression profile and identify genes that best characterise these cell populations. These main steps are shown below <span class="citation">(<span class="citeproc-not-found" data-reference-id="ANDREWS2018114"><strong>???</strong></span>)</span>.</p>
<p><img src="../../Images/Andrews2017_Fig1.png" style="margin:auto; display:block" /></p>
<!--knitr::include_graphics(tmpFn, auto_pdf = TRUE)-->
<p>We’ll first explain dimensionality reduction for visualisation, using Principal Component Analysis, t-SNE and UMAP.</p>
<div id="principal-component-analysis" class="section level2">
<h2><span class="header-section-number">9.1</span> Principal Component Analysis</h2>
<p>In a single cell RNA-seq (scRNASeq) data set, each cell is described by the expression level of thoushands of genes.</p>
<p>The total number of genes measured is referred to as dimensionality. Each gene measured is one dimension in the space characterising the data set. Many genes will little vary across cells and thus be uninformative when comparing cells. Also, because some genes will have correlated expression patterns, some information is redundant. Moreover, we can represent data in three dimensions, not more. So reducing the number of useful dimensions is necessary.</p>
<div id="description" class="section level3">
<h3><span class="header-section-number">9.1.1</span> Description</h3>
<p>The data set: a matrix with one row per sample and one variable per column. Here samples are cells and each variable is the normalised read count for a given gene.</p>
<p>The space: each cell is associated to a point in a multi-dimensional space where each gene is a dimension.</p>
<p>The aim: to find a new set of variables defining a space with fewer dimensions while losing as little information as possible.</p>
<p>Out of a set of variables (read counts), PCA defines new variables called Principal Components (PCs) that best capture the variability observed amongst samples (cells), see <span class="citation">(<span class="citeproc-not-found" data-reference-id="field2012discovering"><strong>???</strong></span>)</span> for example.</p>
<p>The number of variables does not change. Only the fraction of variance captured by each variable differs.
The first PC explains the highest proportion of variance possible (bound by prperties of PCA).
The second PC explains the highest proportion of variance not explained by the first PC.
PCs each explain a decreasing amount of variance not explained by the previous ones.
Each PC is a dimension in the new space.</p>
<p>The total amount of variance explained by the first few PCs is usually such that excluding remaining PCs, ie dimensions, loses little information. The stronger the correlation between the initial variables, the stronger the reduction in dimensionality. PCs to keep can be chosen as those capturing at least as much as the average variance per initial variable or using a scree plot, see below.</p>
<p>PCs are linear combinations of the initial variables. PCs represent the same amount of information as the initial set and enable its restoration. The data is not altered. We only look at it in a different way.</p>
<p>About the mapping function from the old to the new space:</p>
<ul>
<li>it is linear</li>
<li>it is inverse, to restore the original space</li>
<li>it relies on orthogonal PCs so that the total variance remains the same.</li>
</ul>
<p>Two transformations of the data are necessary:</p>
<ul>
<li>center the data so that the sample mean for each column is 0 so the covariance matrix of the intial matrix takes a simple form</li>
<li>scale variance to 1, ie standardize, to avoid PCA loading on variables with large variance.</li>
</ul>
</div>
<div id="example-2" class="section level3">
<h3><span class="header-section-number">9.1.2</span> Example</h3>
<p>Here we will make a simple data set of 100 samples and 2 variables, perform PCA and visualise on the initial plane the data set and PCs <span class="citation">(<span class="citeproc-not-found" data-reference-id="pca_blog_Patcher2014"><strong>???</strong></span>)</span>.</p>
<div class="sourceCode" id="cb786"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb786-1"><a href="dimRedForVizTop.html#cb786-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb786-2"><a href="dimRedForVizTop.html#cb786-2"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>),</span>
<span id="cb786-3"><a href="dimRedForVizTop.html#cb786-3"></a>                  <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p>Let’s make and plot a data set.</p>
<div class="sourceCode" id="cb787"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb787-1"><a href="dimRedForVizTop.html#cb787-1"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)            <span class="co">#sets the seed for random number generation.</span></span>
<span id="cb787-2"><a href="dimRedForVizTop.html#cb787-2"></a> x &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">100</span>              <span class="co">#creates a vector x with numbers from 1 to 100</span></span>
<span id="cb787-3"><a href="dimRedForVizTop.html#cb787-3"></a> ex &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dv">100</span>, <span class="dv">0</span>, <span class="dv">30</span>) <span class="co">#100 normally distributed rand. nos. w/ mean=0, s.d.=30</span></span>
<span id="cb787-4"><a href="dimRedForVizTop.html#cb787-4"></a> ey &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dv">100</span>, <span class="dv">0</span>, <span class="dv">30</span>) <span class="co"># &quot; &quot; </span></span>
<span id="cb787-5"><a href="dimRedForVizTop.html#cb787-5"></a> y &lt;-<span class="st"> </span><span class="dv">30</span> <span class="op">+</span><span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>x         <span class="co">#sets y to be a vector that is a linear function of x</span></span>
<span id="cb787-6"><a href="dimRedForVizTop.html#cb787-6"></a> x_obs &lt;-<span class="st"> </span>x <span class="op">+</span><span class="st"> </span>ex         <span class="co">#adds &quot;noise&quot; to x</span></span>
<span id="cb787-7"><a href="dimRedForVizTop.html#cb787-7"></a> y_obs &lt;-<span class="st"> </span>y <span class="op">+</span><span class="st"> </span>ey         <span class="co">#adds &quot;noise&quot; to y</span></span>
<span id="cb787-8"><a href="dimRedForVizTop.html#cb787-8"></a> P &lt;-<span class="st"> </span><span class="kw">cbind</span>(x_obs,y_obs) <span class="co">#places points in matrix</span></span>
<span id="cb787-9"><a href="dimRedForVizTop.html#cb787-9"></a> <span class="kw">plot</span>(P,<span class="dt">asp=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="dv">1</span>) <span class="co">#plot points</span></span>
<span id="cb787-10"><a href="dimRedForVizTop.html#cb787-10"></a> <span class="kw">points</span>(<span class="kw">mean</span>(x_obs),<span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">3</span>, <span class="dt">pch=</span><span class="dv">19</span>) <span class="co">#show center</span></span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_toy_set-1.png" width="672" /></p>
<p>Center the data and compute covariance matrix.</p>
<div class="sourceCode" id="cb788"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb788-1"><a href="dimRedForVizTop.html#cb788-1"></a>M &lt;-<span class="st"> </span><span class="kw">cbind</span>(x_obs <span class="op">-</span><span class="st"> </span><span class="kw">mean</span>(x_obs), y_obs <span class="op">-</span><span class="st"> </span><span class="kw">mean</span>(y_obs)) <span class="co">#centered matrix</span></span>
<span id="cb788-2"><a href="dimRedForVizTop.html#cb788-2"></a>MCov &lt;-<span class="st"> </span><span class="kw">cov</span>(M)          <span class="co">#creates covariance matrix</span></span></code></pre></div>
<p>Compute the principal axes, ie eigenvectors and corresponding eigenvalues.</p>
<p>An eigenvector is a direction and an eigenvalue is a number measuring the spread of the data in that direction. The eigenvector with the highest eigenvalue is the first principal component.</p>
<p>The eigenvectors of the covariance matrix provide the principal axes, and the eigenvalues quantify the fraction of variance explained in each component.</p>
<div class="sourceCode" id="cb789"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb789-1"><a href="dimRedForVizTop.html#cb789-1"></a>eigenValues &lt;-<span class="st"> </span><span class="kw">eigen</span>(MCov)<span class="op">$</span>values       <span class="co">#compute eigenvalues</span></span>
<span id="cb789-2"><a href="dimRedForVizTop.html#cb789-2"></a>eigenVectors &lt;-<span class="st"> </span><span class="kw">eigen</span>(MCov)<span class="op">$</span>vectors     <span class="co">#compute eigenvectors</span></span>
<span id="cb789-3"><a href="dimRedForVizTop.html#cb789-3"></a></span>
<span id="cb789-4"><a href="dimRedForVizTop.html#cb789-4"></a><span class="co"># or use &#39;singular value decomposition&#39; of the matrix</span></span>
<span id="cb789-5"><a href="dimRedForVizTop.html#cb789-5"></a>d &lt;-<span class="st"> </span><span class="kw">svd</span>(M)<span class="op">$</span>d          <span class="co">#the singular values</span></span>
<span id="cb789-6"><a href="dimRedForVizTop.html#cb789-6"></a>v &lt;-<span class="st"> </span><span class="kw">svd</span>(M)<span class="op">$</span>v          <span class="co">#the right singular vectors</span></span></code></pre></div>
<p>Let’s plot the principal axes.</p>
<p>First PC:</p>
<div class="sourceCode" id="cb790"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb790-1"><a href="dimRedForVizTop.html#cb790-1"></a><span class="co"># PC 1:</span></span>
<span id="cb790-2"><a href="dimRedForVizTop.html#cb790-2"></a> <span class="kw">plot</span>(P,<span class="dt">asp=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="dv">1</span>) <span class="co">#plot points</span></span>
<span id="cb790-3"><a href="dimRedForVizTop.html#cb790-3"></a> <span class="kw">points</span>(<span class="kw">mean</span>(x_obs),<span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">3</span>, <span class="dt">pch=</span><span class="dv">19</span>) <span class="co">#show center</span></span>
<span id="cb790-4"><a href="dimRedForVizTop.html#cb790-4"></a><span class="kw">lines</span>(x_obs,eigenVectors[<span class="dv">2</span>,<span class="dv">1</span>]<span class="op">/</span>eigenVectors[<span class="dv">1</span>,<span class="dv">1</span>]<span class="op">*</span>M[x]<span class="op">+</span><span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">8</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_show_PC1-1.png" width="672" /></p>
<p>Second PC:</p>
<div class="sourceCode" id="cb791"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb791-1"><a href="dimRedForVizTop.html#cb791-1"></a> <span class="kw">plot</span>(P,<span class="dt">asp=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="dv">1</span>) <span class="co">#plot points</span></span>
<span id="cb791-2"><a href="dimRedForVizTop.html#cb791-2"></a> <span class="kw">points</span>(<span class="kw">mean</span>(x_obs),<span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">3</span>, <span class="dt">pch=</span><span class="dv">19</span>) <span class="co">#show center</span></span>
<span id="cb791-3"><a href="dimRedForVizTop.html#cb791-3"></a><span class="co"># PC 1:</span></span>
<span id="cb791-4"><a href="dimRedForVizTop.html#cb791-4"></a><span class="kw">lines</span>(x_obs,eigenVectors[<span class="dv">2</span>,<span class="dv">1</span>]<span class="op">/</span>eigenVectors[<span class="dv">1</span>,<span class="dv">1</span>]<span class="op">*</span>M[x]<span class="op">+</span><span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">8</span>)</span>
<span id="cb791-5"><a href="dimRedForVizTop.html#cb791-5"></a><span class="co"># PC 2:</span></span>
<span id="cb791-6"><a href="dimRedForVizTop.html#cb791-6"></a><span class="kw">lines</span>(x_obs,eigenVectors[<span class="dv">2</span>,<span class="dv">2</span>]<span class="op">/</span>eigenVectors[<span class="dv">1</span>,<span class="dv">2</span>]<span class="op">*</span>M[x]<span class="op">+</span><span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">8</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_show_PC2-1.png" width="672" /></p>
<p>Add the projections of the points onto the first PC:</p>
<div class="sourceCode" id="cb792"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb792-1"><a href="dimRedForVizTop.html#cb792-1"></a><span class="kw">plot</span>(P,<span class="dt">asp=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="dv">1</span>) <span class="co">#plot points</span></span>
<span id="cb792-2"><a href="dimRedForVizTop.html#cb792-2"></a><span class="kw">points</span>(<span class="kw">mean</span>(x_obs),<span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">3</span>, <span class="dt">pch=</span><span class="dv">19</span>) <span class="co">#show center</span></span>
<span id="cb792-3"><a href="dimRedForVizTop.html#cb792-3"></a><span class="co"># PC 1:</span></span>
<span id="cb792-4"><a href="dimRedForVizTop.html#cb792-4"></a><span class="kw">lines</span>(x_obs,eigenVectors[<span class="dv">2</span>,<span class="dv">1</span>]<span class="op">/</span>eigenVectors[<span class="dv">1</span>,<span class="dv">1</span>]<span class="op">*</span>M[x]<span class="op">+</span><span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">8</span>)</span>
<span id="cb792-5"><a href="dimRedForVizTop.html#cb792-5"></a><span class="co"># PC 2:</span></span>
<span id="cb792-6"><a href="dimRedForVizTop.html#cb792-6"></a><span class="kw">lines</span>(x_obs,eigenVectors[<span class="dv">2</span>,<span class="dv">2</span>]<span class="op">/</span>eigenVectors[<span class="dv">1</span>,<span class="dv">2</span>]<span class="op">*</span>M[x]<span class="op">+</span><span class="kw">mean</span>(y_obs),<span class="dt">col=</span><span class="dv">8</span>)</span>
<span id="cb792-7"><a href="dimRedForVizTop.html#cb792-7"></a><span class="co"># add projecions:</span></span>
<span id="cb792-8"><a href="dimRedForVizTop.html#cb792-8"></a>trans &lt;-<span class="st"> </span>(M<span class="op">%*%</span>v[,<span class="dv">1</span>])<span class="op">%*%</span>v[,<span class="dv">1</span>] <span class="co">#compute projections of points</span></span>
<span id="cb792-9"><a href="dimRedForVizTop.html#cb792-9"></a>P_proj &lt;-<span class="st"> </span><span class="kw">scale</span>(trans, <span class="dt">center=</span><span class="op">-</span><span class="kw">cbind</span>(<span class="kw">mean</span>(x_obs),<span class="kw">mean</span>(y_obs)), <span class="dt">scale=</span><span class="ot">FALSE</span>) </span>
<span id="cb792-10"><a href="dimRedForVizTop.html#cb792-10"></a><span class="kw">points</span>(P_proj, <span class="dt">col=</span><span class="dv">4</span>,<span class="dt">pch=</span><span class="dv">19</span>,<span class="dt">cex=</span><span class="fl">0.5</span>) <span class="co">#plot projections</span></span>
<span id="cb792-11"><a href="dimRedForVizTop.html#cb792-11"></a><span class="kw">segments</span>(x_obs,y_obs,P_proj[,<span class="dv">1</span>],P_proj[,<span class="dv">2</span>],<span class="dt">col=</span><span class="dv">4</span>,<span class="dt">lty=</span><span class="dv">2</span>) <span class="co">#connect to points</span></span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_add_projection_onto_PC1-1.png" width="672" /></p>
<p>Compute PCs with prcomp().</p>
<div class="sourceCode" id="cb793"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb793-1"><a href="dimRedForVizTop.html#cb793-1"></a>pca_res &lt;-<span class="st"> </span><span class="kw">prcomp</span>(M)</span></code></pre></div>
<div class="sourceCode" id="cb794"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb794-1"><a href="dimRedForVizTop.html#cb794-1"></a><span class="kw">summary</span>(pca_res)</span></code></pre></div>
<pre><code>## Importance of components:
##                           PC1    PC2
## Standard deviation     73.827 28.279
## Proportion of Variance  0.872  0.128
## Cumulative Proportion   0.872  1.000</code></pre>
<div class="sourceCode" id="cb796"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb796-1"><a href="dimRedForVizTop.html#cb796-1"></a>var_explained &lt;-<span class="st"> </span>pca_res<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span><span class="op">/</span><span class="kw">sum</span>(pca_res<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb796-2"><a href="dimRedForVizTop.html#cb796-2"></a>var_explained</span></code></pre></div>
<pre><code>## [1] 0.8720537 0.1279463</code></pre>
<p>Check amount of variance captured by PCs on a scree plot.</p>
<div class="sourceCode" id="cb798"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb798-1"><a href="dimRedForVizTop.html#cb798-1"></a><span class="co"># Show scree plot:</span></span>
<span id="cb798-2"><a href="dimRedForVizTop.html#cb798-2"></a><span class="kw">plot</span>(pca_res)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_scree-1.png" width="672" /></p>
<p>Plot with ggplot.</p>
<div class="sourceCode" id="cb799"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb799-1"><a href="dimRedForVizTop.html#cb799-1"></a>df_pc &lt;-<span class="st"> </span><span class="kw">data.frame</span>(pca_res<span class="op">$</span>x)</span>
<span id="cb799-2"><a href="dimRedForVizTop.html#cb799-2"></a>g &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df_pc, <span class="kw">aes</span>(PC1, PC2)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb799-3"><a href="dimRedForVizTop.html#cb799-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span><span class="st">   </span><span class="co"># draw points</span></span>
<span id="cb799-4"><a href="dimRedForVizTop.html#cb799-4"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;PCA&quot;</span>, </span>
<span id="cb799-5"><a href="dimRedForVizTop.html#cb799-5"></a>       <span class="dt">subtitle=</span><span class="st">&quot;With principal components PC1 and PC2 as X and Y axis&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb799-6"><a href="dimRedForVizTop.html#cb799-6"></a><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="fl">1.2</span> <span class="op">*</span><span class="st"> </span><span class="kw">c</span>(<span class="kw">min</span>(df_pc<span class="op">$</span>PC1), <span class="kw">max</span>(df_pc<span class="op">$</span>PC1)), </span>
<span id="cb799-7"><a href="dimRedForVizTop.html#cb799-7"></a>                  <span class="dt">ylim =</span> <span class="fl">1.2</span> <span class="op">*</span><span class="st"> </span><span class="kw">c</span>(<span class="kw">min</span>(df_pc<span class="op">$</span>PC2), <span class="kw">max</span>(df_pc<span class="op">$</span>PC2)))</span>
<span id="cb799-8"><a href="dimRedForVizTop.html#cb799-8"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">0</span>)</span>
<span id="cb799-9"><a href="dimRedForVizTop.html#cb799-9"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">xintercept=</span><span class="dv">0</span>)</span>
<span id="cb799-10"><a href="dimRedForVizTop.html#cb799-10"></a>g</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_show_PC_plane_with_ggplot-1.png" width="672" /></p>
<p>Or use ggfortify autoplot().</p>
<div class="sourceCode" id="cb800"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb800-1"><a href="dimRedForVizTop.html#cb800-1"></a><span class="co"># ggfortify</span></span>
<span id="cb800-2"><a href="dimRedForVizTop.html#cb800-2"></a><span class="kw">library</span>(ggfortify)</span>
<span id="cb800-3"><a href="dimRedForVizTop.html#cb800-3"></a>g &lt;-<span class="st"> </span><span class="kw">autoplot</span>(pca_res)</span>
<span id="cb800-4"><a href="dimRedForVizTop.html#cb800-4"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">0</span>)</span>
<span id="cb800-5"><a href="dimRedForVizTop.html#cb800-5"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">xintercept=</span><span class="dv">0</span>)</span>
<span id="cb800-6"><a href="dimRedForVizTop.html#cb800-6"></a>g</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/pca_show_PC_plane_with_ggfortify-1.png" width="672" /></p>
<p>Going from 2D to 3D (figure from <span class="citation">(<span class="citeproc-not-found" data-reference-id="nlpcaPlot"><strong>???</strong></span>)</span>):</p>
<p><img src="../../Images/hemberg_pca.png" style="margin:auto; display:block" /></p>
</div>
</div>
<div id="load-packages-2" class="section level2">
<h2><span class="header-section-number">9.2</span> Load packages</h2>
<div class="sourceCode" id="cb801"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb801-1"><a href="dimRedForVizTop.html#cb801-1"></a><span class="kw">library</span>(scater) <span class="co"># for QC and plots</span></span></code></pre></div>
</div>
<div id="load-data" class="section level2">
<h2><span class="header-section-number">9.3</span> Load data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts for 500 cells per sample.</p>
<div class="sourceCode" id="cb802"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb802-1"><a href="dimRedForVizTop.html#cb802-1"></a>setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb802-2"><a href="dimRedForVizTop.html#cb802-2"></a>setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb802-3"><a href="dimRedForVizTop.html#cb802-3"></a></span>
<span id="cb802-4"><a href="dimRedForVizTop.html#cb802-4"></a><span class="co"># Read object in:</span></span>
<span id="cb802-5"><a href="dimRedForVizTop.html#cb802-5"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb802-6"><a href="dimRedForVizTop.html#cb802-6"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl.Rds&quot;</code></pre>
<div class="sourceCode" id="cb804"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb804-1"><a href="dimRedForVizTop.html#cb804-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb804-2"><a href="dimRedForVizTop.html#cb804-2"></a>{</span>
<span id="cb804-3"><a href="dimRedForVizTop.html#cb804-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb804-4"><a href="dimRedForVizTop.html#cb804-4"></a>}</span>
<span id="cb804-5"><a href="dimRedForVizTop.html#cb804-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb804-6"><a href="dimRedForVizTop.html#cb804-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 16629 5500 
## metadata(0):
## assays(2): counts logcounts
## rownames(16629): ENSG00000237491 ENSG00000225880 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(16): Barcode Run ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb806"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb806-1"><a href="dimRedForVizTop.html#cb806-1"></a><span class="kw">head</span>(<span class="kw">rowData</span>(sce))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 11 columns
##                 ensembl_gene_id external_gene_name chromosome_name
##                     &lt;character&gt;        &lt;character&gt;     &lt;character&gt;
## ENSG00000237491 ENSG00000237491          LINC01409               1
## ENSG00000225880 ENSG00000225880          LINC00115               1
## ENSG00000230368 ENSG00000230368             FAM41C               1
## ENSG00000230699 ENSG00000230699         AL645608.2               1
## ENSG00000188976 ENSG00000188976              NOC2L               1
## ENSG00000187961 ENSG00000187961             KLHL17               1
##                 start_position end_position    strand      Symbol
##                      &lt;integer&gt;    &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## ENSG00000237491         778747       810065         1  AL669831.5
## ENSG00000225880         826206       827522        -1   LINC00115
## ENSG00000230368         868071       876903        -1      FAM41C
## ENSG00000230699         911435       914948         1  AL645608.3
## ENSG00000188976         944203       959309        -1       NOC2L
## ENSG00000187961         960584       965719         1      KLHL17
##                            Type       mean  detected gene_sparsity
##                     &lt;character&gt;  &lt;numeric&gt; &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000237491 Gene Expression 0.02785355  2.706672      0.977951
## ENSG00000225880 Gene Expression 0.01376941  1.340222      0.985699
## ENSG00000230368 Gene Expression 0.02027381  1.946076      0.980821
## ENSG00000230699 Gene Expression 0.00144251  0.144251      0.997704
## ENSG00000188976 Gene Expression 0.17711393 14.511645      0.835565
## ENSG00000187961 Gene Expression 0.00354070  0.348825      0.995935</code></pre>
<div class="sourceCode" id="cb808"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb808-1"><a href="dimRedForVizTop.html#cb808-1"></a><span class="co">#any(duplicated(rowData(nz.sce)$ensembl_gene_id))</span></span>
<span id="cb808-2"><a href="dimRedForVizTop.html#cb808-2"></a><span class="co"># some function(s) used below complain about &#39;strand&#39; already being used in row data,</span></span>
<span id="cb808-3"><a href="dimRedForVizTop.html#cb808-3"></a><span class="co"># so rename that column now:</span></span>
<span id="cb808-4"><a href="dimRedForVizTop.html#cb808-4"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span></span></code></pre></div>
</div>
<div id="pca" class="section level2">
<h2><span class="header-section-number">9.4</span> PCA</h2>
<p>Perform PCA, keep outcome in same object.</p>
<div class="sourceCode" id="cb809"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb809-1"><a href="dimRedForVizTop.html#cb809-1"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb809-2"><a href="dimRedForVizTop.html#cb809-2"></a><span class="co"># compute PCA:</span></span>
<span id="cb809-3"><a href="dimRedForVizTop.html#cb809-3"></a><span class="co">#sce &lt;- runPCA(sce, ncomponents = nbPcToComp, method = &quot;irlba&quot;)</span></span>
<span id="cb809-4"><a href="dimRedForVizTop.html#cb809-4"></a>sce &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce, <span class="dt">ncomponents =</span> nbPcToComp)</span></code></pre></div>
<p>Display scree plot.</p>
<div class="sourceCode" id="cb810"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb810-1"><a href="dimRedForVizTop.html#cb810-1"></a><span class="co"># with reducedDim</span></span>
<span id="cb810-2"><a href="dimRedForVizTop.html#cb810-2"></a>sce.pca &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb810-3"><a href="dimRedForVizTop.html#cb810-3"></a><span class="kw">attributes</span>(sce.pca)<span class="op">$</span>percentVar</span></code></pre></div>
<pre><code>##  [1] 16.1072558  9.6482486  4.4673293  3.7808853  1.9639696  1.8008626
##  [7]  1.6739293  1.2887282  1.0690783  0.8596321  0.8018422  0.6846937
## [13]  0.6139309  0.5946088  0.5411263  0.4377304  0.3900507  0.3815624
## [19]  0.3636303  0.3514861  0.3408887  0.2973306  0.2843550  0.2754252
## [25]  0.2666001  0.2609457  0.2555751  0.2496498  0.2481626  0.2371280
## [31]  0.2320116  0.2304992  0.2270922  0.2257485  0.2242832  0.2198916
## [37]  0.2189744  0.2173953  0.2151230  0.2140901  0.2129180  0.2120306
## [43]  0.2094498  0.2083422  0.2063008  0.2059701  0.2047050  0.2033185
## [49]  0.2025080  0.2017399</code></pre>
<div class="sourceCode" id="cb812"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb812-1"><a href="dimRedForVizTop.html#cb812-1"></a><span class="kw">barplot</span>(<span class="kw">attributes</span>(sce.pca)<span class="op">$</span>percentVar,</span>
<span id="cb812-2"><a href="dimRedForVizTop.html#cb812-2"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;Scree plot for the %s first PCs&quot;</span>, nbPcToComp),</span>
<span id="cb812-3"><a href="dimRedForVizTop.html#cb812-3"></a>        <span class="dt">names.arg=</span><span class="dv">1</span><span class="op">:</span>nbPcToComp,</span>
<span id="cb812-4"><a href="dimRedForVizTop.html#cb812-4"></a>        <span class="dt">cex.names =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_scree_plot-1.png" width="672" /></p>
<p>Display cells on a plot for the first 2 PCs, colouring by ‘Sample’ and setting size to match ‘total_features’.</p>
<p>The proximity of cells reflects the similarity of their expression profiles.</p>
<div class="sourceCode" id="cb813"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb813-1"><a href="dimRedForVizTop.html#cb813-1"></a>g &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(sce,</span>
<span id="cb813-2"><a href="dimRedForVizTop.html#cb813-2"></a>        <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb813-3"><a href="dimRedForVizTop.html#cb813-3"></a>        <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span></span>
<span id="cb813-4"><a href="dimRedForVizTop.html#cb813-4"></a>)         </span>
<span id="cb813-5"><a href="dimRedForVizTop.html#cb813-5"></a>g</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotColorBySample-1.png" width="672" /></p>
<p>One can also split the plot by sample.</p>
<div class="sourceCode" id="cb814"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb814-1"><a href="dimRedForVizTop.html#cb814-1"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st">  </span><span class="kw">facet_wrap</span>(sce<span class="op">$</span>source_name <span class="op">~</span><span class="st"> </span>.)</span>
<span id="cb814-2"><a href="dimRedForVizTop.html#cb814-2"></a>g</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotColorBySample_facetBySample-1.png" width="1152" /></p>
<p>Or plot several PCs at once, using plotReducedDim():</p>
<div class="sourceCode" id="cb815"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb815-1"><a href="dimRedForVizTop.html#cb815-1"></a><span class="kw">plotReducedDim</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">ncomponents=</span><span class="dv">3</span>, </span>
<span id="cb815-2"><a href="dimRedForVizTop.html#cb815-2"></a>        <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotReducedDim-1.png" width="672" /></p>
<div id="correlation-between-pcs-and-the-total-number-of-features-detected" class="section level3">
<h3><span class="header-section-number">9.4.1</span> Correlation between PCs and the total number of features detected</h3>
<p>The PCA plot above shows cells as symbols whose size depends on the total number of features or library size. It suggests there may be a correlation between PCs and these variables. Let’s check:</p>
<div class="sourceCode" id="cb816"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb816-1"><a href="dimRedForVizTop.html#cb816-1"></a><span class="kw">colData</span>(sce)<span class="op">$</span>source_name &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>source_name)</span>
<span id="cb816-2"><a href="dimRedForVizTop.html#cb816-2"></a><span class="kw">colData</span>(sce)<span class="op">$</span>block &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>block)</span>
<span id="cb816-3"><a href="dimRedForVizTop.html#cb816-3"></a>r2mat &lt;-<span class="st"> </span><span class="kw">getExplanatoryPCs</span>(sce)</span>
<span id="cb816-4"><a href="dimRedForVizTop.html#cb816-4"></a><span class="co">#r2mat &lt;- getExplanatoryPCs(sce,</span></span>
<span id="cb816-5"><a href="dimRedForVizTop.html#cb816-5"></a><span class="co">#                           variables = c(&quot;Run&quot;, &quot;source_name&quot;))</span></span>
<span id="cb816-6"><a href="dimRedForVizTop.html#cb816-6"></a><span class="co">#                           #variables = c(&quot;Run&quot;, &quot;Sample.Name&quot;))</span></span>
<span id="cb816-7"><a href="dimRedForVizTop.html#cb816-7"></a><span class="co">#                           #variables = c(&quot;Run&quot;, &quot;Sample.Name&quot;, &quot;source_name&quot;))</span></span>
<span id="cb816-8"><a href="dimRedForVizTop.html#cb816-8"></a></span>
<span id="cb816-9"><a href="dimRedForVizTop.html#cb816-9"></a>r2mat</span></code></pre></div>
<pre><code>##      Barcode      Run Sample.Name source_name         sum    detected
## PC1      NaN 30.85659    30.85659    6.487377  0.43493165  7.08894442
## PC2      NaN 39.71165    39.71165   25.878094  4.25167387  0.58954116
## PC3      NaN 30.70223    30.70223   18.018721 38.24631708 43.80921901
## PC4      NaN 59.99750    59.99750   46.026981  6.29085450  7.52064941
## PC5      NaN 28.72873    28.72873   11.700557  4.57160974  6.47357417
## PC6      NaN 43.68081    43.68081   15.832822  0.39643282  0.04005349
## PC7      NaN 39.80806    39.80806   25.736034  0.34776526  1.87429989
## PC8      NaN 61.48502    61.48502   53.382059  1.26696650  0.69851353
## PC9      NaN 20.48116    20.48116    7.806216  2.68842935  2.55756540
## PC10     NaN 41.24854    41.24854    5.412729  0.01273348  0.43130416
##      subsets_Mito_sum subsets_Mito_detected subsets_Mito_percent       total
## PC1        6.51661947            18.1351642            4.1926231  0.43493165
## PC2        0.10383151             1.9809524           15.5359489  4.25167387
## PC3       25.12527402            10.3594392            2.3470910 38.24631708
## PC4        1.55024880             1.7264436            0.2189570  6.29085450
## PC5        8.47830110            15.8647633            7.0504884  4.57160974
## PC6       11.71821736            14.7324487           40.4069838  0.39643282
## PC7        0.05713278             2.0491925            0.8726234  0.34776526
## PC8        0.05725464             1.8114625            5.5016017  1.26696650
## PC9        4.79687302             4.6900075            3.4547445  2.68842935
## PC10       0.06794968             0.2728095            1.0255645  0.01273348
##          block setName    discard     outlier cell_sparsity  sizeFactor
## PC1   6.487377      NA  0.1175128  0.34195161    7.08948526  6.03083849
## PC2  25.878094      NA  7.2209671  7.03388703    0.59077199  0.21122381
## PC3  18.018721      NA  6.4199167  9.61961429   43.83600840 35.30043641
## PC4  46.026981      NA  2.7802380  4.06365034    7.50975544  8.72178733
## PC5  11.700557      NA  0.6676141  0.25601414    6.47138586  4.57011170
## PC6  15.832822      NA 18.9268506 15.79017131    0.03990797  0.02550611
## PC7  25.736034      NA  1.2853430  1.97448881    1.87469906  0.71227698
## PC8  53.382059      NA  0.8302930  0.69206768    0.69721085  0.76186993
## PC9   7.806216      NA  0.9206323  0.70318630    2.55906696  3.06770581
## PC10  5.412729      NA  0.2997168  0.08553385    0.43088612  0.30545007</code></pre>
<div class="sourceCode" id="cb818"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb818-1"><a href="dimRedForVizTop.html#cb818-1"></a>dat &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">colData</span>(sce)[,<span class="kw">c</span>(<span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb818-2"><a href="dimRedForVizTop.html#cb818-2"></a>                 <span class="st">&quot;source_name&quot;</span>,</span>
<span id="cb818-3"><a href="dimRedForVizTop.html#cb818-3"></a>                 <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb818-4"><a href="dimRedForVizTop.html#cb818-4"></a>                 <span class="st">&quot;detected&quot;</span>,</span>
<span id="cb818-5"><a href="dimRedForVizTop.html#cb818-5"></a>                 <span class="co">#&quot;percent_top_200&quot;,</span></span>
<span id="cb818-6"><a href="dimRedForVizTop.html#cb818-6"></a>                 <span class="st">&quot;subsets_Mito_percent&quot;</span>)],</span>
<span id="cb818-7"><a href="dimRedForVizTop.html#cb818-7"></a>                 <span class="kw">reducedDim</span>(sce,<span class="st">&quot;PCA&quot;</span>))</span>
<span id="cb818-8"><a href="dimRedForVizTop.html#cb818-8"></a>dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(dat)</span>
<span id="cb818-9"><a href="dimRedForVizTop.html#cb818-9"></a>dat<span class="op">$</span>sum &lt;-<span class="st"> </span><span class="kw">log2</span>(dat<span class="op">$</span>sum)</span>
<span id="cb818-10"><a href="dimRedForVizTop.html#cb818-10"></a><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>sum, <span class="dt">y=</span>PC1, <span class="dt">shape=</span>source_name, <span class="dt">col=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb818-11"><a href="dimRedForVizTop.html#cb818-11"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb818-12"><a href="dimRedForVizTop.html#cb818-12"></a><span class="st">    </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span>lm, <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">x=</span>sum, <span class="dt">y=</span>PC1)) </span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotQC_total_features-1.png" width="672" /></p>
<div class="sourceCode" id="cb819"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb819-1"><a href="dimRedForVizTop.html#cb819-1"></a><span class="co">#ggplot(dat, aes(x=percent_top_200, y=PC2, shape=source_name, col=Sample.Name)) +</span></span>
<span id="cb819-2"><a href="dimRedForVizTop.html#cb819-2"></a><span class="co">#    geom_point() +</span></span>
<span id="cb819-3"><a href="dimRedForVizTop.html#cb819-3"></a><span class="co">#    geom_smooth(method=lm, inherit.aes = FALSE, aes(x=percent_top_200, y=PC2)) </span></span>
<span id="cb819-4"><a href="dimRedForVizTop.html#cb819-4"></a><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>detected, <span class="dt">y=</span>PC3, <span class="dt">shape=</span>source_name, <span class="dt">col=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb819-5"><a href="dimRedForVizTop.html#cb819-5"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb819-6"><a href="dimRedForVizTop.html#cb819-6"></a><span class="st">    </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span>lm, <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">x=</span>detected, <span class="dt">y=</span>PC3)) </span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotQC_total_features-2.png" width="672" /></p>
<div class="sourceCode" id="cb820"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb820-1"><a href="dimRedForVizTop.html#cb820-1"></a><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>subsets_Mito_percent, <span class="dt">y=</span>PC2, <span class="dt">shape=</span>source_name, <span class="dt">col=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb820-2"><a href="dimRedForVizTop.html#cb820-2"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb820-3"><a href="dimRedForVizTop.html#cb820-3"></a><span class="st">    </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span>lm, <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">x=</span>subsets_Mito_percent, <span class="dt">y=</span>PC2)) </span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotQC_total_features-3.png" width="672" /></p>
<div class="sourceCode" id="cb821"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb821-1"><a href="dimRedForVizTop.html#cb821-1"></a><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x=</span>source_name, <span class="dt">y=</span>PC7, <span class="dt">shape=</span>source_name, <span class="dt">col=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb821-2"><a href="dimRedForVizTop.html#cb821-2"></a><span class="st">    </span><span class="kw">geom_boxplot</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/sce_pca_plotQC_total_features-4.png" width="672" /></p>
</div>
</div>
<div id="t-sne-t-distributed-stochastic-neighbor-embedding" class="section level2">
<h2><span class="header-section-number">9.5</span> t-SNE: t-Distributed Stochastic Neighbor Embedding</h2>
<!-- https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/latent-spaces.html -->
<p>The Stochastic Neighbor Embedding (SNE) approach address two shortcomings of PCA that captures the global covariance structure with a linear combination of initial variables: by preserving the local structure allowing for non-linear projections. It uses two distributions of the pairwise similarities between data points: in the input data set and in the low-dimensional space.</p>
<p>SNE aims at preserving neighbourhoods. For each points, it computes probabilities of chosing each other point as its neighbour based on a Normal distribution depending on 1) the distance matrix and 2) the size of the neighbourhood (perplexity). SNE aims at finding a low-dimension space (eg 2D-plane) such that the similarity matrix deriving from it is as similar as possible as that from the high-dimension space. To address the fact that in low dimension, points are brought together, the similarity matrix in the low-dimension is allowed to follow a t-distribution.</p>
<p>Two characteristics matter:</p>
<ul>
<li>perplexity, to indicate the relative importance of the local and global patterns in structure of the data set, usually use a value of 50,</li>
<li>stochasticity; running the analysis will produce a different map every time, unless the seed is set.</li>
</ul>
<p>See <a href="https://distill.pub/2016/misread-tsne/">misread-tsne</a>.</p>
<div id="perplexity" class="section level3">
<h3><span class="header-section-number">9.5.1</span> Perplexity</h3>
<p>Compute t-SNE with default perplexity, ie 50.</p>
<div class="sourceCode" id="cb822"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb822-1"><a href="dimRedForVizTop.html#cb822-1"></a><span class="co"># runTSNE default perpexity if min(50, floor(ncol(object)/5))</span></span>
<span id="cb822-2"><a href="dimRedForVizTop.html#cb822-2"></a>sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">perplexity=</span><span class="dv">50</span>, <span class="dt">rand_seed=</span><span class="dv">123</span>)</span></code></pre></div>
<p>Plot t-SNE:</p>
<div class="sourceCode" id="cb823"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb823-1"><a href="dimRedForVizTop.html#cb823-1"></a>tsne50 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce,</span>
<span id="cb823-2"><a href="dimRedForVizTop.html#cb823-2"></a>           <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb823-3"><a href="dimRedForVizTop.html#cb823-3"></a>           <span class="dt">size_by=</span><span class="st">&quot;sum&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb823-4"><a href="dimRedForVizTop.html#cb823-4"></a><span class="st">           </span>fontsize <span class="op">+</span><span class="st"> </span></span>
<span id="cb823-5"><a href="dimRedForVizTop.html#cb823-5"></a><span class="st">           </span><span class="kw">ggtitle</span>(<span class="st">&quot;Perplexity = 50&quot;</span>)</span>
<span id="cb823-6"><a href="dimRedForVizTop.html#cb823-6"></a>tsne50</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotTSNE_perp50-1.png" width="672" /></p>
<!-- Split by sample type: -->
<p>Compute t-SNE for several perplexity values:</p>
<div class="sourceCode" id="cb824"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb824-1"><a href="dimRedForVizTop.html#cb824-1"></a>tsne5.run &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">use_dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">perplexity=</span><span class="dv">5</span>, <span class="dt">rand_seed=</span><span class="dv">123</span>)</span>
<span id="cb824-2"><a href="dimRedForVizTop.html#cb824-2"></a>tsne5 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(tsne5.run, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Perplexity = 5&quot;</span>)</span>
<span id="cb824-3"><a href="dimRedForVizTop.html#cb824-3"></a></span>
<span id="cb824-4"><a href="dimRedForVizTop.html#cb824-4"></a><span class="co">#tsne200.run &lt;- runTSNE(sce, use_dimred=&quot;PCA&quot;, perplexity=200, rand_seed=123)</span></span>
<span id="cb824-5"><a href="dimRedForVizTop.html#cb824-5"></a><span class="co">#tsne200 &lt;- plotTSNE(tsne200.run, colour_by=&quot;Sample.Name&quot;) + fontsize + ggtitle(&quot;Perplexity = 200&quot;)</span></span>
<span id="cb824-6"><a href="dimRedForVizTop.html#cb824-6"></a></span>
<span id="cb824-7"><a href="dimRedForVizTop.html#cb824-7"></a>tsne500.run &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">use_dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">perplexity=</span><span class="dv">500</span>, <span class="dt">rand_seed=</span><span class="dv">123</span>)</span>
<span id="cb824-8"><a href="dimRedForVizTop.html#cb824-8"></a>tsne500 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(tsne500.run, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Perplexity = 500&quot;</span>)</span>
<span id="cb824-9"><a href="dimRedForVizTop.html#cb824-9"></a></span>
<span id="cb824-10"><a href="dimRedForVizTop.html#cb824-10"></a><span class="co">#tsne1000.run &lt;- runTSNE(sce, use_dimred=&quot;PCA&quot;, perplexity=1000, rand_seed=123)</span></span>
<span id="cb824-11"><a href="dimRedForVizTop.html#cb824-11"></a><span class="co">#tsne1000 &lt;- plotTSNE(tsne1000.run, colour_by=&quot;Sample.Name&quot;) + fontsize + ggtitle(&quot;Perplexity = 1000&quot;)</span></span></code></pre></div>
<div class="sourceCode" id="cb825"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb825-1"><a href="dimRedForVizTop.html#cb825-1"></a>tsne5</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotTSNE_perpRange-1.png" width="576" /></p>
<div class="sourceCode" id="cb826"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb826-1"><a href="dimRedForVizTop.html#cb826-1"></a><span class="co">#tsne50</span></span>
<span id="cb826-2"><a href="dimRedForVizTop.html#cb826-2"></a><span class="co">#tsne200</span></span>
<span id="cb826-3"><a href="dimRedForVizTop.html#cb826-3"></a>tsne500</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotTSNE_perpRange-2.png" width="576" /></p>
</div>
<div id="stochasticity" class="section level3">
<h3><span class="header-section-number">9.5.2</span> Stochasticity</h3>
<p>Use a different seed with the same perplexity 50.</p>
<div class="sourceCode" id="cb827"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb827-1"><a href="dimRedForVizTop.html#cb827-1"></a>tsne50.b &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">use_dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">perplexity=</span><span class="dv">50</span>, <span class="dt">rand_seed=</span><span class="dv">456</span>)</span>
<span id="cb827-2"><a href="dimRedForVizTop.html#cb827-2"></a></span>
<span id="cb827-3"><a href="dimRedForVizTop.html#cb827-3"></a>tsne50.b &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(tsne50.b,</span>
<span id="cb827-4"><a href="dimRedForVizTop.html#cb827-4"></a>           <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb827-5"><a href="dimRedForVizTop.html#cb827-5"></a>           <span class="dt">size_by=</span><span class="st">&quot;sum&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb827-6"><a href="dimRedForVizTop.html#cb827-6"></a><span class="st">         </span>fontsize <span class="op">+</span><span class="st"> </span></span>
<span id="cb827-7"><a href="dimRedForVizTop.html#cb827-7"></a><span class="st">         </span><span class="kw">ggtitle</span>(<span class="st">&quot;Perplexity = 50, seed 456&quot;</span>)</span>
<span id="cb827-8"><a href="dimRedForVizTop.html#cb827-8"></a>tsne50.b</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotTSNE_stocha-1.png" width="672" /></p>
</div>
</div>
<div id="umap" class="section level2">
<h2><span class="header-section-number">9.6</span> UMAP</h2>
<p>Another neighbour graph method. Similar to t-SNE, but that is determistic, faster and claims to preserve both local and global structures.</p>
<p>Compute UMAP.</p>
<div class="sourceCode" id="cb828"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb828-1"><a href="dimRedForVizTop.html#cb828-1"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb828-2"><a href="dimRedForVizTop.html#cb828-2"></a>sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<p>Plot UMAP:</p>
<div class="sourceCode" id="cb829"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb829-1"><a href="dimRedForVizTop.html#cb829-1"></a>sce.umap &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(sce,</span>
<span id="cb829-2"><a href="dimRedForVizTop.html#cb829-2"></a>           <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb829-3"><a href="dimRedForVizTop.html#cb829-3"></a>           <span class="dt">size_by=</span><span class="st">&quot;sum&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb829-4"><a href="dimRedForVizTop.html#cb829-4"></a><span class="st">           </span>fontsize <span class="op">+</span><span class="st"> </span></span>
<span id="cb829-5"><a href="dimRedForVizTop.html#cb829-5"></a><span class="st">           </span><span class="kw">ggtitle</span>(<span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb829-6"><a href="dimRedForVizTop.html#cb829-6"></a>sce.umap</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plotUMAP-1.png" width="672" /></p>
<!-- Split by sample: -->
<p>Save SCE object:</p>
<div class="sourceCode" id="cb830"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb830-1"><a href="dimRedForVizTop.html#cb830-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dimRed.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb830-2"><a href="dimRedForVizTop.html#cb830-2"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span></code></pre></div>
</div>
<div id="session-information-5" class="section level2">
<h2><span class="header-section-number">9.7</span> Session information</h2>
<div class="sourceCode" id="cb831"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb831-1"><a href="dimRedForVizTop.html#cb831-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] Cairo_1.5-12.2              edgeR_3.32.1               
##  [3] DelayedArray_0.16.3         RColorBrewer_1.1-2         
##  [5] reticulate_1.18             purrr_0.3.4                
##  [7] tidyr_1.1.3                 tibble_3.1.1               
##  [9] broom_0.7.6                 leiden_0.3.7               
## [11] cluster_2.1.0               dynamicTreeCut_1.63-1      
## [13] bluster_1.0.0               fossil_0.4.0               
## [15] shapefiles_0.7              foreign_0.8-80             
## [17] maps_3.3.0                  sp_1.4-5                   
## [19] clustree_0.4.3              ggraph_2.0.5               
## [21] pheatmap_1.0.12             harmony_1.0                
## [23] Rcpp_1.0.6                  batchelor_1.6.3            
## [25] SeuratObject_4.0.0          Seurat_4.0.1               
## [27] limma_3.46.0                ggfortify_0.4.11           
## [29] BiocSingular_1.6.0          scuttle_1.0.4              
## [31] robustbase_0.93-7           mixtools_1.2.0             
## [33] irlba_2.3.3                 biomaRt_2.46.3             
## [35] Matrix_1.2-18               igraph_1.2.6               
## [37] DropletUtils_1.10.3         scater_1.18.6              
## [39] ggplot2_3.3.3               scran_1.18.7               
## [41] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [43] Biobase_2.50.0              GenomicRanges_1.42.0       
## [45] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [47] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [49] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [51] dplyr_1.0.5                 DT_0.18                    
## [53] knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.1            scattermore_0.7          
##   [3] R.methodsS3_1.8.1         bit64_4.0.5              
##   [5] R.utils_2.10.1            data.table_1.14.0        
##   [7] rpart_4.1-15              RCurl_1.98-1.3           
##   [9] generics_0.1.0            cowplot_1.1.1            
##  [11] RSQLite_2.2.6             RANN_2.6.1               
##  [13] future_1.21.0             bit_4.0.4                
##  [15] spatstat.data_2.1-0       xml2_1.3.2               
##  [17] httpuv_1.5.5              isoband_0.2.4            
##  [19] assertthat_0.2.1          viridis_0.5.1            
##  [21] xfun_0.22                 hms_1.0.0                
##  [23] jquerylib_0.1.3           evaluate_0.14            
##  [25] promises_1.1.1            DEoptimR_1.0-8           
##  [27] fansi_0.4.2               progress_1.2.2           
##  [29] dbplyr_2.0.0              DBI_1.1.1                
##  [31] htmlwidgets_1.5.3         spatstat.geom_2.1-0      
##  [33] ellipsis_0.3.1            RSpectra_0.16-0          
##  [35] crosstalk_1.1.1           backports_1.2.1          
##  [37] bookdown_0.21             deldir_0.2-10            
##  [39] sparseMatrixStats_1.2.1   vctrs_0.3.7              
##  [41] ROCR_1.0-11               abind_1.4-5              
##  [43] cachem_1.0.4              withr_2.4.2              
##  [45] ggforce_0.3.3             sctransform_0.3.2.9005   
##  [47] prettyunits_1.1.1         goftest_1.2-2            
##  [49] segmented_1.3-3           lazyeval_0.2.2           
##  [51] crayon_1.4.1              labeling_0.4.2           
##  [53] pkgconfig_2.0.3           tweenr_1.0.2             
##  [55] nlme_3.1-149              vipor_0.4.5              
##  [57] rlang_0.4.10              globals_0.14.0           
##  [59] lifecycle_1.0.0           miniUI_0.1.1.1           
##  [61] BiocFileCache_1.14.0      rsvd_1.0.3               
##  [63] polyclip_1.10-0           lmtest_0.9-38            
##  [65] Rhdf5lib_1.12.1           zoo_1.8-9                
##  [67] beeswarm_0.2.3            ggridges_0.5.3           
##  [69] png_0.1-7                 viridisLite_0.3.0        
##  [71] bitops_1.0-6              R.oo_1.24.0              
##  [73] KernSmooth_2.23-17        rhdf5filters_1.2.0       
##  [75] blob_1.2.1                DelayedMatrixStats_1.12.3
##  [77] stringr_1.4.0             parallelly_1.24.0        
##  [79] readr_1.4.0               beachmat_2.6.4           
##  [81] scales_1.1.1              memoise_2.0.0            
##  [83] magrittr_2.0.1            plyr_1.8.6               
##  [85] ica_1.0-2                 zlibbioc_1.36.0          
##  [87] compiler_4.0.3            dqrng_0.2.1              
##  [89] fitdistrplus_1.1-3        XVector_0.30.0           
##  [91] listenv_0.8.0             patchwork_1.1.1          
##  [93] pbapply_1.4-3             MASS_7.3-53              
##  [95] mgcv_1.8-33               tidyselect_1.1.0         
##  [97] stringi_1.5.3             highr_0.8                
##  [99] yaml_2.2.1                askpass_1.1              
## [101] locfit_1.5-9.4            ggrepel_0.9.1            
## [103] grid_4.0.3                sass_0.3.1               
## [105] tools_4.0.3               future.apply_1.7.0       
## [107] rstudioapi_0.13           gridExtra_2.3            
## [109] farver_2.1.0              Rtsne_0.15               
## [111] digest_0.6.27             shiny_1.5.0              
## [113] later_1.1.0.1             RcppAnnoy_0.0.18         
## [115] httr_1.4.2                AnnotationDbi_1.52.0     
## [117] kernlab_0.9-29            colorspace_2.0-0         
## [119] XML_3.99-0.5              tensor_1.5               
## [121] splines_4.0.3             uwot_0.1.10              
## [123] statmod_1.4.35            spatstat.utils_2.1-0     
## [125] graphlayouts_0.7.1        plotly_4.9.3             
## [127] xtable_1.8-4              jsonlite_1.7.2           
## [129] tidygraph_1.2.0           R6_2.5.0                 
## [131] pillar_1.6.0              htmltools_0.5.1.1        
## [133] mime_0.10                 glue_1.4.2               
## [135] fastmap_1.1.0             BiocParallel_1.24.1      
## [137] BiocNeighbors_1.8.2       codetools_0.2-16         
## [139] utf8_1.2.1                lattice_0.20-41          
## [141] bslib_0.2.4               spatstat.sparse_2.0-0    
## [143] ResidualMatrix_1.0.0      curl_4.3                 
## [145] ggbeeswarm_0.6.0          openssl_1.4.3            
## [147] survival_3.2-7            rmarkdown_2.7            
## [149] munsell_0.5.0             rhdf5_2.34.0             
## [151] GenomeInfoDbData_1.2.4    HDF5Array_1.18.1         
## [153] reshape2_1.4.4            gtable_0.3.0             
## [155] spatstat.core_2.1-2</code></pre>

<div class="sourceCode" id="cb833"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb833-1"><a href="dimRedForVizTop.html#cb833-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb833-2"><a href="dimRedForVizTop.html#cb833-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb833-3"><a href="dimRedForVizTop.html#cb833-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb833-4"><a href="dimRedForVizTop.html#cb833-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span></code></pre></div>
<div class="sourceCode" id="cb834"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb834-1"><a href="dimRedForVizTop.html#cb834-1"></a><span class="kw">library</span>(knitr)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="NormalisationGSM3872434Top.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identifying-confounding-factors-caron-set.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/dimRedForViz.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
