<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Normalisation - GSM3872434 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Normalisation - GSM3872434 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Normalisation - GSM3872434 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="normalisation-caron-set.html"/>
<link rel="next" href="dimRedForVizTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>17</b> Data integration</a><ul>
<li class="chapter" data-level="17.1" data-path="data-integration.html"><a href="data-integration.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="data-integration.html"><a href="data-integration.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="data-integration.html"><a href="data-integration.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="data-integration.html"><a href="data-integration.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="data-integration.html"><a href="data-integration.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="data-integration.html"><a href="data-integration.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="data-integration.html"><a href="data-integration.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="data-integration.html"><a href="data-integration.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="data-integration.html"><a href="data-integration.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="data-integration.html"><a href="data-integration.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="data-integration.html"><a href="data-integration.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="data-integration.html"><a href="data-integration.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="data-integration.html"><a href="data-integration.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="data-integration.html"><a href="data-integration.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="data-integration.html"><a href="data-integration.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="data-integration.html"><a href="data-integration.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="large-cluster-with-moderate-pbmmc-proportion.html"><a href="large-cluster-with-moderate-pbmmc-proportion.html"><i class="fa fa-check"></i><b>19</b> large cluster with moderate PBMMC proportion:</a></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-7"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-11"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-8"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-9"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-5"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-6"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-12"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="NormalisationGSM3872434Top" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Normalisation - GSM3872434 set</h1>
<p>Sources: chapters on normalisation in the <a href="https://osca.bioconductor.org/normalization.html">OSCA book</a> and the <a href="https://scrnaseq-course.cog.sanger.ac.uk/website/index.html">‘Hemberg group material’</a>.</p>
<p>Why normalise?</p>
<p>Systematic differences in sequencing coverage between libraries occur because of low input material, differences in cDNA capture and PCR amplification.
Normalisation removes such differences so that differences between cells are not technical but biological, allowing meaningful comparison of expression profiles between cells. Normalisation and batch correction have different aims. Normalisation addresses technical differences only, while batch correction considers both technical and biological differences.</p>
<div class="sourceCode" id="cb679"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb679-1"><a href="NormalisationGSM3872434Top.html#cb679-1"></a>normPlotDirBit &lt;-<span class="st"> &quot;Plots/Norm&quot;</span></span>
<span id="cb679-2"><a href="NormalisationGSM3872434Top.html#cb679-2"></a></span>
<span id="cb679-3"><a href="NormalisationGSM3872434Top.html#cb679-3"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb679-4"><a href="NormalisationGSM3872434Top.html#cb679-4"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb679-5"><a href="NormalisationGSM3872434Top.html#cb679-5"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>) {</span>
<span id="cb679-6"><a href="NormalisationGSM3872434Top.html#cb679-6"></a>    setName &lt;-<span class="st"> &quot;GSM3872434&quot;</span></span>
<span id="cb679-7"><a href="NormalisationGSM3872434Top.html#cb679-7"></a>    setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb679-8"><a href="NormalisationGSM3872434Top.html#cb679-8"></a>    dirRel &lt;-<span class="st"> &quot;..&quot;</span></span>
<span id="cb679-9"><a href="NormalisationGSM3872434Top.html#cb679-9"></a>}</span>
<span id="cb679-10"><a href="NormalisationGSM3872434Top.html#cb679-10"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb679-11"><a href="NormalisationGSM3872434Top.html#cb679-11"></a><span class="kw">dir.create</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s&quot;</span>, projDir, outDirBit, normPlotDirBit),</span>
<span id="cb679-12"><a href="NormalisationGSM3872434Top.html#cb679-12"></a>       <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>Load object</p>
<div class="sourceCode" id="cb680"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb680-1"><a href="NormalisationGSM3872434Top.html#cb680-1"></a><span class="co"># Read object in:</span></span>
<span id="cb680-2"><a href="NormalisationGSM3872434Top.html#cb680-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc%s.Rds&quot;</span>,</span>
<span id="cb680-3"><a href="NormalisationGSM3872434Top.html#cb680-3"></a>         projDir, outDirBit, <span class="st">&quot;caron&quot;</span>, setSuf)</span>
<span id="cb680-4"><a href="NormalisationGSM3872434Top.html#cb680-4"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb680-5"><a href="NormalisationGSM3872434Top.html#cb680-5"></a>{</span>
<span id="cb680-6"><a href="NormalisationGSM3872434Top.html#cb680-6"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb680-7"><a href="NormalisationGSM3872434Top.html#cb680-7"></a>}</span>
<span id="cb680-8"><a href="NormalisationGSM3872434Top.html#cb680-8"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb680-9"><a href="NormalisationGSM3872434Top.html#cb680-9"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(1): counts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(16): Sample Barcode ... outlier cell_sparsity
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>Select cells for GSM3872434:</p>
<div class="sourceCode" id="cb682"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb682-1"><a href="NormalisationGSM3872434Top.html#cb682-1"></a><span class="co">##setSuf &lt;- &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb682-2"><a href="NormalisationGSM3872434Top.html#cb682-2"></a><span class="co">##nbCells &lt;- 500</span></span>
<span id="cb682-3"><a href="NormalisationGSM3872434Top.html#cb682-3"></a><span class="co">#setSuf &lt;- &quot;_1kCellPerSpl&quot;</span></span>
<span id="cb682-4"><a href="NormalisationGSM3872434Top.html#cb682-4"></a><span class="co">#nbCells &lt;- 1000</span></span>
<span id="cb682-5"><a href="NormalisationGSM3872434Top.html#cb682-5"></a>setSuf &lt;-<span class="st"> &quot;_GSM3872434&quot;</span></span>
<span id="cb682-6"><a href="NormalisationGSM3872434Top.html#cb682-6"></a><span class="co">##nbCells &lt;- 500</span></span></code></pre></div>
<div class="sourceCode" id="cb683"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb683-1"><a href="NormalisationGSM3872434Top.html#cb683-1"></a><span class="co"># have new list of cell barcodes for each sample</span></span>
<span id="cb683-2"><a href="NormalisationGSM3872434Top.html#cb683-2"></a>sce.nz.master &lt;-<span class="st"> </span>sce</span>
<span id="cb683-3"><a href="NormalisationGSM3872434Top.html#cb683-3"></a>vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.nz.master) <span class="op">%&gt;%</span></span>
<span id="cb683-4"><a href="NormalisationGSM3872434Top.html#cb683-4"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb683-5"><a href="NormalisationGSM3872434Top.html#cb683-5"></a><span class="st">    </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> &quot;GSM3872434&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb683-6"><a href="NormalisationGSM3872434Top.html#cb683-6"></a><span class="st">    </span><span class="kw">group_by</span>(Sample.Name) <span class="op">%&gt;%</span></span>
<span id="cb683-7"><a href="NormalisationGSM3872434Top.html#cb683-7"></a><span class="st">    </span><span class="co">##sample_n(nbCells) %&gt;%</span></span>
<span id="cb683-8"><a href="NormalisationGSM3872434Top.html#cb683-8"></a><span class="st">    </span><span class="kw">pull</span>(Barcode)</span></code></pre></div>
<p>Number of cells in the sample:</p>
<div class="sourceCode" id="cb684"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb684-1"><a href="NormalisationGSM3872434Top.html#cb684-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce.nz.master)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
## 44977  2853</code></pre>
<p>Subset cells from the SCE object:</p>
<div class="sourceCode" id="cb686"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb686-1"><a href="NormalisationGSM3872434Top.html#cb686-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce.nz.master)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb686-2"><a href="NormalisationGSM3872434Top.html#cb686-2"></a>sce &lt;-<span class="st"> </span>sce.nz.master[,tmpInd]</span>
<span id="cb686-3"><a href="NormalisationGSM3872434Top.html#cb686-3"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 2853 
## metadata(0):
## assays(1): counts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(16): Sample Barcode ... outlier cell_sparsity
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>Check columns data:</p>
<div class="sourceCode" id="cb688"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb688-1"><a href="NormalisationGSM3872434Top.html#cb688-1"></a><span class="kw">head</span>(<span class="kw">colData</span>(sce))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 16 columns
##                   Sample            Barcode         Run Sample.Name source_name
##              &lt;character&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;factor&gt;
## 1 /ssd/personal/baller.. AAACCTGAGACTTTCG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 2 /ssd/personal/baller.. AAACCTGGTCTTCAAG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 3 /ssd/personal/baller.. AAACCTGGTGTTGAGG-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 4 /ssd/personal/baller.. AAACCTGTCCCAAGTA-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 5 /ssd/personal/baller.. AAACCTGTCGAATGCT-1  SRR9264343  GSM3872434  ETV6-RUNX1
## 6 /ssd/personal/baller.. AAACGGGCACCATCCT-1  SRR9264343  GSM3872434  ETV6-RUNX1
##         sum  detected subsets_Mito_sum subsets_Mito_detected
##   &lt;numeric&gt; &lt;integer&gt;        &lt;numeric&gt;             &lt;integer&gt;
## 1      6462      1996              290                    12
## 2     11706      3079              567                    12
## 3      7981      2511              423                    12
## 4      8354      2307              519                    13
## 5      1373       701               91                    11
## 6      2142       985              102                    10
##   subsets_Mito_percent     total      block     setName   discard   outlier
##              &lt;numeric&gt; &lt;numeric&gt;   &lt;factor&gt; &lt;character&gt; &lt;logical&gt; &lt;logical&gt;
## 1              4.48777      6462 ETV6-RUNX1       Caron     FALSE     FALSE
## 2              4.84367     11706 ETV6-RUNX1       Caron     FALSE     FALSE
## 3              5.30009      7981 ETV6-RUNX1       Caron     FALSE     FALSE
## 4              6.21259      8354 ETV6-RUNX1       Caron     FALSE     FALSE
## 5              6.62782      1373 ETV6-RUNX1       Caron     FALSE     FALSE
## 6              4.76190      2142 ETV6-RUNX1       Caron     FALSE     FALSE
##   cell_sparsity
##       &lt;numeric&gt;
## 1      0.907503
## 2      0.857361
## 3      0.883637
## 4      0.893091
## 5      0.967515
## 6      0.954400</code></pre>
<div class="sourceCode" id="cb690"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb690-1"><a href="NormalisationGSM3872434Top.html#cb690-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name)</span></code></pre></div>
<pre><code>## 
## GSM3872434 
##       2853</code></pre>
<!--
# mind that genes were filtered using all cells, not just those sampled here.
-->
<p>We write the R object to GSM3872434_sce_nz_postQc_GSM3872434.Rds.</p>
<div class="sourceCode" id="cb692"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb692-1"><a href="NormalisationGSM3872434Top.html#cb692-1"></a><span class="co"># Write object to file</span></span>
<span id="cb692-2"><a href="NormalisationGSM3872434Top.html#cb692-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc%s.Rds&quot;</span>,</span>
<span id="cb692-3"><a href="NormalisationGSM3872434Top.html#cb692-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb692-4"><a href="NormalisationGSM3872434Top.html#cb692-4"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb693"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb693-1"><a href="NormalisationGSM3872434Top.html#cb693-1"></a><span class="co"># Write object to file</span></span>
<span id="cb693-2"><a href="NormalisationGSM3872434Top.html#cb693-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc%s.Rds&quot;</span>,</span>
<span id="cb693-3"><a href="NormalisationGSM3872434Top.html#cb693-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb693-4"><a href="NormalisationGSM3872434Top.html#cb693-4"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
<div id="scaling-normalization-1" class="section level2">
<h2><span class="header-section-number">8.1</span> Scaling normalization</h2>
<p>In scaling normalization, the “normalization factor” is an estimate of the library size relative to the other cells. Steps usually include: computation of a cell-specific ‘scaling’ or ‘size’ factor that represents the relative bias in that cell and division of all counts for the cell by that factor to remove that bias. Assumption: any cell specific bias will affect genes the same way.</p>
<p>Scaling methods typically generate normalised counts-per-million (CPM) or transcripts-per-million (TPM) values that address the effect of sequencing depth. These values however typically have a variance that changes with their mean (heteroscedasticity) while most statistical methods assume a stable variance, which does not vary with the mean (homoscedasticity). A widely used ‘variance stabilising transformation’ is the log transformation (often log2). This works fine for highly expressed genes (as in bulk RNA-seq) but less so for sparse scRNA-seq data.</p>
<div id="cpm-1" class="section level3">
<h3><span class="header-section-number">8.1.1</span> CPM</h3>
<p>Convert raw counts to counts-per-million (CPM) for each cell by dividing counts by the library size then multiplying by 1.000.000. Mind that this does not adress compositional bias caused by highly expressed genes that are also differentially expressed between cells. In <code>scater</code> CPMs are computed with the following code:</p>
<div class="sourceCode" id="cb694"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb694-1"><a href="NormalisationGSM3872434Top.html#cb694-1"></a>calc_cpm &lt;-<span class="st"> </span><span class="cf">function</span> (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) </span>
<span id="cb694-2"><a href="NormalisationGSM3872434Top.html#cb694-2"></a>{</span>
<span id="cb694-3"><a href="NormalisationGSM3872434Top.html#cb694-3"></a>    norm_factor &lt;-<span class="st"> </span><span class="kw">colSums</span>(expr_mat[<span class="op">-</span>spikes, ])</span>
<span id="cb694-4"><a href="NormalisationGSM3872434Top.html#cb694-4"></a>    <span class="kw">return</span>(<span class="kw">t</span>(<span class="kw">t</span>(expr_mat)<span class="op">/</span>norm_factor)) <span class="op">*</span><span class="st"> </span><span class="dv">10</span><span class="op">^</span><span class="dv">6</span></span>
<span id="cb694-5"><a href="NormalisationGSM3872434Top.html#cb694-5"></a>}</span></code></pre></div>
<p>We will use <code>scater</code>’s <code>calculateCPM()</code>.</p>
</div>
<div id="deseqs-size-factor-1" class="section level3">
<h3><span class="header-section-number">8.1.2</span> DESeq’s size factor</h3>
<p>For each gene, compute geometric mean across cells. For each cell compute for each gene the ratio of its expression to its geometric mean, and derive the cell’s size factor as the median ratio across genes. Not suitable for sparse scRNA-seq data as the geometric is computed on non-zero values only. This method is also known as ‘Relative Log Expression’ (RLE) in <code>edgeR</code> and <code>scater</code>.</p>
<p>Example code:</p>
<div class="sourceCode" id="cb695"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb695-1"><a href="NormalisationGSM3872434Top.html#cb695-1"></a>calc_sf &lt;-<span class="st"> </span><span class="cf">function</span> (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) </span>
<span id="cb695-2"><a href="NormalisationGSM3872434Top.html#cb695-2"></a>{</span>
<span id="cb695-3"><a href="NormalisationGSM3872434Top.html#cb695-3"></a>    geomeans &lt;-<span class="st"> </span><span class="kw">exp</span>(<span class="kw">rowMeans</span>(<span class="kw">log</span>(expr_mat[<span class="op">-</span>spikes, ])))</span>
<span id="cb695-4"><a href="NormalisationGSM3872434Top.html#cb695-4"></a>    SF &lt;-<span class="st"> </span><span class="cf">function</span>(cnts) {</span>
<span id="cb695-5"><a href="NormalisationGSM3872434Top.html#cb695-5"></a>        <span class="kw">median</span>((cnts<span class="op">/</span>geomeans)[(<span class="kw">is.finite</span>(geomeans) <span class="op">&amp;</span></span>
<span id="cb695-6"><a href="NormalisationGSM3872434Top.html#cb695-6"></a><span class="st">                </span>geomeans <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)])</span>
<span id="cb695-7"><a href="NormalisationGSM3872434Top.html#cb695-7"></a>    }</span>
<span id="cb695-8"><a href="NormalisationGSM3872434Top.html#cb695-8"></a>    norm_factor &lt;-<span class="st"> </span><span class="kw">apply</span>(expr_mat[<span class="op">-</span>spikes, ], <span class="dv">2</span>, SF)</span>
<span id="cb695-9"><a href="NormalisationGSM3872434Top.html#cb695-9"></a>    <span class="kw">return</span>(<span class="kw">t</span>(<span class="kw">t</span>(expr_mat)<span class="op">/</span>norm_factor))</span>
<span id="cb695-10"><a href="NormalisationGSM3872434Top.html#cb695-10"></a>}</span></code></pre></div>
</div>
<div id="weighted-trimmed-mean-of-m-values-1" class="section level3">
<h3><span class="header-section-number">8.1.3</span> Weighted Trimmed mean of M-values</h3>
<p>To compute weighted Trimmed mean of M-values (TMM), a given cell is chosen as a reference to use in computation for other cells. The M-values are gene-wise log2-fold changes between cells. Trimming entails the removal of the top and bottom 30% of values. The size factor is computed as the average for the remaining cells with a weight according to inverse variances. This method assumes that most genes are not differentially expressed, and the 40% of genes left after trimming may include many zero counts.</p>
<div class="sourceCode" id="cb696"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb696-1"><a href="NormalisationGSM3872434Top.html#cb696-1"></a><span class="kw">sizeFactors</span>(sce) &lt;-<span class="st"> </span>edgeR<span class="op">::</span><span class="kw">calcNormFactors</span>(<span class="kw">counts</span>(sce), <span class="dt">method =</span> <span class="st">&quot;TMM&quot;</span>)</span></code></pre></div>
</div>
<div id="library-size-factor-distribution" class="section level3">
<h3><span class="header-section-number">8.1.4</span> Library size factor distribution</h3>
<p>For each cell, the library size factor is proportional to the library size such that the average size factor across cells is one.</p>
<p>Advantage: normalised counts are on the same scale as the initial counts.</p>
<p>Compute size factors:</p>
<div class="sourceCode" id="cb697"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb697-1"><a href="NormalisationGSM3872434Top.html#cb697-1"></a>lib.sf &lt;-<span class="st"> </span><span class="kw">librarySizeFactors</span>(sce)</span>
<span id="cb697-2"><a href="NormalisationGSM3872434Top.html#cb697-2"></a><span class="kw">summary</span>(lib.sf)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## 0.09666 0.66197 0.92591 1.00000 1.23957 5.49973</code></pre>
<p>Size factor distribution: wide range, typical of scRNA-seq data.</p>
<div class="sourceCode" id="cb699"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb699-1"><a href="NormalisationGSM3872434Top.html#cb699-1"></a><span class="kw">hist</span>(<span class="kw">log10</span>(lib.sf), <span class="dt">xlab=</span><span class="st">&quot;Log10[Size factor]&quot;</span>, <span class="dt">col=</span><span class="st">&#39;grey80&#39;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/librarySizeFactors_hist_norm_GSM3872434_allCells-1.png" width="672" /></p>
<p>Assumption: absence of compositional bias; differential expression between two cells is balanced: upregulation in some genes is accompanied by downregulation of other genes. Not observed.</p>
<p>Inaccurate normalisation due to unaccounted-for composition bias affects the size of the log fold change measured between clusters, but less so the clustering itself. It is thus sufficient to identify clusters and top marker genes.</p>
</div>
<div id="deconvolution-3" class="section level3">
<h3><span class="header-section-number">8.1.5</span> Deconvolution</h3>
<p>Composition bias occurs when differential expression beteween two samples or here cells is not balanced. For a fixed library size, identical in both cells, upregulation of one gene in a cell will means fewer UMIs can be assigned to other genes, which would then appear down regulated. Even if library sizes are allowed to differ, with that for the cell with upregulation being higher, scaling normalisation will reduce normalised counts. Non-upregulated would therefore also appear downregulated.</p>
<p>For bulk RNA-seq, composition bias is removed by assuming that most genes are not differentially expressed between samples, so that differences in non-DE genes would amount to the bias, and used to compute size factors.</p>
<p>Given the sparsity of scRNA-seq data, the methods are not appropriate.</p>
<p>The method below increases read counts by pooling cells into groups, computing size factors within each of these groups and scaling them so they are comparable across clusters. This process is repeated many times, changing pools each time to collect several size factors for each cell, from which is derived a single value for that cell.</p>
<!--
see DESeq2 estimateSizeFactorsFromMatrix
see edgeR calcNormFactors
-->
<div class="sourceCode" id="cb700"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb700-1"><a href="NormalisationGSM3872434Top.html#cb700-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/Images/scran_Fig3.png&quot;</span>, <span class="st">&quot;..&quot;</span>)</span>
<span id="cb700-2"><a href="NormalisationGSM3872434Top.html#cb700-2"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Images/scran_Fig3.png" width="398" /></p>
<div class="sourceCode" id="cb701"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb701-1"><a href="NormalisationGSM3872434Top.html#cb701-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>Cluster cells then normalise.</p>
<div id="cluster-cells-1" class="section level4">
<h4><span class="header-section-number">8.1.5.1</span> Cluster cells</h4>
<div class="sourceCode" id="cb702"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb702-1"><a href="NormalisationGSM3872434Top.html#cb702-1"></a><span class="kw">set.seed</span>(<span class="dv">100</span>) <span class="co"># clusters with PCA from irlba with approximation</span></span>
<span id="cb702-2"><a href="NormalisationGSM3872434Top.html#cb702-2"></a>clust &lt;-<span class="st"> </span><span class="kw">quickCluster</span>(sce) <span class="co"># slow with all cells.</span></span>
<span id="cb702-3"><a href="NormalisationGSM3872434Top.html#cb702-3"></a></span>
<span id="cb702-4"><a href="NormalisationGSM3872434Top.html#cb702-4"></a><span class="co"># write to file</span></span>
<span id="cb702-5"><a href="NormalisationGSM3872434Top.html#cb702-5"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_quickClus%s.Rds&quot;</span>,</span>
<span id="cb702-6"><a href="NormalisationGSM3872434Top.html#cb702-6"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb702-7"><a href="NormalisationGSM3872434Top.html#cb702-7"></a><span class="kw">saveRDS</span>(clust, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb703"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb703-1"><a href="NormalisationGSM3872434Top.html#cb703-1"></a><span class="co"># read from file</span></span>
<span id="cb703-2"><a href="NormalisationGSM3872434Top.html#cb703-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_quickClus%s.Rds&quot;</span>,</span>
<span id="cb703-3"><a href="NormalisationGSM3872434Top.html#cb703-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb703-4"><a href="NormalisationGSM3872434Top.html#cb703-4"></a>clust &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb703-5"><a href="NormalisationGSM3872434Top.html#cb703-5"></a><span class="kw">table</span>(clust)</span></code></pre></div>
<pre><code>## clust
##   1   2   3   4   5   6   7   8   9 
## 232 193 617 315 407 380 398 189 122</code></pre>
</div>
<div id="compute-size-factors-3" class="section level4">
<h4><span class="header-section-number">8.1.5.2</span> Compute size factors</h4>
<div class="sourceCode" id="cb705"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb705-1"><a href="NormalisationGSM3872434Top.html#cb705-1"></a><span class="co">#deconv.sf &lt;- calculateSumFactors(sce, cluster=clust)</span></span>
<span id="cb705-2"><a href="NormalisationGSM3872434Top.html#cb705-2"></a>sce &lt;-<span class="st"> </span><span class="kw">computeSumFactors</span>(sce, <span class="dt">cluster=</span>clust, <span class="dt">min.mean=</span><span class="fl">0.1</span>)</span>
<span id="cb705-3"><a href="NormalisationGSM3872434Top.html#cb705-3"></a>deconv.sf &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(sce)</span>
<span id="cb705-4"><a href="NormalisationGSM3872434Top.html#cb705-4"></a></span>
<span id="cb705-5"><a href="NormalisationGSM3872434Top.html#cb705-5"></a><span class="co"># write to file</span></span>
<span id="cb705-6"><a href="NormalisationGSM3872434Top.html#cb705-6"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_deconvSf%s.Rds&quot;</span>,</span>
<span id="cb705-7"><a href="NormalisationGSM3872434Top.html#cb705-7"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb705-8"><a href="NormalisationGSM3872434Top.html#cb705-8"></a><span class="kw">saveRDS</span>(deconv.sf, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb706"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb706-1"><a href="NormalisationGSM3872434Top.html#cb706-1"></a><span class="co"># read from file</span></span>
<span id="cb706-2"><a href="NormalisationGSM3872434Top.html#cb706-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_deconvSf%s.Rds&quot;</span>,</span>
<span id="cb706-3"><a href="NormalisationGSM3872434Top.html#cb706-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb706-4"><a href="NormalisationGSM3872434Top.html#cb706-4"></a>deconv.sf &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb706-5"><a href="NormalisationGSM3872434Top.html#cb706-5"></a></span>
<span id="cb706-6"><a href="NormalisationGSM3872434Top.html#cb706-6"></a><span class="kw">summary</span>(deconv.sf)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## 0.08207 0.67672 0.91982 1.00000 1.22628 4.06014</code></pre>
<p>Plot deconvolution size factors against library size factors:</p>
<div class="sourceCode" id="cb708"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb708-1"><a href="NormalisationGSM3872434Top.html#cb708-1"></a>deconvDf &lt;-<span class="st"> </span><span class="kw">data.frame</span>(lib.sf, deconv.sf,</span>
<span id="cb708-2"><a href="NormalisationGSM3872434Top.html#cb708-2"></a>            <span class="st">&quot;source_name&quot;</span> =<span class="st"> </span>sce<span class="op">$</span>source_name,</span>
<span id="cb708-3"><a href="NormalisationGSM3872434Top.html#cb708-3"></a>            <span class="st">&quot;sum&quot;</span> =<span class="st"> </span>sce<span class="op">$</span>sum,</span>
<span id="cb708-4"><a href="NormalisationGSM3872434Top.html#cb708-4"></a>            <span class="st">&quot;mito_content&quot;</span> =<span class="st"> </span>sce<span class="op">$</span>subsets_Mito_percent,</span>
<span id="cb708-5"><a href="NormalisationGSM3872434Top.html#cb708-5"></a>            <span class="st">&quot;cell_sparsity&quot;</span> =<span class="st"> </span>sce<span class="op">$</span>cell_sparsity)</span></code></pre></div>
<div class="sourceCode" id="cb709"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb709-1"><a href="NormalisationGSM3872434Top.html#cb709-1"></a>sp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(deconvDf, <span class="kw">aes</span>(<span class="dt">x=</span>lib.sf, <span class="dt">y=</span>deconv.sf, <span class="dt">col=</span>source_name)) <span class="op">+</span></span>
<span id="cb709-2"><a href="NormalisationGSM3872434Top.html#cb709-2"></a><span class="st">  </span><span class="kw">geom_point</span>()</span>
<span id="cb709-3"><a href="NormalisationGSM3872434Top.html#cb709-3"></a></span>
<span id="cb709-4"><a href="NormalisationGSM3872434Top.html#cb709-4"></a><span class="co"># Split by sample type:</span></span>
<span id="cb709-5"><a href="NormalisationGSM3872434Top.html#cb709-5"></a><span class="co">#sp + facet_wrap(~source_name)</span></span></code></pre></div>
<div class="sourceCode" id="cb710"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb710-1"><a href="NormalisationGSM3872434Top.html#cb710-1"></a>sp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(deconvDf, <span class="kw">aes</span>(<span class="dt">x=</span>lib.sf, <span class="dt">y=</span>deconv.sf, <span class="dt">col=</span>cell_sparsity)) <span class="op">+</span></span>
<span id="cb710-2"><a href="NormalisationGSM3872434Top.html#cb710-2"></a><span class="st">  </span><span class="kw">geom_point</span>()</span>
<span id="cb710-3"><a href="NormalisationGSM3872434Top.html#cb710-3"></a>sp</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/scatter_deconvSf_libSf_colBy_cellSpars_norm_GSM3872434_allCells-1.png" width="672" /></p>
</div>
<div id="apply-size-factors-3" class="section level4">
<h4><span class="header-section-number">8.1.5.3</span> Apply size factors</h4>
<p>For each cell, raw counts for genes are divided by the size factor for that cell and log-transformed so downstream analyses focus on genes with strong relative differences. We use <code>scater::logNormCounts()</code>.</p>
<div class="sourceCode" id="cb711"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb711-1"><a href="NormalisationGSM3872434Top.html#cb711-1"></a>sce &lt;-<span class="st"> </span><span class="kw">logNormCounts</span>(sce) <span class="co"># adds logcounts</span></span>
<span id="cb711-2"><a href="NormalisationGSM3872434Top.html#cb711-2"></a><span class="kw">print</span>(<span class="kw">assayNames</span>(sce))</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
</div>
<div id="save-object-3" class="section level4">
<h4><span class="header-section-number">8.1.5.4</span> Save object</h4>
<div class="sourceCode" id="cb713"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb713-1"><a href="NormalisationGSM3872434Top.html#cb713-1"></a><span class="co"># write to file</span></span>
<span id="cb713-2"><a href="NormalisationGSM3872434Top.html#cb713-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb713-3"><a href="NormalisationGSM3872434Top.html#cb713-3"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb714"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb714-1"><a href="NormalisationGSM3872434Top.html#cb714-1"></a>sceDeconv &lt;-<span class="st"> </span>sce</span></code></pre></div>
</div>
</div>
</div>
<div id="sctransform-3" class="section level2">
<h2><span class="header-section-number">8.2</span> SCTransform</h2>
<!--
https://rawgit.com/ChristophH/sctransform/master/inst/doc/variance_stabilizing_transformation.html

vars.to.regress = c("S.Score", "G2M.Score")
vars.to.regress = c("percentMito","Sex")
-->
<p>With scaling normalisation a correlation remains between the mean and variation of expression (heteroskedasticity). This affects downstream dimensionality reduction as the few main new dimensions are usually correlated with library size. SCTransform addresses the issue by regressing library size out of raw counts and providing residuals to use as normalized and variance-stabilized expression values in downstream analysis. We will use the <a href="https://cran.r-project.org/web/packages/sctransform/index.html">sctransform vignette</a>.</p>
<div class="sourceCode" id="cb715"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb715-1"><a href="NormalisationGSM3872434Top.html#cb715-1"></a>counts &lt;-<span class="st"> </span><span class="kw">counts</span>(sce)</span>
<span id="cb715-2"><a href="NormalisationGSM3872434Top.html#cb715-2"></a><span class="kw">print</span>(<span class="kw">class</span>(counts))</span></code></pre></div>
<pre><code>## [1] &quot;dgCMatrix&quot;
## attr(,&quot;package&quot;)
## [1] &quot;Matrix&quot;</code></pre>
<div class="sourceCode" id="cb717"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb717-1"><a href="NormalisationGSM3872434Top.html#cb717-1"></a><span class="kw">print</span>(<span class="kw">dim</span>(counts))</span></code></pre></div>
<pre><code>## [1] 18431  2853</code></pre>
<div class="sourceCode" id="cb719"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb719-1"><a href="NormalisationGSM3872434Top.html#cb719-1"></a><span class="kw">colnames</span>(counts) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Barcode</span></code></pre></div>
<div id="inspect-data-1" class="section level3">
<h3><span class="header-section-number">8.2.1</span> Inspect data</h3>
<p>We will now calculate some properties and visually inspect the data. Our main interest is in the general trends not in individual outliers. Neither genes nor cells that stand out are important at this step, but we focus on the global trends.</p>
<p>Derive gene and cell attributes from the UMI matrix.</p>
<div class="sourceCode" id="cb720"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb720-1"><a href="NormalisationGSM3872434Top.html#cb720-1"></a>gene_attr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">mean =</span> <span class="kw">rowMeans</span>(counts), </span>
<span id="cb720-2"><a href="NormalisationGSM3872434Top.html#cb720-2"></a>                        <span class="dt">detection_rate =</span> <span class="kw">rowMeans</span>(counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>),</span>
<span id="cb720-3"><a href="NormalisationGSM3872434Top.html#cb720-3"></a>                        <span class="dt">var =</span> <span class="kw">apply</span>(counts, <span class="dv">1</span>, var))</span>
<span id="cb720-4"><a href="NormalisationGSM3872434Top.html#cb720-4"></a>gene_attr<span class="op">$</span>log_mean &lt;-<span class="st"> </span><span class="kw">log10</span>(gene_attr<span class="op">$</span>mean)</span>
<span id="cb720-5"><a href="NormalisationGSM3872434Top.html#cb720-5"></a>gene_attr<span class="op">$</span>log_var &lt;-<span class="st"> </span><span class="kw">log10</span>(gene_attr<span class="op">$</span>var)</span>
<span id="cb720-6"><a href="NormalisationGSM3872434Top.html#cb720-6"></a><span class="kw">rownames</span>(gene_attr) &lt;-<span class="st"> </span><span class="kw">rownames</span>(counts)</span>
<span id="cb720-7"><a href="NormalisationGSM3872434Top.html#cb720-7"></a>cell_attr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">n_umi =</span> <span class="kw">colSums</span>(counts),</span>
<span id="cb720-8"><a href="NormalisationGSM3872434Top.html#cb720-8"></a>                        <span class="dt">n_gene =</span> <span class="kw">colSums</span>(counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>))</span>
<span id="cb720-9"><a href="NormalisationGSM3872434Top.html#cb720-9"></a><span class="kw">rownames</span>(cell_attr) &lt;-<span class="st"> </span><span class="kw">colnames</span>(counts)</span></code></pre></div>
<div class="sourceCode" id="cb721"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb721-1"><a href="NormalisationGSM3872434Top.html#cb721-1"></a><span class="kw">dim</span>(gene_attr)</span></code></pre></div>
<pre><code>## [1] 18431     5</code></pre>
<div class="sourceCode" id="cb723"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb723-1"><a href="NormalisationGSM3872434Top.html#cb723-1"></a><span class="kw">head</span>(gene_attr)</span></code></pre></div>
<pre><code>##                         mean detection_rate          var   log_mean    log_var
## ENSG00000238009 0.0007010165   0.0007010165 0.0007007707 -3.1542718 -3.1544241
## ENSG00000237491 0.0364528566   0.0357518402 0.0365388860 -1.4382684 -1.4372447
## ENSG00000225880 0.0318962496   0.0304942166 0.0336947550 -1.4962604 -1.4724377
## ENSG00000230368 0.0140203295   0.0136698212 0.0145298692 -1.8532418 -1.8377383
## ENSG00000230699 0.0021030494   0.0021030494 0.0020993624 -2.6771505 -2.6779126
## ENSG00000188976 0.2958289520   0.2509638977 0.3065639427 -0.5289593 -0.5134789</code></pre>
<div class="sourceCode" id="cb725"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb725-1"><a href="NormalisationGSM3872434Top.html#cb725-1"></a><span class="kw">dim</span>(cell_attr)</span></code></pre></div>
<pre><code>## [1] 2853    2</code></pre>
<div class="sourceCode" id="cb727"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb727-1"><a href="NormalisationGSM3872434Top.html#cb727-1"></a><span class="kw">head</span>(cell_attr)</span></code></pre></div>
<pre><code>##                    n_umi n_gene
## AAACCTGAGACTTTCG-1  6462   1996
## AAACCTGGTCTTCAAG-1 11705   3078
## AAACCTGGTGTTGAGG-1  7981   2511
## AAACCTGTCCCAAGTA-1  8354   2307
## AAACCTGTCGAATGCT-1  1373    701
## AAACGGGCACCATCCT-1  2141    984</code></pre>
<p>Mean-variance relationship</p>
<p>For the genes, we can see that up to a mean UMI count of 0 the variance follows the line through the origin with slop one, i.e. variance and mean are roughly equal as expected under a Poisson model. However, genes with a higher average UMI count show overdispersion compared to Poisson.</p>
<div class="sourceCode" id="cb729"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb729-1"><a href="NormalisationGSM3872434Top.html#cb729-1"></a><span class="kw">ggplot</span>(gene_attr, <span class="kw">aes</span>(log_mean, log_var)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb729-2"><a href="NormalisationGSM3872434Top.html#cb729-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.3</span>, <span class="dt">shape=</span><span class="dv">16</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb729-3"><a href="NormalisationGSM3872434Top.html#cb729-3"></a><span class="st">  </span><span class="kw">geom_density_2d</span>(<span class="dt">size =</span> <span class="fl">0.3</span>) <span class="op">+</span></span>
<span id="cb729-4"><a href="NormalisationGSM3872434Top.html#cb729-4"></a><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope =</span> <span class="dv">1</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/attr_plot_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Mean-detection-rate relationship</p>
<p>In line with the previous plot, we see a lower than expected detection rate in the medium expression range. However, for the highly expressed genes, the rate is at or very close to 1.0 suggesting that there is no zero-inflation in the counts for those genes and that zero-inflation is a result of overdispersion, rather than an independent systematic bias.</p>
<div class="sourceCode" id="cb730"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb730-1"><a href="NormalisationGSM3872434Top.html#cb730-1"></a><span class="co"># add the expected detection rate under Poisson model</span></span>
<span id="cb730-2"><a href="NormalisationGSM3872434Top.html#cb730-2"></a>x =<span class="st"> </span><span class="kw">seq</span>(<span class="dt">from =</span> <span class="dv">-3</span>, <span class="dt">to =</span> <span class="dv">2</span>, <span class="dt">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb730-3"><a href="NormalisationGSM3872434Top.html#cb730-3"></a>poisson_model &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">log_mean =</span> x, <span class="dt">detection_rate =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="kw">dpois</span>(<span class="dv">0</span>, <span class="dt">lambda =</span> <span class="dv">10</span><span class="op">^</span>x))</span>
<span id="cb730-4"><a href="NormalisationGSM3872434Top.html#cb730-4"></a><span class="kw">ggplot</span>(gene_attr, <span class="kw">aes</span>(log_mean, detection_rate)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb730-5"><a href="NormalisationGSM3872434Top.html#cb730-5"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.3</span>, <span class="dt">shape=</span><span class="dv">16</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb730-6"><a href="NormalisationGSM3872434Top.html#cb730-6"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data=</span>poisson_model, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span></span>
<span id="cb730-7"><a href="NormalisationGSM3872434Top.html#cb730-7"></a><span class="st">  </span><span class="kw">theme_gray</span>(<span class="dt">base_size =</span> <span class="dv">8</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/scatter_detecRate_logMean_sct_GSM3872434_allCells-1.png" width="672" /></p>
</div>
<div id="transformation-1" class="section level3">
<h3><span class="header-section-number">8.2.2</span> Transformation</h3>
<p>“Based on the observations above, which are not unique to this particular data set, we propose to model the expression of each gene as a negative binomial random variable with a mean that depends on other variables. Here the other variables can be used to model the differences in sequencing depth between cells and are used as independent variables in a regression model. In order to avoid overfitting, we will first fit model parameters per gene, and then use the relationship between gene mean and parameter values to fit parameters, thereby combining information across genes. Given the fitted model parameters, we transform each observed UMI count into a Pearson residual which can be interpreted as the number of standard deviations an observed count was away from the expected mean. If the model accurately describes the mean-variance relationship and the dependency of mean and latent factors, then the result should have mean zero and a stable variance across the range of expression.” <a href="https://cran.r-project.org/web/packages/sctransform/index.html">sctransform vignette</a>.</p>
<p>Estimate model parameters and transform data</p>
<p>The vst function estimates model parameters and performs the variance stabilizing transformation. Here we use the log10 of the total UMI counts of a cell as variable for sequencing depth for each cell. After data transformation we plot the model parameters as a function of gene mean (geometric mean).</p>
<div class="sourceCode" id="cb731"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb731-1"><a href="NormalisationGSM3872434Top.html#cb731-1"></a><span class="kw">print</span>(<span class="kw">dim</span>(counts))</span></code></pre></div>
<pre><code>## [1] 18431  2853</code></pre>
<div class="sourceCode" id="cb733"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb733-1"><a href="NormalisationGSM3872434Top.html#cb733-1"></a><span class="co"># We use the Future API for parallel processing; set parameters here</span></span>
<span id="cb733-2"><a href="NormalisationGSM3872434Top.html#cb733-2"></a>future<span class="op">::</span><span class="kw">plan</span>(<span class="dt">strategy =</span> <span class="st">&#39;multicore&#39;</span>, <span class="dt">workers =</span> <span class="dv">4</span>)</span>
<span id="cb733-3"><a href="NormalisationGSM3872434Top.html#cb733-3"></a><span class="kw">options</span>(<span class="dt">future.globals.maxSize =</span> <span class="dv">10</span> <span class="op">*</span><span class="st"> </span><span class="dv">1024</span> <span class="op">^</span><span class="st"> </span><span class="dv">3</span>)</span>
<span id="cb733-4"><a href="NormalisationGSM3872434Top.html#cb733-4"></a></span>
<span id="cb733-5"><a href="NormalisationGSM3872434Top.html#cb733-5"></a><span class="kw">set.seed</span>(<span class="dv">44</span>)</span>
<span id="cb733-6"><a href="NormalisationGSM3872434Top.html#cb733-6"></a>vst_out &lt;-<span class="st"> </span>sctransform<span class="op">::</span><span class="kw">vst</span>(counts,</span>
<span id="cb733-7"><a href="NormalisationGSM3872434Top.html#cb733-7"></a>                            <span class="dt">latent_var =</span> <span class="kw">c</span>(<span class="st">&#39;log_umi&#39;</span>),</span>
<span id="cb733-8"><a href="NormalisationGSM3872434Top.html#cb733-8"></a>                            <span class="dt">return_gene_attr =</span> <span class="ot">TRUE</span>,</span>
<span id="cb733-9"><a href="NormalisationGSM3872434Top.html#cb733-9"></a>                            <span class="dt">return_cell_attr =</span> <span class="ot">TRUE</span>,</span>
<span id="cb733-10"><a href="NormalisationGSM3872434Top.html#cb733-10"></a>                            <span class="dt">show_progress =</span> <span class="ot">FALSE</span>)</span>
<span id="cb733-11"><a href="NormalisationGSM3872434Top.html#cb733-11"></a>sctransform<span class="op">::</span><span class="kw">plot_model_pars</span>(vst_out)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/comp_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Inspect model:</p>
<div class="sourceCode" id="cb734"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb734-1"><a href="NormalisationGSM3872434Top.html#cb734-1"></a><span class="kw">print</span>(vst_out<span class="op">$</span>model_str)</span></code></pre></div>
<pre><code>## [1] &quot;y ~ log_umi&quot;</code></pre>
<p>We will look at several genes in more detail.</p>
<div class="sourceCode" id="cb736"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb736-1"><a href="NormalisationGSM3872434Top.html#cb736-1"></a><span class="co">#sctransform::plot_model(vst_out, counts, c(&#39;MALAT1&#39;, &#39;RPL10&#39;, &#39;FTL&#39;), plot_residual = TRUE)</span></span>
<span id="cb736-2"><a href="NormalisationGSM3872434Top.html#cb736-2"></a><span class="kw">rowData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb736-3"><a href="NormalisationGSM3872434Top.html#cb736-3"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span></span>
<span id="cb736-4"><a href="NormalisationGSM3872434Top.html#cb736-4"></a><span class="st">  </span><span class="kw">filter</span>(Symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;MALAT1&#39;</span>, <span class="st">&#39;RPL10&#39;</span>, <span class="st">&#39;FTL&#39;</span>))</span></code></pre></div>
<pre><code>##                 ensembl_gene_id external_gene_name chromosome_name
## ENSG00000147403 ENSG00000147403              RPL10               X
## ENSG00000251562 ENSG00000251562             MALAT1              11
## ENSG00000087086 ENSG00000087086                FTL              19
##                 start_position end_position strand Symbol            Type
## ENSG00000147403      154389955    154409168      1  RPL10 Gene Expression
## ENSG00000251562       65497688     65506516      1 MALAT1 Gene Expression
## ENSG00000087086       48965309     48966879      1    FTL Gene Expression
##                      mean detected gene_sparsity
## ENSG00000147403  48.85548 99.28030   0.011540874
## ENSG00000251562 193.05529 99.09786   0.005352289
## ENSG00000087086  15.48621 96.42481   0.085532093</code></pre>
<div class="sourceCode" id="cb738"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb738-1"><a href="NormalisationGSM3872434Top.html#cb738-1"></a>sctransform<span class="op">::</span><span class="kw">plot_model</span>(vst_out,</span>
<span id="cb738-2"><a href="NormalisationGSM3872434Top.html#cb738-2"></a>                        counts,</span>
<span id="cb738-3"><a href="NormalisationGSM3872434Top.html#cb738-3"></a>                        <span class="kw">c</span>(<span class="st">&#39;ENSG00000251562&#39;</span>, <span class="st">&#39;ENSG00000147403&#39;</span>, <span class="st">&#39;ENSG00000087086&#39;</span>),</span>
<span id="cb738-4"><a href="NormalisationGSM3872434Top.html#cb738-4"></a>                        <span class="dt">plot_residual =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_1_sct_GSM3872434_allCells-1.png" width="672" /></p>
<div class="sourceCode" id="cb739"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb739-1"><a href="NormalisationGSM3872434Top.html#cb739-1"></a>sctransform<span class="op">::</span><span class="kw">plot_model</span>(vst_out,</span>
<span id="cb739-2"><a href="NormalisationGSM3872434Top.html#cb739-2"></a>                        counts,</span>
<span id="cb739-3"><a href="NormalisationGSM3872434Top.html#cb739-3"></a>                        <span class="kw">c</span>(<span class="st">&#39;ENSG00000087086&#39;</span>),</span>
<span id="cb739-4"><a href="NormalisationGSM3872434Top.html#cb739-4"></a>                        <span class="dt">plot_residual =</span> <span class="ot">TRUE</span>,</span>
<span id="cb739-5"><a href="NormalisationGSM3872434Top.html#cb739-5"></a>                        <span class="dt">show_nr =</span> <span class="ot">TRUE</span>,</span>
<span id="cb739-6"><a href="NormalisationGSM3872434Top.html#cb739-6"></a>                        <span class="dt">arrange_vertical =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_2_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Distribution of residual mean:</p>
<div class="sourceCode" id="cb740"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb740-1"><a href="NormalisationGSM3872434Top.html#cb740-1"></a><span class="kw">ggplot</span>(vst_out<span class="op">$</span>gene_attr, <span class="kw">aes</span>(residual_mean)) <span class="op">+</span></span>
<span id="cb740-2"><a href="NormalisationGSM3872434Top.html#cb740-2"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">binwidth=</span><span class="fl">0.01</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_resMean_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Distribution of residual variance:</p>
<div class="sourceCode" id="cb741"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb741-1"><a href="NormalisationGSM3872434Top.html#cb741-1"></a><span class="kw">ggplot</span>(vst_out<span class="op">$</span>gene_attr, <span class="kw">aes</span>(residual_variance)) <span class="op">+</span></span>
<span id="cb741-2"><a href="NormalisationGSM3872434Top.html#cb741-2"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">binwidth=</span><span class="fl">0.1</span>) <span class="op">+</span></span>
<span id="cb741-3"><a href="NormalisationGSM3872434Top.html#cb741-3"></a><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">xintercept=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span></span>
<span id="cb741-4"><a href="NormalisationGSM3872434Top.html#cb741-4"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">10</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_resVar_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Variance against mean (residuals):</p>
<div class="sourceCode" id="cb742"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb742-1"><a href="NormalisationGSM3872434Top.html#cb742-1"></a><span class="kw">ggplot</span>(vst_out<span class="op">$</span>gene_attr, <span class="kw">aes</span>(<span class="dt">x=</span>residual_mean, <span class="dt">y=</span>residual_variance)) <span class="op">+</span></span>
<span id="cb742-2"><a href="NormalisationGSM3872434Top.html#cb742-2"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.3</span>, <span class="dt">shape=</span><span class="dv">16</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb742-3"><a href="NormalisationGSM3872434Top.html#cb742-3"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb742-4"><a href="NormalisationGSM3872434Top.html#cb742-4"></a><span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="dv">10</span>) <span class="op">+</span></span>
<span id="cb742-5"><a href="NormalisationGSM3872434Top.html#cb742-5"></a><span class="st">    </span><span class="kw">geom_density_2d</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_resVar_resMean_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Variance against mean (genes):</p>
<div class="sourceCode" id="cb743"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb743-1"><a href="NormalisationGSM3872434Top.html#cb743-1"></a><span class="kw">ggplot</span>(vst_out<span class="op">$</span>gene_attr, <span class="kw">aes</span>(<span class="kw">log10</span>(gmean), residual_variance)) <span class="op">+</span></span>
<span id="cb743-2"><a href="NormalisationGSM3872434Top.html#cb743-2"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.3</span>, <span class="dt">shape=</span><span class="dv">16</span>) <span class="op">+</span></span>
<span id="cb743-3"><a href="NormalisationGSM3872434Top.html#cb743-3"></a><span class="st">  </span><span class="kw">geom_density_2d</span>(<span class="dt">size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/plot_model_resVar_gMean_sct_GSM3872434_allCells-1.png" width="672" /></p>
<p>Check genes with large residual variance:</p>
<div class="sourceCode" id="cb744"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb744-1"><a href="NormalisationGSM3872434Top.html#cb744-1"></a>dd &lt;-<span class="st"> </span>vst_out<span class="op">$</span>gene_attr <span class="op">%&gt;%</span></span>
<span id="cb744-2"><a href="NormalisationGSM3872434Top.html#cb744-2"></a><span class="st">    </span><span class="kw">arrange</span>(<span class="op">-</span>residual_variance) <span class="op">%&gt;%</span></span>
<span id="cb744-3"><a href="NormalisationGSM3872434Top.html#cb744-3"></a><span class="st">    </span><span class="kw">slice_head</span>(<span class="dt">n =</span> <span class="dv">22</span>) <span class="op">%&gt;%</span></span>
<span id="cb744-4"><a href="NormalisationGSM3872434Top.html#cb744-4"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="kw">across</span>(<span class="kw">where</span>(is.numeric), round, <span class="dv">2</span>))</span>
<span id="cb744-5"><a href="NormalisationGSM3872434Top.html#cb744-5"></a></span>
<span id="cb744-6"><a href="NormalisationGSM3872434Top.html#cb744-6"></a>dd <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb744-7"><a href="NormalisationGSM3872434Top.html#cb744-7"></a><span class="st">    </span><span class="kw">left_join</span>(<span class="kw">as.data.frame</span>(<span class="kw">rowData</span>(sce))[,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;Symbol&quot;</span>)], <span class="st">&quot;ensembl_gene_id&quot;</span>)</span></code></pre></div>
<pre><code>##    ensembl_gene_id detection_rate gmean variance residual_mean
## 1  ENSG00000197061           0.72  2.04   534.32          1.90
## 2  ENSG00000175063           0.21  0.36    26.18          0.68
## 3  ENSG00000164104           0.66  1.74   120.31          1.06
## 4  ENSG00000100097           0.06  0.07     3.85          0.31
## 5  ENSG00000008517           0.01  0.02     0.87          0.20
## 6  ENSG00000123416           0.91  5.22   447.36          1.07
## 7  ENSG00000211679           0.08  0.09     1.46          0.30
## 8  ENSG00000102970           0.03  0.03     2.22          0.15
## 9  ENSG00000170540           0.54  0.86    38.67          0.47
## 10 ENSG00000164611           0.21  0.31    12.05          0.41
## 11 ENSG00000128322           0.33  0.61    11.86          0.69
## 12 ENSG00000244734           0.45  0.49 85356.97          0.15
## 13 ENSG00000188536           0.16  0.15  8088.98          0.11
## 14 ENSG00000206172           0.12  0.11  4995.50          0.09
## 15 ENSG00000271503           0.02  0.02     0.41          0.12
## 16 ENSG00000131747           0.16  0.27     7.92          0.34
## 17 ENSG00000187837           0.67  1.53    21.13          0.78
## 18 ENSG00000026025           0.69  1.92    58.23          0.73
## 19 ENSG00000115523           0.00  0.01     0.24          0.06
## 20 ENSG00000145649           0.00  0.01     0.06          0.08
## 21 ENSG00000158578           0.01  0.01     1.67          0.06
## 22 ENSG00000117399           0.11  0.16     4.07          0.25
##    residual_variance   Symbol
## 1              65.39 HIST1H4C
## 2              20.67    UBE2C
## 3              16.78    HMGB2
## 4              16.15   LGALS1
## 5              15.36     IL32
## 6              13.55   TUBA1B
## 7              12.64    IGLC3
## 8              10.97    CCL17
## 9              10.80  ARL6IP1
## 10             10.77    PTTG1
## 11             10.17    IGLL1
## 12              9.95      HBB
## 13              9.83     HBA2
## 14              9.75     HBA1
## 15              9.02     CCL5
## 16              8.63    TOP2A
## 17              8.44 HIST1H1C
## 18              8.42      VIM
## 19              7.85     GNLY
## 20              7.79     GZMA
## 21              7.78    ALAS2
## 22              7.53    CDC20</code></pre>
<p>Write outcome to file:</p>
<div class="sourceCode" id="cb746"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb746-1"><a href="NormalisationGSM3872434Top.html#cb746-1"></a><span class="co"># write to file</span></span>
<span id="cb746-2"><a href="NormalisationGSM3872434Top.html#cb746-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_vst_out%s.Rds&quot;</span>,</span>
<span id="cb746-3"><a href="NormalisationGSM3872434Top.html#cb746-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb746-4"><a href="NormalisationGSM3872434Top.html#cb746-4"></a><span class="kw">saveRDS</span>(vst_out, tmpFn)</span></code></pre></div>
<p>Check transformed values.</p>
<div class="sourceCode" id="cb747"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb747-1"><a href="NormalisationGSM3872434Top.html#cb747-1"></a><span class="kw">print</span>(<span class="kw">dim</span>(vst_out<span class="op">$</span>y))</span></code></pre></div>
<pre><code>## [1] 14214  2853</code></pre>
<div class="sourceCode" id="cb749"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb749-1"><a href="NormalisationGSM3872434Top.html#cb749-1"></a>vst_out<span class="op">$</span>y[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                 AAACCTGAGACTTTCG-1 AAACCTGGTCTTCAAG-1 AAACCTGGTGTTGAGG-1
## ENSG00000237491        -0.20121617        -0.25944999        -0.22046915
## ENSG00000225880        -0.18668543        -0.24052717        -0.20448848
## ENSG00000230368        -0.12439501        -0.16164785        -0.13667117
## ENSG00000230699        -0.04842402        -0.06042004        -0.05240396
## ENSG00000188976         1.03374030         1.56177764         0.80159808
## ENSG00000187961        -0.05363629        -0.06698983        -0.05806420
## ENSG00000188290        -0.09174430        -0.12027862        -0.10110587
## ENSG00000187608         2.45936165        -0.77304781        -0.65435652
## ENSG00000188157        -0.07078407        -0.09039073        -0.07724886
## ENSG00000131591        -0.16874897        -0.21715523        -0.18475767
##                 AAACCTGTCCCAAGTA-1 AAACCTGTCGAATGCT-1
## ENSG00000237491        -0.22484005        -0.10084504
## ENSG00000225880        -0.20852982        -0.09378858
## ENSG00000230368        -0.13946384        -0.06115474
## ENSG00000230699        -0.05330481        -0.02694137
## ENSG00000188976         0.75303608        -0.27198451
## ENSG00000187961        -0.05906678        -0.02977692
## ENSG00000188290        -0.10324123        -0.04420393
## ENSG00000187608        -0.66778017        -0.28420778
## ENSG00000188157        -0.07871818        -0.03690960
## ENSG00000131591        -0.18839118        -0.08509713</code></pre>
<p>Genes that are expressed in fewer than 5 cells are not used and not returned,
so to add vst_out$y as an assay we need to remove the missing genes.</p>
<div class="sourceCode" id="cb751"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb751-1"><a href="NormalisationGSM3872434Top.html#cb751-1"></a><span class="co"># https://github.com/ChristophH/sctransform/issues/27</span></span>
<span id="cb751-2"><a href="NormalisationGSM3872434Top.html#cb751-2"></a>sceOrig &lt;-<span class="st"> </span>sce</span>
<span id="cb751-3"><a href="NormalisationGSM3872434Top.html#cb751-3"></a>sceOrig</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 2853 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb753"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb753-1"><a href="NormalisationGSM3872434Top.html#cb753-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(sce) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(vst_out<span class="op">$</span>y))</span>
<span id="cb753-2"><a href="NormalisationGSM3872434Top.html#cb753-2"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(sceOrig)</span>
<span id="cb753-3"><a href="NormalisationGSM3872434Top.html#cb753-3"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(sceOrig)</span>
<span id="cb753-4"><a href="NormalisationGSM3872434Top.html#cb753-4"></a>new.counts &lt;-<span class="st"> </span><span class="kw">counts</span>(sceOrig)[tmpInd, ]</span>
<span id="cb753-5"><a href="NormalisationGSM3872434Top.html#cb753-5"></a>sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span>new.counts))</span>
<span id="cb753-6"><a href="NormalisationGSM3872434Top.html#cb753-6"></a></span>
<span id="cb753-7"><a href="NormalisationGSM3872434Top.html#cb753-7"></a><span class="co"># reset the column data on the new object</span></span>
<span id="cb753-8"><a href="NormalisationGSM3872434Top.html#cb753-8"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span>cols.meta</span>
<span id="cb753-9"><a href="NormalisationGSM3872434Top.html#cb753-9"></a><span class="kw">rowData</span>(sce) &lt;-<span class="st"> </span>rows.meta[tmpInd, ]</span>
<span id="cb753-10"><a href="NormalisationGSM3872434Top.html#cb753-10"></a><span class="kw">assayNames</span>(sce)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;</code></pre>
<div class="sourceCode" id="cb755"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb755-1"><a href="NormalisationGSM3872434Top.html#cb755-1"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 14214 2853 
## metadata(0):
## assays(1): counts
## rownames(14214): ENSG00000237491 ENSG00000225880 ... ENSG00000278817
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb757"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb757-1"><a href="NormalisationGSM3872434Top.html#cb757-1"></a>vstMat &lt;-<span class="st"> </span><span class="kw">as</span>(vst_out<span class="op">$</span>y[<span class="kw">rownames</span>(sce),], <span class="st">&quot;dgCMatrix&quot;</span>)</span>
<span id="cb757-2"><a href="NormalisationGSM3872434Top.html#cb757-2"></a><span class="kw">all</span>(<span class="kw">colnames</span>(vstMat) <span class="op">==</span><span class="st"> </span>sce<span class="op">$</span>Barcode)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb759"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb759-1"><a href="NormalisationGSM3872434Top.html#cb759-1"></a><span class="kw">colnames</span>(vstMat) &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb759-2"><a href="NormalisationGSM3872434Top.html#cb759-2"></a><span class="kw">assay</span>(sce, <span class="st">&quot;sctrans_norm&quot;</span>) &lt;-<span class="st"> </span>vstMat <span class="co"># as(vst_out$y[rownames(sce),], &quot;dgCMatrix&quot;)</span></span>
<span id="cb759-3"><a href="NormalisationGSM3872434Top.html#cb759-3"></a><span class="co">#assayNames(sce)</span></span></code></pre></div>
</div>
<div id="save-sce-object-1" class="section level3">
<h3><span class="header-section-number">8.2.3</span> Save SCE object</h3>
<div class="sourceCode" id="cb760"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb760-1"><a href="NormalisationGSM3872434Top.html#cb760-1"></a><span class="co"># write to file</span></span>
<span id="cb760-2"><a href="NormalisationGSM3872434Top.html#cb760-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postSct%s.Rds&quot;</span>,</span>
<span id="cb760-3"><a href="NormalisationGSM3872434Top.html#cb760-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb760-4"><a href="NormalisationGSM3872434Top.html#cb760-4"></a><span class="kw">saveRDS</span>(sce, tmpFn)</span></code></pre></div>
</div>
</div>
<div id="visualisation-1" class="section level2">
<h2><span class="header-section-number">8.3</span> Visualisation</h2>
<div id="log-raw-counts-1" class="section level3">
<h3><span class="header-section-number">8.3.1</span> log raw counts</h3>
<div class="sourceCode" id="cb761"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb761-1"><a href="NormalisationGSM3872434Top.html#cb761-1"></a>typeNorm &lt;-<span class="st"> &quot;logRaw&quot;</span></span>
<span id="cb761-2"><a href="NormalisationGSM3872434Top.html#cb761-2"></a></span>
<span id="cb761-3"><a href="NormalisationGSM3872434Top.html#cb761-3"></a><span class="co">#setSuf &lt;- &quot;_5kCellPerSpl&quot;</span></span>
<span id="cb761-4"><a href="NormalisationGSM3872434Top.html#cb761-4"></a><span class="kw">options</span>(<span class="dt">BiocSingularParam.default=</span><span class="kw">IrlbaParam</span>())</span>
<span id="cb761-5"><a href="NormalisationGSM3872434Top.html#cb761-5"></a></span>
<span id="cb761-6"><a href="NormalisationGSM3872434Top.html#cb761-6"></a><span class="kw">assay</span>(sce, <span class="st">&quot;logcounts_raw&quot;</span>) &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">counts</span>(sce) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb761-7"><a href="NormalisationGSM3872434Top.html#cb761-7"></a></span>
<span id="cb761-8"><a href="NormalisationGSM3872434Top.html#cb761-8"></a>tmp &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb761-9"><a href="NormalisationGSM3872434Top.html#cb761-9"></a>  sce[,],</span>
<span id="cb761-10"><a href="NormalisationGSM3872434Top.html#cb761-10"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts_raw&quot;</span></span>
<span id="cb761-11"><a href="NormalisationGSM3872434Top.html#cb761-11"></a>)</span></code></pre></div>
<p>PCA plot for the logRaw counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb762"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb762-1"><a href="NormalisationGSM3872434Top.html#cb762-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(</span>
<span id="cb762-2"><a href="NormalisationGSM3872434Top.html#cb762-2"></a>    tmp,</span>
<span id="cb762-3"><a href="NormalisationGSM3872434Top.html#cb762-3"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb762-4"><a href="NormalisationGSM3872434Top.html#cb762-4"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb762-5"><a href="NormalisationGSM3872434Top.html#cb762-5"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb762-6"><a href="NormalisationGSM3872434Top.html#cb762-6"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;PCA plot: %s&quot;</span>, typeNorm))</span>
<span id="cb762-7"><a href="NormalisationGSM3872434Top.html#cb762-7"></a></span>
<span id="cb762-8"><a href="NormalisationGSM3872434Top.html#cb762-8"></a><span class="co"># write plot to file:</span></span>
<span id="cb762-9"><a href="NormalisationGSM3872434Top.html#cb762-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb762-10"><a href="NormalisationGSM3872434Top.html#cb762-10"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb762-11"><a href="NormalisationGSM3872434Top.html#cb762-11"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb763"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb763-1"><a href="NormalisationGSM3872434Top.html#cb763-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb763-2"><a href="NormalisationGSM3872434Top.html#cb763-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb763-3"><a href="NormalisationGSM3872434Top.html#cb763-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_logRawPca.png" width="1050" /></p>
<div class="sourceCode" id="cb764"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb764-1"><a href="NormalisationGSM3872434Top.html#cb764-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<!--
Cell-wise RLE for the logRaw counts in the GSM3872434 set. Each cell is represented by a box plot showing the inter-quartile range in grey, wiskers colour-coded by Sample.Name and the median as a black circle. 
-->
</div>
<div id="log-cpm-1" class="section level3">
<h3><span class="header-section-number">8.3.2</span> log CPM</h3>
<div class="sourceCode" id="cb765"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb765-1"><a href="NormalisationGSM3872434Top.html#cb765-1"></a>typeNorm &lt;-<span class="st"> &quot;logCpm&quot;</span></span>
<span id="cb765-2"><a href="NormalisationGSM3872434Top.html#cb765-2"></a></span>
<span id="cb765-3"><a href="NormalisationGSM3872434Top.html#cb765-3"></a><span class="kw">assay</span>(sce, <span class="st">&quot;logCpm&quot;</span>) &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">calculateCPM</span>(sce, <span class="dt">size_factors =</span> <span class="ot">NULL</span>) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb765-4"><a href="NormalisationGSM3872434Top.html#cb765-4"></a></span>
<span id="cb765-5"><a href="NormalisationGSM3872434Top.html#cb765-5"></a>logCpmPca &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb765-6"><a href="NormalisationGSM3872434Top.html#cb765-6"></a>  sce[,],</span>
<span id="cb765-7"><a href="NormalisationGSM3872434Top.html#cb765-7"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;logCpm&quot;</span></span>
<span id="cb765-8"><a href="NormalisationGSM3872434Top.html#cb765-8"></a>)</span></code></pre></div>
<p>PCA plot for the logCpm counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb766"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb766-1"><a href="NormalisationGSM3872434Top.html#cb766-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(</span>
<span id="cb766-2"><a href="NormalisationGSM3872434Top.html#cb766-2"></a>    logCpmPca,</span>
<span id="cb766-3"><a href="NormalisationGSM3872434Top.html#cb766-3"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb766-4"><a href="NormalisationGSM3872434Top.html#cb766-4"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb766-5"><a href="NormalisationGSM3872434Top.html#cb766-5"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb766-6"><a href="NormalisationGSM3872434Top.html#cb766-6"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;PCA plot: %s&quot;</span>, typeNorm))</span>
<span id="cb766-7"><a href="NormalisationGSM3872434Top.html#cb766-7"></a></span>
<span id="cb766-8"><a href="NormalisationGSM3872434Top.html#cb766-8"></a><span class="co"># write plot to file:</span></span>
<span id="cb766-9"><a href="NormalisationGSM3872434Top.html#cb766-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb766-10"><a href="NormalisationGSM3872434Top.html#cb766-10"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb766-11"><a href="NormalisationGSM3872434Top.html#cb766-11"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb767"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb767-1"><a href="NormalisationGSM3872434Top.html#cb767-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb767-2"><a href="NormalisationGSM3872434Top.html#cb767-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb767-3"><a href="NormalisationGSM3872434Top.html#cb767-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_logCpmPca.png" width="1050" /></p>
<div class="sourceCode" id="cb768"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb768-1"><a href="NormalisationGSM3872434Top.html#cb768-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<!--
Cell-wise RLE for the logCpm counts in the GSM3872434 set.
-->
</div>
<div id="scran-1" class="section level3">
<h3><span class="header-section-number">8.3.3</span> scran</h3>
<p>Normalised counts are stored in the ‘logcounts’ assay</p>
<div class="sourceCode" id="cb769"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb769-1"><a href="NormalisationGSM3872434Top.html#cb769-1"></a>typeNorm &lt;-<span class="st"> &quot;scran&quot;</span></span>
<span id="cb769-2"><a href="NormalisationGSM3872434Top.html#cb769-2"></a></span>
<span id="cb769-3"><a href="NormalisationGSM3872434Top.html#cb769-3"></a><span class="co"># assay(sce, &quot;logcounts&quot;)</span></span>
<span id="cb769-4"><a href="NormalisationGSM3872434Top.html#cb769-4"></a></span>
<span id="cb769-5"><a href="NormalisationGSM3872434Top.html#cb769-5"></a>scranPca &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb769-6"><a href="NormalisationGSM3872434Top.html#cb769-6"></a>  sceDeconv[,],</span>
<span id="cb769-7"><a href="NormalisationGSM3872434Top.html#cb769-7"></a>  <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span></span>
<span id="cb769-8"><a href="NormalisationGSM3872434Top.html#cb769-8"></a>)</span></code></pre></div>
<p>PCA plot for the ‘scran’ counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb770"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb770-1"><a href="NormalisationGSM3872434Top.html#cb770-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(</span>
<span id="cb770-2"><a href="NormalisationGSM3872434Top.html#cb770-2"></a>    scranPca,</span>
<span id="cb770-3"><a href="NormalisationGSM3872434Top.html#cb770-3"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb770-4"><a href="NormalisationGSM3872434Top.html#cb770-4"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb770-5"><a href="NormalisationGSM3872434Top.html#cb770-5"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb770-6"><a href="NormalisationGSM3872434Top.html#cb770-6"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;PCA plot: %s&quot;</span>, typeNorm))</span>
<span id="cb770-7"><a href="NormalisationGSM3872434Top.html#cb770-7"></a></span>
<span id="cb770-8"><a href="NormalisationGSM3872434Top.html#cb770-8"></a><span class="co"># write plot to file:</span></span>
<span id="cb770-9"><a href="NormalisationGSM3872434Top.html#cb770-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb770-10"><a href="NormalisationGSM3872434Top.html#cb770-10"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb770-11"><a href="NormalisationGSM3872434Top.html#cb770-11"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb771"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb771-1"><a href="NormalisationGSM3872434Top.html#cb771-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb771-2"><a href="NormalisationGSM3872434Top.html#cb771-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb771-3"><a href="NormalisationGSM3872434Top.html#cb771-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_scranPca.png" width="1050" /></p>
<div class="sourceCode" id="cb772"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb772-1"><a href="NormalisationGSM3872434Top.html#cb772-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>TSNE plot for the ‘scran’ counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb773"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb773-1"><a href="NormalisationGSM3872434Top.html#cb773-1"></a>typeNorm &lt;-<span class="st"> &quot;scran&quot;</span></span>
<span id="cb773-2"><a href="NormalisationGSM3872434Top.html#cb773-2"></a></span>
<span id="cb773-3"><a href="NormalisationGSM3872434Top.html#cb773-3"></a><span class="kw">reducedDim</span>(sceDeconv, <span class="st">&quot;TSNE_scran&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb773-4"><a href="NormalisationGSM3872434Top.html#cb773-4"></a>  <span class="kw">runTSNE</span>(sceDeconv, <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>),</span>
<span id="cb773-5"><a href="NormalisationGSM3872434Top.html#cb773-5"></a>  <span class="st">&quot;TSNE&quot;</span></span>
<span id="cb773-6"><a href="NormalisationGSM3872434Top.html#cb773-6"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb774"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb774-1"><a href="NormalisationGSM3872434Top.html#cb774-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(</span>
<span id="cb774-2"><a href="NormalisationGSM3872434Top.html#cb774-2"></a>  sceDeconv,</span>
<span id="cb774-3"><a href="NormalisationGSM3872434Top.html#cb774-3"></a>  <span class="dt">dimred =</span> <span class="st">&quot;TSNE_scran&quot;</span>,</span>
<span id="cb774-4"><a href="NormalisationGSM3872434Top.html#cb774-4"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb774-5"><a href="NormalisationGSM3872434Top.html#cb774-5"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb774-6"><a href="NormalisationGSM3872434Top.html#cb774-6"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb774-7"><a href="NormalisationGSM3872434Top.html#cb774-7"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;TSNE plot: %s&quot;</span>, typeNorm))</span>
<span id="cb774-8"><a href="NormalisationGSM3872434Top.html#cb774-8"></a></span>
<span id="cb774-9"><a href="NormalisationGSM3872434Top.html#cb774-9"></a><span class="co"># write plot to file:</span></span>
<span id="cb774-10"><a href="NormalisationGSM3872434Top.html#cb774-10"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sTsne.png&quot;</span>,</span>
<span id="cb774-11"><a href="NormalisationGSM3872434Top.html#cb774-11"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb774-12"><a href="NormalisationGSM3872434Top.html#cb774-12"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb775"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb775-1"><a href="NormalisationGSM3872434Top.html#cb775-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sTsne.png&quot;</span>,</span>
<span id="cb775-2"><a href="NormalisationGSM3872434Top.html#cb775-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb775-3"><a href="NormalisationGSM3872434Top.html#cb775-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_scranTsne.png" width="1050" /></p>
<div class="sourceCode" id="cb776"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb776-1"><a href="NormalisationGSM3872434Top.html#cb776-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<!--
UMAP plot for the 'scran' counts in the GSM3872434 set.
-->
<!--
Cell-wise RLE for the 'scran' counts in the GSM3872434 set.
-->
</div>
<div id="sctransform-4" class="section level3">
<h3><span class="header-section-number">8.3.4</span> SCTransform</h3>
<div class="sourceCode" id="cb777"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb777-1"><a href="NormalisationGSM3872434Top.html#cb777-1"></a>typeNorm &lt;-<span class="st"> &quot;sctrans&quot;</span></span>
<span id="cb777-2"><a href="NormalisationGSM3872434Top.html#cb777-2"></a></span>
<span id="cb777-3"><a href="NormalisationGSM3872434Top.html#cb777-3"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_sctrans_norm&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb777-4"><a href="NormalisationGSM3872434Top.html#cb777-4"></a>  <span class="kw">runPCA</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;sctrans_norm&quot;</span>),</span>
<span id="cb777-5"><a href="NormalisationGSM3872434Top.html#cb777-5"></a>  <span class="st">&quot;PCA&quot;</span></span>
<span id="cb777-6"><a href="NormalisationGSM3872434Top.html#cb777-6"></a>)</span></code></pre></div>
<p>PCA plot for the ‘sctrans’ counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb778"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb778-1"><a href="NormalisationGSM3872434Top.html#cb778-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(</span>
<span id="cb778-2"><a href="NormalisationGSM3872434Top.html#cb778-2"></a>  sce,</span>
<span id="cb778-3"><a href="NormalisationGSM3872434Top.html#cb778-3"></a>  <span class="dt">dimred =</span> <span class="st">&quot;PCA_sctrans_norm&quot;</span>,</span>
<span id="cb778-4"><a href="NormalisationGSM3872434Top.html#cb778-4"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb778-5"><a href="NormalisationGSM3872434Top.html#cb778-5"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb778-6"><a href="NormalisationGSM3872434Top.html#cb778-6"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb778-7"><a href="NormalisationGSM3872434Top.html#cb778-7"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;PCA plot: %s&quot;</span>, typeNorm))</span>
<span id="cb778-8"><a href="NormalisationGSM3872434Top.html#cb778-8"></a></span>
<span id="cb778-9"><a href="NormalisationGSM3872434Top.html#cb778-9"></a><span class="co"># write plot to file:</span></span>
<span id="cb778-10"><a href="NormalisationGSM3872434Top.html#cb778-10"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb778-11"><a href="NormalisationGSM3872434Top.html#cb778-11"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb778-12"><a href="NormalisationGSM3872434Top.html#cb778-12"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb779"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb779-1"><a href="NormalisationGSM3872434Top.html#cb779-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sPca.png&quot;</span>,</span>
<span id="cb779-2"><a href="NormalisationGSM3872434Top.html#cb779-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb779-3"><a href="NormalisationGSM3872434Top.html#cb779-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_sctransPca.png" width="1050" /></p>
<div class="sourceCode" id="cb780"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb780-1"><a href="NormalisationGSM3872434Top.html#cb780-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>TSNE plot for the sctrans counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb781"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb781-1"><a href="NormalisationGSM3872434Top.html#cb781-1"></a>typeNorm &lt;-<span class="st"> &quot;sctrans&quot;</span></span>
<span id="cb781-2"><a href="NormalisationGSM3872434Top.html#cb781-2"></a></span>
<span id="cb781-3"><a href="NormalisationGSM3872434Top.html#cb781-3"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE_sctrans_norm&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb781-4"><a href="NormalisationGSM3872434Top.html#cb781-4"></a>  <span class="kw">runTSNE</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;sctrans_norm&quot;</span>),</span>
<span id="cb781-5"><a href="NormalisationGSM3872434Top.html#cb781-5"></a>  <span class="st">&quot;TSNE&quot;</span></span>
<span id="cb781-6"><a href="NormalisationGSM3872434Top.html#cb781-6"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb782"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb782-1"><a href="NormalisationGSM3872434Top.html#cb782-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotReducedDim</span>(</span>
<span id="cb782-2"><a href="NormalisationGSM3872434Top.html#cb782-2"></a>  sce,</span>
<span id="cb782-3"><a href="NormalisationGSM3872434Top.html#cb782-3"></a>  <span class="dt">dimred =</span> <span class="st">&quot;TSNE_sctrans_norm&quot;</span>,</span>
<span id="cb782-4"><a href="NormalisationGSM3872434Top.html#cb782-4"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span>,</span>
<span id="cb782-5"><a href="NormalisationGSM3872434Top.html#cb782-5"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb782-6"><a href="NormalisationGSM3872434Top.html#cb782-6"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;source_name&quot;</span></span>
<span id="cb782-7"><a href="NormalisationGSM3872434Top.html#cb782-7"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;TSNE plot: %s&quot;</span>, typeNorm))</span>
<span id="cb782-8"><a href="NormalisationGSM3872434Top.html#cb782-8"></a></span>
<span id="cb782-9"><a href="NormalisationGSM3872434Top.html#cb782-9"></a><span class="co"># write plot to file:</span></span>
<span id="cb782-10"><a href="NormalisationGSM3872434Top.html#cb782-10"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sTsne.png&quot;</span>,</span>
<span id="cb782-11"><a href="NormalisationGSM3872434Top.html#cb782-11"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb782-12"><a href="NormalisationGSM3872434Top.html#cb782-12"></a><span class="kw">ggsave</span>(<span class="dt">filename=</span>tmpFn, <span class="dt">plot=</span>p)</span></code></pre></div>
<div class="sourceCode" id="cb783"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb783-1"><a href="NormalisationGSM3872434Top.html#cb783-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sTsne.png&quot;</span>,</span>
<span id="cb783-2"><a href="NormalisationGSM3872434Top.html#cb783-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb783-3"><a href="NormalisationGSM3872434Top.html#cb783-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_sctransTsne.png" width="1050" /></p>
<div class="sourceCode" id="cb784"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb784-1"><a href="NormalisationGSM3872434Top.html#cb784-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<!--
UMAP plot for the 'sctrans' counts in the GSM3872434 set.
-->
<p>Cell-wise RLE for the sctrans counts in the GSM3872434 set.</p>
<div class="sourceCode" id="cb785"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb785-1"><a href="NormalisationGSM3872434Top.html#cb785-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotRLE</span>(</span>
<span id="cb785-2"><a href="NormalisationGSM3872434Top.html#cb785-2"></a>    sce,</span>
<span id="cb785-3"><a href="NormalisationGSM3872434Top.html#cb785-3"></a>    <span class="dt">exprs_values =</span> <span class="st">&quot;sctrans_norm&quot;</span>,</span>
<span id="cb785-4"><a href="NormalisationGSM3872434Top.html#cb785-4"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb785-5"><a href="NormalisationGSM3872434Top.html#cb785-5"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;RLE plot: %s&quot;</span>, typeNorm))</span>
<span id="cb785-6"><a href="NormalisationGSM3872434Top.html#cb785-6"></a></span>
<span id="cb785-7"><a href="NormalisationGSM3872434Top.html#cb785-7"></a><span class="co"># write plot to file:</span></span>
<span id="cb785-8"><a href="NormalisationGSM3872434Top.html#cb785-8"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s/%s_sce_nz_postQc%s_%sRle.png&quot;</span>,</span>
<span id="cb785-9"><a href="NormalisationGSM3872434Top.html#cb785-9"></a>         projDir, outDirBit, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb785-10"><a href="NormalisationGSM3872434Top.html#cb785-10"></a><span class="kw">print</span>(<span class="st">&quot;DEV&quot;</span>); <span class="kw">print</span>(<span class="kw">getwd</span>()); <span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;DEV&quot;</code></pre>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/ScriptsMkWiC&quot;</code></pre>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_sctransRle.png&quot;</code></pre>
<div class="sourceCode" id="cb789"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb789-1"><a href="NormalisationGSM3872434Top.html#cb789-1"></a><span class="co">##ggsave(filename=tmpFn, plot=p)</span></span></code></pre></div>
<div class="sourceCode" id="cb790"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb790-1"><a href="NormalisationGSM3872434Top.html#cb790-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/%s_sce_nz_postQc%s_%sRle.png&quot;</span>,</span>
<span id="cb790-2"><a href="NormalisationGSM3872434Top.html#cb790-2"></a>                 dirRel, normPlotDirBit, setName, setSuf, typeNorm)</span>
<span id="cb790-3"><a href="NormalisationGSM3872434Top.html#cb790-3"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(tmpFn, <span class="dt">auto_pdf =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="../Plots/Norm/GSM3872434_sce_nz_postQc_GSM3872434_sctransRle.png" width="1050" /></p>
<div class="sourceCode" id="cb791"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb791-1"><a href="NormalisationGSM3872434Top.html#cb791-1"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
</div>
</div>
<div id="session-information-4" class="section level2">
<h2><span class="header-section-number">8.4</span> Session information</h2>
<div class="sourceCode" id="cb792"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb792-1"><a href="NormalisationGSM3872434Top.html#cb792-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] BiocSingular_1.6.0          scuttle_1.0.4              
##  [3] robustbase_0.93-7           mixtools_1.2.0             
##  [5] dplyr_1.0.5                 DT_0.18                    
##  [7] irlba_2.3.3                 biomaRt_2.46.3             
##  [9] Matrix_1.2-18               igraph_1.2.6               
## [11] DropletUtils_1.10.3         scater_1.18.6              
## [13] ggplot2_3.3.3               scran_1.18.7               
## [15] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [17] Biobase_2.50.0              GenomicRanges_1.42.0       
## [19] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [21] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [23] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [25] knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] ggbeeswarm_0.6.0          colorspace_2.0-0         
##   [3] ellipsis_0.3.1            bluster_1.0.0            
##   [5] XVector_0.30.0            BiocNeighbors_1.8.2      
##   [7] rstudioapi_0.13           bit64_4.0.5              
##   [9] AnnotationDbi_1.52.0      fansi_0.4.2              
##  [11] xml2_1.3.2                codetools_0.2-16         
##  [13] splines_4.0.3             R.methodsS3_1.8.1        
##  [15] sparseMatrixStats_1.2.1   cachem_1.0.4             
##  [17] jsonlite_1.7.2            kernlab_0.9-29           
##  [19] dbplyr_2.0.0              png_0.1-7                
##  [21] R.oo_1.24.0               HDF5Array_1.18.1         
##  [23] readr_1.4.0               compiler_4.0.3           
##  [25] httr_1.4.2                dqrng_0.2.1              
##  [27] assertthat_0.2.1          fastmap_1.1.0            
##  [29] limma_3.46.0              htmltools_0.5.1.1        
##  [31] prettyunits_1.1.1         tools_4.0.3              
##  [33] rsvd_1.0.3                gtable_0.3.0             
##  [35] glue_1.4.2                GenomeInfoDbData_1.2.4   
##  [37] rappdirs_0.3.1            Rcpp_1.0.6               
##  [39] jquerylib_0.1.3           vctrs_0.3.7              
##  [41] rhdf5filters_1.2.0        DelayedMatrixStats_1.12.3
##  [43] xfun_0.22                 stringr_1.4.0            
##  [45] beachmat_2.6.4            lifecycle_1.0.0          
##  [47] statmod_1.4.35            XML_3.99-0.5             
##  [49] DEoptimR_1.0-8            edgeR_3.32.1             
##  [51] MASS_7.3-53               zlibbioc_1.36.0          
##  [53] scales_1.1.1              hms_1.0.0                
##  [55] rhdf5_2.34.0              curl_4.3                 
##  [57] yaml_2.2.1                memoise_2.0.0            
##  [59] gridExtra_2.3             sass_0.3.1               
##  [61] segmented_1.3-3           stringi_1.5.3            
##  [63] RSQLite_2.2.6             highr_0.8                
##  [65] BiocParallel_1.24.1       rlang_0.4.10             
##  [67] pkgconfig_2.0.3           bitops_1.0-6             
##  [69] evaluate_0.14             lattice_0.20-41          
##  [71] purrr_0.3.4               Rhdf5lib_1.12.1          
##  [73] htmlwidgets_1.5.3         bit_4.0.4                
##  [75] tidyselect_1.1.0          magrittr_2.0.1           
##  [77] bookdown_0.21             R6_2.5.0                 
##  [79] generics_0.1.0            DelayedArray_0.16.3      
##  [81] DBI_1.1.1                 pillar_1.6.0             
##  [83] withr_2.4.2               survival_3.2-7           
##  [85] RCurl_1.98-1.3            tibble_3.1.1             
##  [87] crayon_1.4.1              utf8_1.2.1               
##  [89] BiocFileCache_1.14.0      rmarkdown_2.7            
##  [91] viridis_0.5.1             progress_1.2.2           
##  [93] locfit_1.5-9.4            grid_4.0.3               
##  [95] blob_1.2.1                digest_0.6.27            
##  [97] R.utils_2.10.1            openssl_1.4.3            
##  [99] munsell_0.5.0             beeswarm_0.2.3           
## [101] viridisLite_0.3.0         vipor_0.4.5              
## [103] bslib_0.2.4               askpass_1.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="normalisation-caron-set.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dimRedForVizTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/normalisation_GSM3872434_simple.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
