<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Data integration - PBMMC | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Data integration - PBMMC | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Data integration - PBMMC | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dimensionality-reduction-for-analysis.html"/>
<link rel="next" href="dataSetIntegration-ETV6-RUNX1Top.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.10" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.11" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-17"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html"><i class="fa fa-check"></i><b>24</b> Extract T-cells from HCA Dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#variance-modeling"><i class="fa fa-check"></i><b>24.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.2" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#data-integration"><i class="fa fa-check"></i><b>24.2</b> Data integration</a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#clustering-1"><i class="fa fa-check"></i><b>24.4</b> Clustering</a></li>
<li class="chapter" data-level="24.5" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#cell-type-classification"><i class="fa fa-check"></i><b>24.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.6" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#extract-t-cells"><i class="fa fa-check"></i><b>24.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.7" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#ackowledgements"><i class="fa fa-check"></i><b>24.7</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.8" data-path="pseudoTime1Top.html"><a href="pseudoTime1Top.html#session-information-18"><i class="fa fa-check"></i><b>24.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html"><i class="fa fa-check"></i><b>25</b> Setting up the data</a><ul>
<li class="chapter" data-level="25.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>25.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="25.2" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>25.2</b> Find temporally expressed genes</a><ul>
<li class="chapter" data-level="25.2.1" data-path="pseudoTime2Top.html"><a href="pseudoTime2Top.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>25.2.1</b> Pseudotime analysis for another HCA sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="exercise-1.html"><a href="exercise-1.html"><i class="fa fa-check"></i><b>26</b> Exercise 1</a><ul>
<li class="chapter" data-level="26.1" data-path="exercise-1.html"><a href="exercise-1.html#ackowledgements-1"><i class="fa fa-check"></i><b>26.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="26.2" data-path="exercise-1.html"><a href="exercise-1.html#session-information-19"><i class="fa fa-check"></i><b>26.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html"><i class="fa fa-check"></i><b>27</b> <strong>Pseudotime Alignment</strong></a><ul>
<li class="chapter" data-level="27.1" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#ackowledgements-2"><i class="fa fa-check"></i><b>27.1</b> Ackowledgements</a></li>
<li class="chapter" data-level="27.2" data-path="pseudoTime3Top.html"><a href="pseudoTime3Top.html#session-information-20"><i class="fa fa-check"></i><b>27.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>28</b> Doublet detection</a><ul>
<li class="chapter" data-level="28.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>28.2</b> Set up analysis</a></li>
<li class="chapter" data-level="28.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-11"><i class="fa fa-check"></i><b>28.3</b> Load data</a></li>
<li class="chapter" data-level="28.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>28.4</b> Overview</a></li>
<li class="chapter" data-level="28.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>28.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="28.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>28.6</b> Further comments</a></li>
<li class="chapter" data-level="28.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-21"><i class="fa fa-check"></i><b>28.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DataIntegrationPbmmcTop" class="section level1">
<h1><span class="header-section-number">Chapter 14</span> Data integration - PBMMC</h1>
<div class="sourceCode" id="cb1028"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1028-1"><a href="DataIntegrationPbmmcTop.html#cb1028-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb1028-2"><a href="DataIntegrationPbmmcTop.html#cb1028-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb1028-3"><a href="DataIntegrationPbmmcTop.html#cb1028-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb1028-4"><a href="DataIntegrationPbmmcTop.html#cb1028-4"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1028-5"><a href="DataIntegrationPbmmcTop.html#cb1028-5"></a></span>
<span id="cb1028-6"><a href="DataIntegrationPbmmcTop.html#cb1028-6"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1028-7"><a href="DataIntegrationPbmmcTop.html#cb1028-7"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1028-8"><a href="DataIntegrationPbmmcTop.html#cb1028-8"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC&quot;</span></span>
<span id="cb1028-9"><a href="DataIntegrationPbmmcTop.html#cb1028-9"></a>    setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb1028-10"><a href="DataIntegrationPbmmcTop.html#cb1028-10"></a>}</span></code></pre></div>
<p>Source: <a href="https://osca.bioconductor.org/integrating-datasets.html">‘Integrating Datasets’</a> chapter in the OSCA book. Its text is reproduced below with few modifications to adapt it to the data set under scrutiny here.</p>
<div id="abbreviations-1" class="section level2">
<h2><span class="header-section-number">14.1</span> Abbreviations</h2>
<ul>
<li>HVG: highly variable genes</li>
<li>MNN: mutual nearest neighbors</li>
<li>PBMMC: peripheral blood mononuclear cell</li>
<li>SCE: SingleCellExperiment</li>
</ul>
</div>
<div id="motivation" class="section level2">
<h2><span class="header-section-number">14.2</span> Motivation</h2>
<p>Large single-cell RNA sequencing (scRNA-seq) projects usually need to generate data across multiple batches due to logistical constraints. However, the processing of different batches is often subject to uncontrollable differences, e.g., changes in operator, differences in reagent quality. This results in systematic differences in the observed expression in cells from different batches, which we refer to as “batch effects”. Batch effects are problematic as they can be major drivers of heterogeneity in the data, masking the relevant biological differences and complicating interpretation of the results.</p>
<p>Computational correction of these effects is critical for eliminating batch-to-batch variation, allowing data across multiple batches to be combined for common downstream analysis. However, existing methods based on linear models (Ritchie et al. 2015; Leek et al. 2012) assume that the composition of cell populations are either known or the same across batches. To overcome these limitations, bespoke methods have been developed for batch correction of single-cell data (Haghverdi et al. 2018; Butler et al. 2018; Lin et al. 2019) that do not require a priori knowledge about the composition of the population. This allows them to be used in workflows for exploratory analyses of scRNA-seq data where such knowledge is usually unavailable.</p>
</div>
<div id="loading-the-data" class="section level2">
<h2><span class="header-section-number">14.3</span> Loading the data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts, and subset 1000 cells per sample.</p>
<div class="sourceCode" id="cb1029"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1029-1"><a href="DataIntegrationPbmmcTop.html#cb1029-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1029-2"><a href="DataIntegrationPbmmcTop.html#cb1029-2"></a></span>
<span id="cb1029-3"><a href="DataIntegrationPbmmcTop.html#cb1029-3"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_allCells.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1031"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1031-1"><a href="DataIntegrationPbmmcTop.html#cb1031-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb1031-2"><a href="DataIntegrationPbmmcTop.html#cb1031-2"></a>{</span>
<span id="cb1031-3"><a href="DataIntegrationPbmmcTop.html#cb1031-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb1031-4"><a href="DataIntegrationPbmmcTop.html#cb1031-4"></a>}</span>
<span id="cb1031-5"><a href="DataIntegrationPbmmcTop.html#cb1031-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1031-6"><a href="DataIntegrationPbmmcTop.html#cb1031-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1033"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1033-1"><a href="DataIntegrationPbmmcTop.html#cb1033-1"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span></span></code></pre></div>
<p>We next subset the data for the PBMMC sample group:</p>
<div class="sourceCode" id="cb1034"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1034-1"><a href="DataIntegrationPbmmcTop.html#cb1034-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb1034-2"><a href="DataIntegrationPbmmcTop.html#cb1034-2"></a>cb_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/CaronBourque2020/SraRunTable.txt&quot;</span>)</span>
<span id="cb1034-3"><a href="DataIntegrationPbmmcTop.html#cb1034-3"></a>cb_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(cb_sampleSheetFn, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb1034-4"><a href="DataIntegrationPbmmcTop.html#cb1034-4"></a>splVec &lt;-<span class="st"> </span>cb_sampleSheet <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(source_name <span class="op">==</span><span class="st"> </span>splSetToGet) <span class="op">%&gt;%</span></span>
<span id="cb1034-5"><a href="DataIntegrationPbmmcTop.html#cb1034-5"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>()</span>
<span id="cb1034-6"><a href="DataIntegrationPbmmcTop.html#cb1034-6"></a></span>
<span id="cb1034-7"><a href="DataIntegrationPbmmcTop.html#cb1034-7"></a>sourceNames &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>source_name)</span>
<span id="cb1034-8"><a href="DataIntegrationPbmmcTop.html#cb1034-8"></a>sceOrig &lt;-<span class="st"> </span>sce</span>
<span id="cb1034-9"><a href="DataIntegrationPbmmcTop.html#cb1034-9"></a>sce &lt;-<span class="st"> </span>sceOrig[,sce<span class="op">$</span>source_name <span class="op">==</span><span class="st"> </span>splSetToGet ]</span>
<span id="cb1034-10"><a href="DataIntegrationPbmmcTop.html#cb1034-10"></a>nbCells &lt;-<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb1034-11"><a href="DataIntegrationPbmmcTop.html#cb1034-11"></a>all.sce &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1034-12"><a href="DataIntegrationPbmmcTop.html#cb1034-12"></a><span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb1034-13"><a href="DataIntegrationPbmmcTop.html#cb1034-13"></a>{</span>
<span id="cb1034-14"><a href="DataIntegrationPbmmcTop.html#cb1034-14"></a>    vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1034-15"><a href="DataIntegrationPbmmcTop.html#cb1034-15"></a><span class="st">        </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1034-16"><a href="DataIntegrationPbmmcTop.html#cb1034-16"></a><span class="st">        </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb1034-17"><a href="DataIntegrationPbmmcTop.html#cb1034-17"></a><span class="st">        </span><span class="kw">sample_n</span>(nbCells) <span class="op">%&gt;%</span></span>
<span id="cb1034-18"><a href="DataIntegrationPbmmcTop.html#cb1034-18"></a><span class="st">        </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1034-19"><a href="DataIntegrationPbmmcTop.html#cb1034-19"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1034-20"><a href="DataIntegrationPbmmcTop.html#cb1034-20"></a>    </span>
<span id="cb1034-21"><a href="DataIntegrationPbmmcTop.html#cb1034-21"></a>    all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb1034-22"><a href="DataIntegrationPbmmcTop.html#cb1034-22"></a>}</span>
<span id="cb1034-23"><a href="DataIntegrationPbmmcTop.html#cb1034-23"></a>nbSpl &lt;-<span class="st"> </span><span class="kw">length</span>(all.sce)</span></code></pre></div>
<p>We then apply the standard workflow to each sample separately:
* normalisation,
* variance modelling
* dimensionality reduction
* clustering</p>
<div class="sourceCode" id="cb1035"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1035-1"><a href="DataIntegrationPbmmcTop.html#cb1035-1"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb1035-2"><a href="DataIntegrationPbmmcTop.html#cb1035-2"></a><span class="co"># use logNormCounts()</span></span>
<span id="cb1035-3"><a href="DataIntegrationPbmmcTop.html#cb1035-3"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, logNormCounts)</span>
<span id="cb1035-4"><a href="DataIntegrationPbmmcTop.html#cb1035-4"></a></span>
<span id="cb1035-5"><a href="DataIntegrationPbmmcTop.html#cb1035-5"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb1035-6"><a href="DataIntegrationPbmmcTop.html#cb1035-6"></a><span class="co"># model varaince with modelGeneVar()</span></span>
<span id="cb1035-7"><a href="DataIntegrationPbmmcTop.html#cb1035-7"></a><span class="co"># find highly variable genes (HVGs) with getTopHVGs()</span></span>
<span id="cb1035-8"><a href="DataIntegrationPbmmcTop.html#cb1035-8"></a><span class="kw">library</span>(scran)</span>
<span id="cb1035-9"><a href="DataIntegrationPbmmcTop.html#cb1035-9"></a>all.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, modelGeneVar)</span>
<span id="cb1035-10"><a href="DataIntegrationPbmmcTop.html#cb1035-10"></a>all.hvgs &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, getTopHVGs, <span class="dt">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb1035-11"><a href="DataIntegrationPbmmcTop.html#cb1035-11"></a></span>
<span id="cb1035-12"><a href="DataIntegrationPbmmcTop.html#cb1035-12"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb1035-13"><a href="DataIntegrationPbmmcTop.html#cb1035-13"></a><span class="co"># use runPCA()</span></span>
<span id="cb1035-14"><a href="DataIntegrationPbmmcTop.html#cb1035-14"></a><span class="co"># then compute embeddings with runTSNE() and runUMAP()</span></span>
<span id="cb1035-15"><a href="DataIntegrationPbmmcTop.html#cb1035-15"></a><span class="kw">library</span>(BiocSingular)</span>
<span id="cb1035-16"><a href="DataIntegrationPbmmcTop.html#cb1035-16"></a><span class="kw">set.seed</span>(<span class="dv">10000</span>)</span>
<span id="cb1035-17"><a href="DataIntegrationPbmmcTop.html#cb1035-17"></a>all.sce &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="dt">FUN=</span>runPCA, <span class="dt">x=</span>all.sce, <span class="dt">subset_row=</span>all.hvgs, </span>
<span id="cb1035-18"><a href="DataIntegrationPbmmcTop.html#cb1035-18"></a>    <span class="dt">MoreArgs=</span><span class="kw">list</span>(<span class="dt">ncomponents=</span><span class="dv">25</span>, <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>()), </span>
<span id="cb1035-19"><a href="DataIntegrationPbmmcTop.html#cb1035-19"></a>    <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb1035-20"><a href="DataIntegrationPbmmcTop.html#cb1035-20"></a></span>
<span id="cb1035-21"><a href="DataIntegrationPbmmcTop.html#cb1035-21"></a><span class="kw">set.seed</span>(<span class="dv">100000</span>)</span>
<span id="cb1035-22"><a href="DataIntegrationPbmmcTop.html#cb1035-22"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, runTSNE, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1035-23"><a href="DataIntegrationPbmmcTop.html#cb1035-23"></a></span>
<span id="cb1035-24"><a href="DataIntegrationPbmmcTop.html#cb1035-24"></a><span class="kw">set.seed</span>(<span class="dv">1000000</span>)</span>
<span id="cb1035-25"><a href="DataIntegrationPbmmcTop.html#cb1035-25"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, runUMAP, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1035-26"><a href="DataIntegrationPbmmcTop.html#cb1035-26"></a></span>
<span id="cb1035-27"><a href="DataIntegrationPbmmcTop.html#cb1035-27"></a><span class="co">#--- clustering ---#</span></span>
<span id="cb1035-28"><a href="DataIntegrationPbmmcTop.html#cb1035-28"></a><span class="co"># cluster each sample separately</span></span>
<span id="cb1035-29"><a href="DataIntegrationPbmmcTop.html#cb1035-29"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="kw">names</span>(all.sce)) {</span>
<span id="cb1035-30"><a href="DataIntegrationPbmmcTop.html#cb1035-30"></a>    g &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(all.sce[[n]], <span class="dt">k=</span><span class="dv">10</span>, <span class="dt">use.dimred=</span><span class="st">&#39;PCA&#39;</span>)</span>
<span id="cb1035-31"><a href="DataIntegrationPbmmcTop.html#cb1035-31"></a>    clust &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(g)<span class="op">$</span>membership</span>
<span id="cb1035-32"><a href="DataIntegrationPbmmcTop.html#cb1035-32"></a>    <span class="kw">colLabels</span>(all.sce[[n]])  &lt;-<span class="st"> </span><span class="kw">factor</span>(clust)</span>
<span id="cb1035-33"><a href="DataIntegrationPbmmcTop.html#cb1035-33"></a>}</span></code></pre></div>
<p>To prepare for the batch correction:</p>
<ul>
<li>We subset all batches to the common “universe” of features. In this case, it is straightforward as both batches use Ensembl gene annotation.</li>
</ul>
<div class="sourceCode" id="cb1036"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1036-1"><a href="DataIntegrationPbmmcTop.html#cb1036-1"></a>allNames &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){<span class="kw">rownames</span>(x)}))</span>
<span id="cb1036-2"><a href="DataIntegrationPbmmcTop.html#cb1036-2"></a>allNamesNb &lt;-<span class="st"> </span><span class="kw">table</span>(allNames)</span>
<span id="cb1036-3"><a href="DataIntegrationPbmmcTop.html#cb1036-3"></a>universe &lt;-<span class="st"> </span><span class="kw">names</span>(allNamesNb)[allNamesNb<span class="op">==</span>nbSpl] </span>
<span id="cb1036-4"><a href="DataIntegrationPbmmcTop.html#cb1036-4"></a><span class="co">#length(universe)</span></span></code></pre></div>
<ul>
<li>The size of this common “universe” of features here is the number of features shared by all 3 samples is: 18431.</li>
</ul>
<div class="sourceCode" id="cb1037"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1037-1"><a href="DataIntegrationPbmmcTop.html#cb1037-1"></a><span class="co"># Subsetting the SingleCellExperiment object.</span></span>
<span id="cb1037-2"><a href="DataIntegrationPbmmcTop.html#cb1037-2"></a>uni.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){x[universe,]})</span>
<span id="cb1037-3"><a href="DataIntegrationPbmmcTop.html#cb1037-3"></a><span class="co"># Also subsetting the variance modelling results, for convenience.</span></span>
<span id="cb1037-4"><a href="DataIntegrationPbmmcTop.html#cb1037-4"></a>uni.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, <span class="cf">function</span>(x){x[universe,]})</span></code></pre></div>
<ul>
<li>We rescale each batch to adjust for differences in sequencing depth between batches. The <code>multiBatchNorm()</code> function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment (SCE) objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.</li>
</ul>
<div class="sourceCode" id="cb1038"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1038-1"><a href="DataIntegrationPbmmcTop.html#cb1038-1"></a><span class="co"># rescale each batch to adjust for differences in sequencing depth between batches</span></span>
<span id="cb1038-2"><a href="DataIntegrationPbmmcTop.html#cb1038-2"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1038-3"><a href="DataIntegrationPbmmcTop.html#cb1038-3"></a>rescaled &lt;-<span class="st"> </span><span class="kw">multiBatchNorm</span>(uni.sce, <span class="dt">batch =</span> <span class="st">&quot;Sample.Name&quot;</span>)</span></code></pre></div>
<ul>
<li>We perform feature selection by averaging the variance components across all batches with the <code>combineVar()</code> function. We compute the average as it is responsive to batch-specific highly-variable genes (HVGs) while still preserving the within-batch ranking of genes.</li>
</ul>
<div class="sourceCode" id="cb1039"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1039-1"><a href="DataIntegrationPbmmcTop.html#cb1039-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1039-2"><a href="DataIntegrationPbmmcTop.html#cb1039-2"></a><span class="co"># compute average variance components across samples</span></span>
<span id="cb1039-3"><a href="DataIntegrationPbmmcTop.html#cb1039-3"></a>combined.dec &lt;-<span class="st"> </span><span class="kw">combineVar</span>(uni.dec[[<span class="dv">1</span>]], uni.dec[[<span class="dv">2</span>]], uni.dec[[<span class="dv">3</span>]])</span>
<span id="cb1039-4"><a href="DataIntegrationPbmmcTop.html#cb1039-4"></a><span class="co"># identify highly variables genes</span></span>
<span id="cb1039-5"><a href="DataIntegrationPbmmcTop.html#cb1039-5"></a><span class="co"># here as those with a positive biological component</span></span>
<span id="cb1039-6"><a href="DataIntegrationPbmmcTop.html#cb1039-6"></a>chosen.hvgs &lt;-<span class="st"> </span>combined.dec<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb1039-7"><a href="DataIntegrationPbmmcTop.html#cb1039-7"></a><span class="co"># sum(chosen.hvgs)</span></span></code></pre></div>
<p>Number of HVGs: 8604.</p>
<p>When integrating datasets of variable composition, it is generally safer to err on the side of including more genes than are used in a single dataset analysis, to ensure that markers are retained for any dataset-specific subpopulations that might be present. For a top X selection, this means using a larger X (say, ~5000), or in this case, we simply take all genes above the trend.</p>
<p>Alternatively, a more forceful approach to feature selection can be used based on marker genes from within-batch comparisons.</p>
</div>
<div id="diagnosing-batch-effects" class="section level2">
<h2><span class="header-section-number">14.4</span> Diagnosing batch effects</h2>
<p>Before we actually perform any correction, it is worth examining whether there is any batch effect in this dataset. We combine the three SCE objects and perform a PCA on the log-expression values for all genes with positive (average) biological components.</p>
<div class="sourceCode" id="cb1040"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1040-1"><a href="DataIntegrationPbmmcTop.html#cb1040-1"></a><span class="co"># Synchronizing the metadata for cbind()ing.</span></span>
<span id="cb1040-2"><a href="DataIntegrationPbmmcTop.html#cb1040-2"></a></span>
<span id="cb1040-3"><a href="DataIntegrationPbmmcTop.html#cb1040-3"></a><span class="co"># before concatenating matrices,</span></span>
<span id="cb1040-4"><a href="DataIntegrationPbmmcTop.html#cb1040-4"></a><span class="co"># check that features are identical in all matrices: </span></span>
<span id="cb1040-5"><a href="DataIntegrationPbmmcTop.html#cb1040-5"></a><span class="kw">identical</span>(<span class="kw">rowData</span>(rescaled[[<span class="dv">1</span>]]), <span class="kw">rowData</span>(rescaled[[<span class="dv">2</span>]]))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb1042"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1042-1"><a href="DataIntegrationPbmmcTop.html#cb1042-1"></a><span class="kw">identical</span>(<span class="kw">rowData</span>(rescaled[[<span class="dv">1</span>]]), <span class="kw">rowData</span>(rescaled[[<span class="dv">3</span>]]))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb1044"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1044-1"><a href="DataIntegrationPbmmcTop.html#cb1044-1"></a><span class="co"># have a &#39;batch&#39; entry in cell meta data:</span></span>
<span id="cb1044-2"><a href="DataIntegrationPbmmcTop.html#cb1044-2"></a><span class="co"># (not strikly necessary as any column can be used)</span></span>
<span id="cb1044-3"><a href="DataIntegrationPbmmcTop.html#cb1044-3"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(rescaled, <span class="cf">function</span>(x){x<span class="op">$</span>batch &lt;-<span class="st"> </span>x<span class="op">$</span>Sample.Name; x})</span>
<span id="cb1044-4"><a href="DataIntegrationPbmmcTop.html#cb1044-4"></a>rescaled &lt;-<span class="st"> </span>rescaled2</span>
<span id="cb1044-5"><a href="DataIntegrationPbmmcTop.html#cb1044-5"></a><span class="kw">rm</span>(rescaled2)</span>
<span id="cb1044-6"><a href="DataIntegrationPbmmcTop.html#cb1044-6"></a></span>
<span id="cb1044-7"><a href="DataIntegrationPbmmcTop.html#cb1044-7"></a><span class="co"># concat matrices:</span></span>
<span id="cb1044-8"><a href="DataIntegrationPbmmcTop.html#cb1044-8"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">cbind</span>(rescaled[[<span class="dv">1</span>]], rescaled[[<span class="dv">2</span>]], rescaled[[<span class="dv">3</span>]])</span>
<span id="cb1044-9"><a href="DataIntegrationPbmmcTop.html#cb1044-9"></a></span>
<span id="cb1044-10"><a href="DataIntegrationPbmmcTop.html#cb1044-10"></a><span class="co"># Perform PCA</span></span>
<span id="cb1044-11"><a href="DataIntegrationPbmmcTop.html#cb1044-11"></a><span class="co"># Using RandomParam() as it is more efficient for file-backed matrices.</span></span>
<span id="cb1044-12"><a href="DataIntegrationPbmmcTop.html#cb1044-12"></a><span class="kw">library</span>(scater)</span>
<span id="cb1044-13"><a href="DataIntegrationPbmmcTop.html#cb1044-13"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb1044-14"><a href="DataIntegrationPbmmcTop.html#cb1044-14"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runPCA</span>(uncorrected,</span>
<span id="cb1044-15"><a href="DataIntegrationPbmmcTop.html#cb1044-15"></a>              <span class="dt">subset_row=</span>chosen.hvgs,</span>
<span id="cb1044-16"><a href="DataIntegrationPbmmcTop.html#cb1044-16"></a>              <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>())</span></code></pre></div>
<p>We use graph-based clustering on the components to obtain a summary of the population structure.</p>
<p>As the samples should be replicates, each cluster should ideally consist of cells from each batch. However, we instead see clusters that are comprised of cells from a single batch. This indicates that cells of the same type are artificially separated due to technical differences between batches.</p>
<div class="sourceCode" id="cb1045"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1045-1"><a href="DataIntegrationPbmmcTop.html#cb1045-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1045-2"><a href="DataIntegrationPbmmcTop.html#cb1045-2"></a><span class="co"># build shared nearest-neighbour graph</span></span>
<span id="cb1045-3"><a href="DataIntegrationPbmmcTop.html#cb1045-3"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(uncorrected, <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1045-4"><a href="DataIntegrationPbmmcTop.html#cb1045-4"></a><span class="co"># identify cluster with the walk trap method</span></span>
<span id="cb1045-5"><a href="DataIntegrationPbmmcTop.html#cb1045-5"></a>clusters &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1045-6"><a href="DataIntegrationPbmmcTop.html#cb1045-6"></a><span class="co"># get number of cells for each {cluster, batch} pair</span></span>
<span id="cb1045-7"><a href="DataIntegrationPbmmcTop.html#cb1045-7"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters, <span class="dt">Batch=</span>uncorrected<span class="op">$</span>batch)</span>
<span id="cb1045-8"><a href="DataIntegrationPbmmcTop.html#cb1045-8"></a>tab</span></code></pre></div>
<pre><code>##        Batch
## Cluster GSM3872442 GSM3872443 GSM3872444
##      1          23         83         95
##      2         134         12         76
##      3         138         42        295
##      4          16         48         11
##      5           2         12        152
##      6         181          1         94
##      7         187         32        191
##      8           6         58          9
##      9           2         74          0
##      10          0         62          0
##      11          1        154          9
##      12          1         39         14
##      13          0         24          0
##      14         31          8         39
##      15         71          0          6
##      16          0         36          0
##      17         11          7          9
##      18        196          0          0
##      19          0         56          0
##      20          0        252          0</code></pre>
<div class="sourceCode" id="cb1047"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1047-1"><a href="DataIntegrationPbmmcTop.html#cb1047-1"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters, <span class="st">&quot;batch&quot;</span>=uncorrected<span class="op">$</span>batch)</span></code></pre></div>
<p>Cluster size and cell contribution by sample:</p>
<div class="sourceCode" id="cb1048"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1048-1"><a href="DataIntegrationPbmmcTop.html#cb1048-1"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb1048-2"><a href="DataIntegrationPbmmcTop.html#cb1048-2"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb1048-3"><a href="DataIntegrationPbmmcTop.html#cb1048-3"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb1048-4"><a href="DataIntegrationPbmmcTop.html#cb1048-4"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb1048-5"><a href="DataIntegrationPbmmcTop.html#cb1048-5"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1048-6"><a href="DataIntegrationPbmmcTop.html#cb1048-6"></a><span class="st">    </span><span class="kw">geom_col</span>() <span class="op">+</span></span>
<span id="cb1048-7"><a href="DataIntegrationPbmmcTop.html#cb1048-7"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;uncorrected, cell numbers&quot;</span>)</span>
<span id="cb1048-8"><a href="DataIntegrationPbmmcTop.html#cb1048-8"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1048-9"><a href="DataIntegrationPbmmcTop.html#cb1048-9"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb1048-10"><a href="DataIntegrationPbmmcTop.html#cb1048-10"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent) <span class="op">+</span></span>
<span id="cb1048-11"><a href="DataIntegrationPbmmcTop.html#cb1048-11"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;uncorrected, proportions&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1049"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1049-1"><a href="DataIntegrationPbmmcTop.html#cb1049-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The strong separation between cells from different batches is consistent with the clustering results.</p>
<div class="sourceCode" id="cb1050"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1050-1"><a href="DataIntegrationPbmmcTop.html#cb1050-1"></a><span class="kw">set.seed</span>(<span class="dv">1111001</span>)</span>
<span id="cb1050-2"><a href="DataIntegrationPbmmcTop.html#cb1050-2"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(uncorrected, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1050-3"><a href="DataIntegrationPbmmcTop.html#cb1050-3"></a><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<p>Of course, the other explanation for batch-specific clusters is that there are cell types that are unique to each batch. The degree of intermingling of cells from different batches is not an effective diagnostic when the batches involved might actually contain unique cell subpopulations (which is not a consideration in the PBMMC dataset, but the same cannot be said in general). If a cluster only contains cells from a single batch, one can always debate whether that is caused by a failure of the correction method or if there is truly a batch-specific subpopulation. For example, do batch-specific metabolic or differentiation states represent distinct subpopulations? Or should they be merged together? We will not attempt to answer this here, only noting that each batch correction algorithm will make different (and possibly inappropriate) decisions on what constitutes “shared” and “unique” populations.</p>
</div>
<div id="linear-regression" class="section level2">
<h2><span class="header-section-number">14.5</span> Linear regression</h2>
<p>Batch effects in bulk RNA sequencing studies are commonly removed with linear regression. This involves fitting a linear model to each gene’s expression profile, setting the undesirable batch term to zero and recomputing the observations sans the batch effect, yielding a set of corrected expression values for downstream analyses. Linear modelling is the basis of the <code>removeBatchEffect()</code> function from the <code>limma</code> package (Ritchie et al. 2015) as well the <code>comBat()</code> function from the <code>sva</code> package (Leek et al. 2012).</p>
<p>To use this approach in a scRNA-seq context, we assume that the composition of cell subpopulations is the same across batches. We also assume that the batch effect is additive, i.e., any batch-induced fold-change in expression is the same across different cell subpopulations for any given gene. These are strong assumptions as batches derived from different individuals will naturally exhibit variation in cell type abundances and expression. Nonetheless, they may be acceptable when dealing with batches that are technical replicates generated from the same population of cells. (In fact, when its assumptions hold, linear regression is the most statistically efficient as it uses information from all cells to compute the common batch vector.) Linear modelling can also accommodate situations where the composition is known a priori by including the cell type as a factor in the linear model, but this situation is even less common.</p>
<p>We use the <code>rescaleBatches()</code> function from the batchelor package to remove the batch effect. This is roughly equivalent to applying a linear regression to the log-expression values per gene, with some adjustments to improve performance and efficiency. For each gene, the mean expression in each batch is scaled down until it is equal to the lowest mean across all batches. We deliberately choose to scale all expression values down as this mitigates differences in variance when batches lie at different positions on the mean-variance trend. (Specifically, the shrinkage effect of the pseudo-count is greater for smaller counts, suppressing any differences in variance across batches.) An additional feature of <code>rescaleBatches()</code> is that it will preserve sparsity in the input matrix for greater efficiency, whereas other methods like <code>removeBatchEffect()</code> will always return a dense matrix.</p>
<div class="sourceCode" id="cb1051"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1051-1"><a href="DataIntegrationPbmmcTop.html#cb1051-1"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1051-2"><a href="DataIntegrationPbmmcTop.html#cb1051-2"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">rescaleBatches</span>(rescaled)</span>
<span id="cb1051-3"><a href="DataIntegrationPbmmcTop.html#cb1051-3"></a>rescaled2</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 3000 
## metadata(0):
## assays(1): corrected
## rownames(18431): ENSG00000000003 ENSG00000000419 ... ENSG00000285486
##   ENSG00000285492
## rowData names(0):
## colnames: NULL
## colData names(1): batch
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>After clustering, we should observe that most clusters consist of mixtures of cells from the replicate batches, consistent with the removal of the batch effect.</p>
<div class="sourceCode" id="cb1053"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1053-1"><a href="DataIntegrationPbmmcTop.html#cb1053-1"></a><span class="kw">set.seed</span>(<span class="dv">1010101010</span>) <span class="co"># To ensure reproducibility of IRLBA.</span></span>
<span id="cb1053-2"><a href="DataIntegrationPbmmcTop.html#cb1053-2"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">runPCA</span>(rescaled2, <span class="dt">subset_row=</span>chosen.hvgs, <span class="dt">exprs_values=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1053-3"><a href="DataIntegrationPbmmcTop.html#cb1053-3"></a></span>
<span id="cb1053-4"><a href="DataIntegrationPbmmcTop.html#cb1053-4"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(rescaled2, <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1053-5"><a href="DataIntegrationPbmmcTop.html#cb1053-5"></a>clusters.resc &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1053-6"><a href="DataIntegrationPbmmcTop.html#cb1053-6"></a>tab.resc &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.resc, <span class="dt">Batch=</span>rescaled2<span class="op">$</span>batch)</span>
<span id="cb1053-7"><a href="DataIntegrationPbmmcTop.html#cb1053-7"></a>tab.resc</span></code></pre></div>
<pre><code>##        Batch
## Cluster GSM3872442 GSM3872443 GSM3872444
##      1          65         94         89
##      2          53         41          1
##      3           0         53          1
##      4          45         25        132
##      5         131          1         71
##      6          97        251        197
##      7           0         86          2
##      8          11          7          9
##      9          77          1         38
##      10          0         37          0
##      11          1         17          7
##      12        186         39        180
##      13          4          0         21
##      14         50          0         27
##      15         14          0          4
##      16          2          7         42
##      17          1         42          0
##      18          0         21          0
##      19          1         30          1
##      20         19          7         15
##      21        172          4         10
##      22         25          8         39
##      23          7          0          8
##      24         10          6          3
##      25         29         80        102
##      26          0        143          1</code></pre>
<!-- Cluster size and cell contribution by sample: -->
<div class="sourceCode" id="cb1055"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1055-1"><a href="DataIntegrationPbmmcTop.html#cb1055-1"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.resc, <span class="st">&quot;batch&quot;</span>=rescaled2<span class="op">$</span>batch)</span></code></pre></div>
<p>Cluster size and cell contribution by sample, with clusters sorted by size:</p>
<div class="sourceCode" id="cb1056"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1056-1"><a href="DataIntegrationPbmmcTop.html#cb1056-1"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb1056-2"><a href="DataIntegrationPbmmcTop.html#cb1056-2"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb1056-3"><a href="DataIntegrationPbmmcTop.html#cb1056-3"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb1056-4"><a href="DataIntegrationPbmmcTop.html#cb1056-4"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb1056-5"><a href="DataIntegrationPbmmcTop.html#cb1056-5"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1056-6"><a href="DataIntegrationPbmmcTop.html#cb1056-6"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb1056-7"><a href="DataIntegrationPbmmcTop.html#cb1056-7"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1056-8"><a href="DataIntegrationPbmmcTop.html#cb1056-8"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb1056-9"><a href="DataIntegrationPbmmcTop.html#cb1056-9"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span></code></pre></div>
<div class="sourceCode" id="cb1057"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1057-1"><a href="DataIntegrationPbmmcTop.html#cb1057-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-90-1.png" width="672" /></p>
<div class="sourceCode" id="cb1058"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1058-1"><a href="DataIntegrationPbmmcTop.html#cb1058-1"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(rescaled2, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb1058-2"><a href="DataIntegrationPbmmcTop.html#cb1058-2"></a>rescaled2<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(rescaled2<span class="op">$</span>batch)</span>
<span id="cb1058-3"><a href="DataIntegrationPbmmcTop.html#cb1058-3"></a><span class="kw">plotTSNE</span>(rescaled2, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-91-1.png" width="672" /></p>
</div>
<div id="performing-mnn-correction" class="section level2">
<h2><span class="header-section-number">14.6</span> Performing MNN correction</h2>
<div id="algorithm-overview" class="section level3">
<h3><span class="header-section-number">14.6.1</span> Algorithm overview</h3>
<p>Consider a cell a in batch A, and identify the cells in batch B that are nearest neighbors to a in the expression space defined by the selected features. Repeat this for a cell b in batch B, identifying its nearest neighbors in A. Mutual nearest neighbors (MNN) are pairs of cells from different batches that belong in each other’s set of nearest neighbors. The reasoning is that MNN pairs represent cells from the same biological state prior to the application of a batch effect - see Haghverdi et al. (2018) for full theoretical details. Thus, the difference between cells in MNN pairs can be used as an estimate of the batch effect, the subtraction of which yields batch-corrected values.</p>
<p>Compared to linear regression, MNN correction does not assume that the population composition is the same or known beforehand. This is because it learns the shared population structure via identification of MNN pairs and uses this information to obtain an appropriate estimate of the batch effect. Instead, the key assumption of MNN-based approaches is that the batch effect is orthogonal to the biology in high-dimensional expression space. Violations reduce the effectiveness and accuracy of the correction, with the most common case arising from variations in the direction of the batch effect between clusters. Nonetheless, the assumption is usually reasonable as a random vector is very likely to be orthogonal in high-dimensional space.</p>
</div>
<div id="application-to-the-data" class="section level3">
<h3><span class="header-section-number">14.6.2</span> Application to the data</h3>
<p>The <code>batchelor</code> package provides an implementation of the MNN approach via the <code>fastMNN()</code> function. (Unlike the MNN method originally described by Haghverdi et al. (2018), the <code>fastMNN()</code> function performs PCA to reduce the dimensions beforehand and speed up the downstream neighbor detection steps.) We apply it to our batches to remove the batch effect across the highly variable genes in <code>chosen.hvgs</code>. To reduce computational work and technical noise, all cells in all batches are projected into the low-dimensional space defined by the top d principal components. Identification of MNNs and calculation of correction vectors are then performed in this low-dimensional space.</p>
<div class="sourceCode" id="cb1059"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1059-1"><a href="DataIntegrationPbmmcTop.html#cb1059-1"></a><span class="co"># Using randomized SVD here, as this is faster than </span></span>
<span id="cb1059-2"><a href="DataIntegrationPbmmcTop.html#cb1059-2"></a><span class="co"># irlba for file-backed matrices.</span></span>
<span id="cb1059-3"><a href="DataIntegrationPbmmcTop.html#cb1059-3"></a><span class="kw">set.seed</span>(<span class="dv">1000101001</span>)</span>
<span id="cb1059-4"><a href="DataIntegrationPbmmcTop.html#cb1059-4"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(rescaled,</span>
<span id="cb1059-5"><a href="DataIntegrationPbmmcTop.html#cb1059-5"></a>           <span class="dt">auto.merge=</span><span class="ot">TRUE</span>,</span>
<span id="cb1059-6"><a href="DataIntegrationPbmmcTop.html#cb1059-6"></a>           <span class="dt">d=</span><span class="dv">50</span>,</span>
<span id="cb1059-7"><a href="DataIntegrationPbmmcTop.html#cb1059-7"></a>           <span class="dt">k=</span><span class="dv">20</span>,</span>
<span id="cb1059-8"><a href="DataIntegrationPbmmcTop.html#cb1059-8"></a>           <span class="dt">subset.row=</span>chosen.hvgs,</span>
<span id="cb1059-9"><a href="DataIntegrationPbmmcTop.html#cb1059-9"></a>           <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>))</span>
<span id="cb1059-10"><a href="DataIntegrationPbmmcTop.html#cb1059-10"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 8604 3000 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(8604): ENSG00000000003 ENSG00000000457 ... ENSG00000285458
##   ENSG00000285476
## rowData names(1): rotation
## colnames: NULL
## colData names(1): batch
## reducedDimNames(1): corrected
## altExpNames(0):</code></pre>
<p>The function returns a SCE object containing corrected values for downstream analyses like clustering or visualization. Each column of <code>mnn.out</code> corresponds to a cell in one of the batches, while each row corresponds to an input gene in chosen.hvgs. The batch field in the column metadata contains a vector specifying the batch of origin of each cell.</p>
<div class="sourceCode" id="cb1061"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1061-1"><a href="DataIntegrationPbmmcTop.html#cb1061-1"></a><span class="kw">print</span>(<span class="kw">head</span>(mnn.out<span class="op">$</span>batch))</span></code></pre></div>
<pre><code>## [1] &quot;GSM3872442&quot; &quot;GSM3872442&quot; &quot;GSM3872442&quot; &quot;GSM3872442&quot; &quot;GSM3872442&quot;
## [6] &quot;GSM3872442&quot;</code></pre>
<p>The corrected matrix in the <code>reducedDims()</code> contains the low-dimensional corrected coordinates for all cells, which we will use in place of the principal components (PCs) in our downstream analyses.</p>
<div class="sourceCode" id="cb1063"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1063-1"><a href="DataIntegrationPbmmcTop.html#cb1063-1"></a><span class="kw">dim</span>(<span class="kw">reducedDim</span>(mnn.out, <span class="st">&quot;corrected&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 3000   50</code></pre>
<p>A reconstructed matrix in the <code>assays()</code> contains the corrected expression values for each gene in each cell, obtained by projecting the low-dimensional coordinates in corrected back into gene expression space. We do not recommend using this for anything other than visualization.</p>
<div class="sourceCode" id="cb1065"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1065-1"><a href="DataIntegrationPbmmcTop.html#cb1065-1"></a><span class="kw">print</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>))</span></code></pre></div>
<pre><code>## &lt;8604 x 3000&gt; matrix of class LowRankMatrix and type &quot;double&quot;:
##                          [,1]          [,2]          [,3] ...       [,2999]
## ENSG00000000003 -9.693175e-05  1.177007e-04 -1.673004e-04   .  2.055355e-04
## ENSG00000000457  3.065863e-04  2.538019e-04  9.322986e-04   . -1.994157e-04
## ENSG00000000938 -9.292932e-04 -2.415617e-03  1.404325e-03   . -2.982403e-03
## ENSG00000001167  1.860307e-04 -3.200527e-04  3.171867e-04   . -3.775970e-04
## ENSG00000001461 -5.302109e-04  9.167783e-04 -9.285246e-04   . -2.790526e-04
##             ...             .             .             .   .             .
## ENSG00000285437 -1.471949e-03 -2.137265e-03 -3.891309e-03   . -2.256149e-03
## ENSG00000285444  7.318343e-05 -1.387095e-04 -3.779627e-04   . -3.413362e-04
## ENSG00000285447 -1.167881e-04 -5.224395e-05 -2.945133e-04   .  1.850214e-04
## ENSG00000285458 -2.076950e-05  1.076314e-05  1.810845e-04   .  9.591474e-05
## ENSG00000285476 -5.965583e-05  6.250964e-05 -1.062484e-04   .  7.862604e-05
##                       [,3000]
## ENSG00000000003 -2.514390e-04
## ENSG00000000457  1.265962e-04
## ENSG00000000938 -1.385492e-03
## ENSG00000001167 -5.522563e-05
## ENSG00000001461  3.146520e-04
##             ...             .
## ENSG00000285437  1.192830e-03
## ENSG00000285444  1.041148e-04
## ENSG00000285447  2.203375e-04
## ENSG00000285458  1.915324e-04
## ENSG00000285476 -7.949445e-05</code></pre>
<p>The most relevant parameter for tuning <code>fastMNN()</code> is <code>k</code>, which specifies the number of nearest neighbors to consider when defining MNN pairs. This can be interpreted as the minimum anticipated frequency of any shared cell type or state in each batch. Increasing <code>k</code> will generally result in more aggressive merging as the algorithm is more generous in matching subpopulations across batches. It can occasionally be desirable to increase <code>k</code> if one clearly sees that the same cell types are not being adequately merged across batches.</p>
<!--
See Chapter 32 for an example of a more complex fastMNN() merge involving several human pancreas datasets generated by different authors on different patients with different technologies.
-->
</div>
<div id="correction-diagnostics" class="section level3">
<h3><span class="header-section-number">14.6.3</span> Correction diagnostics</h3>
<p>We cluster on the low-dimensional corrected coordinates to obtain a partitioning of the cells that serves as a proxy for the population structure. If the batch effect is successfully corrected, clusters corresponding to shared cell types or states should contain cells from multiple batches. We see that most clusters contain contributions from each batch after correction, consistent with our expectation that the batches are replicates of each other.</p>
<div class="sourceCode" id="cb1067"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1067-1"><a href="DataIntegrationPbmmcTop.html#cb1067-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1067-2"><a href="DataIntegrationPbmmcTop.html#cb1067-2"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(mnn.out, <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1067-3"><a href="DataIntegrationPbmmcTop.html#cb1067-3"></a>clusters.mnn &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb1067-4"><a href="DataIntegrationPbmmcTop.html#cb1067-4"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.mnn, <span class="dt">Batch=</span>mnn.out<span class="op">$</span>batch)</span>
<span id="cb1067-5"><a href="DataIntegrationPbmmcTop.html#cb1067-5"></a>tab.mnn</span></code></pre></div>
<pre><code>##        Batch
## Cluster GSM3872442 GSM3872443 GSM3872444
##      1          16         46         11
##      2         176         38         96
##      3          54         48        144
##      4          77         23         36
##      5          68         32         48
##      6           1         62         14
##      7           3         89          7
##      8           0        139          2
##      9          37         86         65
##      10          9          6          3
##      11        297        267        230
##      12        191         35        185
##      13         11          7          9
##      14         25          7         39
##      15          6         33          9
##      16         29         82        102</code></pre>
<p>Cluster size and cell contribution by sample, with clusters sorted by size:</p>
<div class="sourceCode" id="cb1069"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1069-1"><a href="DataIntegrationPbmmcTop.html#cb1069-1"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.mnn, <span class="st">&quot;batch&quot;</span>=mnn.out<span class="op">$</span>batch)</span>
<span id="cb1069-2"><a href="DataIntegrationPbmmcTop.html#cb1069-2"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb1069-3"><a href="DataIntegrationPbmmcTop.html#cb1069-3"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb1069-4"><a href="DataIntegrationPbmmcTop.html#cb1069-4"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb1069-5"><a href="DataIntegrationPbmmcTop.html#cb1069-5"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb1069-6"><a href="DataIntegrationPbmmcTop.html#cb1069-6"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb1069-7"><a href="DataIntegrationPbmmcTop.html#cb1069-7"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1069-8"><a href="DataIntegrationPbmmcTop.html#cb1069-8"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb1069-9"><a href="DataIntegrationPbmmcTop.html#cb1069-9"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb1069-10"><a href="DataIntegrationPbmmcTop.html#cb1069-10"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb1069-11"><a href="DataIntegrationPbmmcTop.html#cb1069-11"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span></code></pre></div>
<div class="sourceCode" id="cb1070"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1070-1"><a href="DataIntegrationPbmmcTop.html#cb1070-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-98-1.png" width="672" /></p>
<p>We can also compute the variation in the log-abundances to rank the clusters with the greatest variability in their proportional abundances across batches. We can then focus on batch-specific clusters that may be indicative of incomplete batch correction. Obviously, though, this diagnostic is subject to interpretation as the same outcome can be caused by batch-specific populations; some prior knowledge about the biological context is necessary to distinguish between these two possibilities.</p>
<div class="sourceCode" id="cb1071"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1071-1"><a href="DataIntegrationPbmmcTop.html#cb1071-1"></a><span class="co"># Avoid minor difficulties with the &#39;table&#39; class.</span></span>
<span id="cb1071-2"><a href="DataIntegrationPbmmcTop.html#cb1071-2"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">unclass</span>(tab.mnn)</span>
<span id="cb1071-3"><a href="DataIntegrationPbmmcTop.html#cb1071-3"></a></span>
<span id="cb1071-4"><a href="DataIntegrationPbmmcTop.html#cb1071-4"></a><span class="co"># Using a large pseudo.count to avoid unnecessarily</span></span>
<span id="cb1071-5"><a href="DataIntegrationPbmmcTop.html#cb1071-5"></a><span class="co"># large variances when the counts are low.</span></span>
<span id="cb1071-6"><a href="DataIntegrationPbmmcTop.html#cb1071-6"></a>norm &lt;-<span class="st"> </span><span class="kw">normalizeCounts</span>(tab.mnn, <span class="dt">pseudo_count=</span><span class="dv">10</span>)</span>
<span id="cb1071-7"><a href="DataIntegrationPbmmcTop.html#cb1071-7"></a></span>
<span id="cb1071-8"><a href="DataIntegrationPbmmcTop.html#cb1071-8"></a><span class="co"># Ranking clusters by the largest variances.</span></span>
<span id="cb1071-9"><a href="DataIntegrationPbmmcTop.html#cb1071-9"></a>rv &lt;-<span class="st"> </span><span class="kw">rowVars</span>(norm) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">round</span>(<span class="dv">2</span>)</span>
<span id="cb1071-10"><a href="DataIntegrationPbmmcTop.html#cb1071-10"></a><span class="kw">DataFrame</span>(<span class="dt">Batch=</span>tab.mnn, <span class="dt">var=</span>rv)[<span class="kw">order</span>(rv, <span class="dt">decreasing=</span><span class="ot">TRUE</span>),]</span></code></pre></div>
<pre><code>## DataFrame with 16 rows and 4 columns
##     Batch.GSM3872442 Batch.GSM3872443 Batch.GSM3872444       var
##            &lt;integer&gt;        &lt;integer&gt;        &lt;integer&gt; &lt;numeric&gt;
## 8                  0              139                2      4.74
## 7                  3               89                7      2.53
## 6                  1               62               14      1.85
## 12               191               35              185      1.52
## 2                176               38               96      0.96
## ...              ...              ...              ...       ...
## 9                 37               86               65      0.27
## 5                 68               32               48      0.20
## 10                 9                6                3      0.08
## 11               297              267              230      0.03
## 13                11                7                9      0.02</code></pre>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The presence of visual clusters containing cells from all batches provides a comforting illusion that the correction was successful.</p>
<div class="sourceCode" id="cb1073"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1073-1"><a href="DataIntegrationPbmmcTop.html#cb1073-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb1073-2"><a href="DataIntegrationPbmmcTop.html#cb1073-2"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb1073-3"><a href="DataIntegrationPbmmcTop.html#cb1073-3"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1073-4"><a href="DataIntegrationPbmmcTop.html#cb1073-4"></a></span>
<span id="cb1073-5"><a href="DataIntegrationPbmmcTop.html#cb1073-5"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb1073-6"><a href="DataIntegrationPbmmcTop.html#cb1073-6"></a><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-100-1.png" width="672" /></p>
<div class="sourceCode" id="cb1074"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1074-1"><a href="DataIntegrationPbmmcTop.html#cb1074-1"></a><span class="co">#mnn.out$type &lt;- gsub(&quot;_[1-4]&quot;,&quot;&quot;,mnn.out$batch)</span></span>
<span id="cb1074-2"><a href="DataIntegrationPbmmcTop.html#cb1074-2"></a><span class="co">#p &lt;- plotTSNE(mnn.out, colour_by=&quot;batch&quot;, shape_by=&quot;type&quot;)</span></span>
<span id="cb1074-3"><a href="DataIntegrationPbmmcTop.html#cb1074-3"></a><span class="co">#p + facet_wrap(. ~ mnn.out$type)</span></span></code></pre></div>
<p>For <code>fastMNN()</code>, one useful diagnostic is the proportion of variance within each batch that is lost during MNN correction. Specifically, this refers to the within-batch variance that is removed during orthogonalization with respect to the average correction vector at each merge step. This is returned via the <code>lost.var</code> field in the metadata of <code>mnn.out</code>, which contains a matrix of the variance lost in each batch (column) at each merge step (row).</p>
<div class="sourceCode" id="cb1075"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1075-1"><a href="DataIntegrationPbmmcTop.html#cb1075-1"></a><span class="kw">round</span>(<span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var,<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##      GSM3872442 GSM3872443 GSM3872444
## [1,]       0.03       0.00       0.04
## [2,]       0.02       0.17       0.03</code></pre>
<p>Large proportions of lost variance (&gt;10%) suggest that correction is removing genuine biological heterogeneity. This would occur due to violations of the assumption of orthogonality between the batch effect and the biological subspace (Haghverdi et al. 2018). In this case, the proportion of lost variance is small, indicating that non-orthogonality is not a major concern.</p>
<p>The following t-SNE shows clusters identified:</p>
<div class="sourceCode" id="cb1077"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1077-1"><a href="DataIntegrationPbmmcTop.html#cb1077-1"></a>mnn.out<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, clusters.mnn)</span>
<span id="cb1077-2"><a href="DataIntegrationPbmmcTop.html#cb1077-2"></a><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;cluster&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-102-1.png" width="672" /></p>
<p>The following t-SNE plots show expression levels of known cell type marker genes.</p>
<div class="sourceCode" id="cb1078"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1078-1"><a href="DataIntegrationPbmmcTop.html#cb1078-1"></a>genesToShow &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb1078-2"><a href="DataIntegrationPbmmcTop.html#cb1078-2"></a>         <span class="st">&quot;CD79A&quot;</span>, <span class="co"># CD79A   B ***</span></span>
<span id="cb1078-3"><a href="DataIntegrationPbmmcTop.html#cb1078-3"></a>         <span class="st">&quot;CST3&quot;</span>, <span class="co"># CST3     monocytes ***</span></span>
<span id="cb1078-4"><a href="DataIntegrationPbmmcTop.html#cb1078-4"></a>         <span class="st">&quot;CD3D&quot;</span>, <span class="co"># CD3D      T cells ***</span></span>
<span id="cb1078-5"><a href="DataIntegrationPbmmcTop.html#cb1078-5"></a>         <span class="st">&quot;HBA1&quot;</span> <span class="co"># HBA1   erythrocytes ***</span></span>
<span id="cb1078-6"><a href="DataIntegrationPbmmcTop.html#cb1078-6"></a>        )</span>
<span id="cb1078-7"><a href="DataIntegrationPbmmcTop.html#cb1078-7"></a></span>
<span id="cb1078-8"><a href="DataIntegrationPbmmcTop.html#cb1078-8"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(uncorrected)<span class="op">$</span>Symbol <span class="op">%in%</span><span class="st"> </span>genesToShow)</span>
<span id="cb1078-9"><a href="DataIntegrationPbmmcTop.html#cb1078-9"></a>ensToShow &lt;-<span class="st"> </span><span class="kw">rowData</span>(uncorrected)<span class="op">$</span>ensembl_gene_id[tmpInd]</span></code></pre></div>
<p>B cells:</p>
<div class="sourceCode" id="cb1079"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1079-1"><a href="DataIntegrationPbmmcTop.html#cb1079-1"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">1</span>]</span>
<span id="cb1079-2"><a href="DataIntegrationPbmmcTop.html#cb1079-2"></a>    p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1079-3"><a href="DataIntegrationPbmmcTop.html#cb1079-3"></a>    p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1079-4"><a href="DataIntegrationPbmmcTop.html#cb1079-4"></a>            <span class="kw">paste</span>(<span class="st">&quot;B cells&quot;</span>, genex,</span>
<span id="cb1079-5"><a href="DataIntegrationPbmmcTop.html#cb1079-5"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1079-6"><a href="DataIntegrationPbmmcTop.html#cb1079-6"></a>        )</span>
<span id="cb1079-7"><a href="DataIntegrationPbmmcTop.html#cb1079-7"></a>    <span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-104-1.png" width="672" /></p>
<p>T cells:</p>
<div class="sourceCode" id="cb1080"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1080-1"><a href="DataIntegrationPbmmcTop.html#cb1080-1"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">3</span>]</span>
<span id="cb1080-2"><a href="DataIntegrationPbmmcTop.html#cb1080-2"></a>    p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1080-3"><a href="DataIntegrationPbmmcTop.html#cb1080-3"></a>    p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1080-4"><a href="DataIntegrationPbmmcTop.html#cb1080-4"></a>            <span class="kw">paste</span>(<span class="st">&quot;T cells&quot;</span>, genex,</span>
<span id="cb1080-5"><a href="DataIntegrationPbmmcTop.html#cb1080-5"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1080-6"><a href="DataIntegrationPbmmcTop.html#cb1080-6"></a>        )</span>
<span id="cb1080-7"><a href="DataIntegrationPbmmcTop.html#cb1080-7"></a>    <span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-105-1.png" width="672" /></p>
<p>monocytes:</p>
<div class="sourceCode" id="cb1081"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1081-1"><a href="DataIntegrationPbmmcTop.html#cb1081-1"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">2</span>]</span>
<span id="cb1081-2"><a href="DataIntegrationPbmmcTop.html#cb1081-2"></a>    p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1081-3"><a href="DataIntegrationPbmmcTop.html#cb1081-3"></a>    p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1081-4"><a href="DataIntegrationPbmmcTop.html#cb1081-4"></a>            <span class="kw">paste</span>(<span class="st">&quot;monocytes&quot;</span>, genex,</span>
<span id="cb1081-5"><a href="DataIntegrationPbmmcTop.html#cb1081-5"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1081-6"><a href="DataIntegrationPbmmcTop.html#cb1081-6"></a>        )</span>
<span id="cb1081-7"><a href="DataIntegrationPbmmcTop.html#cb1081-7"></a>    <span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-106-1.png" width="672" /></p>
<p>erythrocytes:</p>
<div class="sourceCode" id="cb1082"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1082-1"><a href="DataIntegrationPbmmcTop.html#cb1082-1"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">4</span>]</span>
<span id="cb1082-2"><a href="DataIntegrationPbmmcTop.html#cb1082-2"></a>    p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1082-3"><a href="DataIntegrationPbmmcTop.html#cb1082-3"></a>    p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1082-4"><a href="DataIntegrationPbmmcTop.html#cb1082-4"></a>            <span class="kw">paste</span>(<span class="st">&quot;erythrocytes&quot;</span>, genex,</span>
<span id="cb1082-5"><a href="DataIntegrationPbmmcTop.html#cb1082-5"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1082-6"><a href="DataIntegrationPbmmcTop.html#cb1082-6"></a>        )</span>
<span id="cb1082-7"><a href="DataIntegrationPbmmcTop.html#cb1082-7"></a>    <span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-107-1.png" width="672" /></p>
<p>Other genes (exercise)</p>
<div class="sourceCode" id="cb1083"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1083-1"><a href="DataIntegrationPbmmcTop.html#cb1083-1"></a>genesToShow2 &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb1083-2"><a href="DataIntegrationPbmmcTop.html#cb1083-2"></a>         <span class="st">&quot;IL7R&quot;</span>, <span class="co"># IL7R, CCR7   Naive CD4+ T</span></span>
<span id="cb1083-3"><a href="DataIntegrationPbmmcTop.html#cb1083-3"></a>         <span class="st">&quot;CCR7&quot;</span>, <span class="co"># IL7R, CCR7   Naive CD4+ T</span></span>
<span id="cb1083-4"><a href="DataIntegrationPbmmcTop.html#cb1083-4"></a>         <span class="st">&quot;S100A4&quot;</span>, <span class="co"># IL7R, S100A4   Memory CD4+</span></span>
<span id="cb1083-5"><a href="DataIntegrationPbmmcTop.html#cb1083-5"></a>         <span class="st">&quot;CD14&quot;</span>, <span class="co"># CD14, LYZ    CD14+ Mono</span></span>
<span id="cb1083-6"><a href="DataIntegrationPbmmcTop.html#cb1083-6"></a>         <span class="st">&quot;LYZ&quot;</span>, <span class="co"># CD14, LYZ     CD14+ Mono</span></span>
<span id="cb1083-7"><a href="DataIntegrationPbmmcTop.html#cb1083-7"></a>         <span class="st">&quot;MS4A1&quot;</span>, <span class="co"># MS4A1   B</span></span>
<span id="cb1083-8"><a href="DataIntegrationPbmmcTop.html#cb1083-8"></a>         <span class="st">&quot;CD8A&quot;</span>, <span class="co"># CD8A     CD8+ T</span></span>
<span id="cb1083-9"><a href="DataIntegrationPbmmcTop.html#cb1083-9"></a>         <span class="st">&quot;FCGR3A&quot;</span>, <span class="co"># FCGR3A, MS4A7  FCGR3A+ Mono</span></span>
<span id="cb1083-10"><a href="DataIntegrationPbmmcTop.html#cb1083-10"></a>         <span class="st">&quot;MS4A7&quot;</span>, <span class="co"># FCGR3A, MS4A7   FCGR3A+ Mono</span></span>
<span id="cb1083-11"><a href="DataIntegrationPbmmcTop.html#cb1083-11"></a>         <span class="st">&quot;GNLY&quot;</span>, <span class="co"># GNLY, NKG7   NK</span></span>
<span id="cb1083-12"><a href="DataIntegrationPbmmcTop.html#cb1083-12"></a>         <span class="st">&quot;NKG7&quot;</span>, <span class="co"># GNLY, NKG7   NK</span></span>
<span id="cb1083-13"><a href="DataIntegrationPbmmcTop.html#cb1083-13"></a>         <span class="st">&quot;FCER1A&quot;</span>, <span class="co"># DC</span></span>
<span id="cb1083-14"><a href="DataIntegrationPbmmcTop.html#cb1083-14"></a>         <span class="st">&quot;CST3&quot;</span>, <span class="co"># DC</span></span>
<span id="cb1083-15"><a href="DataIntegrationPbmmcTop.html#cb1083-15"></a>         <span class="st">&quot;PPBP&quot;</span> <span class="co"># Platelet</span></span>
<span id="cb1083-16"><a href="DataIntegrationPbmmcTop.html#cb1083-16"></a>        )</span></code></pre></div>
<div class="sourceCode" id="cb1084"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1084-1"><a href="DataIntegrationPbmmcTop.html#cb1084-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(uncorrected)<span class="op">$</span>Symbol <span class="op">%in%</span><span class="st"> </span>genesToShow2)</span>
<span id="cb1084-2"><a href="DataIntegrationPbmmcTop.html#cb1084-2"></a>ensToShow &lt;-<span class="st"> </span><span class="kw">rowData</span>(uncorrected)<span class="op">$</span>ensembl_gene_id[tmpInd]</span></code></pre></div>
<div class="sourceCode" id="cb1085"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1085-1"><a href="DataIntegrationPbmmcTop.html#cb1085-1"></a><span class="cf">for</span> (genex <span class="cf">in</span> ensToShow)</span>
<span id="cb1085-2"><a href="DataIntegrationPbmmcTop.html#cb1085-2"></a>{</span>
<span id="cb1085-3"><a href="DataIntegrationPbmmcTop.html#cb1085-3"></a>    p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1085-4"><a href="DataIntegrationPbmmcTop.html#cb1085-4"></a>    p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(genex, <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>]))</span>
<span id="cb1085-5"><a href="DataIntegrationPbmmcTop.html#cb1085-5"></a>    <span class="kw">print</span>(p)</span>
<span id="cb1085-6"><a href="DataIntegrationPbmmcTop.html#cb1085-6"></a>}</span></code></pre></div>
</div>
</div>
<div id="preserving-biological-heterogeneity" class="section level2">
<h2><span class="header-section-number">14.7</span> Preserving biological heterogeneity</h2>
<div id="comparison-to-within-batch-clusters" class="section level3">
<h3><span class="header-section-number">14.7.1</span> Comparison to within-batch clusters</h3>
<p>Another useful diagnostic check is to compare the clustering within each batch to the clustering of the merged data. Accurate data integration should preserve variance within each batch as there should be nothing to remove between cells in the same batch. This check complements the previously mentioned diagnostics that only focus on the removal of differences between batches. Specifically, it protects us against cases where the correction method simply aggregates all cells together, which would achieve perfect mixing but also discard the biological heterogeneity of interest.</p>
<p>Ideally, we should see a many-to-1 mapping where the across-batch clustering is nested inside the within-batch clusterings. This indicates that any within-batch structure was preserved after correction while acknowledging that greater resolution is possible with more cells. In practice, more discrepancies can be expected even when the correction is perfect, due to the existence of closely related clusters that were arbitrarily separated in the within-batch clustering. As a general rule, we can be satisfied with the correction if the vast majority of entries are zero, though this may depend on whether specific clusters of interest are gained or lost.</p>
<p>One heatmap is generated for each dataset, where each entry is colored according to the number of cells with each pair of labels (before and after correction), on the log10 scale with a pseudocounts +10 for a smoother color transition (so a minimum value of log10(0+10) == 1).</p>
<div class="sourceCode" id="cb1086"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1086-1"><a href="DataIntegrationPbmmcTop.html#cb1086-1"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1086-2"><a href="DataIntegrationPbmmcTop.html#cb1086-2"></a></span>
<span id="cb1086-3"><a href="DataIntegrationPbmmcTop.html#cb1086-3"></a><span class="co"># have common color scale:</span></span>
<span id="cb1086-4"><a href="DataIntegrationPbmmcTop.html#cb1086-4"></a><span class="co"># for the number of cells on the log10 scale</span></span>
<span id="cb1086-5"><a href="DataIntegrationPbmmcTop.html#cb1086-5"></a>breakVec =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dt">by =</span> <span class="fl">0.2</span>)</span>
<span id="cb1086-6"><a href="DataIntegrationPbmmcTop.html#cb1086-6"></a>colPal &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">7</span>, <span class="dt">name =</span> <span class="st">&quot;RdYlBu&quot;</span>)))(<span class="kw">length</span>(breakVec))</span>
<span id="cb1086-7"><a href="DataIntegrationPbmmcTop.html#cb1086-7"></a><span class="co"># Defines the vector of colors for the legend (it has to be of the same length as breakVec)</span></span>
<span id="cb1086-8"><a href="DataIntegrationPbmmcTop.html#cb1086-8"></a></span>
<span id="cb1086-9"><a href="DataIntegrationPbmmcTop.html#cb1086-9"></a></span>
<span id="cb1086-10"><a href="DataIntegrationPbmmcTop.html#cb1086-10"></a><span class="co"># For the first batch (adding +10 for a smoother color transition</span></span>
<span id="cb1086-11"><a href="DataIntegrationPbmmcTop.html#cb1086-11"></a><span class="co"># from zero to non-zero counts for any given matrix entry).</span></span>
<span id="cb1086-12"><a href="DataIntegrationPbmmcTop.html#cb1086-12"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1086-13"><a href="DataIntegrationPbmmcTop.html#cb1086-13"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb1086-14"><a href="DataIntegrationPbmmcTop.html#cb1086-14"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled[[splIdx]])))</span>
<span id="cb1086-15"><a href="DataIntegrationPbmmcTop.html#cb1086-15"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>), <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1086-16"><a href="DataIntegrationPbmmcTop.html#cb1086-16"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison (number of cells, log10 scale)&quot;</span>, splVec[splIdx]),</span>
<span id="cb1086-17"><a href="DataIntegrationPbmmcTop.html#cb1086-17"></a>        <span class="dt">silent=</span><span class="ot">TRUE</span>,</span>
<span id="cb1086-18"><a href="DataIntegrationPbmmcTop.html#cb1086-18"></a>        <span class="dt">col=</span>colPal)</span>
<span id="cb1086-19"><a href="DataIntegrationPbmmcTop.html#cb1086-19"></a></span>
<span id="cb1086-20"><a href="DataIntegrationPbmmcTop.html#cb1086-20"></a><span class="kw">library</span>(clustree)</span>
<span id="cb1086-21"><a href="DataIntegrationPbmmcTop.html#cb1086-21"></a>combined &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">cl.1=</span><span class="kw">colLabels</span>(rescaled[[splIdx]]),</span>
<span id="cb1086-22"><a href="DataIntegrationPbmmcTop.html#cb1086-22"></a>          <span class="dt">cl.2=</span>clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1086-23"><a href="DataIntegrationPbmmcTop.html#cb1086-23"></a><span class="kw">clustree</span>(combined, <span class="dt">prefix=</span><span class="st">&quot;cl.&quot;</span>, <span class="dt">edge_arrow=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
<div class="sourceCode" id="cb1087"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1087-1"><a href="DataIntegrationPbmmcTop.html#cb1087-1"></a><span class="co"># For the second batch.</span></span>
<span id="cb1087-2"><a href="DataIntegrationPbmmcTop.html#cb1087-2"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1087-3"><a href="DataIntegrationPbmmcTop.html#cb1087-3"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb1087-4"><a href="DataIntegrationPbmmcTop.html#cb1087-4"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled[[splIdx]])))</span>
<span id="cb1087-5"><a href="DataIntegrationPbmmcTop.html#cb1087-5"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>), <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1087-6"><a href="DataIntegrationPbmmcTop.html#cb1087-6"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison (number of cells, log10 scale)&quot;</span>, splVec[splIdx]),</span>
<span id="cb1087-7"><a href="DataIntegrationPbmmcTop.html#cb1087-7"></a>        <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1087-8"><a href="DataIntegrationPbmmcTop.html#cb1087-8"></a></span>
<span id="cb1087-9"><a href="DataIntegrationPbmmcTop.html#cb1087-9"></a><span class="kw">library</span>(clustree)</span>
<span id="cb1087-10"><a href="DataIntegrationPbmmcTop.html#cb1087-10"></a>combined &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">cl.1=</span><span class="kw">colLabels</span>(rescaled[[splIdx]]),</span>
<span id="cb1087-11"><a href="DataIntegrationPbmmcTop.html#cb1087-11"></a>          <span class="dt">cl.2=</span>clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1087-12"><a href="DataIntegrationPbmmcTop.html#cb1087-12"></a><span class="kw">clustree</span>(combined, <span class="dt">prefix=</span><span class="st">&quot;cl.&quot;</span>, <span class="dt">edge_arrow=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-112-1.png" width="672" /></p>
<div class="sourceCode" id="cb1088"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1088-1"><a href="DataIntegrationPbmmcTop.html#cb1088-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-113-1.png" width="672" /></p>
<p>Another evaluation approach is to compute the <strong>coassignment probabilities</strong>, i.e. the probability that cells from two within-batch clusters are clustered together in the across-batch clustering. High probabilities off the diagonal indicate that within-batch clusters are merged in the across-batch analysis. We would generally expect low off-diagonal probabilities for most pairs of clusters, though this may not be reasonably possible if the within-batch clusters were poorly separated in the first place.</p>
<p>Coassignment probabilities for the within-batch clusters, based on coassignment of cells in the across-batch clusters obtained after MNN correction. One heatmap is generated for each sample, where each entry is colored according to the coassignment probability between each pair of within-batch clusters:</p>
<div class="sourceCode" id="cb1089"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1089-1"><a href="DataIntegrationPbmmcTop.html#cb1089-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb1089-2"><a href="DataIntegrationPbmmcTop.html#cb1089-2"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1089-3"><a href="DataIntegrationPbmmcTop.html#cb1089-3"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]]), clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1089-4"><a href="DataIntegrationPbmmcTop.html#cb1089-4"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1089-5"><a href="DataIntegrationPbmmcTop.html#cb1089-5"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1089-6"><a href="DataIntegrationPbmmcTop.html#cb1089-6"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[splIdx]),</span>
<span id="cb1089-7"><a href="DataIntegrationPbmmcTop.html#cb1089-7"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1089-8"><a href="DataIntegrationPbmmcTop.html#cb1089-8"></a></span>
<span id="cb1089-9"><a href="DataIntegrationPbmmcTop.html#cb1089-9"></a><span class="co"># For the second batch.</span></span>
<span id="cb1089-10"><a href="DataIntegrationPbmmcTop.html#cb1089-10"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1089-11"><a href="DataIntegrationPbmmcTop.html#cb1089-11"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]]), clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1089-12"><a href="DataIntegrationPbmmcTop.html#cb1089-12"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1089-13"><a href="DataIntegrationPbmmcTop.html#cb1089-13"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1089-14"><a href="DataIntegrationPbmmcTop.html#cb1089-14"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[splIdx]),</span>
<span id="cb1089-15"><a href="DataIntegrationPbmmcTop.html#cb1089-15"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1089-16"><a href="DataIntegrationPbmmcTop.html#cb1089-16"></a></span>
<span id="cb1089-17"><a href="DataIntegrationPbmmcTop.html#cb1089-17"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-114-1.png" width="672" /></p>
<p>Finally, we can summarize the agreement between clusterings by computing the <strong>Rand index</strong>. This provides a simple metric that we can use to assess the preservation of variation by different correction methods. Larger rand indices (i.e., closer to 1) are more desirable, though this must be balanced against the ability of each method to actually remove the batch effect.</p>
<div class="sourceCode" id="cb1090"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1090-1"><a href="DataIntegrationPbmmcTop.html#cb1090-1"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(fossil))</span>
<span id="cb1090-2"><a href="DataIntegrationPbmmcTop.html#cb1090-2"></a>ri1 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span><span class="st">&quot;GSM3872443&quot;</span>]),</span>
<span id="cb1090-3"><a href="DataIntegrationPbmmcTop.html#cb1090-3"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[<span class="dv">1</span>]])))</span>
<span id="cb1090-4"><a href="DataIntegrationPbmmcTop.html#cb1090-4"></a><span class="co">#ri1</span></span></code></pre></div>
<div class="sourceCode" id="cb1091"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1091-1"><a href="DataIntegrationPbmmcTop.html#cb1091-1"></a><span class="co"># same as above but with `bluster`</span></span>
<span id="cb1091-2"><a href="DataIntegrationPbmmcTop.html#cb1091-2"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1091-3"><a href="DataIntegrationPbmmcTop.html#cb1091-3"></a></span>
<span id="cb1091-4"><a href="DataIntegrationPbmmcTop.html#cb1091-4"></a><span class="kw">library</span>(bluster)</span>
<span id="cb1091-5"><a href="DataIntegrationPbmmcTop.html#cb1091-5"></a>ri1 &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(<span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb1091-6"><a href="DataIntegrationPbmmcTop.html#cb1091-6"></a>             <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1091-7"><a href="DataIntegrationPbmmcTop.html#cb1091-7"></a>             <span class="dt">mode=</span><span class="st">&quot;index&quot;</span>)</span></code></pre></div>
<p>Rand index for GSM3872442: 0.64.</p>
<div class="sourceCode" id="cb1092"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1092-1"><a href="DataIntegrationPbmmcTop.html#cb1092-1"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1092-2"><a href="DataIntegrationPbmmcTop.html#cb1092-2"></a></span>
<span id="cb1092-3"><a href="DataIntegrationPbmmcTop.html#cb1092-3"></a>ri2 &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(<span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb1092-4"><a href="DataIntegrationPbmmcTop.html#cb1092-4"></a>             <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1092-5"><a href="DataIntegrationPbmmcTop.html#cb1092-5"></a>             <span class="dt">mode=</span><span class="st">&quot;index&quot;</span>)</span>
<span id="cb1092-6"><a href="DataIntegrationPbmmcTop.html#cb1092-6"></a><span class="co">#ri2</span></span></code></pre></div>
<p>Rand index for GSM3872443: 0.91</p>
<p>We can also break down the adjusted Rand index (ARI) into per-cluster ratios for more detailed diagnostics. For example, we could see low ratios off the diagonal if distinct clusters in the within-batch clustering were incorrectly aggregated in the merged clustering. Conversely, we might see low ratios on the diagonal if the correction inflated or introduced spurious heterogeneity inside a within-batch cluster.</p>
<div class="sourceCode" id="cb1093"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1093-1"><a href="DataIntegrationPbmmcTop.html#cb1093-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb1093-2"><a href="DataIntegrationPbmmcTop.html#cb1093-2"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1093-3"><a href="DataIntegrationPbmmcTop.html#cb1093-3"></a></span>
<span id="cb1093-4"><a href="DataIntegrationPbmmcTop.html#cb1093-4"></a>tab &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb1093-5"><a href="DataIntegrationPbmmcTop.html#cb1093-5"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1093-6"><a href="DataIntegrationPbmmcTop.html#cb1093-6"></a>  <span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1093-7"><a href="DataIntegrationPbmmcTop.html#cb1093-7"></a>)</span>
<span id="cb1093-8"><a href="DataIntegrationPbmmcTop.html#cb1093-8"></a></span>
<span id="cb1093-9"><a href="DataIntegrationPbmmcTop.html#cb1093-9"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1093-10"><a href="DataIntegrationPbmmcTop.html#cb1093-10"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1093-11"><a href="DataIntegrationPbmmcTop.html#cb1093-11"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s adjusted Rand index&quot;</span>, splVec[splIdx]),</span>
<span id="cb1093-12"><a href="DataIntegrationPbmmcTop.html#cb1093-12"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1093-13"><a href="DataIntegrationPbmmcTop.html#cb1093-13"></a></span>
<span id="cb1093-14"><a href="DataIntegrationPbmmcTop.html#cb1093-14"></a><span class="co"># For the second batch.</span></span>
<span id="cb1093-15"><a href="DataIntegrationPbmmcTop.html#cb1093-15"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1093-16"><a href="DataIntegrationPbmmcTop.html#cb1093-16"></a></span>
<span id="cb1093-17"><a href="DataIntegrationPbmmcTop.html#cb1093-17"></a>tab &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb1093-18"><a href="DataIntegrationPbmmcTop.html#cb1093-18"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1093-19"><a href="DataIntegrationPbmmcTop.html#cb1093-19"></a>  <span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1093-20"><a href="DataIntegrationPbmmcTop.html#cb1093-20"></a>)</span>
<span id="cb1093-21"><a href="DataIntegrationPbmmcTop.html#cb1093-21"></a></span>
<span id="cb1093-22"><a href="DataIntegrationPbmmcTop.html#cb1093-22"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1093-23"><a href="DataIntegrationPbmmcTop.html#cb1093-23"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1093-24"><a href="DataIntegrationPbmmcTop.html#cb1093-24"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s adjusted Rand index&quot;</span>, splVec[splIdx]),</span>
<span id="cb1093-25"><a href="DataIntegrationPbmmcTop.html#cb1093-25"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1093-26"><a href="DataIntegrationPbmmcTop.html#cb1093-26"></a></span>
<span id="cb1093-27"><a href="DataIntegrationPbmmcTop.html#cb1093-27"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<div class="sourceCode" id="cb1094"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1094-1"><a href="DataIntegrationPbmmcTop.html#cb1094-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb1094-2"><a href="DataIntegrationPbmmcTop.html#cb1094-2"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1094-3"><a href="DataIntegrationPbmmcTop.html#cb1094-3"></a></span>
<span id="cb1094-4"><a href="DataIntegrationPbmmcTop.html#cb1094-4"></a>tab &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb1094-5"><a href="DataIntegrationPbmmcTop.html#cb1094-5"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1094-6"><a href="DataIntegrationPbmmcTop.html#cb1094-6"></a>  <span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1094-7"><a href="DataIntegrationPbmmcTop.html#cb1094-7"></a>)</span>
<span id="cb1094-8"><a href="DataIntegrationPbmmcTop.html#cb1094-8"></a>tab1 &lt;-<span class="st"> </span>tab</span>
<span id="cb1094-9"><a href="DataIntegrationPbmmcTop.html#cb1094-9"></a></span>
<span id="cb1094-10"><a href="DataIntegrationPbmmcTop.html#cb1094-10"></a><span class="co">#heat1 &lt;- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,</span></span>
<span id="cb1094-11"><a href="DataIntegrationPbmmcTop.html#cb1094-11"></a><span class="co">#    col=rev(viridis::magma(100)),</span></span>
<span id="cb1094-12"><a href="DataIntegrationPbmmcTop.html#cb1094-12"></a><span class="co">#    main=sprintf(&quot;%s adjusted Rand index&quot;, splVec[splIdx]),</span></span>
<span id="cb1094-13"><a href="DataIntegrationPbmmcTop.html#cb1094-13"></a><span class="co">#    silent=TRUE)</span></span>
<span id="cb1094-14"><a href="DataIntegrationPbmmcTop.html#cb1094-14"></a></span>
<span id="cb1094-15"><a href="DataIntegrationPbmmcTop.html#cb1094-15"></a><span class="co"># For the second batch.</span></span>
<span id="cb1094-16"><a href="DataIntegrationPbmmcTop.html#cb1094-16"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1094-17"><a href="DataIntegrationPbmmcTop.html#cb1094-17"></a></span>
<span id="cb1094-18"><a href="DataIntegrationPbmmcTop.html#cb1094-18"></a>tab &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb1094-19"><a href="DataIntegrationPbmmcTop.html#cb1094-19"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled[[splIdx]])),</span>
<span id="cb1094-20"><a href="DataIntegrationPbmmcTop.html#cb1094-20"></a>  <span class="kw">as.integer</span>(clusters.mnn[rescaled2<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb1094-21"><a href="DataIntegrationPbmmcTop.html#cb1094-21"></a>)</span>
<span id="cb1094-22"><a href="DataIntegrationPbmmcTop.html#cb1094-22"></a>tab2 &lt;-<span class="st"> </span>tab</span>
<span id="cb1094-23"><a href="DataIntegrationPbmmcTop.html#cb1094-23"></a><span class="kw">rm</span>(tab)</span>
<span id="cb1094-24"><a href="DataIntegrationPbmmcTop.html#cb1094-24"></a></span>
<span id="cb1094-25"><a href="DataIntegrationPbmmcTop.html#cb1094-25"></a><span class="co">#heat2 &lt;- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,</span></span>
<span id="cb1094-26"><a href="DataIntegrationPbmmcTop.html#cb1094-26"></a><span class="co">#    col=rev(viridis::magma(100)),</span></span>
<span id="cb1094-27"><a href="DataIntegrationPbmmcTop.html#cb1094-27"></a><span class="co">#    main=sprintf(&quot;%s adjusted Rand index&quot;, splVec[splIdx]),</span></span>
<span id="cb1094-28"><a href="DataIntegrationPbmmcTop.html#cb1094-28"></a><span class="co">#    silent=TRUE)</span></span>
<span id="cb1094-29"><a href="DataIntegrationPbmmcTop.html#cb1094-29"></a></span>
<span id="cb1094-30"><a href="DataIntegrationPbmmcTop.html#cb1094-30"></a><span class="co">#gridExtra::grid.arrange(heat1[[4]], heat2[[4]])</span></span>
<span id="cb1094-31"><a href="DataIntegrationPbmmcTop.html#cb1094-31"></a></span>
<span id="cb1094-32"><a href="DataIntegrationPbmmcTop.html#cb1094-32"></a>  <span class="co">## make breaks from combined range</span></span>
<span id="cb1094-33"><a href="DataIntegrationPbmmcTop.html#cb1094-33"></a>Breaks &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="kw">min</span>(<span class="kw">c</span>(tab1, tab2), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb1094-34"><a href="DataIntegrationPbmmcTop.html#cb1094-34"></a>              <span class="kw">max</span>(<span class="kw">c</span>(tab1, tab2), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb1094-35"><a href="DataIntegrationPbmmcTop.html#cb1094-35"></a>              <span class="dt">length =</span> <span class="dv">100</span>)</span>
<span id="cb1094-36"><a href="DataIntegrationPbmmcTop.html#cb1094-36"></a></span>
<span id="cb1094-37"><a href="DataIntegrationPbmmcTop.html#cb1094-37"></a>splIdx &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb1094-38"><a href="DataIntegrationPbmmcTop.html#cb1094-38"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab1, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1094-39"><a href="DataIntegrationPbmmcTop.html#cb1094-39"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1094-40"><a href="DataIntegrationPbmmcTop.html#cb1094-40"></a>    <span class="dt">breaks=</span>Breaks,</span>
<span id="cb1094-41"><a href="DataIntegrationPbmmcTop.html#cb1094-41"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s adjusted Rand index&quot;</span>, splVec[splIdx]),</span>
<span id="cb1094-42"><a href="DataIntegrationPbmmcTop.html#cb1094-42"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1094-43"><a href="DataIntegrationPbmmcTop.html#cb1094-43"></a></span>
<span id="cb1094-44"><a href="DataIntegrationPbmmcTop.html#cb1094-44"></a>splIdx &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1094-45"><a href="DataIntegrationPbmmcTop.html#cb1094-45"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab2, <span class="dt">cluster_row=</span><span class="ot">FALSE</span>, <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb1094-46"><a href="DataIntegrationPbmmcTop.html#cb1094-46"></a>    <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb1094-47"><a href="DataIntegrationPbmmcTop.html#cb1094-47"></a>    <span class="dt">breaks=</span>Breaks,</span>
<span id="cb1094-48"><a href="DataIntegrationPbmmcTop.html#cb1094-48"></a>    <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s adjusted Rand index&quot;</span>, splVec[splIdx]),</span>
<span id="cb1094-49"><a href="DataIntegrationPbmmcTop.html#cb1094-49"></a>    <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb1094-50"><a href="DataIntegrationPbmmcTop.html#cb1094-50"></a></span>
<span id="cb1094-51"><a href="DataIntegrationPbmmcTop.html#cb1094-51"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-119-1.png" width="672" /></p>
</div>
<div id="encouraging-consistency-with-marker-genes" class="section level3">
<h3><span class="header-section-number">14.7.2</span> Encouraging consistency with marker genes</h3>
<p>In some situations, we will already have performed within-batch analyses to characterize salient aspects of population heterogeneity. This is not uncommon when merging datasets from different sources where each dataset has already been analyzed, annotated and interpreted separately. It is subsequently desirable for the integration procedure to retain these “known interesting” aspects of each dataset in the merged dataset. We can encourage this outcome by using the marker genes within each dataset as our selected feature set for <code>fastMNN()</code> and related methods. This focuses on the relevant heterogeneity and represents a semi-supervised approach.</p>
<p>To illustrate, we apply this strategy to our datasets. We identify the top marker genes from pairwise Wilcoxon ranked sum tests between every pair of clusters within each batch, analogous to the method used by <code>[SingleR](https://www.bioconductor.org/packages/release/bioc/html/SingleR.html)</code>. In this case, we use the top 10 marker genes but any value can be used depending on the acceptable trade-off between signal and noise (and speed). We then take the union across all comparisons in all batches and use that in place of our HVG set in <code>fastMNN()</code>.</p>
<div class="sourceCode" id="cb1095"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1095-1"><a href="DataIntegrationPbmmcTop.html#cb1095-1"></a><span class="co"># Recall that groups for marker detection</span></span>
<span id="cb1095-2"><a href="DataIntegrationPbmmcTop.html#cb1095-2"></a><span class="co"># are automatically defined from &#39;colLabels()&#39;. </span></span>
<span id="cb1095-3"><a href="DataIntegrationPbmmcTop.html#cb1095-3"></a>stats1 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled[[<span class="dv">1</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1095-4"><a href="DataIntegrationPbmmcTop.html#cb1095-4"></a>markers1 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats1[[<span class="dv">1</span>]], stats1[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1095-5"><a href="DataIntegrationPbmmcTop.html#cb1095-5"></a></span>
<span id="cb1095-6"><a href="DataIntegrationPbmmcTop.html#cb1095-6"></a>stats2 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled[[<span class="dv">2</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1095-7"><a href="DataIntegrationPbmmcTop.html#cb1095-7"></a>markers2 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats2[[<span class="dv">1</span>]], stats2[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1095-8"><a href="DataIntegrationPbmmcTop.html#cb1095-8"></a></span>
<span id="cb1095-9"><a href="DataIntegrationPbmmcTop.html#cb1095-9"></a>stats3 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled[[<span class="dv">3</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1095-10"><a href="DataIntegrationPbmmcTop.html#cb1095-10"></a>markers3 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats3[[<span class="dv">1</span>]], stats3[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb1095-11"><a href="DataIntegrationPbmmcTop.html#cb1095-11"></a></span>
<span id="cb1095-12"><a href="DataIntegrationPbmmcTop.html#cb1095-12"></a>marker.set &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">unlist</span>(<span class="kw">c</span>(<span class="kw">unlist</span>(markers1), <span class="kw">unlist</span>(markers2), <span class="kw">unlist</span>(markers3))))</span>
<span id="cb1095-13"><a href="DataIntegrationPbmmcTop.html#cb1095-13"></a><span class="kw">length</span>(marker.set) <span class="co"># getting the total number of genes selected in this manner.</span></span></code></pre></div>
<pre><code>## [1] 435</code></pre>
<p>Total number of genes selected in this manner: 435.</p>
<div class="sourceCode" id="cb1097"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1097-1"><a href="DataIntegrationPbmmcTop.html#cb1097-1"></a><span class="kw">set.seed</span>(<span class="dv">1000110</span>)</span>
<span id="cb1097-2"><a href="DataIntegrationPbmmcTop.html#cb1097-2"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(rescaled, <span class="dt">subset.row=</span>marker.set,</span>
<span id="cb1097-3"><a href="DataIntegrationPbmmcTop.html#cb1097-3"></a>    <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>))</span>
<span id="cb1097-4"><a href="DataIntegrationPbmmcTop.html#cb1097-4"></a><span class="co"># compute t-SNE:</span></span>
<span id="cb1097-5"><a href="DataIntegrationPbmmcTop.html#cb1097-5"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out2, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span></code></pre></div>
<p>We can also visualize the corrected coordinates using a t-SNE plot:</p>
<div class="sourceCode" id="cb1098"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1098-1"><a href="DataIntegrationPbmmcTop.html#cb1098-1"></a><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-122-1.png" width="672" /></p>
<p>A quick inspection indicates that the original within-batch structure is indeed preserved in the corrected data. This highlights the utility of a marker-based feature set for integrating datasets that have already been characterized separately in a manner that preserves existing interpretations of each dataset. We note that some within-batch clusters have merged, most likely due to the lack of robust separation in the first place, though this may also be treated as a diagnostic on the appropriateness of the integration depending on the context.</p>
<div class="sourceCode" id="cb1099"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1099-1"><a href="DataIntegrationPbmmcTop.html#cb1099-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(</span>
<span id="cb1099-2"><a href="DataIntegrationPbmmcTop.html#cb1099-2"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span><span class="st">&quot;GSM3872442&quot;</span>], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled[[<span class="dv">1</span>]]))),</span>
<span id="cb1099-3"><a href="DataIntegrationPbmmcTop.html#cb1099-3"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span><span class="st">&quot;GSM3872443&quot;</span>], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled[[<span class="dv">2</span>]]))),</span>
<span id="cb1099-4"><a href="DataIntegrationPbmmcTop.html#cb1099-4"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span><span class="st">&quot;GSM3872444&quot;</span>], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled[[<span class="dv">3</span>]]))),</span>
<span id="cb1099-5"><a href="DataIntegrationPbmmcTop.html#cb1099-5"></a>    <span class="dt">ncol=</span><span class="dv">2</span></span>
<span id="cb1099-6"><a href="DataIntegrationPbmmcTop.html#cb1099-6"></a>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-123-1.png" width="672" /></p>
<div class="sourceCode" id="cb1100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1100-1"><a href="DataIntegrationPbmmcTop.html#cb1100-1"></a><span class="co"># B cells</span></span>
<span id="cb1100-2"><a href="DataIntegrationPbmmcTop.html#cb1100-2"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">1</span>]</span>
<span id="cb1100-3"><a href="DataIntegrationPbmmcTop.html#cb1100-3"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1100-4"><a href="DataIntegrationPbmmcTop.html#cb1100-4"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1100-5"><a href="DataIntegrationPbmmcTop.html#cb1100-5"></a>        <span class="kw">paste</span>(<span class="st">&quot;B cells&quot;</span>, genex,</span>
<span id="cb1100-6"><a href="DataIntegrationPbmmcTop.html#cb1100-6"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1100-7"><a href="DataIntegrationPbmmcTop.html#cb1100-7"></a>            )</span>
<span id="cb1100-8"><a href="DataIntegrationPbmmcTop.html#cb1100-8"></a><span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-124-1.png" width="672" /></p>
<div class="sourceCode" id="cb1101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1101-1"><a href="DataIntegrationPbmmcTop.html#cb1101-1"></a><span class="co"># cannot find &#39;ENSG00000115523&#39;</span></span>
<span id="cb1101-2"><a href="DataIntegrationPbmmcTop.html#cb1101-2"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: use another gene?</span></span>
<span id="cb1101-3"><a href="DataIntegrationPbmmcTop.html#cb1101-3"></a></span>
<span id="cb1101-4"><a href="DataIntegrationPbmmcTop.html#cb1101-4"></a><span class="co"># erythrocytes:</span></span>
<span id="cb1101-5"><a href="DataIntegrationPbmmcTop.html#cb1101-5"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">4</span>]</span>
<span id="cb1101-6"><a href="DataIntegrationPbmmcTop.html#cb1101-6"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>)</span>
<span id="cb1101-7"><a href="DataIntegrationPbmmcTop.html#cb1101-7"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb1101-8"><a href="DataIntegrationPbmmcTop.html#cb1101-8"></a>            <span class="kw">paste</span>(<span class="st">&quot;erythrocytes&quot;</span>, genex,</span>
<span id="cb1101-9"><a href="DataIntegrationPbmmcTop.html#cb1101-9"></a>            <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb1101-10"><a href="DataIntegrationPbmmcTop.html#cb1101-10"></a>            )</span>
<span id="cb1101-11"><a href="DataIntegrationPbmmcTop.html#cb1101-11"></a><span class="kw">print</span>(p)</span></code></pre></div>
<p>We suggest limiting the use of per-gene corrected values to visualization, e.g., when coloring points on a t-SNE plot by per-cell expression. This can be more aesthetically pleasing than uncorrected expression values that may contain large shifts on the colour scale between cells in different batches. Use of the corrected values in any quantitative procedure should be treated with caution, and should be backed up by similar results from an analysis on the uncorrected values.</p>
<p>B cells:</p>
<div class="sourceCode" id="cb1102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1102-1"><a href="DataIntegrationPbmmcTop.html#cb1102-1"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(uncorrected,</span>
<span id="cb1102-2"><a href="DataIntegrationPbmmcTop.html#cb1102-2"></a>                     clusters.mnn,</span>
<span id="cb1102-3"><a href="DataIntegrationPbmmcTop.html#cb1102-3"></a>                     <span class="dt">block=</span>uncorrected<span class="op">$</span>batch,</span>
<span id="cb1102-4"><a href="DataIntegrationPbmmcTop.html#cb1102-4"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1102-5"><a href="DataIntegrationPbmmcTop.html#cb1102-5"></a>                     <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb1102-6"><a href="DataIntegrationPbmmcTop.html#cb1102-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),</span>
<span id="cb1102-7"><a href="DataIntegrationPbmmcTop.html#cb1102-7"></a>                                                   <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<!-- B cells: -->
<p>Erythrocytes:</p>
<div class="sourceCode" id="cb1103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1103-1"><a href="DataIntegrationPbmmcTop.html#cb1103-1"></a>demo &lt;-<span class="st"> </span>m.out[[<span class="st">&quot;11&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span>data.frame <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(FDR)</span>
<span id="cb1103-2"><a href="DataIntegrationPbmmcTop.html#cb1103-2"></a>demo[ensToShow[<span class="dv">4</span>],]</span></code></pre></div>
<pre><code>##                 ensembl_gene_id Symbol Top p.value FDR summary.logFC   logFC.1
## ENSG00000206172 ENSG00000206172   HBA1 160       1   1     0.1461401 -2.638346
##                   logFC.2    logFC.3   logFC.4     logFC.5   logFC.6   logFC.7
## ENSG00000206172 0.1126643 -0.3672066 0.0768022 -0.04321104 -5.561042 -5.754414
##                   logFC.8    logFC.9  logFC.10   logFC.12   logFC.13   logFC.14
## ENSG00000206172 -5.734749 -0.2643027 0.1461401 -0.1729262 -0.3015218 0.04455094
##                  logFC.15   logFC.16
## ENSG00000206172 -5.313446 -0.1326186</code></pre>
<div class="sourceCode" id="cb1105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1105-1"><a href="DataIntegrationPbmmcTop.html#cb1105-1"></a><span class="kw">as.data.frame</span>(demo[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,<span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]) </span></code></pre></div>
<pre><code>##                 Symbol Top       p.value           FDR
## ENSG00000161970  RPL26   1 1.381128e-154 2.545558e-150
## ENSG00000105193  RPS16   2 2.109919e-146 1.944396e-142
## ENSG00000171858  RPS21   3 1.151263e-143 7.072978e-140
## ENSG00000071082  RPL31   4 4.726090e-142 2.177664e-138
## ENSG00000100316   RPL3   5 1.558616e-138 5.745372e-135
## ENSG00000166441 RPL27A   6 7.723015e-136 2.372381e-132
## ENSG00000145425  RPS3A   4 1.365755e-135 3.596032e-132
## ENSG00000142937   RPS8   1 3.979392e-135 9.168022e-132
## ENSG00000140988   RPS2   9 1.793490e-134 3.672868e-131
## ENSG00000198755 RPL10A   9 6.611618e-134 1.196538e-130
## ENSG00000131469  RPL27  11 7.141185e-134 1.196538e-130
## ENSG00000110700  RPS13   6 1.027873e-133 1.578727e-130
## ENSG00000197958  RPL12   3 3.133807e-133 4.443015e-130
## ENSG00000143947 RPS27A   9 8.860447e-133 1.166478e-129
## ENSG00000156508 EEF1A1  15 4.768881e-129 5.859684e-126
## ENSG00000198242 RPL23A   7 1.515175e-128 1.745387e-125
## ENSG00000198918  RPL39  16 1.104657e-126 1.197643e-123
## ENSG00000231500  RPS18   9 1.350047e-126 1.382373e-123
## ENSG00000109475  RPL34   5 5.450229e-126 5.287009e-123
## ENSG00000167526  RPL13   4 8.175809e-126 7.534417e-123</code></pre>
<p>Expression level for the top gene, ENSG00000161970:</p>
<div class="sourceCode" id="cb1107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1107-1"><a href="DataIntegrationPbmmcTop.html#cb1107-1"></a>geneEnsId &lt;-<span class="st"> </span><span class="kw">rownames</span>(demo)[<span class="dv">1</span>]</span>
<span id="cb1107-2"><a href="DataIntegrationPbmmcTop.html#cb1107-2"></a><span class="kw">plotExpression</span>(uncorrected, <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)), </span>
<span id="cb1107-3"><a href="DataIntegrationPbmmcTop.html#cb1107-3"></a>    <span class="dt">features=</span>geneEnsId, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/unnamed-chunk-130-1.png" width="672" /></p>
</div>
</div>
<div id="session-information-10" class="section level2">
<h2><span class="header-section-number">14.8</span> Session information</h2>
<div class="sourceCode" id="cb1108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1108-1"><a href="DataIntegrationPbmmcTop.html#cb1108-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] splines   parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] cellAlign_0.1.0             celldex_1.0.0              
##  [3] clusterProfiler_3.18.1      msigdbr_7.2.1              
##  [5] viridis_0.5.1               viridisLite_0.3.0          
##  [7] gam_1.20                    foreach_1.5.1              
##  [9] destiny_3.4.0               SingleR_1.4.1              
## [11] forcats_0.5.0               stringr_1.4.0              
## [13] readr_1.4.0                 tidyverse_1.3.0            
## [15] cowplot_1.1.1               edgeR_3.32.1               
## [17] DelayedArray_0.16.3         RColorBrewer_1.1-2         
## [19] reticulate_1.18             purrr_0.3.4                
## [21] tidyr_1.1.3                 tibble_3.1.1               
## [23] broom_0.7.6                 leiden_0.3.7               
## [25] cluster_2.1.0               dynamicTreeCut_1.63-1      
## [27] bluster_1.0.0               fossil_0.4.0               
## [29] shapefiles_0.7              foreign_0.8-80             
## [31] maps_3.3.0                  sp_1.4-5                   
## [33] clustree_0.4.3              ggraph_2.0.5               
## [35] pheatmap_1.0.12             harmony_1.0                
## [37] Rcpp_1.0.6                  batchelor_1.6.3            
## [39] SeuratObject_4.0.0          Seurat_4.0.1               
## [41] limma_3.46.0                ggfortify_0.4.11           
## [43] BiocSingular_1.6.0          scuttle_1.0.4              
## [45] robustbase_0.93-7           mixtools_1.2.0             
## [47] Cairo_1.5-12.2              dplyr_1.0.5                
## [49] DT_0.18                     irlba_2.3.3                
## [51] biomaRt_2.46.3              Matrix_1.2-18              
## [53] igraph_1.2.6                DropletUtils_1.10.3        
## [55] scater_1.18.6               ggplot2_3.3.3              
## [57] scran_1.18.7                SingleCellExperiment_1.12.0
## [59] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [61] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [63] IRanges_2.24.1              S4Vectors_0.28.1           
## [65] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [67] matrixStats_0.58.0          knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                    vcd_1.4-8                    
##   [3] ica_1.0-2                     class_7.3-17                 
##   [5] ps_1.6.0                      lmtest_0.9-38                
##   [7] crayon_1.4.1                  laeken_0.5.1                 
##   [9] spatstat.core_2.1-2           MASS_7.3-53                  
##  [11] rhdf5filters_1.2.0            nlme_3.1-149                 
##  [13] backports_1.2.1               reprex_0.3.0                 
##  [15] GOSemSim_2.16.1               rlang_0.4.10                 
##  [17] XVector_0.30.0                ROCR_1.0-11                  
##  [19] readxl_1.3.1                  smoother_1.1                 
##  [21] BiocParallel_1.24.1           bit64_4.0.5                  
##  [23] glue_1.4.2                    sctransform_0.3.2.9005       
##  [25] vipor_0.4.5                   spatstat.sparse_2.0-0        
##  [27] AnnotationDbi_1.52.0          DOSE_3.16.0                  
##  [29] spatstat.geom_2.1-0           haven_2.3.1                  
##  [31] tidyselect_1.1.0              rio_0.5.16                   
##  [33] fitdistrplus_1.1-3            XML_3.99-0.5                 
##  [35] zoo_1.8-9                     xtable_1.8-4                 
##  [37] RcppHNSW_0.3.0                magrittr_2.0.1               
##  [39] evaluate_0.14                 cli_2.4.0                    
##  [41] zlibbioc_1.36.0               rstudioapi_0.13              
##  [43] miniUI_0.1.1.1                bslib_0.2.4                  
##  [45] rpart_4.1-15                  fastmatch_1.1-0              
##  [47] treeio_1.14.3                 RcppEigen_0.3.3.9.1          
##  [49] shiny_1.5.0                   xfun_0.22                    
##  [51] askpass_1.1                   tidygraph_1.2.0              
##  [53] pcaMethods_1.82.0             interactiveDisplayBase_1.28.0
##  [55] ggrepel_0.9.1                 ape_5.4-1                    
##  [57] listenv_0.8.0                 png_0.1-7                    
##  [59] future_1.21.0                 withr_2.4.2                  
##  [61] bitops_1.0-6                  ggforce_0.3.3                
##  [63] ranger_0.12.1                 plyr_1.8.6                   
##  [65] cellranger_1.1.0              e1071_1.7-4                  
##  [67] dqrng_0.2.1                   pillar_1.6.0                 
##  [69] cachem_1.0.4                  fs_1.5.0                     
##  [71] kernlab_0.9-29                scatterplot3d_0.3-41         
##  [73] TTR_0.24.2                    xts_0.12.1                   
##  [75] DelayedMatrixStats_1.12.3     vctrs_0.3.7                  
##  [77] ellipsis_0.3.1                generics_0.1.0               
##  [79] tools_4.0.3                   beeswarm_0.2.3               
##  [81] munsell_0.5.0                 tweenr_1.0.2                 
##  [83] fgsea_1.16.0                  proxy_0.4-25                 
##  [85] fastmap_1.1.0                 compiler_4.0.3               
##  [87] abind_1.4-5                   httpuv_1.5.5                 
##  [89] ExperimentHub_1.16.1          segmented_1.3-3              
##  [91] plotly_4.9.3                  GenomeInfoDbData_1.2.4       
##  [93] gridExtra_2.3                 lattice_0.20-41              
##  [95] deldir_0.2-10                 utf8_1.2.1                   
##  [97] later_1.1.0.1                 BiocFileCache_1.14.0         
##  [99] jsonlite_1.7.2                ggplot.multistats_1.0.0      
## [101] scales_1.1.1                  tidytree_0.3.3               
## [103] pbapply_1.4-3                 carData_3.0-4                
## [105] sparseMatrixStats_1.2.1       lazyeval_0.2.2               
## [107] promises_1.1.1                car_3.0-10                   
## [109] R.utils_2.10.1                goftest_1.2-2                
## [111] spatstat.utils_2.1-0          rmarkdown_2.7                
## [113] openxlsx_4.2.3                statmod_1.4.35               
## [115] Rtsne_0.15                    downloader_0.4               
## [117] uwot_0.1.10                   HDF5Array_1.18.1             
## [119] survival_3.2-7                ResidualMatrix_1.0.0         
## [121] yaml_2.2.1                    htmltools_0.5.1.1            
## [123] memoise_2.0.0                 locfit_1.5-9.4               
## [125] graphlayouts_0.7.1            digest_0.6.27                
## [127] assertthat_0.2.1              mime_0.10                    
## [129] rappdirs_0.3.1                RSQLite_2.2.6                
## [131] future.apply_1.7.0            data.table_1.14.0            
## [133] blob_1.2.1                    R.oo_1.24.0                  
## [135] labeling_0.4.2                Rhdf5lib_1.12.1              
## [137] AnnotationHub_2.22.1          RCurl_1.98-1.3               
## [139] hms_1.0.0                     modelr_0.1.8                 
## [141] rhdf5_2.34.0                  colorspace_2.0-0             
## [143] BiocManager_1.30.12           ggbeeswarm_0.6.0             
## [145] aplot_0.0.6                   nnet_7.3-14                  
## [147] sass_0.3.1                    bookdown_0.21                
## [149] RANN_2.6.1                    enrichplot_1.11.2.994        
## [151] fansi_0.4.2                   VIM_6.1.0                    
## [153] parallelly_1.24.0             R6_2.5.0                     
## [155] grid_4.0.3                    ggridges_0.5.3               
## [157] lifecycle_1.0.0               zip_2.1.1                    
## [159] curl_4.3                      jquerylib_0.1.3              
## [161] DO.db_2.9                     qvalue_2.22.0                
## [163] RcppAnnoy_0.0.18              iterators_1.0.13             
## [165] htmlwidgets_1.5.3             beachmat_2.6.4               
## [167] polyclip_1.10-0               shadowtext_0.0.7             
## [169] rvest_0.3.6                   mgcv_1.8-33                  
## [171] globals_0.14.0                openssl_1.4.3                
## [173] patchwork_1.1.1               dtw_1.22-3                   
## [175] codetools_0.2-16              lubridate_1.7.10             
## [177] GO.db_3.12.1                  prettyunits_1.1.1            
## [179] dbplyr_2.0.0                  RSpectra_0.16-0              
## [181] R.methodsS3_1.8.1             gtable_0.3.0                 
## [183] DBI_1.1.1                     highr_0.8                    
## [185] tensor_1.5                    httr_1.4.2                   
## [187] KernSmooth_2.23-17            stringi_1.5.3                
## [189] progress_1.2.2                reshape2_1.4.4               
## [191] farver_2.1.0                  ggthemes_4.2.4               
## [193] hexbin_1.28.2                 ggtree_2.4.1                 
## [195] xml2_1.3.2                    rvcheck_0.1.8                
## [197] boot_1.3-25                   BiocNeighbors_1.8.2          
## [199] scattermore_0.7               BiocVersion_3.12.0           
## [201] DEoptimR_1.0-8                bit_4.0.4                    
## [203] scatterpie_0.1.5              spatstat.data_2.1-0          
## [205] pkgconfig_2.0.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dimensionality-reduction-for-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dataSetIntegration-ETV6-RUNX1Top.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/dataSetIntegration_PBMMC.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
