<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 24 Pseudotime analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 24 Pseudotime analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 24 Pseudotime analysis | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-expression-and-abundance-between-conditions.html"/>
<link rel="next" href="doubletDetectionTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="5.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>6</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#library-size-normalization-2"><i class="fa fa-check"></i><b>7.1.4</b> Library size normalization</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-caron-set.html"><a href="normalisation-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>8</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="8.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>8.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="8.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>8.1.1</b> CPM</a></li>
<li class="chapter" data-level="8.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>8.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="8.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>8.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="8.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>8.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="8.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>8.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="8.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="8.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>8.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="8.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>8.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-3"><i class="fa fa-check"></i><b>8.2</b> SCTransform</a><ul>
<li class="chapter" data-level="8.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>8.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="8.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>8.2.2</b> Transformation</a></li>
<li class="chapter" data-level="8.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>8.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>8.3</b> Visualisation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>8.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="8.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>8.3.2</b> log CPM</a></li>
<li class="chapter" data-level="8.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>8.3.3</b> scran</a></li>
<li class="chapter" data-level="8.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>8.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-4"><i class="fa fa-check"></i><b>8.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>9.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="9.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>9.1.1</b> Description</a></li>
<li class="chapter" data-level="9.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-2"><i class="fa fa-check"></i><b>9.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-2"><i class="fa fa-check"></i><b>9.2</b> Load packages</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>9.3</b> Load data</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>9.4</b> PCA</a><ul>
<li class="chapter" data-level="9.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>9.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>9.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>9.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="9.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>9.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>9.6</b> UMAP</a></li>
<li class="chapter" data-level="9.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-5"><i class="fa fa-check"></i><b>9.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>10</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="10.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>10.1</b> Load object</a></li>
<li class="chapter" data-level="10.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>10.2</b> PCA</a></li>
<li class="chapter" data-level="10.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>10.3</b> Normalised counts</a></li>
<li class="chapter" data-level="10.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-6"><i class="fa fa-check"></i><b>10.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>11</b> Feature selection</a><ul>
<li class="chapter" data-level="11.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>11.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="11.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>11.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="11.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>11.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-7"><i class="fa fa-check"></i><b>11.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>12</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>12.1</b> Prepare data</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>12.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>12.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>12.4</b> Batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>12.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-5"><i class="fa fa-check"></i><b>12.5</b> SCTransform</a><ul>
<li class="chapter" data-level="12.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>12.5.1</b> Batch only</a></li>
<li class="chapter" data-level="12.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>12.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>12.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="12.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>12.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="12.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>12.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>12.7</b> fastMNN</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>12.8</b> Harmony</a></li>
<li class="chapter" data-level="12.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-8"><i class="fa fa-check"></i><b>12.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>13</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="13.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>13.1</b> Load data</a></li>
<li class="chapter" data-level="13.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>13.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="13.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>13.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="13.4" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#session-information-9"><i class="fa fa-check"></i><b>13.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html"><i class="fa fa-check"></i><b>14</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="14.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#abbreviations-1"><i class="fa fa-check"></i><b>14.1</b> Abbreviations</a></li>
<li class="chapter" data-level="14.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#motivation"><i class="fa fa-check"></i><b>14.2</b> Motivation</a></li>
<li class="chapter" data-level="14.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#loading-the-data"><i class="fa fa-check"></i><b>14.3</b> Loading the data</a></li>
<li class="chapter" data-level="14.4" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>14.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="14.5" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#linear-regression"><i class="fa fa-check"></i><b>14.5</b> Linear regression</a></li>
<li class="chapter" data-level="14.6" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>14.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="14.6.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#algorithm-overview"><i class="fa fa-check"></i><b>14.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="14.6.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#application-to-the-data"><i class="fa fa-check"></i><b>14.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="14.6.3" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>14.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>14.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="14.7.1" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>14.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="14.7.2" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>14.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="DataIntegrationPbmmcTop.html"><a href="DataIntegrationPbmmcTop.html#session-information-10"><i class="fa fa-check"></i><b>14.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>15</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="15.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>15.1</b> Abbreviations</a></li>
<li class="chapter" data-level="15.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>15.2</b> Motivation</a></li>
<li class="chapter" data-level="15.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#load-data-3"><i class="fa fa-check"></i><b>15.3</b> Load data</a></li>
<li class="chapter" data-level="15.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>15.4</b> Loading the data</a></li>
<li class="chapter" data-level="15.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>15.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="15.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>15.6</b> Linear regression</a></li>
<li class="chapter" data-level="15.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>15.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="15.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>15.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="15.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>15.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>15.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="15.8.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches"><i class="fa fa-check"></i><b>15.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="15.8.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>15.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="15.8.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>15.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="15.8.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities"><i class="fa fa-check"></i><b>15.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="15.8.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index"><i class="fa fa-check"></i><b>15.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>15.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="15.10" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>16</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="16.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-3"><i class="fa fa-check"></i><b>16.1</b> Abbreviations</a></li>
<li class="chapter" data-level="16.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-2"><i class="fa fa-check"></i><b>16.2</b> Motivation</a></li>
<li class="chapter" data-level="16.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-4"><i class="fa fa-check"></i><b>16.3</b> Load data</a></li>
<li class="chapter" data-level="16.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>16.4</b> Loading the data</a></li>
<li class="chapter" data-level="16.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>16.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="16.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-2"><i class="fa fa-check"></i><b>16.6</b> Linear regression</a></li>
<li class="chapter" data-level="16.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>16.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="16.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>16.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="16.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>16.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>16.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="16.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>16.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="16.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>16.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="16.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>16.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="16.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>16.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="16.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-1"><i class="fa fa-check"></i><b>16.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>16.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="16.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-12"><i class="fa fa-check"></i><b>16.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>17</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="17.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-4"><i class="fa fa-check"></i><b>17.1</b> Abbreviations</a></li>
<li class="chapter" data-level="17.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-3"><i class="fa fa-check"></i><b>17.2</b> Motivation</a></li>
<li class="chapter" data-level="17.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-5"><i class="fa fa-check"></i><b>17.3</b> Load data</a></li>
<li class="chapter" data-level="17.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>17.4</b> Loading the data</a></li>
<li class="chapter" data-level="17.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>17.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="17.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-3"><i class="fa fa-check"></i><b>17.6</b> Linear regression</a></li>
<li class="chapter" data-level="17.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>17.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="17.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>17.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="17.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>17.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>17.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="17.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>17.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="17.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>17.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="17.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>17.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="17.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>17.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="17.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-2"><i class="fa fa-check"></i><b>17.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>17.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="17.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-13"><i class="fa fa-check"></i><b>17.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>18</b> Data integration - PBMMC + ETV6-RUNX</a><ul>
<li class="chapter" data-level="18.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#abbreviations-5"><i class="fa fa-check"></i><b>18.1</b> Abbreviations</a></li>
<li class="chapter" data-level="18.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#motivation-4"><i class="fa fa-check"></i><b>18.2</b> Motivation</a></li>
<li class="chapter" data-level="18.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#load-data-6"><i class="fa fa-check"></i><b>18.3</b> Load data</a></li>
<li class="chapter" data-level="18.4" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#loading-the-data-4"><i class="fa fa-check"></i><b>18.4</b> Loading the data</a></li>
<li class="chapter" data-level="18.5" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>18.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="18.6" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>18.6</b> Linear regression</a></li>
<li class="chapter" data-level="18.7" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#performing-mnn-correction-4"><i class="fa fa-check"></i><b>18.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="18.7.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>18.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="18.7.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>18.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>18.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="18.8.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>18.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="18.8.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>18.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="18.8.2.1" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>18.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="18.8.2.2" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>18.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="18.8.2.3" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#rand-index-3"><i class="fa fa-check"></i><b>18.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>18.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="18.10" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>18.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="18.11" data-path="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html"><a href="dataSetIntegration-PBMMC-ETV6-RUNX1Top.html#session-information-14"><i class="fa fa-check"></i><b>18.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html"><i class="fa fa-check"></i><b>19</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="19.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#motivation-5"><i class="fa fa-check"></i><b>19.1</b> Motivation</a></li>
<li class="chapter" data-level="19.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#load-data-7"><i class="fa fa-check"></i><b>19.2</b> Load data</a></li>
<li class="chapter" data-level="19.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>19.3</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="19.4" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#linear-regression-5"><i class="fa fa-check"></i><b>19.4</b> Linear regression</a></li>
<li class="chapter" data-level="19.5" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#performing-mnn-correction-5"><i class="fa fa-check"></i><b>19.5</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="19.5.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>19.5.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="19.5.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>19.5.2</b> Application to the data</a></li>
<li class="chapter" data-level="19.5.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>19.5.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>19.6</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="19.6.1" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>19.6.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="19.6.2" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>19.6.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="19.6.3" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>19.6.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>19.7</b> <strong>Challenge</strong> Same but with an ordered merging</a></li>
<li class="chapter" data-level="19.8" data-path="dataSetIntegration-allSetsTop.html"><a href="dataSetIntegration-allSetsTop.html#session-information-15"><i class="fa fa-check"></i><b>19.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>20</b> Clustering with ‘’PBMMC,ETV6-RUNX1’’ and ’_5hCellPerSpl’ {#clustering’PBMMC_ETV6-RUNX1__5hCellPerSpl’Top}</a><ul>
<li class="chapter" data-level="20.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-8"><i class="fa fa-check"></i><b>20.1</b> Load data</a></li>
<li class="chapter" data-level="20.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>20.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="20.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>20.2.1</b> Defining cell clusters from expression data</a><ul>
<li class="chapter" data-level="20.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>20.2.1.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="20.2.1.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>20.2.1.1.1</b> Clustering</a></li>
<li class="chapter" data-level="20.2.1.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>20.2.1.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>20.2.1.2</b> k-means</a><ul>
<li class="chapter" data-level="20.2.1.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>20.2.1.2.1</b> Description</a></li>
<li class="chapter" data-level="20.2.1.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-3"><i class="fa fa-check"></i><b>20.2.1.2.2</b> Example</a></li>
<li class="chapter" data-level="20.2.1.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>20.2.1.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="20.2.1.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>20.2.1.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="20.2.1.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>20.2.1.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="20.2.1.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>20.2.1.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="20.2.1.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>20.2.1.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="20.2.1.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>20.2.1.3.1</b> Modularity</a></li>
<li class="chapter" data-level="20.2.1.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-via-short-random-walks-walktrap"><i class="fa fa-check"></i><b>20.2.1.3.2</b> Community detection via short random walks: ‘walktrap’</a></li>
<li class="chapter" data-level="20.2.1.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-louvain-method"><i class="fa fa-check"></i><b>20.2.1.3.3</b> Community detection with the ‘Louvain’ method</a></li>
<li class="chapter" data-level="20.2.1.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#community-detection-with-the-leiden-method"><i class="fa fa-check"></i><b>20.2.1.3.4</b> Community detection with the ‘Leiden’ method</a></li>
<li class="chapter" data-level="20.2.1.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>20.2.1.3.5</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>20.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="20.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>20.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="20.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>20.5</b> Save data</a></li>
<li class="chapter" data-level="20.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-16"><i class="fa fa-check"></i><b>20.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html"><i class="fa fa-check"></i><b>21</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="21.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#learning-objectives"><i class="fa fa-check"></i><b>21.1</b> Learning objectives</a></li>
<li class="chapter" data-level="21.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#load-data-9"><i class="fa fa-check"></i><b>21.2</b> Load data</a></li>
<li class="chapter" data-level="21.3" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>21.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="21.3.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#differential-expression-analysis"><i class="fa fa-check"></i><b>21.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="21.3.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#heatmap"><i class="fa fa-check"></i><b>21.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>21.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="21.5" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>21.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="21.6" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>21.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="21.7" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#using-a-binomial-test"><i class="fa fa-check"></i><b>21.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="21.8" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>21.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="21.9" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>21.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="21.9.1" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#from-data-snooping"><i class="fa fa-check"></i><b>21.9.1</b> 11.5.1 From data snooping</a></li>
<li class="chapter" data-level="21.9.2" data-path="cluster-marker-genes.html"><a href="cluster-marker-genes.html#nature-of-replication"><i class="fa fa-check"></i><b>21.9.2</b> Nature of replication</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>22</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="22.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-10"><i class="fa fa-check"></i><b>22.1</b> Load data</a></li>
<li class="chapter" data-level="22.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-6"><i class="fa fa-check"></i><b>22.2</b> Motivation</a></li>
<li class="chapter" data-level="22.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>22.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="22.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>22.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="22.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>22.5</b> Regressing out cell cycle phase</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html"><i class="fa fa-check"></i><b>23</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#motivation-7"><i class="fa fa-check"></i><b>23.1</b> Motivation</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#setting-up-the-data"><i class="fa fa-check"></i><b>23.2</b> Setting up the data</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>23.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="23.3.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>23.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="23.3.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>23.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="23.3.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#introduction-4"><i class="fa fa-check"></i><b>23.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="23.3.2.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#pre-processing"><i class="fa fa-check"></i><b>23.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="23.3.2.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#statistical-modelling"><i class="fa fa-check"></i><b>23.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="23.3.2.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>23.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="23.3.2.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>23.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="23.3.2.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#list-of-degs"><i class="fa fa-check"></i><b>23.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="23.3.2.4.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>23.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.3.3" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#putting-it-all-together"><i class="fa fa-check"></i><b>23.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>23.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="23.4.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#overview"><i class="fa fa-check"></i><b>23.4.1</b> Overview</a></li>
<li class="chapter" data-level="23.4.2" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#handling-composition-effects"><i class="fa fa-check"></i><b>23.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="23.4.2.1" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#background"><i class="fa fa-check"></i><b>23.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="differential-expression-and-abundance-between-conditions.html"><a href="differential-expression-and-abundance-between-conditions.html#session-information-17"><i class="fa fa-check"></i><b>23.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>24</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="24.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>24.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="24.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>24.0.2</b> Data integration</a></li>
<li class="chapter" data-level="24.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>24.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="24.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>24.0.4</b> Clustering</a></li>
<li class="chapter" data-level="24.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cell-type-classification"><i class="fa fa-check"></i><b>24.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="24.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>24.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="24.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>24.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="24.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>24.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="24.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>24.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="24.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>24.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="24.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>24.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="24.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>24.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="24.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>24.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="24.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-18"><i class="fa fa-check"></i><b>24.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>25</b> Doublet detection</a><ul>
<li class="chapter" data-level="25.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-1"><i class="fa fa-check"></i><b>25.1</b> Learning objectives</a></li>
<li class="chapter" data-level="25.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>25.2</b> Set up analysis</a></li>
<li class="chapter" data-level="25.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-11"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>25.4</b> Overview</a></li>
<li class="chapter" data-level="25.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>25.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="25.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>25.6</b> Further comments</a></li>
<li class="chapter" data-level="25.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-19"><i class="fa fa-check"></i><b>25.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pseudoTimeTop" class="section level1">
<h1><span class="header-section-number">Chapter 24</span> Pseudotime analysis</h1>
<div class="sourceCode" id="cb1930"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1930-1"><a href="pseudoTimeTop.html#cb1930-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb1930-2"><a href="pseudoTimeTop.html#cb1930-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb1930-3"><a href="pseudoTimeTop.html#cb1930-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb1930-4"><a href="pseudoTimeTop.html#cb1930-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb1930-5"><a href="pseudoTimeTop.html#cb1930-5"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb1930-6"><a href="pseudoTimeTop.html#cb1930-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb1930-7"><a href="pseudoTimeTop.html#cb1930-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb1930-8"><a href="pseudoTimeTop.html#cb1930-8"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1930-9"><a href="pseudoTimeTop.html#cb1930-9"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1930-10"><a href="pseudoTimeTop.html#cb1930-10"></a>    setName &lt;-<span class="st"> &quot;hca&quot;</span></span>
<span id="cb1930-11"><a href="pseudoTimeTop.html#cb1930-11"></a>    splSetToGet &lt;-<span class="st"> &quot;dummy&quot;</span></span>
<span id="cb1930-12"><a href="pseudoTimeTop.html#cb1930-12"></a>    setSuf &lt;-<span class="st"> &quot;_5kCellPerSpl&quot;</span></span>
<span id="cb1930-13"><a href="pseudoTimeTop.html#cb1930-13"></a>    dsiSuf &lt;-<span class="st"> &#39;_dummy&#39;</span></span>
<span id="cb1930-14"><a href="pseudoTimeTop.html#cb1930-14"></a>}</span>
<span id="cb1930-15"><a href="pseudoTimeTop.html#cb1930-15"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1930-16"><a href="pseudoTimeTop.html#cb1930-16"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1930-17"><a href="pseudoTimeTop.html#cb1930-17"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1930-18"><a href="pseudoTimeTop.html#cb1930-18"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
<!--
These are the libraries we will need in this workbook
-->
<div class="sourceCode" id="cb1931"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1931-1"><a href="pseudoTimeTop.html#cb1931-1"></a><span class="kw">library</span>(SingleCellExperiment)</span>
<span id="cb1931-2"><a href="pseudoTimeTop.html#cb1931-2"></a><span class="kw">library</span>(scran)</span>
<span id="cb1931-3"><a href="pseudoTimeTop.html#cb1931-3"></a><span class="kw">library</span>(scater)</span>
<span id="cb1931-4"><a href="pseudoTimeTop.html#cb1931-4"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb1931-5"><a href="pseudoTimeTop.html#cb1931-5"></a><span class="kw">library</span>(cowplot)</span>
<span id="cb1931-6"><a href="pseudoTimeTop.html#cb1931-6"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1931-7"><a href="pseudoTimeTop.html#cb1931-7"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb1931-8"><a href="pseudoTimeTop.html#cb1931-8"></a><span class="kw">library</span>(SingleR)</span>
<span id="cb1931-9"><a href="pseudoTimeTop.html#cb1931-9"></a><span class="kw">library</span>(destiny)</span>
<span id="cb1931-10"><a href="pseudoTimeTop.html#cb1931-10"></a><span class="kw">library</span>(gam)</span>
<span id="cb1931-11"><a href="pseudoTimeTop.html#cb1931-11"></a><span class="kw">library</span>(viridis)</span>
<span id="cb1931-12"><a href="pseudoTimeTop.html#cb1931-12"></a><span class="kw">library</span>(msigdbr)</span>
<span id="cb1931-13"><a href="pseudoTimeTop.html#cb1931-13"></a><span class="kw">library</span>(clusterProfiler)</span>
<span id="cb1931-14"><a href="pseudoTimeTop.html#cb1931-14"></a><span class="kw">library</span>(cellAlign) <span class="co"># https://github.com/shenorrLab/cellAlign</span></span>
<span id="cb1931-15"><a href="pseudoTimeTop.html#cb1931-15"></a><span class="kw">library</span>(Cairo)</span></code></pre></div>
<p>## Extract T-cells from HCA Dataset {#pseudoTimeExtractTCell}</p>
<p>In this section, we are starting our analysis with normalized <a href="https://preview.data.humancellatlas.org">HCA</a> data and perform integration, clustering and dimensionality reduction. Our aim is to extract T-cells from this dataset and proceed with pseudotime analysis in the next section.</p>
<p>We are going to work with HCA data. This data set has been pre-processed and normalized before.</p>
<!--
Add foldable code for recap.
Or link to chapter.
-->
<div class="sourceCode" id="cb1932"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1932-1"><a href="pseudoTimeTop.html#cb1932-1"></a><span class="co">#sce&lt;-readRDS(file=&quot;~/Course_Materials/scRNAseq/pseudotime/hca_sce.bone.RDS&quot;)</span></span>
<span id="cb1932-2"><a href="pseudoTimeTop.html#cb1932-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>,</span>
<span id="cb1932-3"><a href="pseudoTimeTop.html#cb1932-3"></a>                 projDir, outDirBit, setName, setSuf)</span>
<span id="cb1932-4"><a href="pseudoTimeTop.html#cb1932-4"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/hca_sce_nz_postDeconv_5kCellPerSpl.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1934"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1934-1"><a href="pseudoTimeTop.html#cb1934-1"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>tmpFn)</span></code></pre></div>
<p>We use symbols in place of ENSEMBL IDs for easier interpretation later.</p>
<div class="sourceCode" id="cb1935"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1935-1"><a href="pseudoTimeTop.html#cb1935-1"></a><span class="co"># BiocManager::install(&quot;EnsDb.Hsapiens.v86&quot;)</span></span>
<span id="cb1935-2"><a href="pseudoTimeTop.html#cb1935-2"></a><span class="co">#rowData(sce)$Chr &lt;- mapIds(EnsDb.Hsapiens.v86, keys=rownames(sce), column=&quot;SEQNAME&quot;, keytype=&quot;GENEID&quot;)</span></span>
<span id="cb1935-3"><a href="pseudoTimeTop.html#cb1935-3"></a><span class="kw">rownames</span>(sce) &lt;-<span class="st"> </span><span class="kw">uniquifyFeatureNames</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id, <span class="dt">names =</span> <span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol)</span></code></pre></div>
<div id="variance-modeling" class="section level3">
<h3><span class="header-section-number">24.0.1</span> Variance modeling</h3>
<p>We block on the donor of origin to mitigate batch effects during highly variable gene (HVG) selection.
We select a larger number of HVGs to capture any batch-specific variation that might be present.</p>
<div class="sourceCode" id="cb1936"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1936-1"><a href="pseudoTimeTop.html#cb1936-1"></a>dec.hca &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce, <span class="dt">block=</span>sce<span class="op">$</span>Sample.Name)</span>
<span id="cb1936-2"><a href="pseudoTimeTop.html#cb1936-2"></a>top.hca &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(dec.hca, <span class="dt">n=</span><span class="dv">5000</span>)</span></code></pre></div>
</div>
<div id="data-integration" class="section level3">
<h3><span class="header-section-number">24.0.2</span> Data integration</h3>
<p>The <code>batchelor</code> package provides an implementation of the Mutual Nearest Neighbours (MNN) approach via the fastMNN() function. We apply it to our HCA data to remove the donor specific effects across the highly variable genes in <code>top.hca</code>. To reduce computational work and technical noise, all cells in all samples are projected into the low-dimensional space defined by the top d principal components. Identification of MNNs and calculation of correction vectors are then performed in this low-dimensional space.
The corrected matrix in the reducedDims() contains the low-dimensional corrected coordinates for all cells, which we will use in place of the PCs in our downstream analyses. We store it in ‘MNN’ slot in the main sce object.</p>
<div class="sourceCode" id="cb1937"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1937-1"><a href="pseudoTimeTop.html#cb1937-1"></a><span class="kw">set.seed</span>(<span class="dv">1010001</span>)</span>
<span id="cb1937-2"><a href="pseudoTimeTop.html#cb1937-2"></a>merged.hca &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(sce,</span>
<span id="cb1937-3"><a href="pseudoTimeTop.html#cb1937-3"></a>              <span class="dt">batch =</span> sce<span class="op">$</span>Sample.Name,</span>
<span id="cb1937-4"><a href="pseudoTimeTop.html#cb1937-4"></a>              <span class="dt">subset.row =</span> top.hca)</span>
<span id="cb1937-5"><a href="pseudoTimeTop.html#cb1937-5"></a><span class="kw">reducedDim</span>(sce, <span class="st">&#39;MNN&#39;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(merged.hca, <span class="st">&#39;corrected&#39;</span>)</span></code></pre></div>
</div>
<div id="dimensionality-reduction" class="section level3">
<h3><span class="header-section-number">24.0.3</span> Dimensionality Reduction</h3>
<p>We cluster on the low-dimensional corrected coordinates to obtain a partitioning of the cells that serves as a proxy for the population structure. If the batch effect is successfully corrected, clusters corresponding to shared cell types or states should contain cells from multiple samples. We see that all clusters contain contributions from each sample after correction.</p>
<div class="sourceCode" id="cb1938"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1938-1"><a href="pseudoTimeTop.html#cb1938-1"></a><span class="kw">set.seed</span>(<span class="dv">01010100</span>)</span>
<span id="cb1938-2"><a href="pseudoTimeTop.html#cb1938-2"></a>sce &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;MNN&quot;</span>)</span>
<span id="cb1938-3"><a href="pseudoTimeTop.html#cb1938-3"></a>sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;MNN&quot;</span>)</span>
<span id="cb1938-4"><a href="pseudoTimeTop.html#cb1938-4"></a>sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;MNN&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1939"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1939-1"><a href="pseudoTimeTop.html#cb1939-1"></a><span class="kw">plotPCA</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/dimRed_plot_pseudotime1-1.png" width="672" /></p>
<div class="sourceCode" id="cb1940"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1940-1"><a href="pseudoTimeTop.html#cb1940-1"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;tSNE&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/dimRed_plot_pseudotime1-2.png" width="672" /></p>
<div class="sourceCode" id="cb1941"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1941-1"><a href="pseudoTimeTop.html#cb1941-1"></a><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/dimRed_plot_pseudotime1-3.png" width="672" /></p>
</div>
<div id="clustering-1" class="section level3">
<h3><span class="header-section-number">24.0.4</span> Clustering</h3>
<p>Graph-based clustering generates an excessively large intermediate graph so we will instead use a two-step approach with k-means. We generate 1000 small clusters that are subsequently aggregated into more interpretable groups with a graph-based method.</p>
<div class="sourceCode" id="cb1942"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1942-1"><a href="pseudoTimeTop.html#cb1942-1"></a><span class="kw">set.seed</span>(<span class="dv">1000</span>)</span>
<span id="cb1942-2"><a href="pseudoTimeTop.html#cb1942-2"></a>clust.hca &lt;-<span class="st"> </span><span class="kw">clusterSNNGraph</span>(sce,</span>
<span id="cb1942-3"><a href="pseudoTimeTop.html#cb1942-3"></a>                             <span class="dt">use.dimred=</span><span class="st">&quot;MNN&quot;</span>,</span>
<span id="cb1942-4"><a href="pseudoTimeTop.html#cb1942-4"></a>                             <span class="dt">use.kmeans=</span><span class="ot">TRUE</span>,</span>
<span id="cb1942-5"><a href="pseudoTimeTop.html#cb1942-5"></a>                             <span class="dt">kmeans.centers=</span><span class="dv">1000</span>)</span>
<span id="cb1942-6"><a href="pseudoTimeTop.html#cb1942-6"></a></span>
<span id="cb1942-7"><a href="pseudoTimeTop.html#cb1942-7"></a><span class="kw">colLabels</span>(sce) &lt;-<span class="st"> </span><span class="kw">factor</span>(clust.hca)</span>
<span id="cb1942-8"><a href="pseudoTimeTop.html#cb1942-8"></a><span class="kw">table</span>(<span class="kw">colLabels</span>(sce))</span></code></pre></div>
<pre><code>## 
##     1     2     3     4     5     6     7     8     9    10    11    12    13 
##  5366  4405  2419  2828  2367  6035   897  1749 10177  1069  1674   623   391</code></pre>
<div class="sourceCode" id="cb1944"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1944-1"><a href="pseudoTimeTop.html#cb1944-1"></a><span class="kw">plotPCA</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/clustering_plot_pseudotime1-1.png" width="672" /></p>
<div class="sourceCode" id="cb1945"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1945-1"><a href="pseudoTimeTop.html#cb1945-1"></a><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/clustering_plot_pseudotime1-2.png" width="672" /></p>
<div class="sourceCode" id="cb1946"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1946-1"><a href="pseudoTimeTop.html#cb1946-1"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;label&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;tSNE&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/clustering_plot_pseudotime1-3.png" width="672" /></p>
</div>
<div id="cell-type-classification" class="section level3">
<h3><span class="header-section-number">24.0.5</span> Cell type classification</h3>
<p>We perform automated cell type classification using a reference dataset to annotate each cluster based on its pseudo-bulk profile. This is for a quick assignment of cluster identity. We are going to use Human Primary Cell Atlas (HPCA) data for that. <code>HumanPrimaryCellAtlasData</code> function provides normalized expression values for 713 microarray samples from HPCA (<a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-14-632">Mabbott et al., 2013</a>).
These 713 samples were processed and normalized as described in <a href="https://www.nature.com/articles/s41590-018-0276-y">Aran, Looney and Liu et al. (2019)</a>.
Each sample has been assigned to one of 37 main cell types and 157 subtypes.</p>
<div class="sourceCode" id="cb1947"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1947-1"><a href="pseudoTimeTop.html#cb1947-1"></a>se.aggregated &lt;-<span class="st"> </span><span class="kw">sumCountsAcrossCells</span>(sce, <span class="dt">id=</span><span class="kw">colLabels</span>(sce))</span>
<span id="cb1947-2"><a href="pseudoTimeTop.html#cb1947-2"></a>hpc &lt;-<span class="st"> </span>celldex<span class="op">::</span><span class="kw">HumanPrimaryCellAtlasData</span>()</span>
<span id="cb1947-3"><a href="pseudoTimeTop.html#cb1947-3"></a>anno.hca &lt;-<span class="st"> </span><span class="kw">SingleR</span>(se.aggregated, <span class="dt">ref =</span> hpc, <span class="dt">labels =</span> hpc<span class="op">$</span>label.main, <span class="dt">assay.type.test=</span><span class="st">&quot;sum&quot;</span>)</span>
<span id="cb1947-4"><a href="pseudoTimeTop.html#cb1947-4"></a>anno.hca</span></code></pre></div>
<pre><code>## DataFrame with 13 rows and 5 columns
##                             scores     first.labels     tuning.scores
##                           &lt;matrix&gt;      &lt;character&gt;       &lt;DataFrame&gt;
## 1   0.259339:0.585119:0.666284:...         Monocyte 0.562065:0.256610
## 2   0.281126:0.632594:0.588652:...          NK_cell 0.579341:0.463134
## 3   0.308231:0.630162:0.730628:... Pre-B_cell_CD34- 0.285299:0.249503
## 4   0.255632:0.745508:0.615614:...           B_cell 0.678913:0.181606
## 5   0.372877:0.567884:0.660298:...              MEP 0.362612:0.341907
## ...                            ...              ...               ...
## 9   0.289160:0.641716:0.586731:...          T_cells 0.769493:0.717388
## 10  0.285185:0.639443:0.600310:...          NK_cell 0.808176:0.725154
## 11  0.269728:0.759889:0.598751:...           B_cell 0.759889:0.684991
## 12  0.326027:0.676683:0.576854:...           B_cell 0.495728:0.264335
## 13  0.288975:0.641663:0.609441:... Pre-B_cell_CD34- 0.212197:0.200732
##               labels    pruned.labels
##          &lt;character&gt;      &lt;character&gt;
## 1           Monocyte         Monocyte
## 2            T_cells          T_cells
## 3   Pre-B_cell_CD34- Pre-B_cell_CD34-
## 4             B_cell           B_cell
## 5         BM &amp; Prog.       BM &amp; Prog.
## ...              ...              ...
## 9            T_cells          T_cells
## 10           NK_cell          NK_cell
## 11            B_cell           B_cell
## 12            B_cell           B_cell
## 13  Pre-B_cell_CD34- Pre-B_cell_CD34-</code></pre>
<div class="sourceCode" id="cb1949"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1949-1"><a href="pseudoTimeTop.html#cb1949-1"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(anno.hca<span class="op">$</span>labels, <span class="kw">colnames</span>(se.aggregated))</span>
<span id="cb1949-2"><a href="pseudoTimeTop.html#cb1949-2"></a><span class="co"># Adding a pseudo-count of 10 to avoid strong color jumps with just 1 cell.</span></span>
<span id="cb1949-3"><a href="pseudoTimeTop.html#cb1949-3"></a><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>))</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/cellType_heatmap_pseudotime1-1.png" width="672" /></p>
<div class="sourceCode" id="cb1950"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1950-1"><a href="pseudoTimeTop.html#cb1950-1"></a>sce<span class="op">$</span>cell_type&lt;-<span class="kw">recode</span>(sce<span class="op">$</span>label,</span>
<span id="cb1950-2"><a href="pseudoTimeTop.html#cb1950-2"></a>       <span class="st">&quot;1&quot;</span> =<span class="st"> &quot;T_cells&quot;</span>, </span>
<span id="cb1950-3"><a href="pseudoTimeTop.html#cb1950-3"></a>       <span class="st">&quot;2&quot;</span> =<span class="st"> &quot;Monocyte&quot;</span>, </span>
<span id="cb1950-4"><a href="pseudoTimeTop.html#cb1950-4"></a>       <span class="st">&quot;3&quot;</span>=<span class="st">&quot;B_cell&quot;</span>,</span>
<span id="cb1950-5"><a href="pseudoTimeTop.html#cb1950-5"></a>       <span class="st">&quot;4&quot;</span>=<span class="st">&quot;MEP&quot;</span>, </span>
<span id="cb1950-6"><a href="pseudoTimeTop.html#cb1950-6"></a>       <span class="st">&quot;5&quot;</span>=<span class="st">&quot;B_cell&quot;</span>, </span>
<span id="cb1950-7"><a href="pseudoTimeTop.html#cb1950-7"></a>       <span class="st">&quot;6&quot;</span>=<span class="st">&quot;CMP&quot;</span>, </span>
<span id="cb1950-8"><a href="pseudoTimeTop.html#cb1950-8"></a>       <span class="st">&quot;7&quot;</span>=<span class="st">&quot;T_cells&quot;</span>,</span>
<span id="cb1950-9"><a href="pseudoTimeTop.html#cb1950-9"></a>      <span class="st">&quot;8&quot;</span>=<span class="st">&quot;Monocyte&quot;</span>,</span>
<span id="cb1950-10"><a href="pseudoTimeTop.html#cb1950-10"></a>      <span class="st">&quot;9&quot;</span>=<span class="st">&quot;T_cells&quot;</span>,</span>
<span id="cb1950-11"><a href="pseudoTimeTop.html#cb1950-11"></a>      <span class="st">&quot;10&quot;</span>=<span class="st">&quot;Pro-B_cell_CD34+&quot;</span>,</span>
<span id="cb1950-12"><a href="pseudoTimeTop.html#cb1950-12"></a>      <span class="st">&quot;11&quot;</span>=<span class="st">&quot;NK_cell&quot;</span>,</span>
<span id="cb1950-13"><a href="pseudoTimeTop.html#cb1950-13"></a>      <span class="st">&quot;12&quot;</span>=<span class="st">&quot;B_cell&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1951"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1951-1"><a href="pseudoTimeTop.html#cb1951-1"></a><span class="co">#level_key &lt;- anno.hca %&gt;%</span></span>
<span id="cb1951-2"><a href="pseudoTimeTop.html#cb1951-2"></a><span class="co">#  data.frame() %&gt;%</span></span>
<span id="cb1951-3"><a href="pseudoTimeTop.html#cb1951-3"></a><span class="co">#  rownames_to_column(&quot;clu&quot;) %&gt;%</span></span>
<span id="cb1951-4"><a href="pseudoTimeTop.html#cb1951-4"></a><span class="co">#  #select(clu, labels)</span></span>
<span id="cb1951-5"><a href="pseudoTimeTop.html#cb1951-5"></a><span class="co">#  pull(labels)</span></span>
<span id="cb1951-6"><a href="pseudoTimeTop.html#cb1951-6"></a>level_key &lt;-<span class="st"> </span>anno.hca<span class="op">$</span>labels</span>
<span id="cb1951-7"><a href="pseudoTimeTop.html#cb1951-7"></a><span class="kw">names</span>(level_key) &lt;-<span class="st"> </span><span class="kw">row.names</span>(anno.hca)</span>
<span id="cb1951-8"><a href="pseudoTimeTop.html#cb1951-8"></a>sce<span class="op">$</span>cell_type &lt;-<span class="st"> </span><span class="kw">recode</span>(sce<span class="op">$</span>label, <span class="op">!!!</span>level_key)</span></code></pre></div>
<p>We can now use the predicted cell types to color PCA, UMAP and tSNE.</p>
<div class="sourceCode" id="cb1952"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1952-1"><a href="pseudoTimeTop.html#cb1952-1"></a><span class="kw">plotPCA</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;cell_type&quot;</span>, <span class="dt">text_by=</span><span class="st">&quot;cell_type&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/cellType_plot_pseudotime1-1.png" width="672" /></p>
<div class="sourceCode" id="cb1953"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1953-1"><a href="pseudoTimeTop.html#cb1953-1"></a><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;cell_type&quot;</span>, <span class="dt">text_by=</span><span class="st">&quot;cell_type&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/cellType_plot_pseudotime1-2.png" width="672" /></p>
<div class="sourceCode" id="cb1954"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1954-1"><a href="pseudoTimeTop.html#cb1954-1"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;cell_type&quot;</span>, <span class="dt">text_by=</span><span class="st">&quot;cell_type&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;tSNE&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/cellType_plot_pseudotime1-3.png" width="672" /></p>
<p>We can also check expression of some marker genes.</p>
<p>CD3D and TRAC are used as marker genes for T-cells <a href="https://www.nature.com/articles/s41467-019-12464-3">Szabo et al. 2019</a>.</p>
<div class="sourceCode" id="cb1955"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1955-1"><a href="pseudoTimeTop.html#cb1955-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">features=</span><span class="kw">c</span>(<span class="st">&quot;CD3D&quot;</span>), <span class="dt">x=</span><span class="st">&quot;label&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;cell_type&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/tCellMark1_plot_pseudotime1-1.png" width="672" /></p>
<div class="sourceCode" id="cb1956"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1956-1"><a href="pseudoTimeTop.html#cb1956-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">features=</span><span class="kw">c</span>(<span class="st">&quot;TRAC&quot;</span>), <span class="dt">x=</span><span class="st">&quot;label&quot;</span>, <span class="dt">colour_by=</span><span class="st">&quot;cell_type&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/tCellMark2_plot_pseudotime1-1.png" width="672" /></p>
</div>
<div id="extract-t-cells" class="section level3">
<h3><span class="header-section-number">24.0.6</span> Extract T-cells</h3>
<p>We will now extract T-cells and store in a new SCE object to use in pseudotime analysis.</p>
<p>Pull barcodes for T-cells</p>
<div class="sourceCode" id="cb1957"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1957-1"><a href="pseudoTimeTop.html#cb1957-1"></a>tcell.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb1957-2"><a href="pseudoTimeTop.html#cb1957-2"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1957-3"><a href="pseudoTimeTop.html#cb1957-3"></a><span class="st">    </span><span class="kw">group_by</span>(cell_type) <span class="op">%&gt;%</span></span>
<span id="cb1957-4"><a href="pseudoTimeTop.html#cb1957-4"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(cell_type <span class="op">==</span><span class="st"> &quot;T_cells&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1957-5"><a href="pseudoTimeTop.html#cb1957-5"></a><span class="st">    </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1957-6"><a href="pseudoTimeTop.html#cb1957-6"></a></span>
<span id="cb1957-7"><a href="pseudoTimeTop.html#cb1957-7"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>tcell.bc)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
## 19383 20617</code></pre>
<p>Create a new SingleCellExperiment object for T-cells</p>
<div class="sourceCode" id="cb1959"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1959-1"><a href="pseudoTimeTop.html#cb1959-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>tcell.bc)</span>
<span id="cb1959-2"><a href="pseudoTimeTop.html#cb1959-2"></a>sce.tcell &lt;-<span class="st"> </span>sce[,tmpInd]</span></code></pre></div>
<div class="sourceCode" id="cb1960"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1960-1"><a href="pseudoTimeTop.html#cb1960-1"></a><span class="co">#saveRDS(sce.tcell,&quot;~/Course_Materials/scRNAseq/pseudotime/sce.tcell.RDS&quot;)</span></span>
<span id="cb1960-2"><a href="pseudoTimeTop.html#cb1960-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_tcell.Rds&quot;</span>,</span>
<span id="cb1960-3"><a href="pseudoTimeTop.html#cb1960-3"></a>                 projDir, outDirBit, setName, setSuf)</span>
<span id="cb1960-4"><a href="pseudoTimeTop.html#cb1960-4"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/hca_sce_nz_postDeconv_5kCellPerSpl_tcell.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1962"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1962-1"><a href="pseudoTimeTop.html#cb1962-1"></a><span class="kw">saveRDS</span>(sce.tcell, tmpFn)</span>
<span id="cb1962-2"><a href="pseudoTimeTop.html#cb1962-2"></a><span class="kw">rm</span>(sce.tcell)</span></code></pre></div>
</div>
<div id="pseudoTimeSetUp" class="section level2">
<h2><span class="header-section-number">24.1</span> Setting up the data</h2>
<p>In many situations, one is studying a process where cells change continuously. This includes, for example, many differentiation processes taking place during development: following a stimulus, cells will change from one cell-type to another. Ideally, we would like to monitor the expression levels of an individual cell over time. Unfortunately, such monitoring is not possible with scRNA-seq since the cell is lysed (destroyed) when the RNA is extracted.</p>
<p>Instead, we must sample at multiple time-points and obtain snapshots of the gene expression profiles. Since some of the cells will proceed faster along the differentiation than others, each snapshot may contain cells at varying points along the developmental progression. We use statistical methods to order the cells along one or more trajectories which represent the underlying developmental trajectories, this ordering is referred to as “pseudotime”.</p>
<p>A recent benchmarking paper by <a href="https://doi.org/10.1038/s41587-019-0071-9">Saelens et al</a> provides a detailed summary of the various computational methods for trajectory inference from single-cell transcriptomics. They discuss 45 tools and evaluate them across various aspects including accuracy, scalability, and usability. They provide <a href="https://dynverse.org">dynverse</a>, an open set of packages to benchmark, construct and interpret single-cell trajectories (currently they have a uniform interface for 60 methods).
We load the SCE object we have generated previously. This object contains only the T-cells from 8 healthy donors. We will first prepare the data by identifying variable genes, integrating the data across donors and calculating principal components.</p>
<div class="sourceCode" id="cb1963"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1963-1"><a href="pseudoTimeTop.html#cb1963-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_tcell.Rds&quot;</span>,</span>
<span id="cb1963-2"><a href="pseudoTimeTop.html#cb1963-2"></a>                 projDir, outDirBit, setName, setSuf)</span>
<span id="cb1963-3"><a href="pseudoTimeTop.html#cb1963-3"></a>sce.tcell &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>tmpFn)</span>
<span id="cb1963-4"><a href="pseudoTimeTop.html#cb1963-4"></a>sce.tcell</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 21579 20617 
## metadata(0):
## assays(2): counts logcounts
## rownames(21579): AL627309.1 AL669831.5 ... AC233755.1 AC240274.1
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(19): Sample Barcode ... label cell_type
## reducedDimNames(4): MNN PCA UMAP TSNE
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1965"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1965-1"><a href="pseudoTimeTop.html#cb1965-1"></a>dec.tcell &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce.tcell, <span class="dt">block=</span>sce.tcell<span class="op">$</span>Sample.Name)</span>
<span id="cb1965-2"><a href="pseudoTimeTop.html#cb1965-2"></a>top.tcell &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(dec.tcell, <span class="dt">n=</span><span class="dv">5000</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1966"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1966-1"><a href="pseudoTimeTop.html#cb1966-1"></a><span class="kw">set.seed</span>(<span class="dv">1010001</span>)</span>
<span id="cb1966-2"><a href="pseudoTimeTop.html#cb1966-2"></a>merged.tcell &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(sce.tcell, <span class="dt">batch =</span> sce.tcell<span class="op">$</span>Sample.Name, <span class="dt">subset.row =</span> top.tcell)</span>
<span id="cb1966-3"><a href="pseudoTimeTop.html#cb1966-3"></a><span class="kw">reducedDim</span>(sce.tcell, <span class="st">&#39;MNN&#39;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(merged.tcell, <span class="st">&#39;corrected&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1967"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1967-1"><a href="pseudoTimeTop.html#cb1967-1"></a>sce.tcell &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce.tcell, <span class="dt">dimred=</span><span class="st">&quot;MNN&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1968"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1968-1"><a href="pseudoTimeTop.html#cb1968-1"></a><span class="kw">plotPCA</span>(sce.tcell, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/dimRed_plot_pseudotime2-1.png" width="672" /></p>
<div id="trajectory-inference-with-destiny" class="section level3">
<h3><span class="header-section-number">24.1.1</span> Trajectory inference with destiny</h3>
<p><a href="https://en.wikipedia.org/wiki/Diffusion_map">Diffusion maps</a> were introduced by <a href="http://www.sciencedirect.com/science/article/pii/S1063520306000546">Ronald Coifman and Stephane Lafon</a>, and the underlying idea is to assume that the data are samples from a diffusion process. The method infers the low-dimensional manifold by estimating the eigenvalues and eigenvectors for the diffusion operator related to the data. <a href="https://academic.oup.com/bioinformatics/article/32/8/1241/1744143">Angerer et al</a> have applied the diffusion maps concept to the analysis of single-cell RNA-seq data to create an R package called <code>destiny.</code></p>
<p>For ease of computation, we will perform pseudotime analysis only on one sample, and we will downsample the object to 1000 cells. We will select the sample named <code>MantonBM1</code>.</p>
<div class="sourceCode" id="cb1969"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1969-1"><a href="pseudoTimeTop.html#cb1969-1"></a><span class="co"># pull the barcodes for MantonBM1 sample &amp; and downsample the set to 1000 genes </span></span>
<span id="cb1969-2"><a href="pseudoTimeTop.html#cb1969-2"></a>vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.tcell) <span class="op">%&gt;%</span></span>
<span id="cb1969-3"><a href="pseudoTimeTop.html#cb1969-3"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1969-4"><a href="pseudoTimeTop.html#cb1969-4"></a><span class="st">    </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> &quot;MantonBM1&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1969-5"><a href="pseudoTimeTop.html#cb1969-5"></a><span class="st">    </span><span class="kw">group_by</span>(Sample.Name) <span class="op">%&gt;%</span></span>
<span id="cb1969-6"><a href="pseudoTimeTop.html#cb1969-6"></a><span class="st">    </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span></span>
<span id="cb1969-7"><a href="pseudoTimeTop.html#cb1969-7"></a><span class="st">    </span><span class="kw">pull</span>(Barcode)</span></code></pre></div>
<p>Number of cells in the sample:</p>
<div class="sourceCode" id="cb1970"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1970-1"><a href="pseudoTimeTop.html#cb1970-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce.tcell)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
## 19617  1000</code></pre>
<p>Subset cells from the main SCE object:</p>
<div class="sourceCode" id="cb1972"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1972-1"><a href="pseudoTimeTop.html#cb1972-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce.tcell)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1972-2"><a href="pseudoTimeTop.html#cb1972-2"></a>sce.tcell.BM1 &lt;-<span class="st"> </span>sce.tcell[,tmpInd]</span>
<span id="cb1972-3"><a href="pseudoTimeTop.html#cb1972-3"></a>sce.tcell.BM1</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 21579 1000 
## metadata(0):
## assays(2): counts logcounts
## rownames(21579): AL627309.1 AL669831.5 ... AC233755.1 AC240274.1
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(19): Sample Barcode ... label cell_type
## reducedDimNames(4): MNN PCA UMAP TSNE
## altExpNames(0):</code></pre>
<p>Identify top 500 highly variable genes</p>
<div class="sourceCode" id="cb1974"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1974-1"><a href="pseudoTimeTop.html#cb1974-1"></a>dec.tcell.BM1 &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce.tcell.BM1)</span>
<span id="cb1974-2"><a href="pseudoTimeTop.html#cb1974-2"></a>top.tcell.BM1 &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(dec.tcell.BM1, <span class="dt">n=</span><span class="dv">500</span>)</span></code></pre></div>
<p>We will extract normalized counts for HVG to use in pseudotime alignment</p>
<div class="sourceCode" id="cb1975"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1975-1"><a href="pseudoTimeTop.html#cb1975-1"></a>tcell.BM1_counts &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce.tcell.BM1)</span>
<span id="cb1975-2"><a href="pseudoTimeTop.html#cb1975-2"></a>tcell.BM1_counts &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">as.matrix</span>(tcell.BM1_counts[top.tcell.BM1,]))</span>
<span id="cb1975-3"><a href="pseudoTimeTop.html#cb1975-3"></a>cellLabels &lt;-<span class="st"> </span>sce.tcell.BM1<span class="op">$</span>Barcode</span>
<span id="cb1975-4"><a href="pseudoTimeTop.html#cb1975-4"></a><span class="kw">rownames</span>(tcell.BM1_counts) &lt;-<span class="st"> </span>cellLabels</span></code></pre></div>
<div class="sourceCode" id="cb1976"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1976-1"><a href="pseudoTimeTop.html#cb1976-1"></a>tcell.BM1_counts[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</span></code></pre></div>
<pre><code>##                         CCL5   S100A4     NKG7 KLRB1
## AAACCTGAGCAGCGTA-13 3.734481 2.838991 1.609628     0
## AAACCTGAGCGATATA-13 0.000000 3.891127 0.000000     0
## AAACCTGCAAAGCGGT-13 0.000000 0.000000 0.000000     0
## AAACCTGCAAGAGGCT-13 0.000000 1.250165 0.000000     0</code></pre>
<p>And finally, we can run pseudotime alignment with destiny</p>
<div class="sourceCode" id="cb1978"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1978-1"><a href="pseudoTimeTop.html#cb1978-1"></a>dm_tcell_BM1 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(tcell.BM1_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span></code></pre></div>
<p>Plot diffusion component 1 vs diffusion component 2 (DC1 vs DC2).</p>
<div class="sourceCode" id="cb1979"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1979-1"><a href="pseudoTimeTop.html#cb1979-1"></a>tmp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">DC1 =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">1</span>],</span>
<span id="cb1979-2"><a href="pseudoTimeTop.html#cb1979-2"></a>                  <span class="dt">DC2 =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">2</span>])</span>
<span id="cb1979-3"><a href="pseudoTimeTop.html#cb1979-3"></a></span>
<span id="cb1979-4"><a href="pseudoTimeTop.html#cb1979-4"></a><span class="kw">ggplot</span>(tmp, <span class="kw">aes</span>(<span class="dt">x =</span> DC1, <span class="dt">y =</span> DC2)) <span class="op">+</span></span>
<span id="cb1979-5"><a href="pseudoTimeTop.html#cb1979-5"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb1979-6"><a href="pseudoTimeTop.html#cb1979-6"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Diffusion component 1&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1979-7"><a href="pseudoTimeTop.html#cb1979-7"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Diffusion component 2&quot;</span>) <span class="op">+</span></span>
<span id="cb1979-8"><a href="pseudoTimeTop.html#cb1979-8"></a><span class="st">    </span><span class="kw">theme_classic</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/MantonBM1_diffusMap_plot_pseudotime2-1.png" width="672" /></p>
<p>Stash diffusion components to SCE object</p>
<div class="sourceCode" id="cb1980"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1980-1"><a href="pseudoTimeTop.html#cb1980-1"></a>sce.tcell.BM1<span class="op">$</span>pseudotime_destiny_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">1</span>]</span>
<span id="cb1980-2"><a href="pseudoTimeTop.html#cb1980-2"></a>sce.tcell.BM1<span class="op">$</span>pseudotime_destiny_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">2</span>]</span></code></pre></div>
</div>
<div id="find-temporally-expressed-genes" class="section level3">
<h3><span class="header-section-number">24.1.2</span> Find temporally expressed genes</h3>
<p>After running destiny, an interesting next step may be to find genes that change their expression over the course of time We demonstrate one possible method for this type of analysis on the 500 most variable genes. We will regress each gene on the pseudotime variable we have generated, using a general additive model (GAM). This allows us to detect non-linear patterns in gene expression. We are going to use HVG we identified in the previous step, but this analysis can also be done using the whole transcriptome.</p>
<div class="sourceCode" id="cb1981"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1981-1"><a href="pseudoTimeTop.html#cb1981-1"></a><span class="co"># Only look at the 500 most variable genes when identifying temporally expressesd genes.</span></span>
<span id="cb1981-2"><a href="pseudoTimeTop.html#cb1981-2"></a><span class="co"># Identify the variable genes by ranking all genes by their variance.</span></span>
<span id="cb1981-3"><a href="pseudoTimeTop.html#cb1981-3"></a><span class="co"># We will use the first diffusion components as a measure of pseudotime </span></span>
<span id="cb1981-4"><a href="pseudoTimeTop.html#cb1981-4"></a>Y&lt;-<span class="kw">log2</span>(<span class="kw">counts</span>(sce.tcell.BM1)<span class="op">+</span><span class="dv">1</span>)</span>
<span id="cb1981-5"><a href="pseudoTimeTop.html#cb1981-5"></a><span class="kw">colnames</span>(Y)&lt;-cellLabels</span>
<span id="cb1981-6"><a href="pseudoTimeTop.html#cb1981-6"></a>Y&lt;-Y[top.tcell.BM1,]</span>
<span id="cb1981-7"><a href="pseudoTimeTop.html#cb1981-7"></a><span class="co"># Fit GAM for each gene using pseudotime as independent variable.</span></span>
<span id="cb1981-8"><a href="pseudoTimeTop.html#cb1981-8"></a>t &lt;-<span class="st"> </span><span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">1</span>]</span>
<span id="cb1981-9"><a href="pseudoTimeTop.html#cb1981-9"></a>gam.pval &lt;-<span class="st"> </span><span class="kw">apply</span>(Y, <span class="dv">1</span>, <span class="cf">function</span>(z){</span>
<span id="cb1981-10"><a href="pseudoTimeTop.html#cb1981-10"></a>  d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">z=</span>z, <span class="dt">t=</span>t)</span>
<span id="cb1981-11"><a href="pseudoTimeTop.html#cb1981-11"></a>  tmp &lt;-<span class="st"> </span><span class="kw">gam</span>(z <span class="op">~</span><span class="st"> </span><span class="kw">lo</span>(t), <span class="dt">data=</span>d)</span>
<span id="cb1981-12"><a href="pseudoTimeTop.html#cb1981-12"></a>  p &lt;-<span class="st"> </span><span class="kw">summary</span>(tmp)[<span class="dv">4</span>][[<span class="dv">1</span>]][<span class="dv">1</span>,<span class="dv">5</span>]</span>
<span id="cb1981-13"><a href="pseudoTimeTop.html#cb1981-13"></a>  p</span>
<span id="cb1981-14"><a href="pseudoTimeTop.html#cb1981-14"></a>})</span></code></pre></div>
<p>Select top 30 genes for visualization</p>
<div class="sourceCode" id="cb1982"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1982-1"><a href="pseudoTimeTop.html#cb1982-1"></a><span class="co"># Identify genes with the most significant time-dependent model fit.</span></span>
<span id="cb1982-2"><a href="pseudoTimeTop.html#cb1982-2"></a>topgenes &lt;-<span class="st"> </span><span class="kw">names</span>(<span class="kw">sort</span>(gam.pval, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>))[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]  </span></code></pre></div>
<p>Visualize these genes in a heatmap</p>
<div class="sourceCode" id="cb1983"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1983-1"><a href="pseudoTimeTop.html#cb1983-1"></a>heatmapdata &lt;-<span class="st"> </span>Y[topgenes,]</span>
<span id="cb1983-2"><a href="pseudoTimeTop.html#cb1983-2"></a>heatmapdata &lt;-<span class="st"> </span>heatmapdata[,<span class="kw">order</span>(t, <span class="dt">na.last =</span> <span class="ot">NA</span>)]</span>
<span id="cb1983-3"><a href="pseudoTimeTop.html#cb1983-3"></a>t_ann&lt;-<span class="kw">as.data.frame</span>(t)</span>
<span id="cb1983-4"><a href="pseudoTimeTop.html#cb1983-4"></a><span class="kw">colnames</span>(t_ann)&lt;-<span class="st">&quot;pseudotime&quot;</span></span>
<span id="cb1983-5"><a href="pseudoTimeTop.html#cb1983-5"></a><span class="kw">pheatmap</span>(heatmapdata, <span class="dt">cluster_rows =</span> T, <span class="dt">cluster_cols =</span> F, <span class="dt">color =</span> <span class="kw">plasma</span>(<span class="dv">200</span>), <span class="dt">show_colnames =</span> F, <span class="dt">annotation_col =</span> t_ann)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/MantonBM1_tempExprGenes_heatmap_pseudotime2-1.png" width="672" /></p>
<p><strong>Visualize how some of the temporally expressed genes change in time</strong></p>
<p>Following individual genes is very helpful for identifying genes that play an important role in the differentiation process. We illustrate the procedure using the GZMA gene. We have added the pseudotime values computed with destiny to the colData slot of the SCE object. Having done that, the full plotting capabilities of the scater package can be used to investigate relationships between gene expression, cell populations and pseudotime.</p>
<div class="sourceCode" id="cb1984"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1984-1"><a href="pseudoTimeTop.html#cb1984-1"></a><span class="kw">plotExpression</span>(sce.tcell.BM1,</span>
<span id="cb1984-2"><a href="pseudoTimeTop.html#cb1984-2"></a>           <span class="st">&quot;GZMA&quot;</span>,</span>
<span id="cb1984-3"><a href="pseudoTimeTop.html#cb1984-3"></a>           <span class="dt">x =</span> <span class="st">&quot;pseudotime_destiny_1&quot;</span>, </span>
<span id="cb1984-4"><a href="pseudoTimeTop.html#cb1984-4"></a>               <span class="dt">show_violin =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1984-5"><a href="pseudoTimeTop.html#cb1984-5"></a>               <span class="dt">show_smooth =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/MantonBM1_tempExprGenes_GZMA_pseudotime2-1.png" width="672" /></p>
</div>
<div id="pseudotime-analysis-for-another-hca-sample" class="section level3">
<h3><span class="header-section-number">24.1.3</span> Pseudotime analysis for another HCA sample</h3>
<div class="sourceCode" id="cb1985"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1985-1"><a href="pseudoTimeTop.html#cb1985-1"></a><span class="co"># pull barcodes for MantonBM2 </span></span>
<span id="cb1985-2"><a href="pseudoTimeTop.html#cb1985-2"></a>vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce.tcell) <span class="op">%&gt;%</span></span>
<span id="cb1985-3"><a href="pseudoTimeTop.html#cb1985-3"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1985-4"><a href="pseudoTimeTop.html#cb1985-4"></a><span class="st">    </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> &quot;MantonBM2&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1985-5"><a href="pseudoTimeTop.html#cb1985-5"></a><span class="st">    </span><span class="kw">group_by</span>(Sample.Name) <span class="op">%&gt;%</span></span>
<span id="cb1985-6"><a href="pseudoTimeTop.html#cb1985-6"></a><span class="st">    </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span></span>
<span id="cb1985-7"><a href="pseudoTimeTop.html#cb1985-7"></a><span class="st">    </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb1985-8"><a href="pseudoTimeTop.html#cb1985-8"></a></span>
<span id="cb1985-9"><a href="pseudoTimeTop.html#cb1985-9"></a><span class="co"># create another object for MantonBM2</span></span>
<span id="cb1985-10"><a href="pseudoTimeTop.html#cb1985-10"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce.tcell)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1985-11"><a href="pseudoTimeTop.html#cb1985-11"></a>sce.tcell.BM2 &lt;-<span class="st"> </span>sce.tcell[,tmpInd]</span>
<span id="cb1985-12"><a href="pseudoTimeTop.html#cb1985-12"></a></span>
<span id="cb1985-13"><a href="pseudoTimeTop.html#cb1985-13"></a><span class="co"># Identift HVG</span></span>
<span id="cb1985-14"><a href="pseudoTimeTop.html#cb1985-14"></a>dec.tcell.BM2 &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce.tcell.BM2)</span>
<span id="cb1985-15"><a href="pseudoTimeTop.html#cb1985-15"></a>top.tcell.BM2 &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(dec.tcell.BM2, <span class="dt">n=</span><span class="dv">500</span>)</span>
<span id="cb1985-16"><a href="pseudoTimeTop.html#cb1985-16"></a></span>
<span id="cb1985-17"><a href="pseudoTimeTop.html#cb1985-17"></a><span class="co"># extract normalized count data for HVG </span></span>
<span id="cb1985-18"><a href="pseudoTimeTop.html#cb1985-18"></a>tcell.BM2_counts&lt;-<span class="kw">logcounts</span>(sce.tcell.BM2)</span>
<span id="cb1985-19"><a href="pseudoTimeTop.html#cb1985-19"></a>tcell.BM2_counts&lt;-<span class="kw">t</span>(<span class="kw">as.matrix</span>(tcell.BM2_counts[top.tcell.BM2,]))</span>
<span id="cb1985-20"><a href="pseudoTimeTop.html#cb1985-20"></a>cellLabels &lt;-<span class="st"> </span>sce.tcell.BM2<span class="op">$</span>Barcode</span>
<span id="cb1985-21"><a href="pseudoTimeTop.html#cb1985-21"></a><span class="kw">rownames</span>(tcell.BM2_counts) &lt;-<span class="st"> </span>cellLabels</span>
<span id="cb1985-22"><a href="pseudoTimeTop.html#cb1985-22"></a></span>
<span id="cb1985-23"><a href="pseudoTimeTop.html#cb1985-23"></a>dm_tcell_BM2 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(tcell.BM2_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span>
<span id="cb1985-24"><a href="pseudoTimeTop.html#cb1985-24"></a></span>
<span id="cb1985-25"><a href="pseudoTimeTop.html#cb1985-25"></a>tmp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">DC1 =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM2)[, <span class="dv">1</span>],</span>
<span id="cb1985-26"><a href="pseudoTimeTop.html#cb1985-26"></a>                  <span class="dt">DC2 =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM2)[, <span class="dv">2</span>])</span>
<span id="cb1985-27"><a href="pseudoTimeTop.html#cb1985-27"></a></span>
<span id="cb1985-28"><a href="pseudoTimeTop.html#cb1985-28"></a><span class="kw">ggplot</span>(tmp, <span class="kw">aes</span>(<span class="dt">x =</span> DC1, <span class="dt">y =</span> DC2)) <span class="op">+</span></span>
<span id="cb1985-29"><a href="pseudoTimeTop.html#cb1985-29"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb1985-30"><a href="pseudoTimeTop.html#cb1985-30"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Diffusion component 1&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1985-31"><a href="pseudoTimeTop.html#cb1985-31"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Diffusion component 2&quot;</span>) <span class="op">+</span></span>
<span id="cb1985-32"><a href="pseudoTimeTop.html#cb1985-32"></a><span class="st">    </span><span class="kw">theme_classic</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/MantonBM2_preProc_pseudotime2-1.png" width="672" /></p>
<div class="sourceCode" id="cb1986"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1986-1"><a href="pseudoTimeTop.html#cb1986-1"></a><span class="co"># tidy</span></span>
<span id="cb1986-2"><a href="pseudoTimeTop.html#cb1986-2"></a><span class="kw">rm</span>(sce.tcell)</span></code></pre></div>
</div>
<div id="exercise-1" class="section level3">
<h3><span class="header-section-number">24.1.4</span> Exercise 1</h3>
<p>Obtain pseudotime for one of the Caron samples.</p>
<div class="sourceCode" id="cb1987"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1987-1"><a href="pseudoTimeTop.html#cb1987-1"></a><span class="co">#sce.PRET1&lt;-readRDS(&quot;~/Course_Materials/scRNAseq/pseudotime/sce_caron_PRET1.RDS&quot;)</span></span>
<span id="cb1987-2"><a href="pseudoTimeTop.html#cb1987-2"></a>setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1987-3"><a href="pseudoTimeTop.html#cb1987-3"></a>setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb1987-4"><a href="pseudoTimeTop.html#cb1987-4"></a><span class="co">#tmpFn &lt;- sprintf(&quot;%s/%s/Robjects/%s_sce_nz_postQc%s.Rds&quot;, # no norm counts yet</span></span>
<span id="cb1987-5"><a href="pseudoTimeTop.html#cb1987-5"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>,</span>
<span id="cb1987-6"><a href="pseudoTimeTop.html#cb1987-6"></a>                 projDir, outDirBit, setName, setSuf)</span>
<span id="cb1987-7"><a href="pseudoTimeTop.html#cb1987-7"></a>x &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1987-8"><a href="pseudoTimeTop.html#cb1987-8"></a>spl.x &lt;-<span class="st"> </span><span class="kw">colData</span>(x) <span class="op">%&gt;%</span></span>
<span id="cb1987-9"><a href="pseudoTimeTop.html#cb1987-9"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1987-10"><a href="pseudoTimeTop.html#cb1987-10"></a><span class="st">    </span><span class="kw">filter</span>(source_name <span class="op">==</span><span class="st"> &quot;PRE-T&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1987-11"><a href="pseudoTimeTop.html#cb1987-11"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name) <span class="op">%&gt;%</span></span>
<span id="cb1987-12"><a href="pseudoTimeTop.html#cb1987-12"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>(<span class="dv">1</span>)</span></code></pre></div>
<p>We will use GSM3872440:</p>
<div class="sourceCode" id="cb1988"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1988-1"><a href="pseudoTimeTop.html#cb1988-1"></a><span class="co"># downsample by randomly choosing 1000 cells:</span></span>
<span id="cb1988-2"><a href="pseudoTimeTop.html#cb1988-2"></a>vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(x) <span class="op">%&gt;%</span></span>
<span id="cb1988-3"><a href="pseudoTimeTop.html#cb1988-3"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1988-4"><a href="pseudoTimeTop.html#cb1988-4"></a><span class="st">  </span><span class="kw">filter</span>(Sample.Name <span class="op">==</span><span class="st"> </span>spl.x) <span class="op">%&gt;%</span></span>
<span id="cb1988-5"><a href="pseudoTimeTop.html#cb1988-5"></a><span class="st">  </span><span class="kw">group_by</span>(Sample.Name) <span class="op">%&gt;%</span></span>
<span id="cb1988-6"><a href="pseudoTimeTop.html#cb1988-6"></a><span class="st">  </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span></span>
<span id="cb1988-7"><a href="pseudoTimeTop.html#cb1988-7"></a><span class="st">  </span><span class="kw">pull</span>(Barcode)</span></code></pre></div>
<p>Number of cells in the sample:</p>
<div class="sourceCode" id="cb1989"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1989-1"><a href="pseudoTimeTop.html#cb1989-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(x)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
## 46830  1000</code></pre>
<p>Subset cells from the main SCE object:</p>
<div class="sourceCode" id="cb1991"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1991-1"><a href="pseudoTimeTop.html#cb1991-1"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(x)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb1991-2"><a href="pseudoTimeTop.html#cb1991-2"></a>sce.PRET1 &lt;-<span class="st"> </span>x[,tmpInd]</span>
<span id="cb1991-3"><a href="pseudoTimeTop.html#cb1991-3"></a><span class="kw">rownames</span>(sce.PRET1) &lt;-<span class="st"> </span><span class="kw">uniquifyFeatureNames</span>(<span class="kw">rowData</span>(sce.PRET1)<span class="op">$</span>ensembl_gene_id,</span>
<span id="cb1991-4"><a href="pseudoTimeTop.html#cb1991-4"></a>                                            <span class="dt">names =</span> <span class="kw">rowData</span>(sce.PRET1)<span class="op">$</span>Symbol)</span>
<span id="cb1991-5"><a href="pseudoTimeTop.html#cb1991-5"></a></span>
<span id="cb1991-6"><a href="pseudoTimeTop.html#cb1991-6"></a>sce.PRET1</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 1000 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): AL627309.1 AL669831.5 ... AC233755.1 AC240274.1
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb1993"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1993-1"><a href="pseudoTimeTop.html#cb1993-1"></a><span class="kw">rm</span>(x, vec.bc)</span></code></pre></div>
<p>You need to perform:
* variance remodelling
* HVG identification
* extract normalized counts
* run destiny
* visualize diffusion components</p>
<div class="sourceCode" id="cb1994"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1994-1"><a href="pseudoTimeTop.html#cb1994-1"></a><span class="co"># Identify HVG</span></span>
<span id="cb1994-2"><a href="pseudoTimeTop.html#cb1994-2"></a>dec.caron.PRET1 &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce.PRET1)</span>
<span id="cb1994-3"><a href="pseudoTimeTop.html#cb1994-3"></a>top.caron.PRET1 &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(dec.caron.PRET1, <span class="dt">n=</span><span class="dv">500</span>)</span>
<span id="cb1994-4"><a href="pseudoTimeTop.html#cb1994-4"></a></span>
<span id="cb1994-5"><a href="pseudoTimeTop.html#cb1994-5"></a><span class="co"># extract normalized count data for HVG </span></span>
<span id="cb1994-6"><a href="pseudoTimeTop.html#cb1994-6"></a>caron.PRET1_counts &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce.PRET1)</span>
<span id="cb1994-7"><a href="pseudoTimeTop.html#cb1994-7"></a>caron.PRET1_counts &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">as.matrix</span>(caron.PRET1_counts[top.caron.PRET1,]))</span>
<span id="cb1994-8"><a href="pseudoTimeTop.html#cb1994-8"></a>cellLabels &lt;-<span class="st"> </span>sce.PRET1<span class="op">$</span>Barcode</span>
<span id="cb1994-9"><a href="pseudoTimeTop.html#cb1994-9"></a><span class="kw">rownames</span>(caron.PRET1_counts) &lt;-<span class="st"> </span>cellLabels</span>
<span id="cb1994-10"><a href="pseudoTimeTop.html#cb1994-10"></a></span>
<span id="cb1994-11"><a href="pseudoTimeTop.html#cb1994-11"></a>dm_caron.PRET1 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(caron.PRET1_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span>
<span id="cb1994-12"><a href="pseudoTimeTop.html#cb1994-12"></a></span>
<span id="cb1994-13"><a href="pseudoTimeTop.html#cb1994-13"></a>tmp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">DC1 =</span> <span class="kw">eigenvectors</span>(dm_caron.PRET1)[, <span class="dv">1</span>],</span>
<span id="cb1994-14"><a href="pseudoTimeTop.html#cb1994-14"></a>                  <span class="dt">DC2 =</span> <span class="kw">eigenvectors</span>(dm_caron.PRET1)[, <span class="dv">2</span>])</span>
<span id="cb1994-15"><a href="pseudoTimeTop.html#cb1994-15"></a></span>
<span id="cb1994-16"><a href="pseudoTimeTop.html#cb1994-16"></a><span class="kw">ggplot</span>(tmp, <span class="kw">aes</span>(<span class="dt">x =</span> DC1, <span class="dt">y =</span> DC2)) <span class="op">+</span></span>
<span id="cb1994-17"><a href="pseudoTimeTop.html#cb1994-17"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb1994-18"><a href="pseudoTimeTop.html#cb1994-18"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Diffusion component 1&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1994-19"><a href="pseudoTimeTop.html#cb1994-19"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Diffusion component 2&quot;</span>) <span class="op">+</span></span>
<span id="cb1994-20"><a href="pseudoTimeTop.html#cb1994-20"></a><span class="st">    </span><span class="kw">theme_classic</span>()</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/PRET1_analysis_pseudotime2-1.png" width="672" /></p>
<div class="sourceCode" id="cb1995"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1995-1"><a href="pseudoTimeTop.html#cb1995-1"></a>Y &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">counts</span>(sce.PRET1)<span class="op">+</span><span class="dv">1</span>)</span>
<span id="cb1995-2"><a href="pseudoTimeTop.html#cb1995-2"></a><span class="kw">colnames</span>(Y) &lt;-<span class="st"> </span>cellLabels</span>
<span id="cb1995-3"><a href="pseudoTimeTop.html#cb1995-3"></a>Y &lt;-<span class="st"> </span>Y[top.caron.PRET1,]</span>
<span id="cb1995-4"><a href="pseudoTimeTop.html#cb1995-4"></a><span class="co"># Fit GAM for each gene using pseudotime as independent variable.</span></span>
<span id="cb1995-5"><a href="pseudoTimeTop.html#cb1995-5"></a>t &lt;-<span class="st"> </span><span class="kw">eigenvectors</span>(dm_caron.PRET1)[, <span class="dv">1</span>]</span>
<span id="cb1995-6"><a href="pseudoTimeTop.html#cb1995-6"></a>gam.pval &lt;-<span class="st"> </span><span class="kw">apply</span>(Y, <span class="dv">1</span>, <span class="cf">function</span>(z){</span>
<span id="cb1995-7"><a href="pseudoTimeTop.html#cb1995-7"></a>  d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">z=</span>z, <span class="dt">t=</span>t)</span>
<span id="cb1995-8"><a href="pseudoTimeTop.html#cb1995-8"></a>  tmp &lt;-<span class="st"> </span><span class="kw">gam</span>(z <span class="op">~</span><span class="st"> </span><span class="kw">lo</span>(t), <span class="dt">data=</span>d)</span>
<span id="cb1995-9"><a href="pseudoTimeTop.html#cb1995-9"></a>  p &lt;-<span class="st"> </span><span class="kw">summary</span>(tmp)[<span class="dv">4</span>][[<span class="dv">1</span>]][<span class="dv">1</span>,<span class="dv">5</span>]</span>
<span id="cb1995-10"><a href="pseudoTimeTop.html#cb1995-10"></a>  p</span>
<span id="cb1995-11"><a href="pseudoTimeTop.html#cb1995-11"></a>})</span></code></pre></div>
<p>Select top 30 genes for visualization</p>
<div class="sourceCode" id="cb1996"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1996-1"><a href="pseudoTimeTop.html#cb1996-1"></a><span class="co"># Identify genes with the most significant time-dependent model fit.</span></span>
<span id="cb1996-2"><a href="pseudoTimeTop.html#cb1996-2"></a>topgenes &lt;-<span class="st"> </span><span class="kw">names</span>(<span class="kw">sort</span>(gam.pval, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>))[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]  </span></code></pre></div>
<p>Visualize these genes in a heatmap</p>
<div class="sourceCode" id="cb1997"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1997-1"><a href="pseudoTimeTop.html#cb1997-1"></a>heatmapdata &lt;-<span class="st"> </span>Y[topgenes,]</span>
<span id="cb1997-2"><a href="pseudoTimeTop.html#cb1997-2"></a>heatmapdata &lt;-<span class="st"> </span>heatmapdata[,<span class="kw">order</span>(t, <span class="dt">na.last =</span> <span class="ot">NA</span>)]</span>
<span id="cb1997-3"><a href="pseudoTimeTop.html#cb1997-3"></a>t_ann&lt;-<span class="kw">as.data.frame</span>(t)</span>
<span id="cb1997-4"><a href="pseudoTimeTop.html#cb1997-4"></a><span class="kw">colnames</span>(t_ann)&lt;-<span class="st">&quot;pseudotime&quot;</span></span>
<span id="cb1997-5"><a href="pseudoTimeTop.html#cb1997-5"></a><span class="kw">pheatmap</span>(heatmapdata, <span class="dt">cluster_rows =</span> T, <span class="dt">cluster_cols =</span> F, <span class="dt">color =</span> <span class="kw">plasma</span>(<span class="dv">200</span>), <span class="dt">show_colnames =</span> F, <span class="dt">annotation_col =</span> t_ann)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/PRET1_tempExprGenes_heatmap_pseudotime2-1.png" width="672" /></p>
<p><strong>What kind of a dynamic process might be taking place in this cancer cell?</strong></p>
<p>We can quickly check in which pathways these top genes are enriched using MSigDB.</p>
<p>Molecular Signatures Database contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>We are going to use hallmark gene sets (<code>H</code>) and perform a hypergeometric test with our top 30 genes for all HALLMARK sets.</p>
<div class="sourceCode" id="cb1998"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1998-1"><a href="pseudoTimeTop.html#cb1998-1"></a>msigdb_hallmark &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>,</span>
<span id="cb1998-2"><a href="pseudoTimeTop.html#cb1998-2"></a>                         <span class="dt">category =</span> <span class="st">&quot;H&quot;</span>)  <span class="op">%&gt;%</span></span>
<span id="cb1998-3"><a href="pseudoTimeTop.html#cb1998-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, gene_symbol)</span>
<span id="cb1998-4"><a href="pseudoTimeTop.html#cb1998-4"></a>em &lt;-<span class="st"> </span><span class="kw">enricher</span>(topgenes,</span>
<span id="cb1998-5"><a href="pseudoTimeTop.html#cb1998-5"></a>               <span class="dt">TERM2GENE =</span> msigdb_hallmark)</span>
<span id="cb1998-6"><a href="pseudoTimeTop.html#cb1998-6"></a><span class="kw">head</span>(em)[,<span class="st">&quot;qvalue&quot;</span>,drop=F]</span></code></pre></div>
<pre><code>##                                qvalue
## HALLMARK_E2F_TARGETS     2.744673e-17
## HALLMARK_G2M_CHECKPOINT  4.008755e-14
## HALLMARK_MITOTIC_SPINDLE 5.776092e-03</code></pre>
<p>Let’s also check the ‘oncogenic signatures’ gene sets (<code>C6</code>) and perform a hypergeometric test with our top 30 genes for all these gene sets.</p>
<div class="sourceCode" id="cb2000"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2000-1"><a href="pseudoTimeTop.html#cb2000-1"></a>msigdb_c6 &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>,</span>
<span id="cb2000-2"><a href="pseudoTimeTop.html#cb2000-2"></a>                         <span class="dt">category =</span> <span class="st">&quot;C6&quot;</span>)  <span class="op">%&gt;%</span></span>
<span id="cb2000-3"><a href="pseudoTimeTop.html#cb2000-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, gene_symbol)</span>
<span id="cb2000-4"><a href="pseudoTimeTop.html#cb2000-4"></a>em &lt;-<span class="st"> </span><span class="kw">enricher</span>(topgenes,</span>
<span id="cb2000-5"><a href="pseudoTimeTop.html#cb2000-5"></a>               <span class="dt">TERM2GENE =</span> msigdb_c6)</span>
<span id="cb2000-6"><a href="pseudoTimeTop.html#cb2000-6"></a><span class="kw">head</span>(em)[,<span class="st">&quot;qvalue&quot;</span>,drop=F]</span></code></pre></div>
<pre><code>##                                          qvalue
## GCNP_SHH_UP_LATE.V1_UP             6.029194e-06
## CORDENONSI_YAP_CONSERVED_SIGNATURE 1.937933e-05
## VEGF_A_UP.V1_DN                    7.781542e-05
## RB_P107_DN.V1_UP                   2.584490e-04
## CSR_LATE_UP.V1_UP                  7.412233e-03
## MTOR_UP.V1_UP                      7.412233e-03</code></pre>
<p>Let’s write to file for future use the normalised counts for the highly variables genes:</p>
<div class="sourceCode" id="cb2002"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2002-1"><a href="pseudoTimeTop.html#cb2002-1"></a>setName &lt;-<span class="st"> &quot;hca&quot;</span></span>
<span id="cb2002-2"><a href="pseudoTimeTop.html#cb2002-2"></a>setSuf &lt;-<span class="st"> &quot;_5kCellPerSpl&quot;</span></span>
<span id="cb2002-3"><a href="pseudoTimeTop.html#cb2002-3"></a></span>
<span id="cb2002-4"><a href="pseudoTimeTop.html#cb2002-4"></a><span class="co"># BM1</span></span>
<span id="cb2002-5"><a href="pseudoTimeTop.html#cb2002-5"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2002-6"><a href="pseudoTimeTop.html#cb2002-6"></a>                 projDir, outDirBit, setName, setSuf, <span class="st">&quot;MantonBM1&quot;</span>)</span>
<span id="cb2002-7"><a href="pseudoTimeTop.html#cb2002-7"></a><span class="kw">saveRDS</span>(tcell.BM1_counts, <span class="dt">file=</span>tmpFn)</span>
<span id="cb2002-8"><a href="pseudoTimeTop.html#cb2002-8"></a></span>
<span id="cb2002-9"><a href="pseudoTimeTop.html#cb2002-9"></a><span class="co"># BM2</span></span>
<span id="cb2002-10"><a href="pseudoTimeTop.html#cb2002-10"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2002-11"><a href="pseudoTimeTop.html#cb2002-11"></a>                 projDir, outDirBit, setName, setSuf, <span class="st">&quot;MantonBM2&quot;</span>)</span>
<span id="cb2002-12"><a href="pseudoTimeTop.html#cb2002-12"></a><span class="kw">saveRDS</span>(tcell.BM2_counts, <span class="dt">file=</span>tmpFn)</span>
<span id="cb2002-13"><a href="pseudoTimeTop.html#cb2002-13"></a></span>
<span id="cb2002-14"><a href="pseudoTimeTop.html#cb2002-14"></a><span class="co"># PRE-T 1</span></span>
<span id="cb2002-15"><a href="pseudoTimeTop.html#cb2002-15"></a>setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb2002-16"><a href="pseudoTimeTop.html#cb2002-16"></a>setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb2002-17"><a href="pseudoTimeTop.html#cb2002-17"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2002-18"><a href="pseudoTimeTop.html#cb2002-18"></a>                 projDir, outDirBit, setName, setSuf, spl.x)</span>
<span id="cb2002-19"><a href="pseudoTimeTop.html#cb2002-19"></a><span class="kw">saveRDS</span>(caron.PRET1_counts, <span class="dt">file=</span>tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb2003"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2003-1"><a href="pseudoTimeTop.html#cb2003-1"></a><span class="kw">rm</span>(sce.tcell.BM1, sce.tcell.BM2, sce.tcell.PRET1)</span>
<span id="cb2003-2"><a href="pseudoTimeTop.html#cb2003-2"></a><span class="kw">rm</span>(tcell.BM1_counts, tcell.BM2_counts, caron.PRET1_counts)</span></code></pre></div>
</div>
</div>
<div id="pseudoTime3Top" class="section level2">
<h2><span class="header-section-number">24.2</span> <strong>Pseudotime Alignment</strong></h2>
<p>CellAlign is a tool for quantitative comparison of expression dynamics within or between single-cell trajectories. The input to the CellAlign workflow is any trajectory vector that orders single cell expression with a pseudo-time spacing and the expression matrix for the cells used to define the trajectory. cellAlign has 3 essential steps:</p>
<ol style="list-style-type: decimal">
<li><p>Interpolate the data to have N evenly spaced points along the scaled pseudotime vector using a sliding window of Gaussian weights</p></li>
<li><p>Determine the genes of interest for alignment</p></li>
<li><p>Align your trajectory among the selected genes either along the whole trajectory or along a partial segment.</p></li>
</ol>
<p>The first step is to interpolate the data along the trajectory to represent the data by N (default 200) equally spaced points along the pseudotime trajectory. We included this step because single-cell measurements are often sparse or heterogeneous along the trajectory, leaving gaps that cannot be aligned. Cell-Align interpolates the gene-expression values of equally spaced artificial points using the real single-cell expression data. The expression values of the interpolated points are calculated using all cells, with each single cell assigned a weight given by a Gaussian distribution centered at the interpolated point and a width assigned by a parameter called <code>winSz</code>. The default <code>winSz</code> is 0.1, as this is the range that preserves the dynamics of the trajectory without including excessive noise for standard single cell data sets.</p>
<p>We will use the samples analysed in the previous session: two ABMMC and one PRE-T:</p>
<div class="sourceCode" id="cb2004"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2004-1"><a href="pseudoTimeTop.html#cb2004-1"></a><span class="co"># HCA ABMMCs:</span></span>
<span id="cb2004-2"><a href="pseudoTimeTop.html#cb2004-2"></a>setName &lt;-<span class="st"> &quot;hca&quot;</span></span>
<span id="cb2004-3"><a href="pseudoTimeTop.html#cb2004-3"></a>setSuf &lt;-<span class="st"> &quot;_5kCellPerSpl&quot;</span></span>
<span id="cb2004-4"><a href="pseudoTimeTop.html#cb2004-4"></a></span>
<span id="cb2004-5"><a href="pseudoTimeTop.html#cb2004-5"></a><span class="co"># BM1</span></span>
<span id="cb2004-6"><a href="pseudoTimeTop.html#cb2004-6"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2004-7"><a href="pseudoTimeTop.html#cb2004-7"></a>                 projDir, outDirBit, setName, setSuf, <span class="st">&quot;MantonBM1&quot;</span>)</span>
<span id="cb2004-8"><a href="pseudoTimeTop.html#cb2004-8"></a>tcell.BM1_counts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb2004-9"><a href="pseudoTimeTop.html#cb2004-9"></a></span>
<span id="cb2004-10"><a href="pseudoTimeTop.html#cb2004-10"></a><span class="co"># BM2</span></span>
<span id="cb2004-11"><a href="pseudoTimeTop.html#cb2004-11"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2004-12"><a href="pseudoTimeTop.html#cb2004-12"></a>                 projDir, outDirBit, setName, setSuf, <span class="st">&quot;MantonBM2&quot;</span>)</span>
<span id="cb2004-13"><a href="pseudoTimeTop.html#cb2004-13"></a>tcell.BM2_counts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb2004-14"><a href="pseudoTimeTop.html#cb2004-14"></a></span>
<span id="cb2004-15"><a href="pseudoTimeTop.html#cb2004-15"></a><span class="co"># Caron PRE-T 1</span></span>
<span id="cb2004-16"><a href="pseudoTimeTop.html#cb2004-16"></a>setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb2004-17"><a href="pseudoTimeTop.html#cb2004-17"></a>setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb2004-18"><a href="pseudoTimeTop.html#cb2004-18"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_%s.Rds&quot;</span>,</span>
<span id="cb2004-19"><a href="pseudoTimeTop.html#cb2004-19"></a>                 projDir, outDirBit, setName, setSuf, <span class="st">&quot;GSM3872440&quot;</span>)</span>
<span id="cb2004-20"><a href="pseudoTimeTop.html#cb2004-20"></a>caron.PRET1_counts &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
<div class="sourceCode" id="cb2005"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2005-1"><a href="pseudoTimeTop.html#cb2005-1"></a>dm_tcell_BM1 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(tcell.BM1_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span>
<span id="cb2005-2"><a href="pseudoTimeTop.html#cb2005-2"></a>interGlobal_hcaBM1 &lt;-<span class="st"> </span><span class="kw">interWeights</span>(<span class="dt">expDataBatch =</span> <span class="kw">t</span>(tcell.BM1_counts), </span>
<span id="cb2005-3"><a href="pseudoTimeTop.html#cb2005-3"></a>                                                    <span class="dt">trajCond =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM1)[, <span class="dv">1</span>], </span>
<span id="cb2005-4"><a href="pseudoTimeTop.html#cb2005-4"></a>                                                    <span class="dt">winSz =</span> <span class="fl">0.1</span>, <span class="dt">numPts=</span><span class="dv">200</span>)</span>
<span id="cb2005-5"><a href="pseudoTimeTop.html#cb2005-5"></a></span>
<span id="cb2005-6"><a href="pseudoTimeTop.html#cb2005-6"></a>dm_tcell_BM2 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(tcell.BM2_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span>
<span id="cb2005-7"><a href="pseudoTimeTop.html#cb2005-7"></a>interGlobal_hcaBM2 &lt;-<span class="st"> </span><span class="kw">interWeights</span>(<span class="dt">expDataBatch =</span> <span class="kw">t</span>(tcell.BM2_counts), </span>
<span id="cb2005-8"><a href="pseudoTimeTop.html#cb2005-8"></a>                                                    <span class="dt">trajCond =</span> <span class="kw">eigenvectors</span>(dm_tcell_BM2)[, <span class="dv">1</span>], </span>
<span id="cb2005-9"><a href="pseudoTimeTop.html#cb2005-9"></a>                                                    <span class="dt">winSz =</span> <span class="fl">0.1</span>, <span class="dt">numPts=</span><span class="dv">200</span>)</span>
<span id="cb2005-10"><a href="pseudoTimeTop.html#cb2005-10"></a></span>
<span id="cb2005-11"><a href="pseudoTimeTop.html#cb2005-11"></a>dm_caron.PRET1 &lt;-<span class="st"> </span><span class="kw">DiffusionMap</span>(caron.PRET1_counts,<span class="dt">n_pcs =</span> <span class="dv">50</span>)</span>
<span id="cb2005-12"><a href="pseudoTimeTop.html#cb2005-12"></a>interGlobal_caronPRET1 &lt;-<span class="st"> </span><span class="kw">interWeights</span>(<span class="dt">expDataBatch =</span> <span class="kw">t</span>(caron.PRET1_counts), </span>
<span id="cb2005-13"><a href="pseudoTimeTop.html#cb2005-13"></a>                                                    <span class="dt">trajCond =</span> <span class="kw">eigenvectors</span>(dm_caron.PRET1)[, <span class="dv">1</span>], </span>
<span id="cb2005-14"><a href="pseudoTimeTop.html#cb2005-14"></a>                                                    <span class="dt">winSz =</span> <span class="fl">0.1</span>, <span class="dt">numPts=</span><span class="dv">200</span>)</span></code></pre></div>
<p>Scale the expression matrix</p>
<div class="sourceCode" id="cb2006"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2006-1"><a href="pseudoTimeTop.html#cb2006-1"></a>interGlobal_caronPRET1_scaled =<span class="st"> </span><span class="kw">scaleInterpolate</span>(interGlobal_caronPRET1)</span>
<span id="cb2006-2"><a href="pseudoTimeTop.html#cb2006-2"></a>interGlobal_hcaBM1_scaled =<span class="st"> </span><span class="kw">scaleInterpolate</span>(interGlobal_hcaBM1)</span>
<span id="cb2006-3"><a href="pseudoTimeTop.html#cb2006-3"></a>interGlobal_hcaBM2_scaled =<span class="st"> </span><span class="kw">scaleInterpolate</span>(interGlobal_hcaBM2)</span></code></pre></div>
<p>Identify the shared genes across datasets</p>
<div class="sourceCode" id="cb2007"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2007-1"><a href="pseudoTimeTop.html#cb2007-1"></a>sharedMarkers =<span class="st"> </span><span class="kw">Reduce</span>(intersect,</span>
<span id="cb2007-2"><a href="pseudoTimeTop.html#cb2007-2"></a>                       <span class="kw">list</span>(<span class="kw">rownames</span>(interGlobal_caronPRET1<span class="op">$</span>interpolatedVals),</span>
<span id="cb2007-3"><a href="pseudoTimeTop.html#cb2007-3"></a>                          <span class="kw">rownames</span>(interGlobal_hcaBM1<span class="op">$</span>interpolatedVals),</span>
<span id="cb2007-4"><a href="pseudoTimeTop.html#cb2007-4"></a>                          <span class="kw">rownames</span>(interGlobal_hcaBM2<span class="op">$</span>interpolatedVals)))</span>
<span id="cb2007-5"><a href="pseudoTimeTop.html#cb2007-5"></a><span class="kw">length</span>(sharedMarkers)</span></code></pre></div>
<pre><code>## [1] 84</code></pre>
<p>Finally, there is the alignment step. CellAlign operates much like sequence alignment algorithms, quantifying overall similarity in expression throughout the trajectory (global alignment), or finding areas of highly conserved expression (local alignment). Cell-Align then finds a path through the matrix that minimizes the overall distance while adhering to the following constraints:</p>
<ul>
<li><p>for global alignment the alignment must cover the entire extent of both trajectories, always starting in the upper left of the dissimilarity matrix and ending in the lower right.</p></li>
<li><p>for local alignment the alignment is restricted only to highly similar cells, yielding as output regions with conserved expression dynamics</p></li>
</ul>
<p>Intuitively, the optimal alignment runs along a “valley” within the dissimilarity matrix.</p>
<div class="sourceCode" id="cb2009"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2009-1"><a href="pseudoTimeTop.html#cb2009-1"></a>A &lt;-<span class="st"> </span><span class="kw">calcDistMat</span>(interGlobal_caronPRET1_scaled<span class="op">$</span>scaledData[sharedMarkers,],</span>
<span id="cb2009-2"><a href="pseudoTimeTop.html#cb2009-2"></a>         interGlobal_hcaBM1_scaled<span class="op">$</span>scaledData[sharedMarkers,],</span>
<span id="cb2009-3"><a href="pseudoTimeTop.html#cb2009-3"></a>         <span class="dt">dist.method =</span> <span class="st">&#39;Euclidean&#39;</span>)</span>
<span id="cb2009-4"><a href="pseudoTimeTop.html#cb2009-4"></a>alignment =<span class="st"> </span><span class="kw">globalAlign</span>(A)</span>
<span id="cb2009-5"><a href="pseudoTimeTop.html#cb2009-5"></a><span class="kw">plotAlign</span>(alignment)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/align_caronPRET1_pseudotime3-1.png" width="672" /></p>
<pre><code>## [1] NA</code></pre>
<div class="sourceCode" id="cb2011"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2011-1"><a href="pseudoTimeTop.html#cb2011-1"></a>B &lt;-<span class="st"> </span><span class="kw">calcDistMat</span>(interGlobal_hcaBM1_scaled<span class="op">$</span>scaledData[sharedMarkers,],</span>
<span id="cb2011-2"><a href="pseudoTimeTop.html#cb2011-2"></a>                 interGlobal_hcaBM2_scaled<span class="op">$</span>scaledData[sharedMarkers,],</span>
<span id="cb2011-3"><a href="pseudoTimeTop.html#cb2011-3"></a>                 <span class="dt">dist.method =</span> <span class="st">&#39;Euclidean&#39;</span>)</span>
<span id="cb2011-4"><a href="pseudoTimeTop.html#cb2011-4"></a>alignment =<span class="st"> </span><span class="kw">globalAlign</span>(B)</span>
<span id="cb2011-5"><a href="pseudoTimeTop.html#cb2011-5"></a><span class="kw">plotAlign</span>(alignment)</span></code></pre></div>
<p><img src="MkWiC_files/figure-html/align_hcaBM2_pseudotime3-1.png" width="672" /></p>
<pre><code>## [1] NA</code></pre>
</div>
<div id="ackowledgements" class="section level2">
<h2><span class="header-section-number">24.3</span> Ackowledgements</h2>
<p>This notebook uses material from:</p>
<ul>
<li><a href="https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/index.html">SVI course</a>,</li>
<li><a href="https://osca.bioconductor.org">OSCA Book</a>,</li>
<li><a href="https://broadinstitute.github.io/2020_scWorkshop/">Broad Institute Workshop</a>,</li>
<li><a href="https://scrnaseq-course.cog.sanger.ac.uk/website/index.html">Hemberg Group Course</a>,</li>
<li><a href="https://github.com/shenorrLab/cellAlign">cellAlign</a>.</li>
</ul>
</div>
<div id="session-information-18" class="section level2">
<h2><span class="header-section-number">24.4</span> Session information</h2>
<div class="sourceCode" id="cb2013"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2013-1"><a href="pseudoTimeTop.html#cb2013-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] splines   stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] BiocNeighbors_1.8.2         BiocParallel_1.24.1        
##  [3] scDblFinder_1.4.0           cellAlign_0.1.0            
##  [5] celldex_1.0.0               clusterProfiler_3.18.1     
##  [7] msigdbr_7.2.1               viridis_0.5.1              
##  [9] viridisLite_0.3.0           gam_1.20                   
## [11] foreach_1.5.1               destiny_3.4.0              
## [13] SingleR_1.4.1               forcats_0.5.0              
## [15] stringr_1.4.0               readr_1.4.0                
## [17] tidyverse_1.3.0             cowplot_1.1.1              
## [19] Cairo_1.5-12.2              edgeR_3.32.1               
## [21] DelayedArray_0.16.3         RColorBrewer_1.1-2         
## [23] reticulate_1.18             purrr_0.3.4                
## [25] tidyr_1.1.3                 tibble_3.1.1               
## [27] broom_0.7.6                 leiden_0.3.7               
## [29] cluster_2.1.0               dynamicTreeCut_1.63-1      
## [31] bluster_1.0.0               fossil_0.4.0               
## [33] shapefiles_0.7              foreign_0.8-80             
## [35] maps_3.3.0                  sp_1.4-5                   
## [37] clustree_0.4.3              ggraph_2.0.5               
## [39] pheatmap_1.0.12             harmony_1.0                
## [41] Rcpp_1.0.6                  batchelor_1.6.3            
## [43] SeuratObject_4.0.0          Seurat_4.0.1               
## [45] limma_3.46.0                ggfortify_0.4.11           
## [47] BiocSingular_1.6.0          scuttle_1.0.4              
## [49] robustbase_0.93-7           mixtools_1.2.0             
## [51] irlba_2.3.3                 biomaRt_2.46.3             
## [53] Matrix_1.2-18               igraph_1.2.6               
## [55] DropletUtils_1.10.3         scater_1.18.6              
## [57] ggplot2_3.3.3               scran_1.18.7               
## [59] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [61] Biobase_2.50.0              GenomicRanges_1.42.0       
## [63] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [65] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [67] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [69] dplyr_1.0.5                 DT_0.18                    
## [71] knitr_1.32                 
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                    vcd_1.4-8                    
##   [3] ica_1.0-2                     class_7.3-17                 
##   [5] ps_1.6.0                      lmtest_0.9-38                
##   [7] crayon_1.4.1                  laeken_0.5.1                 
##   [9] spatstat.core_2.1-2           MASS_7.3-53                  
##  [11] rhdf5filters_1.2.0            nlme_3.1-149                 
##  [13] backports_1.2.1               reprex_0.3.0                 
##  [15] GOSemSim_2.16.1               rlang_0.4.10                 
##  [17] XVector_0.30.0                ROCR_1.0-11                  
##  [19] readxl_1.3.1                  smoother_1.1                 
##  [21] xgboost_1.3.2.1               bit64_4.0.5                  
##  [23] glue_1.4.2                    sctransform_0.3.2.9005       
##  [25] vipor_0.4.5                   spatstat.sparse_2.0-0        
##  [27] AnnotationDbi_1.52.0          DOSE_3.16.0                  
##  [29] spatstat.geom_2.1-0           haven_2.3.1                  
##  [31] tidyselect_1.1.0              rio_0.5.16                   
##  [33] fitdistrplus_1.1-3            XML_3.99-0.5                 
##  [35] zoo_1.8-9                     xtable_1.8-4                 
##  [37] RcppHNSW_0.3.0                magrittr_2.0.1               
##  [39] evaluate_0.14                 cli_2.4.0                    
##  [41] zlibbioc_1.36.0               rstudioapi_0.13              
##  [43] miniUI_0.1.1.1                bslib_0.2.4                  
##  [45] rpart_4.1-15                  fastmatch_1.1-0              
##  [47] treeio_1.14.3                 RcppEigen_0.3.3.9.1          
##  [49] shiny_1.5.0                   xfun_0.22                    
##  [51] askpass_1.1                   tidygraph_1.2.0              
##  [53] pcaMethods_1.82.0             interactiveDisplayBase_1.28.0
##  [55] ggrepel_0.9.1                 ape_5.4-1                    
##  [57] listenv_0.8.0                 png_0.1-7                    
##  [59] future_1.21.0                 withr_2.4.2                  
##  [61] bitops_1.0-6                  ggforce_0.3.3                
##  [63] ranger_0.12.1                 plyr_1.8.6                   
##  [65] cellranger_1.1.0              e1071_1.7-4                  
##  [67] dqrng_0.2.1                   pillar_1.6.0                 
##  [69] cachem_1.0.4                  fs_1.5.0                     
##  [71] kernlab_0.9-29                scatterplot3d_0.3-41         
##  [73] TTR_0.24.2                    xts_0.12.1                   
##  [75] DelayedMatrixStats_1.12.3     vctrs_0.3.7                  
##  [77] ellipsis_0.3.1                generics_0.1.0               
##  [79] tools_4.0.3                   beeswarm_0.2.3               
##  [81] munsell_0.5.0                 tweenr_1.0.2                 
##  [83] fgsea_1.16.0                  proxy_0.4-25                 
##  [85] fastmap_1.1.0                 compiler_4.0.3               
##  [87] abind_1.4-5                   httpuv_1.5.5                 
##  [89] ExperimentHub_1.16.1          segmented_1.3-3              
##  [91] plotly_4.9.3                  GenomeInfoDbData_1.2.4       
##  [93] gridExtra_2.3                 lattice_0.20-41              
##  [95] deldir_0.2-10                 utf8_1.2.1                   
##  [97] later_1.1.0.1                 BiocFileCache_1.14.0         
##  [99] jsonlite_1.7.2                ggplot.multistats_1.0.0      
## [101] scales_1.1.1                  tidytree_0.3.3               
## [103] pbapply_1.4-3                 carData_3.0-4                
## [105] sparseMatrixStats_1.2.1       lazyeval_0.2.2               
## [107] promises_1.1.1                car_3.0-10                   
## [109] R.utils_2.10.1                goftest_1.2-2                
## [111] checkmate_2.0.0               spatstat.utils_2.1-0         
## [113] rmarkdown_2.7                 openxlsx_4.2.3               
## [115] statmod_1.4.35                Rtsne_0.15                   
## [117] downloader_0.4                uwot_0.1.10                  
## [119] HDF5Array_1.18.1              survival_3.2-7               
## [121] ResidualMatrix_1.0.0          yaml_2.2.1                   
## [123] htmltools_0.5.1.1             memoise_2.0.0                
## [125] locfit_1.5-9.4                graphlayouts_0.7.1           
## [127] digest_0.6.27                 assertthat_0.2.1             
## [129] mime_0.10                     rappdirs_0.3.1               
## [131] RSQLite_2.2.6                 future.apply_1.7.0           
## [133] data.table_1.14.0             blob_1.2.1                   
## [135] R.oo_1.24.0                   labeling_0.4.2               
## [137] Rhdf5lib_1.12.1               AnnotationHub_2.22.1         
## [139] RCurl_1.98-1.3                hms_1.0.0                    
## [141] modelr_0.1.8                  rhdf5_2.34.0                 
## [143] colorspace_2.0-0              BiocManager_1.30.12          
## [145] ggbeeswarm_0.6.0              aplot_0.0.6                  
## [147] nnet_7.3-14                   sass_0.3.1                   
## [149] mclust_5.4.7                  bookdown_0.21                
## [151] RANN_2.6.1                    enrichplot_1.11.2.994        
## [153] fansi_0.4.2                   VIM_6.1.0                    
## [155] parallelly_1.24.0             R6_2.5.0                     
## [157] grid_4.0.3                    ggridges_0.5.3               
## [159] lifecycle_1.0.0               zip_2.1.1                    
## [161] curl_4.3                      jquerylib_0.1.3              
## [163] DO.db_2.9                     qvalue_2.22.0                
## [165] RcppAnnoy_0.0.18              iterators_1.0.13             
## [167] htmlwidgets_1.5.3             beachmat_2.6.4               
## [169] polyclip_1.10-0               crosstalk_1.1.1              
## [171] shadowtext_0.0.7              rvest_0.3.6                  
## [173] mgcv_1.8-33                   globals_0.14.0               
## [175] openssl_1.4.3                 patchwork_1.1.1              
## [177] dtw_1.22-3                    codetools_0.2-16             
## [179] lubridate_1.7.10              FNN_1.1.3                    
## [181] GO.db_3.12.1                  prettyunits_1.1.1            
## [183] dbplyr_2.0.0                  RSpectra_0.16-0              
## [185] R.methodsS3_1.8.1             gtable_0.3.0                 
## [187] DBI_1.1.1                     highr_0.8                    
## [189] tensor_1.5                    httr_1.4.2                   
## [191] KernSmooth_2.23-17            stringi_1.5.3                
## [193] progress_1.2.2                reshape2_1.4.4               
## [195] farver_2.1.0                  ggthemes_4.2.4               
## [197] hexbin_1.28.2                 ggtree_2.4.1                 
## [199] xml2_1.3.2                    rvcheck_0.1.8                
## [201] boot_1.3-25                   scattermore_0.7              
## [203] BiocVersion_3.12.0            DEoptimR_1.0-8               
## [205] bit_4.0.4                     scatterpie_0.1.5             
## [207] spatstat.data_2.1-0           pkgconfig_2.0.3              
## [209] knn.covertree_1.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-expression-and-abundance-between-conditions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="doubletDetectionTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/pseudoTime.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWiC.pdf", "MkWiC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
