<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Batch correction - GSM3872442 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Batch correction - GSM3872442 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Batch correction - GSM3872442 set | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dimRedForAnaTop.html"/>
<link rel="next" href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information"><i class="fa fa-check"></i><b>4.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>5</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#learning-objectives"><i class="fa fa-check"></i><b>5.1</b> Learning objectives</a></li>
<li class="chapter" data-level="5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>5.2</b> Scaling normalization</a><ul>
<li class="chapter" data-level="5.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>5.2.1</b> CPM</a></li>
<li class="chapter" data-level="5.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>5.2.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="5.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>5.2.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="5.2.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>5.2.4</b> Library size normalization</a></li>
<li class="chapter" data-level="5.2.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution"><i class="fa fa-check"></i><b>5.2.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="5.2.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>5.2.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="5.2.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>5.2.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="5.2.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>5.2.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="5.2.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object"><i class="fa fa-check"></i><b>5.2.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform"><i class="fa fa-check"></i><b>5.3</b> SCTransform</a><ul>
<li class="chapter" data-level="5.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>5.3.1</b> Inspect data</a></li>
<li class="chapter" data-level="5.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>5.3.2</b> Transformation</a></li>
<li class="chapter" data-level="5.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>5.3.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>5.4</b> Visualisation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>5.4.1</b> log raw counts</a></li>
<li class="chapter" data-level="5.4.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>5.4.2</b> log CPM</a></li>
<li class="chapter" data-level="5.4.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>5.4.3</b> scran</a></li>
<li class="chapter" data-level="5.4.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>5.4.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-1"><i class="fa fa-check"></i><b>5.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>6</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="6.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-1"><i class="fa fa-check"></i><b>6.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-1"><i class="fa fa-check"></i><b>6.2</b> Load packages</a></li>
<li class="chapter" data-level="6.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>6.3</b> Load data</a></li>
<li class="chapter" data-level="6.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>6.4</b> PCA</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>6.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>6.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="6.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>6.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>6.6</b> UMAP</a></li>
<li class="chapter" data-level="6.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-2"><i class="fa fa-check"></i><b>6.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>7</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>7.1</b> Load object</a></li>
<li class="chapter" data-level="7.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>7.3</b> Normalised counts</a></li>
<li class="chapter" data-level="7.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>8</b> Feature selection</a><ul>
<li class="chapter" data-level="8.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>8.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="8.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>8.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="8.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>8.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-4"><i class="fa fa-check"></i><b>8.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>9.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>9.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-5"><i class="fa fa-check"></i><b>9.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>10</b> Batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="10.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#learning-objectives-1"><i class="fa fa-check"></i><b>10.1</b> Learning objectives</a></li>
<li class="chapter" data-level="10.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>10.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="10.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>10.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="10.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>10.4</b> Batch correction</a><ul>
<li class="chapter" data-level="10.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>10.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-2"><i class="fa fa-check"></i><b>10.5</b> SCTransform</a></li>
<li class="chapter" data-level="10.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>10.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="10.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>10.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="10.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>10.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>10.7</b> fastMNN</a></li>
<li class="chapter" data-level="10.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>10.8</b> Harmony</a></li>
<li class="chapter" data-level="10.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-6"><i class="fa fa-check"></i><b>10.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>11</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="11.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#learning-objectives-2"><i class="fa fa-check"></i><b>11.1</b> Learning objectives</a></li>
<li class="chapter" data-level="11.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#abbreviations-1"><i class="fa fa-check"></i><b>11.2</b> Abbreviations</a></li>
<li class="chapter" data-level="11.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#motivation"><i class="fa fa-check"></i><b>11.3</b> Motivation</a></li>
<li class="chapter" data-level="11.4" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#load-the-data"><i class="fa fa-check"></i><b>11.4</b> Load the data</a></li>
<li class="chapter" data-level="11.5" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>11.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="11.6" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#linear-regression"><i class="fa fa-check"></i><b>11.6</b> Linear regression</a></li>
<li class="chapter" data-level="11.7" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>11.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="11.7.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#algorithm-overview"><i class="fa fa-check"></i><b>11.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="11.7.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#application-to-the-data"><i class="fa fa-check"></i><b>11.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>11.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="11.8.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>11.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="11.8.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>11.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="11.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>11.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="11.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#coassignment-probabilities"><i class="fa fa-check"></i><b>11.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="11.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#rand-index"><i class="fa fa-check"></i><b>11.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>11.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="11.10" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>11.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="11.11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#session-information-7"><i class="fa fa-check"></i><b>11.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>12</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="12.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#learning-objectives-3"><i class="fa fa-check"></i><b>12.1</b> Learning objectives</a></li>
<li class="chapter" data-level="12.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>12.2</b> Abbreviations</a></li>
<li class="chapter" data-level="12.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>12.3</b> Motivation</a></li>
<li class="chapter" data-level="12.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data-1"><i class="fa fa-check"></i><b>12.4</b> Load the data</a></li>
<li class="chapter" data-level="12.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>12.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="12.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>12.6</b> Linear regression</a></li>
<li class="chapter" data-level="12.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>12.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="12.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>12.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="12.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>12.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>12.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="12.8.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>12.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="12.8.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>12.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="12.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>12.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="12.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>12.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="12.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-1"><i class="fa fa-check"></i><b>12.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>12.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="12.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>12.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="12.11" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-8"><i class="fa fa-check"></i><b>12.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>13</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="13.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#learning-objectives-4"><i class="fa fa-check"></i><b>13.1</b> Learning objectives</a></li>
<li class="chapter" data-level="13.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-2"><i class="fa fa-check"></i><b>13.2</b> Motivation</a></li>
<li class="chapter" data-level="13.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-2"><i class="fa fa-check"></i><b>13.3</b> Load the data</a></li>
<li class="chapter" data-level="13.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>13.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="13.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-2"><i class="fa fa-check"></i><b>13.5</b> Linear regression</a></li>
<li class="chapter" data-level="13.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>13.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="13.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>13.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="13.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>13.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>13.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="13.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>13.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>13.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>13.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="13.7.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-2"><i class="fa fa-check"></i><b>13.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="13.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>13.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="13.7.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>13.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>13.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="13.8.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>13.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.8.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>13.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.8.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-3"><i class="fa fa-check"></i><b>13.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="13.8.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>13.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>13.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="13.10" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-9"><i class="fa fa-check"></i><b>13.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>14</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="14.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-3"><i class="fa fa-check"></i><b>14.1</b> Load data</a></li>
<li class="chapter" data-level="14.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>14.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="14.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>14.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="14.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>14.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="14.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>14.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="14.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>14.2.2</b> k-means</a><ul>
<li class="chapter" data-level="14.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>14.2.2.1</b> Description</a></li>
<li class="chapter" data-level="14.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-2"><i class="fa fa-check"></i><b>14.2.2.2</b> Example</a></li>
<li class="chapter" data-level="14.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>14.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="14.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>14.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="14.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>14.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="14.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>14.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="14.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>14.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="14.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>14.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="14.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>14.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="14.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>14.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="14.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>14.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="14.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>14.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="14.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>14.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>14.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="14.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>14.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="14.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>14.5</b> Save data</a></li>
<li class="chapter" data-level="14.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-10"><i class="fa fa-check"></i><b>14.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>15</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="15.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives-5"><i class="fa fa-check"></i><b>15.1</b> Learning objectives</a></li>
<li class="chapter" data-level="15.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-4"><i class="fa fa-check"></i><b>15.2</b> Load data</a></li>
<li class="chapter" data-level="15.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>15.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="15.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>15.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="15.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>15.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>15.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="15.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>15.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="15.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>15.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="15.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>15.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="15.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>15.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="15.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>15.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="15.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>15.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="15.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>15.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html"><i class="fa fa-check"></i><b>16</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="16.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#motivation-3"><i class="fa fa-check"></i><b>16.1</b> Motivation</a></li>
<li class="chapter" data-level="16.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#setting-up-the-data"><i class="fa fa-check"></i><b>16.2</b> Setting up the data</a></li>
<li class="chapter" data-level="16.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>16.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="16.3.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>16.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="16.3.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>16.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="16.3.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#introduction-3"><i class="fa fa-check"></i><b>16.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="16.3.2.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#pre-processing"><i class="fa fa-check"></i><b>16.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="16.3.2.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#statistical-modelling"><i class="fa fa-check"></i><b>16.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="16.3.2.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>16.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="16.3.2.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>16.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="16.3.2.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#list-of-degs"><i class="fa fa-check"></i><b>16.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="16.3.2.4.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>16.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.3.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#putting-it-all-together"><i class="fa fa-check"></i><b>16.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>16.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="16.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#overview"><i class="fa fa-check"></i><b>16.4.1</b> Overview</a></li>
<li class="chapter" data-level="16.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#handling-composition-effects"><i class="fa fa-check"></i><b>16.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="16.4.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#background"><i class="fa fa-check"></i><b>16.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#session-information-12"><i class="fa fa-check"></i><b>16.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>17</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="17.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>17.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="17.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>17.0.2</b> Data integration</a></li>
<li class="chapter" data-level="17.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>17.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="17.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>17.0.4</b> Clustering</a></li>
<li class="chapter" data-level="17.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>17.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="17.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>17.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="17.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>17.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>17.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="17.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>17.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="17.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>17.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="17.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>17.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>17.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="17.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>17.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="17.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-13"><i class="fa fa-check"></i><b>17.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>18</b> Annex</a></li>
<li class="chapter" data-level="19" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>19</b> Gene annotation</a><ul>
<li class="chapter" data-level="19.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-4"><i class="fa fa-check"></i><b>19.1</b> Introduction</a></li>
<li class="chapter" data-level="19.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>19.2</b> Sample sheet</a></li>
<li class="chapter" data-level="19.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>19.3</b> SRR9264343</a></li>
<li class="chapter" data-level="19.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-14"><i class="fa fa-check"></i><b>19.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>20</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="20.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-5"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages-2"><i class="fa fa-check"></i><b>20.2</b> Load packages</a></li>
<li class="chapter" data-level="20.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>20.3</b> Sample sheet</a></li>
<li class="chapter" data-level="20.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation-1"><i class="fa fa-check"></i><b>20.4</b> Data representation</a></li>
<li class="chapter" data-level="20.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example-3"><i class="fa fa-check"></i><b>20.5</b> Example</a></li>
<li class="chapter" data-level="20.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>20.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="20.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>20.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="20.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>20.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>20.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="20.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model-1"><i class="fa fa-check"></i><b>20.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="20.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>20.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="20.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point-1"><i class="fa fa-check"></i><b>20.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="20.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>20.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="20.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>20.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="20.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>20.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="20.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>20.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="20.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control-1"><i class="fa fa-check"></i><b>20.10</b> Quality control</a><ul>
<li class="chapter" data-level="20.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>20.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="20.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>20.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="20.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size-1"><i class="fa fa-check"></i><b>20.11.1</b> Library size</a></li>
<li class="chapter" data-level="20.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>20.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="20.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>20.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="20.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary-1"><i class="fa fa-check"></i><b>20.11.4</b> Summary</a></li>
<li class="chapter" data-level="20.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>20.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="20.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions-1"><i class="fa fa-check"></i><b>20.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="20.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>20.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="20.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>20.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="20.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>20.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="20.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>20.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="20.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>20.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="20.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>20.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="20.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca-1"><i class="fa fa-check"></i><b>20.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="20.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>20.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="20.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>20.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty-1"><i class="fa fa-check"></i><b>20.13</b> Novelty</a></li>
<li class="chapter" data-level="20.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>20.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="20.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>20.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="20.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>20.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="20.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters-1"><i class="fa fa-check"></i><b>20.14.3</b> Filters</a></li>
<li class="chapter" data-level="20.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>20.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="20.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only-1"><i class="fa fa-check"></i><b>20.14.5</b> Caron only</a></li>
<li class="chapter" data-level="20.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only-1"><i class="fa fa-check"></i><b>20.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="20.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>20.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="20.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information-15"><i class="fa fa-check"></i><b>20.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>21</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="21.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>21.1</b> Caron</a><ul>
<li class="chapter" data-level="21.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>21.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>21.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>21.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>21.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>21.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>21.2</b> Hca</a><ul>
<li class="chapter" data-level="21.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>21.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>21.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>21.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>21.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-2"><i class="fa fa-check"></i><b>21.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform-3"><i class="fa fa-check"></i><b>21.3</b> SCTransform</a><ul>
<li class="chapter" data-level="21.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>21.3.1</b> Caron</a></li>
<li class="chapter" data-level="21.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>21.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-16"><i class="fa fa-check"></i><b>21.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>22</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="22.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>22.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="22.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>22.1.1</b> CPM</a></li>
<li class="chapter" data-level="22.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>22.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="22.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>22.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="22.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>22.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="22.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>22.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="22.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>22.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="22.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="22.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="22.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>22.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>22.2</b> SCTransform</a><ul>
<li class="chapter" data-level="22.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>22.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="22.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>22.2.2</b> Transformation</a></li>
<li class="chapter" data-level="22.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>22.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>22.3</b> Visualisation</a><ul>
<li class="chapter" data-level="22.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>22.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="22.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>22.3.2</b> log CPM</a></li>
<li class="chapter" data-level="22.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>22.3.3</b> scran</a></li>
<li class="chapter" data-level="22.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-5"><i class="fa fa-check"></i><b>22.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-17"><i class="fa fa-check"></i><b>22.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>23</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="23.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-3"><i class="fa fa-check"></i><b>23.1</b> Abbreviations</a></li>
<li class="chapter" data-level="23.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation-4"><i class="fa fa-check"></i><b>23.2</b> Motivation</a></li>
<li class="chapter" data-level="23.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>23.3</b> Loading the data</a></li>
<li class="chapter" data-level="23.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>23.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="23.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression-3"><i class="fa fa-check"></i><b>23.5</b> Linear regression</a></li>
<li class="chapter" data-level="23.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>23.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="23.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>23.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="23.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>23.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>23.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="23.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>23.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>23.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="23.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>23.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="23.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>23.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-18"><i class="fa fa-check"></i><b>23.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>24</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="24.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-4"><i class="fa fa-check"></i><b>24.1</b> Abbreviations</a></li>
<li class="chapter" data-level="24.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-5"><i class="fa fa-check"></i><b>24.2</b> Motivation</a></li>
<li class="chapter" data-level="24.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>24.3</b> Loading the data</a></li>
<li class="chapter" data-level="24.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>24.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="24.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>24.5</b> Linear regression</a></li>
<li class="chapter" data-level="24.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>24.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="24.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>24.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="24.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>24.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>24.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="24.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>24.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="24.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>24.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="24.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>24.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="24.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>24.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="24.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index-4"><i class="fa fa-check"></i><b>24.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>24.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="24.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-19"><i class="fa fa-check"></i><b>24.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>25</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="25.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-5"><i class="fa fa-check"></i><b>25.1</b> Abbreviations</a></li>
<li class="chapter" data-level="25.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-6"><i class="fa fa-check"></i><b>25.2</b> Motivation</a></li>
<li class="chapter" data-level="25.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-5"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>25.4</b> Loading the data</a></li>
<li class="chapter" data-level="25.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>25.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="25.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-5"><i class="fa fa-check"></i><b>25.6</b> Linear regression</a></li>
<li class="chapter" data-level="25.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>25.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="25.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>25.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="25.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>25.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>25.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="25.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>25.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="25.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>25.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="25.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>25.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="25.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>25.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="25.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-5"><i class="fa fa-check"></i><b>25.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>25.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="25.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-20"><i class="fa fa-check"></i><b>25.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>26</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="26.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-6"><i class="fa fa-check"></i><b>26.1</b> Abbreviations</a></li>
<li class="chapter" data-level="26.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-7"><i class="fa fa-check"></i><b>26.2</b> Motivation</a></li>
<li class="chapter" data-level="26.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-6"><i class="fa fa-check"></i><b>26.3</b> Load data</a></li>
<li class="chapter" data-level="26.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>26.4</b> Loading the data</a></li>
<li class="chapter" data-level="26.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>26.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="26.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-6"><i class="fa fa-check"></i><b>26.6</b> Linear regression</a></li>
<li class="chapter" data-level="26.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>26.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="26.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-6"><i class="fa fa-check"></i><b>26.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="26.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-6"><i class="fa fa-check"></i><b>26.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="26.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>26.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="26.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>26.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="26.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>26.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="26.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>26.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="26.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>26.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="26.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-6"><i class="fa fa-check"></i><b>26.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>26.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="26.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-21"><i class="fa fa-check"></i><b>26.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#learning-objectives-6"><i class="fa fa-check"></i><b>27.1</b> Learning objectives</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#abbreviations-7"><i class="fa fa-check"></i><b>27.2</b> Abbreviations</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#motivation-8"><i class="fa fa-check"></i><b>27.3</b> Motivation</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#load-the-data-3"><i class="fa fa-check"></i><b>27.4</b> Load the data</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>27.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#linear-regression-7"><i class="fa fa-check"></i><b>27.6</b> Linear regression</a></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>27.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>27.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>27.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>27.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.8.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mixing-between-batches-6"><i class="fa fa-check"></i><b>27.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.8.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>27.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.8.2.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-5"><i class="fa fa-check"></i><b>27.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.8.2.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>27.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.8.2.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#rand-index-7"><i class="fa fa-check"></i><b>27.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>27.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>27.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.11" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#session-information-22"><i class="fa fa-check"></i><b>27.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>28</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#learning-objectives-7"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-8"><i class="fa fa-check"></i><b>28.2</b> Abbreviations</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-9"><i class="fa fa-check"></i><b>28.3</b> Motivation</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-4"><i class="fa fa-check"></i><b>28.4</b> Load the data</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-8"><i class="fa fa-check"></i><b>28.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-8"><i class="fa fa-check"></i><b>28.6</b> Linear regression</a></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-4"><i class="fa fa-check"></i><b>28.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-8"><i class="fa fa-check"></i><b>28.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-8"><i class="fa fa-check"></i><b>28.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-8"><i class="fa fa-check"></i><b>28.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.8.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-7"><i class="fa fa-check"></i><b>28.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="28.8.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.8.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-6"><i class="fa fa-check"></i><b>28.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="28.8.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-7"><i class="fa fa-check"></i><b>28.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.8.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-8"><i class="fa fa-check"></i><b>28.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-8"><i class="fa fa-check"></i><b>28.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-4"><i class="fa fa-check"></i><b>28.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.11" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-23"><i class="fa fa-check"></i><b>28.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>29</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="29.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#learning-objectives-8"><i class="fa fa-check"></i><b>29.1</b> Learning objectives</a></li>
<li class="chapter" data-level="29.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-10"><i class="fa fa-check"></i><b>29.2</b> Motivation</a></li>
<li class="chapter" data-level="29.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-5"><i class="fa fa-check"></i><b>29.3</b> Load the data</a></li>
<li class="chapter" data-level="29.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-9"><i class="fa fa-check"></i><b>29.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="29.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-9"><i class="fa fa-check"></i><b>29.5</b> Linear regression</a></li>
<li class="chapter" data-level="29.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-5"><i class="fa fa-check"></i><b>29.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="29.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-9"><i class="fa fa-check"></i><b>29.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="29.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-9"><i class="fa fa-check"></i><b>29.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="29.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-9"><i class="fa fa-check"></i><b>29.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="29.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-10"><i class="fa fa-check"></i><b>29.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>29.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-8"><i class="fa fa-check"></i><b>29.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="29.7.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-9"><i class="fa fa-check"></i><b>29.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="29.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-9"><i class="fa fa-check"></i><b>29.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="29.7.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>29.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="29.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>29.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="29.8.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-11"><i class="fa fa-check"></i><b>29.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.8.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>29.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.8.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-10"><i class="fa fa-check"></i><b>29.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="29.8.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>29.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="29.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-5"><i class="fa fa-check"></i><b>29.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="29.10" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-24"><i class="fa fa-check"></i><b>29.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>30</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="30.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>30.1</b> Load data</a></li>
<li class="chapter" data-level="30.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-11"><i class="fa fa-check"></i><b>30.2</b> Motivation</a></li>
<li class="chapter" data-level="30.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>30.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="30.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>30.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="30.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>30.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="30.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-25"><i class="fa fa-check"></i><b>30.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>31</b> Doublet detection</a><ul>
<li class="chapter" data-level="31.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-9"><i class="fa fa-check"></i><b>31.1</b> Learning objectives</a></li>
<li class="chapter" data-level="31.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>31.2</b> Set up analysis</a></li>
<li class="chapter" data-level="31.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>31.3</b> Load data</a></li>
<li class="chapter" data-level="31.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>31.4</b> Overview</a></li>
<li class="chapter" data-level="31.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>31.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="31.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>31.6</b> Further comments</a></li>
<li class="chapter" data-level="31.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-26"><i class="fa fa-check"></i><b>31.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="batch-correction---gsm3872442-set" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> Batch correction - GSM3872442 set</h1>
<div id="learning-objectives-1" class="section level2">
<h2><span class="header-section-number">10.1</span> Learning objectives</h2>
<style>
div.blue {background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<ul>
<li>bla</li>
<li>bla</li>
<li>bla</li>
</ul>
</div>
<p>## Data</p>
<p>GSM3872442 is a single PBMMC sample sequenced as a pool of two libraries:
SRR9264351 and SRR9264352.</p>
<p>We will use this sample to illustrate batch correction.</p>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="batch-correction-gsm3872442-set.html#cb509-1"></a>qcPlotDirBit &lt;-<span class="st"> &quot;Plots/Norm&quot;</span> <span class="co"># not used </span><span class="al">TODO</span></span>
<span id="cb509-2"><a href="batch-correction-gsm3872442-set.html#cb509-2"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb509-3"><a href="batch-correction-gsm3872442-set.html#cb509-3"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb509-4"><a href="batch-correction-gsm3872442-set.html#cb509-4"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb509-5"><a href="batch-correction-gsm3872442-set.html#cb509-5"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb509-6"><a href="batch-correction-gsm3872442-set.html#cb509-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb509-7"><a href="batch-correction-gsm3872442-set.html#cb509-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb509-8"><a href="batch-correction-gsm3872442-set.html#cb509-8"></a></span>
<span id="cb509-9"><a href="batch-correction-gsm3872442-set.html#cb509-9"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>) {</span>
<span id="cb509-10"><a href="batch-correction-gsm3872442-set.html#cb509-10"></a>    setName &lt;-<span class="st"> &quot;GSM3872442&quot;</span></span>
<span id="cb509-11"><a href="batch-correction-gsm3872442-set.html#cb509-11"></a>    setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb509-12"><a href="batch-correction-gsm3872442-set.html#cb509-12"></a>    dirRel &lt;-<span class="st"> &quot;..&quot;</span></span>
<span id="cb509-13"><a href="batch-correction-gsm3872442-set.html#cb509-13"></a>}</span></code></pre></div>
<p>Load object</p>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="batch-correction-gsm3872442-set.html#cb510-1"></a>setSuf &lt;-<span class="st"> &quot;&quot;</span></span>
<span id="cb510-2"><a href="batch-correction-gsm3872442-set.html#cb510-2"></a></span>
<span id="cb510-3"><a href="batch-correction-gsm3872442-set.html#cb510-3"></a><span class="co"># Read object in:</span></span>
<span id="cb510-4"><a href="batch-correction-gsm3872442-set.html#cb510-4"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postQc%s.Rds&quot;</span>,</span>
<span id="cb510-5"><a href="batch-correction-gsm3872442-set.html#cb510-5"></a>         projDir, outDirBit, <span class="st">&quot;caron&quot;</span>, setSuf)</span>
<span id="cb510-6"><a href="batch-correction-gsm3872442-set.html#cb510-6"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span></code></pre></div>
<p>Select the GSM3872442 cells:</p>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="batch-correction-gsm3872442-set.html#cb511-1"></a>sample1.nz.sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span><span class="kw">counts</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264351&quot;</span>)])),</span>
<span id="cb511-2"><a href="batch-correction-gsm3872442-set.html#cb511-2"></a>                                       <span class="dt">colData=</span><span class="kw">colData</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264351&quot;</span>)]))</span>
<span id="cb511-3"><a href="batch-correction-gsm3872442-set.html#cb511-3"></a></span>
<span id="cb511-4"><a href="batch-correction-gsm3872442-set.html#cb511-4"></a>sample2.nz.sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span><span class="kw">counts</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264352&quot;</span>)])),</span>
<span id="cb511-5"><a href="batch-correction-gsm3872442-set.html#cb511-5"></a>                                       <span class="dt">colData=</span><span class="kw">colData</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264352&quot;</span>)]))</span></code></pre></div>
</div>
<div id="normalise-each-separately-and-re-pool" class="section level2">
<h2><span class="header-section-number">10.2</span> Normalise each separately and re-pool</h2>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="batch-correction-gsm3872442-set.html#cb512-1"></a>sample1.clusters &lt;-<span class="st"> </span><span class="kw">quickCluster</span>(sample1.nz.sce, <span class="dt">method=</span><span class="st">&quot;igraph&quot;</span>)</span>
<span id="cb512-2"><a href="batch-correction-gsm3872442-set.html#cb512-2"></a>sample1.nz.sce &lt;-<span class="st"> </span><span class="kw">computeSumFactors</span>(sample1.nz.sce, <span class="dt">min.mean=</span><span class="fl">0.1</span>, <span class="dt">cluster=</span>sample1.clusters)</span>
<span id="cb512-3"><a href="batch-correction-gsm3872442-set.html#cb512-3"></a>sample1.nz.sce &lt;-<span class="st"> </span><span class="kw">logNormCounts</span>(sample1.nz.sce)</span>
<span id="cb512-4"><a href="batch-correction-gsm3872442-set.html#cb512-4"></a></span>
<span id="cb512-5"><a href="batch-correction-gsm3872442-set.html#cb512-5"></a>sample2.clusters &lt;-<span class="st"> </span><span class="kw">quickCluster</span>(sample2.nz.sce, <span class="dt">method=</span><span class="st">&quot;igraph&quot;</span>)</span>
<span id="cb512-6"><a href="batch-correction-gsm3872442-set.html#cb512-6"></a>sample2.nz.sce &lt;-<span class="st"> </span><span class="kw">computeSumFactors</span>(sample2.nz.sce, <span class="dt">min.mean=</span><span class="fl">0.1</span>, <span class="dt">cluster=</span>sample2.clusters)</span>
<span id="cb512-7"><a href="batch-correction-gsm3872442-set.html#cb512-7"></a>sample2.nz.sce &lt;-<span class="st"> </span><span class="kw">logNormCounts</span>(sample2.nz.sce)</span>
<span id="cb512-8"><a href="batch-correction-gsm3872442-set.html#cb512-8"></a></span>
<span id="cb512-9"><a href="batch-correction-gsm3872442-set.html#cb512-9"></a><span class="kw">rm</span>(sample1.clusters, sample2.clusters)</span></code></pre></div>
<p>Re-pool:</p>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="batch-correction-gsm3872442-set.html#cb513-1"></a><span class="co"># recombine the normalized samples together</span></span>
<span id="cb513-2"><a href="batch-correction-gsm3872442-set.html#cb513-2"></a>all.samp.exprs &lt;-<span class="st"> </span><span class="kw">do.call</span>(cbind,</span>
<span id="cb513-3"><a href="batch-correction-gsm3872442-set.html#cb513-3"></a>                          <span class="kw">list</span>(<span class="st">&quot;SRR9264351&quot;</span>=<span class="kw">exprs</span>(sample1.nz.sce),</span>
<span id="cb513-4"><a href="batch-correction-gsm3872442-set.html#cb513-4"></a>                               <span class="st">&quot;SRR9264352&quot;</span>=<span class="kw">exprs</span>(sample2.nz.sce)))</span>
<span id="cb513-5"><a href="batch-correction-gsm3872442-set.html#cb513-5"></a><span class="kw">colnames</span>(all.samp.exprs) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">as.character</span>(<span class="kw">colData</span>(sample1.nz.sce)<span class="op">$</span>Barcode),</span>
<span id="cb513-6"><a href="batch-correction-gsm3872442-set.html#cb513-6"></a>                              <span class="kw">as.character</span>(<span class="kw">colData</span>(sample2.nz.sce)<span class="op">$</span>Barcode))</span></code></pre></div>
<p>For the PCA we want to quickly select the genes that are most informative. We will use the top 2000 genes with the highest variance.</p>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="batch-correction-gsm3872442-set.html#cb514-1"></a>gene.variances &lt;-<span class="st"> </span><span class="kw">apply</span>(all.samp.exprs, <span class="dv">1</span>, var)</span>
<span id="cb514-2"><a href="batch-correction-gsm3872442-set.html#cb514-2"></a><span class="kw">names</span>(gene.variances) &lt;-<span class="st"> </span><span class="kw">rownames</span>(all.samp.exprs)</span>
<span id="cb514-3"><a href="batch-correction-gsm3872442-set.html#cb514-3"></a>highly.variable.genes &lt;-<span class="st"> </span><span class="kw">names</span>(gene.variances[<span class="kw">order</span>(gene.variances, <span class="dt">decreasing=</span><span class="ot">TRUE</span>)])[<span class="dv">1</span><span class="op">:</span><span class="dv">2000</span>]</span>
<span id="cb514-4"><a href="batch-correction-gsm3872442-set.html#cb514-4"></a><span class="kw">rm</span>(gene.variances)</span></code></pre></div>
<p>Perform PCA:</p>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="batch-correction-gsm3872442-set.html#cb515-1"></a><span class="co"># we need to use a fast approximate algorithm for PCA on large data sets</span></span>
<span id="cb515-2"><a href="batch-correction-gsm3872442-set.html#cb515-2"></a><span class="co"># this algorithm has a stochastic component,</span></span>
<span id="cb515-3"><a href="batch-correction-gsm3872442-set.html#cb515-3"></a><span class="co"># so we need to fix the seed number to get the same result each time</span></span>
<span id="cb515-4"><a href="batch-correction-gsm3872442-set.html#cb515-4"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb515-5"><a href="batch-correction-gsm3872442-set.html#cb515-5"></a>separate.hvg.pca &lt;-<span class="st"> </span>irlba<span class="op">::</span><span class="kw">prcomp_irlba</span>(<span class="kw">t</span>(all.samp.exprs[highly.variable.genes, ]), <span class="dt">n=</span><span class="dv">5</span>) <span class="co"># we only need a few components</span></span>
<span id="cb515-6"><a href="batch-correction-gsm3872442-set.html#cb515-6"></a>separate.hvg.pcs &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(separate.hvg.pca<span class="op">$</span>x) <span class="co"># extract the principal components</span></span>
<span id="cb515-7"><a href="batch-correction-gsm3872442-set.html#cb515-7"></a>separate.hvg.pcs<span class="op">$</span>Cell &lt;-<span class="st"> </span><span class="kw">colnames</span>(all.samp.exprs) <span class="co"># set the sample column as the cell IDs</span></span>
<span id="cb515-8"><a href="batch-correction-gsm3872442-set.html#cb515-8"></a></span>
<span id="cb515-9"><a href="batch-correction-gsm3872442-set.html#cb515-9"></a><span class="co"># combine the PCs with the sample information into a single data frame for plotting</span></span>
<span id="cb515-10"><a href="batch-correction-gsm3872442-set.html#cb515-10"></a>samples.info &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Cell&quot;</span>=<span class="kw">colnames</span>(all.samp.exprs),</span>
<span id="cb515-11"><a href="batch-correction-gsm3872442-set.html#cb515-11"></a>                           <span class="st">&quot;Run&quot;</span>=<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;SRR9264351&quot;</span>, <span class="kw">ncol</span>(sample1.nz.sce)), </span>
<span id="cb515-12"><a href="batch-correction-gsm3872442-set.html#cb515-12"></a>                                   <span class="kw">rep</span>(<span class="st">&quot;SRR9264352&quot;</span>, <span class="kw">ncol</span>(sample2.nz.sce))))</span>
<span id="cb515-13"><a href="batch-correction-gsm3872442-set.html#cb515-13"></a></span>
<span id="cb515-14"><a href="batch-correction-gsm3872442-set.html#cb515-14"></a><span class="co"># merge the two data frames together</span></span>
<span id="cb515-15"><a href="batch-correction-gsm3872442-set.html#cb515-15"></a>separate.pca.merge &lt;-<span class="st"> </span><span class="kw">merge</span>(separate.hvg.pcs, samples.info, <span class="dt">by=</span><span class="st">&#39;Cell&#39;</span>)</span>
<span id="cb515-16"><a href="batch-correction-gsm3872442-set.html#cb515-16"></a></span>
<span id="cb515-17"><a href="batch-correction-gsm3872442-set.html#cb515-17"></a><span class="co"># tidy</span></span>
<span id="cb515-18"><a href="batch-correction-gsm3872442-set.html#cb515-18"></a><span class="kw">rm</span>(all.samp.exprs, separate.hvg.pca, separate.hvg.pcs, samples.info)</span></code></pre></div>
<p>Plot PC1-PC2 plane, with cells colored by ‘Run’ (and sized according to library size):</p>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="batch-correction-gsm3872442-set.html#cb516-1"></a>sce.sep &lt;-<span class="st"> </span><span class="kw">cbind</span>(sample1.nz.sce, sample2.nz.sce)</span>
<span id="cb516-2"><a href="batch-correction-gsm3872442-set.html#cb516-2"></a><span class="kw">rm</span>(sample1.nz.sce, sample2.nz.sce)</span>
<span id="cb516-3"><a href="batch-correction-gsm3872442-set.html#cb516-3"></a>sce.sep &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce.sep)</span>
<span id="cb516-4"><a href="batch-correction-gsm3872442-set.html#cb516-4"></a><span class="kw">plotPCA</span>(sce.sep, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/sep_cbind_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="batch-correction-gsm3872442-set.html#cb517-1"></a>sce.sep &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce.sep, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb517-2"><a href="batch-correction-gsm3872442-set.html#cb517-2"></a><span class="kw">plotTSNE</span>(sce.sep, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/sep_tsne_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb518-1"><a href="batch-correction-gsm3872442-set.html#cb518-1"></a>sce.sep &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce.sep, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb518-2"><a href="batch-correction-gsm3872442-set.html#cb518-2"></a><span class="kw">plotUMAP</span>(sce.sep, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/sep_umap_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb519-1"><a href="batch-correction-gsm3872442-set.html#cb519-1"></a><span class="kw">rm</span>(sce.sep)</span></code></pre></div>
</div>
<div id="normalise-batches-together" class="section level2">
<h2><span class="header-section-number">10.3</span> Normalise batches together</h2>
<div class="sourceCode" id="cb520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb520-1"><a href="batch-correction-gsm3872442-set.html#cb520-1"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span><span class="kw">counts</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264351&quot;</span>, <span class="st">&quot;SRR9264352&quot;</span>)])),</span>
<span id="cb520-2"><a href="batch-correction-gsm3872442-set.html#cb520-2"></a>                                       <span class="dt">colData=</span><span class="kw">colData</span>(sce[, sce<span class="op">$</span>Run <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SRR9264351&quot;</span>, <span class="st">&quot;SRR9264352&quot;</span>)]))</span>
<span id="cb520-3"><a href="batch-correction-gsm3872442-set.html#cb520-3"></a></span>
<span id="cb520-4"><a href="batch-correction-gsm3872442-set.html#cb520-4"></a>sample3.clusters &lt;-<span class="st"> </span><span class="kw">quickCluster</span>(sample3.nz.sce, <span class="dt">method=</span><span class="st">&quot;igraph&quot;</span>)</span>
<span id="cb520-5"><a href="batch-correction-gsm3872442-set.html#cb520-5"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">computeSumFactors</span>(sample3.nz.sce, <span class="dt">min.mean=</span><span class="fl">0.1</span>, <span class="dt">cluster=</span>sample3.clusters)</span>
<span id="cb520-6"><a href="batch-correction-gsm3872442-set.html#cb520-6"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">logNormCounts</span>(sample3.nz.sce)</span>
<span id="cb520-7"><a href="batch-correction-gsm3872442-set.html#cb520-7"></a></span>
<span id="cb520-8"><a href="batch-correction-gsm3872442-set.html#cb520-8"></a>pool.exprs &lt;-<span class="st"> </span><span class="kw">exprs</span>(sample3.nz.sce)</span>
<span id="cb520-9"><a href="batch-correction-gsm3872442-set.html#cb520-9"></a><span class="kw">colnames</span>(pool.exprs) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="kw">colData</span>(sample3.nz.sce)<span class="op">$</span>Barcode, <span class="dt">pattern=</span><span class="st">&quot;-&quot;</span>, <span class="dt">replacement=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb520-10"><a href="batch-correction-gsm3872442-set.html#cb520-10"></a></span>
<span id="cb520-11"><a href="batch-correction-gsm3872442-set.html#cb520-11"></a><span class="kw">rm</span>(sample3.clusters, sce)</span></code></pre></div>
<p>Find the 2000 genes with the highest variance:</p>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="batch-correction-gsm3872442-set.html#cb521-1"></a>gene.variances &lt;-<span class="st"> </span><span class="kw">apply</span>(pool.exprs, <span class="dv">1</span>, var)</span>
<span id="cb521-2"><a href="batch-correction-gsm3872442-set.html#cb521-2"></a><span class="kw">names</span>(gene.variances) &lt;-<span class="st"> </span><span class="kw">rownames</span>(pool.exprs)</span>
<span id="cb521-3"><a href="batch-correction-gsm3872442-set.html#cb521-3"></a>highly.variable.genes &lt;-<span class="st"> </span><span class="kw">names</span>(gene.variances[<span class="kw">order</span>(gene.variances, <span class="dt">decreasing=</span><span class="ot">TRUE</span>)])[<span class="dv">1</span><span class="op">:</span><span class="dv">2000</span>]</span>
<span id="cb521-4"><a href="batch-correction-gsm3872442-set.html#cb521-4"></a><span class="kw">rm</span>(gene.variances)</span></code></pre></div>
<p>Perform PCA:</p>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="batch-correction-gsm3872442-set.html#cb522-1"></a><span class="co"># we need to use a fast approximate algorithm for PCA on large data sets</span></span>
<span id="cb522-2"><a href="batch-correction-gsm3872442-set.html#cb522-2"></a><span class="co"># this algorithm has a stochastic component, so we need to fix the seed number to get the same result each time</span></span>
<span id="cb522-3"><a href="batch-correction-gsm3872442-set.html#cb522-3"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb522-4"><a href="batch-correction-gsm3872442-set.html#cb522-4"></a>combined.hvg.pca &lt;-<span class="st"> </span>irlba<span class="op">::</span><span class="kw">prcomp_irlba</span>(<span class="kw">t</span>(pool.exprs[highly.variable.genes, ]), <span class="dt">n=</span><span class="dv">5</span>) <span class="co"># we only need a few components</span></span>
<span id="cb522-5"><a href="batch-correction-gsm3872442-set.html#cb522-5"></a>combined.hvg.pcs &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(combined.hvg.pca<span class="op">$</span>x) <span class="co"># extract the principal components</span></span>
<span id="cb522-6"><a href="batch-correction-gsm3872442-set.html#cb522-6"></a>combined.hvg.pcs<span class="op">$</span>Cell &lt;-<span class="st"> </span><span class="kw">colnames</span>(pool.exprs) <span class="co"># set the sample column as the cell IDs</span></span>
<span id="cb522-7"><a href="batch-correction-gsm3872442-set.html#cb522-7"></a></span>
<span id="cb522-8"><a href="batch-correction-gsm3872442-set.html#cb522-8"></a><span class="co"># combine the PCs with the sample information into a single data frame for plotting</span></span>
<span id="cb522-9"><a href="batch-correction-gsm3872442-set.html#cb522-9"></a>samples.info &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Cell&quot;</span>=<span class="kw">colnames</span>(pool.exprs),</span>
<span id="cb522-10"><a href="batch-correction-gsm3872442-set.html#cb522-10"></a>                           <span class="st">&quot;Run&quot;</span>=<span class="kw">colData</span>(sample3.nz.sce)<span class="op">$</span>Run)</span>
<span id="cb522-11"><a href="batch-correction-gsm3872442-set.html#cb522-11"></a></span>
<span id="cb522-12"><a href="batch-correction-gsm3872442-set.html#cb522-12"></a><span class="co"># merge the two data frames together</span></span>
<span id="cb522-13"><a href="batch-correction-gsm3872442-set.html#cb522-13"></a>combined.pca.merge &lt;-<span class="st"> </span><span class="kw">merge</span>(combined.hvg.pcs, samples.info, <span class="dt">by=</span><span class="st">&#39;Cell&#39;</span>)</span>
<span id="cb522-14"><a href="batch-correction-gsm3872442-set.html#cb522-14"></a></span>
<span id="cb522-15"><a href="batch-correction-gsm3872442-set.html#cb522-15"></a><span class="kw">rm</span>(all.samp.exprs, combined.hvg.pca, combined.hvg.pcs, samples.info)</span></code></pre></div>
<p>Plot PC1-PC2 plane, with cells colored by ‘Run’ (and sized according to library size):</p>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="batch-correction-gsm3872442-set.html#cb523-1"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sample3.nz.sce)</span>
<span id="cb523-2"><a href="batch-correction-gsm3872442-set.html#cb523-2"></a><span class="kw">plotPCA</span>(sample3.nz.sce, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/tog_show_pca_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="batch-correction-gsm3872442-set.html#cb524-1"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sample3.nz.sce, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb524-2"><a href="batch-correction-gsm3872442-set.html#cb524-2"></a><span class="kw">plotTSNE</span>(sample3.nz.sce, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/tog_show_tsne_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="batch-correction-gsm3872442-set.html#cb525-1"></a>sample3.nz.sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sample3.nz.sce, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb525-2"><a href="batch-correction-gsm3872442-set.html#cb525-2"></a><span class="kw">plotUMAP</span>(sample3.nz.sce, <span class="dt">colour_by=</span><span class="st">&quot;Run&quot;</span>, <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/tog_show_umap_batch_GSM3872442-1.png" width="672" /></p>
</div>
<div id="batch-correction" class="section level2">
<h2><span class="header-section-number">10.4</span> Batch correction</h2>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="batch-correction-gsm3872442-set.html#cb526-1"></a>sample3.nz.sce<span class="op">$</span>Run &lt;-<span class="st"> </span><span class="kw">factor</span>(sample3.nz.sce<span class="op">$</span>Run)</span>
<span id="cb526-2"><a href="batch-correction-gsm3872442-set.html#cb526-2"></a>sample3.nz.sce<span class="op">$</span>batch &lt;-<span class="st"> </span>sample3.nz.sce<span class="op">$</span>Run</span>
<span id="cb526-3"><a href="batch-correction-gsm3872442-set.html#cb526-3"></a>sce &lt;-<span class="st"> </span>sample3.nz.sce</span></code></pre></div>
<div id="gaussian-normal-linear-models" class="section level3">
<h3><span class="header-section-number">10.4.1</span> Gaussian (normal) linear models</h3>
<!-- 7.6.2.1 Gaussian (normal) linear models -->
<p>Limma</p>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="batch-correction-gsm3872442-set.html#cb527-1"></a><span class="kw">suppressMessages</span>(<span class="kw">require</span>(limma))</span>
<span id="cb527-2"><a href="batch-correction-gsm3872442-set.html#cb527-2"></a>lm_design_batch &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>batch, <span class="dt">data =</span> <span class="kw">colData</span>(sce))</span>
<span id="cb527-3"><a href="batch-correction-gsm3872442-set.html#cb527-3"></a>fit_lm_batch &lt;-<span class="st"> </span><span class="kw">lmFit</span>(<span class="kw">logcounts</span>(sce), lm_design_batch)</span>
<span id="cb527-4"><a href="batch-correction-gsm3872442-set.html#cb527-4"></a>resids_lm_batch &lt;-<span class="st"> </span><span class="kw">residuals</span>(fit_lm_batch, <span class="kw">logcounts</span>(sce))</span>
<span id="cb527-5"><a href="batch-correction-gsm3872442-set.html#cb527-5"></a><span class="kw">assay</span>(sce, <span class="st">&quot;lm_batch&quot;</span>) &lt;-<span class="st"> </span>resids_lm_batch</span>
<span id="cb527-6"><a href="batch-correction-gsm3872442-set.html#cb527-6"></a></span>
<span id="cb527-7"><a href="batch-correction-gsm3872442-set.html#cb527-7"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_lm_batch&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb527-8"><a href="batch-correction-gsm3872442-set.html#cb527-8"></a>  <span class="kw">runPCA</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;lm_batch&quot;</span>), <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb527-9"><a href="batch-correction-gsm3872442-set.html#cb527-9"></a></span>
<span id="cb527-10"><a href="batch-correction-gsm3872442-set.html#cb527-10"></a><span class="kw">plotReducedDim</span>(sce, <span class="dt">dimred =</span> <span class="st">&quot;PCA_lm_batch&quot;</span>,</span>
<span id="cb527-11"><a href="batch-correction-gsm3872442-set.html#cb527-11"></a>        <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>, </span>
<span id="cb527-12"><a href="batch-correction-gsm3872442-set.html#cb527-12"></a>        <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb527-13"><a href="batch-correction-gsm3872442-set.html#cb527-13"></a>        <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb527-14"><a href="batch-correction-gsm3872442-set.html#cb527-14"></a>        ) <span class="op">+</span></span>
<span id="cb527-15"><a href="batch-correction-gsm3872442-set.html#cb527-15"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;LM - regress out batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/linReg_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="batch-correction-gsm3872442-set.html#cb528-1"></a><span class="co">#scePreSct &lt;- sce # not used </span><span class="al">TODO</span><span class="co"> delete</span></span></code></pre></div>
</div>
</div>
<div id="sctransform-2" class="section level2">
<h2><span class="header-section-number">10.5</span> SCTransform</h2>
<p>First make a copy of the SCE object (we will need one later).</p>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="batch-correction-gsm3872442-set.html#cb529-1"></a><span class="co"># have log lib size</span></span>
<span id="cb529-2"><a href="batch-correction-gsm3872442-set.html#cb529-2"></a>sce<span class="op">$</span>log10sum &lt;-<span class="st"> </span><span class="kw">log10</span>(sce<span class="op">$</span>sum)</span>
<span id="cb529-3"><a href="batch-correction-gsm3872442-set.html#cb529-3"></a><span class="co"># keep copy of SCE to draw from after SCTransform,</span></span>
<span id="cb529-4"><a href="batch-correction-gsm3872442-set.html#cb529-4"></a><span class="co"># which discard some genes </span><span class="al">TODO</span><span class="co"> check-again/mention slow &#39;return all&#39; option</span></span>
<span id="cb529-5"><a href="batch-correction-gsm3872442-set.html#cb529-5"></a>sceOrig &lt;-<span class="st"> </span>sce</span></code></pre></div>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="batch-correction-gsm3872442-set.html#cb530-1"></a>counts &lt;-<span class="st"> </span><span class="kw">counts</span>(sce)</span>
<span id="cb530-2"><a href="batch-correction-gsm3872442-set.html#cb530-2"></a><span class="kw">class</span>(counts)</span></code></pre></div>
<pre><code>## [1] &quot;dgCMatrix&quot;
## attr(,&quot;package&quot;)
## [1] &quot;Matrix&quot;</code></pre>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="batch-correction-gsm3872442-set.html#cb532-1"></a><span class="co"># inspect data</span></span>
<span id="cb532-2"><a href="batch-correction-gsm3872442-set.html#cb532-2"></a>gene_attr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">mean =</span> <span class="kw">rowMeans</span>(counts),</span>
<span id="cb532-3"><a href="batch-correction-gsm3872442-set.html#cb532-3"></a>                        <span class="dt">detection_rate =</span> <span class="kw">rowMeans</span>(counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>),</span>
<span id="cb532-4"><a href="batch-correction-gsm3872442-set.html#cb532-4"></a>                        <span class="dt">var =</span> <span class="kw">apply</span>(counts, <span class="dv">1</span>, var))</span>
<span id="cb532-5"><a href="batch-correction-gsm3872442-set.html#cb532-5"></a>gene_attr<span class="op">$</span>log_mean &lt;-<span class="st"> </span><span class="kw">log10</span>(gene_attr<span class="op">$</span>mean)</span>
<span id="cb532-6"><a href="batch-correction-gsm3872442-set.html#cb532-6"></a>gene_attr<span class="op">$</span>log_var &lt;-<span class="st"> </span><span class="kw">log10</span>(gene_attr<span class="op">$</span>var)</span>
<span id="cb532-7"><a href="batch-correction-gsm3872442-set.html#cb532-7"></a><span class="kw">rownames</span>(gene_attr) &lt;-<span class="st"> </span><span class="kw">rownames</span>(counts)</span>
<span id="cb532-8"><a href="batch-correction-gsm3872442-set.html#cb532-8"></a>cell_attr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">n_umi =</span> <span class="kw">colSums</span>(counts),</span>
<span id="cb532-9"><a href="batch-correction-gsm3872442-set.html#cb532-9"></a>                        <span class="dt">n_gene =</span> <span class="kw">colSums</span>(counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>))</span>
<span id="cb532-10"><a href="batch-correction-gsm3872442-set.html#cb532-10"></a><span class="kw">rownames</span>(cell_attr) &lt;-<span class="st"> </span><span class="kw">colnames</span>(counts)</span>
<span id="cb532-11"><a href="batch-correction-gsm3872442-set.html#cb532-11"></a></span>
<span id="cb532-12"><a href="batch-correction-gsm3872442-set.html#cb532-12"></a><span class="co"># plot</span></span>
<span id="cb532-13"><a href="batch-correction-gsm3872442-set.html#cb532-13"></a><span class="kw">ggplot</span>(gene_attr, <span class="kw">aes</span>(log_mean, log_var)) <span class="op">+</span></span>
<span id="cb532-14"><a href="batch-correction-gsm3872442-set.html#cb532-14"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.3</span>, <span class="dt">shape =</span> <span class="dv">16</span>) <span class="op">+</span></span>
<span id="cb532-15"><a href="batch-correction-gsm3872442-set.html#cb532-15"></a><span class="st">  </span><span class="kw">geom_density_2d</span>(<span class="dt">size =</span> <span class="fl">0.3</span>) <span class="op">+</span></span>
<span id="cb532-16"><a href="batch-correction-gsm3872442-set.html#cb532-16"></a><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySctCheck_batch_GSM3872442-1.png" width="672" /></p>
<p>Mean-variance relationship</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="batch-correction-gsm3872442-set.html#cb533-1"></a><span class="co"># Mean-variance relationship</span></span>
<span id="cb533-2"><a href="batch-correction-gsm3872442-set.html#cb533-2"></a><span class="co"># add the expected detection rate under Poisson model</span></span>
<span id="cb533-3"><a href="batch-correction-gsm3872442-set.html#cb533-3"></a>x =<span class="st"> </span><span class="kw">seq</span>(<span class="dt">from =</span> <span class="dv">-3</span>, <span class="dt">to =</span> <span class="dv">2</span>, <span class="dt">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb533-4"><a href="batch-correction-gsm3872442-set.html#cb533-4"></a>poisson_model &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">log_mean =</span> x, <span class="dt">detection_rate =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="kw">dpois</span>(<span class="dv">0</span>, <span class="dt">lambda =</span> <span class="dv">10</span><span class="op">^</span>x))</span>
<span id="cb533-5"><a href="batch-correction-gsm3872442-set.html#cb533-5"></a><span class="kw">ggplot</span>(gene_attr, <span class="kw">aes</span>(log_mean, detection_rate)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.3</span>, <span class="dt">shape =</span> <span class="dv">16</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb533-6"><a href="batch-correction-gsm3872442-set.html#cb533-6"></a><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">data =</span> poisson_model, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>(<span class="dt">base_size =</span> <span class="dv">8</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_meanVarRel_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="batch-correction-gsm3872442-set.html#cb534-1"></a><span class="kw">rm</span>(gene_attr)</span></code></pre></div>
<p>Mean-detection-rate relationship</p>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="batch-correction-gsm3872442-set.html#cb535-1"></a><span class="co"># Mean-detection-rate relationship </span></span>
<span id="cb535-2"><a href="batch-correction-gsm3872442-set.html#cb535-2"></a><span class="kw">ggplot</span>(cell_attr, <span class="kw">aes</span>(n_umi, n_gene)) <span class="op">+</span></span>
<span id="cb535-3"><a href="batch-correction-gsm3872442-set.html#cb535-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.3</span>, <span class="dt">shape =</span> <span class="dv">16</span>) <span class="op">+</span></span>
<span id="cb535-4"><a href="batch-correction-gsm3872442-set.html#cb535-4"></a><span class="st">  </span><span class="kw">geom_density_2d</span>(<span class="dt">size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_meanDetect_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="batch-correction-gsm3872442-set.html#cb536-1"></a><span class="kw">rm</span>(cell_attr)</span></code></pre></div>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="batch-correction-gsm3872442-set.html#cb537-1"></a>counts &lt;-<span class="st"> </span><span class="kw">counts</span>(sce)</span>
<span id="cb537-2"><a href="batch-correction-gsm3872442-set.html#cb537-2"></a><span class="kw">colnames</span>(counts) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Barcode</span>
<span id="cb537-3"><a href="batch-correction-gsm3872442-set.html#cb537-3"></a></span>
<span id="cb537-4"><a href="batch-correction-gsm3872442-set.html#cb537-4"></a>cellAttr &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(sce))[,<span class="kw">c</span>(<span class="st">&quot;log10sum&quot;</span>, <span class="st">&quot;batch&quot;</span>)]</span>
<span id="cb537-5"><a href="batch-correction-gsm3872442-set.html#cb537-5"></a><span class="kw">rownames</span>(cellAttr) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Barcode</span>
<span id="cb537-6"><a href="batch-correction-gsm3872442-set.html#cb537-6"></a></span>
<span id="cb537-7"><a href="batch-correction-gsm3872442-set.html#cb537-7"></a><span class="co">#https://github.com/satijalab/seurat/issues/3925</span></span>
<span id="cb537-8"><a href="batch-correction-gsm3872442-set.html#cb537-8"></a><span class="co"># remotes::install_github(&quot;ChristophH/sctransform@develop&quot;)</span></span>
<span id="cb537-9"><a href="batch-correction-gsm3872442-set.html#cb537-9"></a></span>
<span id="cb537-10"><a href="batch-correction-gsm3872442-set.html#cb537-10"></a><span class="co">### Genes expressed in at least 5 cells will be kept</span></span>
<span id="cb537-11"><a href="batch-correction-gsm3872442-set.html#cb537-11"></a>sctnorm_data &lt;-<span class="st"> </span>sctransform<span class="op">::</span><span class="kw">vst</span>(<span class="dt">umi =</span> counts,</span>
<span id="cb537-12"><a href="batch-correction-gsm3872442-set.html#cb537-12"></a>                                 <span class="dt">min_cells =</span> <span class="dv">5</span>,</span>
<span id="cb537-13"><a href="batch-correction-gsm3872442-set.html#cb537-13"></a>                                 <span class="co">#min_cells = 10,</span></span>
<span id="cb537-14"><a href="batch-correction-gsm3872442-set.html#cb537-14"></a>                                 <span class="co">#method = &quot;nb_fast&quot;,</span></span>
<span id="cb537-15"><a href="batch-correction-gsm3872442-set.html#cb537-15"></a>                                 <span class="co">#n_genes = 3000,</span></span>
<span id="cb537-16"><a href="batch-correction-gsm3872442-set.html#cb537-16"></a>                                 <span class="co">#bw_adjust = 2, # 3</span></span>
<span id="cb537-17"><a href="batch-correction-gsm3872442-set.html#cb537-17"></a>                                 <span class="dt">cell_attr =</span> cellAttr,</span>
<span id="cb537-18"><a href="batch-correction-gsm3872442-set.html#cb537-18"></a>                                 <span class="dt">latent_var =</span> <span class="kw">c</span>(<span class="st">&quot;batch&quot;</span>),</span>
<span id="cb537-19"><a href="batch-correction-gsm3872442-set.html#cb537-19"></a>                                 <span class="co">#latent_var = c(&quot;log10sum&quot;, &quot;batch&quot;),</span></span>
<span id="cb537-20"><a href="batch-correction-gsm3872442-set.html#cb537-20"></a>                                 <span class="dt">return_gene_attr =</span> <span class="ot">TRUE</span>,</span>
<span id="cb537-21"><a href="batch-correction-gsm3872442-set.html#cb537-21"></a>                                 <span class="dt">return_cell_attr =</span> <span class="ot">TRUE</span>,</span>
<span id="cb537-22"><a href="batch-correction-gsm3872442-set.html#cb537-22"></a>                                 <span class="dt">verbosity =</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Check model used:</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="batch-correction-gsm3872442-set.html#cb538-1"></a><span class="co"># model:</span></span>
<span id="cb538-2"><a href="batch-correction-gsm3872442-set.html#cb538-2"></a><span class="kw">print</span>(sctnorm_data<span class="op">$</span>model_str)</span></code></pre></div>
<pre><code>## [1] &quot;y ~ batch&quot;</code></pre>
<p>Check new values (here 3 rows and 3 columns only):</p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="batch-correction-gsm3872442-set.html#cb540-1"></a>sctnorm_data<span class="op">$</span>y[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##                 AAACCTGCACTTCGAA-9 AAACCTGCAGACGCAA-9 AAACCTGTCATCACCC-9
## ENSG00000237491         -0.1805847         -0.1805847         -0.1805847
## ENSG00000225880         -0.1052960         -0.1052960         -0.1052960
## ENSG00000230368         -0.1977572         -0.1977572         -0.1977572</code></pre>
<p>Check object:</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="batch-correction-gsm3872442-set.html#cb542-1"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 16629 2099 
## metadata(0):
## assays(3): counts logcounts lm_batch
## rownames(16629): ENSG00000237491 ENSG00000225880 ... ENSG00000275063
##   ENSG00000271254
## rowData names(0):
## colnames: NULL
## colData names(18): Barcode Run ... batch log10sum
## reducedDimNames(4): PCA TSNE UMAP PCA_lm_batch
## altExpNames(0):</code></pre>
<p>Some genes were not included in the transformation and excluded from the output, so we will remove them from the SCE object too.</p>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="batch-correction-gsm3872442-set.html#cb544-1"></a><span class="co"># exclude genes that were not used in the transformation: </span></span>
<span id="cb544-2"><a href="batch-correction-gsm3872442-set.html#cb544-2"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(sce) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(sctnorm_data<span class="op">$</span>y))</span>
<span id="cb544-3"><a href="batch-correction-gsm3872442-set.html#cb544-3"></a>cols.meta &lt;-<span class="st"> </span><span class="kw">colData</span>(sceOrig)</span>
<span id="cb544-4"><a href="batch-correction-gsm3872442-set.html#cb544-4"></a>rows.meta &lt;-<span class="st"> </span><span class="kw">rowData</span>(sceOrig)</span>
<span id="cb544-5"><a href="batch-correction-gsm3872442-set.html#cb544-5"></a></span>
<span id="cb544-6"><a href="batch-correction-gsm3872442-set.html#cb544-6"></a>new.counts &lt;-<span class="st"> </span><span class="kw">counts</span>(sceOrig)[tmpInd, ]</span>
<span id="cb544-7"><a href="batch-correction-gsm3872442-set.html#cb544-7"></a>sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span>new.counts))</span>
<span id="cb544-8"><a href="batch-correction-gsm3872442-set.html#cb544-8"></a></span>
<span id="cb544-9"><a href="batch-correction-gsm3872442-set.html#cb544-9"></a><span class="co"># reset the column data on the new object</span></span>
<span id="cb544-10"><a href="batch-correction-gsm3872442-set.html#cb544-10"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span>cols.meta</span>
<span id="cb544-11"><a href="batch-correction-gsm3872442-set.html#cb544-11"></a><span class="kw">rowData</span>(sce) &lt;-<span class="st"> </span>rows.meta[tmpInd, ]</span></code></pre></div>
<p>We now copy the transformation output to the SCE object:</p>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="batch-correction-gsm3872442-set.html#cb545-1"></a>vstMat &lt;-<span class="st"> </span><span class="kw">as</span>(sctnorm_data<span class="op">$</span>y[<span class="kw">rownames</span>(sce),], <span class="st">&quot;dgCMatrix&quot;</span>)</span>
<span id="cb545-2"><a href="batch-correction-gsm3872442-set.html#cb545-2"></a><span class="kw">all</span>(<span class="kw">colnames</span>(vstMat) <span class="op">==</span><span class="st"> </span>sce<span class="op">$</span>Barcode)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="batch-correction-gsm3872442-set.html#cb547-1"></a><span class="kw">dim</span>(vstMat)</span></code></pre></div>
<pre><code>## [1] 13784  2099</code></pre>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="batch-correction-gsm3872442-set.html#cb549-1"></a><span class="kw">colnames</span>(vstMat) &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb549-2"><a href="batch-correction-gsm3872442-set.html#cb549-2"></a><span class="kw">assay</span>(sce, <span class="st">&quot;sctrans_norm_batchOnly&quot;</span>) &lt;-<span class="st"> </span>vstMat <span class="co"># as(vst_out$y[rownames(sce),], &quot;dgCMatrix&quot;)</span></span></code></pre></div>
<p>Also copy ‘logcounts’:</p>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="batch-correction-gsm3872442-set.html#cb550-1"></a>assayX &lt;-<span class="st"> &quot;logcounts&quot;</span></span>
<span id="cb550-2"><a href="batch-correction-gsm3872442-set.html#cb550-2"></a>tmpAssay &lt;-<span class="st"> </span><span class="kw">assay</span>(sceOrig, assayX)</span>
<span id="cb550-3"><a href="batch-correction-gsm3872442-set.html#cb550-3"></a><span class="kw">assay</span>(sce, assayX) &lt;-<span class="st"> </span>tmpAssay[tmpInd, ]</span></code></pre></div>
<p>Diagnostic plots are shown below:</p>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="batch-correction-gsm3872442-set.html#cb551-1"></a>sctransform<span class="op">::</span><span class="kw">plot_model_pars</span>(sctnorm_data)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_modelPars_batch_GSM3872442-1.png" width="672" /></p>
<p>The reduced dimension plots below show improved mixing of cells from the two sets:</p>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="batch-correction-gsm3872442-set.html#cb552-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_sctrans_norm_batchOnly&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb552-2"><a href="batch-correction-gsm3872442-set.html#cb552-2"></a>  <span class="kw">runPCA</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;sctrans_norm_batchOnly&quot;</span>), <span class="st">&quot;PCA&quot;</span></span>
<span id="cb552-3"><a href="batch-correction-gsm3872442-set.html#cb552-3"></a>)</span>
<span id="cb552-4"><a href="batch-correction-gsm3872442-set.html#cb552-4"></a><span class="kw">plotReducedDim</span>(</span>
<span id="cb552-5"><a href="batch-correction-gsm3872442-set.html#cb552-5"></a>  sce,</span>
<span id="cb552-6"><a href="batch-correction-gsm3872442-set.html#cb552-6"></a>  <span class="dt">dimred =</span> <span class="st">&quot;PCA_sctrans_norm_batchOnly&quot;</span>,</span>
<span id="cb552-7"><a href="batch-correction-gsm3872442-set.html#cb552-7"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,</span>
<span id="cb552-8"><a href="batch-correction-gsm3872442-set.html#cb552-8"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb552-9"><a href="batch-correction-gsm3872442-set.html#cb552-9"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb552-10"><a href="batch-correction-gsm3872442-set.html#cb552-10"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA plot: sctransform normalization - batch only&quot;</span>) </span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_pca_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="batch-correction-gsm3872442-set.html#cb553-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_sctrans_norm_batchOnly&quot;</span>, <span class="dt">name=</span><span class="st">&quot;TSNE_sctrans_norm_batchOnly&quot;</span>)</span>
<span id="cb553-2"><a href="batch-correction-gsm3872442-set.html#cb553-2"></a><span class="kw">plotReducedDim</span>(</span>
<span id="cb553-3"><a href="batch-correction-gsm3872442-set.html#cb553-3"></a>  sce,</span>
<span id="cb553-4"><a href="batch-correction-gsm3872442-set.html#cb553-4"></a>  <span class="dt">dimred =</span> <span class="st">&quot;TSNE_sctrans_norm_batchOnly&quot;</span>,</span>
<span id="cb553-5"><a href="batch-correction-gsm3872442-set.html#cb553-5"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,</span>
<span id="cb553-6"><a href="batch-correction-gsm3872442-set.html#cb553-6"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb553-7"><a href="batch-correction-gsm3872442-set.html#cb553-7"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb553-8"><a href="batch-correction-gsm3872442-set.html#cb553-8"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;TSNE plot: sctransform normalization - batch only&quot;</span>) </span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_tsne_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="batch-correction-gsm3872442-set.html#cb554-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_sctrans_norm_batchOnly&quot;</span>, <span class="dt">name=</span><span class="st">&quot;UMAP_sctrans_norm_batchOnly&quot;</span>)</span>
<span id="cb554-2"><a href="batch-correction-gsm3872442-set.html#cb554-2"></a><span class="kw">plotReducedDim</span>(</span>
<span id="cb554-3"><a href="batch-correction-gsm3872442-set.html#cb554-3"></a>  sce,</span>
<span id="cb554-4"><a href="batch-correction-gsm3872442-set.html#cb554-4"></a>  <span class="dt">dimred =</span> <span class="st">&quot;UMAP_sctrans_norm_batchOnly&quot;</span>,</span>
<span id="cb554-5"><a href="batch-correction-gsm3872442-set.html#cb554-5"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,</span>
<span id="cb554-6"><a href="batch-correction-gsm3872442-set.html#cb554-6"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb554-7"><a href="batch-correction-gsm3872442-set.html#cb554-7"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb554-8"><a href="batch-correction-gsm3872442-set.html#cb554-8"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;UMAP plot: sctransform normalization - batch only&quot;</span>) </span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/batchOnlySct_umap_batch_GSM3872442-1.png" width="672" /></p>
<p>Keep copy of SCE object for later:</p>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="batch-correction-gsm3872442-set.html#cb555-1"></a>sce_batchOnly &lt;-<span class="st"> </span>sce</span></code></pre></div>
</div>
<div id="mnncorrect" class="section level2">
<h2><span class="header-section-number">10.6</span> mnnCorrect</h2>
<!-- #https://bioconductor.org/packages/release/bioc/vignettes/batchelor/inst/doc/correction.html -->
<div id="check-presence-of-batch-effect" class="section level3">
<h3><span class="header-section-number">10.6.1</span> Check presence of batch effect</h3>
<p>Same as above but with batchelor commands to make the two batches and identify highly variable genes for faster dimensionality reduction.</p>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="batch-correction-gsm3872442-set.html#cb556-1"></a>sce &lt;-<span class="st"> </span>sample3.nz.sce</span>
<span id="cb556-2"><a href="batch-correction-gsm3872442-set.html#cb556-2"></a><span class="kw">rm</span>(sample3.nz.sce)</span>
<span id="cb556-3"><a href="batch-correction-gsm3872442-set.html#cb556-3"></a></span>
<span id="cb556-4"><a href="batch-correction-gsm3872442-set.html#cb556-4"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb556-5"><a href="batch-correction-gsm3872442-set.html#cb556-5"></a><span class="co"># Mind assayNames()</span></span>
<span id="cb556-6"><a href="batch-correction-gsm3872442-set.html#cb556-6"></a>sce1 &lt;-<span class="st"> </span>sce[, sce<span class="op">$</span>Run <span class="op">==</span><span class="st"> &quot;SRR9264351&quot;</span>]</span>
<span id="cb556-7"><a href="batch-correction-gsm3872442-set.html#cb556-7"></a>sce2 &lt;-<span class="st"> </span>sce[, sce<span class="op">$</span>Run <span class="op">==</span><span class="st"> &quot;SRR9264352&quot;</span>]</span></code></pre></div>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="batch-correction-gsm3872442-set.html#cb557-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb557-2"><a href="batch-correction-gsm3872442-set.html#cb557-2"></a>dec1 &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce1)</span>
<span id="cb557-3"><a href="batch-correction-gsm3872442-set.html#cb557-3"></a>dec2 &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce2)</span>
<span id="cb557-4"><a href="batch-correction-gsm3872442-set.html#cb557-4"></a>combined.dec &lt;-<span class="st"> </span><span class="kw">combineVar</span>(dec1, dec2)</span>
<span id="cb557-5"><a href="batch-correction-gsm3872442-set.html#cb557-5"></a>chosen.hvgs &lt;-<span class="st"> </span>combined.dec<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb557-6"><a href="batch-correction-gsm3872442-set.html#cb557-6"></a><span class="kw">summary</span>(chosen.hvgs)</span></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE 
## logical    7655    8974</code></pre>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="batch-correction-gsm3872442-set.html#cb559-1"></a><span class="kw">rm</span>(dec1, dec2)</span></code></pre></div>
<p>As a diagnostic, we check that there actually is a batch effect across these datasets by checking that they cluster separately. Here, we combine the two SingleCellExperiment objects without any correction using the NoCorrectParam() flag, and we informally verify that cells from different batches are separated using a t-SNE plot.</p>
<p>There is a moderate batch effect.</p>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="batch-correction-gsm3872442-set.html#cb560-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb560-2"><a href="batch-correction-gsm3872442-set.html#cb560-2"></a>combined &lt;-<span class="st"> </span><span class="kw">correctExperiments</span>(<span class="dt">A=</span>sce1, <span class="dt">B=</span>sce2, <span class="dt">PARAM=</span><span class="kw">NoCorrectParam</span>())</span>
<span id="cb560-3"><a href="batch-correction-gsm3872442-set.html#cb560-3"></a>combined &lt;-<span class="st"> </span><span class="kw">runPCA</span>(combined, <span class="dt">subset_row=</span>chosen.hvgs)</span>
<span id="cb560-4"><a href="batch-correction-gsm3872442-set.html#cb560-4"></a>combined &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(combined, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb560-5"><a href="batch-correction-gsm3872442-set.html#cb560-5"></a>combined &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(combined, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb560-6"><a href="batch-correction-gsm3872442-set.html#cb560-6"></a><span class="kw">plotPCA</span>(combined, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/noCor_redDim_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="batch-correction-gsm3872442-set.html#cb561-1"></a><span class="kw">plotTSNE</span>(combined, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/noCor_redDim_batch_GSM3872442-2.png" width="672" /></p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="batch-correction-gsm3872442-set.html#cb562-1"></a><span class="kw">plotUMAP</span>(combined, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/noCor_redDim_batch_GSM3872442-3.png" width="672" /></p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="batch-correction-gsm3872442-set.html#cb563-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_noCor&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(combined, <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb563-2"><a href="batch-correction-gsm3872442-set.html#cb563-2"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE_noCor&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(combined, <span class="st">&quot;TSNE&quot;</span>)</span>
<span id="cb563-3"><a href="batch-correction-gsm3872442-set.html#cb563-3"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;UMAP_noCor&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(combined, <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb563-4"><a href="batch-correction-gsm3872442-set.html#cb563-4"></a><span class="kw">rm</span>(combined)</span></code></pre></div>
</div>
<div id="correct-batch-effect-with-mnncorrect" class="section level3">
<h3><span class="header-section-number">10.6.2</span> Correct batch effect with mnnCorrect</h3>
<p>This is the initial method. It uses gene expression values to identify cells with similar expression patterns in both batches.</p>
<p>Let us get the normalised counts:</p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="batch-correction-gsm3872442-set.html#cb564-1"></a>batch1 &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce1)</span>
<span id="cb564-2"><a href="batch-correction-gsm3872442-set.html#cb564-2"></a>batch2 &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce2)</span>
<span id="cb564-3"><a href="batch-correction-gsm3872442-set.html#cb564-3"></a><span class="kw">rm</span>(sce1, sce2)</span></code></pre></div>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="batch-correction-gsm3872442-set.html#cb565-1"></a>y &lt;-<span class="st"> </span>batchelor<span class="op">::</span><span class="kw">mnnCorrect</span>(</span>
<span id="cb565-2"><a href="batch-correction-gsm3872442-set.html#cb565-2"></a>          batch1, batch2,  </span>
<span id="cb565-3"><a href="batch-correction-gsm3872442-set.html#cb565-3"></a>      <span class="co">#subset.row = fewer.hvgs,</span></span>
<span id="cb565-4"><a href="batch-correction-gsm3872442-set.html#cb565-4"></a>      <span class="dt">correct.all =</span> <span class="ot">TRUE</span>,</span>
<span id="cb565-5"><a href="batch-correction-gsm3872442-set.html#cb565-5"></a>          <span class="dt">k =</span> <span class="dv">20</span>,</span>
<span id="cb565-6"><a href="batch-correction-gsm3872442-set.html#cb565-6"></a>          <span class="dt">sigma =</span> <span class="fl">0.1</span>,</span>
<span id="cb565-7"><a href="batch-correction-gsm3872442-set.html#cb565-7"></a>          <span class="dt">cos.norm.in =</span> <span class="ot">TRUE</span>,</span>
<span id="cb565-8"><a href="batch-correction-gsm3872442-set.html#cb565-8"></a>          <span class="dt">svd.dim =</span> <span class="dv">2</span></span>
<span id="cb565-9"><a href="batch-correction-gsm3872442-set.html#cb565-9"></a>        )</span></code></pre></div>
<p>Copy the corrected values to the SCE object:</p>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="batch-correction-gsm3872442-set.html#cb566-1"></a><span class="kw">assay</span>(sce, <span class="st">&quot;mnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">assay</span>(y, <span class="st">&quot;corrected&quot;</span>)</span></code></pre></div>
<p>Show reduced dimension plots and check for improved mixing of cells from the two sets:</p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="batch-correction-gsm3872442-set.html#cb567-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runPCA</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;mnn&quot;</span>)</span>
<span id="cb567-2"><a href="batch-correction-gsm3872442-set.html#cb567-2"></a><span class="kw">plotPCA</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/mnnCor_pca_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="batch-correction-gsm3872442-set.html#cb568-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_mnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="batch-correction-gsm3872442-set.html#cb569-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_mnn&quot;</span>)</span>
<span id="cb569-2"><a href="batch-correction-gsm3872442-set.html#cb569-2"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/mnnCor_tsne_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="batch-correction-gsm3872442-set.html#cb570-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE_mnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="batch-correction-gsm3872442-set.html#cb571-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_mnn&quot;</span>)</span>
<span id="cb571-2"><a href="batch-correction-gsm3872442-set.html#cb571-2"></a><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/mnnCor_umap_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb572"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb572-1"><a href="batch-correction-gsm3872442-set.html#cb572-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;UMAP_mnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="batch-correction-gsm3872442-set.html#cb573-1"></a><span class="kw">rm</span>(combined.dec, chosen.hvgs)</span></code></pre></div>
</div>
</div>
<div id="fastmnn" class="section level2">
<h2><span class="header-section-number">10.7</span> fastMNN</h2>
<p>This method is faster than mnnCorrect as it identifies nearest neighbours after dimensionality reduction.</p>
<div class="sourceCode" id="cb574"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb574-1"><a href="batch-correction-gsm3872442-set.html#cb574-1"></a>fx &lt;-<span class="st"> </span>batchelor<span class="op">::</span><span class="kw">fastMNN</span>(</span>
<span id="cb574-2"><a href="batch-correction-gsm3872442-set.html#cb574-2"></a>                      sce,</span>
<span id="cb574-3"><a href="batch-correction-gsm3872442-set.html#cb574-3"></a>              <span class="co">#correct.all = TRUE,</span></span>
<span id="cb574-4"><a href="batch-correction-gsm3872442-set.html#cb574-4"></a>                      <span class="dt">batch =</span> sce<span class="op">$</span>Run</span>
<span id="cb574-5"><a href="batch-correction-gsm3872442-set.html#cb574-5"></a>            )</span>
<span id="cb574-6"><a href="batch-correction-gsm3872442-set.html#cb574-6"></a><span class="kw">class</span>(fx)</span></code></pre></div>
<pre><code>## [1] &quot;SingleCellExperiment&quot;
## attr(,&quot;package&quot;)
## [1] &quot;SingleCellExperiment&quot;</code></pre>
<p>Copy the corrected values to the SCE object:</p>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="batch-correction-gsm3872442-set.html#cb576-1"></a><span class="co"># fastMNN may drop some genes</span></span>
<span id="cb576-2"><a href="batch-correction-gsm3872442-set.html#cb576-2"></a><span class="co"># so we may not be able to keep the outcome in &#39;assay&#39;</span></span>
<span id="cb576-3"><a href="batch-correction-gsm3872442-set.html#cb576-3"></a><span class="kw">assay</span>(sce, <span class="st">&quot;fastmnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">assay</span>(fx, <span class="st">&quot;reconstructed&quot;</span>)</span></code></pre></div>
<p>Show reduced dimension plots and check for improved mixing of cells from the two sets:</p>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="batch-correction-gsm3872442-set.html#cb577-1"></a>fastmnn_pca &lt;-<span class="st"> </span><span class="kw">runPCA</span>(<span class="kw">assay</span>(sce, <span class="st">&quot;fastmnn&quot;</span>), <span class="dt">rank=</span><span class="dv">2</span>) <span class="co"># slow</span></span>
<span id="cb577-2"><a href="batch-correction-gsm3872442-set.html#cb577-2"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_fastmnn&quot;</span>) &lt;-<span class="st"> </span>fastmnn_pca<span class="op">$</span>rotation</span>
<span id="cb577-3"><a href="batch-correction-gsm3872442-set.html#cb577-3"></a><span class="kw">rm</span>(fastmnn_pca)</span>
<span id="cb577-4"><a href="batch-correction-gsm3872442-set.html#cb577-4"></a></span>
<span id="cb577-5"><a href="batch-correction-gsm3872442-set.html#cb577-5"></a><span class="kw">plotReducedDim</span>(</span>
<span id="cb577-6"><a href="batch-correction-gsm3872442-set.html#cb577-6"></a>  sce,</span>
<span id="cb577-7"><a href="batch-correction-gsm3872442-set.html#cb577-7"></a>  <span class="dt">dimred =</span> <span class="st">&quot;PCA_fastmnn&quot;</span>,</span>
<span id="cb577-8"><a href="batch-correction-gsm3872442-set.html#cb577-8"></a>  <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,</span>
<span id="cb577-9"><a href="batch-correction-gsm3872442-set.html#cb577-9"></a>  <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb577-10"><a href="batch-correction-gsm3872442-set.html#cb577-10"></a>  <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb577-11"><a href="batch-correction-gsm3872442-set.html#cb577-11"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA plot: fastMNN&quot;</span>) </span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/fastMnn_pca_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb578"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb578-1"><a href="batch-correction-gsm3872442-set.html#cb578-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_fastmnn&quot;</span>)</span>
<span id="cb578-2"><a href="batch-correction-gsm3872442-set.html#cb578-2"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/fastMnn_tsne_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="batch-correction-gsm3872442-set.html#cb579-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE_fastmnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="batch-correction-gsm3872442-set.html#cb580-1"></a>sce &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(sce, <span class="dt">dimred=</span><span class="st">&quot;PCA_fastmnn&quot;</span>)</span>
<span id="cb580-2"><a href="batch-correction-gsm3872442-set.html#cb580-2"></a><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/fastMnn_umap_batch_GSM3872442-1.png" width="672" /></p>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="batch-correction-gsm3872442-set.html#cb581-1"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;UMAP_fastmnn&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
</div>
<div id="harmony" class="section level2">
<h2><span class="header-section-number">10.8</span> Harmony</h2>
<p>Harmony [Korsunsky2018fast] is a newer batch correction method, which is designed to operate on PC space. The algorithm proceeds to iteratively cluster the cells, with the objective function formulated to promote cells from multiple datasets within each cluster. Once a clustering is obtained, the positions of the centroids of each dataset are obtained on a per-cluster basis and the coordinates are corrected. This procedure is iterated until convergence. Harmony comes with a theta parameter that controls the degree of batch correction (higher values lead to more dataset integration), and can account for multiple experimental and biological factors on input (see <a href="https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/normalization-confounders-and-batch-correction.html#harmony">variant of the ‘Hemberg course’</a>).</p>
<div class="sourceCode" id="cb582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb582-1"><a href="batch-correction-gsm3872442-set.html#cb582-1"></a><span class="kw">library</span>(harmony)</span>
<span id="cb582-2"><a href="batch-correction-gsm3872442-set.html#cb582-2"></a></span>
<span id="cb582-3"><a href="batch-correction-gsm3872442-set.html#cb582-3"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_logcounts&quot;</span>) &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(</span>
<span id="cb582-4"><a href="batch-correction-gsm3872442-set.html#cb582-4"></a>  <span class="kw">runPCA</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb582-5"><a href="batch-correction-gsm3872442-set.html#cb582-5"></a>)</span>
<span id="cb582-6"><a href="batch-correction-gsm3872442-set.html#cb582-6"></a></span>
<span id="cb582-7"><a href="batch-correction-gsm3872442-set.html#cb582-7"></a><span class="co">#Seeing how the end result of Harmony is an altered dimensional reduction space created on the basis of PCA, we plot the obtained manifold here and exclude it from the rest of the follow-ups in the section.</span></span>
<span id="cb582-8"><a href="batch-correction-gsm3872442-set.html#cb582-8"></a></span>
<span id="cb582-9"><a href="batch-correction-gsm3872442-set.html#cb582-9"></a>pca &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA_logcounts&quot;</span>))</span>
<span id="cb582-10"><a href="batch-correction-gsm3872442-set.html#cb582-10"></a>harmony_emb &lt;-<span class="st"> </span><span class="kw">HarmonyMatrix</span>(pca,</span>
<span id="cb582-11"><a href="batch-correction-gsm3872442-set.html#cb582-11"></a>                 sce<span class="op">$</span>batch,</span>
<span id="cb582-12"><a href="batch-correction-gsm3872442-set.html#cb582-12"></a>                 <span class="dt">theta=</span><span class="dv">2</span>,</span>
<span id="cb582-13"><a href="batch-correction-gsm3872442-set.html#cb582-13"></a>                 <span class="dt">do_pca=</span><span class="ot">FALSE</span>)</span>
<span id="cb582-14"><a href="batch-correction-gsm3872442-set.html#cb582-14"></a><span class="kw">reducedDim</span>(sce, <span class="st">&quot;harmony&quot;</span>) &lt;-<span class="st"> </span>harmony_emb</span>
<span id="cb582-15"><a href="batch-correction-gsm3872442-set.html#cb582-15"></a></span>
<span id="cb582-16"><a href="batch-correction-gsm3872442-set.html#cb582-16"></a><span class="kw">plotReducedDim</span>(</span>
<span id="cb582-17"><a href="batch-correction-gsm3872442-set.html#cb582-17"></a>    sce,</span>
<span id="cb582-18"><a href="batch-correction-gsm3872442-set.html#cb582-18"></a>    <span class="dt">dimred =</span> <span class="st">&#39;harmony&#39;</span>,</span>
<span id="cb582-19"><a href="batch-correction-gsm3872442-set.html#cb582-19"></a>    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,</span>
<span id="cb582-20"><a href="batch-correction-gsm3872442-set.html#cb582-20"></a>    <span class="dt">size_by =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb582-21"><a href="batch-correction-gsm3872442-set.html#cb582-21"></a>    <span class="dt">shape_by =</span> <span class="st">&quot;Sample.Name&quot;</span></span>
<span id="cb582-22"><a href="batch-correction-gsm3872442-set.html#cb582-22"></a>)</span></code></pre></div>
<p><img src="batch_GSM3872442_files/figure-html/harmony_run_batch_GSM3872442-1.png" width="672" /></p>
</div>
<div id="session-information-6" class="section level2">
<h2><span class="header-section-number">10.9</span> Session information</h2>
<details>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="batch-correction-gsm3872442-set.html#cb583-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] harmony_1.0                 Rcpp_1.0.6                 
##  [3] batchelor_1.6.3             limma_3.46.0               
##  [5] Cairo_1.5-12.2              BiocSingular_1.6.0         
##  [7] dplyr_1.0.6                 scran_1.18.7               
##  [9] scater_1.18.6               ggplot2_3.3.3              
## [11] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [13] Biobase_2.50.0              GenomicRanges_1.42.0       
## [15] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [17] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [19] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [21] knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7              sctransform_0.3.2.9005   
##  [3] tools_4.0.3               bslib_0.2.5              
##  [5] ResidualMatrix_1.0.0      utf8_1.2.1               
##  [7] R6_2.5.0                  irlba_2.3.3              
##  [9] vipor_0.4.5               uwot_0.1.10              
## [11] DBI_1.1.1                 colorspace_2.0-1         
## [13] withr_2.4.2               tidyselect_1.1.1         
## [15] gridExtra_2.3             compiler_4.0.3           
## [17] BiocNeighbors_1.8.2       isoband_0.2.4            
## [19] DelayedArray_0.16.3       labeling_0.4.2           
## [21] bookdown_0.22             sass_0.4.0               
## [23] scales_1.1.1              stringr_1.4.0            
## [25] digest_0.6.27             rmarkdown_2.8            
## [27] XVector_0.30.0            pkgconfig_2.0.3          
## [29] htmltools_0.5.1.1         parallelly_1.25.0        
## [31] sparseMatrixStats_1.2.1   highr_0.9                
## [33] rlang_0.4.11              FNN_1.1.3                
## [35] DelayedMatrixStats_1.12.3 jquerylib_0.1.4          
## [37] generics_0.1.0            farver_2.1.0             
## [39] jsonlite_1.7.2            BiocParallel_1.24.1      
## [41] RCurl_1.98-1.3            magrittr_2.0.1           
## [43] GenomeInfoDbData_1.2.4    scuttle_1.0.4            
## [45] Matrix_1.3-3              ggbeeswarm_0.6.0         
## [47] munsell_0.5.0             fansi_0.4.2              
## [49] viridis_0.6.1             lifecycle_1.0.0          
## [51] stringi_1.6.1             yaml_2.2.1               
## [53] edgeR_3.32.1              MASS_7.3-54              
## [55] zlibbioc_1.36.0           plyr_1.8.6               
## [57] Rtsne_0.15                grid_4.0.3               
## [59] listenv_0.8.0             dqrng_0.3.0              
## [61] crayon_1.4.1              lattice_0.20-44          
## [63] cowplot_1.1.1             beachmat_2.6.4           
## [65] locfit_1.5-9.4            pillar_1.6.1             
## [67] igraph_1.2.6              future.apply_1.7.0       
## [69] reshape2_1.4.4            codetools_0.2-18         
## [71] glue_1.4.2                evaluate_0.14            
## [73] vctrs_0.3.8               tidyr_1.1.3              
## [75] gtable_0.3.0              purrr_0.3.4              
## [77] future_1.21.0             assertthat_0.2.1         
## [79] xfun_0.23                 rsvd_1.0.5               
## [81] RSpectra_0.16-0           viridisLite_0.4.0        
## [83] tibble_3.1.2              beeswarm_0.3.1           
## [85] globals_0.14.0            bluster_1.0.0            
## [87] statmod_1.4.36            ellipsis_0.3.2</code></pre>
</details>

<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="batch-correction-gsm3872442-set.html#cb585-1"></a><span class="co">#qcPlotDirBit &lt;- &quot;NormPlots&quot;</span></span>
<span id="cb585-2"><a href="batch-correction-gsm3872442-set.html#cb585-2"></a><span class="co">#setNameUpp &lt;- &quot;Caron&quot;</span></span>
<span id="cb585-3"><a href="batch-correction-gsm3872442-set.html#cb585-3"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb585-4"><a href="batch-correction-gsm3872442-set.html#cb585-4"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb585-5"><a href="batch-correction-gsm3872442-set.html#cb585-5"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb585-6"><a href="batch-correction-gsm3872442-set.html#cb585-6"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span></code></pre></div>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="batch-correction-gsm3872442-set.html#cb586-1"></a><span class="kw">library</span>(knitr)</span></code></pre></div>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="batch-correction-gsm3872442-set.html#cb587-1"></a>isChild &lt;-<span class="st"> &quot;yes&quot;</span> <span class="co"># TRUE</span></span>
<span id="cb587-2"><a href="batch-correction-gsm3872442-set.html#cb587-2"></a>src &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">c</span>(<span class="st">&quot;_5hCellPerSpl&quot;</span>),</span>
<span id="cb587-3"><a href="batch-correction-gsm3872442-set.html#cb587-3"></a>          <span class="cf">function</span>(setSuf, isChild) {</span>
<span id="cb587-4"><a href="batch-correction-gsm3872442-set.html#cb587-4"></a>              <span class="kw">knit_expand</span>(<span class="dt">file =</span> <span class="st">&quot;dataSetIntegration_PBMMC_PRE-T.Rmd&quot;</span>)</span>
<span id="cb587-5"><a href="batch-correction-gsm3872442-set.html#cb587-5"></a>          })</span></code></pre></div>
<!--
  setSuf: "_allCells"
-->
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="batch-correction-gsm3872442-set.html#cb588-1"></a>params2 &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb588-2"><a href="batch-correction-gsm3872442-set.html#cb588-2"></a>  <span class="st">&quot;projDir&quot;</span> =<span class="st"> &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020&quot;</span>,</span>
<span id="cb588-3"><a href="batch-correction-gsm3872442-set.html#cb588-3"></a>  <span class="st">&quot;dirRel&quot;</span> =<span class="st"> &quot;..&quot;</span>,</span>
<span id="cb588-4"><a href="batch-correction-gsm3872442-set.html#cb588-4"></a>  <span class="st">&quot;inpDirBit&quot;</span> =<span class="st"> &quot;AnaWiSce/AnaCourse1&quot;</span>,</span>
<span id="cb588-5"><a href="batch-correction-gsm3872442-set.html#cb588-5"></a>  <span class="st">&quot;outDirBit&quot;</span> =<span class="st"> &quot;AnaWiSce/AnaCourse1&quot;</span>,</span>
<span id="cb588-6"><a href="batch-correction-gsm3872442-set.html#cb588-6"></a>  <span class="st">&quot;bookType&quot;</span> =<span class="st"> &quot;km&quot;</span>,</span>
<span id="cb588-7"><a href="batch-correction-gsm3872442-set.html#cb588-7"></a>  <span class="st">&quot;cacheBool&quot;</span>  =<span class="st"> </span><span class="ot">FALSE</span>,  </span>
<span id="cb588-8"><a href="batch-correction-gsm3872442-set.html#cb588-8"></a>  <span class="st">&quot;setName&quot;</span> =<span class="st"> &quot;caron&quot;</span>,</span>
<span id="cb588-9"><a href="batch-correction-gsm3872442-set.html#cb588-9"></a>  <span class="st">&quot;splSetToGet&quot;</span> =<span class="st"> &quot;PBMMC,PRE-T&quot;</span>,</span>
<span id="cb588-10"><a href="batch-correction-gsm3872442-set.html#cb588-10"></a>  <span class="co">#&quot;setSuf&quot; = &quot;_5hCellPerSpl&quot;,</span></span>
<span id="cb588-11"><a href="batch-correction-gsm3872442-set.html#cb588-11"></a>  <span class="st">&quot;setSuf&quot;</span> =<span class="st"> &quot;_allCells&quot;</span>,</span>
<span id="cb588-12"><a href="batch-correction-gsm3872442-set.html#cb588-12"></a>  <span class="st">&quot;dsiSuf&quot;</span> =<span class="st"> &#39;_dsi&#39;</span>)</span>
<span id="cb588-13"><a href="batch-correction-gsm3872442-set.html#cb588-13"></a></span>
<span id="cb588-14"><a href="batch-correction-gsm3872442-set.html#cb588-14"></a><span class="cf">if</span>(<span class="kw">interactive</span>()) {</span>
<span id="cb588-15"><a href="batch-correction-gsm3872442-set.html#cb588-15"></a>  paramsToUse &lt;-<span class="st"> </span>params2</span>
<span id="cb588-16"><a href="batch-correction-gsm3872442-set.html#cb588-16"></a>} <span class="cf">else</span> {</span>
<span id="cb588-17"><a href="batch-correction-gsm3872442-set.html#cb588-17"></a>  paramsToUse &lt;-<span class="st"> </span>params</span>
<span id="cb588-18"><a href="batch-correction-gsm3872442-set.html#cb588-18"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="batch-correction-gsm3872442-set.html#cb589-1"></a><span class="co"># need string for analysis name to use in header identifier</span></span>
<span id="cb589-2"><a href="batch-correction-gsm3872442-set.html#cb589-2"></a>setSuf &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>setSuf</span>
<span id="cb589-3"><a href="batch-correction-gsm3872442-set.html#cb589-3"></a><span class="cf">if</span>(paramsToUse<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>) {setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span>}</span>
<span id="cb589-4"><a href="batch-correction-gsm3872442-set.html#cb589-4"></a><span class="cf">if</span>(<span class="kw">exists</span>(<span class="st">&quot;isChild&quot;</span>)) { setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span> }</span>
<span id="cb589-5"><a href="batch-correction-gsm3872442-set.html#cb589-5"></a>anaStg &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;-&quot;</span>, setSuf)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dimRedForAnaTop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/batch_GSM3872442.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC2.pdf", "KmWiC2.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
