<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Cluster marker genes | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Cluster marker genes | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Cluster marker genes | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"/>
<link rel="next" href="multiSplCompTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information"><i class="fa fa-check"></i><b>4.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>5</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#learning-objectives"><i class="fa fa-check"></i><b>5.1</b> Learning objectives</a></li>
<li class="chapter" data-level="5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>5.2</b> Scaling normalization</a><ul>
<li class="chapter" data-level="5.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>5.2.1</b> CPM</a></li>
<li class="chapter" data-level="5.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>5.2.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="5.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>5.2.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="5.2.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>5.2.4</b> Library size normalization</a></li>
<li class="chapter" data-level="5.2.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution"><i class="fa fa-check"></i><b>5.2.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="5.2.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>5.2.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="5.2.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>5.2.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="5.2.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>5.2.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="5.2.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object"><i class="fa fa-check"></i><b>5.2.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform"><i class="fa fa-check"></i><b>5.3</b> SCTransform</a><ul>
<li class="chapter" data-level="5.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>5.3.1</b> Inspect data</a></li>
<li class="chapter" data-level="5.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>5.3.2</b> Transformation</a></li>
<li class="chapter" data-level="5.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>5.3.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>5.4</b> Visualisation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>5.4.1</b> log raw counts</a></li>
<li class="chapter" data-level="5.4.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>5.4.2</b> log CPM</a></li>
<li class="chapter" data-level="5.4.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>5.4.3</b> scran</a></li>
<li class="chapter" data-level="5.4.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>5.4.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-1"><i class="fa fa-check"></i><b>5.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>6</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="6.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-1"><i class="fa fa-check"></i><b>6.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-1"><i class="fa fa-check"></i><b>6.2</b> Load packages</a></li>
<li class="chapter" data-level="6.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>6.3</b> Load data</a></li>
<li class="chapter" data-level="6.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>6.4</b> PCA</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>6.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>6.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="6.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>6.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>6.6</b> UMAP</a></li>
<li class="chapter" data-level="6.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-2"><i class="fa fa-check"></i><b>6.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>7</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>7.1</b> Load object</a></li>
<li class="chapter" data-level="7.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>7.3</b> Normalised counts</a></li>
<li class="chapter" data-level="7.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>8</b> Feature selection</a><ul>
<li class="chapter" data-level="8.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>8.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="8.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>8.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="8.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>8.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-4"><i class="fa fa-check"></i><b>8.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>9.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>9.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-5"><i class="fa fa-check"></i><b>9.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>10</b> Batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="10.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#learning-objectives-1"><i class="fa fa-check"></i><b>10.1</b> Learning objectives</a></li>
<li class="chapter" data-level="10.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>10.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="10.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>10.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="10.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>10.4</b> Batch correction</a><ul>
<li class="chapter" data-level="10.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>10.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-2"><i class="fa fa-check"></i><b>10.5</b> SCTransform</a></li>
<li class="chapter" data-level="10.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>10.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="10.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>10.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="10.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>10.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>10.7</b> fastMNN</a></li>
<li class="chapter" data-level="10.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>10.8</b> Harmony</a></li>
<li class="chapter" data-level="10.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-6"><i class="fa fa-check"></i><b>10.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>11</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="11.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#learning-objectives-2"><i class="fa fa-check"></i><b>11.1</b> Learning objectives</a></li>
<li class="chapter" data-level="11.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#abbreviations-1"><i class="fa fa-check"></i><b>11.2</b> Abbreviations</a></li>
<li class="chapter" data-level="11.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#motivation"><i class="fa fa-check"></i><b>11.3</b> Motivation</a></li>
<li class="chapter" data-level="11.4" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#load-the-data"><i class="fa fa-check"></i><b>11.4</b> Load the data</a></li>
<li class="chapter" data-level="11.5" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>11.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="11.6" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#linear-regression"><i class="fa fa-check"></i><b>11.6</b> Linear regression</a></li>
<li class="chapter" data-level="11.7" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>11.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="11.7.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#algorithm-overview"><i class="fa fa-check"></i><b>11.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="11.7.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#application-to-the-data"><i class="fa fa-check"></i><b>11.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>11.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="11.8.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>11.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="11.8.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>11.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="11.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>11.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="11.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#coassignment-probabilities"><i class="fa fa-check"></i><b>11.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="11.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#rand-index"><i class="fa fa-check"></i><b>11.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>11.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="11.10" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>11.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="11.11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#session-information-7"><i class="fa fa-check"></i><b>11.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>12</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="12.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#learning-objectives-3"><i class="fa fa-check"></i><b>12.1</b> Learning objectives</a></li>
<li class="chapter" data-level="12.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>12.2</b> Abbreviations</a></li>
<li class="chapter" data-level="12.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>12.3</b> Motivation</a></li>
<li class="chapter" data-level="12.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data-1"><i class="fa fa-check"></i><b>12.4</b> Load the data</a></li>
<li class="chapter" data-level="12.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>12.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="12.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>12.6</b> Linear regression</a></li>
<li class="chapter" data-level="12.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>12.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="12.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>12.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="12.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>12.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>12.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="12.8.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>12.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="12.8.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>12.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="12.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>12.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="12.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>12.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="12.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-1"><i class="fa fa-check"></i><b>12.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>12.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="12.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>12.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="12.11" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-8"><i class="fa fa-check"></i><b>12.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>13</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="13.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#learning-objectives-4"><i class="fa fa-check"></i><b>13.1</b> Learning objectives</a></li>
<li class="chapter" data-level="13.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-2"><i class="fa fa-check"></i><b>13.2</b> Motivation</a></li>
<li class="chapter" data-level="13.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-2"><i class="fa fa-check"></i><b>13.3</b> Load the data</a></li>
<li class="chapter" data-level="13.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>13.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="13.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-2"><i class="fa fa-check"></i><b>13.5</b> Linear regression</a></li>
<li class="chapter" data-level="13.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>13.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="13.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>13.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="13.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>13.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>13.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="13.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>13.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>13.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>13.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="13.7.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-2"><i class="fa fa-check"></i><b>13.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="13.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>13.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="13.7.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>13.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>13.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="13.8.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>13.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.8.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>13.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.8.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-3"><i class="fa fa-check"></i><b>13.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="13.8.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>13.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>13.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="13.10" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-9"><i class="fa fa-check"></i><b>13.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>14</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="14.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-3"><i class="fa fa-check"></i><b>14.1</b> Load data</a></li>
<li class="chapter" data-level="14.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>14.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="14.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>14.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="14.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>14.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="14.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>14.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="14.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>14.2.2</b> k-means</a><ul>
<li class="chapter" data-level="14.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>14.2.2.1</b> Description</a></li>
<li class="chapter" data-level="14.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-2"><i class="fa fa-check"></i><b>14.2.2.2</b> Example</a></li>
<li class="chapter" data-level="14.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>14.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="14.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>14.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="14.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>14.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="14.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>14.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="14.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>14.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="14.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>14.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="14.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>14.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="14.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>14.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="14.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>14.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="14.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>14.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="14.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>14.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>14.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="14.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>14.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="14.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>14.5</b> Save data</a></li>
<li class="chapter" data-level="14.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-10"><i class="fa fa-check"></i><b>14.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>15</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="15.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives-5"><i class="fa fa-check"></i><b>15.1</b> Learning objectives</a></li>
<li class="chapter" data-level="15.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-4"><i class="fa fa-check"></i><b>15.2</b> Load data</a></li>
<li class="chapter" data-level="15.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>15.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="15.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>15.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="15.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>15.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>15.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="15.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>15.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="15.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>15.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="15.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>15.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="15.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>15.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="15.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>15.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="15.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>15.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="15.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>15.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html"><i class="fa fa-check"></i><b>16</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="16.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#motivation-3"><i class="fa fa-check"></i><b>16.1</b> Motivation</a></li>
<li class="chapter" data-level="16.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#setting-up-the-data"><i class="fa fa-check"></i><b>16.2</b> Setting up the data</a></li>
<li class="chapter" data-level="16.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>16.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="16.3.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>16.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="16.3.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>16.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="16.3.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#introduction-3"><i class="fa fa-check"></i><b>16.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="16.3.2.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#pre-processing"><i class="fa fa-check"></i><b>16.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="16.3.2.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#statistical-modelling"><i class="fa fa-check"></i><b>16.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="16.3.2.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>16.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="16.3.2.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>16.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="16.3.2.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#list-of-degs"><i class="fa fa-check"></i><b>16.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="16.3.2.4.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>16.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.3.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#putting-it-all-together"><i class="fa fa-check"></i><b>16.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>16.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="16.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#overview"><i class="fa fa-check"></i><b>16.4.1</b> Overview</a></li>
<li class="chapter" data-level="16.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#handling-composition-effects"><i class="fa fa-check"></i><b>16.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="16.4.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#background"><i class="fa fa-check"></i><b>16.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#session-information-12"><i class="fa fa-check"></i><b>16.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>17</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="17.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>17.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="17.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>17.0.2</b> Data integration</a></li>
<li class="chapter" data-level="17.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>17.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="17.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>17.0.4</b> Clustering</a></li>
<li class="chapter" data-level="17.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>17.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="17.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>17.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="17.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>17.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>17.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="17.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>17.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="17.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>17.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="17.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>17.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>17.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="17.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>17.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="17.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-13"><i class="fa fa-check"></i><b>17.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>18</b> Annex</a></li>
<li class="chapter" data-level="19" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>19</b> Gene annotation</a><ul>
<li class="chapter" data-level="19.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-4"><i class="fa fa-check"></i><b>19.1</b> Introduction</a></li>
<li class="chapter" data-level="19.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>19.2</b> Sample sheet</a></li>
<li class="chapter" data-level="19.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>19.3</b> SRR9264343</a></li>
<li class="chapter" data-level="19.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-14"><i class="fa fa-check"></i><b>19.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>20</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="20.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-5"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages-2"><i class="fa fa-check"></i><b>20.2</b> Load packages</a></li>
<li class="chapter" data-level="20.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>20.3</b> Sample sheet</a></li>
<li class="chapter" data-level="20.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation-1"><i class="fa fa-check"></i><b>20.4</b> Data representation</a></li>
<li class="chapter" data-level="20.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example-3"><i class="fa fa-check"></i><b>20.5</b> Example</a></li>
<li class="chapter" data-level="20.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>20.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="20.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>20.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="20.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>20.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>20.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="20.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model-1"><i class="fa fa-check"></i><b>20.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="20.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>20.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="20.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point-1"><i class="fa fa-check"></i><b>20.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="20.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>20.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="20.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>20.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="20.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>20.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="20.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>20.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="20.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control-1"><i class="fa fa-check"></i><b>20.10</b> Quality control</a><ul>
<li class="chapter" data-level="20.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>20.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="20.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>20.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="20.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size-1"><i class="fa fa-check"></i><b>20.11.1</b> Library size</a></li>
<li class="chapter" data-level="20.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>20.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="20.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>20.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="20.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary-1"><i class="fa fa-check"></i><b>20.11.4</b> Summary</a></li>
<li class="chapter" data-level="20.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>20.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="20.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions-1"><i class="fa fa-check"></i><b>20.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="20.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>20.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="20.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>20.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="20.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>20.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="20.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>20.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="20.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>20.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="20.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>20.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="20.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca-1"><i class="fa fa-check"></i><b>20.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="20.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>20.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="20.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>20.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty-1"><i class="fa fa-check"></i><b>20.13</b> Novelty</a></li>
<li class="chapter" data-level="20.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>20.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="20.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>20.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="20.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>20.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="20.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters-1"><i class="fa fa-check"></i><b>20.14.3</b> Filters</a></li>
<li class="chapter" data-level="20.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>20.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="20.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only-1"><i class="fa fa-check"></i><b>20.14.5</b> Caron only</a></li>
<li class="chapter" data-level="20.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only-1"><i class="fa fa-check"></i><b>20.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="20.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>20.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="20.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information-15"><i class="fa fa-check"></i><b>20.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>21</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="21.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>21.1</b> Caron</a><ul>
<li class="chapter" data-level="21.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>21.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>21.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>21.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>21.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>21.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>21.2</b> Hca</a><ul>
<li class="chapter" data-level="21.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>21.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>21.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>21.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>21.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-2"><i class="fa fa-check"></i><b>21.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform-3"><i class="fa fa-check"></i><b>21.3</b> SCTransform</a><ul>
<li class="chapter" data-level="21.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>21.3.1</b> Caron</a></li>
<li class="chapter" data-level="21.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>21.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-16"><i class="fa fa-check"></i><b>21.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>22</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="22.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>22.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="22.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>22.1.1</b> CPM</a></li>
<li class="chapter" data-level="22.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>22.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="22.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>22.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="22.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>22.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="22.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>22.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="22.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>22.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="22.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="22.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="22.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>22.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>22.2</b> SCTransform</a><ul>
<li class="chapter" data-level="22.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>22.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="22.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>22.2.2</b> Transformation</a></li>
<li class="chapter" data-level="22.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>22.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>22.3</b> Visualisation</a><ul>
<li class="chapter" data-level="22.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>22.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="22.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>22.3.2</b> log CPM</a></li>
<li class="chapter" data-level="22.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>22.3.3</b> scran</a></li>
<li class="chapter" data-level="22.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-5"><i class="fa fa-check"></i><b>22.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-17"><i class="fa fa-check"></i><b>22.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>23</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="23.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-3"><i class="fa fa-check"></i><b>23.1</b> Abbreviations</a></li>
<li class="chapter" data-level="23.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation-4"><i class="fa fa-check"></i><b>23.2</b> Motivation</a></li>
<li class="chapter" data-level="23.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>23.3</b> Loading the data</a></li>
<li class="chapter" data-level="23.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>23.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="23.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression-3"><i class="fa fa-check"></i><b>23.5</b> Linear regression</a></li>
<li class="chapter" data-level="23.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>23.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="23.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>23.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="23.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>23.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>23.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="23.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>23.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>23.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="23.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>23.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="23.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>23.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-18"><i class="fa fa-check"></i><b>23.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>24</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="24.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-4"><i class="fa fa-check"></i><b>24.1</b> Abbreviations</a></li>
<li class="chapter" data-level="24.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-5"><i class="fa fa-check"></i><b>24.2</b> Motivation</a></li>
<li class="chapter" data-level="24.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>24.3</b> Loading the data</a></li>
<li class="chapter" data-level="24.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>24.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="24.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>24.5</b> Linear regression</a></li>
<li class="chapter" data-level="24.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>24.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="24.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>24.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="24.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>24.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>24.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="24.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>24.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="24.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>24.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="24.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>24.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="24.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>24.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="24.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index-4"><i class="fa fa-check"></i><b>24.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>24.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="24.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-19"><i class="fa fa-check"></i><b>24.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>25</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="25.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-5"><i class="fa fa-check"></i><b>25.1</b> Abbreviations</a></li>
<li class="chapter" data-level="25.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-6"><i class="fa fa-check"></i><b>25.2</b> Motivation</a></li>
<li class="chapter" data-level="25.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-5"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>25.4</b> Loading the data</a></li>
<li class="chapter" data-level="25.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>25.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="25.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-5"><i class="fa fa-check"></i><b>25.6</b> Linear regression</a></li>
<li class="chapter" data-level="25.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>25.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="25.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>25.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="25.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>25.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>25.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="25.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>25.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="25.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>25.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="25.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>25.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="25.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>25.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="25.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-5"><i class="fa fa-check"></i><b>25.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>25.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="25.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-20"><i class="fa fa-check"></i><b>25.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>26</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="26.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-6"><i class="fa fa-check"></i><b>26.1</b> Abbreviations</a></li>
<li class="chapter" data-level="26.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-7"><i class="fa fa-check"></i><b>26.2</b> Motivation</a></li>
<li class="chapter" data-level="26.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-6"><i class="fa fa-check"></i><b>26.3</b> Load data</a></li>
<li class="chapter" data-level="26.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>26.4</b> Loading the data</a></li>
<li class="chapter" data-level="26.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>26.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="26.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-6"><i class="fa fa-check"></i><b>26.6</b> Linear regression</a></li>
<li class="chapter" data-level="26.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>26.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="26.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-6"><i class="fa fa-check"></i><b>26.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="26.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-6"><i class="fa fa-check"></i><b>26.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="26.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>26.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="26.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>26.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="26.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>26.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="26.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>26.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="26.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>26.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="26.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-6"><i class="fa fa-check"></i><b>26.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>26.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="26.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-21"><i class="fa fa-check"></i><b>26.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#learning-objectives-6"><i class="fa fa-check"></i><b>27.1</b> Learning objectives</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#abbreviations-7"><i class="fa fa-check"></i><b>27.2</b> Abbreviations</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#motivation-8"><i class="fa fa-check"></i><b>27.3</b> Motivation</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#load-the-data-3"><i class="fa fa-check"></i><b>27.4</b> Load the data</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>27.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#linear-regression-7"><i class="fa fa-check"></i><b>27.6</b> Linear regression</a></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>27.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>27.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>27.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>27.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.8.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mixing-between-batches-6"><i class="fa fa-check"></i><b>27.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.8.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>27.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.8.2.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-5"><i class="fa fa-check"></i><b>27.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.8.2.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>27.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.8.2.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#rand-index-7"><i class="fa fa-check"></i><b>27.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>27.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>27.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.11" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#session-information-22"><i class="fa fa-check"></i><b>27.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>28</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#learning-objectives-7"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-8"><i class="fa fa-check"></i><b>28.2</b> Abbreviations</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-9"><i class="fa fa-check"></i><b>28.3</b> Motivation</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-4"><i class="fa fa-check"></i><b>28.4</b> Load the data</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-8"><i class="fa fa-check"></i><b>28.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-8"><i class="fa fa-check"></i><b>28.6</b> Linear regression</a></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-4"><i class="fa fa-check"></i><b>28.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-8"><i class="fa fa-check"></i><b>28.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-8"><i class="fa fa-check"></i><b>28.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-8"><i class="fa fa-check"></i><b>28.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.8.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-7"><i class="fa fa-check"></i><b>28.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="28.8.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.8.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-6"><i class="fa fa-check"></i><b>28.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="28.8.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-7"><i class="fa fa-check"></i><b>28.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.8.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-8"><i class="fa fa-check"></i><b>28.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-8"><i class="fa fa-check"></i><b>28.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-4"><i class="fa fa-check"></i><b>28.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.11" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-23"><i class="fa fa-check"></i><b>28.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>29</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="29.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#learning-objectives-8"><i class="fa fa-check"></i><b>29.1</b> Learning objectives</a></li>
<li class="chapter" data-level="29.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-10"><i class="fa fa-check"></i><b>29.2</b> Motivation</a></li>
<li class="chapter" data-level="29.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-5"><i class="fa fa-check"></i><b>29.3</b> Load the data</a></li>
<li class="chapter" data-level="29.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-9"><i class="fa fa-check"></i><b>29.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="29.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-9"><i class="fa fa-check"></i><b>29.5</b> Linear regression</a></li>
<li class="chapter" data-level="29.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-5"><i class="fa fa-check"></i><b>29.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="29.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-9"><i class="fa fa-check"></i><b>29.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="29.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-9"><i class="fa fa-check"></i><b>29.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="29.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-9"><i class="fa fa-check"></i><b>29.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="29.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-10"><i class="fa fa-check"></i><b>29.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>29.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-8"><i class="fa fa-check"></i><b>29.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="29.7.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-9"><i class="fa fa-check"></i><b>29.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="29.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-9"><i class="fa fa-check"></i><b>29.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="29.7.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>29.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="29.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>29.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="29.8.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-11"><i class="fa fa-check"></i><b>29.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.8.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>29.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.8.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-10"><i class="fa fa-check"></i><b>29.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="29.8.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>29.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="29.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-5"><i class="fa fa-check"></i><b>29.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="29.10" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-24"><i class="fa fa-check"></i><b>29.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>30</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="30.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>30.1</b> Load data</a></li>
<li class="chapter" data-level="30.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-11"><i class="fa fa-check"></i><b>30.2</b> Motivation</a></li>
<li class="chapter" data-level="30.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>30.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="30.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>30.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="30.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>30.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="30.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-25"><i class="fa fa-check"></i><b>30.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>31</b> Doublet detection</a><ul>
<li class="chapter" data-level="31.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-9"><i class="fa fa-check"></i><b>31.1</b> Learning objectives</a></li>
<li class="chapter" data-level="31.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>31.2</b> Set up analysis</a></li>
<li class="chapter" data-level="31.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>31.3</b> Load data</a></li>
<li class="chapter" data-level="31.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>31.4</b> Overview</a></li>
<li class="chapter" data-level="31.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>31.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="31.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>31.6</b> Further comments</a></li>
<li class="chapter" data-level="31.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-26"><i class="fa fa-check"></i><b>31.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clusterMarkerGenesTop" class="section level1">
<h1><span class="header-section-number">Chapter 15</span> Cluster marker genes</h1>
<p><img src="../../Images/Andrews2017_Fig1.png" style="margin:auto; display:block" /></p>
<div class="sourceCode" id="cb1095"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1095-1"><a href="clusterMarkerGenesTop.html#cb1095-1"></a><span class="cf">if</span>(<span class="kw">interactive</span>()) {</span>
<span id="cb1095-2"><a href="clusterMarkerGenesTop.html#cb1095-2"></a>  paramsToUse &lt;-<span class="st"> </span>params2</span>
<span id="cb1095-3"><a href="clusterMarkerGenesTop.html#cb1095-3"></a>} <span class="cf">else</span> {</span>
<span id="cb1095-4"><a href="clusterMarkerGenesTop.html#cb1095-4"></a>  paramsToUse &lt;-<span class="st"> </span>params</span>
<span id="cb1095-5"><a href="clusterMarkerGenesTop.html#cb1095-5"></a>}</span>
<span id="cb1095-6"><a href="clusterMarkerGenesTop.html#cb1095-6"></a></span>
<span id="cb1095-7"><a href="clusterMarkerGenesTop.html#cb1095-7"></a>projDir &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>projDir</span>
<span id="cb1095-8"><a href="clusterMarkerGenesTop.html#cb1095-8"></a>dirRel &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>dirRel</span>
<span id="cb1095-9"><a href="clusterMarkerGenesTop.html#cb1095-9"></a>outDirBit &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>outDirBit</span>
<span id="cb1095-10"><a href="clusterMarkerGenesTop.html#cb1095-10"></a>cacheBool &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>cacheBool</span>
<span id="cb1095-11"><a href="clusterMarkerGenesTop.html#cb1095-11"></a>setName &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>setName</span>
<span id="cb1095-12"><a href="clusterMarkerGenesTop.html#cb1095-12"></a>splSetToGet &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>splSetToGet</span>
<span id="cb1095-13"><a href="clusterMarkerGenesTop.html#cb1095-13"></a>setSuf &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>setSuf</span>
<span id="cb1095-14"><a href="clusterMarkerGenesTop.html#cb1095-14"></a>dsiSuf &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1095-15"><a href="clusterMarkerGenesTop.html#cb1095-15"></a><span class="cf">if</span>(paramsToUse<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1095-16"><a href="clusterMarkerGenesTop.html#cb1095-16"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1095-17"><a href="clusterMarkerGenesTop.html#cb1095-17"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span></span>
<span id="cb1095-18"><a href="clusterMarkerGenesTop.html#cb1095-18"></a>    setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb1095-19"><a href="clusterMarkerGenesTop.html#cb1095-19"></a>    dsiSuf &lt;-<span class="st"> &#39;_dsi&#39;</span></span>
<span id="cb1095-20"><a href="clusterMarkerGenesTop.html#cb1095-20"></a>}</span>
<span id="cb1095-21"><a href="clusterMarkerGenesTop.html#cb1095-21"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1095-22"><a href="clusterMarkerGenesTop.html#cb1095-22"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1095-23"><a href="clusterMarkerGenesTop.html#cb1095-23"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1095-24"><a href="clusterMarkerGenesTop.html#cb1095-24"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
<div class="sourceCode" id="cb1096"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1096-1"><a href="clusterMarkerGenesTop.html#cb1096-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb1096-2"><a href="clusterMarkerGenesTop.html#cb1096-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb1096-3"><a href="clusterMarkerGenesTop.html#cb1096-3"></a><span class="kw">library</span>(scran)</span>
<span id="cb1096-4"><a href="clusterMarkerGenesTop.html#cb1096-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb1096-5"><a href="clusterMarkerGenesTop.html#cb1096-5"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb1096-6"><a href="clusterMarkerGenesTop.html#cb1096-6"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1096-7"><a href="clusterMarkerGenesTop.html#cb1096-7"></a><span class="kw">library</span>(Cairo)</span>
<span id="cb1096-8"><a href="clusterMarkerGenesTop.html#cb1096-8"></a><span class="kw">library</span>(glue)</span>
<span id="cb1096-9"><a href="clusterMarkerGenesTop.html#cb1096-9"></a></span>
<span id="cb1096-10"><a href="clusterMarkerGenesTop.html#cb1096-10"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>), <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p>Source: we will follow the <a href="https://bioconductor.org/books/release/OSCA/marker-detection.html">OSCA chapter on marker detection</a> (with some of its text copied here with little modification). See also the Hemberg group chapter on <a href="https://scrnaseq-course.cog.sanger.ac.uk/website/biological-analysis.html#dechapter">differential analysis section</a>.</p>
<p>To interpret our clustering results, we identify the genes that drive separation between clusters. These marker genes allow us to assign biological meaning to each cluster based on their functional annotation. In the most obvious case, the marker genes for each cluster are a priori associated with particular cell types, allowing us to treat the clustering as a proxy for cell type identity. The same principle can be applied to discover more subtle differences between clusters (e.g., changes in activation or differentiation state) based on the behavior of genes in the affected pathways.</p>
<p>Identification of marker genes is usually based around the retrospective detection of differential expression between clusters. Genes that are more strongly DE are more likely to have caused separate clustering of cells in the first place. Several different statistical tests are available to quantify the differences in expression profiles, and different approaches can be used to consolidate test results into a single ranking of genes for each cluster.</p>
<div id="learning-objectives-5" class="section level2">
<h2><span class="header-section-number">15.1</span> Learning objectives</h2>
<ul>
<li>identify genes that differentially expressed between clusters,</li>
<li>exclusively or not,</li>
<li>using different methods that test:
<ul>
<li>the mean expression level,</li>
<li>the whole distribution,</li>
<li>or the proportion of cells expressing the gene</li>
</ul></li>
<li>compile a summary table.</li>
</ul>
</div>
<div id="load-data-4" class="section level2">
<h2><span class="header-section-number">15.2</span> Load data</h2>
<p>We will load the R file keeping the SCE (SingleCellExperiment) object with the outcome of the clustering performed after feature selection and dimensionality reduction of the normalised counts for 500 cells per sample.</p>
<div class="sourceCode" id="cb1097"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1097-1"><a href="clusterMarkerGenesTop.html#cb1097-1"></a><span class="co"># read uncorrected counts</span></span>
<span id="cb1097-2"><a href="clusterMarkerGenesTop.html#cb1097-2"></a>fn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_uncorr.Rds&quot;</span>,</span>
<span id="cb1097-3"><a href="clusterMarkerGenesTop.html#cb1097-3"></a>              projDir,</span>
<span id="cb1097-4"><a href="clusterMarkerGenesTop.html#cb1097-4"></a>              outDirBit,</span>
<span id="cb1097-5"><a href="clusterMarkerGenesTop.html#cb1097-5"></a>              setName,</span>
<span id="cb1097-6"><a href="clusterMarkerGenesTop.html#cb1097-6"></a>              setSuf,</span>
<span id="cb1097-7"><a href="clusterMarkerGenesTop.html#cb1097-7"></a>              dsiSuf,</span>
<span id="cb1097-8"><a href="clusterMarkerGenesTop.html#cb1097-8"></a>              splSetToGet2) <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1097-9"><a href="clusterMarkerGenesTop.html#cb1097-9"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>fn)</span>
<span id="cb1097-10"><a href="clusterMarkerGenesTop.html#cb1097-10"></a></span>
<span id="cb1097-11"><a href="clusterMarkerGenesTop.html#cb1097-11"></a><span class="co"># Read object in:</span></span>
<span id="cb1097-12"><a href="clusterMarkerGenesTop.html#cb1097-12"></a><span class="co">#tmpFn &lt;- sprintf(&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_clustered.Rds&quot;, projDir, outDirBit, setName, setSuf)</span></span>
<span id="cb1097-13"><a href="clusterMarkerGenesTop.html#cb1097-13"></a><span class="co">#tmpFn &lt;- sprintf(&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_clust.Rds&quot;,</span></span>
<span id="cb1097-14"><a href="clusterMarkerGenesTop.html#cb1097-14"></a><span class="co">#                 projDir, outDirBit, setName, setSuf)</span></span>
<span id="cb1097-15"><a href="clusterMarkerGenesTop.html#cb1097-15"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_clust.Rds&quot;</span>,</span>
<span id="cb1097-16"><a href="clusterMarkerGenesTop.html#cb1097-16"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1097-17"><a href="clusterMarkerGenesTop.html#cb1097-17"></a></span>
<span id="cb1097-18"><a href="clusterMarkerGenesTop.html#cb1097-18"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1_clust.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1099"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1099-1"><a href="clusterMarkerGenesTop.html#cb1099-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb1099-2"><a href="clusterMarkerGenesTop.html#cb1099-2"></a>{</span>
<span id="cb1099-3"><a href="clusterMarkerGenesTop.html#cb1099-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb1099-4"><a href="clusterMarkerGenesTop.html#cb1099-4"></a>}</span>
<span id="cb1099-5"><a href="clusterMarkerGenesTop.html#cb1099-5"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb1099-6"><a href="clusterMarkerGenesTop.html#cb1099-6"></a><span class="kw">rm</span>(tmpFn)</span>
<span id="cb1099-7"><a href="clusterMarkerGenesTop.html#cb1099-7"></a></span>
<span id="cb1099-8"><a href="clusterMarkerGenesTop.html#cb1099-8"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb1099-9"><a href="clusterMarkerGenesTop.html#cb1099-9"></a></span>
<span id="cb1099-10"><a href="clusterMarkerGenesTop.html#cb1099-10"></a><span class="co"># copy clustering output to uncorrected SCE</span></span>
<span id="cb1099-11"><a href="clusterMarkerGenesTop.html#cb1099-11"></a>sce &lt;-<span class="st"> </span>uncorrected</span>
<span id="cb1099-12"><a href="clusterMarkerGenesTop.html#cb1099-12"></a>x &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out)[,<span class="kw">c</span>(<span class="st">&quot;Barcode&quot;</span>,<span class="st">&quot;leiden&quot;</span>)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb1099-13"><a href="clusterMarkerGenesTop.html#cb1099-13"></a><span class="kw">colData</span>(sce) &lt;-<span class="st"> </span><span class="kw">colData</span>(uncorrected) <span class="op">%&gt;%</span></span>
<span id="cb1099-14"><a href="clusterMarkerGenesTop.html#cb1099-14"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1099-15"><a href="clusterMarkerGenesTop.html#cb1099-15"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(x, <span class="dt">by=</span><span class="st">&quot;Barcode&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1099-16"><a href="clusterMarkerGenesTop.html#cb1099-16"></a><span class="st">  </span>DataFrame</span>
<span id="cb1099-17"><a href="clusterMarkerGenesTop.html#cb1099-17"></a><span class="kw">colData</span>(sce)<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>leiden</span>
<span id="cb1099-18"><a href="clusterMarkerGenesTop.html#cb1099-18"></a>sce<span class="op">$</span>clusterStg &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, sce<span class="op">$</span>cluster),</span>
<span id="cb1099-19"><a href="clusterMarkerGenesTop.html#cb1099-19"></a>             <span class="dt">levels =</span> <span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, <span class="kw">levels</span>(sce<span class="op">$</span>cluster)))</span>
<span id="cb1099-20"><a href="clusterMarkerGenesTop.html#cb1099-20"></a><span class="kw">rm</span>(uncorrected)</span></code></pre></div>
<p>Number of cells: .</p>
<p>Number of genes: 16629.</p>
</div>
<div id="detecting-genes-differentially-expressed-between-clusters" class="section level2">
<h2><span class="header-section-number">15.3</span> Detecting genes differentially expressed between clusters</h2>
<p>The TSNE plots below show the structure observed with and without data set integration
Cells colored by cluster:</p>
<div class="sourceCode" id="cb1100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1100-1"><a href="clusterMarkerGenesTop.html#cb1100-1"></a>tsney &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;leiden&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1100-2"><a href="clusterMarkerGenesTop.html#cb1100-2"></a>tsnex &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1100-3"><a href="clusterMarkerGenesTop.html#cb1100-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(tsney, tsnex, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>Cells colored by sample name:</p>
<div class="sourceCode" id="cb1101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1101-1"><a href="clusterMarkerGenesTop.html#cb1101-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1101-2"><a href="clusterMarkerGenesTop.html#cb1101-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1101-3"><a href="clusterMarkerGenesTop.html#cb1101-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<div class="sourceCode" id="cb1102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1102-1"><a href="clusterMarkerGenesTop.html#cb1102-1"></a><span class="kw">rm</span>(p1, p2)</span></code></pre></div>
<p>The UMAP plots below show the structure observed with and without data set integration:</p>
<p>Cells colored by cluster:</p>
<div class="sourceCode" id="cb1103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1103-1"><a href="clusterMarkerGenesTop.html#cb1103-1"></a>umapy &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;leiden&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1103-2"><a href="clusterMarkerGenesTop.html#cb1103-2"></a>umapx &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1103-3"><a href="clusterMarkerGenesTop.html#cb1103-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(umapy, umapx, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Cells colored by sample name:</p>
<div class="sourceCode" id="cb1104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1104-1"><a href="clusterMarkerGenesTop.html#cb1104-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1104-2"><a href="clusterMarkerGenesTop.html#cb1104-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1104-3"><a href="clusterMarkerGenesTop.html#cb1104-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode" id="cb1105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1105-1"><a href="clusterMarkerGenesTop.html#cb1105-1"></a><span class="kw">rm</span>(p1, p2)</span></code></pre></div>
<div id="differential-expression-analysis" class="section level3">
<h3><span class="header-section-number">15.3.1</span> Differential expression analysis</h3>
<p>For each cluster, we will identify genes whose expression differ to that of other clusters, for each pair of cluster, using <code>scran::findMarkers()</code>. The function fits a linear model to the log-expression values for each gene using limma <span class="citation">(<span class="citeproc-not-found" data-reference-id="doi:10.1093/nar/gkv007"><strong>???</strong></span>)</span> and allows testing for differential expression in each cluster compared to the others while accounting for known, uninteresting factors.</p>
<p>We will first identify genes whose average expression differ between clusters, using the Welch t-test (default) with a log-fold change
threshold of 0 (default) and ranking genes based on the outcome of any of the pairwise comparisons (default).</p>
<div class="sourceCode" id="cb1106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1106-1"><a href="clusterMarkerGenesTop.html#cb1106-1"></a><span class="co"># Clusters called with igraph::cluster_walktrap() are named with digits.</span></span>
<span id="cb1106-2"><a href="clusterMarkerGenesTop.html#cb1106-2"></a><span class="co"># We add a &#39;c&#39; prefix (for-cluster) to avoid any confusion: these values are labels, not numerical values</span></span>
<span id="cb1106-3"><a href="clusterMarkerGenesTop.html#cb1106-3"></a></span>
<span id="cb1106-4"><a href="clusterMarkerGenesTop.html#cb1106-4"></a><span class="co"># run scran::findMarkers()</span></span>
<span id="cb1106-5"><a href="clusterMarkerGenesTop.html#cb1106-5"></a><span class="co"># with default parameters for now</span></span>
<span id="cb1106-6"><a href="clusterMarkerGenesTop.html#cb1106-6"></a><span class="co"># check the function&#39;s manual for details (?scran::findMarkers)</span></span>
<span id="cb1106-7"><a href="clusterMarkerGenesTop.html#cb1106-7"></a><span class="co"># (the test.type and pval.type options are covered below)</span></span>
<span id="cb1106-8"><a href="clusterMarkerGenesTop.html#cb1106-8"></a>markersWoBatch &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1106-9"><a href="clusterMarkerGenesTop.html#cb1106-9"></a>                       <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg)</span></code></pre></div>
<div class="sourceCode" id="cb1107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1107-1"><a href="clusterMarkerGenesTop.html#cb1107-1"></a>markersWiBatch &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1107-2"><a href="clusterMarkerGenesTop.html#cb1107-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1107-3"><a href="clusterMarkerGenesTop.html#cb1107-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1107-4"><a href="clusterMarkerGenesTop.html#cb1107-4"></a>                     <span class="co">#direction=&quot;up&quot;,</span></span>
<span id="cb1107-5"><a href="clusterMarkerGenesTop.html#cb1107-5"></a>                     <span class="co">#lfc=1,</span></span>
<span id="cb1107-6"><a href="clusterMarkerGenesTop.html#cb1107-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1107-7"><a href="clusterMarkerGenesTop.html#cb1107-7"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1107-8"><a href="clusterMarkerGenesTop.html#cb1107-8"></a>                                           ,</span>
<span id="cb1107-9"><a href="clusterMarkerGenesTop.html#cb1107-9"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span>
<span id="cb1107-10"><a href="clusterMarkerGenesTop.html#cb1107-10"></a></span>
<span id="cb1107-11"><a href="clusterMarkerGenesTop.html#cb1107-11"></a> <span class="co"># A cell type of some sort:</span></span>
<span id="cb1107-12"><a href="clusterMarkerGenesTop.html#cb1107-12"></a>demo &lt;-<span class="st"> </span>markersWiBatch[[<span class="st">&quot;c4&quot;</span>]]</span></code></pre></div>
<div class="sourceCode" id="cb1108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1108-1"><a href="clusterMarkerGenesTop.html#cb1108-1"></a><span class="kw">as.data.frame</span>(demo[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,<span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]) </span></code></pre></div>
<pre><code>##                  Symbol Top       p.value           FDR
## ENSG00000117632   STMN1   1 1.289261e-255 4.287824e-252
## ENSG00000123416  TUBA1B   1 2.206400e-240 5.241461e-237
## ENSG00000124795     DEK   1 5.629134e-102  1.088452e-99
## ENSG00000164104   HMGB2   1 5.471063e-198 5.054350e-195
## ENSG00000177954   RPS27   1 7.116798e-300 5.917262e-296
## ENSG00000189403   HMGB1   1 1.710272e-217 2.187701e-214
## ENSG00000213585   VDAC1   1  3.551566e-62  2.938259e-60
## ENSG00000244734     HBB   1  0.000000e+00  0.000000e+00
## ENSG00000087086     FTL   2 8.984861e-117 2.298604e-114
## ENSG00000117748    RPA2   2  1.813523e-60  1.429245e-58
## ENSG00000124766    SOX4   2  1.835594e-79  2.438379e-77
## ENSG00000196230    TUBB   2 1.033082e-233 1.908792e-230
## ENSG00000206172    HBA1   2 1.091060e-286 6.047743e-283
## ENSG00000122026   RPL21   3 9.779062e-235 2.032700e-231
## ENSG00000128218  VPREB3   3 4.163338e-128 1.357493e-125
## ENSG00000128322   IGLL1   3  1.027465e-91  1.779762e-89
## ENSG00000178741   COX5A   3  4.164852e-59  3.133816e-57
## ENSG00000188536    HBA2   3 7.344382e-284 3.053243e-280
## ENSG00000141552 ANAPC11   4  9.869646e-55  6.361331e-53
## ENSG00000164032   H2AFZ   4 1.942589e-155 1.076777e-152</code></pre>
<div class="sourceCode" id="cb1110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1110-1"><a href="clusterMarkerGenesTop.html#cb1110-1"></a>featx &lt;-<span class="st"> </span>demo[<span class="dv">1</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>]</span>
<span id="cb1110-2"><a href="clusterMarkerGenesTop.html#cb1110-2"></a>featy &lt;-<span class="st"> </span>demo <span class="op">%&gt;%</span><span class="st"> </span>data.frame <span class="op">%&gt;%</span></span>
<span id="cb1110-3"><a href="clusterMarkerGenesTop.html#cb1110-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">==</span><span class="st"> </span>featx) <span class="op">%&gt;%</span></span>
<span id="cb1110-4"><a href="clusterMarkerGenesTop.html#cb1110-4"></a><span class="st">  </span><span class="kw">pull</span>(<span class="st">&quot;Symbol&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1111-1"><a href="clusterMarkerGenesTop.html#cb1111-1"></a><span class="kw">plotExpression</span>(sce,</span>
<span id="cb1111-2"><a href="clusterMarkerGenesTop.html#cb1111-2"></a>               <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(sce<span class="op">$</span>clusterStg)),</span>
<span id="cb1111-3"><a href="clusterMarkerGenesTop.html#cb1111-3"></a>               <span class="dt">features=</span>featx, <span class="co"># &quot;ENSG00000007312&quot;,</span></span>
<span id="cb1111-4"><a href="clusterMarkerGenesTop.html#cb1111-4"></a>               <span class="dt">colour_by=</span><span class="st">&quot;block&quot;</span>) <span class="op">+</span></span>
<span id="cb1111-5"><a href="clusterMarkerGenesTop.html#cb1111-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by) <span class="op">+</span></span>
<span id="cb1111-6"><a href="clusterMarkerGenesTop.html#cb1111-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">glue</span>(<span class="st">&#39;{featx} aka {featy}&#39;</span>))</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode" id="cb1112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1112-1"><a href="clusterMarkerGenesTop.html#cb1112-1"></a>markers &lt;-<span class="st"> </span>markersWiBatch</span></code></pre></div>
<p>Results are compiled in a single table per cluster that stores the outcome of comparisons against the other clusters.
One can then select differentially expressed genes from each pairwise comparison between clusters.</p>
<p>We will define a set of genes for cluster 1 by selecting the top 10 genes of each comparison, and check the test output, eg adjusted p-values and log-fold changes.</p>
<div class="sourceCode" id="cb1113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1113-1"><a href="clusterMarkerGenesTop.html#cb1113-1"></a>clux &lt;-<span class="st"> &quot;c1&quot;</span></span>
<span id="cb1113-2"><a href="clusterMarkerGenesTop.html#cb1113-2"></a></span>
<span id="cb1113-3"><a href="clusterMarkerGenesTop.html#cb1113-3"></a><span class="co"># get output table for cluster of interest:</span></span>
<span id="cb1113-4"><a href="clusterMarkerGenesTop.html#cb1113-4"></a>marker.set &lt;-<span class="st"> </span>markers[[clux]]</span>
<span id="cb1113-5"><a href="clusterMarkerGenesTop.html#cb1113-5"></a><span class="kw">head</span>(marker.set, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 24 columns
##                 ensembl_gene_id external_gene_name chromosome_name
##                     &lt;character&gt;        &lt;character&gt;     &lt;character&gt;
## ENSG00000123416 ENSG00000123416             TUBA1B              12
## ENSG00000124766 ENSG00000124766               SOX4               6
## ENSG00000128218 ENSG00000128218             VPREB3              22
##                 start_position end_position strandNum      Symbol
##                      &lt;integer&gt;    &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## ENSG00000123416       49127782     49131397        -1      TUBA1B
## ENSG00000124766       21593751     21598619         1        SOX4
## ENSG00000128218       23752743     23754425        -1      VPREB3
##                            Type      mean  detected gene_sparsity       Top
##                     &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;     &lt;numeric&gt; &lt;integer&gt;
## ENSG00000123416 Gene Expression   6.13696   55.0540      0.331165         1
## ENSG00000124766 Gene Expression   2.45547   40.8965      0.387268         1
## ENSG00000128218 Gene Expression   1.11073   28.6456      0.547111         1
##                      p.value          FDR summary.logFC   logFC.c2  logFC.c3
##                    &lt;numeric&gt;    &lt;numeric&gt;     &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
## ENSG00000123416 3.67661e-161 6.11383e-158      -1.86877  0.2403615  0.742311
## ENSG00000124766  6.33797e-82  1.69990e-79       1.17633  0.2043029  1.037998
## ENSG00000128218 3.18154e-177 8.81764e-174       1.29414 -0.0149894  1.339206
##                  logFC.c4  logFC.c5  logFC.c6  logFC.c7  logFC.c8  logFC.c9
##                 &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## ENSG00000123416 -1.868766 -0.452086  0.032636  0.591062  0.709009  0.386867
## ENSG00000124766  0.367174  1.139207  0.734617  1.150730  1.090125  0.334170
## ENSG00000128218  0.177627  1.077620  0.855558  0.702230  1.294140  0.376922
##                 logFC.c10
##                 &lt;numeric&gt;
## ENSG00000123416  0.491658
## ENSG00000124766  1.176332
## ENSG00000128218  1.086622</code></pre>
<div class="sourceCode" id="cb1115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1115-1"><a href="clusterMarkerGenesTop.html#cb1115-1"></a><span class="co"># add gene annotation:</span></span>
<span id="cb1115-2"><a href="clusterMarkerGenesTop.html#cb1115-2"></a><span class="co">##tmpDf &lt;- marker.set</span></span>
<span id="cb1115-3"><a href="clusterMarkerGenesTop.html#cb1115-3"></a><span class="co">##tmpDf$ensembl_gene_id &lt;- rownames(tmpDf)</span></span>
<span id="cb1115-4"><a href="clusterMarkerGenesTop.html#cb1115-4"></a><span class="co">##tmpDf2 &lt;- base::merge(tmpDf, rowData(sce), by=&quot;ensembl_gene_id&quot;, all.x=TRUE, all.y=F, sort=F)</span></span></code></pre></div>
<p>Write table to file:</p>
<div class="sourceCode" id="cb1116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1116-1"><a href="clusterMarkerGenesTop.html#cb1116-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_%s.tsv&quot;</span>,</span>
<span id="cb1116-2"><a href="clusterMarkerGenesTop.html#cb1116-2"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2, clux)</span>
<span id="cb1116-3"><a href="clusterMarkerGenesTop.html#cb1116-3"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1_c1.tsv&quot;</code></pre>
<div class="sourceCode" id="cb1118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1118-1"><a href="clusterMarkerGenesTop.html#cb1118-1"></a><span class="kw">write.table</span>(marker.set, <span class="dt">file=</span>tmpFn, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</span>
<span id="cb1118-2"><a href="clusterMarkerGenesTop.html#cb1118-2"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>Show expression of marker on t-SNE and UMAP:</p>
<div class="sourceCode" id="cb1119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1119-1"><a href="clusterMarkerGenesTop.html#cb1119-1"></a>mrkSet &lt;-<span class="st"> </span>marker.set <span class="op">%&gt;%</span></span>
<span id="cb1119-2"><a href="clusterMarkerGenesTop.html#cb1119-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1119-3"><a href="clusterMarkerGenesTop.html#cb1119-3"></a><span class="st">  </span><span class="kw">filter</span>(Symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">rowData</span>(mnn.out)<span class="op">$</span>Symbol) <span class="op">%&gt;%</span></span>
<span id="cb1119-4"><a href="clusterMarkerGenesTop.html#cb1119-4"></a><span class="st">  </span><span class="kw">arrange</span>(FDR)</span>
<span id="cb1119-5"><a href="clusterMarkerGenesTop.html#cb1119-5"></a></span>
<span id="cb1119-6"><a href="clusterMarkerGenesTop.html#cb1119-6"></a>geneInd &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb1119-7"><a href="clusterMarkerGenesTop.html#cb1119-7"></a>geneEns &lt;-<span class="st"> </span>mrkSet[geneInd,<span class="st">&quot;ensembl_gene_id&quot;</span>]</span>
<span id="cb1119-8"><a href="clusterMarkerGenesTop.html#cb1119-8"></a>geneExt &lt;-<span class="st"> </span>mrkSet[geneInd,<span class="st">&quot;external_gene_name&quot;</span>] <span class="co"># external_gene_name ~ Symbol</span></span>
<span id="cb1119-9"><a href="clusterMarkerGenesTop.html#cb1119-9"></a></span>
<span id="cb1119-10"><a href="clusterMarkerGenesTop.html#cb1119-10"></a><span class="co">#tsne1 &lt;- plotTSNE(sce, colour_by=mrkSet[geneInd,&quot;ensembl_gene_id&quot;]) + fontsize</span></span>
<span id="cb1119-11"><a href="clusterMarkerGenesTop.html#cb1119-11"></a>tsne1 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span>geneEns) <span class="op">+</span></span>
<span id="cb1119-12"><a href="clusterMarkerGenesTop.html#cb1119-12"></a><span class="st">  </span>fontsize <span class="op">+</span></span>
<span id="cb1119-13"><a href="clusterMarkerGenesTop.html#cb1119-13"></a><span class="st">  </span><span class="kw">scale_color_continuous</span>(<span class="dt">type =</span> <span class="st">&#39;viridis&#39;</span>,</span>
<span id="cb1119-14"><a href="clusterMarkerGenesTop.html#cb1119-14"></a>                         <span class="dt">guide =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> geneExt))</span>
<span id="cb1119-15"><a href="clusterMarkerGenesTop.html#cb1119-15"></a></span>
<span id="cb1119-16"><a href="clusterMarkerGenesTop.html#cb1119-16"></a>umap1 &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(sce, <span class="dt">colour_by=</span>geneEns) <span class="op">+</span></span>
<span id="cb1119-17"><a href="clusterMarkerGenesTop.html#cb1119-17"></a><span class="st">  </span>fontsize <span class="op">+</span></span>
<span id="cb1119-18"><a href="clusterMarkerGenesTop.html#cb1119-18"></a><span class="st">  </span><span class="kw">scale_color_continuous</span>(<span class="dt">type =</span> <span class="st">&#39;viridis&#39;</span>,</span>
<span id="cb1119-19"><a href="clusterMarkerGenesTop.html#cb1119-19"></a>                         <span class="dt">guide =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> geneExt))</span>
<span id="cb1119-20"><a href="clusterMarkerGenesTop.html#cb1119-20"></a></span>
<span id="cb1119-21"><a href="clusterMarkerGenesTop.html#cb1119-21"></a>tsne1b &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb1119-22"><a href="clusterMarkerGenesTop.html#cb1119-22"></a>                   <span class="dt">colour_by=</span>geneExt,</span>
<span id="cb1119-23"><a href="clusterMarkerGenesTop.html#cb1119-23"></a>                   <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1119-24"><a href="clusterMarkerGenesTop.html#cb1119-24"></a></span>
<span id="cb1119-25"><a href="clusterMarkerGenesTop.html#cb1119-25"></a>umap1b &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(mnn.out,</span>
<span id="cb1119-26"><a href="clusterMarkerGenesTop.html#cb1119-26"></a>                   <span class="dt">colour_by=</span>mrkSet[geneInd,<span class="st">&quot;Symbol&quot;</span>],</span>
<span id="cb1119-27"><a href="clusterMarkerGenesTop.html#cb1119-27"></a>                   <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span></code></pre></div>
<p>Expresion levels of ENSG00000244734 aka HBB:</p>
<div class="sourceCode" id="cb1120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1120-1"><a href="clusterMarkerGenesTop.html#cb1120-1"></a><span class="kw">plotExpression</span>(sce,</span>
<span id="cb1120-2"><a href="clusterMarkerGenesTop.html#cb1120-2"></a>               <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(sce<span class="op">$</span>clusterStg)),</span>
<span id="cb1120-3"><a href="clusterMarkerGenesTop.html#cb1120-3"></a>               <span class="dt">features=</span>geneEns, <span class="co"># &quot;ENSG00000007312&quot;,</span></span>
<span id="cb1120-4"><a href="clusterMarkerGenesTop.html#cb1120-4"></a>               <span class="dt">colour_by=</span><span class="st">&quot;block&quot;</span>) <span class="op">+</span></span>
<span id="cb1120-5"><a href="clusterMarkerGenesTop.html#cb1120-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by) <span class="op">+</span></span>
<span id="cb1120-6"><a href="clusterMarkerGenesTop.html#cb1120-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">glue</span>(<span class="st">&#39;{geneEns} aka {geneExt}&#39;</span>))</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Without data set integration (sce)</p>
<div class="sourceCode" id="cb1121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1121-1"><a href="clusterMarkerGenesTop.html#cb1121-1"></a><span class="co">#tsne1</span></span>
<span id="cb1121-2"><a href="clusterMarkerGenesTop.html#cb1121-2"></a><span class="co">#umap1</span></span>
<span id="cb1121-3"><a href="clusterMarkerGenesTop.html#cb1121-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(tsne1, umap1, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-10-1.png" width="672" />
With data set integration (mnn.out):</p>
<div class="sourceCode" id="cb1122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1122-1"><a href="clusterMarkerGenesTop.html#cb1122-1"></a><span class="co">#tsne1b</span></span>
<span id="cb1122-2"><a href="clusterMarkerGenesTop.html#cb1122-2"></a><span class="co">#umap1b</span></span>
<span id="cb1122-3"><a href="clusterMarkerGenesTop.html#cb1122-3"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(tsne1b, umap1b, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>With data integration (mnn.out), TSNE with leiden clusters and HBB levels:</p>
<div class="sourceCode" id="cb1123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1123-1"><a href="clusterMarkerGenesTop.html#cb1123-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(tsney, tsne1b, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>Without data integration (sce), TSNE with leiden clusters and HBB levels:</p>
<div class="sourceCode" id="cb1124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1124-1"><a href="clusterMarkerGenesTop.html#cb1124-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(tsnex, tsne1, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>With data integration (mnn.out), UMAP with leiden clusters and HBB levels:</p>
<div class="sourceCode" id="cb1125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1125-1"><a href="clusterMarkerGenesTop.html#cb1125-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(umapy, umap1b, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Without data integration (sce), UMAP with leiden clusters and HBB levels:</p>
<div class="sourceCode" id="cb1126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1126-1"><a href="clusterMarkerGenesTop.html#cb1126-1"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(umapx, umap1, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb1127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1127-1"><a href="clusterMarkerGenesTop.html#cb1127-1"></a><span class="kw">rm</span>(tsne1, tsne1b, umap1, umap1b)</span>
<span id="cb1127-2"><a href="clusterMarkerGenesTop.html#cb1127-2"></a><span class="kw">rm</span>(tsnex, tsney, umapx, umapy)</span></code></pre></div>
<p>Gene set enrichment analyses used for bulk RNA-seq may be used to characterise clusters further.</p>
</div>
<div id="heatmap" class="section level3">
<h3><span class="header-section-number">15.3.2</span> Heatmap</h3>
<p>As for bulk RNA, differences in expression profiles of the top genes can be visualised with a heatmap.</p>
<p>Normalised counts:</p>
<div class="sourceCode" id="cb1128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1128-1"><a href="clusterMarkerGenesTop.html#cb1128-1"></a><span class="co"># select some top genes:</span></span>
<span id="cb1128-2"><a href="clusterMarkerGenesTop.html#cb1128-2"></a>top.markers &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.set)[marker.set<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">10</span>]</span>
<span id="cb1128-3"><a href="clusterMarkerGenesTop.html#cb1128-3"></a></span>
<span id="cb1128-4"><a href="clusterMarkerGenesTop.html#cb1128-4"></a><span class="co"># have matrix to annotate sample with cluster and sample:</span></span>
<span id="cb1128-5"><a href="clusterMarkerGenesTop.html#cb1128-5"></a>tmpData &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce)[top.markers,]</span>
<span id="cb1128-6"><a href="clusterMarkerGenesTop.html#cb1128-6"></a><span class="co"># concat sample and barcode names to make unique name across the whole data set</span></span>
<span id="cb1128-7"><a href="clusterMarkerGenesTop.html#cb1128-7"></a>tmpCellNames &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name, <span class="kw">colData</span>(sce)<span class="op">$</span>Barcode, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1128-8"><a href="clusterMarkerGenesTop.html#cb1128-8"></a><span class="co"># use these to namecolumn of matrix the show as heatmap:</span></span>
<span id="cb1128-9"><a href="clusterMarkerGenesTop.html#cb1128-9"></a><span class="kw">colnames</span>(tmpData) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode                    </span></span>
<span id="cb1128-10"><a href="clusterMarkerGenesTop.html#cb1128-10"></a></span>
<span id="cb1128-11"><a href="clusterMarkerGenesTop.html#cb1128-11"></a><span class="co"># columns annotation with cell name:</span></span>
<span id="cb1128-12"><a href="clusterMarkerGenesTop.html#cb1128-12"></a>mat_col &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">cluster =</span> sce<span class="op">$</span>cluster,</span>
<span id="cb1128-13"><a href="clusterMarkerGenesTop.html#cb1128-13"></a>              <span class="dt">sample =</span> sce<span class="op">$</span>Sample.Name,</span>
<span id="cb1128-14"><a href="clusterMarkerGenesTop.html#cb1128-14"></a>              <span class="dt">type =</span> sce<span class="op">$</span>source_name</span>
<span id="cb1128-15"><a href="clusterMarkerGenesTop.html#cb1128-15"></a>        )</span>
<span id="cb1128-16"><a href="clusterMarkerGenesTop.html#cb1128-16"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span><span class="kw">colnames</span>(tmpData)</span>
<span id="cb1128-17"><a href="clusterMarkerGenesTop.html#cb1128-17"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode</span></span>
<span id="cb1128-18"><a href="clusterMarkerGenesTop.html#cb1128-18"></a></span>
<span id="cb1128-19"><a href="clusterMarkerGenesTop.html#cb1128-19"></a><span class="co"># Prepare colours for clusters:</span></span>
<span id="cb1128-20"><a href="clusterMarkerGenesTop.html#cb1128-20"></a>colourCount =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(sce<span class="op">$</span>cluster))</span>
<span id="cb1128-21"><a href="clusterMarkerGenesTop.html#cb1128-21"></a>getPalette =<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb1128-22"><a href="clusterMarkerGenesTop.html#cb1128-22"></a></span>
<span id="cb1128-23"><a href="clusterMarkerGenesTop.html#cb1128-23"></a>mat_colors &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">group =</span> <span class="kw">getPalette</span>(colourCount))</span>
<span id="cb1128-24"><a href="clusterMarkerGenesTop.html#cb1128-24"></a><span class="kw">names</span>(mat_colors<span class="op">$</span>group) &lt;-<span class="st"> </span><span class="kw">unique</span>(sce<span class="op">$</span>cluster)</span>
<span id="cb1128-25"><a href="clusterMarkerGenesTop.html#cb1128-25"></a></span>
<span id="cb1128-26"><a href="clusterMarkerGenesTop.html#cb1128-26"></a>breaksVec =<span class="st"> </span><span class="kw">seq</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="dt">by =</span> <span class="fl">0.1</span>)</span>
<span id="cb1128-27"><a href="clusterMarkerGenesTop.html#cb1128-27"></a></span>
<span id="cb1128-28"><a href="clusterMarkerGenesTop.html#cb1128-28"></a><span class="co"># plot heatmap:</span></span>
<span id="cb1128-29"><a href="clusterMarkerGenesTop.html#cb1128-29"></a><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb1128-30"><a href="clusterMarkerGenesTop.html#cb1128-30"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1128-31"><a href="clusterMarkerGenesTop.html#cb1128-31"></a>           <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1128-32"><a href="clusterMarkerGenesTop.html#cb1128-32"></a>           <span class="co">#show_rownames     = FALSE,</span></span>
<span id="cb1128-33"><a href="clusterMarkerGenesTop.html#cb1128-33"></a>           <span class="dt">show_rownames     =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1128-34"><a href="clusterMarkerGenesTop.html#cb1128-34"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1128-35"><a href="clusterMarkerGenesTop.html#cb1128-35"></a>           <span class="dt">labels_row        =</span> <span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(tmpData),<span class="st">&quot;Symbol&quot;</span>],</span>
<span id="cb1128-36"><a href="clusterMarkerGenesTop.html#cb1128-36"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1128-37"><a href="clusterMarkerGenesTop.html#cb1128-37"></a>           <span class="dt">annotation_colors =</span> mat_colors,</span>
<span id="cb1128-38"><a href="clusterMarkerGenesTop.html#cb1128-38"></a>           <span class="dt">color             =</span> <span class="kw">colorRampPalette</span>(</span>
<span id="cb1128-39"><a href="clusterMarkerGenesTop.html#cb1128-39"></a>             <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">7</span>,</span>
<span id="cb1128-40"><a href="clusterMarkerGenesTop.html#cb1128-40"></a>                            <span class="dt">name =</span> <span class="st">&quot;RdYlBu&quot;</span>)))(<span class="kw">length</span>(breaksVec)),</span>
<span id="cb1128-41"><a href="clusterMarkerGenesTop.html#cb1128-41"></a>           <span class="dt">breaks            =</span> breaksVec,</span>
<span id="cb1128-42"><a href="clusterMarkerGenesTop.html#cb1128-42"></a>       <span class="dt">fontsize_row      =</span> <span class="dv">7</span></span>
<span id="cb1128-43"><a href="clusterMarkerGenesTop.html#cb1128-43"></a>           )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/marker_set_clu1_heatmap_unsorted-1.png" width="672" /></p>
<div class="sourceCode" id="cb1129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1129-1"><a href="clusterMarkerGenesTop.html#cb1129-1"></a><span class="co"># remove MALAT1</span></span>
<span id="cb1129-2"><a href="clusterMarkerGenesTop.html#cb1129-2"></a>tmpData2 &lt;-<span class="st"> </span>tmpData <span class="op">%&gt;%</span></span>
<span id="cb1129-3"><a href="clusterMarkerGenesTop.html#cb1129-3"></a><span class="st">  </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span></span>
<span id="cb1129-4"><a href="clusterMarkerGenesTop.html#cb1129-4"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1129-5"><a href="clusterMarkerGenesTop.html#cb1129-5"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;ensId&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1129-6"><a href="clusterMarkerGenesTop.html#cb1129-6"></a><span class="st">  </span><span class="kw">filter</span>(ensId <span class="op">!=</span><span class="st"> &quot;ENSG00000251562&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1129-7"><a href="clusterMarkerGenesTop.html#cb1129-7"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="st">&quot;ensId&quot;</span>)</span>
<span id="cb1129-8"><a href="clusterMarkerGenesTop.html#cb1129-8"></a><span class="kw">colnames</span>(tmpData2) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(tmpData2))</span>
<span id="cb1129-9"><a href="clusterMarkerGenesTop.html#cb1129-9"></a></span>
<span id="cb1129-10"><a href="clusterMarkerGenesTop.html#cb1129-10"></a><span class="kw">pheatmap</span>(tmpData2,</span>
<span id="cb1129-11"><a href="clusterMarkerGenesTop.html#cb1129-11"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1129-12"><a href="clusterMarkerGenesTop.html#cb1129-12"></a>           <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1129-13"><a href="clusterMarkerGenesTop.html#cb1129-13"></a>           <span class="co">#show_rownames     = FALSE,</span></span>
<span id="cb1129-14"><a href="clusterMarkerGenesTop.html#cb1129-14"></a>           <span class="dt">show_rownames     =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1129-15"><a href="clusterMarkerGenesTop.html#cb1129-15"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1129-16"><a href="clusterMarkerGenesTop.html#cb1129-16"></a>           <span class="dt">labels_row        =</span> <span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(tmpData2),<span class="st">&quot;Symbol&quot;</span>],</span>
<span id="cb1129-17"><a href="clusterMarkerGenesTop.html#cb1129-17"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1129-18"><a href="clusterMarkerGenesTop.html#cb1129-18"></a>           <span class="dt">annotation_colors =</span> mat_colors,</span>
<span id="cb1129-19"><a href="clusterMarkerGenesTop.html#cb1129-19"></a>           <span class="dt">color             =</span> <span class="kw">colorRampPalette</span>(</span>
<span id="cb1129-20"><a href="clusterMarkerGenesTop.html#cb1129-20"></a>             <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">7</span>,</span>
<span id="cb1129-21"><a href="clusterMarkerGenesTop.html#cb1129-21"></a>                            <span class="dt">name =</span> <span class="st">&quot;RdYlBu&quot;</span>)))(<span class="kw">length</span>(breaksVec)),</span>
<span id="cb1129-22"><a href="clusterMarkerGenesTop.html#cb1129-22"></a>           <span class="dt">breaks            =</span> breaksVec,</span>
<span id="cb1129-23"><a href="clusterMarkerGenesTop.html#cb1129-23"></a>       <span class="dt">fontsize_row      =</span> <span class="dv">7</span></span>
<span id="cb1129-24"><a href="clusterMarkerGenesTop.html#cb1129-24"></a>           )</span>
<span id="cb1129-25"><a href="clusterMarkerGenesTop.html#cb1129-25"></a><span class="kw">rm</span>(tmpData2)</span></code></pre></div>
<!--
One can sort both the gene and sample dendrograms to improve the heatmap.
-->
<div class="sourceCode" id="cb1130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1130-1"><a href="clusterMarkerGenesTop.html#cb1130-1"></a><span class="kw">library</span>(dendsort)</span>
<span id="cb1130-2"><a href="clusterMarkerGenesTop.html#cb1130-2"></a></span>
<span id="cb1130-3"><a href="clusterMarkerGenesTop.html#cb1130-3"></a>mat &lt;-<span class="st"> </span>tmpData</span>
<span id="cb1130-4"><a href="clusterMarkerGenesTop.html#cb1130-4"></a>mat_cluster_cols &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(<span class="kw">t</span>(mat)))</span>
<span id="cb1130-5"><a href="clusterMarkerGenesTop.html#cb1130-5"></a></span>
<span id="cb1130-6"><a href="clusterMarkerGenesTop.html#cb1130-6"></a>sort_hclust &lt;-<span class="st"> </span><span class="cf">function</span>(...) <span class="kw">as.hclust</span>(<span class="kw">dendsort</span>(<span class="kw">as.dendrogram</span>(...)))</span>
<span id="cb1130-7"><a href="clusterMarkerGenesTop.html#cb1130-7"></a></span>
<span id="cb1130-8"><a href="clusterMarkerGenesTop.html#cb1130-8"></a>mat_cluster_cols &lt;-<span class="st"> </span><span class="kw">sort_hclust</span>(mat_cluster_cols)</span>
<span id="cb1130-9"><a href="clusterMarkerGenesTop.html#cb1130-9"></a><span class="co">#plot(mat_cluster_cols, main = &quot;Sorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)</span></span>
<span id="cb1130-10"><a href="clusterMarkerGenesTop.html#cb1130-10"></a></span>
<span id="cb1130-11"><a href="clusterMarkerGenesTop.html#cb1130-11"></a>mat_cluster_rows &lt;-<span class="st"> </span><span class="kw">sort_hclust</span>(<span class="kw">hclust</span>(<span class="kw">dist</span>(mat)))</span>
<span id="cb1130-12"><a href="clusterMarkerGenesTop.html#cb1130-12"></a><span class="kw">rm</span>(mat)</span>
<span id="cb1130-13"><a href="clusterMarkerGenesTop.html#cb1130-13"></a></span>
<span id="cb1130-14"><a href="clusterMarkerGenesTop.html#cb1130-14"></a><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb1130-15"><a href="clusterMarkerGenesTop.html#cb1130-15"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1130-16"><a href="clusterMarkerGenesTop.html#cb1130-16"></a>           <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1130-17"><a href="clusterMarkerGenesTop.html#cb1130-17"></a>           <span class="dt">show_rownames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1130-18"><a href="clusterMarkerGenesTop.html#cb1130-18"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1130-19"><a href="clusterMarkerGenesTop.html#cb1130-19"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1130-20"><a href="clusterMarkerGenesTop.html#cb1130-20"></a>           <span class="dt">annotation_colors =</span> mat_colors,</span>
<span id="cb1130-21"><a href="clusterMarkerGenesTop.html#cb1130-21"></a>           <span class="dt">cluster_cols      =</span> mat_cluster_cols,</span>
<span id="cb1130-22"><a href="clusterMarkerGenesTop.html#cb1130-22"></a>           <span class="dt">cluster_rows      =</span> mat_cluster_rows</span>
<span id="cb1130-23"><a href="clusterMarkerGenesTop.html#cb1130-23"></a>         )</span></code></pre></div>
<p>Counts corrected for batch effect (‘corrected’):</p>
<div class="sourceCode" id="cb1131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1131-1"><a href="clusterMarkerGenesTop.html#cb1131-1"></a><span class="co"># select some top genes:</span></span>
<span id="cb1131-2"><a href="clusterMarkerGenesTop.html#cb1131-2"></a><span class="co">#top.markers &lt;- rownames(marker.set)[marker.set$Top &lt;= 10]</span></span>
<span id="cb1131-3"><a href="clusterMarkerGenesTop.html#cb1131-3"></a>top.markers &lt;-<span class="st"> </span>marker.set[marker.set<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">10</span>,] <span class="op">%&gt;%</span></span>
<span id="cb1131-4"><a href="clusterMarkerGenesTop.html#cb1131-4"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1131-5"><a href="clusterMarkerGenesTop.html#cb1131-5"></a><span class="st">  </span><span class="kw">pull</span>(Symbol)</span>
<span id="cb1131-6"><a href="clusterMarkerGenesTop.html#cb1131-6"></a></span>
<span id="cb1131-7"><a href="clusterMarkerGenesTop.html#cb1131-7"></a><span class="co"># have matrix to annotate sample with cluster and sample:</span></span>
<span id="cb1131-8"><a href="clusterMarkerGenesTop.html#cb1131-8"></a>tmpData3 &lt;-<span class="st"> </span><span class="kw">assay</span>(mnn.out) <span class="op">%&gt;%</span></span>
<span id="cb1131-9"><a href="clusterMarkerGenesTop.html#cb1131-9"></a><span class="st">  </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span></span>
<span id="cb1131-10"><a href="clusterMarkerGenesTop.html#cb1131-10"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1131-11"><a href="clusterMarkerGenesTop.html#cb1131-11"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;Symbol&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb1131-12"><a href="clusterMarkerGenesTop.html#cb1131-12"></a><span class="st">  </span><span class="kw">filter</span>(Symbol <span class="op">%in%</span><span class="st"> </span>top.markers) <span class="op">%&gt;%</span></span>
<span id="cb1131-13"><a href="clusterMarkerGenesTop.html#cb1131-13"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="st">&quot;Symbol&quot;</span>)</span>
<span id="cb1131-14"><a href="clusterMarkerGenesTop.html#cb1131-14"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out)<span class="op">$</span>Barcode</span>
<span id="cb1131-15"><a href="clusterMarkerGenesTop.html#cb1131-15"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(tmpData3))</span>
<span id="cb1131-16"><a href="clusterMarkerGenesTop.html#cb1131-16"></a></span>
<span id="cb1131-17"><a href="clusterMarkerGenesTop.html#cb1131-17"></a><span class="co"># concat sample and barcode names to make unique name across the whole data set</span></span>
<span id="cb1131-18"><a href="clusterMarkerGenesTop.html#cb1131-18"></a>tmpCellNames &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">colData</span>(mnn.out)<span class="op">$</span>Sample.Name,</span>
<span id="cb1131-19"><a href="clusterMarkerGenesTop.html#cb1131-19"></a>                      <span class="kw">colData</span>(mnn.out)<span class="op">$</span>Barcode, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1131-20"><a href="clusterMarkerGenesTop.html#cb1131-20"></a><span class="co"># use these to name columns of matrix the show as heatmap:</span></span>
<span id="cb1131-21"><a href="clusterMarkerGenesTop.html#cb1131-21"></a><span class="kw">colnames</span>(tmpData3) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode                    </span></span>
<span id="cb1131-22"><a href="clusterMarkerGenesTop.html#cb1131-22"></a></span>
<span id="cb1131-23"><a href="clusterMarkerGenesTop.html#cb1131-23"></a><span class="co"># columns annotation with cell name:</span></span>
<span id="cb1131-24"><a href="clusterMarkerGenesTop.html#cb1131-24"></a>mat_col &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">cluster =</span> sce<span class="op">$</span>cluster,</span>
<span id="cb1131-25"><a href="clusterMarkerGenesTop.html#cb1131-25"></a>              <span class="dt">sample =</span> sce<span class="op">$</span>Sample.Name,</span>
<span id="cb1131-26"><a href="clusterMarkerGenesTop.html#cb1131-26"></a>              <span class="dt">type =</span> sce<span class="op">$</span>source_name</span>
<span id="cb1131-27"><a href="clusterMarkerGenesTop.html#cb1131-27"></a>        )</span>
<span id="cb1131-28"><a href="clusterMarkerGenesTop.html#cb1131-28"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span><span class="kw">colnames</span>(tmpData)</span>
<span id="cb1131-29"><a href="clusterMarkerGenesTop.html#cb1131-29"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span>tmpCellNames <span class="co"># colData(sce)$Barcode</span></span>
<span id="cb1131-30"><a href="clusterMarkerGenesTop.html#cb1131-30"></a></span>
<span id="cb1131-31"><a href="clusterMarkerGenesTop.html#cb1131-31"></a><span class="co"># Prepare colours for clusters:</span></span>
<span id="cb1131-32"><a href="clusterMarkerGenesTop.html#cb1131-32"></a>colourCount =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(sce<span class="op">$</span>cluster))</span>
<span id="cb1131-33"><a href="clusterMarkerGenesTop.html#cb1131-33"></a>getPalette =<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb1131-34"><a href="clusterMarkerGenesTop.html#cb1131-34"></a></span>
<span id="cb1131-35"><a href="clusterMarkerGenesTop.html#cb1131-35"></a>mat_colors &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">group =</span> <span class="kw">getPalette</span>(colourCount))</span>
<span id="cb1131-36"><a href="clusterMarkerGenesTop.html#cb1131-36"></a><span class="kw">names</span>(mat_colors<span class="op">$</span>group) &lt;-<span class="st"> </span><span class="kw">unique</span>(sce<span class="op">$</span>cluster)</span>
<span id="cb1131-37"><a href="clusterMarkerGenesTop.html#cb1131-37"></a></span>
<span id="cb1131-38"><a href="clusterMarkerGenesTop.html#cb1131-38"></a><span class="co"># plot heatmap:</span></span>
<span id="cb1131-39"><a href="clusterMarkerGenesTop.html#cb1131-39"></a><span class="kw">pheatmap</span>(tmpData3[,<span class="kw">order</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>clusterStg)],</span>
<span id="cb1131-40"><a href="clusterMarkerGenesTop.html#cb1131-40"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb1131-41"><a href="clusterMarkerGenesTop.html#cb1131-41"></a>           <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1131-42"><a href="clusterMarkerGenesTop.html#cb1131-42"></a>           <span class="co">#show_rownames     = FALSE,</span></span>
<span id="cb1131-43"><a href="clusterMarkerGenesTop.html#cb1131-43"></a>           <span class="dt">show_rownames     =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1131-44"><a href="clusterMarkerGenesTop.html#cb1131-44"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb1131-45"><a href="clusterMarkerGenesTop.html#cb1131-45"></a>           <span class="dt">labels_row        =</span> <span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(tmpData),<span class="st">&quot;Symbol&quot;</span>],</span>
<span id="cb1131-46"><a href="clusterMarkerGenesTop.html#cb1131-46"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1131-47"><a href="clusterMarkerGenesTop.html#cb1131-47"></a>           <span class="dt">annotation_colors =</span> mat_colors,</span>
<span id="cb1131-48"><a href="clusterMarkerGenesTop.html#cb1131-48"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1131-49"><a href="clusterMarkerGenesTop.html#cb1131-49"></a>           <span class="dt">fontsize_row      =</span> <span class="dv">7</span></span>
<span id="cb1131-50"><a href="clusterMarkerGenesTop.html#cb1131-50"></a>           )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/marker_set_clu1_heatmap_unsorted_recon-1.png" width="672" /></p>
<div class="sourceCode" id="cb1132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1132-1"><a href="clusterMarkerGenesTop.html#cb1132-1"></a><span class="kw">rm</span>(tmpData3)</span>
<span id="cb1132-2"><a href="clusterMarkerGenesTop.html#cb1132-2"></a><span class="kw">rm</span>(mnn.out, tmpData)</span></code></pre></div>
<p>To demonstrate how to interpret the results, we will use cluster 9 as our cluster of interest. The relevant DataFrame contains log2-fold changes of expression in cluster 9 over each other cluster, along with several statistics obtained by combining p-values (Simes 1986) across the pairwise comparisons involving cluster 9.</p>
<div class="sourceCode" id="cb1133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1133-1"><a href="clusterMarkerGenesTop.html#cb1133-1"></a>chosen &lt;-<span class="st"> &quot;c9&quot;</span></span>
<span id="cb1133-2"><a href="clusterMarkerGenesTop.html#cb1133-2"></a>interesting &lt;-<span class="st"> </span>markers[[chosen]]</span>
<span id="cb1133-3"><a href="clusterMarkerGenesTop.html#cb1133-3"></a><span class="kw">print</span>(<span class="kw">colnames</span>(interesting))</span></code></pre></div>
<pre><code>##  [1] &quot;ensembl_gene_id&quot;    &quot;external_gene_name&quot; &quot;chromosome_name&quot;   
##  [4] &quot;start_position&quot;     &quot;end_position&quot;       &quot;strandNum&quot;         
##  [7] &quot;Symbol&quot;             &quot;Type&quot;               &quot;mean&quot;              
## [10] &quot;detected&quot;           &quot;gene_sparsity&quot;      &quot;Top&quot;               
## [13] &quot;p.value&quot;            &quot;FDR&quot;                &quot;summary.logFC&quot;     
## [16] &quot;logFC.c1&quot;           &quot;logFC.c2&quot;           &quot;logFC.c3&quot;          
## [19] &quot;logFC.c4&quot;           &quot;logFC.c5&quot;           &quot;logFC.c6&quot;          
## [22] &quot;logFC.c7&quot;           &quot;logFC.c8&quot;           &quot;logFC.c10&quot;</code></pre>
<p>Of particular interest is the Top field. The set of genes with Top ≤X is the union of the top X genes (ranked by p-value) from each pairwise comparison involving cluster 9. For example, the set of all genes with Top values of 1 contains the gene with the lowest p-value from each comparison. Similarly, the set of genes with Top values less than or equal to 10 contains the top 10 genes from each comparison. The Top field represents findMarkers()’s approach to consolidating multiple pairwise comparisons into a single ranking for each cluster; each DataFrame produced by findMarkers() will order genes based on the Top value by default.</p>
<div class="sourceCode" id="cb1135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1135-1"><a href="clusterMarkerGenesTop.html#cb1135-1"></a>interesting.coluToShow &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;FDR&quot;</span>, <span class="st">&quot;summary.logFC&quot;</span>)</span>
<span id="cb1135-2"><a href="clusterMarkerGenesTop.html#cb1135-2"></a>interesting[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.coluToShow]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 4 columns
##                      Symbol       Top          FDR summary.logFC
##                 &lt;character&gt; &lt;integer&gt;    &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000113387        SUB1         1  6.32981e-12     -2.745184
## ENSG00000127314       RAP1B         1  3.33268e-39     -0.613188
## ENSG00000147454    SLC25A37         1 1.71443e-142     -4.430710
## ENSG00000169100     SLC25A6         1  1.04622e-41     -1.297843
## ENSG00000198851        CD3E         1  4.16157e-66     -1.126555
## ENSG00000205542      TMSB4X         1  6.87247e-18     -2.391993
## ENSG00000211898        IGHD         1  5.34296e-25     -1.270077
## ENSG00000213585       VDAC1         1  3.11388e-65     -0.358800
## ENSG00000105374        NKG7         2  4.07717e-17     -1.765356
## ENSG00000117748        RPA2         2  1.03990e-57     -0.337331</code></pre>
<p>We use the Top field to identify a set of genes that is guaranteed to distinguish cluster 9 from any other cluster. Here, we examine the top 6 genes from each pairwise comparison.</p>
<div class="sourceCode" id="cb1137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1137-1"><a href="clusterMarkerGenesTop.html#cb1137-1"></a>best.set &lt;-<span class="st"> </span>interesting[interesting<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">6</span>,]</span>
<span id="cb1137-2"><a href="clusterMarkerGenesTop.html#cb1137-2"></a>logFCs &lt;-<span class="st"> </span><span class="kw">getMarkerEffects</span>(best.set)</span>
<span id="cb1137-3"><a href="clusterMarkerGenesTop.html#cb1137-3"></a>logFCs.ens &lt;-<span class="st"> </span><span class="kw">rownames</span>(logFCs)</span>
<span id="cb1137-4"><a href="clusterMarkerGenesTop.html#cb1137-4"></a><span class="kw">rownames</span>(logFCs) &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(logFCs), <span class="st">&quot;Symbol&quot;</span>]</span>
<span id="cb1137-5"><a href="clusterMarkerGenesTop.html#cb1137-5"></a></span>
<span id="cb1137-6"><a href="clusterMarkerGenesTop.html#cb1137-6"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1137-7"><a href="clusterMarkerGenesTop.html#cb1137-7"></a><span class="kw">pheatmap</span>(logFCs, <span class="dt">breaks=</span><span class="kw">seq</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="dt">length.out=</span><span class="dv">101</span>))</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-19-1.png" width="576" /></p>
</div>
</div>
<div id="using-the-log-fold-change" class="section level2">
<h2><span class="header-section-number">15.4</span> Using the log-fold change</h2>
<p>Our previous findMarkers() call considers both up- and downregulated genes to be potential markers. However, downregulated genes are less appealing as markers as it is more difficult to interpret and experimentally validate an absence of expression. To focus on up-regulated markers, we can instead perform a one-sided t-test to identify genes that are upregulated in each cluster compared to the others. This is achieved by setting direction=“up” in the findMarkers() call.</p>
<div class="sourceCode" id="cb1138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1138-1"><a href="clusterMarkerGenesTop.html#cb1138-1"></a>markersWoBatch.up &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1138-2"><a href="clusterMarkerGenesTop.html#cb1138-2"></a>                 <span class="dt">groups =</span> sce<span class="op">$</span>clusterStg,</span>
<span id="cb1138-3"><a href="clusterMarkerGenesTop.html#cb1138-3"></a>                 <span class="dt">direction =</span> <span class="st">&quot;up&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1139-1"><a href="clusterMarkerGenesTop.html#cb1139-1"></a>markersWiBatch.up &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x =</span> sce,</span>
<span id="cb1139-2"><a href="clusterMarkerGenesTop.html#cb1139-2"></a>                     <span class="dt">groups =</span> sce<span class="op">$</span>clusterStg,</span>
<span id="cb1139-3"><a href="clusterMarkerGenesTop.html#cb1139-3"></a>                     <span class="dt">block =</span> sce<span class="op">$</span>block,</span>
<span id="cb1139-4"><a href="clusterMarkerGenesTop.html#cb1139-4"></a>                     <span class="dt">direction =</span> <span class="st">&quot;up&quot;</span>,</span>
<span id="cb1139-5"><a href="clusterMarkerGenesTop.html#cb1139-5"></a>                     <span class="co">#lfc=1,</span></span>
<span id="cb1139-6"><a href="clusterMarkerGenesTop.html#cb1139-6"></a>                     <span class="dt">row.data =</span> <span class="kw">rowData</span>(sce)[,</span>
<span id="cb1139-7"><a href="clusterMarkerGenesTop.html#cb1139-7"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1139-8"><a href="clusterMarkerGenesTop.html#cb1139-8"></a>                                           ,</span>
<span id="cb1139-9"><a href="clusterMarkerGenesTop.html#cb1139-9"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span>
<span id="cb1139-10"><a href="clusterMarkerGenesTop.html#cb1139-10"></a>markers.up &lt;-<span class="st"> </span>markersWiBatch.up</span></code></pre></div>
<div class="sourceCode" id="cb1140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1140-1"><a href="clusterMarkerGenesTop.html#cb1140-1"></a>interesting.up &lt;-<span class="st"> </span>markers.up[[chosen]]</span>
<span id="cb1140-2"><a href="clusterMarkerGenesTop.html#cb1140-2"></a>interesting.up[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.coluToShow]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 4 columns
##                      Symbol       Top         FDR summary.logFC
##                 &lt;character&gt; &lt;integer&gt;   &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000019582        CD74         1 5.39531e-19       2.53434
## ENSG00000198840      MT-ND3         1 6.24399e-32       2.89198
## ENSG00000198899     MT-ATP6         1 3.93950e-18       2.11267
## ENSG00000251562      MALAT1         1 2.06433e-24       2.46650
## ENSG00000198763      MT-ND2         2 6.00337e-24       2.50605
## ENSG00000198938      MT-CO3         2 8.60452e-14       2.06609
## ENSG00000198712      MT-CO2         3 3.28988e-19       1.99352
## ENSG00000198886      MT-ND4         4 2.10326e-14       1.84384
## ENSG00000198727      MT-CYB         5 7.63237e-11       1.80483
## ENSG00000204287     HLA-DRA         5 4.53827e-11       2.14766</code></pre>
<p>The t-test also allows us to specify a non-zero log-fold change as the null hypothesis. This allows us to consider the magnitude of the log-fold change in our p-value calculations, in a manner that is more rigorous than simply filtering directly on the log-fold changes (McCarthy and Smyth 2009). (Specifically, a simple threshold does not consider the variance and can enrich for genes that have both large log-fold changes and large variances.) We perform this by setting lfc= in our findMarkers() call - when combined with direction=, this tests for genes with log-fold changes that are significantly greater than 1:</p>
<div class="sourceCode" id="cb1142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1142-1"><a href="clusterMarkerGenesTop.html#cb1142-1"></a>markersWoBatch.up2 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>, <span class="dt">lfc=</span><span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1143-1"><a href="clusterMarkerGenesTop.html#cb1143-1"></a>markersWiBatch.up2 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1143-2"><a href="clusterMarkerGenesTop.html#cb1143-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1143-3"><a href="clusterMarkerGenesTop.html#cb1143-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1143-4"><a href="clusterMarkerGenesTop.html#cb1143-4"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1143-5"><a href="clusterMarkerGenesTop.html#cb1143-5"></a>                     <span class="co">#lfc=1,</span></span>
<span id="cb1143-6"><a href="clusterMarkerGenesTop.html#cb1143-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1143-7"><a href="clusterMarkerGenesTop.html#cb1143-7"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1143-8"><a href="clusterMarkerGenesTop.html#cb1143-8"></a>                                           ,</span>
<span id="cb1143-9"><a href="clusterMarkerGenesTop.html#cb1143-9"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span>
<span id="cb1143-10"><a href="clusterMarkerGenesTop.html#cb1143-10"></a>markers.up2 &lt;-<span class="st"> </span>markersWiBatch.up2</span></code></pre></div>
<div class="sourceCode" id="cb1144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1144-1"><a href="clusterMarkerGenesTop.html#cb1144-1"></a>interesting.up2 &lt;-<span class="st"> </span>markers.up2[[chosen]]</span>
<span id="cb1144-2"><a href="clusterMarkerGenesTop.html#cb1144-2"></a>interesting.up2[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.coluToShow]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 4 columns
##                      Symbol       Top         FDR summary.logFC
##                 &lt;character&gt; &lt;integer&gt;   &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000019582        CD74         1 5.39531e-19       2.53434
## ENSG00000198840      MT-ND3         1 6.24399e-32       2.89198
## ENSG00000198899     MT-ATP6         1 3.93950e-18       2.11267
## ENSG00000251562      MALAT1         1 2.06433e-24       2.46650
## ENSG00000198763      MT-ND2         2 6.00337e-24       2.50605
## ENSG00000198938      MT-CO3         2 8.60452e-14       2.06609
## ENSG00000198712      MT-CO2         3 3.28988e-19       1.99352
## ENSG00000198886      MT-ND4         4 2.10326e-14       1.84384
## ENSG00000198727      MT-CYB         5 7.63237e-11       1.80483
## ENSG00000204287     HLA-DRA         5 4.53827e-11       2.14766</code></pre>
<p>These two settings yield a more focused set of candidate marker genes that are upregulated in cluster 9.</p>
<div class="sourceCode" id="cb1146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1146-1"><a href="clusterMarkerGenesTop.html#cb1146-1"></a>best.set &lt;-<span class="st"> </span>interesting.up2[interesting.up2<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">5</span>,]</span>
<span id="cb1146-2"><a href="clusterMarkerGenesTop.html#cb1146-2"></a>logFCs &lt;-<span class="st"> </span><span class="kw">getMarkerEffects</span>(best.set)</span>
<span id="cb1146-3"><a href="clusterMarkerGenesTop.html#cb1146-3"></a>logFCs.ens &lt;-<span class="st"> </span><span class="kw">rownames</span>(logFCs)</span>
<span id="cb1146-4"><a href="clusterMarkerGenesTop.html#cb1146-4"></a><span class="kw">rownames</span>(logFCs) &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(logFCs), <span class="st">&quot;Symbol&quot;</span>]</span>
<span id="cb1146-5"><a href="clusterMarkerGenesTop.html#cb1146-5"></a></span>
<span id="cb1146-6"><a href="clusterMarkerGenesTop.html#cb1146-6"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1146-7"><a href="clusterMarkerGenesTop.html#cb1146-7"></a><span class="kw">pheatmap</span>(logFCs, <span class="dt">breaks=</span><span class="kw">seq</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="dt">length.out=</span><span class="dv">101</span>))</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<p>Of course, this increased stringency is not without cost. If only upregulated genes are requested from findMarkers(), any cluster defined by downregulation of a marker gene will not contain that gene among the top set of features in its DataFrame. This is occasionally relevant for subtypes or other states that are distinguished by high versus low expression of particular genes. Similarly, setting an excessively high log-fold change threshold may discard otherwise useful genes. For example, a gene upregulated in a small proportion of cells of a cluster will have a small log-fold change but can still be an effective marker if the focus is on specificity rather than sensitivity.</p>
</div>
<div id="finding-cluster-specific-markers" class="section level2">
<h2><span class="header-section-number">15.5</span> Finding cluster-specific markers</h2>
<p>By default, findMarkers() will give a high ranking to genes that are differentially expressed in any pairwise comparison. This is because a gene only needs a very low p
-value in a single pairwise comparison to achieve a low Top value. A more stringent approach would only consider genes that are differentially expressed in all pairwise comparisons involving the cluster of interest. To achieve this, we set pval.type=“all” in findMarkers() to use an intersection-union test (Berger and Hsu 1996) where the combined p-value for each gene is the maximum of the p-values from all pairwise comparisons. A gene will only achieve a low combined p-value if it is strongly DE in all comparisons to other clusters.</p>
<div class="sourceCode" id="cb1147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1147-1"><a href="clusterMarkerGenesTop.html#cb1147-1"></a><span class="co"># We can combine this with &#39;direction=&#39;.</span></span>
<span id="cb1147-2"><a href="clusterMarkerGenesTop.html#cb1147-2"></a>markersWoBatch.up3 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1147-3"><a href="clusterMarkerGenesTop.html#cb1147-3"></a>                  <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1147-4"><a href="clusterMarkerGenesTop.html#cb1147-4"></a>                  <span class="dt">pval.type=</span><span class="st">&quot;all&quot;</span>,</span>
<span id="cb1147-5"><a href="clusterMarkerGenesTop.html#cb1147-5"></a>                  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1148-1"><a href="clusterMarkerGenesTop.html#cb1148-1"></a>markersWiBatch.up3 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1148-2"><a href="clusterMarkerGenesTop.html#cb1148-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1148-3"><a href="clusterMarkerGenesTop.html#cb1148-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1148-4"><a href="clusterMarkerGenesTop.html#cb1148-4"></a>                     <span class="dt">pval.type=</span><span class="st">&quot;all&quot;</span>,</span>
<span id="cb1148-5"><a href="clusterMarkerGenesTop.html#cb1148-5"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1148-6"><a href="clusterMarkerGenesTop.html#cb1148-6"></a>                     <span class="co">#lfc=1,</span></span>
<span id="cb1148-7"><a href="clusterMarkerGenesTop.html#cb1148-7"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1148-8"><a href="clusterMarkerGenesTop.html#cb1148-8"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1148-9"><a href="clusterMarkerGenesTop.html#cb1148-9"></a>                                           ,</span>
<span id="cb1148-10"><a href="clusterMarkerGenesTop.html#cb1148-10"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span>
<span id="cb1148-11"><a href="clusterMarkerGenesTop.html#cb1148-11"></a>markers.up3 &lt;-<span class="st"> </span>markersWiBatch.up3</span></code></pre></div>
<div class="sourceCode" id="cb1149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1149-1"><a href="clusterMarkerGenesTop.html#cb1149-1"></a>interesting.up3 &lt;-<span class="st"> </span>markers.up3[[chosen]]</span>
<span id="cb1149-2"><a href="clusterMarkerGenesTop.html#cb1149-2"></a>interesting.colu &lt;-<span class="st"> </span><span class="kw">intersect</span>(interesting.coluToShow, <span class="kw">colnames</span>(interesting.up3))</span>
<span id="cb1149-3"><a href="clusterMarkerGenesTop.html#cb1149-3"></a>interesting.up3[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.colu]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 3 columns
##                      Symbol        FDR summary.logFC
##                 &lt;character&gt;  &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000198899     MT-ATP6 0.00160704      1.032391
## ENSG00000198840      MT-ND3 0.00160704      2.293836
## ENSG00000198763      MT-ND2 0.00160704      1.081879
## ENSG00000251562      MALAT1 0.00394722      0.580444
## ENSG00000198712      MT-CO2 0.00536471      1.625607
## ENSG00000198938      MT-CO3 0.01466755      1.339647
## ENSG00000198886      MT-ND4 0.03277462      1.396687
## ENSG00000198727      MT-CYB 1.00000000      0.950516
## ENSG00000244754     N4BP2L2 1.00000000      0.513462
## ENSG00000198786      MT-ND5 1.00000000      0.521689</code></pre>
<p>This strategy will only report genes that are highly specific to the cluster of interest. When it works, it can be highly effective as it generates a small focused set of candidate markers. However, any gene that is expressed at the same level in two or more clusters will simply not be detected. This is likely to discard many interesting genes, especially if the clusters are finely resolved with weak separation. To give a concrete example, consider a mixed population of CD4+-only, CD8+-only, double-positive and double-negative T cells. With pval.type=“all”, neither Cd4 or Cd8 would be detected as subpopulation-specific markers because each gene is expressed in two subpopulations. In comparison, pval.type=“any” will detect both of these genes as they will be DE between at least one pair of subpopulations.</p>
<p>If pval.type=“all” is too stringent yet pval.type=“any” is too generous, a compromise is to set pval.type=“some”. For each gene, we apply the Holm-Bonferroni correction across its p
-values and take the middle-most value as the combined p-value. This effectively tests the global null hypothesis that at least 50% of the individual pairwise comparisons exhibit no DE. We then rank the genes by their combined p-values to obtain an ordered set of marker candidates. The aim is to improve the conciseness of the top markers for defining a cluster while mitigating the risk of discarding useful genes that are not DE to all other clusters. The downside is that taking this compromise position sacrifices the theoretical guarantees offered at the other two extremes.</p>
<div class="sourceCode" id="cb1151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1151-1"><a href="clusterMarkerGenesTop.html#cb1151-1"></a>markersWoBatch.up4 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1151-2"><a href="clusterMarkerGenesTop.html#cb1151-2"></a>                  <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1151-3"><a href="clusterMarkerGenesTop.html#cb1151-3"></a>                  <span class="dt">pval.type=</span><span class="st">&quot;some&quot;</span>,</span>
<span id="cb1151-4"><a href="clusterMarkerGenesTop.html#cb1151-4"></a>                  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1152-1"><a href="clusterMarkerGenesTop.html#cb1152-1"></a>markersWiBatch.up4 &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1152-2"><a href="clusterMarkerGenesTop.html#cb1152-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1152-3"><a href="clusterMarkerGenesTop.html#cb1152-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1152-4"><a href="clusterMarkerGenesTop.html#cb1152-4"></a>                     <span class="dt">pval.type=</span><span class="st">&quot;some&quot;</span>,</span>
<span id="cb1152-5"><a href="clusterMarkerGenesTop.html#cb1152-5"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1152-6"><a href="clusterMarkerGenesTop.html#cb1152-6"></a>                     <span class="co">#lfc=1,</span></span>
<span id="cb1152-7"><a href="clusterMarkerGenesTop.html#cb1152-7"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1152-8"><a href="clusterMarkerGenesTop.html#cb1152-8"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1152-9"><a href="clusterMarkerGenesTop.html#cb1152-9"></a>                                           ,</span>
<span id="cb1152-10"><a href="clusterMarkerGenesTop.html#cb1152-10"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb1153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1153-1"><a href="clusterMarkerGenesTop.html#cb1153-1"></a>markers.up4 &lt;-<span class="st"> </span>markersWiBatch.up4</span>
<span id="cb1153-2"><a href="clusterMarkerGenesTop.html#cb1153-2"></a>interesting.up4 &lt;-<span class="st"> </span>markers.up4[[chosen]]</span>
<span id="cb1153-3"><a href="clusterMarkerGenesTop.html#cb1153-3"></a>interesting.colu &lt;-<span class="st"> </span><span class="kw">intersect</span>(interesting.coluToShow, <span class="kw">colnames</span>(interesting.up4))</span>
<span id="cb1153-4"><a href="clusterMarkerGenesTop.html#cb1153-4"></a>interesting.up4[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.colu]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 3 columns
##                      Symbol         FDR summary.logFC
##                 &lt;character&gt;   &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000198840      MT-ND3 6.08619e-13      1.937097
## ENSG00000251562      MALAT1 5.27427e-11      1.261153
## ENSG00000198763      MT-ND2 8.27824e-11      1.660871
## ENSG00000198712      MT-CO2 1.26691e-09      1.299312
## ENSG00000198899     MT-ATP6 6.20982e-09      1.482559
## ENSG00000198938      MT-CO3 5.95867e-08      1.496933
## ENSG00000198886      MT-ND4 4.48305e-05      1.120305
## ENSG00000198727      MT-CYB 9.89275e-03      0.917140
## ENSG00000198888      MT-ND1 1.00711e-01      1.084596
## ENSG00000100099        HPS4 1.00711e-01      0.684222</code></pre>
<p>In both cases, a different method is used to compute the summary effect size compared to pval.type=“any”. For pval.type=“all”, the summary log-fold change is defined as that corresponding to the pairwise comparison with the largest p-value, while for pval.type=“some”, it is defined as the log-fold change for the comparison with the middle-most p-value. This reflects the calculation of the combined p-value and avoids focusing on genes with strong changes in only one comparison.</p>
</div>
<div id="using-the-wilcoxon-rank-sum-test" class="section level2">
<h2><span class="header-section-number">15.6</span> Using the Wilcoxon rank sum test</h2>
<p>The Wilcoxon rank sum test (also known as the Wilcoxon-Mann-Whitney test, or WMW test) is another widely used method for pairwise comparisons between groups of observations. Its strength lies in the fact that it directly assesses separation between the expression distributions of different clusters. The WMW test statistic is proportional to the area-under-the-curve (AUC), i.e., the concordance probability, which is the probability of a random cell from one cluster having higher expression than a random cell from another cluster. In a pairwise comparison, AUCs of 1 or 0 indicate that the two clusters have perfectly separated expression distributions. Thus, the WMW test directly addresses the most desirable property of a candidate marker gene, while the t-test only does so indirectly via the difference in the means and the intra-group variance.</p>
<p>We perform WMW tests by again using the findMarkers() function, this time with test=“wilcox”. This returns a list of DataFrames containing ranked candidate markers for each cluster. The direction=, lfc= and pval.type= arguments can be specified and have the same interpretation as described for t-tests. We demonstrate below by detecting upregulated genes in each cluster with direction=“up”.</p>
<div class="sourceCode" id="cb1155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1155-1"><a href="clusterMarkerGenesTop.html#cb1155-1"></a>markers.wmw &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">test=</span><span class="st">&quot;wilcox&quot;</span>, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1156-1"><a href="clusterMarkerGenesTop.html#cb1156-1"></a>markersWiBatch.wmw &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1156-2"><a href="clusterMarkerGenesTop.html#cb1156-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1156-3"><a href="clusterMarkerGenesTop.html#cb1156-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1156-4"><a href="clusterMarkerGenesTop.html#cb1156-4"></a>                     <span class="dt">test=</span><span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb1156-5"><a href="clusterMarkerGenesTop.html#cb1156-5"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1156-6"><a href="clusterMarkerGenesTop.html#cb1156-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1156-7"><a href="clusterMarkerGenesTop.html#cb1156-7"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1156-8"><a href="clusterMarkerGenesTop.html#cb1156-8"></a>                                           ,</span>
<span id="cb1156-9"><a href="clusterMarkerGenesTop.html#cb1156-9"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb1157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1157-1"><a href="clusterMarkerGenesTop.html#cb1157-1"></a>markers.wmw &lt;-<span class="st"> </span>markersWiBatch.wmw</span>
<span id="cb1157-2"><a href="clusterMarkerGenesTop.html#cb1157-2"></a><span class="co">#print(names(markers.wmw))</span></span></code></pre></div>
<p>To explore the results in more detail, we focus on the DataFrame for cluster 9. The interpretation of Top is the same as described for t-tests, and Simes’ method is again used to combine p-values across pairwise comparisons. If we want more focused sets, we can also change pval.type= as previously described.</p>
<div class="sourceCode" id="cb1158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1158-1"><a href="clusterMarkerGenesTop.html#cb1158-1"></a>interesting.coluToShow &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;FDR&quot;</span>, <span class="st">&quot;summary.AUC&quot;</span>)</span>
<span id="cb1158-2"><a href="clusterMarkerGenesTop.html#cb1158-2"></a></span>
<span id="cb1158-3"><a href="clusterMarkerGenesTop.html#cb1158-3"></a>interesting.wmw &lt;-<span class="st"> </span>markers.wmw[[chosen]]</span>
<span id="cb1158-4"><a href="clusterMarkerGenesTop.html#cb1158-4"></a>interesting.wmw[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,interesting.coluToShow]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 4 columns
##                      Symbol       Top         FDR summary.AUC
##                 &lt;character&gt; &lt;integer&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## ENSG00000019582        CD74         1 5.05721e-24    0.912979
## ENSG00000198840      MT-ND3         1 7.45593e-23    0.959038
## ENSG00000251562      MALAT1         1 5.05721e-24    0.971495
## ENSG00000196628        TCF4         2 4.90193e-23    0.680557
## ENSG00000198712      MT-CO2         2 1.43378e-17    0.905674
## ENSG00000198763      MT-ND2         2 8.03850e-20    0.926217
## ENSG00000198938      MT-CO3         2 3.72034e-16    0.887331
## ENSG00000128218      VPREB3         3 5.30296e-23    0.686888
## ENSG00000105369       CD79A         4 1.09836e-21    0.734337
## ENSG00000198899     MT-ATP6         4 1.20658e-15    0.881893</code></pre>
<p>The DataFrame contains the AUCs from comparing cluster 9 to every other cluster. A value greater than 0.5 indicates that the gene is upregulated in the current cluster compared to the other cluster, while values less than 0.5 correspond to downregulation. We would typically expect AUCs of 0.7-0.8 for a strongly upregulated candidate marker.</p>
<div class="sourceCode" id="cb1160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1160-1"><a href="clusterMarkerGenesTop.html#cb1160-1"></a>best.set &lt;-<span class="st"> </span>interesting.wmw[interesting.wmw<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">5</span>,]</span>
<span id="cb1160-2"><a href="clusterMarkerGenesTop.html#cb1160-2"></a>AUCs &lt;-<span class="st"> </span><span class="kw">getMarkerEffects</span>(best.set, <span class="dt">prefix=</span><span class="st">&quot;AUC&quot;</span>)</span>
<span id="cb1160-3"><a href="clusterMarkerGenesTop.html#cb1160-3"></a>AUCs.ens &lt;-<span class="st"> </span><span class="kw">rownames</span>(AUCs)</span>
<span id="cb1160-4"><a href="clusterMarkerGenesTop.html#cb1160-4"></a><span class="kw">rownames</span>(AUCs) &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(AUCs), <span class="st">&quot;Symbol&quot;</span>]</span>
<span id="cb1160-5"><a href="clusterMarkerGenesTop.html#cb1160-5"></a></span>
<span id="cb1160-6"><a href="clusterMarkerGenesTop.html#cb1160-6"></a></span>
<span id="cb1160-7"><a href="clusterMarkerGenesTop.html#cb1160-7"></a><span class="kw">library</span>(pheatmap)</span>
<span id="cb1160-8"><a href="clusterMarkerGenesTop.html#cb1160-8"></a><span class="kw">pheatmap</span>(AUCs,</span>
<span id="cb1160-9"><a href="clusterMarkerGenesTop.html#cb1160-9"></a>     <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length.out=</span><span class="dv">21</span>),</span>
<span id="cb1160-10"><a href="clusterMarkerGenesTop.html#cb1160-10"></a>     <span class="dt">color =</span> viridis<span class="op">::</span><span class="kw">viridis</span>(<span class="dv">21</span>))</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<p>One practical advantage of the WMW test over the Welch t-test is that it is symmetric with respect to differences in the size of the groups being compared. This means that, all else being equal, the top-ranked genes on each side of a DE comparison will have similar expression profiles regardless of the number of cells in each group. In contrast, the t-test will favor genes where the larger group has the higher relative variance as this increases the estimated degrees of freedom and decreases the resulting p-value. This can lead to unappealing rankings when the aim is to identify genes upregulated in smaller groups. The WMW test is not completely immune to variance effects - for example, it will slightly favor detection of DEGs at low average abundance where the greater number of ties at zero deflates the approximate variance of the rank sum statistic - but this is relatively benign as the selected genes are still fairly interesting.</p>
<!-- We observe both of these effects in a comparison between alpha and gamma cells in the human pancreas data set from Lawlor et al. (2017) (Figure 11.4). -->
<div class="sourceCode" id="cb1161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1161-1"><a href="clusterMarkerGenesTop.html#cb1161-1"></a><span class="kw">print</span>(<span class="kw">head</span>(sce<span class="op">$</span>clusterStg))</span></code></pre></div>
<div class="sourceCode" id="cb1162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1162-1"><a href="clusterMarkerGenesTop.html#cb1162-1"></a>type1 &lt;-<span class="st"> &quot;c1&quot;</span></span>
<span id="cb1162-2"><a href="clusterMarkerGenesTop.html#cb1162-2"></a><span class="co">#type2 &lt;- &quot;c10&quot;</span></span>
<span id="cb1162-3"><a href="clusterMarkerGenesTop.html#cb1162-3"></a>type2 &lt;-<span class="st"> </span><span class="kw">levels</span>(sce<span class="op">$</span>clusterStg)[<span class="kw">nlevels</span>(sce<span class="op">$</span>clusterStg)]</span>
<span id="cb1162-4"><a href="clusterMarkerGenesTop.html#cb1162-4"></a>marker.t &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1162-5"><a href="clusterMarkerGenesTop.html#cb1162-5"></a>                        <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1162-6"><a href="clusterMarkerGenesTop.html#cb1162-6"></a>                        <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1162-7"><a href="clusterMarkerGenesTop.html#cb1162-7"></a>                        <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1162-8"><a href="clusterMarkerGenesTop.html#cb1162-8"></a>                        <span class="dt">restrict=</span><span class="kw">c</span>(type1, type2),</span>
<span id="cb1162-9"><a href="clusterMarkerGenesTop.html#cb1162-9"></a>                        <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1162-10"><a href="clusterMarkerGenesTop.html#cb1162-10"></a>                                           <span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;ensembl_gene_id&quot;</span>),</span>
<span id="cb1162-11"><a href="clusterMarkerGenesTop.html#cb1162-11"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span>
<span id="cb1162-12"><a href="clusterMarkerGenesTop.html#cb1162-12"></a>marker.w &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1162-13"><a href="clusterMarkerGenesTop.html#cb1162-13"></a>                        <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1162-14"><a href="clusterMarkerGenesTop.html#cb1162-14"></a>                        <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1162-15"><a href="clusterMarkerGenesTop.html#cb1162-15"></a>                        <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1162-16"><a href="clusterMarkerGenesTop.html#cb1162-16"></a>                        <span class="dt">restrict=</span><span class="kw">c</span>(type1, type2),</span>
<span id="cb1162-17"><a href="clusterMarkerGenesTop.html#cb1162-17"></a>                        <span class="dt">test.type=</span><span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb1162-18"><a href="clusterMarkerGenesTop.html#cb1162-18"></a>                        <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1162-19"><a href="clusterMarkerGenesTop.html#cb1162-19"></a>                                           <span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;ensembl_gene_id&quot;</span>),</span>
<span id="cb1162-20"><a href="clusterMarkerGenesTop.html#cb1162-20"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>]</span>
<span id="cb1162-21"><a href="clusterMarkerGenesTop.html#cb1162-21"></a>                        )</span></code></pre></div>
<div class="sourceCode" id="cb1163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1163-1"><a href="clusterMarkerGenesTop.html#cb1163-1"></a><span class="co"># Upregulated in type 1:</span></span>
<span id="cb1163-2"><a href="clusterMarkerGenesTop.html#cb1163-2"></a>marker.type1.t &lt;-<span class="st"> </span>marker.t[[type1]]</span>
<span id="cb1163-3"><a href="clusterMarkerGenesTop.html#cb1163-3"></a>marker.type1.w &lt;-<span class="st"> </span>marker.w[[type1]]</span>
<span id="cb1163-4"><a href="clusterMarkerGenesTop.html#cb1163-4"></a>chosen.type1.t &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.type1.t)[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]</span>
<span id="cb1163-5"><a href="clusterMarkerGenesTop.html#cb1163-5"></a>chosen.type1.w &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.type1.w)[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]</span>
<span id="cb1163-6"><a href="clusterMarkerGenesTop.html#cb1163-6"></a>u.type1.t &lt;-<span class="st"> </span><span class="kw">setdiff</span>(chosen.type1.t, chosen.type1.w)</span>
<span id="cb1163-7"><a href="clusterMarkerGenesTop.html#cb1163-7"></a>u.type1.w &lt;-<span class="st"> </span><span class="kw">setdiff</span>(chosen.type1.w, chosen.type1.t)</span>
<span id="cb1163-8"><a href="clusterMarkerGenesTop.html#cb1163-8"></a></span>
<span id="cb1163-9"><a href="clusterMarkerGenesTop.html#cb1163-9"></a><span class="co"># Upregulated in gamma:</span></span>
<span id="cb1163-10"><a href="clusterMarkerGenesTop.html#cb1163-10"></a>marker.type2.t &lt;-<span class="st"> </span>marker.t[[type2]]</span>
<span id="cb1163-11"><a href="clusterMarkerGenesTop.html#cb1163-11"></a>marker.type2.w &lt;-<span class="st"> </span>marker.w[[type2]]</span>
<span id="cb1163-12"><a href="clusterMarkerGenesTop.html#cb1163-12"></a>chosen.type2.t &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.type2.t)[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]</span>
<span id="cb1163-13"><a href="clusterMarkerGenesTop.html#cb1163-13"></a>chosen.type2.w &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.type2.w)[<span class="dv">1</span><span class="op">:</span><span class="dv">30</span>]</span>
<span id="cb1163-14"><a href="clusterMarkerGenesTop.html#cb1163-14"></a>u.type2.t &lt;-<span class="st"> </span><span class="kw">setdiff</span>(chosen.type2.t, chosen.type2.w)</span>
<span id="cb1163-15"><a href="clusterMarkerGenesTop.html#cb1163-15"></a>u.type2.w &lt;-<span class="st"> </span><span class="kw">setdiff</span>(chosen.type2.w, chosen.type2.t)</span>
<span id="cb1163-16"><a href="clusterMarkerGenesTop.html#cb1163-16"></a></span>
<span id="cb1163-17"><a href="clusterMarkerGenesTop.html#cb1163-17"></a>u.type1.t &lt;-<span class="st"> </span>u.type1.t[<span class="dv">1</span><span class="op">:</span><span class="kw">min</span>(<span class="dv">4</span>,<span class="kw">length</span>(u.type1.t))]</span>
<span id="cb1163-18"><a href="clusterMarkerGenesTop.html#cb1163-18"></a>u.type1.w &lt;-<span class="st"> </span>u.type1.w[<span class="dv">1</span><span class="op">:</span><span class="kw">min</span>(<span class="dv">4</span>,<span class="kw">length</span>(u.type1.w))]</span>
<span id="cb1163-19"><a href="clusterMarkerGenesTop.html#cb1163-19"></a>u.type2.t &lt;-<span class="st"> </span>u.type2.t[<span class="dv">1</span><span class="op">:</span><span class="kw">min</span>(<span class="dv">4</span>,<span class="kw">length</span>(u.type2.t))]</span>
<span id="cb1163-20"><a href="clusterMarkerGenesTop.html#cb1163-20"></a>u.type2.w &lt;-<span class="st"> </span>u.type2.w[<span class="dv">1</span><span class="op">:</span><span class="kw">min</span>(<span class="dv">4</span>,<span class="kw">length</span>(u.type2.w))]</span></code></pre></div>
<p>u.type1.t</p>
<div class="sourceCode" id="cb1164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1164-1"><a href="clusterMarkerGenesTop.html#cb1164-1"></a>marker.type1.t[,] <span class="op">%&gt;%</span></span>
<span id="cb1164-2"><a href="clusterMarkerGenesTop.html#cb1164-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1164-3"><a href="clusterMarkerGenesTop.html#cb1164-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>u.type1.t)</span></code></pre></div>
<pre><code>##                  Symbol ensembl_gene_id Top      p.value          FDR
## ENSG00000242574 HLA-DMB ENSG00000242574   3 1.591031e-43 8.819083e-40
## ENSG00000138795    LEF1 ENSG00000138795   4 7.601769e-42 3.160246e-38
## ENSG00000066923   STAG3 ENSG00000066923   5 6.797967e-41 2.260868e-37
## ENSG00000175727   MLXIP ENSG00000175727   6 4.038754e-39 1.119341e-35
##                 summary.logFC logFC.c10
## ENSG00000242574     0.3400197 0.3400197
## ENSG00000138795     0.4215269 0.4215269
## ENSG00000066923     0.3138304 0.3138304
## ENSG00000175727     0.5284546 0.5284546</code></pre>
<p>u.type1.w</p>
<div class="sourceCode" id="cb1166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1166-1"><a href="clusterMarkerGenesTop.html#cb1166-1"></a>marker.type1.w[,] <span class="op">%&gt;%</span></span>
<span id="cb1166-2"><a href="clusterMarkerGenesTop.html#cb1166-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1166-3"><a href="clusterMarkerGenesTop.html#cb1166-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>u.type1.w)</span></code></pre></div>
<pre><code>##                   Symbol ensembl_gene_id Top      p.value        FDR
## ENSG00000204287  HLA-DRA ENSG00000204287   1 3.657964e-06 0.06082829
## ENSG00000223865 HLA-DPB1 ENSG00000223865   2 2.745894e-05 0.13655164
## ENSG00000100721    TCL1A ENSG00000100721   3 3.854490e-05 0.13655164
## ENSG00000019582     CD74 ENSG00000019582   4 3.988134e-05 0.13655164
##                 summary.AUC   AUC.c10
## ENSG00000204287   0.9923039 0.9923039
## ENSG00000223865   0.9436445 0.9436445
## ENSG00000100721   0.9304866 0.9304866
## ENSG00000019582   0.9364449 0.9364449</code></pre>
<p>u.type2.t</p>
<div class="sourceCode" id="cb1168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1168-1"><a href="clusterMarkerGenesTop.html#cb1168-1"></a>marker.type2.t[,] <span class="op">%&gt;%</span></span>
<span id="cb1168-2"><a href="clusterMarkerGenesTop.html#cb1168-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1168-3"><a href="clusterMarkerGenesTop.html#cb1168-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>u.type2.t)</span></code></pre></div>
<pre><code>##                 Symbol ensembl_gene_id Top      p.value          FDR
## ENSG00000166562 SEC11C ENSG00000166562   1 6.096755e-12 8.385620e-08
## ENSG00000051523   CYBA ENSG00000051523   2 1.008554e-11 8.385620e-08
## ENSG00000180879   SSR4 ENSG00000180879   3 3.543074e-11 1.963926e-07
## ENSG00000172183  ISG20 ENSG00000172183   4 4.825016e-11 2.005880e-07
##                 summary.logFC logFC.c1
## ENSG00000166562      2.028202 2.028202
## ENSG00000051523      1.589030 1.589030
## ENSG00000180879      2.455082 2.455082
## ENSG00000172183      1.416760 1.416760</code></pre>
<p>u.type2.w</p>
<div class="sourceCode" id="cb1170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1170-1"><a href="clusterMarkerGenesTop.html#cb1170-1"></a>marker.type2.w[,] <span class="op">%&gt;%</span></span>
<span id="cb1170-2"><a href="clusterMarkerGenesTop.html#cb1170-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb1170-3"><a href="clusterMarkerGenesTop.html#cb1170-3"></a><span class="st">  </span><span class="kw">filter</span>(ensembl_gene_id <span class="op">%in%</span><span class="st"> </span>u.type2.w)</span></code></pre></div>
<pre><code>##                    Symbol ensembl_gene_id Top       p.value           FDR
## ENSG00000254709     IGLL5 ENSG00000254709   1 1.634885e-141 2.718650e-137
## ENSG00000138185    ENTPD1 ENSG00000138185   3 1.452621e-117 6.038908e-114
## ENSG00000240505 TNFRSF13B ENSG00000240505   4 1.452621e-117 6.038908e-114
## ENSG00000211890     IGHA2 ENSG00000211890   5  1.874610e-98  6.234577e-95
##                 summary.AUC    AUC.c1
## ENSG00000254709   0.9852284 0.9852284
## ENSG00000138185   0.8990814 0.8990814
## ENSG00000240505   0.8990814 0.8990814
## ENSG00000211890   0.8987090 0.8987090</code></pre>
<div class="sourceCode" id="cb1172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1172-1"><a href="clusterMarkerGenesTop.html#cb1172-1"></a><span class="co"># Examining all uniquely detected markers in each direction.</span></span>
<span id="cb1172-2"><a href="clusterMarkerGenesTop.html#cb1172-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb1172-3"><a href="clusterMarkerGenesTop.html#cb1172-3"></a>subset &lt;-<span class="st"> </span>sce[,sce<span class="op">$</span>clusterStg <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(type1, type2)]</span>
<span id="cb1172-4"><a href="clusterMarkerGenesTop.html#cb1172-4"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(</span>
<span id="cb1172-5"><a href="clusterMarkerGenesTop.html#cb1172-5"></a>    <span class="kw">plotExpression</span>(subset, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>, <span class="dt">features=</span>u.type1.t, <span class="dt">ncol=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb1172-6"><a href="clusterMarkerGenesTop.html#cb1172-6"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;Upregulated in %s, t-test-only&quot;</span>, type1)),</span>
<span id="cb1172-7"><a href="clusterMarkerGenesTop.html#cb1172-7"></a>    <span class="kw">plotExpression</span>(subset, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>, <span class="dt">features=</span>u.type1.w, <span class="dt">ncol=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb1172-8"><a href="clusterMarkerGenesTop.html#cb1172-8"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;Upregulated in %s, WMW-test-only&quot;</span>, type1)),</span>
<span id="cb1172-9"><a href="clusterMarkerGenesTop.html#cb1172-9"></a>    <span class="kw">plotExpression</span>(subset, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>, <span class="dt">features=</span>u.type2.t, <span class="dt">ncol=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb1172-10"><a href="clusterMarkerGenesTop.html#cb1172-10"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;Upregulated in %s, t-test-only&quot;</span>, type2)),</span>
<span id="cb1172-11"><a href="clusterMarkerGenesTop.html#cb1172-11"></a>    <span class="kw">plotExpression</span>(subset, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>, <span class="dt">features=</span>u.type2.w, <span class="dt">ncol=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb1172-12"><a href="clusterMarkerGenesTop.html#cb1172-12"></a><span class="st">        </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;Upregulated in %s, WMW-test-only&quot;</span>, type2)),</span>
<span id="cb1172-13"><a href="clusterMarkerGenesTop.html#cb1172-13"></a>    <span class="dt">ncol=</span><span class="dv">2</span></span>
<span id="cb1172-14"><a href="clusterMarkerGenesTop.html#cb1172-14"></a>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<div class="sourceCode" id="cb1173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1173-1"><a href="clusterMarkerGenesTop.html#cb1173-1"></a><span class="kw">rm</span>(u.type1.t, u.type1.w, u.type2.t, u.type2.w)</span></code></pre></div>
<p>The main disadvantage of the WMW test is that the AUCs are much slower to compute compared to t-statistics. This may be inconvenient for interactive analyses involving multiple iterations of marker detection. We can mitigate this to some extent by parallelizing these calculations using the BPPARAM= argument in findMarkers().</p>
</div>
<div id="using-a-binomial-test" class="section level2">
<h2><span class="header-section-number">15.7</span> Using a binomial test</h2>
<p>The binomial test identifies genes that differ in the proportion of expressing cells between clusters. (For the purposes of this section, a cell is considered to express a gene simply if it has non-zero expression for that gene.) This represents a much more stringent definition of marker genes compared to the other methods, as differences in expression between clusters are effectively ignored if both distributions of expression values are not near zero. The premise is that genes are more likely to contribute to important biological decisions if they were active in one cluster and silent in another, compared to more subtle “tuning” effects from changing the expression of an active gene. From a practical perspective, a binary measure of presence/absence is easier to validate.</p>
<p>We perform pairwise binomial tests between clusters using the findMarkers() function with test=“binom”. This returns a list of DataFrames containing marker statistics for each cluster such as the Top rank and its p-value. Here, the effect size is reported as the log-fold change in this proportion between each pair of clusters. Large positive log-fold changes indicate that the gene is more frequently expressed in one cluster compared to the other. We focus on genes that are upregulated in each cluster compared to the others by setting direction=“up”.</p>
<div class="sourceCode" id="cb1174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1174-1"><a href="clusterMarkerGenesTop.html#cb1174-1"></a>markersWoBatch.binom &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce,</span>
<span id="cb1174-2"><a href="clusterMarkerGenesTop.html#cb1174-2"></a>                    <span class="dt">test=</span><span class="st">&quot;binom&quot;</span>,</span>
<span id="cb1174-3"><a href="clusterMarkerGenesTop.html#cb1174-3"></a>                    <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1174-4"><a href="clusterMarkerGenesTop.html#cb1174-4"></a>                    <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg)</span></code></pre></div>
<div class="sourceCode" id="cb1175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1175-1"><a href="clusterMarkerGenesTop.html#cb1175-1"></a>markersWiBatch.binom &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(<span class="dt">x=</span>sce,</span>
<span id="cb1175-2"><a href="clusterMarkerGenesTop.html#cb1175-2"></a>                     <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg,</span>
<span id="cb1175-3"><a href="clusterMarkerGenesTop.html#cb1175-3"></a>                     <span class="dt">block=</span>sce<span class="op">$</span>block,</span>
<span id="cb1175-4"><a href="clusterMarkerGenesTop.html#cb1175-4"></a>                     <span class="dt">test=</span><span class="st">&quot;binom&quot;</span>,</span>
<span id="cb1175-5"><a href="clusterMarkerGenesTop.html#cb1175-5"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb1175-6"><a href="clusterMarkerGenesTop.html#cb1175-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(sce)[,</span>
<span id="cb1175-7"><a href="clusterMarkerGenesTop.html#cb1175-7"></a>                                           <span class="co">#c(&quot;Symbol&quot;, &quot;ensembl_gene_id&quot;),</span></span>
<span id="cb1175-8"><a href="clusterMarkerGenesTop.html#cb1175-8"></a>                                           ,</span>
<span id="cb1175-9"><a href="clusterMarkerGenesTop.html#cb1175-9"></a>                                           <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb1176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1176-1"><a href="clusterMarkerGenesTop.html#cb1176-1"></a>markers.binom &lt;-<span class="st"> </span>markersWiBatch.binom</span>
<span id="cb1176-2"><a href="clusterMarkerGenesTop.html#cb1176-2"></a><span class="kw">print</span>(<span class="kw">names</span>(markers.binom))</span></code></pre></div>
<pre><code>##  [1] &quot;c1&quot;  &quot;c2&quot;  &quot;c3&quot;  &quot;c4&quot;  &quot;c5&quot;  &quot;c6&quot;  &quot;c7&quot;  &quot;c8&quot;  &quot;c9&quot;  &quot;c10&quot;</code></pre>
<div class="sourceCode" id="cb1178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1178-1"><a href="clusterMarkerGenesTop.html#cb1178-1"></a>interesting.binom &lt;-<span class="st"> </span>markers.binom[[chosen]]</span>
<span id="cb1178-2"><a href="clusterMarkerGenesTop.html#cb1178-2"></a><span class="kw">print</span>(<span class="kw">colnames</span>(interesting.binom))</span></code></pre></div>
<pre><code>##  [1] &quot;ensembl_gene_id&quot;    &quot;external_gene_name&quot; &quot;chromosome_name&quot;   
##  [4] &quot;start_position&quot;     &quot;end_position&quot;       &quot;strandNum&quot;         
##  [7] &quot;Symbol&quot;             &quot;Type&quot;               &quot;mean&quot;              
## [10] &quot;detected&quot;           &quot;gene_sparsity&quot;      &quot;Top&quot;               
## [13] &quot;p.value&quot;            &quot;FDR&quot;                &quot;summary.logFC&quot;     
## [16] &quot;logFC.c1&quot;           &quot;logFC.c2&quot;           &quot;logFC.c3&quot;          
## [19] &quot;logFC.c4&quot;           &quot;logFC.c5&quot;           &quot;logFC.c6&quot;          
## [22] &quot;logFC.c7&quot;           &quot;logFC.c8&quot;           &quot;logFC.c10&quot;</code></pre>
<p>The plot below confirms that the top genes exhibit strong differences in the proportion of expressing cells in cluster 9 compared to the others.</p>
<div class="sourceCode" id="cb1180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1180-1"><a href="clusterMarkerGenesTop.html#cb1180-1"></a>top.genes &lt;-<span class="st"> </span><span class="kw">head</span>(<span class="kw">rownames</span>(interesting.binom))</span>
<span id="cb1180-2"><a href="clusterMarkerGenesTop.html#cb1180-2"></a><span class="co">#plotExpression(sce, x=&quot;clusterStg&quot;, features=top.genes)</span></span>
<span id="cb1180-3"><a href="clusterMarkerGenesTop.html#cb1180-3"></a><span class="kw">plotExpression</span>(sce, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1180-4"><a href="clusterMarkerGenesTop.html#cb1180-4"></a>               <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1180-5"><a href="clusterMarkerGenesTop.html#cb1180-5"></a>               <span class="dt">features=</span>top.genes[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>] )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-47-1.png" width="672" /></p>
<div class="sourceCode" id="cb1181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1181-1"><a href="clusterMarkerGenesTop.html#cb1181-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1181-2"><a href="clusterMarkerGenesTop.html#cb1181-2"></a>               <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1181-3"><a href="clusterMarkerGenesTop.html#cb1181-3"></a>               <span class="dt">features=</span>top.genes[<span class="dv">1</span>] )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
<div class="sourceCode" id="cb1182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1182-1"><a href="clusterMarkerGenesTop.html#cb1182-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1182-2"><a href="clusterMarkerGenesTop.html#cb1182-2"></a>               <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1182-3"><a href="clusterMarkerGenesTop.html#cb1182-3"></a>               <span class="dt">features=</span>top.genes[<span class="dv">2</span>] )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-48-2.png" width="672" /></p>
<div class="sourceCode" id="cb1183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1183-1"><a href="clusterMarkerGenesTop.html#cb1183-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1183-2"><a href="clusterMarkerGenesTop.html#cb1183-2"></a>               <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1183-3"><a href="clusterMarkerGenesTop.html#cb1183-3"></a>               <span class="dt">features=</span>top.genes[<span class="dv">3</span>] )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-48-3.png" width="672" /></p>
<div class="sourceCode" id="cb1184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1184-1"><a href="clusterMarkerGenesTop.html#cb1184-1"></a><span class="kw">plotExpression</span>(sce, <span class="dt">x=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1184-2"><a href="clusterMarkerGenesTop.html#cb1184-2"></a>               <span class="dt">colour_by=</span><span class="st">&quot;clusterStg&quot;</span>,</span>
<span id="cb1184-3"><a href="clusterMarkerGenesTop.html#cb1184-3"></a>               <span class="dt">features=</span>top.genes[<span class="dv">4</span>] )</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-48-4.png" width="672" /></p>
</div>
<div id="combining-multiple-marker-statistics" class="section level2">
<h2><span class="header-section-number">15.8</span> Combining multiple marker statistics</h2>
<p>On occasion, we might want to combine marker statistics from several testing regimes into a single DataFrame. This allows us to easily inspect multiple statistics at once to verify that a particular gene is a strong candidate marker. For example, a large AUC from the WMW test indicates that the expression distributions are well-separated between clusters, while the log-fold change reported with the t-test provides a more interpretable measure of the magnitude of the change in expression. We use the multiMarkerStats() to merge the results of separate findMarkers() calls into one DataFrame per cluster, with statistics interleaved to facilitate a direct comparison between different test regimes.</p>
<div class="sourceCode" id="cb1185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1185-1"><a href="clusterMarkerGenesTop.html#cb1185-1"></a>combined &lt;-<span class="st"> </span><span class="kw">multiMarkerStats</span>(</span>
<span id="cb1185-2"><a href="clusterMarkerGenesTop.html#cb1185-2"></a>    <span class="dt">t=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>),</span>
<span id="cb1185-3"><a href="clusterMarkerGenesTop.html#cb1185-3"></a>    <span class="dt">wilcox=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">test=</span><span class="st">&quot;wilcox&quot;</span>, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>),</span>
<span id="cb1185-4"><a href="clusterMarkerGenesTop.html#cb1185-4"></a>    <span class="dt">binom=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">test=</span><span class="st">&quot;binom&quot;</span>, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb1185-5"><a href="clusterMarkerGenesTop.html#cb1185-5"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb1186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1186-1"><a href="clusterMarkerGenesTop.html#cb1186-1"></a>combined &lt;-<span class="st"> </span><span class="kw">multiMarkerStats</span>(</span>
<span id="cb1186-2"><a href="clusterMarkerGenesTop.html#cb1186-2"></a>    <span class="dt">t=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>, <span class="dt">block=</span>sce<span class="op">$</span>block),</span>
<span id="cb1186-3"><a href="clusterMarkerGenesTop.html#cb1186-3"></a>    <span class="dt">wilcox=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">test=</span><span class="st">&quot;wilcox&quot;</span>, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>, <span class="dt">block=</span>sce<span class="op">$</span>block),</span>
<span id="cb1186-4"><a href="clusterMarkerGenesTop.html#cb1186-4"></a>    <span class="dt">binom=</span><span class="kw">findMarkers</span>(sce, <span class="dt">groups=</span>sce<span class="op">$</span>clusterStg, <span class="dt">test=</span><span class="st">&quot;binom&quot;</span>, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>, <span class="dt">block=</span>sce<span class="op">$</span>block)</span>
<span id="cb1186-5"><a href="clusterMarkerGenesTop.html#cb1186-5"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb1187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1187-1"><a href="clusterMarkerGenesTop.html#cb1187-1"></a><span class="co"># Interleaved marker statistics from both tests for each cluster.</span></span>
<span id="cb1187-2"><a href="clusterMarkerGenesTop.html#cb1187-2"></a><span class="kw">print</span>(<span class="kw">colnames</span>(combined[[<span class="st">&quot;c1&quot;</span>]]))</span></code></pre></div>
<pre><code>##  [1] &quot;Top&quot;                 &quot;p.value&quot;             &quot;FDR&quot;                
##  [4] &quot;t.Top&quot;               &quot;wilcox.Top&quot;          &quot;binom.Top&quot;          
##  [7] &quot;t.p.value&quot;           &quot;wilcox.p.value&quot;      &quot;binom.p.value&quot;      
## [10] &quot;t.FDR&quot;               &quot;wilcox.FDR&quot;          &quot;binom.FDR&quot;          
## [13] &quot;t.summary.logFC&quot;     &quot;wilcox.summary.AUC&quot;  &quot;binom.summary.logFC&quot;
## [16] &quot;t.logFC.c2&quot;          &quot;wilcox.AUC.c2&quot;       &quot;binom.logFC.c2&quot;     
## [19] &quot;t.logFC.c3&quot;          &quot;wilcox.AUC.c3&quot;       &quot;binom.logFC.c3&quot;     
## [22] &quot;t.logFC.c4&quot;          &quot;wilcox.AUC.c4&quot;       &quot;binom.logFC.c4&quot;     
## [25] &quot;t.logFC.c5&quot;          &quot;wilcox.AUC.c5&quot;       &quot;binom.logFC.c5&quot;     
## [28] &quot;t.logFC.c6&quot;          &quot;wilcox.AUC.c6&quot;       &quot;binom.logFC.c6&quot;     
## [31] &quot;t.logFC.c7&quot;          &quot;wilcox.AUC.c7&quot;       &quot;binom.logFC.c7&quot;     
## [34] &quot;t.logFC.c8&quot;          &quot;wilcox.AUC.c8&quot;       &quot;binom.logFC.c8&quot;     
## [37] &quot;t.logFC.c9&quot;          &quot;wilcox.AUC.c9&quot;       &quot;binom.logFC.c9&quot;     
## [40] &quot;t.logFC.c10&quot;         &quot;wilcox.AUC.c10&quot;      &quot;binom.logFC.c10&quot;</code></pre>
<div class="sourceCode" id="cb1189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1189-1"><a href="clusterMarkerGenesTop.html#cb1189-1"></a><span class="co">#head(combined[[&quot;c1&quot;]][,1:9])</span></span>
<span id="cb1189-2"><a href="clusterMarkerGenesTop.html#cb1189-2"></a>combined[[<span class="st">&quot;c1&quot;</span>]]<span class="op">$</span>Symbol &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[<span class="kw">rownames</span>(combined[[<span class="st">&quot;c1&quot;</span>]]), <span class="st">&quot;Symbol&quot;</span>]</span>
<span id="cb1189-3"><a href="clusterMarkerGenesTop.html#cb1189-3"></a>tmpCol &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="kw">colnames</span>(combined[[<span class="st">&quot;c1&quot;</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">9</span>])</span>
<span id="cb1189-4"><a href="clusterMarkerGenesTop.html#cb1189-4"></a><span class="kw">head</span>(combined[[<span class="st">&quot;c1&quot;</span>]][,tmpCol])</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 10 columns
##                      Symbol       Top     p.value         FDR     t.Top
##                 &lt;character&gt; &lt;integer&gt;   &lt;numeric&gt;   &lt;numeric&gt; &lt;integer&gt;
## ENSG00000196549         MME         2 1.86762e-44 3.45074e-41         2
## ENSG00000198771       RCSD1         3 7.21345e-40 6.80436e-37         3
## ENSG00000124766        SOX4         4 2.37596e-38 1.88142e-35         1
## ENSG00000128218      VPREB3         4 1.67646e-55 1.39389e-51         1
## ENSG00000231389    HLA-DPA1         4 6.79103e-48 2.82320e-44         1
## ENSG00000081189       MEF2C         5 4.45585e-57 7.40963e-53         1
##                 wilcox.Top binom.Top    t.p.value wilcox.p.value binom.p.value
##                  &lt;integer&gt; &lt;integer&gt;    &lt;numeric&gt;      &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000196549          2         2 1.05815e-119    1.47856e-55   1.86762e-44
## ENSG00000198771          2         2  9.11728e-88    5.91283e-65   7.21345e-40
## ENSG00000124766          4         1  3.16898e-82    8.46994e-54   2.37596e-38
## ENSG00000128218          4         1 1.59077e-177    2.36837e-75   1.67646e-55
## ENSG00000231389          3         4 5.09978e-218    3.79418e-86   6.79103e-48
## ENSG00000081189          5         1 3.37642e-166    1.48401e-71   4.45585e-57</code></pre>
<div class="sourceCode" id="cb1191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1191-1"><a href="clusterMarkerGenesTop.html#cb1191-1"></a><span class="kw">rm</span>(sce)</span></code></pre></div>
<p>In addition, multiMarkerStats() will compute a number of new statistics by combining the per-regime statistics. The combined Top value is obtained by simply taking the largest Top value across all tests for a given gene, while the reported p.value is obtained by taking the largest p-value. Ranking on either metric focuses on genes with robust differences that are highly ranked and detected by each of the individual testing regimes. Of course, this might be considered an overly conservative approach in practice, so it is entirely permissible to re-rank the DataFrame according to the Top or p.value for an individual regime (effectively limiting the use of the other regimes’ statistics to diagnostics only).</p>
<p>Write list to file:</p>
<div class="sourceCode" id="cb1192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1192-1"><a href="clusterMarkerGenesTop.html#cb1192-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_clustMarkCombi.Rds&quot;</span>,</span>
<span id="cb1192-2"><a href="clusterMarkerGenesTop.html#cb1192-2"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1192-3"><a href="clusterMarkerGenesTop.html#cb1192-3"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1_clustMarkCombi.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1194-1"><a href="clusterMarkerGenesTop.html#cb1194-1"></a><span class="kw">saveRDS</span>(combined, <span class="dt">file=</span>tmpFn)</span>
<span id="cb1194-2"><a href="clusterMarkerGenesTop.html#cb1194-2"></a><span class="kw">rm</span>(combined,tmpFn)</span></code></pre></div>
</div>
<div id="invalidity-of-p-values" class="section level2">
<h2><span class="header-section-number">15.9</span> Invalidity of p-values</h2>
<div id="from-data-snooping" class="section level3">
<h3><span class="header-section-number">15.9.1</span> From data snooping</h3>
<p>All of our DE strategies for detecting marker genes between clusters are statistically flawed to some extent. The DE analysis is performed on the same data used to obtain the clusters, which represents “data dredging” (also known as fishing or data snooping). The hypothesis of interest - are there differences between clusters? - is formulated from the data, so we are more likely to get a positive result when we re-use the data set to test that hypothesis.</p>
<p>The practical effect of data dredging is best illustrated with a simple simulation. We simulate i.i.d. normal values, perform k-means clustering and test for DE between clusters of cells with findMarkers(). The resulting distribution of p-values is heavily skewed towards low values. Thus, we can detect “significant” differences between clusters even in the absence of any real substructure in the data. This effect arises from the fact that clustering, by definition, yields groups of cells that are separated in expression space. Testing for DE genes between clusters will inevitably yield some significant results as that is how the clusters were defined.</p>
<p>Distribution of <span class="math inline">\(p\)</span>-values from a DE analysis between two clusters in a simulation with no true subpopulation structure:</p>
<div class="sourceCode" id="cb1195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1195-1"><a href="clusterMarkerGenesTop.html#cb1195-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb1195-2"><a href="clusterMarkerGenesTop.html#cb1195-2"></a><span class="kw">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb1195-3"><a href="clusterMarkerGenesTop.html#cb1195-3"></a>y &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rnorm</span>(<span class="dv">100000</span>), <span class="dt">ncol=</span><span class="dv">200</span>)</span>
<span id="cb1195-4"><a href="clusterMarkerGenesTop.html#cb1195-4"></a>clusters &lt;-<span class="st"> </span><span class="kw">kmeans</span>(<span class="kw">t</span>(y), <span class="dt">centers=</span><span class="dv">2</span>)<span class="op">$</span>cluster</span>
<span id="cb1195-5"><a href="clusterMarkerGenesTop.html#cb1195-5"></a>out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(y, clusters)</span>
<span id="cb1195-6"><a href="clusterMarkerGenesTop.html#cb1195-6"></a><span class="kw">hist</span>(out[[<span class="dv">1</span>]]<span class="op">$</span>p.value, <span class="dt">col=</span><span class="st">&quot;grey80&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;p-value&quot;</span>)</span></code></pre></div>
<p><img src="clusterMarkerGenes_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
<p>For marker gene detection, this effect is largely harmless as the p-values are used only for ranking. However, it becomes an issue when the p-values are used to define “significant differences” between clusters with respect to an error rate threshold. Meaningful interpretation of error rates require consideration of the long-run behavior, i.e., the rate of incorrect rejections if the experiment were repeated many times. The concept of statistical significance for differences between clusters is not applicable if clusters and their interpretations are not stably reproducible across (hypothetical) replicate experiments.</p>
</div>
<div id="nature-of-replication" class="section level3">
<h3><span class="header-section-number">15.9.2</span> Nature of replication</h3>
<p>The naive application of DE analysis methods will treat counts from the same cluster of cells as replicate observations. This is not the most relevant level of replication when cells are derived from the same biological sample (i.e., cell culture, animal or patient). DE analyses that treat cells as replicates fail to properly model the sample-to-sample variability (Lun and Marioni 2017). The latter is arguably the more important level of replication as different samples will necessarily be generated if the experiment is to be replicated. Indeed, the use of cells as replicates only masks the fact that the sample size is actually one in an experiment involving a single biological sample. This reinforces the inappropriateness of using the marker gene p-values to perform statistical inference.</p>
<p>“We strongly recommend selecting some markers for use in validation studies with an independent replicate population of cells. A typical strategy is to identify a corresponding subset of cells that express the upregulated markers and do not express the downregulated markers. Ideally, a different technique for quantifying expression would also be used during validation, e.g., fluorescent in situ hybridisation or quantitative PCR. This confirms that the subpopulation genuinely exists and is not an artifact of the scRNA-seq protocol or the computational analysis.”</p>
<p>See the OSCA chapter on <a href="https://osca.bioconductor.org/clustering.html">Marker gene detection</a></p>
<p><strong>Challenge</strong> Identify markers for a different cluster and try to identify the cell type.</p>
</div>
</div>
<div id="session-information-11" class="section level2">
<h2><span class="header-section-number">15.10</span> Session information</h2>
<details>
<div class="sourceCode" id="cb1196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1196-1"><a href="clusterMarkerGenesTop.html#cb1196-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] glue_1.4.2                  Cairo_1.5-12.2             
##  [3] pheatmap_1.0.12             RColorBrewer_1.1-2         
##  [5] dplyr_1.0.6                 scran_1.18.7               
##  [7] scater_1.18.6               SingleCellExperiment_1.12.0
##  [9] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [11] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [13] IRanges_2.24.1              S4Vectors_0.28.1           
## [15] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [17] matrixStats_0.58.0          ggplot2_3.3.3              
## [19] knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7              tools_4.0.3              
##  [3] bslib_0.2.5               utf8_1.2.1               
##  [5] R6_2.5.0                  irlba_2.3.3              
##  [7] vipor_0.4.5               uwot_0.1.10              
##  [9] DBI_1.1.1                 colorspace_2.0-1         
## [11] withr_2.4.2               tidyselect_1.1.1         
## [13] gridExtra_2.3             compiler_4.0.3           
## [15] BiocNeighbors_1.8.2       DelayedArray_0.16.3      
## [17] labeling_0.4.2            bookdown_0.22            
## [19] sass_0.4.0                scales_1.1.1             
## [21] stringr_1.4.0             digest_0.6.27            
## [23] rmarkdown_2.8             XVector_0.30.0           
## [25] pkgconfig_2.0.3           htmltools_0.5.1.1        
## [27] sparseMatrixStats_1.2.1   highr_0.9                
## [29] limma_3.46.0              rlang_0.4.11             
## [31] FNN_1.1.3                 DelayedMatrixStats_1.12.3
## [33] farver_2.1.0              jquerylib_0.1.4          
## [35] generics_0.1.0            jsonlite_1.7.2           
## [37] BiocParallel_1.24.1       RCurl_1.98-1.3           
## [39] magrittr_2.0.1            BiocSingular_1.6.0       
## [41] GenomeInfoDbData_1.2.4    scuttle_1.0.4            
## [43] Matrix_1.3-3              Rcpp_1.0.6               
## [45] ggbeeswarm_0.6.0          munsell_0.5.0            
## [47] fansi_0.4.2               viridis_0.6.1            
## [49] lifecycle_1.0.0           stringi_1.6.1            
## [51] yaml_2.2.1                edgeR_3.32.1             
## [53] zlibbioc_1.36.0           grid_4.0.3               
## [55] dqrng_0.3.0               crayon_1.4.1             
## [57] lattice_0.20-44           cowplot_1.1.1            
## [59] beachmat_2.6.4            locfit_1.5-9.4           
## [61] pillar_1.6.1              igraph_1.2.6             
## [63] codetools_0.2-18          evaluate_0.14            
## [65] vctrs_0.3.8               gtable_0.3.0             
## [67] purrr_0.3.4               assertthat_0.2.1         
## [69] xfun_0.23                 rsvd_1.0.5               
## [71] RSpectra_0.16-0           viridisLite_0.4.0        
## [73] tibble_3.1.2              beeswarm_0.3.1           
## [75] bluster_1.0.0             statmod_1.4.36           
## [77] ellipsis_0.3.2</code></pre>
</details>

<!--
  setSuf: "_5hCellPerSpl"
  setSuf: "_5hCps"
  setSuf: "_allCells"
-->
<div class="sourceCode" id="cb1198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1198-1"><a href="clusterMarkerGenesTop.html#cb1198-1"></a>params2 &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb1198-2"><a href="clusterMarkerGenesTop.html#cb1198-2"></a>  <span class="st">&quot;projDir&quot;</span> =<span class="st"> &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020&quot;</span>,</span>
<span id="cb1198-3"><a href="clusterMarkerGenesTop.html#cb1198-3"></a>  <span class="st">&quot;dirRel&quot;</span> =<span class="st"> &quot;..&quot;</span>,</span>
<span id="cb1198-4"><a href="clusterMarkerGenesTop.html#cb1198-4"></a>  <span class="st">&quot;inpDirBit&quot;</span> =<span class="st"> &quot;AnaWiSce/AnaCourse1&quot;</span>,</span>
<span id="cb1198-5"><a href="clusterMarkerGenesTop.html#cb1198-5"></a>  <span class="st">&quot;outDirBit&quot;</span> =<span class="st"> &quot;AnaWiSce/AnaCourse1&quot;</span>,</span>
<span id="cb1198-6"><a href="clusterMarkerGenesTop.html#cb1198-6"></a>  <span class="st">&quot;bookType&quot;</span> =<span class="st"> &quot;km&quot;</span>,</span>
<span id="cb1198-7"><a href="clusterMarkerGenesTop.html#cb1198-7"></a>  <span class="st">&quot;cacheBool&quot;</span>  =<span class="st"> </span><span class="ot">FALSE</span>,  </span>
<span id="cb1198-8"><a href="clusterMarkerGenesTop.html#cb1198-8"></a>  <span class="st">&quot;setName&quot;</span> =<span class="st"> &quot;caron&quot;</span>,</span>
<span id="cb1198-9"><a href="clusterMarkerGenesTop.html#cb1198-9"></a>  <span class="st">&quot;splSetToGet&quot;</span> =<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span>,</span>
<span id="cb1198-10"><a href="clusterMarkerGenesTop.html#cb1198-10"></a>  <span class="st">&quot;setSuf&quot;</span> =<span class="st"> &quot;_5hCellPerSpl&quot;</span>,</span>
<span id="cb1198-11"><a href="clusterMarkerGenesTop.html#cb1198-11"></a>  <span class="st">&quot;dsiSuf&quot;</span> =<span class="st"> &#39;_dsi&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1199-1"><a href="clusterMarkerGenesTop.html#cb1199-1"></a><span class="cf">if</span>(<span class="kw">interactive</span>()) {</span>
<span id="cb1199-2"><a href="clusterMarkerGenesTop.html#cb1199-2"></a>  paramsToUse &lt;-<span class="st"> </span>params2</span>
<span id="cb1199-3"><a href="clusterMarkerGenesTop.html#cb1199-3"></a>} <span class="cf">else</span> {</span>
<span id="cb1199-4"><a href="clusterMarkerGenesTop.html#cb1199-4"></a>  paramsToUse &lt;-<span class="st"> </span>params</span>
<span id="cb1199-5"><a href="clusterMarkerGenesTop.html#cb1199-5"></a>}</span>
<span id="cb1199-6"><a href="clusterMarkerGenesTop.html#cb1199-6"></a></span>
<span id="cb1199-7"><a href="clusterMarkerGenesTop.html#cb1199-7"></a>projDir &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>projDir</span>
<span id="cb1199-8"><a href="clusterMarkerGenesTop.html#cb1199-8"></a>dirRel &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>dirRel</span>
<span id="cb1199-9"><a href="clusterMarkerGenesTop.html#cb1199-9"></a>outDirBit &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>outDirBit</span>
<span id="cb1199-10"><a href="clusterMarkerGenesTop.html#cb1199-10"></a>cacheBool &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>cacheBool</span>
<span id="cb1199-11"><a href="clusterMarkerGenesTop.html#cb1199-11"></a>splSetToGet &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>splSetToGet</span>
<span id="cb1199-12"><a href="clusterMarkerGenesTop.html#cb1199-12"></a>setName &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>setName</span>
<span id="cb1199-13"><a href="clusterMarkerGenesTop.html#cb1199-13"></a>setSuf &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>setSuf</span>
<span id="cb1199-14"><a href="clusterMarkerGenesTop.html#cb1199-14"></a>dsiSuf &lt;-<span class="st"> </span>paramsToUse<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb1199-15"><a href="clusterMarkerGenesTop.html#cb1199-15"></a></span>
<span id="cb1199-16"><a href="clusterMarkerGenesTop.html#cb1199-16"></a><span class="cf">if</span>(paramsToUse<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb1199-17"><a href="clusterMarkerGenesTop.html#cb1199-17"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb1199-18"><a href="clusterMarkerGenesTop.html#cb1199-18"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span></span>
<span id="cb1199-19"><a href="clusterMarkerGenesTop.html#cb1199-19"></a>    setSuf &lt;-<span class="st"> &quot;_5hCps&quot;</span></span>
<span id="cb1199-20"><a href="clusterMarkerGenesTop.html#cb1199-20"></a>}</span>
<span id="cb1199-21"><a href="clusterMarkerGenesTop.html#cb1199-21"></a></span>
<span id="cb1199-22"><a href="clusterMarkerGenesTop.html#cb1199-22"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb1199-23"><a href="clusterMarkerGenesTop.html#cb1199-23"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb1199-24"><a href="clusterMarkerGenesTop.html#cb1199-24"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb1199-25"><a href="clusterMarkerGenesTop.html#cb1199-25"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multiSplCompTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/clusterMarkerGenes.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC2.pdf", "KmWiC2.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
