<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop} | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop} | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop} | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dsi-5hCellPerSpl-allSetsTop.html"/>
<link rel="next" href="clusterMarkerGenesTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information"><i class="fa fa-check"></i><b>4.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>5</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#learning-objectives"><i class="fa fa-check"></i><b>5.1</b> Learning objectives</a></li>
<li class="chapter" data-level="5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>5.2</b> Scaling normalization</a><ul>
<li class="chapter" data-level="5.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>5.2.1</b> CPM</a></li>
<li class="chapter" data-level="5.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>5.2.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="5.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>5.2.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="5.2.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>5.2.4</b> Library size normalization</a></li>
<li class="chapter" data-level="5.2.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution"><i class="fa fa-check"></i><b>5.2.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="5.2.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>5.2.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="5.2.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>5.2.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="5.2.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>5.2.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="5.2.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object"><i class="fa fa-check"></i><b>5.2.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform"><i class="fa fa-check"></i><b>5.3</b> SCTransform</a><ul>
<li class="chapter" data-level="5.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>5.3.1</b> Inspect data</a></li>
<li class="chapter" data-level="5.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>5.3.2</b> Transformation</a></li>
<li class="chapter" data-level="5.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>5.3.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>5.4</b> Visualisation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>5.4.1</b> log raw counts</a></li>
<li class="chapter" data-level="5.4.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>5.4.2</b> log CPM</a></li>
<li class="chapter" data-level="5.4.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>5.4.3</b> scran</a></li>
<li class="chapter" data-level="5.4.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>5.4.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-1"><i class="fa fa-check"></i><b>5.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>6</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="6.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-1"><i class="fa fa-check"></i><b>6.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-1"><i class="fa fa-check"></i><b>6.2</b> Load packages</a></li>
<li class="chapter" data-level="6.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>6.3</b> Load data</a></li>
<li class="chapter" data-level="6.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>6.4</b> PCA</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>6.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>6.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="6.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>6.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>6.6</b> UMAP</a></li>
<li class="chapter" data-level="6.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-2"><i class="fa fa-check"></i><b>6.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>7</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>7.1</b> Load object</a></li>
<li class="chapter" data-level="7.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>7.3</b> Normalised counts</a></li>
<li class="chapter" data-level="7.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>8</b> Feature selection</a><ul>
<li class="chapter" data-level="8.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>8.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="8.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>8.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="8.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>8.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-4"><i class="fa fa-check"></i><b>8.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>9.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>9.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-5"><i class="fa fa-check"></i><b>9.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>10</b> Batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="10.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#learning-objectives-1"><i class="fa fa-check"></i><b>10.1</b> Learning objectives</a></li>
<li class="chapter" data-level="10.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>10.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="10.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>10.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="10.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>10.4</b> Batch correction</a><ul>
<li class="chapter" data-level="10.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>10.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-2"><i class="fa fa-check"></i><b>10.5</b> SCTransform</a></li>
<li class="chapter" data-level="10.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>10.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="10.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>10.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="10.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>10.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>10.7</b> fastMNN</a></li>
<li class="chapter" data-level="10.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>10.8</b> Harmony</a></li>
<li class="chapter" data-level="10.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-6"><i class="fa fa-check"></i><b>10.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>11</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="11.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#learning-objectives-2"><i class="fa fa-check"></i><b>11.1</b> Learning objectives</a></li>
<li class="chapter" data-level="11.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#abbreviations-1"><i class="fa fa-check"></i><b>11.2</b> Abbreviations</a></li>
<li class="chapter" data-level="11.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#motivation"><i class="fa fa-check"></i><b>11.3</b> Motivation</a></li>
<li class="chapter" data-level="11.4" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#load-the-data"><i class="fa fa-check"></i><b>11.4</b> Load the data</a></li>
<li class="chapter" data-level="11.5" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>11.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="11.6" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#linear-regression"><i class="fa fa-check"></i><b>11.6</b> Linear regression</a></li>
<li class="chapter" data-level="11.7" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>11.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="11.7.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#algorithm-overview"><i class="fa fa-check"></i><b>11.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="11.7.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#application-to-the-data"><i class="fa fa-check"></i><b>11.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>11.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="11.8.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>11.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="11.8.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>11.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="11.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>11.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="11.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#coassignment-probabilities"><i class="fa fa-check"></i><b>11.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="11.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#rand-index"><i class="fa fa-check"></i><b>11.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>11.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="11.10" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>11.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="11.11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#session-information-7"><i class="fa fa-check"></i><b>11.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>12</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="12.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#learning-objectives-3"><i class="fa fa-check"></i><b>12.1</b> Learning objectives</a></li>
<li class="chapter" data-level="12.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>12.2</b> Abbreviations</a></li>
<li class="chapter" data-level="12.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>12.3</b> Motivation</a></li>
<li class="chapter" data-level="12.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data-1"><i class="fa fa-check"></i><b>12.4</b> Load the data</a></li>
<li class="chapter" data-level="12.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>12.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="12.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>12.6</b> Linear regression</a></li>
<li class="chapter" data-level="12.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>12.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="12.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>12.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="12.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>12.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>12.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="12.8.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>12.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="12.8.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>12.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="12.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>12.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="12.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>12.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="12.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-1"><i class="fa fa-check"></i><b>12.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>12.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="12.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>12.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="12.11" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-8"><i class="fa fa-check"></i><b>12.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>13</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="13.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#learning-objectives-4"><i class="fa fa-check"></i><b>13.1</b> Learning objectives</a></li>
<li class="chapter" data-level="13.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-2"><i class="fa fa-check"></i><b>13.2</b> Motivation</a></li>
<li class="chapter" data-level="13.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-2"><i class="fa fa-check"></i><b>13.3</b> Load the data</a></li>
<li class="chapter" data-level="13.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>13.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="13.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-2"><i class="fa fa-check"></i><b>13.5</b> Linear regression</a></li>
<li class="chapter" data-level="13.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>13.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="13.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>13.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="13.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>13.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>13.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="13.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>13.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>13.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>13.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="13.7.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-2"><i class="fa fa-check"></i><b>13.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="13.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>13.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="13.7.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>13.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>13.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="13.8.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>13.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.8.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>13.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.8.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-3"><i class="fa fa-check"></i><b>13.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="13.8.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>13.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>13.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="13.10" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-9"><i class="fa fa-check"></i><b>13.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>14</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="14.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-3"><i class="fa fa-check"></i><b>14.1</b> Load data</a></li>
<li class="chapter" data-level="14.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>14.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="14.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>14.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="14.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>14.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="14.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>14.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="14.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>14.2.2</b> k-means</a><ul>
<li class="chapter" data-level="14.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>14.2.2.1</b> Description</a></li>
<li class="chapter" data-level="14.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-2"><i class="fa fa-check"></i><b>14.2.2.2</b> Example</a></li>
<li class="chapter" data-level="14.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>14.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="14.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>14.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="14.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>14.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="14.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>14.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="14.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>14.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="14.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>14.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="14.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>14.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="14.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>14.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="14.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>14.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="14.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>14.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="14.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>14.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>14.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="14.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>14.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="14.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>14.5</b> Save data</a></li>
<li class="chapter" data-level="14.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-10"><i class="fa fa-check"></i><b>14.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>15</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="15.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives-5"><i class="fa fa-check"></i><b>15.1</b> Learning objectives</a></li>
<li class="chapter" data-level="15.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-4"><i class="fa fa-check"></i><b>15.2</b> Load data</a></li>
<li class="chapter" data-level="15.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>15.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="15.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>15.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="15.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>15.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>15.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="15.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>15.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="15.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>15.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="15.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>15.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="15.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>15.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="15.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>15.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="15.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>15.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="15.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>15.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html"><i class="fa fa-check"></i><b>16</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="16.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#motivation-3"><i class="fa fa-check"></i><b>16.1</b> Motivation</a></li>
<li class="chapter" data-level="16.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#setting-up-the-data"><i class="fa fa-check"></i><b>16.2</b> Setting up the data</a></li>
<li class="chapter" data-level="16.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>16.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="16.3.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>16.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="16.3.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>16.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="16.3.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#introduction-3"><i class="fa fa-check"></i><b>16.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="16.3.2.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#pre-processing"><i class="fa fa-check"></i><b>16.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="16.3.2.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#statistical-modelling"><i class="fa fa-check"></i><b>16.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="16.3.2.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>16.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="16.3.2.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>16.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="16.3.2.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#list-of-degs"><i class="fa fa-check"></i><b>16.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="16.3.2.4.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>16.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.3.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#putting-it-all-together"><i class="fa fa-check"></i><b>16.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>16.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="16.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#overview"><i class="fa fa-check"></i><b>16.4.1</b> Overview</a></li>
<li class="chapter" data-level="16.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#handling-composition-effects"><i class="fa fa-check"></i><b>16.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="16.4.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#background"><i class="fa fa-check"></i><b>16.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#session-information-12"><i class="fa fa-check"></i><b>16.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>17</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="17.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>17.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="17.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>17.0.2</b> Data integration</a></li>
<li class="chapter" data-level="17.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>17.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="17.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>17.0.4</b> Clustering</a></li>
<li class="chapter" data-level="17.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>17.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="17.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>17.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="17.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>17.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>17.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="17.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>17.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="17.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>17.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="17.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>17.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>17.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="17.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>17.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="17.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-13"><i class="fa fa-check"></i><b>17.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>18</b> Annex</a></li>
<li class="chapter" data-level="19" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>19</b> Gene annotation</a><ul>
<li class="chapter" data-level="19.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-4"><i class="fa fa-check"></i><b>19.1</b> Introduction</a></li>
<li class="chapter" data-level="19.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>19.2</b> Sample sheet</a></li>
<li class="chapter" data-level="19.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>19.3</b> SRR9264343</a></li>
<li class="chapter" data-level="19.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-14"><i class="fa fa-check"></i><b>19.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>20</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="20.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-5"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages-2"><i class="fa fa-check"></i><b>20.2</b> Load packages</a></li>
<li class="chapter" data-level="20.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>20.3</b> Sample sheet</a></li>
<li class="chapter" data-level="20.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation-1"><i class="fa fa-check"></i><b>20.4</b> Data representation</a></li>
<li class="chapter" data-level="20.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example-3"><i class="fa fa-check"></i><b>20.5</b> Example</a></li>
<li class="chapter" data-level="20.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>20.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="20.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>20.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="20.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>20.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>20.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="20.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model-1"><i class="fa fa-check"></i><b>20.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="20.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>20.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="20.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point-1"><i class="fa fa-check"></i><b>20.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="20.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>20.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="20.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>20.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="20.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>20.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="20.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>20.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="20.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control-1"><i class="fa fa-check"></i><b>20.10</b> Quality control</a><ul>
<li class="chapter" data-level="20.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>20.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="20.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>20.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="20.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size-1"><i class="fa fa-check"></i><b>20.11.1</b> Library size</a></li>
<li class="chapter" data-level="20.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>20.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="20.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>20.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="20.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary-1"><i class="fa fa-check"></i><b>20.11.4</b> Summary</a></li>
<li class="chapter" data-level="20.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>20.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="20.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions-1"><i class="fa fa-check"></i><b>20.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="20.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>20.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="20.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>20.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="20.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>20.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="20.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>20.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="20.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>20.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="20.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>20.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="20.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca-1"><i class="fa fa-check"></i><b>20.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="20.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>20.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="20.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>20.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty-1"><i class="fa fa-check"></i><b>20.13</b> Novelty</a></li>
<li class="chapter" data-level="20.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>20.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="20.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>20.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="20.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>20.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="20.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters-1"><i class="fa fa-check"></i><b>20.14.3</b> Filters</a></li>
<li class="chapter" data-level="20.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>20.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="20.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only-1"><i class="fa fa-check"></i><b>20.14.5</b> Caron only</a></li>
<li class="chapter" data-level="20.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only-1"><i class="fa fa-check"></i><b>20.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="20.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>20.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="20.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information-15"><i class="fa fa-check"></i><b>20.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>21</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="21.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>21.1</b> Caron</a><ul>
<li class="chapter" data-level="21.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>21.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>21.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>21.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>21.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>21.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>21.2</b> Hca</a><ul>
<li class="chapter" data-level="21.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>21.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>21.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>21.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>21.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-2"><i class="fa fa-check"></i><b>21.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform-3"><i class="fa fa-check"></i><b>21.3</b> SCTransform</a><ul>
<li class="chapter" data-level="21.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>21.3.1</b> Caron</a></li>
<li class="chapter" data-level="21.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>21.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-16"><i class="fa fa-check"></i><b>21.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>22</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="22.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>22.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="22.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>22.1.1</b> CPM</a></li>
<li class="chapter" data-level="22.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>22.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="22.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>22.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="22.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>22.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="22.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>22.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="22.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>22.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="22.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="22.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="22.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>22.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>22.2</b> SCTransform</a><ul>
<li class="chapter" data-level="22.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>22.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="22.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>22.2.2</b> Transformation</a></li>
<li class="chapter" data-level="22.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>22.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>22.3</b> Visualisation</a><ul>
<li class="chapter" data-level="22.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>22.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="22.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>22.3.2</b> log CPM</a></li>
<li class="chapter" data-level="22.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>22.3.3</b> scran</a></li>
<li class="chapter" data-level="22.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-5"><i class="fa fa-check"></i><b>22.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-17"><i class="fa fa-check"></i><b>22.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>23</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="23.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-3"><i class="fa fa-check"></i><b>23.1</b> Abbreviations</a></li>
<li class="chapter" data-level="23.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation-4"><i class="fa fa-check"></i><b>23.2</b> Motivation</a></li>
<li class="chapter" data-level="23.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>23.3</b> Loading the data</a></li>
<li class="chapter" data-level="23.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>23.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="23.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression-3"><i class="fa fa-check"></i><b>23.5</b> Linear regression</a></li>
<li class="chapter" data-level="23.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>23.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="23.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>23.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="23.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>23.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>23.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="23.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>23.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>23.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="23.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>23.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="23.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>23.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-18"><i class="fa fa-check"></i><b>23.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>24</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="24.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-4"><i class="fa fa-check"></i><b>24.1</b> Abbreviations</a></li>
<li class="chapter" data-level="24.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-5"><i class="fa fa-check"></i><b>24.2</b> Motivation</a></li>
<li class="chapter" data-level="24.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>24.3</b> Loading the data</a></li>
<li class="chapter" data-level="24.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>24.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="24.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>24.5</b> Linear regression</a></li>
<li class="chapter" data-level="24.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>24.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="24.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>24.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="24.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>24.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>24.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="24.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>24.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="24.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>24.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="24.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>24.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="24.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>24.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="24.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index-4"><i class="fa fa-check"></i><b>24.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>24.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="24.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-19"><i class="fa fa-check"></i><b>24.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>25</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="25.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-5"><i class="fa fa-check"></i><b>25.1</b> Abbreviations</a></li>
<li class="chapter" data-level="25.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-6"><i class="fa fa-check"></i><b>25.2</b> Motivation</a></li>
<li class="chapter" data-level="25.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-5"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>25.4</b> Loading the data</a></li>
<li class="chapter" data-level="25.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>25.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="25.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-5"><i class="fa fa-check"></i><b>25.6</b> Linear regression</a></li>
<li class="chapter" data-level="25.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>25.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="25.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>25.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="25.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>25.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>25.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="25.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>25.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="25.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>25.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="25.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>25.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="25.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>25.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="25.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-5"><i class="fa fa-check"></i><b>25.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>25.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="25.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-20"><i class="fa fa-check"></i><b>25.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>26</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="26.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-6"><i class="fa fa-check"></i><b>26.1</b> Abbreviations</a></li>
<li class="chapter" data-level="26.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-7"><i class="fa fa-check"></i><b>26.2</b> Motivation</a></li>
<li class="chapter" data-level="26.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-6"><i class="fa fa-check"></i><b>26.3</b> Load data</a></li>
<li class="chapter" data-level="26.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>26.4</b> Loading the data</a></li>
<li class="chapter" data-level="26.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>26.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="26.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-6"><i class="fa fa-check"></i><b>26.6</b> Linear regression</a></li>
<li class="chapter" data-level="26.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>26.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="26.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-6"><i class="fa fa-check"></i><b>26.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="26.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-6"><i class="fa fa-check"></i><b>26.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="26.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>26.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="26.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>26.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="26.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>26.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="26.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>26.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="26.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>26.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="26.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-6"><i class="fa fa-check"></i><b>26.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>26.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="26.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-21"><i class="fa fa-check"></i><b>26.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#learning-objectives-6"><i class="fa fa-check"></i><b>27.1</b> Learning objectives</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#abbreviations-7"><i class="fa fa-check"></i><b>27.2</b> Abbreviations</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#motivation-8"><i class="fa fa-check"></i><b>27.3</b> Motivation</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#load-the-data-3"><i class="fa fa-check"></i><b>27.4</b> Load the data</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>27.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#linear-regression-7"><i class="fa fa-check"></i><b>27.6</b> Linear regression</a></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>27.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>27.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>27.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>27.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.8.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mixing-between-batches-6"><i class="fa fa-check"></i><b>27.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.8.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>27.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.8.2.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-5"><i class="fa fa-check"></i><b>27.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.8.2.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>27.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.8.2.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#rand-index-7"><i class="fa fa-check"></i><b>27.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>27.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>27.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.11" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#session-information-22"><i class="fa fa-check"></i><b>27.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>28</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#learning-objectives-7"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-8"><i class="fa fa-check"></i><b>28.2</b> Abbreviations</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-9"><i class="fa fa-check"></i><b>28.3</b> Motivation</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-4"><i class="fa fa-check"></i><b>28.4</b> Load the data</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-8"><i class="fa fa-check"></i><b>28.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-8"><i class="fa fa-check"></i><b>28.6</b> Linear regression</a></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-4"><i class="fa fa-check"></i><b>28.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-8"><i class="fa fa-check"></i><b>28.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-8"><i class="fa fa-check"></i><b>28.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-8"><i class="fa fa-check"></i><b>28.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.8.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-7"><i class="fa fa-check"></i><b>28.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="28.8.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.8.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-6"><i class="fa fa-check"></i><b>28.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="28.8.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-7"><i class="fa fa-check"></i><b>28.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.8.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-8"><i class="fa fa-check"></i><b>28.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-8"><i class="fa fa-check"></i><b>28.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-4"><i class="fa fa-check"></i><b>28.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.11" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-23"><i class="fa fa-check"></i><b>28.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>29</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="29.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#learning-objectives-8"><i class="fa fa-check"></i><b>29.1</b> Learning objectives</a></li>
<li class="chapter" data-level="29.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-10"><i class="fa fa-check"></i><b>29.2</b> Motivation</a></li>
<li class="chapter" data-level="29.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-5"><i class="fa fa-check"></i><b>29.3</b> Load the data</a></li>
<li class="chapter" data-level="29.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-9"><i class="fa fa-check"></i><b>29.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="29.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-9"><i class="fa fa-check"></i><b>29.5</b> Linear regression</a></li>
<li class="chapter" data-level="29.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-5"><i class="fa fa-check"></i><b>29.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="29.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-9"><i class="fa fa-check"></i><b>29.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="29.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-9"><i class="fa fa-check"></i><b>29.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="29.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-9"><i class="fa fa-check"></i><b>29.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="29.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-10"><i class="fa fa-check"></i><b>29.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>29.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-8"><i class="fa fa-check"></i><b>29.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="29.7.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-9"><i class="fa fa-check"></i><b>29.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="29.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-9"><i class="fa fa-check"></i><b>29.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="29.7.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>29.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="29.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>29.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="29.8.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-11"><i class="fa fa-check"></i><b>29.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.8.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>29.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.8.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-10"><i class="fa fa-check"></i><b>29.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="29.8.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>29.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="29.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-5"><i class="fa fa-check"></i><b>29.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="29.10" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-24"><i class="fa fa-check"></i><b>29.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>30</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="30.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>30.1</b> Load data</a></li>
<li class="chapter" data-level="30.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-11"><i class="fa fa-check"></i><b>30.2</b> Motivation</a></li>
<li class="chapter" data-level="30.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>30.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="30.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>30.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="30.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>30.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="30.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-25"><i class="fa fa-check"></i><b>30.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>31</b> Doublet detection</a><ul>
<li class="chapter" data-level="31.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-9"><i class="fa fa-check"></i><b>31.1</b> Learning objectives</a></li>
<li class="chapter" data-level="31.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>31.2</b> Set up analysis</a></li>
<li class="chapter" data-level="31.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>31.3</b> Load data</a></li>
<li class="chapter" data-level="31.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>31.4</b> Overview</a></li>
<li class="chapter" data-level="31.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>31.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="31.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>31.6</b> Further comments</a></li>
<li class="chapter" data-level="31.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-26"><i class="fa fa-check"></i><b>31.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clustering-with-pbmmcetv6-runx1-and-_5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop" class="section level1">
<h1><span class="header-section-number">Chapter 14</span> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</h1>
<!-- TODO: get image file
<img src="../Images/Andrews2017_Fig1.png" style="margin:auto; display:block" />
-->
<p>Source: <a href="https://hemberg-lab.github.io/scRNA.seq.course/biological-analysis.html##clustering-methods">clustering methods</a> in the Hemberg group material and <a href="https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/clustering-and-cell-annotation.html#clustering-methods.">its ‘biocellgen variant’</a>, with some of its text copied with few edits only. Also see the <a href="https://osca.bioconductor.org/clustering.html">OSCA chapter on clustering</a> and a <a href="https://f1000research.com/articles/7-1297#ref-2">benchmark study</a>.</p>
<p>Once we have normalized the data and removed confounders we can carry out analyses that are relevant to the biological questions at hand. The exact nature of the analysis depends on the dataset. One of the most promising applications of scRNA-seq is <em>de novo</em> discovery and annotation of cell-types based on transcription profiles. This requires the identification of groups of cells based on the similarities of the transcriptomes without any prior knowledge of the labels, or unsupervised clustering. To avoid the challenges caused by the noise and high dimensionality of the scRNA-seq data, clustering is performed after feature selection and dimensionality reduction, usually on the PCA output.</p>
<p>We will introduce three widely used clustering methods: 1) hierarchical, 2) k-means and 3) graph-based clustering. In short, the first two were developed first and are faster for small data sets, while the third is more recent and better suited for scRNA-seq, especially large data sets. All three identify non-overlapping clusters.</p>
<p>We will apply them to the denoised log-expression values on the data set studied and measure clustering quality.</p>
<div id="load-data-3" class="section level2">
<h2><span class="header-section-number">14.1</span> Load data</h2>
<!--
We will load the R file keeping the SCE (SingleCellExperiment) object with the normalised counts for 500 cells per sample and the outcome of feature selection followed by dimensionality reduction.
-->
<div class="sourceCode" id="cb992"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb992-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-1"></a><span class="co">#setName &lt;- &quot;caron&quot;</span></span>
<span id="cb992-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-2"></a><span class="co">#setSuf &lt;- &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb992-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-3"></a><span class="co">#setSuf &lt;- &quot;&quot; # dev with 1k/spl, should be _1kCellPerSpl</span></span>
<span id="cb992-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-4"></a><span class="co">#setSuf &lt;- &quot;_allCells&quot; # dev with 1k/spl, should be _1kCellPerSpl</span></span>
<span id="cb992-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-5"></a></span>
<span id="cb992-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-6"></a><span class="co"># need library size</span></span>
<span id="cb992-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-7"></a><span class="co"># all sample types</span></span>
<span id="cb992-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-8"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb992-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-9"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb992-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-10"></a>{</span>
<span id="cb992-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-11"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb992-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-12"></a>}</span>
<span id="cb992-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-13"></a>sce0 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb992-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-14"></a></span>
<span id="cb992-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-15"></a><span class="co"># Read mnn.out object in:</span></span>
<span id="cb992-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-16"></a><span class="co"># only sample types in splSetToGet2</span></span>
<span id="cb992-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-17"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb992-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-18"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi_%s.Rds&quot;</span>,</span>
<span id="cb992-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-19"></a>        projDir, outDirBit, setName, setSuf, splSetToGet2)</span>
<span id="cb992-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb992-20"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1.Rds&quot;</code></pre>
<div class="sourceCode" id="cb994"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb994-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb994-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-2"></a>{</span>
<span id="cb994-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb994-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-4"></a>}</span>
<span id="cb994-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn) <span class="co"># ex mnn.out</span></span>
<span id="cb994-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-6"></a><span class="co">#sce2 &lt;- readRDS(tmpFn) # ex mnn.out</span></span>
<span id="cb994-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-7"></a><span class="co">#rownames(sce) &lt;- rownames(sce2)</span></span>
<span id="cb994-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb994-8"></a><span class="kw">rowData</span>(sce) &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">rowData</span>(sce), <span class="kw">rowData</span>(sce0)[<span class="kw">rownames</span>(sce),]) <span class="op">%&gt;%</span><span class="st"> </span>DataFrame</span></code></pre></div>
<p>Number of cells: .</p>
<p>Number of genes: 7930.</p>
</div>
<div id="clustering-cells-into-putative-subpopulations" class="section level2">
<h2><span class="header-section-number">14.2</span> Clustering cells into putative subpopulations</h2>
<!--
See https://hemberg-lab.github.io/scRNA.seq.course/index.html for three types of clustering.
See https://www.ncbi.nlm.nih.gov/pubmed/27303057 for review
-->
<!-- Defining cell clusters from expression data -->
<div id="hierarchical-clustering" class="section level3">
<h3><span class="header-section-number">14.2.1</span> Hierarchical clustering</h3>
<p>Hierarchical clustering builds a hierarchy of clusters yielding a dendrogram that groups together cells with similar expression patterns across the chosen genes.</p>
<p>There are two types of strategies:</p>
<ul>
<li>Agglomerative (bottom-up): each observation starts in its own cluster, and pairs of clusters are merged as one moves up the hierarchy.</li>
<li>Divisive (top-down): all observations start in one cluster, and splits are performed recursively as one moves down the hierarchy.</li>
</ul>
<!--
<img src="../Images/bioCellGenHierar1.png" style="margin:auto; display:block" />
<img src="../Images/bioCellGenHierar2.png" style="margin:auto; display:block" />
-->
<p>The raw data:</p>
<div class="sourceCode" id="cb995"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb995-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb995-1"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(<span class="st">&quot;../Images/bioCellGenHierar1.png&quot;</span>)</span></code></pre></div>
<p><img src="../Images/bioCellGenHierar1.png" width="30%" style="display: block; margin: auto;" /></p>
<p>The hierarchical clustering dendrogram:</p>
<div class="sourceCode" id="cb996"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb996-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb996-1"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(<span class="st">&quot;../Images/bioCellGenHierar2.png&quot;</span>)</span></code></pre></div>
<p><img src="../Images/bioCellGenHierar2.png" width="50%" style="display: block; margin: auto;" /></p>
<ul>
<li>Pros:
<ul>
<li>deterministic method</li>
<li>returns partitions at all levels along the dendrogram</li>
</ul></li>
<li>Cons:
<ul>
<li>computationally expensive in time and memory that increase proportionally to the square of the number of data points</li>
</ul></li>
</ul>
<div id="clustering" class="section level4">
<h4><span class="header-section-number">14.2.1.1</span> Clustering</h4>
<p>Here we will apply hierarchical clustering on the Euclidean distances between cells, using the Ward D2 criterion to minimize the total variance within each cluster.</p>
<div class="sourceCode" id="cb997"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb997-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-1"></a><span class="co"># get PCs</span></span>
<span id="cb997-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-2"></a><span class="co">#pcs &lt;- reducedDim(sce, &quot;PCA&quot;)</span></span>
<span id="cb997-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-3"></a>pcs &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;corrected&quot;</span>)</span>
<span id="cb997-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-4"></a><span class="co"># compute distance</span></span>
<span id="cb997-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-5"></a>pc.dist &lt;-<span class="st"> </span><span class="kw">dist</span>(pcs)</span>
<span id="cb997-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-6"></a><span class="co"># derive tree</span></span>
<span id="cb997-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-7"></a>hc.tree &lt;-<span class="st"> </span><span class="kw">hclust</span>(pc.dist, <span class="dt">method=</span><span class="st">&quot;ward.D2&quot;</span>)</span>
<span id="cb997-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb997-8"></a>hcd &lt;-<span class="st"> </span><span class="kw">as.dendrogram</span>(hc.tree)</span></code></pre></div>
<p>The dendrogram below shows each cell as a leaf.</p>
<div class="sourceCode" id="cb998"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb998-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb998-1"></a><span class="co">#plot(hc.tree, labels = FALSE, xlab = NULL, sub = NULL)</span></span>
<span id="cb998-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb998-2"></a><span class="kw">plot</span>(hcd, <span class="dt">type =</span> <span class="st">&quot;rectangle&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Height&quot;</span>, <span class="dt">leaflab =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_tree_hierar-1.png" width="672" /></p>
<p>Clusters are identified in the dendrogram using the shape of branches as well as their height (‘dynamic tree cut’ <span class="citation">(<span class="citeproc-not-found" data-reference-id="doi:10.1093/bioinformatics/btm563"><strong>???</strong></span>)</span>). <!-- https://academic.oup.com/bioinformatics/article/24/5/719/200751 --></p>
<div class="sourceCode" id="cb999"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb999-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb999-1"></a><span class="co"># identify clusters by cutting branches, requesting a minimum cluster size of 20 cells.</span></span>
<span id="cb999-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb999-2"></a>hc.clusters &lt;-<span class="st"> </span><span class="kw">unname</span>(<span class="kw">cutreeDynamic</span>(hc.tree,</span>
<span id="cb999-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb999-3"></a>                    <span class="dt">distM =</span> <span class="kw">as.matrix</span>(pc.dist),</span>
<span id="cb999-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb999-4"></a>                    <span class="dt">minClusterSize =</span> <span class="dv">20</span>,</span>
<span id="cb999-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb999-5"></a>                    <span class="dt">verbose =</span> <span class="dv">0</span>))</span></code></pre></div>
<p>Cell counts for each cluster (rows) and each sample group (columns):</p>
<div class="sourceCode" id="cb1000"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1000-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1000-1"></a><span class="co"># per sample group</span></span>
<span id="cb1000-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1000-2"></a><span class="kw">table</span>(hc.clusters, sce<span class="op">$</span>source_name)</span></code></pre></div>
<pre><code>##            
## hc.clusters ABMMC ETV6-RUNX1 HHD PBMMC PRE-T
##          1      0        609   0   183     0
##          2      0        189   0   441     0
##          3      0        512   0    94     0
##          4      0        340   0   151     0
##          5      0         99   0   107     0
##          6      0         76   0   126     0
##          7      0         13   0   167     0
##          8      0         45   0    70     0
##          9      0         73   0    25     0
##          10     0         24   0    68     0
##          11     0         20   0    68     0</code></pre>
<p>Cell counts for each cluster (rows) and each sample (columns):</p>
<div class="sourceCode" id="cb1002"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1002-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1002-1"></a><span class="co"># per sample</span></span>
<span id="cb1002-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1002-2"></a><span class="kw">table</span>(hc.clusters, sce<span class="op">$</span>Sample.Name)</span></code></pre></div>
<pre><code>##            
## hc.clusters GSM3872434 GSM3872435 GSM3872436 GSM3872437 GSM3872442 GSM3872443
##          1          50        301        159         99         73         18
##          2           1         15        154         19        185        125
##          3         299         83         28        102         57         15
##          4         121         61         38        120         90         11
##          5           0          0         21         78          0        101
##          6           2          9         51         14         17         54
##          7           1          0          5          7         38         50
##          8           0          1         11         33          1         57
##          9          20         28         11         14          6         10
##          10          4          1          8         11         14         36
##          11          2          1         14          3         19         23
##            
## hc.clusters GSM3872444
##          1          92
##          2         131
##          3          22
##          4          50
##          5           6
##          6          55
##          7          79
##          8          12
##          9           9
##          10         18
##          11         26</code></pre>
<p>This data (cell number) may also be shown as heatmap.</p>
<div class="sourceCode" id="cb1004"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1004-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-1"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(hc.clusters, sce<span class="op">$</span>Sample.Name)</span>
<span id="cb1004-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-2"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;hc&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1004-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-3"></a></span>
<span id="cb1004-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-4"></a><span class="co"># columns annotation with cell name:</span></span>
<span id="cb1004-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-5"></a>mat_col &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(Sample.Name, source_name) <span class="op">%&gt;%</span><span class="st"> </span>unique</span>
<span id="cb1004-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-6"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span>mat_col<span class="op">$</span>Sample.Name</span>
<span id="cb1004-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-7"></a>mat_col<span class="op">$</span>Sample.Name &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb1004-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-8"></a></span>
<span id="cb1004-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-9"></a><span class="co"># Prepare colours for clusters:</span></span>
<span id="cb1004-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-10"></a>colourCount =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(mat_col<span class="op">$</span>source_name))</span>
<span id="cb1004-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-11"></a>getPalette =<span class="st"> </span>grDevices<span class="op">::</span><span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb1004-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-12"></a></span>
<span id="cb1004-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-13"></a>mat_colors &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">source_name =</span> <span class="kw">getPalette</span>(colourCount))</span>
<span id="cb1004-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1004-14"></a><span class="kw">names</span>(mat_colors<span class="op">$</span>source_name) &lt;-<span class="st"> </span><span class="kw">unique</span>(mat_col<span class="op">$</span>source_name)</span></code></pre></div>
<p>Heatmap, with samples ordered as in sample sheet:</p>
<div class="sourceCode" id="cb1005"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1005-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1005-1"></a><span class="co"># without column clustering</span></span>
<span id="cb1005-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1005-2"></a><span class="kw">pheatmap</span>(tmpTab,</span>
<span id="cb1005-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1005-3"></a>           <span class="dt">cluster_cols =</span> <span class="ot">FALSE</span>,</span>
<span id="cb1005-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1005-4"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1005-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1005-5"></a>           <span class="dt">annotation_colors =</span> mat_colors)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Heatmap, with samples ordered by similarity of cell distribution across clusters:</p>
<div class="sourceCode" id="cb1006"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1006-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1006-1"></a><span class="co"># with column clustering</span></span>
<span id="cb1006-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1006-2"></a><span class="kw">pheatmap</span>(tmpTab,</span>
<span id="cb1006-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1006-3"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1006-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1006-4"></a>           <span class="dt">annotation_colors =</span> mat_colors)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>If clusters mostly include cells from one sample or the other, it suggests that the samples differ, and/or the presence of batch effect.</p>
<p>Let us show cluster assignments on the t-SNE, with cells shaped by cell type,
colored by cluster and sized total UMI counts (‘sum’).</p>
<div class="sourceCode" id="cb1007"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1007-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-1"></a><span class="co"># store cluster assignment in SCE object:</span></span>
<span id="cb1007-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-2"></a>sce<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">factor</span>(hc.clusters)</span>
<span id="cb1007-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-3"></a><span class="co"># make, store and show TSNE plot:</span></span>
<span id="cb1007-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-4"></a><span class="co">#g &lt;- plotTSNE(sce, colour_by = &quot;cluster&quot;, size_by = &quot;sum&quot;, shape_by=&quot;source_name&quot;)</span></span>
<span id="cb1007-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-5"></a>g &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by =</span> <span class="st">&quot;cluster&quot;</span>, <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>)</span>
<span id="cb1007-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1007-6"></a>g</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_tsne_hierar-1.png" width="768" /></p>
<p>Split by sample group:</p>
<div class="sourceCode" id="cb1008"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1008-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1008-1"></a><span class="co"># split by sample and show:</span></span>
<span id="cb1008-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1008-2"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(. <span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span>
<span id="cb1008-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1008-3"></a>g</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_tsne_hierar_facet-1.png" width="960" /></p>
<p>In some areas cells are not all assigned to the same cluster.</p>
</div>
<div id="separatedness" class="section level4">
<h4><span class="header-section-number">14.2.1.2</span> Separatedness</h4>
<p>The congruence of clusters may be assessed by computing the sillhouette for each cell.
The larger the value the closer the cell to cells in its cluster than to cells in other clusters.
Cells closer to cells in other clusters have a negative value.
Good cluster separation is indicated by clusters whose cells have large silhouette values.</p>
<p>We first compute silhouette values.</p>
<div class="sourceCode" id="cb1009"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1009-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1009-1"></a>sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(hc.clusters, <span class="dt">dist =</span> pc.dist)</span></code></pre></div>
<p>We then plot silhouettes with one color per cluster and cells with a negative silhouette with the color of their closest cluster.
We also add the average silhouette for each cluster and all cells.</p>
<div class="sourceCode" id="cb1010"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1010-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-1"></a><span class="co"># prepare colours:</span></span>
<span id="cb1010-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-2"></a>clust.col &lt;-<span class="st"> </span>scater<span class="op">:::</span><span class="kw">.get_palette</span>(<span class="st">&quot;tableau10medium&quot;</span>) <span class="co"># hidden scater colours</span></span>
<span id="cb1010-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-3"></a>sil.cols &lt;-<span class="st"> </span>clust.col[<span class="kw">ifelse</span>(sil[,<span class="dv">3</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, sil[,<span class="dv">1</span>], sil[,<span class="dv">2</span>])]</span>
<span id="cb1010-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-4"></a>sil.cols &lt;-<span class="st"> </span>sil.cols[<span class="kw">order</span>(<span class="op">-</span>sil[,<span class="dv">1</span>], sil[,<span class="dv">3</span>])]</span>
<span id="cb1010-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-5"></a><span class="co"># plot:</span></span>
<span id="cb1010-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-6"></a><span class="kw">plot</span>(sil,</span>
<span id="cb1010-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-7"></a>     <span class="dt">main =</span> <span class="kw">paste</span>(<span class="kw">length</span>(<span class="kw">unique</span>(hc.clusters)), <span class="st">&quot;clusters&quot;</span>),</span>
<span id="cb1010-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-8"></a>     <span class="dt">border =</span> sil.cols,</span>
<span id="cb1010-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-9"></a>     <span class="dt">col =</span> sil.cols,</span>
<span id="cb1010-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1010-10"></a>     <span class="dt">do.col.sort =</span> <span class="ot">FALSE</span>) </span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_silhouette_hierar-1.png" width="672" /></p>
<p>The plot shows cells with negative silhouette indicating that too many clusters may have been defined.
The method and parameters used defined clusters with properties that may not fit the data set, eg clusters with the same diameter.</p>
</div>
</div>
<div id="k-means" class="section level3">
<h3><span class="header-section-number">14.2.2</span> k-means</h3>
<div id="description-1" class="section level4">
<h4><span class="header-section-number">14.2.2.1</span> Description</h4>
<p>In k-means clustering, the goal is to partition N cells into k different clusters. In an iterative manner, cluster centers are defined and each cell is assigned to its nearest cluster.</p>
<p>The aim is to minimise within-cluster variation and maximise between-cluster variation, using the following steps:</p>
<ul>
<li>randomly select k data points to serve as initial cluster centers,</li>
<li>for each point, compute the distance between that point and each of the centroids and assign the point to the cluster with the closest centroid,</li>
<li>calculate the mean of each cluster (the ‘mean’ in ‘k-mean’) to define its centroid, and for each point compute the distance to these means to choose the closest, repeat until the distance between centroids and data points is minimal (ie clusters do not change) or the maximum number of iterations is reached,</li>
<li>compute the total variation within clusters,</li>
<li>assign new centroids and repeat steps above</li>
</ul>
<p><img src="../Images/bioCellGenKmean.png" style="margin:auto; display:block" /></p>
<ul>
<li>Pros:
<ul>
<li>fast</li>
</ul></li>
<li>Cons:
<ul>
<li>assumes a pre-determined number of clusters</li>
<li>sensitive to outliers</li>
<li>tends to define equally-sized clusters</li>
</ul></li>
</ul>
</div>
<div id="example-2" class="section level4">
<h4><span class="header-section-number">14.2.2.2</span> Example</h4>
<p>The dendogram built above suggests there may be six large cell populations.</p>
<p>Let us define six clusters.</p>
<div class="sourceCode" id="cb1011"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1011-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-1"></a><span class="co"># define clusters with kmeans()</span></span>
<span id="cb1011-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-2"></a><span class="co"># because results depend on the initial cluster centers,</span></span>
<span id="cb1011-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-3"></a><span class="co"># it is usually best to try several times,</span></span>
<span id="cb1011-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-4"></a><span class="co"># by setting &#39;nstart&#39; to say 20,</span></span>
<span id="cb1011-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-5"></a><span class="co"># kmeans() will then retain the run with the lowest within cluster variation.</span></span>
<span id="cb1011-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-6"></a>kclust &lt;-<span class="st"> </span><span class="kw">kmeans</span>(pcs, <span class="dt">centers=</span><span class="dv">6</span>, <span class="dt">nstart =</span> <span class="dv">20</span>) </span>
<span id="cb1011-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-7"></a></span>
<span id="cb1011-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-8"></a><span class="co"># compute silhouette</span></span>
<span id="cb1011-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-9"></a>sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(kclust<span class="op">$</span>cluster, <span class="kw">dist</span>(pcs))</span>
<span id="cb1011-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-10"></a></span>
<span id="cb1011-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-11"></a><span class="co"># plot silhouette</span></span>
<span id="cb1011-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-12"></a>clust.col &lt;-<span class="st"> </span>scater<span class="op">:::</span><span class="kw">.get_palette</span>(<span class="st">&quot;tableau10medium&quot;</span>) <span class="co"># hidden scater colours</span></span>
<span id="cb1011-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-13"></a>sil.cols &lt;-<span class="st"> </span>clust.col[<span class="kw">ifelse</span>(sil[,<span class="dv">3</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, sil[,<span class="dv">1</span>], sil[,<span class="dv">2</span>])]</span>
<span id="cb1011-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-14"></a>sil.cols &lt;-<span class="st"> </span>sil.cols[<span class="kw">order</span>(<span class="op">-</span>sil[,<span class="dv">1</span>], sil[,<span class="dv">3</span>])]</span>
<span id="cb1011-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-15"></a><span class="kw">plot</span>(sil, <span class="dt">main =</span> <span class="kw">paste</span>(<span class="kw">length</span>(<span class="kw">unique</span>(kclust<span class="op">$</span>cluster)), <span class="st">&quot;clusters&quot;</span>), </span>
<span id="cb1011-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1011-16"></a>    <span class="dt">border=</span>sil.cols, <span class="dt">col=</span>sil.cols, <span class="dt">do.col.sort=</span><span class="ot">FALSE</span>) </span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/comp_kmeans_k6-1.png" width="672" /></p>
<p>The t-SNE plot below shows cells color-coded by cluster membership:</p>
<div class="sourceCode" id="cb1012"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1012-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-1"></a><span class="co"># get cell coordinates from &quot;TSNE&quot; slot:</span></span>
<span id="cb1012-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-2"></a>tSneCoord &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE&quot;</span>))</span>
<span id="cb1012-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-3"></a><span class="kw">colnames</span>(tSneCoord) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TSNE1&quot;</span>, <span class="st">&quot;TSNE2&quot;</span>)</span>
<span id="cb1012-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-4"></a><span class="co"># add sample type:</span></span>
<span id="cb1012-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-5"></a>tSneCoord<span class="op">$</span>source_name &lt;-<span class="st"> </span>sce<span class="op">$</span>source_name</span>
<span id="cb1012-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-6"></a><span class="co"># add cluster info:</span></span>
<span id="cb1012-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-7"></a>tSneCoord<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">as.factor</span>(kclust<span class="op">$</span>cluster)</span>
<span id="cb1012-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-8"></a><span class="co"># draw plot</span></span>
<span id="cb1012-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-9"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tSneCoord, <span class="kw">aes</span>(<span class="dt">x=</span>TSNE1, <span class="dt">y=</span>TSNE2,</span>
<span id="cb1012-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-10"></a>                <span class="dt">colour=</span>cluster,</span>
<span id="cb1012-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-11"></a>                <span class="dt">shape=</span>source_name)) <span class="op">+</span></span>
<span id="cb1012-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-12"></a><span class="st">    </span><span class="kw">geom_point</span>()</span>
<span id="cb1012-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1012-13"></a>p2</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_tSNE_kmeans_k6-1.png" width="768" /></p>
<p>Split by sample type:</p>
<div class="sourceCode" id="cb1013"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1013-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1013-1"></a>p2 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_tSNE_kmeans_k6_split-1.png" width="960" /></p>
<p>To find the most appropriate number of clusters, one performs the analysis for a series of values for k and for each k value computes a ‘measure of fit’ of the clusters defined. Several metrics exist. We will look at:
* within-cluster sum-of-squares
* silhouette
* gap statistic</p>
</div>
<div id="within-cluster-sum-of-squares" class="section level4">
<h4><span class="header-section-number">14.2.2.3</span> Within-cluster sum-of-squares</h4>
<p>The within-cluster sum-of-squares is the sum of the squared deviations from each observation and the cluster centroid. This metric measures the variability of observations within a cluster. It decreases as k increases, by an amount that decreases with k. Indeed, for low k values increasing k usually improves clustering, as shown by a sharp drop in the within-cluster sum-of-squares. For higher k values, increasing k does not reduce the within-cluster sum-of-squares much: dividing clusters further only slightly reduces variablility inside clusters. On a plot of the within-cluster sum-of-squares against k, the curve shows two parts: the first with a steep negative slope and the second with a small slope. The ‘elbow’ of the curve indicates the most appropriate number of clusters.</p>
<div class="sourceCode" id="cb1014"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1014-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-1"></a><span class="kw">library</span>(broom)</span>
<span id="cb1014-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-2"></a><span class="kw">library</span>(tibble)</span>
<span id="cb1014-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-3"></a><span class="kw">library</span>(tidyr)</span>
<span id="cb1014-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-4"></a><span class="kw">library</span>(purrr)</span>
<span id="cb1014-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-5"></a></span>
<span id="cb1014-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-6"></a><span class="co"># get PCA matrix (i.e. rotated values)</span></span>
<span id="cb1014-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-7"></a>points &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(pcs)</span>
<span id="cb1014-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-8"></a></span>
<span id="cb1014-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-9"></a><span class="co"># define clusters for different number of clusters</span></span>
<span id="cb1014-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-10"></a><span class="co"># from 1 to 20:</span></span>
<span id="cb1014-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-11"></a>kclusts &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">k =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>) <span class="op">%&gt;%</span></span>
<span id="cb1014-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-12"></a><span class="st">  </span><span class="kw">mutate</span>(</span>
<span id="cb1014-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-13"></a>    <span class="dt">kclust =</span> <span class="kw">map</span>(k, <span class="op">~</span><span class="kw">kmeans</span>(points, .x)), <span class="co"># define clusters</span></span>
<span id="cb1014-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-14"></a>    <span class="dt">tidied =</span> <span class="kw">map</span>(kclust, tidy), <span class="co"># &#39;flatten&#39; kmeans() output into tibble</span></span>
<span id="cb1014-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-15"></a>    <span class="dt">glanced =</span> <span class="kw">map</span>(kclust, glance), <span class="co"># convert model or other R object to convert to single-row data frame</span></span>
<span id="cb1014-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-16"></a>    <span class="dt">augmented =</span> <span class="kw">map</span>(kclust, augment, points) <span class="co"># extract per-observation information</span></span>
<span id="cb1014-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-17"></a>  )</span>
<span id="cb1014-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-18"></a></span>
<span id="cb1014-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-19"></a><span class="co"># get cluster assignments</span></span>
<span id="cb1014-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-20"></a><span class="co"># unnest a list column with unnest(),</span></span>
<span id="cb1014-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-21"></a><span class="co"># i.e. make each element of the list its own row.</span></span>
<span id="cb1014-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-22"></a>clusters &lt;-<span class="st"> </span>kclusts <span class="op">%&gt;%</span></span>
<span id="cb1014-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-23"></a><span class="st">  </span><span class="kw">unnest</span>(tidied)</span>
<span id="cb1014-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-24"></a></span>
<span id="cb1014-25"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-25"></a><span class="co"># get assignments</span></span>
<span id="cb1014-26"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-26"></a>assignments &lt;-<span class="st"> </span>kclusts <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1014-27"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-27"></a><span class="st">  </span><span class="kw">unnest</span>(augmented)</span>
<span id="cb1014-28"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-28"></a></span>
<span id="cb1014-29"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-29"></a><span class="co"># get clustering outcome</span></span>
<span id="cb1014-30"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-30"></a>clusterings &lt;-<span class="st"> </span>kclusts <span class="op">%&gt;%</span></span>
<span id="cb1014-31"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1014-31"></a><span class="st">  </span><span class="kw">unnest</span>(glanced)</span></code></pre></div>
<p>We now plot the total within-cluster sum-of-squares and decide on k.</p>
<div class="sourceCode" id="cb1015"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1015-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1015-1"></a><span class="kw">ggplot</span>(clusterings, <span class="kw">aes</span>(k, tot.withinss)) <span class="op">+</span></span>
<span id="cb1015-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1015-2"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb1015-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1015-3"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_withinss-1.png" width="672" /></p>
<div class="sourceCode" id="cb1016"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1016-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1016-1"></a>clusterings <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1016-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1016-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tot.withinss.diff =</span> tot.withinss <span class="op">-</span><span class="st"> </span><span class="kw">lag</span>(tot.withinss)) <span class="op">%&gt;%</span></span>
<span id="cb1016-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1016-3"></a><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(tot.withinss.diff))</span></code></pre></div>
<pre><code>## # A tibble: 20 x 9
##        k kclust  tidied       totss tot.withinss betweenss  iter augmented      
##    &lt;int&gt; &lt;list&gt;  &lt;list&gt;       &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;int&gt; &lt;list&gt;         
##  1    17 &lt;kmean… &lt;tibble [17…  418.         126.  2.92e+ 2     5 &lt;tibble [3,500…
##  2    20 &lt;kmean… &lt;tibble [20…  418.         120.  2.98e+ 2     6 &lt;tibble [3,500…
##  3     8 &lt;kmean… &lt;tibble [8 …  418.         170.  2.48e+ 2     4 &lt;tibble [3,500…
##  4    19 &lt;kmean… &lt;tibble [19…  418.         120.  2.98e+ 2     6 &lt;tibble [3,500…
##  5    14 &lt;kmean… &lt;tibble [14…  418.         133.  2.85e+ 2     4 &lt;tibble [3,500…
##  6    16 &lt;kmean… &lt;tibble [16…  418.         124.  2.94e+ 2     5 &lt;tibble [3,500…
##  7    12 &lt;kmean… &lt;tibble [12…  418.         138.  2.80e+ 2     4 &lt;tibble [3,500…
##  8    13 &lt;kmean… &lt;tibble [13…  418.         135.  2.84e+ 2     4 &lt;tibble [3,500…
##  9    11 &lt;kmean… &lt;tibble [11…  418.         142.  2.76e+ 2     4 &lt;tibble [3,500…
## 10    15 &lt;kmean… &lt;tibble [15…  418.         127.  2.91e+ 2     4 &lt;tibble [3,500…
## 11    18 &lt;kmean… &lt;tibble [18…  418.         120.  2.98e+ 2     4 &lt;tibble [3,500…
## 12    10 &lt;kmean… &lt;tibble [10…  418.         146.  2.72e+ 2     5 &lt;tibble [3,500…
## 13     7 &lt;kmean… &lt;tibble [7 …  418.         170.  2.49e+ 2     4 &lt;tibble [3,500…
## 14     6 &lt;kmean… &lt;tibble [6 …  418.         182.  2.37e+ 2     5 &lt;tibble [3,500…
## 15     9 &lt;kmean… &lt;tibble [9 …  418.         156.  2.62e+ 2     5 &lt;tibble [3,500…
## 16     5 &lt;kmean… &lt;tibble [5 …  418.         195.  2.23e+ 2     4 &lt;tibble [3,500…
## 17     4 &lt;kmean… &lt;tibble [4 …  418.         219.  1.99e+ 2     3 &lt;tibble [3,500…
## 18     3 &lt;kmean… &lt;tibble [3 …  418.         249.  1.69e+ 2     3 &lt;tibble [3,500…
## 19     2 &lt;kmean… &lt;tibble [2 …  418.         305.  1.13e+ 2     1 &lt;tibble [3,500…
## 20     1 &lt;kmean… &lt;tibble [1 …  418.         418. -2.27e-13     1 &lt;tibble [3,500…
## # … with 1 more variable: tot.withinss.diff &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb1018"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1018-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-1"></a><span class="co"># get the smallest negative drop</span></span>
<span id="cb1018-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-2"></a>k_neg &lt;-<span class="st"> </span>clusterings <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1018-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tot.withinss.diff =</span> tot.withinss <span class="op">-</span><span class="st"> </span><span class="kw">lag</span>(tot.withinss)) <span class="op">%&gt;%</span></span>
<span id="cb1018-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-4"></a><span class="st">  </span><span class="kw">filter</span>(tot.withinss.diff <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb1018-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-5"></a><span class="st">  </span><span class="co">#arrange(desc(tot.withinss.diff)) %&gt;%</span></span>
<span id="cb1018-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-6"></a><span class="st">  </span><span class="kw">slice_max</span>(tot.withinss.diff, <span class="dt">n =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb1018-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-7"></a><span class="st">  </span><span class="kw">pull</span>(k)</span>
<span id="cb1018-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-8"></a>k_neg &lt;-<span class="st"> </span>k_neg <span class="dv">-1</span></span>
<span id="cb1018-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-9"></a><span class="co"># get the first positive diff</span></span>
<span id="cb1018-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-10"></a>k_pos &lt;-<span class="st"> </span>clusterings <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1018-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tot.withinss.diff =</span> tot.withinss <span class="op">-</span><span class="st"> </span><span class="kw">lag</span>(tot.withinss)) <span class="op">%&gt;%</span></span>
<span id="cb1018-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-12"></a><span class="st">  </span><span class="kw">filter</span>(tot.withinss.diff <span class="op">&gt;=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb1018-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-13"></a><span class="st">  </span><span class="co">#arrange(desc(tot.withinss.diff)) %&gt;%</span></span>
<span id="cb1018-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-14"></a><span class="st">  </span><span class="kw">slice_min</span>(k, <span class="dt">n =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb1018-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-15"></a><span class="st">  </span><span class="kw">pull</span>(k)</span>
<span id="cb1018-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1018-16"></a>k_pos &lt;-<span class="st"> </span>k_pos <span class="dv">-1</span></span></code></pre></div>
<p>The plot above suggests reasonable values for k may be:</p>
<ul>
<li>13 (where the drop in total within-cluster sum-of-squares is the smallest)</li>
<li>7 (lowest cluster number where the difference in consecutive total within-cluster sum-of-squares is positive)</li>
</ul>
</div>
<div id="silhouette" class="section level4">
<h4><span class="header-section-number">14.2.2.4</span> Silhouette</h4>
<p>We now compute the Silhouette values for each set of clusters defined above for the series of values of k.</p>
<div class="sourceCode" id="cb1019"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1019-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-1"></a><span class="co"># compute pairwise distance between cells in the PC space,</span></span>
<span id="cb1019-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-2"></a><span class="co"># as it is needed to compute silhouette:</span></span>
<span id="cb1019-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-3"></a>pcDist &lt;-<span class="st"> </span><span class="kw">dist</span>(pcs)</span>
<span id="cb1019-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-4"></a><span class="co"># compute silhouette for each set of clusters defined above for a series of values of k:</span></span>
<span id="cb1019-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-5"></a>Ks=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">20</span>,</span>
<span id="cb1019-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-6"></a>        <span class="cf">function</span>(i) {</span>
<span id="cb1019-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-7"></a>            tmpClu &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(kclusts<span class="op">$</span>augmented[[i]]<span class="op">$</span>.cluster)</span>
<span id="cb1019-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-8"></a>            <span class="co">#table(tmpClu)</span></span>
<span id="cb1019-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-9"></a>            sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(tmpClu, pcDist)</span>
<span id="cb1019-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-10"></a>            <span class="kw">summary</span>(sil)<span class="op">$</span>avg.width</span>
<span id="cb1019-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-11"></a>        }</span>
<span id="cb1019-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-12"></a>)</span>
<span id="cb1019-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-13"></a><span class="co"># plot average width against k:</span></span>
<span id="cb1019-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-14"></a><span class="kw">plot</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">20</span>, Ks,</span>
<span id="cb1019-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-15"></a>     <span class="dt">xlab=</span><span class="st">&quot;k&quot;</span>,</span>
<span id="cb1019-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-16"></a>     <span class="dt">ylab=</span><span class="st">&quot;av. silhouette&quot;</span>,</span>
<span id="cb1019-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-17"></a>     <span class="dt">type=</span><span class="st">&quot;b&quot;</span>,</span>
<span id="cb1019-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1019-18"></a>     <span class="dt">pch=</span><span class="dv">19</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<!--
The plot above shows that silhouette values vary little for k between 4 and 11, though higher values are observed for k at 5, 8, 9 and 10.
-->
</div>
<div id="gap-statistic" class="section level4">
<h4><span class="header-section-number">14.2.2.5</span> Gap statistic</h4>
<p>Because the variation quantity decreases as the number of clusters increases,
a more reliable approach is to compare the observed variation to that expected from a null distribution.
With the gap statistic, the expected variation is computed:</p>
<ul>
<li>by generating several reference data sets using the uniform distribution (no cluster),</li>
<li>for each such set and the series of k values, by clustering and computing the within-cluster variation.</li>
</ul>
<p>The gap statistic measures for a given k the difference between the expected and observed variation.
The most appropriate number of clusters is that with the higest gap value.</p>
<p>We will use <code>cluster::clusGap()</code> to compute the gap statistic for k between 1 and 20.</p>
<div class="sourceCode" id="cb1020"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1020-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-1"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb1020-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-2"></a>gaps &lt;-<span class="st"> </span>cluster<span class="op">::</span><span class="kw">clusGap</span>(</span>
<span id="cb1020-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-3"></a>  <span class="dt">x =</span> pcs,</span>
<span id="cb1020-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-4"></a>  <span class="dt">FUNcluster =</span> kmeans,</span>
<span id="cb1020-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-5"></a>  <span class="dt">K.max =</span> <span class="dv">20</span>,</span>
<span id="cb1020-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-6"></a>  <span class="dt">nstart =</span> <span class="dv">5</span>, <span class="co"># low for expediency here but should use higher</span></span>
<span id="cb1020-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-7"></a>  <span class="dt">B =</span> <span class="dv">10</span> <span class="co"># low for expediency here but should use higher</span></span>
<span id="cb1020-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-8"></a>)</span>
<span id="cb1020-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-9"></a></span>
<span id="cb1020-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-10"></a><span class="co">## find the &quot;best&quot; k</span></span>
<span id="cb1020-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-11"></a>best.k &lt;-<span class="st"> </span>cluster<span class="op">::</span><span class="kw">maxSE</span>(gaps<span class="op">$</span>Tab[, <span class="st">&quot;gap&quot;</span>], gaps<span class="op">$</span>Tab[, <span class="st">&quot;SE.sim&quot;</span>])</span>
<span id="cb1020-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-12"></a><span class="co"># in case the the best k was &#39;1&#39;,</span></span>
<span id="cb1020-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-13"></a><span class="co"># skip first row of output table:</span></span>
<span id="cb1020-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-14"></a>gapsTab &lt;-<span class="st"> </span>gaps<span class="op">$</span>Tab[<span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(gaps<span class="op">$</span>Tab),]</span>
<span id="cb1020-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-15"></a>best.k &lt;-<span class="st"> </span>cluster<span class="op">::</span><span class="kw">maxSE</span>(gapsTab[, <span class="st">&quot;gap&quot;</span>], gapsTab[, <span class="st">&quot;SE.sim&quot;</span>])</span>
<span id="cb1020-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1020-16"></a><span class="co">#best.k</span></span></code></pre></div>
<p>The “optimal” k value is 19.</p>
<div class="sourceCode" id="cb1021"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1021-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1021-1"></a><span class="co"># try other method to choose best k:</span></span>
<span id="cb1021-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1021-2"></a><span class="co">##best.k &lt;- cluster::maxSE(gapsTab[, &quot;gap&quot;], gapsTab[, &quot;SE.sim&quot;], method=&quot;Tibs2001SEmax&quot;)</span></span>
<span id="cb1021-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1021-3"></a><span class="co">##best.k &lt;- cluster::maxSE(gapsTab[, &quot;gap&quot;], gapsTab[, &quot;SE.sim&quot;], method=&quot;firstmax&quot;)</span></span>
<span id="cb1021-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1021-4"></a><span class="co">##best.k</span></span></code></pre></div>
<p>We next copy the cluster assignment to the SCE object.</p>
<div class="sourceCode" id="cb1022"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1022-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1022-1"></a>df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(assignments)</span>
<span id="cb1022-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1022-2"></a><span class="co"># here only copy outcome for k == 10</span></span>
<span id="cb1022-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1022-3"></a>sce<span class="op">$</span>kmeans10 &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(df[df<span class="op">$</span>k <span class="op">==</span><span class="st"> </span><span class="dv">10</span>, <span class="st">&quot;.cluster&quot;</span>])</span></code></pre></div>
<p>We now check silhouette values for a k of 10:</p>
<div class="sourceCode" id="cb1023"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1023-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-1"></a><span class="kw">library</span>(cluster)</span>
<span id="cb1023-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-2"></a>clust.col &lt;-<span class="st"> </span>scater<span class="op">:::</span><span class="kw">.get_palette</span>(<span class="st">&quot;tableau10medium&quot;</span>) <span class="co"># hidden scater colours</span></span>
<span id="cb1023-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-3"></a>sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(sce<span class="op">$</span>kmeans10, <span class="dt">dist =</span> pc.dist)</span>
<span id="cb1023-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-4"></a>sil.cols &lt;-<span class="st"> </span>clust.col[<span class="kw">ifelse</span>(sil[,<span class="dv">3</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, sil[,<span class="dv">1</span>], sil[,<span class="dv">2</span>])]</span>
<span id="cb1023-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-5"></a>sil.cols &lt;-<span class="st"> </span>sil.cols[<span class="kw">order</span>(<span class="op">-</span>sil[,<span class="dv">1</span>], sil[,<span class="dv">3</span>])]</span>
<span id="cb1023-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-6"></a><span class="kw">plot</span>(sil, <span class="dt">main =</span> <span class="kw">paste</span>(<span class="kw">length</span>(<span class="kw">unique</span>(sce<span class="op">$</span>kmeans10)), <span class="st">&quot;clusters&quot;</span>), </span>
<span id="cb1023-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1023-7"></a>    <span class="dt">border=</span>sil.cols, <span class="dt">col=</span>sil.cols, <span class="dt">do.col.sort=</span><span class="ot">FALSE</span>) </span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/silhouette_kmeans_k10-1.png" width="672" /></p>
<div class="sourceCode" id="cb1024"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1024-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1024-1"></a>df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(assignments)</span>
<span id="cb1024-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1024-2"></a>sce<span class="op">$</span>kmeans.best &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(df[df<span class="op">$</span>k <span class="op">==</span><span class="st"> </span>best.k, <span class="st">&quot;.cluster&quot;</span>])</span></code></pre></div>
<p>We now check silhouette values for a k of 19:</p>
<div class="sourceCode" id="cb1025"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1025-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-1"></a><span class="kw">library</span>(cluster)</span>
<span id="cb1025-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-2"></a>clust.col &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&quot;RdBu&quot;</span>))(best.k)</span>
<span id="cb1025-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-3"></a>sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(sce<span class="op">$</span>kmeans.best, <span class="dt">dist =</span> pc.dist)</span>
<span id="cb1025-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-4"></a>sil.cols &lt;-<span class="st"> </span>clust.col[<span class="kw">ifelse</span>(sil[,<span class="dv">3</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, sil[,<span class="dv">1</span>], sil[,<span class="dv">2</span>])]</span>
<span id="cb1025-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-5"></a>sil.cols &lt;-<span class="st"> </span>sil.cols[<span class="kw">order</span>(<span class="op">-</span>sil[,<span class="dv">1</span>], sil[,<span class="dv">3</span>])]</span>
<span id="cb1025-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-6"></a><span class="kw">plot</span>(sil, <span class="dt">main =</span> <span class="kw">paste</span>(<span class="kw">length</span>(<span class="kw">unique</span>(sce<span class="op">$</span>kmeans.best)), <span class="st">&quot;clusters&quot;</span>), </span>
<span id="cb1025-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1025-7"></a>    <span class="dt">border=</span>sil.cols, <span class="dt">col=</span>sil.cols, <span class="dt">do.col.sort=</span><span class="ot">FALSE</span>) </span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/silhouette_kmeans_kbest-1.png" width="672" /></p>
<p>For the same number of clusters (10), fewer cells have negative values with k-means than with hierarchical clustering. The average silhouette is similar though, and all clusters show a wide range.</p>
</div>
<div id="multi-metric" class="section level4">
<h4><span class="header-section-number">14.2.2.6</span> Multi-metric</h4>
<!-- TODO Keep? Implies proper choice of metrics ... -->
<p>One may also use multiple metrics rather than only one and check how they concord, using <code>NbClust::NbClust()</code> (see manual for a description of metrics computed):</p>
<div class="sourceCode" id="cb1026"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1026-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-1"></a><span class="co"># run only once and save outcome to file to load later.</span></span>
<span id="cb1026-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-2"></a></span>
<span id="cb1026-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-3"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_500CellK2to20NbClust.Rds&quot;</span>, projDir, outDirBit, setName, setSuf)</span>
<span id="cb1026-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-4"></a><span class="cf">if</span>(<span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb1026-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-5"></a>{</span>
<span id="cb1026-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-6"></a>    nb &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span>tmpFn)</span>
<span id="cb1026-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-7"></a>} <span class="cf">else</span> {</span>
<span id="cb1026-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-8"></a>    <span class="kw">library</span>(NbClust)</span>
<span id="cb1026-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-9"></a>    <span class="co"># tried with k from 2 to 20 but that takes too long with all cells.</span></span>
<span id="cb1026-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-10"></a>    <span class="co"># subsample 500 cells</span></span>
<span id="cb1026-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-11"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">nrow</span>(pcs), <span class="dv">500</span>)</span>
<span id="cb1026-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-12"></a>        nb =<span class="st"> </span><span class="kw">NbClust</span>(<span class="dt">data=</span>pcs[tmpInd,], </span>
<span id="cb1026-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-13"></a>                 <span class="dt">distance =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb1026-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-14"></a>                 <span class="dt">min.nc =</span> <span class="dv">2</span>,</span>
<span id="cb1026-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-15"></a>              <span class="dt">max.nc =</span> <span class="dv">20</span>,</span>
<span id="cb1026-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-16"></a>                 <span class="co">#min.nc = 8,</span></span>
<span id="cb1026-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-17"></a>                 <span class="co">#max.nc = 16,</span></span>
<span id="cb1026-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-18"></a>                 <span class="dt">method =</span> <span class="st">&quot;kmeans&quot;</span>,</span>
<span id="cb1026-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-19"></a>                 <span class="dt">index=</span><span class="kw">c</span>(<span class="st">&quot;kl&quot;</span>,<span class="st">&quot;ch&quot;</span>,<span class="st">&quot;cindex&quot;</span>,<span class="st">&quot;db&quot;</span>,<span class="st">&quot;silhouette&quot;</span>,</span>
<span id="cb1026-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-20"></a>                    <span class="st">&quot;duda&quot;</span>,<span class="st">&quot;pseudot2&quot;</span>,<span class="st">&quot;beale&quot;</span>,<span class="st">&quot;ratkowsky&quot;</span>,</span>
<span id="cb1026-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-21"></a>                    <span class="st">&quot;gap&quot;</span>,<span class="st">&quot;gamma&quot;</span>,<span class="st">&quot;mcclain&quot;</span>,<span class="st">&quot;gplus&quot;</span>,</span>
<span id="cb1026-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-22"></a>                    <span class="st">&quot;tau&quot;</span>,<span class="st">&quot;sdindex&quot;</span>,<span class="st">&quot;sdbw&quot;</span>))</span>
<span id="cb1026-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-23"></a>    <span class="kw">saveRDS</span>(nb, <span class="dt">file=</span>tmpFn)</span>
<span id="cb1026-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-24"></a>    <span class="kw">rm</span>(tmpInd)</span>
<span id="cb1026-25"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-25"></a>}</span>
<span id="cb1026-26"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1026-26"></a><span class="kw">rm</span>(tmpFn)</span></code></pre></div>
<p>Table showing the number of methods selecting values of k as the best choice:</p>
<div class="sourceCode" id="cb1027"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1027-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1027-1"></a><span class="kw">table</span>(nb<span class="op">$</span>Best.nc[<span class="dv">1</span>,]) <span class="co"># consensus seems to be 3 clusters</span></span></code></pre></div>
<pre><code>## 
##  2  3  7 11 14 20 
##  7  4  1  1  1  2</code></pre>
<div class="sourceCode" id="cb1029"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1029-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1029-1"></a><span class="kw">rm</span>(nb)</span></code></pre></div>
<p>No strong support for a single sensible value of k is observed.</p>
</div>
</div>
<div id="graph-based-clustering" class="section level3">
<h3><span class="header-section-number">14.2.3</span> Graph-based clustering</h3>
<p>Graph-based clustering entails building a nearest-neighbour (NN) graph using cells as nodes and their similarity as edges, then identifying ‘communities’ of cells within the network (more details <a href="https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/clustering-and-cell-annotation.html#unsupervised-clustering-methods">there</a>). In this k-NN graph two nodes (cells), say A and B, are connected by an edge if the distance between them is amongst the k smallest distances from A to other cells (‘KNN’), and from B to other cells (shared-NN, ‘SNN’). An edge may be weighted based on the number or closeness of shared nearest neighbours. Clusters are identified using metrics related to the number of neighbours (‘connections’) to find groups of highly interconnected cells. “The value of ‘k’ can be roughly interpreted as the anticipated size of the smallest subpopulation” (see <code>scran</code>’s <code>buildSNNGraph()</code> manual).</p>
<ul>
<li>Pros:
<ul>
<li>fast and memory efficient (avoids the need to construct a distance matrix for all pairs of cells)</li>
<li>no assumptions on the shape of the clusters or the distribution of cells within each cluster</li>
<li>no need to specify a number of clusters to identify (but the size of the neigbourhood used affects the size of clusters)</li>
</ul></li>
<li>Cons:
<ul>
<li>loss of information beyond neighboring cells, which can affect community detection in regions with many cells.</li>
</ul></li>
</ul>
<p>The plot below shows the same data set as a network built using three different numbers of neighbours: 5, 15 and 25 (from <a href="https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/clustering-and-cell-annotation.html#clustering-methods.">here</a>).</p>
<p><img src="../Images/bioCellGenGraphDeng.png" style="margin:auto; display:block" /></p>
<p>We will:</p>
<ul>
<li>build the graph,</li>
<li>define clusters,</li>
<li>check membership across samples,</li>
<li>show membership on a t-SNE plot,</li>
<li>assess clustering quality.</li>
</ul>
<p>We now build the shared nearest neighbour (SNN) graph, using <code>scran</code>’s <code>buildSNNGraph()</code> with:</p>
<ul>
<li>the reduced and denoised data set (PCs)</li>
<li>the default number of neighbours (k=10)</li>
<li>the default type of edge weight (type=“rank”)</li>
</ul>
<div class="sourceCode" id="cb1030"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1030-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1030-1"></a><span class="co"># compute graph using buildSNNGraph</span></span>
<span id="cb1030-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1030-2"></a><span class="co"># check manual with: ?buildSNNGraph</span></span>
<span id="cb1030-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1030-3"></a><span class="co">#snn.gr &lt;- buildSNNGraph(sce, use.dimred=&quot;PCA&quot;)</span></span>
<span id="cb1030-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1030-4"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(sce, <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>)</span></code></pre></div>
<p>The number of cells in the data set is large so we randomly choose 1000 nodes (cells) in the network before plotting the resulting smaller network. Cells are color-coded by sample type:</p>
<div class="sourceCode" id="cb1031"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1031-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-1"></a><span class="co"># subset graph down to 1000 cells</span></span>
<span id="cb1031-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-2"></a><span class="co"># https://igraph.org/r/doc/subgraph.html</span></span>
<span id="cb1031-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-3"></a></span>
<span id="cb1031-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-4"></a><span class="co"># Add vertices (nodes. ie cells) annotation</span></span>
<span id="cb1031-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-5"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>Sample.Name &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name</span>
<span id="cb1031-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-6"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>source_name &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>source_name)</span>
<span id="cb1031-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-7"></a></span>
<span id="cb1031-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-8"></a><span class="co"># pick 1000 nodes randomly</span></span>
<span id="cb1031-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-9"></a>edgesToGet &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">nrow</span>(snn.gr[]), <span class="dv">1000</span>)</span>
<span id="cb1031-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-10"></a></span>
<span id="cb1031-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-11"></a><span class="co"># subset graph for these 1000 ramdomly chosen nodes</span></span>
<span id="cb1031-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-12"></a>snn.gr.subset &lt;-<span class="st"> </span><span class="kw">subgraph</span>(snn.gr, edgesToGet)</span>
<span id="cb1031-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-13"></a>g &lt;-<span class="st"> </span>snn.gr.subset</span>
<span id="cb1031-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-14"></a></span>
<span id="cb1031-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-15"></a><span class="co"># set colors for clusters</span></span>
<span id="cb1031-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-16"></a><span class="cf">if</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>source_name)) <span class="op">&lt;=</span><span class="st"> </span><span class="dv">2</span>)</span>
<span id="cb1031-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-17"></a>{</span>
<span id="cb1031-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-18"></a>  cols &lt;-<span class="st"> </span>colorspace<span class="op">::</span><span class="kw">rainbow_hcl</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>source_name)))</span>
<span id="cb1031-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-19"></a>} <span class="cf">else</span> {</span>
<span id="cb1031-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-20"></a>  cols &lt;-<span class="st"> </span>RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>source_name)), <span class="dt">name =</span> <span class="st">&quot;RdYlBu&quot;</span>)</span>
<span id="cb1031-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-21"></a>}</span>
<span id="cb1031-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-22"></a><span class="kw">names</span>(cols) &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>source_name)</span>
<span id="cb1031-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-23"></a></span>
<span id="cb1031-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-24"></a><span class="co"># plot graph</span></span>
<span id="cb1031-25"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-25"></a><span class="kw">plot.igraph</span>(</span>
<span id="cb1031-26"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-26"></a>  g, <span class="dt">layout =</span> <span class="kw">layout_with_fr</span>(g),</span>
<span id="cb1031-27"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-27"></a>  <span class="dt">vertex.size =</span> <span class="dv">3</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>,</span>
<span id="cb1031-28"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-28"></a>  <span class="dt">vertex.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>source_name],</span>
<span id="cb1031-29"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-29"></a>  <span class="dt">frame.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>source_name],</span>
<span id="cb1031-30"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-30"></a>  <span class="dt">main =</span> <span class="st">&quot;default parameters&quot;</span></span>
<span id="cb1031-31"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-31"></a>)</span>
<span id="cb1031-32"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-32"></a></span>
<span id="cb1031-33"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-33"></a><span class="co"># add legend</span></span>
<span id="cb1031-34"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-34"></a><span class="kw">legend</span>(<span class="st">&#39;bottomright&#39;</span>,</span>
<span id="cb1031-35"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-35"></a>       <span class="dt">legend=</span><span class="kw">names</span>(cols),</span>
<span id="cb1031-36"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-36"></a>       <span class="dt">pch=</span><span class="dv">21</span>,</span>
<span id="cb1031-37"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-37"></a>       <span class="dt">col=</span>cols, <span class="co"># &quot;#777777&quot;,</span></span>
<span id="cb1031-38"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-38"></a>       <span class="dt">pt.bg=</span>cols,</span>
<span id="cb1031-39"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1031-39"></a>       <span class="dt">pt.cex=</span><span class="dv">1</span>, <span class="dt">cex=</span>.<span class="dv">6</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ncol=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_walktrap_graph-1.png" width="100%" /></p>
<div id="modularity" class="section level4">
<h4><span class="header-section-number">14.2.3.1</span> Modularity</h4>
<p>Several methods to detect clusters (‘communities’) in networks rely on the ‘modulatrity’ metric. For a given partition of cells into clusters, modularity measures how separated clusters are from each other, based on the difference between the observed and expected weight of edges between nodes. For the whole graph, the closer to 1 the better.</p>
</div>
<div id="walktrap-method" class="section level4">
<h4><span class="header-section-number">14.2.3.2</span> Walktrap method</h4>
<!-- #### Community detection via short random walks: 'walktrap' -->
<p>The walktrap method relies on short random walks (a few steps) through the network. These walks tend to be ‘trapped’ in highly-connected regions of the network. Node similarity is measured based on these walks. Nodes are first each assigned their own community. Pairwise distances are computed and the two closest communities are grouped. These steps are repeated a given number of times to produce a dendrogram. Hierarchical clustering is then applied to the distance matrix. The best partition is that with the highest modularity. <!-- https://link.springer.com/article/10.1007/s11227-019-03018-x --></p>
<!--
BIB Pons and Latapy [18] Pons P, Latapy M (2006) Computing communities in large networks using random walks. J Graph Algorithms Appl 10(2):191–218
reliable (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5102890/)
-->
<!--
see pipeComp on performance of walktrap
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5102890/
-->
<div class="sourceCode" id="cb1032"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1032-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1032-1"></a><span class="co"># identify clusters with walktrap</span></span>
<span id="cb1032-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1032-2"></a><span class="co"># default number of steps: 4</span></span>
<span id="cb1032-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1032-3"></a>cluster.out &lt;-<span class="st"> </span><span class="kw">cluster_walktrap</span>(snn.gr)</span></code></pre></div>
<p>We will now count cells in each cluster for each sample type and each sample.</p>
<p>We first retrieve membership.</p>
<div class="sourceCode" id="cb1033"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1033-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1033-1"></a><span class="co"># cluster assignments are stored in the membership slot</span></span>
<span id="cb1033-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1033-2"></a>wt.clusters &lt;-<span class="st"> </span>cluster.out<span class="op">$</span>membership</span></code></pre></div>
<p>Cluster number and sizes (number of cells):</p>
<div class="sourceCode" id="cb1034"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1034-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1034-1"></a><span class="kw">table</span>(wt.clusters)</span></code></pre></div>
<pre><code>## wt.clusters
##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19 
## 709 193 341  63 137  33  66  23 629  56  35  98 127 536 100 191  86  59  18</code></pre>
<p>The table below shows the cluster distribution across sample types. Many clusters are observed in only one sample type.</p>
<div class="sourceCode" id="cb1036"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1036-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1036-1"></a><span class="kw">table</span>(wt.clusters, sce<span class="op">$</span>source_name)</span></code></pre></div>
<pre><code>##            
## wt.clusters ABMMC ETV6-RUNX1 HHD PBMMC PRE-T
##          1      0        543   0   166     0
##          2      0        100   0    93     0
##          3      0        250   0    91     0
##          4      0         27   0    36     0
##          5      0         68   0    69     0
##          6      0          6   0    27     0
##          7      0          0   0    66     0
##          8      0          6   0    17     0
##          9      0        191   0   438     0
##          10     0         26   0    30     0
##          11     0          1   0    34     0
##          12     0          3   0    95     0
##          13     0         58   0    69     0
##          14     0        521   0    15     0
##          15     0         23   0    77     0
##          16     0         75   0   116     0
##          17     0         44   0    42     0
##          18     0         48   0    11     0
##          19     0         10   0     8     0</code></pre>
<p>The table below shows the cluster distribution across samples. Most clusters comprise cells from several replicates of a same sample type, while few are observed in only one sample.</p>
<div class="sourceCode" id="cb1038"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1038-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-1"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(wt.clusters, sce<span class="op">$</span>Sample.Name)</span>
<span id="cb1038-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-2"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;4-step&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1038-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-3"></a></span>
<span id="cb1038-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-4"></a><span class="co"># columns annotation with cell name:</span></span>
<span id="cb1038-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-5"></a>mat_col &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(Sample.Name, source_name) <span class="op">%&gt;%</span><span class="st"> </span>unique</span>
<span id="cb1038-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-6"></a><span class="kw">rownames</span>(mat_col) &lt;-<span class="st"> </span>mat_col<span class="op">$</span>Sample.Name</span>
<span id="cb1038-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-7"></a>mat_col<span class="op">$</span>Sample.Name &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb1038-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-8"></a></span>
<span id="cb1038-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-9"></a><span class="co"># Prepare colours for clusters:</span></span>
<span id="cb1038-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-10"></a>colourCount =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(mat_col<span class="op">$</span>source_name))</span>
<span id="cb1038-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-11"></a>getPalette =<span class="st"> </span>grDevices<span class="op">::</span><span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb1038-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-12"></a></span>
<span id="cb1038-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-13"></a>mat_colors &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">source_name =</span> <span class="kw">getPalette</span>(colourCount))</span>
<span id="cb1038-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-14"></a><span class="kw">names</span>(mat_colors<span class="op">$</span>source_name) &lt;-<span class="st"> </span><span class="kw">unique</span>(mat_col<span class="op">$</span>source_name)</span>
<span id="cb1038-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-15"></a></span>
<span id="cb1038-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-16"></a><span class="kw">pheatmap</span>(tmpTab,</span>
<span id="cb1038-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-17"></a>           <span class="dt">annotation_col    =</span> mat_col,</span>
<span id="cb1038-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1038-18"></a>           <span class="dt">annotation_colors =</span> mat_colors)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/rep_clusters_samples-1.png" width="672" /></p>
<p>We next check the effect of the number of steps on the clusters detected by
increasing it (default number of steps: 4).</p>
<p><em>With 5-step walks:</em></p>
<p>Cluster sizes:</p>
<div class="sourceCode" id="cb1039"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1039-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1039-1"></a>cluster.out.s5 &lt;-<span class="st"> </span><span class="kw">cluster_walktrap</span>(snn.gr, <span class="dt">steps =</span> <span class="dv">5</span>)</span>
<span id="cb1039-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1039-2"></a>wt.clusters.s5 &lt;-<span class="st"> </span>cluster.out.s5<span class="op">$</span>membership</span>
<span id="cb1039-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1039-3"></a><span class="kw">table</span>(wt.clusters.s5)</span></code></pre></div>
<pre><code>## wt.clusters.s5
##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19 
## 338 525  63  80 131 779  23 629  38 140 128  84 194  39  47 100  85  59  18</code></pre>
<p>Comparison with 4-step clusters:</p>
<div class="sourceCode" id="cb1041"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1041-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1041-1"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(wt.clusters, wt.clusters.s5)</span>
<span id="cb1041-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1041-2"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;4-step&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1041-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1041-3"></a><span class="kw">colnames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;5-step&quot;</span>, <span class="kw">colnames</span>(tmpTab) , <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1041-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1041-4"></a><span class="kw">pheatmap</span>(tmpTab)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p><em>With 15-step walks:</em></p>
<p>Cluster sizes:</p>
<div class="sourceCode" id="cb1042"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1042-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1042-1"></a>cluster.out.s15 &lt;-<span class="st"> </span><span class="kw">cluster_walktrap</span>(snn.gr, <span class="dt">steps =</span> <span class="dv">15</span>)</span>
<span id="cb1042-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1042-2"></a>wt.clusters.s15 &lt;-<span class="st"> </span>cluster.out.s15<span class="op">$</span>membership</span>
<span id="cb1042-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1042-3"></a><span class="kw">table</span>(wt.clusters.s15)</span></code></pre></div>
<pre><code>## wt.clusters.s15
##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18 
##  65  23 126  62 664 249 181  37 131 250 187  91 562  39 122  49 603  59</code></pre>
<p>Comparison with 4-step clusters:</p>
<div class="sourceCode" id="cb1044"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1044-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1044-1"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(wt.clusters, wt.clusters.s15)</span>
<span id="cb1044-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1044-2"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;4-step&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1044-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1044-3"></a><span class="kw">colnames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;15-step&quot;</span>, <span class="kw">colnames</span>(tmpTab) , <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1044-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1044-4"></a><span class="kw">pheatmap</span>(tmpTab)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>The network may also be built and clusters identified at once with <code>scran::clusterSNNGraph()</code> that also allows for large data sets to first cluster cells with k-means into a large number of clusters and then perform graph-based clustering on cluster centroids.</p>
<div class="sourceCode" id="cb1045"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1045-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-1"></a><span class="co"># set use.kmeans to TRUE to &#39;create centroids that are then subjected to graph-based clustering&#39; (see manual)</span></span>
<span id="cb1045-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-2"></a><span class="co"># with kmeans.centers default (square root of the number of cells)</span></span>
<span id="cb1045-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-3"></a><span class="co"># or set kmeans.centers to another value, eg 30</span></span>
<span id="cb1045-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-4"></a>clustersOnce &lt;-<span class="st"> </span><span class="kw">clusterSNNGraph</span>(sce,</span>
<span id="cb1045-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-5"></a>                 <span class="co">#use.dimred=&quot;PCA&quot;,</span></span>
<span id="cb1045-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-6"></a>                 <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb1045-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-7"></a>                 <span class="dt">use.kmeans=</span><span class="ot">TRUE</span>,</span>
<span id="cb1045-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-8"></a>                 <span class="dt">kmeans.centers =</span> <span class="dv">200</span>,</span>
<span id="cb1045-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1045-9"></a>                 <span class="dt">full.stats=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<p>Distribution of the size of the 200 k-means clusters:</p>
<div class="sourceCode" id="cb1046"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1046-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1046-1"></a><span class="co"># &#39;kmeans&#39;: assignment of each cell to a k-means cluster</span></span>
<span id="cb1046-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1046-2"></a><span class="co">##table(clustersOnce$kmeans) # no good with many clusters</span></span>
<span id="cb1046-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1046-3"></a><span class="kw">summary</span>(<span class="kw">data.frame</span>(<span class="kw">table</span>(clustersOnce<span class="op">$</span>kmeans))<span class="op">$</span>Freq)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     1.0    11.0    16.0    17.5    23.0    48.0</code></pre>
<p>Graph-based cluster sizes in cell number:</p>
<div class="sourceCode" id="cb1048"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1048-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1048-1"></a><span class="co"># &#39;igraph&#39;: assignment of each cell to a graph-based cluster operating on the k-means clusters</span></span>
<span id="cb1048-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1048-2"></a><span class="kw">table</span>(clustersOnce<span class="op">$</span>igraph)</span></code></pre></div>
<pre><code>## 
##    1    2    3    4    5    6 
##  733 1387  479  364  181  356</code></pre>
<p>The heatmap below displays the contingency table for the graph-based clusters (rows) and the k-means clusters (columns) used to build the network (color-code by size, i.e. number of cells).</p>
<div class="sourceCode" id="cb1050"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1050-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-1"></a><span class="co">#table(clustersOnce$igraph,</span></span>
<span id="cb1050-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-2"></a><span class="co">#   clustersOnce$kmeans)</span></span>
<span id="cb1050-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-3"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(clustersOnce<span class="op">$</span>igraph,</span>
<span id="cb1050-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-4"></a>        clustersOnce<span class="op">$</span>kmeans)</span>
<span id="cb1050-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-5"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;igraph&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1050-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-6"></a><span class="kw">colnames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;kmeans&quot;</span>, <span class="kw">colnames</span>(tmpTab) , <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1050-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1050-7"></a><span class="kw">pheatmap</span>(tmpTab, <span class="dt">show_colnames =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Graph-based cluster sizes in k-mean cluster number:</p>
<div class="sourceCode" id="cb1051"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1051-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1051-1"></a><span class="co"># ‘membership’, the graph-based cluster to which each node is assigned</span></span>
<span id="cb1051-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1051-2"></a><span class="kw">table</span>(<span class="kw">metadata</span>(clustersOnce)<span class="op">$</span>membership)</span></code></pre></div>
<pre><code>## 
##  1  2  3  4  5  6 
## 52 69 28 22 13 16</code></pre>
<p>The plot below show the network build with the 200 k-means clusters used the
identify the graph-based clusters.</p>
<div class="sourceCode" id="cb1053"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1053-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-1"></a>g &lt;-<span class="st"> </span><span class="kw">metadata</span>(clustersOnce)<span class="op">$</span>graph</span>
<span id="cb1053-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-2"></a><span class="kw">V</span>(g)<span class="op">$</span>membership &lt;-<span class="st"> </span><span class="kw">metadata</span>(clustersOnce)<span class="op">$</span>membership</span>
<span id="cb1053-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-3"></a></span>
<span id="cb1053-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-4"></a><span class="co"># set colors for clusters</span></span>
<span id="cb1053-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-5"></a>rgb.palette &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;purple&quot;</span>,<span class="st">&quot;yellow&quot;</span>), <span class="dt">space=</span><span class="st">&quot;rgb&quot;</span>) <span class="co"># Seurat-like</span></span>
<span id="cb1053-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-6"></a>cols &lt;-<span class="st"> </span><span class="kw">rgb.palette</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>membership)))</span>
<span id="cb1053-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-7"></a><span class="kw">names</span>(cols) &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>membership)))</span>
<span id="cb1053-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-8"></a></span>
<span id="cb1053-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-9"></a><span class="co"># plot graph</span></span>
<span id="cb1053-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-10"></a><span class="kw">plot.igraph</span>(</span>
<span id="cb1053-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-11"></a>  g, <span class="dt">layout =</span> <span class="kw">layout_with_fr</span>(g),</span>
<span id="cb1053-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-12"></a>  <span class="dt">vertex.size =</span> <span class="dv">3</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>,</span>
<span id="cb1053-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-13"></a>  <span class="dt">vertex.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>membership],</span>
<span id="cb1053-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-14"></a>  <span class="dt">frame.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>membership],</span>
<span id="cb1053-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-15"></a>  <span class="dt">main =</span> <span class="st">&quot;with 200 k-means clusters&quot;</span></span>
<span id="cb1053-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-16"></a>)</span>
<span id="cb1053-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-17"></a></span>
<span id="cb1053-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-18"></a><span class="co"># add legend</span></span>
<span id="cb1053-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-19"></a><span class="kw">legend</span>(<span class="st">&#39;bottomright&#39;</span>,</span>
<span id="cb1053-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-20"></a>       <span class="dt">legend=</span><span class="kw">names</span>(cols),</span>
<span id="cb1053-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-21"></a>       <span class="dt">pch=</span><span class="dv">21</span>,</span>
<span id="cb1053-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-22"></a>       <span class="dt">col=</span>cols, <span class="co"># &quot;#777777&quot;,</span></span>
<span id="cb1053-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-23"></a>       <span class="dt">pt.bg=</span>cols,</span>
<span id="cb1053-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1053-24"></a>       <span class="dt">pt.cex=</span><span class="dv">1</span>, <span class="dt">cex=</span>.<span class="dv">6</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-24-1.png" width="100%" /></p>
</div>
<div id="louvain-method" class="section level4">
<h4><span class="header-section-number">14.2.3.3</span> Louvain method</h4>
<!-- #### Community detection with the 'Louvain' method -->
<p>With the Louvain method, nodes are also first assigned their own community. This hierarchical agglomerative method then progresses in two-step iterations: 1) nodes are re-assigned one at a time to the community for which they increase modularity the most, if any, 2) a new, ‘aggregate’ network is built where nodes are the communities formed in the previous step. This is repeated until modularity stops increasing. The diagram below is copied from <a href="https://www.nature.com/articles/s41598-019-41695-z#Fig1">this article</a>.</p>
<p><img src="../Images/leiden_Fig1_HTML.png" style="margin:auto; display:block" /></p>
<!-- BIBREF https://www.nature.com/articles/s41598-019-41695-z#Fig1 -->
<!--
multilevel community method
BIBREF https://arxiv.org/abs/1810.08473
-->
<p>We now apply the Louvain approach, store its outcome in the SCE object and show cluster sizes.</p>
<div class="sourceCode" id="cb1054"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1054-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-1"></a>ig.louvain &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_louvain</span>(snn.gr)</span>
<span id="cb1054-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-2"></a>cl &lt;-<span class="st"> </span>ig.louvain<span class="op">$</span>membership</span>
<span id="cb1054-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-3"></a>cl &lt;-<span class="st"> </span><span class="kw">factor</span>(cl)</span>
<span id="cb1054-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-4"></a><span class="co"># store membership</span></span>
<span id="cb1054-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-5"></a>sce<span class="op">$</span>louvain &lt;-<span class="st"> </span>cl</span>
<span id="cb1054-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-6"></a><span class="co"># show cluster sizes:</span></span>
<span id="cb1054-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1054-7"></a><span class="kw">table</span>(sce<span class="op">$</span>louvain)</span></code></pre></div>
<pre><code>## 
##   1   2   3   4   5   6   7   8   9  10 
## 653 194  60 605 272 552 519 152 124 369</code></pre>
<p>The plot below displays the network with nodes (the 1000 randomly chosen cells) color-coded by cluster membership:</p>
<div class="sourceCode" id="cb1056"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1056-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-1"></a><span class="co">## Define a set of colors to use (must be at least as many as the number of</span></span>
<span id="cb1056-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-2"></a><span class="co">## communities)</span></span>
<span id="cb1056-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-3"></a><span class="co">##cols &lt;- RColorBrewer::brewer.pal(n = nlevels(sce$cluster), name = &quot;RdYlBu&quot;)</span></span>
<span id="cb1056-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-4"></a><span class="co">##cols &lt;- grDevices:::colorRampPalette(RColorBrewer::brewer.pal(nlevels(sce$cluster), &quot;RdYlBu&quot;))</span></span>
<span id="cb1056-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-5"></a></span>
<span id="cb1056-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-6"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>Sample.Name &lt;-<span class="st"> </span><span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name</span>
<span id="cb1056-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-7"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>source_name &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>source_name)</span>
<span id="cb1056-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-8"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>louvain &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>louvain)</span>
<span id="cb1056-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-9"></a></span>
<span id="cb1056-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-10"></a><span class="co"># once only # edgesToGet &lt;- sample(nrow(snn.gr[]), 1000)</span></span>
<span id="cb1056-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-11"></a></span>
<span id="cb1056-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-12"></a>snn.gr.subset &lt;-<span class="st"> </span><span class="kw">subgraph</span>(snn.gr, edgesToGet)</span>
<span id="cb1056-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-13"></a>g &lt;-<span class="st"> </span>snn.gr.subset</span>
<span id="cb1056-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-14"></a></span>
<span id="cb1056-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-15"></a>tmpLayout &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g)</span>
<span id="cb1056-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-16"></a></span>
<span id="cb1056-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-17"></a>rgb.palette &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;purple&quot;</span>,<span class="st">&quot;yellow&quot;</span>), <span class="dt">space=</span><span class="st">&quot;rgb&quot;</span>) <span class="co"># Seurat-like</span></span>
<span id="cb1056-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-18"></a><span class="co">#rgb.palette &lt;- colorRampPalette(c(&quot;yellow2&quot;,&quot;goldenrod&quot;,&quot;darkred&quot;), space=&quot;rgb&quot;)</span></span>
<span id="cb1056-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-19"></a><span class="co">##rgb.palette &lt;- colorRampPalette(c(&quot;red&quot;,&quot;orange&quot;,&quot;blue&quot;), space=&quot;rgb&quot;)</span></span>
<span id="cb1056-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-20"></a>cols &lt;-<span class="st"> </span><span class="kw">rgb.palette</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>louvain)))</span>
<span id="cb1056-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-21"></a><span class="kw">names</span>(cols) &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>louvain)))</span>
<span id="cb1056-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-22"></a></span>
<span id="cb1056-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-23"></a><span class="co"># Plot the graph, color by cluster assignment</span></span>
<span id="cb1056-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-24"></a>igraph<span class="op">::</span><span class="kw">plot.igraph</span>(</span>
<span id="cb1056-25"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-25"></a>  g, <span class="dt">layout =</span> tmpLayout,</span>
<span id="cb1056-26"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-26"></a>  <span class="dt">vertex.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>louvain],</span>
<span id="cb1056-27"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-27"></a>  <span class="dt">frame.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>louvain],</span>
<span id="cb1056-28"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-28"></a>  <span class="dt">vertex.size =</span> <span class="dv">3</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">main =</span> <span class="st">&quot;Louvain&quot;</span></span>
<span id="cb1056-29"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-29"></a>)</span>
<span id="cb1056-30"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-30"></a></span>
<span id="cb1056-31"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-31"></a><span class="co"># add legend</span></span>
<span id="cb1056-32"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-32"></a><span class="kw">legend</span>(<span class="st">&#39;bottomright&#39;</span>,</span>
<span id="cb1056-33"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-33"></a>       <span class="dt">legend=</span><span class="kw">names</span>(cols),</span>
<span id="cb1056-34"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-34"></a>       <span class="dt">pch=</span><span class="dv">21</span>,</span>
<span id="cb1056-35"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-35"></a>       <span class="dt">col=</span>cols, <span class="co"># &quot;#777777&quot;,</span></span>
<span id="cb1056-36"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-36"></a>       <span class="dt">pt.bg=</span>cols,</span>
<span id="cb1056-37"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1056-37"></a>       <span class="dt">pt.cex=</span><span class="dv">1</span>, <span class="dt">cex=</span>.<span class="dv">6</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_louvain_graph-1.png" width="100%" /></p>
<p>The t-SNE plot shows cells color-coded by cluster membership:</p>
<div class="sourceCode" id="cb1057"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1057-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1057-1"></a><span class="co"># show clusters on TSNE</span></span>
<span id="cb1057-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1057-2"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;louvain&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1057-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1057-3"></a>p</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_louvain_tsne_1-1.png" width="768" /></p>
<p>Split by sample type:</p>
<div class="sourceCode" id="cb1058"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1058-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1058-1"></a><span class="co"># facet by sample group:</span></span>
<span id="cb1058-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1058-2"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_louvain_tsne_2-1.png" width="960" /></p>
</div>
<div id="leiden-method" class="section level4">
<h4><span class="header-section-number">14.2.3.4</span> Leiden method</h4>
<!-- #### Community detection with the 'Leiden' method -->
<!-- (igraph_community_leiden is only for the C lib, not the R package) -->
<p>The Leiden method improves on the Louvain method by garanteeing that at each iteration clusters are connected and well-separated. The method includes an extra step in the iterations: after nodes are moved (step 1), the resulting partition is refined (step2) and only then the new aggregate network made, and refined (step 3). The diagram below is copied from <a href="https://www.nature.com/articles/s41598-019-41695-z#Fig3">this article</a>.</p>
<p><img src="../Images/leiden_Fig3_HTML.png" style="margin:auto; display:block" /></p>
<!-- BIBREF https://www.nature.com/articles/s41598-019-41695-z#Fig3 -->
<div class="sourceCode" id="cb1059"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1059-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-1"></a><span class="co"># have logical to check we can use the required python modules:</span></span>
<span id="cb1059-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-2"></a>leidenOk &lt;-<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb1059-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-3"></a><span class="co"># use the reticulate R package to use python modules</span></span>
<span id="cb1059-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-4"></a><span class="kw">library</span>(reticulate)</span>
<span id="cb1059-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-5"></a><span class="co"># set path to python</span></span>
<span id="cb1059-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-6"></a><span class="kw">use_python</span>(<span class="st">&quot;/home/baller01/.local/share/r-miniconda/envs/r-reticulate/bin/python&quot;</span>)</span>
<span id="cb1059-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-7"></a><span class="co"># check availability of python modules and set logical accordingly</span></span>
<span id="cb1059-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-8"></a><span class="cf">if</span>(reticulate<span class="op">::</span><span class="kw">py_module_available</span>(<span class="st">&quot;igraph&quot;</span>)</span>
<span id="cb1059-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-9"></a><span class="op">&amp;</span><span class="st"> </span>reticulate<span class="op">::</span><span class="kw">py_module_available</span>(<span class="st">&quot;leidenalg&quot;</span>))</span>
<span id="cb1059-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-10"></a>{</span>
<span id="cb1059-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-11"></a>    leidenOk &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb1059-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1059-12"></a>}</span></code></pre></div>
<p>We now apply the Leiden approach, store its outcome in the SCE object and show cluster sizes.</p>
<div class="sourceCode" id="cb1060"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1060-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-1"></a><span class="co"># mind eval above is set to the logical set above</span></span>
<span id="cb1060-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-2"></a><span class="co"># so that chunk will only run if python modules were found.</span></span>
<span id="cb1060-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-3"></a><span class="kw">library</span>(<span class="st">&quot;leiden&quot;</span>)</span>
<span id="cb1060-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-4"></a>adjacency_matrix &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">as_adjacency_matrix</span>(snn.gr)</span>
<span id="cb1060-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-5"></a>partition &lt;-<span class="st"> </span><span class="kw">leiden</span>(adjacency_matrix)</span>
<span id="cb1060-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-6"></a><span class="co"># store membership</span></span>
<span id="cb1060-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-7"></a>sce<span class="op">$</span>leiden &lt;-<span class="st"> </span><span class="kw">factor</span>(partition)</span>
<span id="cb1060-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-8"></a><span class="co"># show cluster sizes:</span></span>
<span id="cb1060-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1060-9"></a><span class="kw">table</span>(sce<span class="op">$</span>leiden)</span></code></pre></div>
<pre><code>## 
##   1   2   3   4   5   6   7   8   9  10 
## 643 639 604 514 401 297 197 124  59  22</code></pre>
<p>The plot below displays the network with nodes (the 1000 randomly chosen cells) color-coded by cluster membership:</p>
<div class="sourceCode" id="cb1062"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1062-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-1"></a><span class="co">## Define a set of colors to use (must be at least as many as the number of</span></span>
<span id="cb1062-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-2"></a><span class="co">## communities)</span></span>
<span id="cb1062-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-3"></a><span class="co">##cols &lt;- RColorBrewer::brewer.pal(n = nlevels(sce$cluster), name = &quot;RdYlBu&quot;)</span></span>
<span id="cb1062-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-4"></a><span class="co">##cols &lt;- grDevices:::colorRampPalette(RColorBrewer::brewer.pal(nlevels(sce$cluster), &quot;RdYlBu&quot;))</span></span>
<span id="cb1062-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-5"></a></span>
<span id="cb1062-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-6"></a><span class="co">##V(snn.gr)$Sample.Name &lt;- colData(sce)$Sample.Name</span></span>
<span id="cb1062-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-7"></a><span class="co">##V(snn.gr)$source_name &lt;- as.character(colData(sce)$source_name)</span></span>
<span id="cb1062-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-8"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>leiden &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>leiden)</span>
<span id="cb1062-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-9"></a></span>
<span id="cb1062-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-10"></a><span class="co"># once only # edgesToGet &lt;- sample(nrow(snn.gr[]), 1000)</span></span>
<span id="cb1062-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-11"></a></span>
<span id="cb1062-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-12"></a>snn.gr.subset &lt;-<span class="st"> </span><span class="kw">subgraph</span>(snn.gr, edgesToGet)</span>
<span id="cb1062-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-13"></a>g &lt;-<span class="st"> </span>snn.gr.subset</span>
<span id="cb1062-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-14"></a></span>
<span id="cb1062-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-15"></a><span class="co">#cols &lt;- RColorBrewer::brewer.pal(n = length(unique(V(g)$leiden)), name = &quot;RdYlBu&quot;) # 11 max</span></span>
<span id="cb1062-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-16"></a><span class="co">#cols &lt;- RColorBrewer::brewer.pal(n = length(unique(V(g)$leiden)), name = &quot;Spectral&quot;) # 11 max</span></span>
<span id="cb1062-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-17"></a><span class="co">#names(cols) &lt;- unique(V(g)$leiden)</span></span>
<span id="cb1062-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-18"></a></span>
<span id="cb1062-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-19"></a>rgb.palette &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;purple&quot;</span>,<span class="st">&quot;yellow&quot;</span>), <span class="dt">space=</span><span class="st">&quot;rgb&quot;</span>) <span class="co"># Seurat-like</span></span>
<span id="cb1062-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-20"></a><span class="co">#rgb.palette &lt;- colorRampPalette(c(&quot;yellow2&quot;,&quot;goldenrod&quot;,&quot;darkred&quot;), space=&quot;rgb&quot;)</span></span>
<span id="cb1062-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-21"></a><span class="co">##rgb.palette &lt;- colorRampPalette(c(&quot;red&quot;,&quot;orange&quot;,&quot;blue&quot;), space=&quot;rgb&quot;)</span></span>
<span id="cb1062-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-22"></a>cols &lt;-<span class="st"> </span><span class="kw">rgb.palette</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>leiden)))</span>
<span id="cb1062-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-23"></a><span class="kw">names</span>(cols) &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>leiden)))</span>
<span id="cb1062-24"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-24"></a></span>
<span id="cb1062-25"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-25"></a><span class="co">## Plot the graph, color by cluster assignment</span></span>
<span id="cb1062-26"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-26"></a>igraph<span class="op">::</span><span class="kw">plot.igraph</span>(</span>
<span id="cb1062-27"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-27"></a>  g, <span class="dt">layout =</span> tmpLayout,</span>
<span id="cb1062-28"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-28"></a>  <span class="dt">vertex.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>leiden],</span>
<span id="cb1062-29"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-29"></a>  <span class="dt">frame.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>leiden],</span>
<span id="cb1062-30"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-30"></a>  <span class="dt">vertex.size =</span> <span class="dv">3</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">main =</span> <span class="st">&quot;Leiden&quot;</span></span>
<span id="cb1062-31"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-31"></a>)</span>
<span id="cb1062-32"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-32"></a><span class="kw">legend</span>(<span class="st">&#39;bottomright&#39;</span>,</span>
<span id="cb1062-33"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-33"></a>       <span class="dt">legend=</span><span class="kw">names</span>(cols),</span>
<span id="cb1062-34"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-34"></a>       <span class="dt">pch=</span><span class="dv">21</span>,</span>
<span id="cb1062-35"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-35"></a>       <span class="dt">col=</span>cols, <span class="co"># &quot;#777777&quot;,</span></span>
<span id="cb1062-36"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-36"></a>       <span class="dt">pt.bg=</span>cols,</span>
<span id="cb1062-37"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1062-37"></a>       <span class="dt">pt.cex=</span><span class="dv">1</span>, <span class="dt">cex=</span>.<span class="dv">6</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_leiden_graph-1.png" width="100%" />
The t-SNE plot below shows cells color-coded by cluster membership:</p>
<div class="sourceCode" id="cb1063"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1063-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1063-1"></a><span class="co"># show clusters on TSNE</span></span>
<span id="cb1063-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1063-2"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;leiden&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb1063-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1063-3"></a>p</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_leiden_tsne_1-1.png" width="768" /></p>
<p>Split by sample type:</p>
<div class="sourceCode" id="cb1064"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1064-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1064-1"></a><span class="co"># facet by sample group:</span></span>
<span id="cb1064-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1064-2"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_leiden_tsne_2-1.png" width="960" /></p>
</div>
<div id="modularity-to-assess-clusters-quality" class="section level4">
<h4><span class="header-section-number">14.2.3.5</span> Modularity to assess clusters quality</h4>
<p>We now compute modularity to assess clusters quality (the division of cells into several groups, ‘clusters’, is compared to that in an equivalent random network with no structure).</p>
<p>The overall modularity obtained with the Louvain method is 0.72.</p>
<p>Modularity can also be computed for each cluster, with <code>scran</code>’s <code>clusterModularity()</code> (now in the <a href="http://bioconductor.org/packages/devel/bioc/html/bluster.html">bluster package</a>). This function returns an upper diagonal matrix with values for each pair of clusters.</p>
<div class="sourceCode" id="cb1065"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1065-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-1"></a><span class="co"># compute cluster-wise modularities,</span></span>
<span id="cb1065-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-2"></a><span class="co"># with clusterModularity(),</span></span>
<span id="cb1065-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-3"></a><span class="co"># which returns a matrix keeping values for each pair of clusters</span></span>
<span id="cb1065-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-4"></a><span class="co">##mod.out &lt;- clusterModularity(snn.gr, wt.clusters, get.weights=TRUE)</span></span>
<span id="cb1065-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-5"></a>mod.out &lt;-<span class="st"> </span>bluster<span class="op">::</span><span class="kw">pairwiseModularity</span>(snn.gr,</span>
<span id="cb1065-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-6"></a>                                       <span class="co">#wt.clusters,</span></span>
<span id="cb1065-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-7"></a>                                       sce<span class="op">$</span>louvain,</span>
<span id="cb1065-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-8"></a>                                       <span class="dt">get.weights=</span><span class="ot">TRUE</span>)</span>
<span id="cb1065-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-9"></a><span class="co"># Modularity is proportional to the cluster size,</span></span>
<span id="cb1065-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-10"></a><span class="co"># so we compute the ratio of the observed to expected weights</span></span>
<span id="cb1065-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-11"></a><span class="co"># for each cluster pair</span></span>
<span id="cb1065-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-12"></a>ratio &lt;-<span class="st"> </span>mod.out<span class="op">$</span>observed<span class="op">/</span>mod.out<span class="op">$</span>expected</span>
<span id="cb1065-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1065-13"></a>lratio &lt;-<span class="st"> </span><span class="kw">log10</span>(ratio <span class="op">+</span><span class="st"> </span><span class="dv">1</span>) <span class="co"># on log scale to improve colour range</span></span></code></pre></div>
<p>We display below the cluster-wise modularity on a heatmap. Clusters that are well separated mostly comprise intra-cluster edges and harbour a high modularity score on the diagonal and low scores off that diagonal. Two poorly separated clusters will share edges and the pair will have a high score.</p>
<div class="sourceCode" id="cb1066"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1066-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1066-1"></a><span class="kw">pheatmap</span>(lratio, <span class="dt">cluster_rows=</span><span class="ot">FALSE</span>, <span class="dt">cluster_cols=</span><span class="ot">FALSE</span>, </span>
<span id="cb1066-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1066-2"></a>    <span class="dt">color=</span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;blue&quot;</span>))(<span class="dv">100</span>))</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/show_clusterModularity_snn-1.png" width="672" /></p>
<p>The cluster-wise modularity matrix may also be shown as a network, with clusters as nodes and edges weighted by modularity:</p>
<div class="sourceCode" id="cb1067"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1067-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-1"></a>cluster.gr &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">graph_from_adjacency_matrix</span>(</span>
<span id="cb1067-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-2"></a>    ratio, </span>
<span id="cb1067-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-3"></a>    <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>,</span>
<span id="cb1067-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-4"></a>    <span class="dt">weighted=</span><span class="ot">TRUE</span>,</span>
<span id="cb1067-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-5"></a>    <span class="dt">diag=</span><span class="ot">FALSE</span>)</span>
<span id="cb1067-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-6"></a><span class="kw">plot</span>(cluster.gr,</span>
<span id="cb1067-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-7"></a>     <span class="dt">label.cex =</span> <span class="fl">0.2</span>,</span>
<span id="cb1067-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1067-8"></a>     <span class="dt">edge.width =</span> igraph<span class="op">::</span><span class="kw">E</span>(cluster.gr)<span class="op">$</span>weight)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/plot_clusterNetwork_snn-1.png" width="100%" /></p>
</div>
<div id="choose-the-number-of-clusters" class="section level4">
<h4><span class="header-section-number">14.2.3.6</span> Choose the number of clusters</h4>
<p>Some community detection methods are hierarchical, e.g. walktrap. With these methods, one can choose the number of clusters.</p>
<p>We can now retrieve a given number of clusters, e.g. 10 here to match the k-means clusters defined above:</p>
<div class="sourceCode" id="cb1068"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1068-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1068-1"></a><span class="co"># remember cluster.out keeps the walktrap outcome</span></span>
<span id="cb1068-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1068-2"></a>cluster.out<span class="fl">.10</span> &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cut_at</span>(cluster.out, <span class="dt">no =</span> <span class="dv">10</span>)</span>
<span id="cb1068-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1068-3"></a>cluster.out<span class="fl">.10</span> &lt;-<span class="st"> </span><span class="kw">factor</span>(cluster.out<span class="fl">.10</span>)</span>
<span id="cb1068-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1068-4"></a><span class="kw">table</span>(cluster.out<span class="fl">.10</span>)</span></code></pre></div>
<pre><code>## cluster.out.10
##    1    2    3    4    5    6    7    8    9   10 
##  269  182   96 1497  478   23  629   35  100  191</code></pre>
<p>The plot below displays the network with nodes (the 1000 randomly chosen cells) color-coded by cluster membership, split by sample type:</p>
<div class="sourceCode" id="cb1070"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1070-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-1"></a><span class="co"># annotate nodes</span></span>
<span id="cb1070-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-2"></a><span class="kw">V</span>(snn.gr)<span class="op">$</span>walkCutAt10 &lt;-<span class="st"> </span><span class="kw">as.character</span>(cluster.out<span class="fl">.10</span>)</span>
<span id="cb1070-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-3"></a></span>
<span id="cb1070-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-4"></a><span class="co"># subset graph</span></span>
<span id="cb1070-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-5"></a>snn.gr.subset &lt;-<span class="st"> </span><span class="kw">subgraph</span>(snn.gr, edgesToGet)</span>
<span id="cb1070-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-6"></a>g &lt;-<span class="st"> </span>snn.gr.subset</span>
<span id="cb1070-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-7"></a></span>
<span id="cb1070-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-8"></a><span class="co"># make colors:</span></span>
<span id="cb1070-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-9"></a>cols &lt;-<span class="st"> </span><span class="kw">rgb.palette</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>walkCutAt10)))</span>
<span id="cb1070-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-10"></a><span class="kw">names</span>(cols) &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">V</span>(g)<span class="op">$</span>walkCutAt10)))</span>
<span id="cb1070-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-11"></a></span>
<span id="cb1070-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-12"></a><span class="co"># Plot the graph, color by cluster assignment</span></span>
<span id="cb1070-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-13"></a>igraph<span class="op">::</span><span class="kw">plot.igraph</span>(</span>
<span id="cb1070-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-14"></a>  g, <span class="dt">layout =</span> tmpLayout,</span>
<span id="cb1070-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-15"></a>  <span class="dt">vertex.color =</span> cols[<span class="kw">V</span>(g)<span class="op">$</span>walkCutAt10],</span>
<span id="cb1070-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-16"></a>  <span class="dt">vertex.size =</span> <span class="dv">3</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">main =</span> <span class="st">&quot;walkCutAt10&quot;</span></span>
<span id="cb1070-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-17"></a>)</span>
<span id="cb1070-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-18"></a><span class="kw">legend</span>(<span class="st">&#39;bottomright&#39;</span>,</span>
<span id="cb1070-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-19"></a>      <span class="dt">legend=</span><span class="kw">names</span>(cols),</span>
<span id="cb1070-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-20"></a>       <span class="dt">pch=</span><span class="dv">21</span>,</span>
<span id="cb1070-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-21"></a>       <span class="dt">col=</span><span class="st">&quot;#777777&quot;</span>,</span>
<span id="cb1070-22"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-22"></a>       <span class="dt">pt.bg=</span>cols,</span>
<span id="cb1070-23"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1070-23"></a>       <span class="dt">pt.cex=</span><span class="dv">2</span>, <span class="dt">cex=</span>.<span class="dv">8</span>, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-27-1.png" width="100%" /></p>
<p>The t-SNE plot below shows cells color-coded by cluster membership split by
sample type:</p>
<div class="sourceCode" id="cb1071"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1071-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-1"></a> <span class="co"># ```{r, out.width = &#39;100%&#39;}</span></span>
<span id="cb1071-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-2"></a>sce<span class="op">$</span>cluster.out<span class="fl">.10</span> &lt;-<span class="st"> </span>cluster.out<span class="fl">.10</span></span>
<span id="cb1071-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-3"></a>sce<span class="op">$</span>cluster.out.<span class="fl">10.</span>col &lt;-<span class="st"> </span>cols[cluster.out<span class="fl">.10</span>]</span>
<span id="cb1071-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-4"></a></span>
<span id="cb1071-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-5"></a><span class="co"># show clusters on TSNE</span></span>
<span id="cb1071-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-6"></a><span class="co">#p &lt;- plotTSNE(sce, colour_by=&quot;cluster.out.10&quot;) + fontsize</span></span>
<span id="cb1071-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-7"></a>tSneCoord &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">reducedDim</span>(sce, <span class="st">&quot;TSNE&quot;</span>))</span>
<span id="cb1071-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-8"></a><span class="kw">colnames</span>(tSneCoord) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TSNE1&quot;</span>, <span class="st">&quot;TSNE2&quot;</span>)</span>
<span id="cb1071-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-9"></a><span class="co"># add sample type:</span></span>
<span id="cb1071-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-10"></a>tSneCoord<span class="op">$</span>source_name &lt;-<span class="st"> </span>sce<span class="op">$</span>source_name</span>
<span id="cb1071-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-11"></a><span class="co"># add cluster info:</span></span>
<span id="cb1071-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-12"></a>tSneCoord<span class="op">$</span>cluster.out.<span class="fl">10.</span>col &lt;-<span class="st"> </span>sce<span class="op">$</span>cluster.out.<span class="fl">10.</span>col</span>
<span id="cb1071-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-13"></a><span class="co"># draw plot</span></span>
<span id="cb1071-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-14"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tSneCoord, <span class="kw">aes</span>(<span class="dt">x=</span>TSNE1, <span class="dt">y=</span>TSNE2,</span>
<span id="cb1071-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-15"></a>                <span class="dt">color=</span>cluster.out<span class="fl">.10</span>,</span>
<span id="cb1071-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-16"></a>                <span class="dt">shape=</span>source_name)) <span class="op">+</span></span>
<span id="cb1071-17"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-17"></a><span class="st">    </span><span class="kw">geom_point</span>()</span>
<span id="cb1071-18"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-18"></a>p2 &lt;-<span class="st"> </span>p2 <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> cols)</span>
<span id="cb1071-19"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-19"></a><span class="co"># facet by sample group:</span></span>
<span id="cb1071-20"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-20"></a>p2 <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>sce<span class="op">$</span>source_name) <span class="op">+</span></span>
<span id="cb1071-21"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1071-21"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="kw">rel</span>(<span class="fl">0.7</span>)))</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-28-1.png" width="960" /></p>
</div>
</div>
</div>
<div id="comparing-two-sets-of-clusters" class="section level2">
<h2><span class="header-section-number">14.3</span> Comparing two sets of clusters</h2>
<p>The contingency table below shows the concordance between the walk-trap (rows) and k-means (columns) clustering methods:</p>
<div class="sourceCode" id="cb1072"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1072-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1072-1"></a><span class="co"># rows: walktrap</span></span>
<span id="cb1072-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1072-2"></a><span class="co"># columns: kmeans</span></span>
<span id="cb1072-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1072-3"></a><span class="co">##tmpTab &lt;- table(cluster.out$membership, sce$kmeans10)</span></span>
<span id="cb1072-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1072-4"></a>tmpTab &lt;-<span class="st"> </span><span class="kw">table</span>(sce<span class="op">$</span>cluster.out<span class="fl">.10</span>, sce<span class="op">$</span>kmeans10)</span>
<span id="cb1072-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1072-5"></a>tmpTab</span></code></pre></div>
<pre><code>##     
##        1   2   3   4   5   6   7   8   9  10
##   1    0   0   0   0   0   0   0 252  17   0
##   2    0   0   0   1  19   1 161   0   0   0
##   3    0   0   0   0   0   0   0   0  96   0
##   4   14 748   2   5 184  10   0   0   0 534
##   5    0  11   0   0  49 418   0   0   0   0
##   6   19   0   0   0   2   0   0   0   1   1
##   7    0   0 610  19   0   0   0   0   0   0
##   8    0   0   0   0  23   0   0   0  12   0
##   9    0   0   0 100   0   0   0   0   0   0
##   10 191   0   0   0   0   0   0   0   0   0</code></pre>
<p>Concordance is moderate, with some clusters defined with one method comprising cells assigned to several clusters defined with the other method.</p>
<p>Concordance can also be displayed on a heatmap:</p>
<div class="sourceCode" id="cb1074"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1074-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1074-1"></a><span class="kw">rownames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;walk&quot;</span>, <span class="kw">rownames</span>(tmpTab), <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1074-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1074-2"></a><span class="kw">colnames</span>(tmpTab) =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;kmeans10&quot;</span>, <span class="kw">colnames</span>(tmpTab) , <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1074-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1074-3"></a><span class="kw">pheatmap</span>(tmpTab)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<p>Concordance can also be measured with the adjusted Rand index that ranges between 0 and 1 from random partitions to perfect agreement. Here, the adjusted Rand index is 0.600216.</p>
</div>
<div id="expression-of-known-marker-genes" class="section level2">
<h2><span class="header-section-number">14.4</span> Expression of known marker genes</h2>
<p>We first show clusters on a t-SNE plot:</p>
<div class="sourceCode" id="cb1075"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1075-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1075-1"></a>cluToUse &lt;-<span class="st"> &quot;cluster.out.10&quot;</span></span>
<span id="cb1075-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1075-2"></a><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by =</span> cluToUse) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;walktrap, 10 clusters&quot;</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-31-1.png" width="768" /></p>
<p>Having identified clusters, we now display the level of expression of cell type marker genes to quickly match clusters with cell types (e.g. MS4A1 for B cell marker gene). For each marker we will plot its expression on a t-SNE, and show distribution across each cluster on a violin plot.</p>
<p>Example with B cell marker genes MS4A1 and CD79A:</p>
<div class="sourceCode" id="cb1076"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1076-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-1"></a><span class="kw">rownames</span>(sce) &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw">uniquifyFeatureNames</span>(</span>
<span id="cb1076-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-2"></a>  <span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id,</span>
<span id="cb1076-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-3"></a>  <span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol)</span>
<span id="cb1076-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-4"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;MS4A1&quot;</span>) <span class="op">+</span></span>
<span id="cb1076-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-5"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;B cells: MS4A1&quot;</span>)</span>
<span id="cb1076-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-6"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(sce, <span class="dt">by_exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;CD79A&quot;</span>) <span class="op">+</span></span>
<span id="cb1076-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-7"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;B cells: CD79A&quot;</span>)</span>
<span id="cb1076-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1076-8"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/exprKnownMarkers_plot1_clusteringPostDsi-1.png" width="960" /></p>
<div class="sourceCode" id="cb1077"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1077-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1077-1"></a><span class="kw">rm</span>(p1, p2)</span></code></pre></div>
<div class="sourceCode" id="cb1078"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1078-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotExpression</span>(sce,</span>
<span id="cb1078-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-2"></a>               <span class="dt">exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb1078-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-3"></a>               <span class="dt">x=</span>cluToUse,</span>
<span id="cb1078-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-4"></a>               <span class="dt">colour_by=</span>cluToUse,</span>
<span id="cb1078-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-5"></a>               <span class="dt">features=</span> <span class="st">&quot;MS4A1&quot;</span>) <span class="op">+</span></span>
<span id="cb1078-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;B cells&quot;</span>)</span>
<span id="cb1078-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-7"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotExpression</span>(sce,</span>
<span id="cb1078-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-8"></a>               <span class="dt">exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb1078-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-9"></a>               <span class="dt">x=</span>cluToUse,</span>
<span id="cb1078-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-10"></a>               <span class="dt">colour_by=</span>cluToUse,</span>
<span id="cb1078-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-11"></a>               <span class="dt">features=</span> <span class="st">&quot;CD79A&quot;</span>) <span class="op">+</span></span>
<span id="cb1078-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-12"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;B cells&quot;</span>)</span>
<span id="cb1078-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1078-13"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-32-1.png" width="960" /></p>
<div class="sourceCode" id="cb1079"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1079-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1079-1"></a><span class="kw">rm</span>(p1, p2)</span></code></pre></div>
<p>We now show the distribution of expression levels for marker genes of other PBMC types.</p>
<div class="sourceCode" id="cb1080"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1080-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-1"></a><span class="co">#markGenes &lt;- c(&quot;IL7R&quot;, &quot;CCR7&quot;, &quot;S100A4&quot;, &quot;CD14&quot;, &quot;LYZ&quot;, &quot;MS4A1&quot;, &quot;CD8A&quot;, &quot;FCGR3A&quot;, &quot;MS4A7&quot;, &quot;GNLY&quot;, &quot;NKG7&quot;, &quot;FCER1A&quot;, &quot;CST3&quot;, &quot;PPBP&quot;)</span></span>
<span id="cb1080-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-2"></a>markGenesFull &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb1080-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-3"></a>  <span class="st">&quot;Naive CD4+ T cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;IL7R&quot;</span>, <span class="st">&quot;CCR7&quot;</span>),</span>
<span id="cb1080-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-4"></a>  <span class="st">&quot;Memory CD4+ T cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;IL7R&quot;</span>, <span class="st">&quot;S100A4&quot;</span>),</span>
<span id="cb1080-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-5"></a>  <span class="st">&quot;B cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;MS4A1&quot;</span>),</span>
<span id="cb1080-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-6"></a>  <span class="st">&quot;CD8+ T cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;CD8A&quot;</span>),</span>
<span id="cb1080-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-7"></a>  <span class="st">&quot;NK cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;GNLY&quot;</span>, <span class="st">&quot;NKG7&quot;</span>),</span>
<span id="cb1080-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-8"></a>  <span class="st">&quot;CD14+ Monocytes&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;LYZ&quot;</span>),</span>
<span id="cb1080-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-9"></a>  <span class="st">&quot;FCGR3A+ Monocytes&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;FCGR3A&quot;</span>, <span class="st">&quot;MS4A7&quot;</span>),</span>
<span id="cb1080-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-10"></a>  <span class="st">&quot;Dendritic Cells&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;FCER1A&quot;</span>, <span class="st">&quot;CST3&quot;</span>),</span>
<span id="cb1080-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-11"></a>  <span class="st">&quot;Platelets&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;PPBP&quot;</span>)</span>
<span id="cb1080-12"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-12"></a>)</span>
<span id="cb1080-13"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-13"></a>markGenesAvail &lt;-<span class="st"> </span><span class="kw">lapply</span>(markGenesFull,</span>
<span id="cb1080-14"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-14"></a>                         <span class="cf">function</span>(x){</span>
<span id="cb1080-15"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-15"></a>                           x[x <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(sce)]</span>
<span id="cb1080-16"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1080-16"></a>                         })</span></code></pre></div>
<div class="sourceCode" id="cb1081"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1081-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-1"></a>plotExpression2 &lt;-<span class="st"> </span><span class="cf">function</span>(sce, ctx, markGenesAvail) {</span>
<span id="cb1081-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-2"></a>  <span class="cf">if</span>(<span class="kw">length</span>(markGenesAvail[[ctx]])<span class="op">&gt;</span><span class="dv">0</span>) {</span>
<span id="cb1081-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-3"></a>    <span class="kw">plotExpression</span>(sce, <span class="dt">exprs_values =</span> <span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb1081-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-4"></a>                   <span class="dt">x=</span>cluToUse, <span class="dt">colour_by=</span>cluToUse,</span>
<span id="cb1081-5"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-5"></a>                   <span class="dt">features=</span>markGenesAvail[[ctx]]) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(ctx)</span>
<span id="cb1081-6"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-6"></a>  }</span>
<span id="cb1081-7"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-7"></a>}</span>
<span id="cb1081-8"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-8"></a></span>
<span id="cb1081-9"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-9"></a><span class="co"># Naive CD4+ T</span></span>
<span id="cb1081-10"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-10"></a>ctx &lt;-<span class="st"> &quot;Naive CD4+ T cells&quot;</span></span>
<span id="cb1081-11"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1081-11"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-1.png" width="960" /></p>
<div class="sourceCode" id="cb1082"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1082-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1082-1"></a><span class="co"># Memory CD4+</span></span>
<span id="cb1082-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1082-2"></a>ctx &lt;-<span class="st"> &quot;Memory CD4+ T cells&quot;</span></span>
<span id="cb1082-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1082-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-2.png" width="960" /></p>
<div class="sourceCode" id="cb1083"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1083-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1083-1"></a><span class="co"># B cells</span></span>
<span id="cb1083-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1083-2"></a><span class="co">##plotExpression(sce, exprs_values = &quot;reconstructed&quot;, x=cluToUse, colour_by=cluToUse, features=markGenes[6]) + ggtitle(&quot;B cells&quot;)</span></span>
<span id="cb1083-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1083-3"></a>ctx &lt;-<span class="st"> &quot;B cells&quot;</span></span>
<span id="cb1083-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1083-4"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-3.png" width="960" /></p>
<div class="sourceCode" id="cb1084"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1084-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1084-1"></a><span class="co"># CD8+ T</span></span>
<span id="cb1084-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1084-2"></a>ctx &lt;-<span class="st"> &quot;CD8+ T cells&quot;</span></span>
<span id="cb1084-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1084-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-4.png" width="960" /></p>
<div class="sourceCode" id="cb1085"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1085-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1085-1"></a><span class="co"># NK</span></span>
<span id="cb1085-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1085-2"></a>ctx &lt;-<span class="st"> &quot;NK cells&quot;</span></span>
<span id="cb1085-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1085-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-5.png" width="960" /></p>
<div class="sourceCode" id="cb1086"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1086-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1086-1"></a><span class="co"># CD14+ Mono</span></span>
<span id="cb1086-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1086-2"></a>ctx &lt;-<span class="st"> &quot;CD14+ Monocytes&quot;</span></span>
<span id="cb1086-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1086-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-6.png" width="960" /></p>
<div class="sourceCode" id="cb1087"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1087-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1087-1"></a><span class="co"># FCGR3A+ Mono</span></span>
<span id="cb1087-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1087-2"></a>ctx &lt;-<span class="st"> &quot;FCGR3A+ Monocytes&quot;</span></span>
<span id="cb1087-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1087-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-7.png" width="960" /></p>
<div class="sourceCode" id="cb1088"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1088-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1088-1"></a><span class="co"># DC</span></span>
<span id="cb1088-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1088-2"></a>ctx &lt;-<span class="st"> &quot;Dendritic Cells&quot;</span></span>
<span id="cb1088-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1088-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-8.png" width="960" /></p>
<div class="sourceCode" id="cb1089"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1089-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1089-1"></a><span class="co"># Platelet</span></span>
<span id="cb1089-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1089-2"></a>ctx &lt;-<span class="st"> &quot;Platelets&quot;</span></span>
<span id="cb1089-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1089-3"></a><span class="kw">plotExpression2</span>(sce, ctx, markGenesAvail)</span></code></pre></div>
<p><img src="clusteringPostDsi_files/figure-html/unnamed-chunk-34-9.png" width="960" /></p>
<!-- Add clusters to initial sce (not the output fastMNN) ?TODO? -->
</div>
<div id="save-data" class="section level2">
<h2><span class="header-section-number">14.5</span> Save data</h2>
<p>Write SCE object to file.</p>
<div class="sourceCode" id="cb1090"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1090-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1090-1"></a><span class="co">#tmpFn &lt;- sprintf(&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_clustered.Rds&quot;,</span></span>
<span id="cb1090-2"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1090-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s%s_%s_clust.Rds&quot;</span>,</span>
<span id="cb1090-3"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1090-3"></a>                 projDir, outDirBit, setName, setSuf, dsiSuf, splSetToGet2)</span>
<span id="cb1090-4"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1090-4"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dsi_PBMMC_ETV6-RUNX1_clust.Rds&quot;</code></pre>
<div class="sourceCode" id="cb1092"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1092-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1092-1"></a><span class="kw">saveRDS</span>(sce, <span class="dt">file=</span>tmpFn)</span></code></pre></div>
</div>
<div id="session-information-10" class="section level2">
<h2><span class="header-section-number">14.6</span> Session information</h2>
<details>
<div class="sourceCode" id="cb1093"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1093-1"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#cb1093-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] reticulate_1.20             NbClust_3.0                
##  [3] purrr_0.3.4                 tidyr_1.1.3                
##  [5] tibble_3.1.2                broom_0.7.6                
##  [7] Cairo_1.5-12.2              pheatmap_1.0.12            
##  [9] leiden_0.3.7                igraph_1.2.6               
## [11] cluster_2.1.2               dynamicTreeCut_1.63-1      
## [13] dplyr_1.0.6                 scran_1.18.7               
## [15] scater_1.18.6               SingleCellExperiment_1.12.0
## [17] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [19] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [21] IRanges_2.24.1              S4Vectors_0.28.1           
## [23] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [25] matrixStats_0.58.0          ggplot2_3.3.3              
## [27] knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7              RColorBrewer_1.1-2       
##  [3] backports_1.2.1           tools_4.0.3              
##  [5] bslib_0.2.5               utf8_1.2.1               
##  [7] R6_2.5.0                  irlba_2.3.3              
##  [9] vipor_0.4.5               DBI_1.1.1                
## [11] colorspace_2.0-1          withr_2.4.2              
## [13] tidyselect_1.1.1          gridExtra_2.3            
## [15] compiler_4.0.3            cli_2.5.0                
## [17] BiocNeighbors_1.8.2       DelayedArray_0.16.3      
## [19] labeling_0.4.2            bookdown_0.22            
## [21] sass_0.4.0                scales_1.1.1             
## [23] rappdirs_0.3.3            stringr_1.4.0            
## [25] digest_0.6.27             rmarkdown_2.8            
## [27] XVector_0.30.0            pkgconfig_2.0.3          
## [29] htmltools_0.5.1.1         sparseMatrixStats_1.2.1  
## [31] highr_0.9                 limma_3.46.0             
## [33] rlang_0.4.11              rstudioapi_0.13          
## [35] DelayedMatrixStats_1.12.3 farver_2.1.0             
## [37] jquerylib_0.1.4           generics_0.1.0           
## [39] jsonlite_1.7.2            mclust_5.4.7             
## [41] BiocParallel_1.24.1       RCurl_1.98-1.3           
## [43] magrittr_2.0.1            BiocSingular_1.6.0       
## [45] GenomeInfoDbData_1.2.4    scuttle_1.0.4            
## [47] Matrix_1.3-3              Rcpp_1.0.6               
## [49] ggbeeswarm_0.6.0          munsell_0.5.0            
## [51] fansi_0.4.2               viridis_0.6.1            
## [53] lifecycle_1.0.0           stringi_1.6.1            
## [55] yaml_2.2.1                edgeR_3.32.1             
## [57] zlibbioc_1.36.0           grid_4.0.3               
## [59] dqrng_0.3.0               crayon_1.4.1             
## [61] lattice_0.20-44           cowplot_1.1.1            
## [63] beachmat_2.6.4            locfit_1.5-9.4           
## [65] ps_1.6.0                  pillar_1.6.1             
## [67] codetools_0.2-18          glue_1.4.2               
## [69] evaluate_0.14             vctrs_0.3.8              
## [71] png_0.1-7                 gtable_0.3.0             
## [73] assertthat_0.2.1          xfun_0.23                
## [75] rsvd_1.0.5                viridisLite_0.4.0        
## [77] beeswarm_0.3.1            bluster_1.0.0            
## [79] statmod_1.4.36            ellipsis_0.3.2</code></pre>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dsi-5hCellPerSpl-allSetsTop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="clusterMarkerGenesTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/clusteringPostDsi.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC2.pdf", "KmWiC2.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
