<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 29 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 29 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 29 Data integration - all Caron sample types | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-05-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"/>
<link rel="next" href="cellCyclePhasesTop.html"/>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.3</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="PreProcTop.html"><a href="PreProcTop.html"><i class="fa fa-check"></i><b>4</b> Quality Control - with 500 cells per sample</a><ul>
<li class="chapter" data-level="4.1" data-path="PreProcTop.html"><a href="PreProcTop.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="PreProcTop.html"><a href="PreProcTop.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="PreProcTop.html"><a href="PreProcTop.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="PreProcTop.html"><a href="PreProcTop.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="PreProcTop.html"><a href="PreProcTop.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="PreProcTop.html"><a href="PreProcTop.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="PreProcTop.html"><a href="PreProcTop.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="PreProcTop.html"><a href="PreProcTop.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="PreProcTop.html"><a href="PreProcTop.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="PreProcTop.html"><a href="PreProcTop.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="PreProcTop.html"><a href="PreProcTop.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="PreProcTop.html"><a href="PreProcTop.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="PreProcTop.html"><a href="PreProcTop.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="PreProcTop.html"><a href="PreProcTop.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="PreProcTop.html"><a href="PreProcTop.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="PreProcTop.html"><a href="PreProcTop.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="PreProcTop.html"><a href="PreProcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="PreProcTop.html"><a href="PreProcTop.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="PreProcTop.html"><a href="PreProcTop.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="PreProcTop.html"><a href="PreProcTop.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="PreProcTop.html"><a href="PreProcTop.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="PreProcTop.html"><a href="PreProcTop.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="PreProcTop.html"><a href="PreProcTop.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="PreProcTop.html"><a href="PreProcTop.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="PreProcTop.html"><a href="PreProcTop.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="PreProcTop.html"><a href="PreProcTop.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="PreProcTop.html"><a href="PreProcTop.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="PreProcTop.html"><a href="PreProcTop.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="PreProcTop.html"><a href="PreProcTop.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="PreProcTop.html"><a href="PreProcTop.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="PreProcTop.html"><a href="PreProcTop.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="PreProcTop.html"><a href="PreProcTop.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="PreProcTop.html"><a href="PreProcTop.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="PreProcTop.html"><a href="PreProcTop.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="PreProcTop.html"><a href="PreProcTop.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="PreProcTop.html"><a href="PreProcTop.html#session-information"><i class="fa fa-check"></i><b>4.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html"><i class="fa fa-check"></i><b>5</b> Normalisation - Caron set</a><ul>
<li class="chapter" data-level="5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#learning-objectives"><i class="fa fa-check"></i><b>5.1</b> Learning objectives</a></li>
<li class="chapter" data-level="5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scaling-normalization"><i class="fa fa-check"></i><b>5.2</b> Scaling normalization</a><ul>
<li class="chapter" data-level="5.2.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cpm"><i class="fa fa-check"></i><b>5.2.1</b> CPM</a></li>
<li class="chapter" data-level="5.2.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deseqs-size-factor"><i class="fa fa-check"></i><b>5.2.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="5.2.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>5.2.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="5.2.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#library-size-normalization"><i class="fa fa-check"></i><b>5.2.4</b> Library size normalization</a></li>
<li class="chapter" data-level="5.2.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#deconvolution"><i class="fa fa-check"></i><b>5.2.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="5.2.5.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#cluster-cells"><i class="fa fa-check"></i><b>5.2.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="5.2.5.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#compute-size-factors"><i class="fa fa-check"></i><b>5.2.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="5.2.5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#apply-size-factors"><i class="fa fa-check"></i><b>5.2.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="5.2.5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-object"><i class="fa fa-check"></i><b>5.2.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform"><i class="fa fa-check"></i><b>5.3</b> SCTransform</a><ul>
<li class="chapter" data-level="5.3.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#inspect-data"><i class="fa fa-check"></i><b>5.3.1</b> Inspect data</a></li>
<li class="chapter" data-level="5.3.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#transformation"><i class="fa fa-check"></i><b>5.3.2</b> Transformation</a></li>
<li class="chapter" data-level="5.3.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#save-sce-object"><i class="fa fa-check"></i><b>5.3.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#visualisation"><i class="fa fa-check"></i><b>5.4</b> Visualisation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-raw-counts"><i class="fa fa-check"></i><b>5.4.1</b> log raw counts</a></li>
<li class="chapter" data-level="5.4.2" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#log-cpm"><i class="fa fa-check"></i><b>5.4.2</b> log CPM</a></li>
<li class="chapter" data-level="5.4.3" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#scran"><i class="fa fa-check"></i><b>5.4.3</b> scran</a></li>
<li class="chapter" data-level="5.4.4" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#sctransform-1"><i class="fa fa-check"></i><b>5.4.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="NormalisationCaron5hcpsTop.html"><a href="NormalisationCaron5hcpsTop.html#session-information-1"><i class="fa fa-check"></i><b>5.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html"><i class="fa fa-check"></i><b>6</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="6.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#description"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#example-1"><i class="fa fa-check"></i><b>6.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-packages-1"><i class="fa fa-check"></i><b>6.2</b> Load packages</a></li>
<li class="chapter" data-level="6.3" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#load-data"><i class="fa fa-check"></i><b>6.3</b> Load data</a></li>
<li class="chapter" data-level="6.4" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#pca"><i class="fa fa-check"></i><b>6.4</b> PCA</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>6.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#perplexity"><i class="fa fa-check"></i><b>6.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="6.5.2" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#stochasticity"><i class="fa fa-check"></i><b>6.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#umap"><i class="fa fa-check"></i><b>6.6</b> UMAP</a></li>
<li class="chapter" data-level="6.7" data-path="dimRedForVizTop.html"><a href="dimRedForVizTop.html#session-information-2"><i class="fa fa-check"></i><b>6.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html"><i class="fa fa-check"></i><b>7</b> Identifying confounding factors - Caron set</a><ul>
<li class="chapter" data-level="7.1" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#load-object"><i class="fa fa-check"></i><b>7.1</b> Load object</a></li>
<li class="chapter" data-level="7.2" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#pca-1"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#normalised-counts"><i class="fa fa-check"></i><b>7.3</b> Normalised counts</a></li>
<li class="chapter" data-level="7.4" data-path="identifying-confounding-factors-caron-set.html"><a href="identifying-confounding-factors-caron-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="featSelecTop.html"><a href="featSelecTop.html"><i class="fa fa-check"></i><b>8</b> Feature selection</a><ul>
<li class="chapter" data-level="8.1" data-path="featSelecTop.html"><a href="featSelecTop.html#load-data-1"><i class="fa fa-check"></i><b>8.1</b> Load data</a></li>
<li class="chapter" data-level="8.2" data-path="featSelecTop.html"><a href="featSelecTop.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>8.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="8.2.1" data-path="featSelecTop.html"><a href="featSelecTop.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>8.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="8.2.2" data-path="featSelecTop.html"><a href="featSelecTop.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>8.2.2</b> Choosing some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="featSelecTop.html"><a href="featSelecTop.html#session-information-4"><i class="fa fa-check"></i><b>8.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html"><i class="fa fa-check"></i><b>9</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#load-data-2"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>9.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="9.3" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>9.3</b> Visualise expression patterns of some HVGs</a></li>
<li class="chapter" data-level="9.4" data-path="dimRedForAnaTop.html"><a href="dimRedForAnaTop.html#session-information-5"><i class="fa fa-check"></i><b>9.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>10</b> Batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="10.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#learning-objectives-1"><i class="fa fa-check"></i><b>10.1</b> Learning objectives</a></li>
<li class="chapter" data-level="10.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>10.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="10.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>10.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="10.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>10.4</b> Batch correction</a><ul>
<li class="chapter" data-level="10.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>10.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-2"><i class="fa fa-check"></i><b>10.5</b> SCTransform</a></li>
<li class="chapter" data-level="10.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>10.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="10.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>10.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="10.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>10.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>10.7</b> fastMNN</a></li>
<li class="chapter" data-level="10.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>10.8</b> Harmony</a></li>
<li class="chapter" data-level="10.9" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#session-information-6"><i class="fa fa-check"></i><b>10.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>11</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="11.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#learning-objectives-2"><i class="fa fa-check"></i><b>11.1</b> Learning objectives</a></li>
<li class="chapter" data-level="11.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#abbreviations-1"><i class="fa fa-check"></i><b>11.2</b> Abbreviations</a></li>
<li class="chapter" data-level="11.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#motivation"><i class="fa fa-check"></i><b>11.3</b> Motivation</a></li>
<li class="chapter" data-level="11.4" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#load-the-data"><i class="fa fa-check"></i><b>11.4</b> Load the data</a></li>
<li class="chapter" data-level="11.5" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>11.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="11.6" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#linear-regression"><i class="fa fa-check"></i><b>11.6</b> Linear regression</a></li>
<li class="chapter" data-level="11.7" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction"><i class="fa fa-check"></i><b>11.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="11.7.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#algorithm-overview"><i class="fa fa-check"></i><b>11.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="11.7.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#application-to-the-data"><i class="fa fa-check"></i><b>11.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#correction-diagnostics"><i class="fa fa-check"></i><b>11.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="11.8.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#mixing-between-batches"><i class="fa fa-check"></i><b>11.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="11.8.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>11.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="11.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction"><i class="fa fa-check"></i><b>11.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="11.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#coassignment-probabilities"><i class="fa fa-check"></i><b>11.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="11.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#rand-index"><i class="fa fa-check"></i><b>11.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>11.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="11.10" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells"><i class="fa fa-check"></i><b>11.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="11.11" data-path="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html"><a href="dsi-5hCellPerSpl-PBMMC-PRE-TTop.html#session-information-7"><i class="fa fa-check"></i><b>11.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>12</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="12.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#learning-objectives-3"><i class="fa fa-check"></i><b>12.1</b> Learning objectives</a></li>
<li class="chapter" data-level="12.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#abbreviations-2"><i class="fa fa-check"></i><b>12.2</b> Abbreviations</a></li>
<li class="chapter" data-level="12.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#motivation-1"><i class="fa fa-check"></i><b>12.3</b> Motivation</a></li>
<li class="chapter" data-level="12.4" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#load-the-data-1"><i class="fa fa-check"></i><b>12.4</b> Load the data</a></li>
<li class="chapter" data-level="12.5" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-1"><i class="fa fa-check"></i><b>12.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="12.6" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#linear-regression-1"><i class="fa fa-check"></i><b>12.6</b> Linear regression</a></li>
<li class="chapter" data-level="12.7" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-1"><i class="fa fa-check"></i><b>12.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="12.7.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-1"><i class="fa fa-check"></i><b>12.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="12.7.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-1"><i class="fa fa-check"></i><b>12.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-1"><i class="fa fa-check"></i><b>12.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="12.8.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-1"><i class="fa fa-check"></i><b>12.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="12.8.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-1"><i class="fa fa-check"></i><b>12.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="12.8.2.1" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-1"><i class="fa fa-check"></i><b>12.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="12.8.2.2" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-1"><i class="fa fa-check"></i><b>12.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="12.8.2.3" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#rand-index-1"><i class="fa fa-check"></i><b>12.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-1"><i class="fa fa-check"></i><b>12.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="12.10" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-1"><i class="fa fa-check"></i><b>12.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="12.11" data-path="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-5hCellPerSpl-PBMMC-ETV6-RUNX1Top.html#session-information-8"><i class="fa fa-check"></i><b>12.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html"><i class="fa fa-check"></i><b>13</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="13.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#learning-objectives-4"><i class="fa fa-check"></i><b>13.1</b> Learning objectives</a></li>
<li class="chapter" data-level="13.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#motivation-2"><i class="fa fa-check"></i><b>13.2</b> Motivation</a></li>
<li class="chapter" data-level="13.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#load-the-data-2"><i class="fa fa-check"></i><b>13.3</b> Load the data</a></li>
<li class="chapter" data-level="13.4" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#diagnosing-batch-effects-2"><i class="fa fa-check"></i><b>13.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="13.5" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#linear-regression-2"><i class="fa fa-check"></i><b>13.5</b> Linear regression</a></li>
<li class="chapter" data-level="13.6" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#mutual-nearest-neighbour-correction-2"><i class="fa fa-check"></i><b>13.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="13.6.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#algorithm-overview-2"><i class="fa fa-check"></i><b>13.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="13.6.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#application-to-the-data-2"><i class="fa fa-check"></i><b>13.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#correction-diagnostics-2"><i class="fa fa-check"></i><b>13.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="13.7.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-2"><i class="fa fa-check"></i><b>13.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.7.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>13.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.7.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#coassignment-probabilities-2"><i class="fa fa-check"></i><b>13.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="13.7.1.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-2"><i class="fa fa-check"></i><b>13.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="13.7.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#encouraging-consistency-with-marker-genes-2"><i class="fa fa-check"></i><b>13.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="13.7.3" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#using-the-corrected-values"><i class="fa fa-check"></i><b>13.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#challenge-same-but-with-an-ordered-merging"><i class="fa fa-check"></i><b>13.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="13.8.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#preserving-biological-heterogeneity-3"><i class="fa fa-check"></i><b>13.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="13.8.1.1" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#comparison-to-within-batch-clusters-1"><i class="fa fa-check"></i><b>13.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="13.8.1.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#rand-index-3"><i class="fa fa-check"></i><b>13.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="13.8.2" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#cluster-markers"><i class="fa fa-check"></i><b>13.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#identify-clusters-with-pbmmc-cells-2"><i class="fa fa-check"></i><b>13.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="13.10" data-path="dsi-5hCellPerSpl-allSetsTop.html"><a href="dsi-5hCellPerSpl-allSetsTop.html#session-information-9"><i class="fa fa-check"></i><b>13.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><i class="fa fa-check"></i><b>14</b> Clustering with ‘PBMMC,ETV6-RUNX1’ and ’_5hCellPerSpl’ {#clusteringPBMMC-ETV6-RUNX1-5hCellPerSplTop}</a><ul>
<li class="chapter" data-level="14.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#load-data-3"><i class="fa fa-check"></i><b>14.1</b> Load data</a></li>
<li class="chapter" data-level="14.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering-cells-into-putative-subpopulations"><i class="fa fa-check"></i><b>14.2</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="14.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#hierarchical-clustering"><i class="fa fa-check"></i><b>14.2.1</b> Hierarchical clustering</a><ul>
<li class="chapter" data-level="14.2.1.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#clustering"><i class="fa fa-check"></i><b>14.2.1.1</b> Clustering</a></li>
<li class="chapter" data-level="14.2.1.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#separatedness"><i class="fa fa-check"></i><b>14.2.1.2</b> Separatedness</a></li>
</ul></li>
<li class="chapter" data-level="14.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#k-means"><i class="fa fa-check"></i><b>14.2.2</b> k-means</a><ul>
<li class="chapter" data-level="14.2.2.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#description-1"><i class="fa fa-check"></i><b>14.2.2.1</b> Description</a></li>
<li class="chapter" data-level="14.2.2.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#example-2"><i class="fa fa-check"></i><b>14.2.2.2</b> Example</a></li>
<li class="chapter" data-level="14.2.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#within-cluster-sum-of-squares"><i class="fa fa-check"></i><b>14.2.2.3</b> Within-cluster sum-of-squares</a></li>
<li class="chapter" data-level="14.2.2.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#silhouette"><i class="fa fa-check"></i><b>14.2.2.4</b> Silhouette</a></li>
<li class="chapter" data-level="14.2.2.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#gap-statistic"><i class="fa fa-check"></i><b>14.2.2.5</b> Gap statistic</a></li>
<li class="chapter" data-level="14.2.2.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#multi-metric"><i class="fa fa-check"></i><b>14.2.2.6</b> Multi-metric</a></li>
</ul></li>
<li class="chapter" data-level="14.2.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#graph-based-clustering"><i class="fa fa-check"></i><b>14.2.3</b> Graph-based clustering</a><ul>
<li class="chapter" data-level="14.2.3.1" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity"><i class="fa fa-check"></i><b>14.2.3.1</b> Modularity</a></li>
<li class="chapter" data-level="14.2.3.2" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#walktrap-method"><i class="fa fa-check"></i><b>14.2.3.2</b> Walktrap method</a></li>
<li class="chapter" data-level="14.2.3.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#louvain-method"><i class="fa fa-check"></i><b>14.2.3.3</b> Louvain method</a></li>
<li class="chapter" data-level="14.2.3.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#leiden-method"><i class="fa fa-check"></i><b>14.2.3.4</b> Leiden method</a></li>
<li class="chapter" data-level="14.2.3.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#modularity-to-assess-clusters-quality"><i class="fa fa-check"></i><b>14.2.3.5</b> Modularity to assess clusters quality</a></li>
<li class="chapter" data-level="14.2.3.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#choose-the-number-of-clusters"><i class="fa fa-check"></i><b>14.2.3.6</b> Choose the number of clusters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#comparing-two-sets-of-clusters"><i class="fa fa-check"></i><b>14.3</b> Comparing two sets of clusters</a></li>
<li class="chapter" data-level="14.4" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#expression-of-known-marker-genes"><i class="fa fa-check"></i><b>14.4</b> Expression of known marker genes</a></li>
<li class="chapter" data-level="14.5" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#save-data"><i class="fa fa-check"></i><b>14.5</b> Save data</a></li>
<li class="chapter" data-level="14.6" data-path="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html"><a href="clustering-with-pbmmcetv6-runx1-and-5hcellperspl-clusteringpbmmc-etv6-runx1-5hcellperspltop.html#session-information-10"><i class="fa fa-check"></i><b>14.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html"><i class="fa fa-check"></i><b>15</b> Cluster marker genes</a><ul>
<li class="chapter" data-level="15.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#learning-objectives-5"><i class="fa fa-check"></i><b>15.1</b> Learning objectives</a></li>
<li class="chapter" data-level="15.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#load-data-4"><i class="fa fa-check"></i><b>15.2</b> Load data</a></li>
<li class="chapter" data-level="15.3" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#detecting-genes-differentially-expressed-between-clusters"><i class="fa fa-check"></i><b>15.3</b> Detecting genes differentially expressed between clusters</a><ul>
<li class="chapter" data-level="15.3.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#differential-expression-analysis"><i class="fa fa-check"></i><b>15.3.1</b> Differential expression analysis</a></li>
<li class="chapter" data-level="15.3.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#heatmap"><i class="fa fa-check"></i><b>15.3.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-log-fold-change"><i class="fa fa-check"></i><b>15.4</b> Using the log-fold change</a></li>
<li class="chapter" data-level="15.5" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#finding-cluster-specific-markers"><i class="fa fa-check"></i><b>15.5</b> Finding cluster-specific markers</a></li>
<li class="chapter" data-level="15.6" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-the-wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>15.6</b> Using the Wilcoxon rank sum test</a></li>
<li class="chapter" data-level="15.7" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#using-a-binomial-test"><i class="fa fa-check"></i><b>15.7</b> Using a binomial test</a></li>
<li class="chapter" data-level="15.8" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#combining-multiple-marker-statistics"><i class="fa fa-check"></i><b>15.8</b> Combining multiple marker statistics</a></li>
<li class="chapter" data-level="15.9" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#invalidity-of-p-values"><i class="fa fa-check"></i><b>15.9</b> Invalidity of p-values</a><ul>
<li class="chapter" data-level="15.9.1" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#from-data-snooping"><i class="fa fa-check"></i><b>15.9.1</b> From data snooping</a></li>
<li class="chapter" data-level="15.9.2" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#nature-of-replication"><i class="fa fa-check"></i><b>15.9.2</b> Nature of replication</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="clusterMarkerGenesTop.html"><a href="clusterMarkerGenesTop.html#session-information-11"><i class="fa fa-check"></i><b>15.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html"><i class="fa fa-check"></i><b>16</b> Differential expression and abundance between conditions</a><ul>
<li class="chapter" data-level="16.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#motivation-3"><i class="fa fa-check"></i><b>16.1</b> Motivation</a></li>
<li class="chapter" data-level="16.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#setting-up-the-data"><i class="fa fa-check"></i><b>16.2</b> Setting up the data</a></li>
<li class="chapter" data-level="16.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-between-conditions"><i class="fa fa-check"></i><b>16.3</b> Differential expression between conditions</a><ul>
<li class="chapter" data-level="16.3.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#creating-pseudo-bulk-samples"><i class="fa fa-check"></i><b>16.3.1</b> Creating pseudo-bulk samples</a></li>
<li class="chapter" data-level="16.3.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#performing-the-de-analysis"><i class="fa fa-check"></i><b>16.3.2</b> Performing the DE analysis</a><ul>
<li class="chapter" data-level="16.3.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#introduction-3"><i class="fa fa-check"></i><b>16.3.2.1</b> Introduction</a></li>
<li class="chapter" data-level="16.3.2.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#pre-processing"><i class="fa fa-check"></i><b>16.3.2.2</b> Pre-processing</a></li>
<li class="chapter" data-level="16.3.2.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#statistical-modelling"><i class="fa fa-check"></i><b>16.3.2.3</b> Statistical modelling</a></li>
<li class="chapter" data-level="16.3.2.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-expression-for-each-cluster"><i class="fa fa-check"></i><b>16.3.2.4</b> Differential expression for each cluster</a><ul>
<li class="chapter" data-level="16.3.2.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-degs-by-cluster-and-direction"><i class="fa fa-check"></i><b>16.3.2.4.1</b> Number of DEGs by cluster and direction</a></li>
<li class="chapter" data-level="16.3.2.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#list-of-degs"><i class="fa fa-check"></i><b>16.3.2.4.2</b> List of DEGs</a></li>
<li class="chapter" data-level="16.3.2.4.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#number-of-clusters-skipped"><i class="fa fa-check"></i><b>16.3.2.4.3</b> Number of clusters skipped</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.3.3" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#putting-it-all-together"><i class="fa fa-check"></i><b>16.3.3</b> Putting it all together</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#differential-abundance-between-conditions"><i class="fa fa-check"></i><b>16.4</b> Differential abundance between conditions</a><ul>
<li class="chapter" data-level="16.4.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#overview"><i class="fa fa-check"></i><b>16.4.1</b> Overview</a></li>
<li class="chapter" data-level="16.4.2" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#handling-composition-effects"><i class="fa fa-check"></i><b>16.4.2</b> Handling composition effects</a><ul>
<li class="chapter" data-level="16.4.2.1" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#background"><i class="fa fa-check"></i><b>16.4.2.1</b> Background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="multiSplCompTop.html"><a href="multiSplCompTop.html#session-information-12"><i class="fa fa-check"></i><b>16.5</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html"><i class="fa fa-check"></i><b>17</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="17.0.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#variance-modeling"><i class="fa fa-check"></i><b>17.0.1</b> Variance modeling</a></li>
<li class="chapter" data-level="17.0.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#data-integration"><i class="fa fa-check"></i><b>17.0.2</b> Data integration</a></li>
<li class="chapter" data-level="17.0.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#dimensionality-reduction"><i class="fa fa-check"></i><b>17.0.3</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="17.0.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#clustering-1"><i class="fa fa-check"></i><b>17.0.4</b> Clustering</a></li>
<li class="chapter" data-level="17.0.5" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#cellTypeAnnotation"><i class="fa fa-check"></i><b>17.0.5</b> Cell type classification</a></li>
<li class="chapter" data-level="17.0.6" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#extract-t-cells"><i class="fa fa-check"></i><b>17.0.6</b> Extract T-cells</a></li>
<li class="chapter" data-level="17.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTimeSetUp"><i class="fa fa-check"></i><b>17.1</b> Setting up the data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#trajectory-inference-with-destiny"><i class="fa fa-check"></i><b>17.1.1</b> Trajectory inference with destiny</a></li>
<li class="chapter" data-level="17.1.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>17.1.2</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="17.1.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudotime-analysis-for-another-hca-sample"><i class="fa fa-check"></i><b>17.1.3</b> Pseudotime analysis for another HCA sample</a></li>
<li class="chapter" data-level="17.1.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#exercise-1"><i class="fa fa-check"></i><b>17.1.4</b> Exercise 1</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#pseudoTime3Top"><i class="fa fa-check"></i><b>17.2</b> <strong>Pseudotime Alignment</strong></a></li>
<li class="chapter" data-level="17.3" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#ackowledgements"><i class="fa fa-check"></i><b>17.3</b> Ackowledgements</a></li>
<li class="chapter" data-level="17.4" data-path="pseudoTimeTop.html"><a href="pseudoTimeTop.html#session-information-13"><i class="fa fa-check"></i><b>17.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="annex.html"><a href="annex.html"><i class="fa fa-check"></i><b>18</b> Annex</a></li>
<li class="chapter" data-level="19" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html"><i class="fa fa-check"></i><b>19</b> Gene annotation</a><ul>
<li class="chapter" data-level="19.1" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#introduction-4"><i class="fa fa-check"></i><b>19.1</b> Introduction</a></li>
<li class="chapter" data-level="19.2" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>19.2</b> Sample sheet</a></li>
<li class="chapter" data-level="19.3" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#srr9264343"><i class="fa fa-check"></i><b>19.3</b> SRR9264343</a></li>
<li class="chapter" data-level="19.4" data-path="GeneAnnotTop.html"><a href="GeneAnnotTop.html#session-information-14"><i class="fa fa-check"></i><b>19.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html"><i class="fa fa-check"></i><b>20</b> Quality Control - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="20.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#introduction-5"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-packages-2"><i class="fa fa-check"></i><b>20.2</b> Load packages</a></li>
<li class="chapter" data-level="20.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sample-sheet-3"><i class="fa fa-check"></i><b>20.3</b> Sample sheet</a></li>
<li class="chapter" data-level="20.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#data-representation-1"><i class="fa fa-check"></i><b>20.4</b> Data representation</a></li>
<li class="chapter" data-level="20.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#example-3"><i class="fa fa-check"></i><b>20.5</b> Example</a></li>
<li class="chapter" data-level="20.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>20.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="20.6.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>20.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="20.6.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>20.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>20.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="20.7.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mixture-model-1"><i class="fa fa-check"></i><b>20.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="20.7.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>20.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="20.7.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#inflection-point-1"><i class="fa fa-check"></i><b>20.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="20.7.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>20.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="20.7.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>20.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="20.8" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>20.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="20.9" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>20.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="20.10" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#quality-control-1"><i class="fa fa-check"></i><b>20.10</b> Quality control</a><ul>
<li class="chapter" data-level="20.10.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>20.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="20.11" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>20.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="20.11.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#library-size-1"><i class="fa fa-check"></i><b>20.11.1</b> Library size</a></li>
<li class="chapter" data-level="20.11.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>20.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="20.11.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>20.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="20.11.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#summary-1"><i class="fa fa-check"></i><b>20.11.4</b> Summary</a></li>
<li class="chapter" data-level="20.11.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>20.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="20.11.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#assumptions-1"><i class="fa fa-check"></i><b>20.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="20.12" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>20.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="20.12.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>20.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="20.12.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>20.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="20.12.1.1.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>20.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="20.12.1.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>20.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="20.12.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>20.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="20.12.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-pca-1"><i class="fa fa-check"></i><b>20.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="20.12.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>20.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="20.12.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>20.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#novelty-1"><i class="fa fa-check"></i><b>20.13</b> Novelty</a></li>
<li class="chapter" data-level="20.14" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>20.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="20.14.1" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>20.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="20.14.2" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>20.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="20.14.3" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#filters-1"><i class="fa fa-check"></i><b>20.14.3</b> Filters</a></li>
<li class="chapter" data-level="20.14.4" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>20.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="20.14.5" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#caron-only-1"><i class="fa fa-check"></i><b>20.14.5</b> Caron only</a></li>
<li class="chapter" data-level="20.14.6" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#hca-only-1"><i class="fa fa-check"></i><b>20.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="20.15" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#subsample-hca-set"><i class="fa fa-check"></i><b>20.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="20.16" data-path="PreProcAllCellsTop.html"><a href="PreProcAllCellsTop.html#session-information-15"><i class="fa fa-check"></i><b>20.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html"><i class="fa fa-check"></i><b>21</b> Normalisation - with 2-5k cells per sample</a><ul>
<li class="chapter" data-level="21.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron"><i class="fa fa-check"></i><b>21.1</b> Caron</a><ul>
<li class="chapter" data-level="21.1.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-1"><i class="fa fa-check"></i><b>21.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.1.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-1"><i class="fa fa-check"></i><b>21.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.1.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-1"><i class="fa fa-check"></i><b>21.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.1.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-1"><i class="fa fa-check"></i><b>21.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.1.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-1"><i class="fa fa-check"></i><b>21.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca"><i class="fa fa-check"></i><b>21.2</b> Hca</a><ul>
<li class="chapter" data-level="21.2.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#library-size-normalization-2"><i class="fa fa-check"></i><b>21.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="21.2.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#deconvolution-2"><i class="fa fa-check"></i><b>21.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="21.2.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#compute-size-factors-2"><i class="fa fa-check"></i><b>21.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="21.2.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#apply-size-factors-2"><i class="fa fa-check"></i><b>21.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="21.2.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#save-object-2"><i class="fa fa-check"></i><b>21.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#sctransform-3"><i class="fa fa-check"></i><b>21.3</b> SCTransform</a><ul>
<li class="chapter" data-level="21.3.1" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#caron-1"><i class="fa fa-check"></i><b>21.3.1</b> Caron</a></li>
<li class="chapter" data-level="21.3.2" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#hca-1"><i class="fa fa-check"></i><b>21.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="NormalisationAllCellsTop.html"><a href="NormalisationAllCellsTop.html#session-information-16"><i class="fa fa-check"></i><b>21.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html"><i class="fa fa-check"></i><b>22</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="22.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scaling-normalization-1"><i class="fa fa-check"></i><b>22.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="22.1.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cpm-1"><i class="fa fa-check"></i><b>22.1.1</b> CPM</a></li>
<li class="chapter" data-level="22.1.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deseqs-size-factor-1"><i class="fa fa-check"></i><b>22.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="22.1.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#weighted-trimmed-mean-of-m-values-1"><i class="fa fa-check"></i><b>22.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="22.1.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>22.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="22.1.5" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#deconvolution-3"><i class="fa fa-check"></i><b>22.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="22.1.5.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#cluster-cells-1"><i class="fa fa-check"></i><b>22.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="22.1.5.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#compute-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="22.1.5.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#apply-size-factors-3"><i class="fa fa-check"></i><b>22.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="22.1.5.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-object-3"><i class="fa fa-check"></i><b>22.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-4"><i class="fa fa-check"></i><b>22.2</b> SCTransform</a><ul>
<li class="chapter" data-level="22.2.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#inspect-data-1"><i class="fa fa-check"></i><b>22.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="22.2.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#transformation-1"><i class="fa fa-check"></i><b>22.2.2</b> Transformation</a></li>
<li class="chapter" data-level="22.2.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#save-sce-object-1"><i class="fa fa-check"></i><b>22.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#visualisation-1"><i class="fa fa-check"></i><b>22.3</b> Visualisation</a><ul>
<li class="chapter" data-level="22.3.1" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-raw-counts-1"><i class="fa fa-check"></i><b>22.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="22.3.2" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#log-cpm-1"><i class="fa fa-check"></i><b>22.3.2</b> log CPM</a></li>
<li class="chapter" data-level="22.3.3" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#scran-1"><i class="fa fa-check"></i><b>22.3.3</b> scran</a></li>
<li class="chapter" data-level="22.3.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#sctransform-5"><i class="fa fa-check"></i><b>22.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="NormalisationGSM3872434Top.html"><a href="NormalisationGSM3872434Top.html#session-information-17"><i class="fa fa-check"></i><b>22.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html"><i class="fa fa-check"></i><b>23</b> Data integration - PBMMC</a><ul>
<li class="chapter" data-level="23.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#abbreviations-3"><i class="fa fa-check"></i><b>23.1</b> Abbreviations</a></li>
<li class="chapter" data-level="23.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#motivation-4"><i class="fa fa-check"></i><b>23.2</b> Motivation</a></li>
<li class="chapter" data-level="23.3" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#loading-the-data"><i class="fa fa-check"></i><b>23.3</b> Loading the data</a></li>
<li class="chapter" data-level="23.4" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#diagnosing-batch-effects-3"><i class="fa fa-check"></i><b>23.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="23.5" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#linear-regression-3"><i class="fa fa-check"></i><b>23.5</b> Linear regression</a></li>
<li class="chapter" data-level="23.6" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#performing-mnn-correction"><i class="fa fa-check"></i><b>23.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="23.6.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#algorithm-overview-3"><i class="fa fa-check"></i><b>23.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="23.6.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#application-to-the-data-3"><i class="fa fa-check"></i><b>23.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#correction-diagnostics-3"><i class="fa fa-check"></i><b>23.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="23.7.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#mixing-between-batches-2"><i class="fa fa-check"></i><b>23.7.1</b> Mixing between batches</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#preserving-biological-heterogeneity-4"><i class="fa fa-check"></i><b>23.8</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="23.8.1" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#comparison-to-within-batch-clusters-2"><i class="fa fa-check"></i><b>23.8.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="23.8.2" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#encouraging-consistency-with-marker-genes-3"><i class="fa fa-check"></i><b>23.8.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="DataIntegration-PBMMCTop.html"><a href="DataIntegration-PBMMCTop.html#session-information-18"><i class="fa fa-check"></i><b>23.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>24</b> Data integration - ETV6-RUNX1</a><ul>
<li class="chapter" data-level="24.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#abbreviations-4"><i class="fa fa-check"></i><b>24.1</b> Abbreviations</a></li>
<li class="chapter" data-level="24.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#motivation-5"><i class="fa fa-check"></i><b>24.2</b> Motivation</a></li>
<li class="chapter" data-level="24.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#loading-the-data-1"><i class="fa fa-check"></i><b>24.3</b> Loading the data</a></li>
<li class="chapter" data-level="24.4" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#diagnosing-batch-effects-4"><i class="fa fa-check"></i><b>24.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="24.5" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#linear-regression-4"><i class="fa fa-check"></i><b>24.5</b> Linear regression</a></li>
<li class="chapter" data-level="24.6" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#performing-mnn-correction-1"><i class="fa fa-check"></i><b>24.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="24.6.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#algorithm-overview-4"><i class="fa fa-check"></i><b>24.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="24.6.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#application-to-the-data-4"><i class="fa fa-check"></i><b>24.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#correction-diagnostics-4"><i class="fa fa-check"></i><b>24.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="24.7.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#mixing-between-batches-3"><i class="fa fa-check"></i><b>24.7.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="24.7.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-5"><i class="fa fa-check"></i><b>24.7.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="24.7.2.1" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-2"><i class="fa fa-check"></i><b>24.7.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="24.7.2.2" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#coassignment-probabilities-3"><i class="fa fa-check"></i><b>24.7.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="24.7.2.3" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#rand-index-4"><i class="fa fa-check"></i><b>24.7.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24.8" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-4"><i class="fa fa-check"></i><b>24.8</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="24.9" data-path="dataSetIntegration-ETV6-RUNX1Top.html"><a href="dataSetIntegration-ETV6-RUNX1Top.html#session-information-19"><i class="fa fa-check"></i><b>24.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html"><i class="fa fa-check"></i><b>25</b> Data integration - HDD</a><ul>
<li class="chapter" data-level="25.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#abbreviations-5"><i class="fa fa-check"></i><b>25.1</b> Abbreviations</a></li>
<li class="chapter" data-level="25.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#motivation-6"><i class="fa fa-check"></i><b>25.2</b> Motivation</a></li>
<li class="chapter" data-level="25.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#load-data-5"><i class="fa fa-check"></i><b>25.3</b> Load data</a></li>
<li class="chapter" data-level="25.4" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#loading-the-data-2"><i class="fa fa-check"></i><b>25.4</b> Loading the data</a></li>
<li class="chapter" data-level="25.5" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#diagnosing-batch-effects-5"><i class="fa fa-check"></i><b>25.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="25.6" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#linear-regression-5"><i class="fa fa-check"></i><b>25.6</b> Linear regression</a></li>
<li class="chapter" data-level="25.7" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#performing-mnn-correction-2"><i class="fa fa-check"></i><b>25.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="25.7.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#algorithm-overview-5"><i class="fa fa-check"></i><b>25.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="25.7.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#application-to-the-data-5"><i class="fa fa-check"></i><b>25.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#correction-diagnostics-5"><i class="fa fa-check"></i><b>25.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="25.8.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#mixing-between-batches-4"><i class="fa fa-check"></i><b>25.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="25.8.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#preserving-biological-heterogeneity-6"><i class="fa fa-check"></i><b>25.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="25.8.2.1" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-3"><i class="fa fa-check"></i><b>25.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="25.8.2.2" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#coassignment-probabilities-4"><i class="fa fa-check"></i><b>25.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="25.8.2.3" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#rand-index-5"><i class="fa fa-check"></i><b>25.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#encouraging-consistency-with-marker-genes-5"><i class="fa fa-check"></i><b>25.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="25.10" data-path="dataSetIntegration-HHDTop.html"><a href="dataSetIntegration-HHDTop.html#session-information-20"><i class="fa fa-check"></i><b>25.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html"><i class="fa fa-check"></i><b>26</b> Data integration - PRE-T</a><ul>
<li class="chapter" data-level="26.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#abbreviations-6"><i class="fa fa-check"></i><b>26.1</b> Abbreviations</a></li>
<li class="chapter" data-level="26.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#motivation-7"><i class="fa fa-check"></i><b>26.2</b> Motivation</a></li>
<li class="chapter" data-level="26.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#load-data-6"><i class="fa fa-check"></i><b>26.3</b> Load data</a></li>
<li class="chapter" data-level="26.4" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#loading-the-data-3"><i class="fa fa-check"></i><b>26.4</b> Loading the data</a></li>
<li class="chapter" data-level="26.5" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#diagnosing-batch-effects-6"><i class="fa fa-check"></i><b>26.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="26.6" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#linear-regression-6"><i class="fa fa-check"></i><b>26.6</b> Linear regression</a></li>
<li class="chapter" data-level="26.7" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#performing-mnn-correction-3"><i class="fa fa-check"></i><b>26.7</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="26.7.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#algorithm-overview-6"><i class="fa fa-check"></i><b>26.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="26.7.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#application-to-the-data-6"><i class="fa fa-check"></i><b>26.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="26.8" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#correction-diagnostics-6"><i class="fa fa-check"></i><b>26.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="26.8.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#mixing-between-batches-5"><i class="fa fa-check"></i><b>26.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="26.8.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#preserving-biological-heterogeneity-7"><i class="fa fa-check"></i><b>26.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="26.8.2.1" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-4"><i class="fa fa-check"></i><b>26.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="26.8.2.2" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#coassignment-probabilities-5"><i class="fa fa-check"></i><b>26.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="26.8.2.3" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#rand-index-6"><i class="fa fa-check"></i><b>26.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26.9" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#encouraging-consistency-with-marker-genes-6"><i class="fa fa-check"></i><b>26.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="26.10" data-path="DataIntegrationPretTop.html"><a href="DataIntegrationPretTop.html#session-information-21"><i class="fa fa-check"></i><b>26.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html"><i class="fa fa-check"></i><b>27</b> Data integration - PBMMC and PRE-T</a><ul>
<li class="chapter" data-level="27.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#learning-objectives-6"><i class="fa fa-check"></i><b>27.1</b> Learning objectives</a></li>
<li class="chapter" data-level="27.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#abbreviations-7"><i class="fa fa-check"></i><b>27.2</b> Abbreviations</a></li>
<li class="chapter" data-level="27.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#motivation-8"><i class="fa fa-check"></i><b>27.3</b> Motivation</a></li>
<li class="chapter" data-level="27.4" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#load-the-data-3"><i class="fa fa-check"></i><b>27.4</b> Load the data</a></li>
<li class="chapter" data-level="27.5" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#diagnosing-batch-effects-7"><i class="fa fa-check"></i><b>27.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="27.6" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#linear-regression-7"><i class="fa fa-check"></i><b>27.6</b> Linear regression</a></li>
<li class="chapter" data-level="27.7" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mutual-nearest-neighbour-correction-3"><i class="fa fa-check"></i><b>27.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="27.7.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#algorithm-overview-7"><i class="fa fa-check"></i><b>27.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="27.7.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#application-to-the-data-7"><i class="fa fa-check"></i><b>27.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#correction-diagnostics-7"><i class="fa fa-check"></i><b>27.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="27.8.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#mixing-between-batches-6"><i class="fa fa-check"></i><b>27.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="27.8.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#preserving-biological-heterogeneity-8"><i class="fa fa-check"></i><b>27.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="27.8.2.1" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-5"><i class="fa fa-check"></i><b>27.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="27.8.2.2" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#coassignment-probabilities-6"><i class="fa fa-check"></i><b>27.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="27.8.2.3" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#rand-index-7"><i class="fa fa-check"></i><b>27.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27.9" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#encouraging-consistency-with-marker-genes-7"><i class="fa fa-check"></i><b>27.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="27.10" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#identify-clusters-with-pbmmc-cells-3"><i class="fa fa-check"></i><b>27.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="27.11" data-path="dsi-allCells-PBMMC-PRE-TTop.html"><a href="dsi-allCells-PBMMC-PRE-TTop.html#session-information-22"><i class="fa fa-check"></i><b>27.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><i class="fa fa-check"></i><b>28</b> Data integration - PBMMC and ETV6-RUNX</a><ul>
<li class="chapter" data-level="28.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#learning-objectives-7"><i class="fa fa-check"></i><b>28.1</b> Learning objectives</a></li>
<li class="chapter" data-level="28.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#abbreviations-8"><i class="fa fa-check"></i><b>28.2</b> Abbreviations</a></li>
<li class="chapter" data-level="28.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#motivation-9"><i class="fa fa-check"></i><b>28.3</b> Motivation</a></li>
<li class="chapter" data-level="28.4" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#load-the-data-4"><i class="fa fa-check"></i><b>28.4</b> Load the data</a></li>
<li class="chapter" data-level="28.5" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#diagnosing-batch-effects-8"><i class="fa fa-check"></i><b>28.5</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="28.6" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#linear-regression-8"><i class="fa fa-check"></i><b>28.6</b> Linear regression</a></li>
<li class="chapter" data-level="28.7" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mutual-nearest-neighbour-correction-4"><i class="fa fa-check"></i><b>28.7</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="28.7.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#algorithm-overview-8"><i class="fa fa-check"></i><b>28.7.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="28.7.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#application-to-the-data-8"><i class="fa fa-check"></i><b>28.7.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#correction-diagnostics-8"><i class="fa fa-check"></i><b>28.8</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="28.8.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#mixing-between-batches-7"><i class="fa fa-check"></i><b>28.8.1</b> Mixing between batches</a></li>
<li class="chapter" data-level="28.8.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#preserving-biological-heterogeneity-9"><i class="fa fa-check"></i><b>28.8.2</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="28.8.2.1" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#comparison-between-within-batch-clusters-and-across-batch-clusters-obtained-after-mnn-correction-6"><i class="fa fa-check"></i><b>28.8.2.1</b> Comparison between within-batch clusters and across-batch clusters obtained after MNN correction</a></li>
<li class="chapter" data-level="28.8.2.2" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#coassignment-probabilities-7"><i class="fa fa-check"></i><b>28.8.2.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="28.8.2.3" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#rand-index-8"><i class="fa fa-check"></i><b>28.8.2.3</b> Rand index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28.9" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#encouraging-consistency-with-marker-genes-8"><i class="fa fa-check"></i><b>28.9</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="28.10" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#identify-clusters-with-pbmmc-cells-4"><i class="fa fa-check"></i><b>28.10</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="28.11" data-path="dsi-allCells-PBMMC-ETV6-RUNX1Top.html"><a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html#session-information-23"><i class="fa fa-check"></i><b>28.11</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html"><i class="fa fa-check"></i><b>29</b> Data integration - all Caron sample types</a><ul>
<li class="chapter" data-level="29.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#learning-objectives-8"><i class="fa fa-check"></i><b>29.1</b> Learning objectives</a></li>
<li class="chapter" data-level="29.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#motivation-10"><i class="fa fa-check"></i><b>29.2</b> Motivation</a></li>
<li class="chapter" data-level="29.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#load-the-data-5"><i class="fa fa-check"></i><b>29.3</b> Load the data</a></li>
<li class="chapter" data-level="29.4" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#diagnosing-batch-effects-9"><i class="fa fa-check"></i><b>29.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="29.5" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#linear-regression-9"><i class="fa fa-check"></i><b>29.5</b> Linear regression</a></li>
<li class="chapter" data-level="29.6" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#mutual-nearest-neighbour-correction-5"><i class="fa fa-check"></i><b>29.6</b> Mutual Nearest Neighbour correction</a><ul>
<li class="chapter" data-level="29.6.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#algorithm-overview-9"><i class="fa fa-check"></i><b>29.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="29.6.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#application-to-the-data-9"><i class="fa fa-check"></i><b>29.6.2</b> Application to the data</a></li>
</ul></li>
<li class="chapter" data-level="29.7" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#correction-diagnostics-9"><i class="fa fa-check"></i><b>29.7</b> Correction diagnostics</a><ul>
<li class="chapter" data-level="29.7.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-10"><i class="fa fa-check"></i><b>29.7.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.7.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-3"><i class="fa fa-check"></i><b>29.7.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.7.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#coassignment-probabilities-8"><i class="fa fa-check"></i><b>29.7.1.2</b> Coassignment probabilities</a></li>
<li class="chapter" data-level="29.7.1.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-9"><i class="fa fa-check"></i><b>29.7.1.3</b> Rand index</a></li>
</ul></li>
<li class="chapter" data-level="29.7.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#encouraging-consistency-with-marker-genes-9"><i class="fa fa-check"></i><b>29.7.2</b> Encouraging consistency with marker genes</a></li>
<li class="chapter" data-level="29.7.3" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#using-the-corrected-values-1"><i class="fa fa-check"></i><b>29.7.3</b> Using the corrected values</a></li>
</ul></li>
<li class="chapter" data-level="29.8" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#challenge-same-but-with-an-ordered-merging-1"><i class="fa fa-check"></i><b>29.8</b> <strong>Challenge</strong> Same but with an ordered merging</a><ul>
<li class="chapter" data-level="29.8.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#preserving-biological-heterogeneity-11"><i class="fa fa-check"></i><b>29.8.1</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="29.8.1.1" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#comparison-to-within-batch-clusters-4"><i class="fa fa-check"></i><b>29.8.1.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="29.8.1.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#rand-index-10"><i class="fa fa-check"></i><b>29.8.1.2</b> Rand index:</a></li>
</ul></li>
<li class="chapter" data-level="29.8.2" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#cluster-markers-1"><i class="fa fa-check"></i><b>29.8.2</b> Cluster markers:</a></li>
</ul></li>
<li class="chapter" data-level="29.9" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#identify-clusters-with-pbmmc-cells-5"><i class="fa fa-check"></i><b>29.9</b> Identify clusters with PBMMC cells</a></li>
<li class="chapter" data-level="29.10" data-path="dsi-allCells-allSetsTop.html"><a href="dsi-allCells-allSetsTop.html#session-information-24"><i class="fa fa-check"></i><b>29.10</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html"><i class="fa fa-check"></i><b>30</b> Cell cycle assignment</a><ul>
<li class="chapter" data-level="30.1" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#load-data-7"><i class="fa fa-check"></i><b>30.1</b> Load data</a></li>
<li class="chapter" data-level="30.2" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#motivation-11"><i class="fa fa-check"></i><b>30.2</b> Motivation</a></li>
<li class="chapter" data-level="30.3" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclins"><i class="fa fa-check"></i><b>30.3</b> Using the cyclins</a></li>
<li class="chapter" data-level="30.4" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#using-the-cyclone-classifier"><i class="fa fa-check"></i><b>30.4</b> Using the cyclone() classifier</a></li>
<li class="chapter" data-level="30.5" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#regressing-out-cell-cycle-phase"><i class="fa fa-check"></i><b>30.5</b> Regressing out cell cycle phase</a></li>
<li class="chapter" data-level="30.6" data-path="cellCyclePhasesTop.html"><a href="cellCyclePhasesTop.html#session-information-25"><i class="fa fa-check"></i><b>30.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html"><i class="fa fa-check"></i><b>31</b> Doublet detection</a><ul>
<li class="chapter" data-level="31.1" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#learning-objectives-9"><i class="fa fa-check"></i><b>31.1</b> Learning objectives</a></li>
<li class="chapter" data-level="31.2" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#set-up-analysis"><i class="fa fa-check"></i><b>31.2</b> Set up analysis</a></li>
<li class="chapter" data-level="31.3" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#load-data-8"><i class="fa fa-check"></i><b>31.3</b> Load data</a></li>
<li class="chapter" data-level="31.4" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#overview-1"><i class="fa fa-check"></i><b>31.4</b> Overview</a></li>
<li class="chapter" data-level="31.5" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#doublet-detection-by-simulation"><i class="fa fa-check"></i><b>31.5</b> Doublet detection by simulation</a></li>
<li class="chapter" data-level="31.6" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#further-comments"><i class="fa fa-check"></i><b>31.6</b> Further comments</a></li>
<li class="chapter" data-level="31.7" data-path="doubletDetectionTop.html"><a href="doubletDetectionTop.html#session-information-26"><i class="fa fa-check"></i><b>31.7</b> Session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dsi-allCells-allSetsTop" class="section level1">
<h1><span class="header-section-number">Chapter 29</span> Data integration - all Caron sample types</h1>
<div id="learning-objectives-8" class="section level2">
<h2><span class="header-section-number">29.1</span> Learning objectives</h2>
<style>
div.blue {background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<ul>
<li>bla</li>
<li>bla</li>
<li>bla</li>
</ul>
</div>
<div class="sourceCode" id="cb2368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2368-1"><a href="dsi-allCells-allSetsTop.html#cb2368-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb2368-2"><a href="dsi-allCells-allSetsTop.html#cb2368-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb2368-3"><a href="dsi-allCells-allSetsTop.html#cb2368-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb2368-4"><a href="dsi-allCells-allSetsTop.html#cb2368-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb2368-5"><a href="dsi-allCells-allSetsTop.html#cb2368-5"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb2368-6"><a href="dsi-allCells-allSetsTop.html#cb2368-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb2368-7"><a href="dsi-allCells-allSetsTop.html#cb2368-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb2368-8"><a href="dsi-allCells-allSetsTop.html#cb2368-8"></a><span class="cf">if</span>(<span class="kw">exists</span>(<span class="st">&quot;isChild&quot;</span>)) { setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span> }</span>
<span id="cb2368-9"><a href="dsi-allCells-allSetsTop.html#cb2368-9"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb2368-10"><a href="dsi-allCells-allSetsTop.html#cb2368-10"></a></span>
<span id="cb2368-11"><a href="dsi-allCells-allSetsTop.html#cb2368-11"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb2368-12"><a href="dsi-allCells-allSetsTop.html#cb2368-12"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb2368-13"><a href="dsi-allCells-allSetsTop.html#cb2368-13"></a>    splSetToGet &lt;-<span class="st"> &quot;dummy&quot;</span></span>
<span id="cb2368-14"><a href="dsi-allCells-allSetsTop.html#cb2368-14"></a>    setSuf &lt;-<span class="st"> &quot;_allCells&quot;</span></span>
<span id="cb2368-15"><a href="dsi-allCells-allSetsTop.html#cb2368-15"></a>}</span>
<span id="cb2368-16"><a href="dsi-allCells-allSetsTop.html#cb2368-16"></a></span>
<span id="cb2368-17"><a href="dsi-allCells-allSetsTop.html#cb2368-17"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb2368-18"><a href="dsi-allCells-allSetsTop.html#cb2368-18"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb2368-19"><a href="dsi-allCells-allSetsTop.html#cb2368-19"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb2368-20"><a href="dsi-allCells-allSetsTop.html#cb2368-20"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span>
<span id="cb2368-21"><a href="dsi-allCells-allSetsTop.html#cb2368-21"></a></span>
<span id="cb2368-22"><a href="dsi-allCells-allSetsTop.html#cb2368-22"></a><span class="kw">library</span>(BiocParallel)</span>
<span id="cb2368-23"><a href="dsi-allCells-allSetsTop.html#cb2368-23"></a>bpp &lt;-<span class="st"> </span><span class="kw">MulticoreParam</span>(<span class="dv">8</span>)</span></code></pre></div>
<p>Source: <a href="https://osca.bioconductor.org/integrating-datasets.html">Integrating Datasets</a> of the OSCA book and the <a href="https://rdrr.io/github/LTLA/batchelor/man/fastMNN.html">fastMNN manual</a>.</p>
</div>
<div id="motivation-10" class="section level2">
<h2><span class="header-section-number">29.2</span> Motivation</h2>
<p>Large single-cell RNA sequencing (scRNA-seq) projects usually need to generate data across multiple batches due to logistical constraints. However, the processing of different batches is often subject to uncontrollable differences, e.g., changes in operator, differences in reagent quality. This results in systematic differences in the observed expression in cells from different batches, which we refer to as “batch effects”. Batch effects are problematic as they can be major drivers of heterogeneity in the data, masking the relevant biological differences and complicating interpretation of the results.</p>
<p>Computational correction of these effects is critical for eliminating batch-to-batch variation, allowing data across multiple batches to be combined for common downstream analysis. However, existing methods based on linear models (Ritchie et al. 2015; Leek et al. 2012) assume that the composition of cell populations are either known or the same across batches. To overcome these limitations, bespoke methods have been developed for batch correction of single-cell data (Haghverdi et al. 2018; Butler et al. 2018; Lin et al. 2019) that do not require a priori knowledge about the composition of the population. This allows them to be used in workflows for exploratory analyses of scRNA-seq data where such knowledge is usually unavailable.</p>
</div>
<div id="load-the-data-5" class="section level2">
<h2><span class="header-section-number">29.3</span> Load the data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts.</p>
<div class="sourceCode" id="cb2369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2369-1"><a href="dsi-allCells-allSetsTop.html#cb2369-1"></a><span class="co">#setName &lt;- &quot;caron&quot;</span></span>
<span id="cb2369-2"><a href="dsi-allCells-allSetsTop.html#cb2369-2"></a><span class="co"># Read object in:</span></span>
<span id="cb2369-3"><a href="dsi-allCells-allSetsTop.html#cb2369-3"></a><span class="co">#setSuf &lt;- &quot;&quot;</span></span>
<span id="cb2369-4"><a href="dsi-allCells-allSetsTop.html#cb2369-4"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds&quot;</span>,</span>
<span id="cb2369-5"><a href="dsi-allCells-allSetsTop.html#cb2369-5"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb2369-6"><a href="dsi-allCells-allSetsTop.html#cb2369-6"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/AnaCourse1/Robjects/caron_sce_nz_postDeconv_allCells.Rds&quot;</code></pre>
<div class="sourceCode" id="cb2371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2371-1"><a href="dsi-allCells-allSetsTop.html#cb2371-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb2371-2"><a href="dsi-allCells-allSetsTop.html#cb2371-2"></a>{</span>
<span id="cb2371-3"><a href="dsi-allCells-allSetsTop.html#cb2371-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb2371-4"><a href="dsi-allCells-allSetsTop.html#cb2371-4"></a>}</span>
<span id="cb2371-5"><a href="dsi-allCells-allSetsTop.html#cb2371-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb2371-6"><a href="dsi-allCells-allSetsTop.html#cb2371-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(2): counts logcounts
## rownames(18431): ENSG00000238009 ENSG00000237491 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(17): Sample Barcode ... cell_sparsity sizeFactor
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb2373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2373-1"><a href="dsi-allCells-allSetsTop.html#cb2373-1"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span> <span class="co"># to avoid error later</span></span>
<span id="cb2373-2"><a href="dsi-allCells-allSetsTop.html#cb2373-2"></a></span>
<span id="cb2373-3"><a href="dsi-allCells-allSetsTop.html#cb2373-3"></a><span class="co">#head(rowData(sce))</span></span>
<span id="cb2373-4"><a href="dsi-allCells-allSetsTop.html#cb2373-4"></a><span class="co">#head(colData(sce))</span></span>
<span id="cb2373-5"><a href="dsi-allCells-allSetsTop.html#cb2373-5"></a><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>block)</span></code></pre></div>
<pre><code>## 
##      ABMMC ETV6-RUNX1        HHD      PBMMC      PRE-T 
##          0      19488       9988      10885       7469</code></pre>
<div class="sourceCode" id="cb2375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2375-1"><a href="dsi-allCells-allSetsTop.html#cb2375-1"></a><span class="co">#assayNames(sce)</span></span>
<span id="cb2375-2"><a href="dsi-allCells-allSetsTop.html#cb2375-2"></a><span class="co">#reducedDimNames(sce)</span></span></code></pre></div>
<p>Read in the sample sheet:</p>
<div class="sourceCode" id="cb2376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2376-1"><a href="dsi-allCells-allSetsTop.html#cb2376-1"></a><span class="co"># CaronBourque2020</span></span>
<span id="cb2376-2"><a href="dsi-allCells-allSetsTop.html#cb2376-2"></a>cb_sampleSheetFn &lt;-<span class="st"> </span><span class="kw">file.path</span>(projDir, <span class="st">&quot;Data/CaronBourque2020/SraRunTable.txt&quot;</span>)</span>
<span id="cb2376-3"><a href="dsi-allCells-allSetsTop.html#cb2376-3"></a>cb_sampleSheet &lt;-<span class="st"> </span><span class="kw">read.table</span>(cb_sampleSheetFn, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb2376-4"><a href="dsi-allCells-allSetsTop.html#cb2376-4"></a>cb_sampleSheet &lt;-<span class="st">  </span>cb_sampleSheet <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>Run <span class="op">==</span><span class="st"> &quot;SRR9264351&quot;</span>)</span>
<span id="cb2376-5"><a href="dsi-allCells-allSetsTop.html#cb2376-5"></a>cb_sampleSheet[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,]</span></code></pre></div>
<pre><code>##          Run Assay.Type AvgSpotLen       Bases  BioProject    BioSample
## 1 SRR9264343    RNA-Seq        132 27850288884 PRJNA548203 SAMN12011162
## 2 SRR9264344    RNA-Seq        132 43613421192 PRJNA548203 SAMN12011172
##         Bytes                                                      Cell_type
## 1 18644549905 Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
## 2 27638885644 Pre-B t(12;21) [ETV6-RUNX1] acute lymphoblastic leukemia cells
##   Center.Name Consent DATASTORE.filetype DATASTORE.provider
## 1         GEO  public          fastq,sra         gs,ncbi,s3
## 2         GEO  public          fastq,sra         gs,ncbi,s3
##                 DATASTORE.region                          disease_state
## 1 gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
## 2 gs.US,ncbi.public,s3.us-east-1 Childhood acute lymphoblastic leukemia
##   Experiment GEO_Accession..exp.          Instrument LibraryLayout
## 1 SRX6034681          GSM3872434 Illumina HiSeq 4000        PAIRED
## 2 SRX6034682          GSM3872435 Illumina HiSeq 4000        PAIRED
##   LibrarySelection  LibrarySource     Organism Platform          ReleaseDate
## 1             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
## 2             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA 2020-02-14T00:00:00Z
##   Sample.Name source_name SRA.Study
## 1  GSM3872434  ETV6-RUNX1 SRP201012
## 2  GSM3872435  ETV6-RUNX1 SRP201012</code></pre>
<p>Have informative sample names:</p>
<div class="sourceCode" id="cb2378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2378-1"><a href="dsi-allCells-allSetsTop.html#cb2378-1"></a>cb_sampleSheet<span class="op">$</span>Sample.Name2 &lt;-<span class="st"> &quot;NA&quot;</span></span>
<span id="cb2378-2"><a href="dsi-allCells-allSetsTop.html#cb2378-2"></a>typeVec &lt;-<span class="st"> </span><span class="kw">unique</span>(cb_sampleSheet<span class="op">$</span>source_name)</span>
<span id="cb2378-3"><a href="dsi-allCells-allSetsTop.html#cb2378-3"></a><span class="cf">for</span> (tx <span class="cf">in</span> typeVec)</span>
<span id="cb2378-4"><a href="dsi-allCells-allSetsTop.html#cb2378-4"></a>{</span>
<span id="cb2378-5"><a href="dsi-allCells-allSetsTop.html#cb2378-5"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(cb_sampleSheet<span class="op">$</span>source_name <span class="op">==</span><span class="st"> </span>tx)</span>
<span id="cb2378-6"><a href="dsi-allCells-allSetsTop.html#cb2378-6"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(tmpInd))</span>
<span id="cb2378-7"><a href="dsi-allCells-allSetsTop.html#cb2378-7"></a>    {</span>
<span id="cb2378-8"><a href="dsi-allCells-allSetsTop.html#cb2378-8"></a>        cb_sampleSheet<span class="op">$</span>Sample.Name2[tmpInd[i]] &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s_%s&quot;</span>, tx, i)</span>
<span id="cb2378-9"><a href="dsi-allCells-allSetsTop.html#cb2378-9"></a>    }</span>
<span id="cb2378-10"><a href="dsi-allCells-allSetsTop.html#cb2378-10"></a>}</span>
<span id="cb2378-11"><a href="dsi-allCells-allSetsTop.html#cb2378-11"></a></span>
<span id="cb2378-12"><a href="dsi-allCells-allSetsTop.html#cb2378-12"></a><span class="kw">colData</span>(sce)<span class="op">$</span>Sample.Name2 &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb2378-13"><a href="dsi-allCells-allSetsTop.html#cb2378-13"></a><span class="st">    </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2378-14"><a href="dsi-allCells-allSetsTop.html#cb2378-14"></a><span class="st">    </span><span class="kw">left_join</span>( cb_sampleSheet, <span class="dt">by=</span><span class="st">&quot;Sample.Name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb2378-15"><a href="dsi-allCells-allSetsTop.html#cb2378-15"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name2)</span>
<span id="cb2378-16"><a href="dsi-allCells-allSetsTop.html#cb2378-16"></a></span>
<span id="cb2378-17"><a href="dsi-allCells-allSetsTop.html#cb2378-17"></a>splVec &lt;-<span class="st"> </span>cb_sampleSheet <span class="op">%&gt;%</span></span>
<span id="cb2378-18"><a href="dsi-allCells-allSetsTop.html#cb2378-18"></a><span class="st">    </span><span class="co">#filter(source_name == &quot;ETV6-RUNX1&quot;) %&gt;%</span></span>
<span id="cb2378-19"><a href="dsi-allCells-allSetsTop.html#cb2378-19"></a><span class="st">    </span><span class="kw">pull</span>(Sample.Name2)</span>
<span id="cb2378-20"><a href="dsi-allCells-allSetsTop.html#cb2378-20"></a>splVec</span></code></pre></div>
<pre><code>##  [1] &quot;ETV6-RUNX1_1&quot; &quot;ETV6-RUNX1_2&quot; &quot;ETV6-RUNX1_3&quot; &quot;ETV6-RUNX1_4&quot; &quot;HHD_1&quot;       
##  [6] &quot;HHD_2&quot;        &quot;PRE-T_1&quot;      &quot;PRE-T_2&quot;      &quot;PBMMC_1&quot;      &quot;PBMMC_2&quot;     
## [11] &quot;PBMMC_3&quot;</code></pre>
<!--
Subset cells (e.g 500 or 1000 per sample for a faster analysis):
-->
<div class="sourceCode" id="cb2380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2380-1"><a href="dsi-allCells-allSetsTop.html#cb2380-1"></a><span class="co"># mind we now have a downsampled set to use all along</span></span>
<span id="cb2380-2"><a href="dsi-allCells-allSetsTop.html#cb2380-2"></a><span class="co"># so avoid doing it again</span></span>
<span id="cb2380-3"><a href="dsi-allCells-allSetsTop.html#cb2380-3"></a><span class="co"># also, changes setsuf to shorter form eg _5hCps, which we could use from the start </span><span class="al">TODO</span></span>
<span id="cb2380-4"><a href="dsi-allCells-allSetsTop.html#cb2380-4"></a></span>
<span id="cb2380-5"><a href="dsi-allCells-allSetsTop.html#cb2380-5"></a>all.sce &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb2380-6"><a href="dsi-allCells-allSetsTop.html#cb2380-6"></a></span>
<span id="cb2380-7"><a href="dsi-allCells-allSetsTop.html#cb2380-7"></a><span class="cf">if</span>(setSuf <span class="op">==</span><span class="st"> &quot;_allCells&quot;</span> <span class="op">|</span><span class="st"> </span>setSuf <span class="op">==</span><span class="st"> &quot;_5hCellPerSpl&quot;</span>)</span>
<span id="cb2380-8"><a href="dsi-allCells-allSetsTop.html#cb2380-8"></a>{</span>
<span id="cb2380-9"><a href="dsi-allCells-allSetsTop.html#cb2380-9"></a>  <span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb2380-10"><a href="dsi-allCells-allSetsTop.html#cb2380-10"></a>  {</span>
<span id="cb2380-11"><a href="dsi-allCells-allSetsTop.html#cb2380-11"></a>    vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb2380-12"><a href="dsi-allCells-allSetsTop.html#cb2380-12"></a><span class="st">        </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2380-13"><a href="dsi-allCells-allSetsTop.html#cb2380-13"></a><span class="st">        </span><span class="kw">filter</span>(Sample.Name2 <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb2380-14"><a href="dsi-allCells-allSetsTop.html#cb2380-14"></a><span class="st">        </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb2380-15"><a href="dsi-allCells-allSetsTop.html#cb2380-15"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb2380-16"><a href="dsi-allCells-allSetsTop.html#cb2380-16"></a>    all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb2380-17"><a href="dsi-allCells-allSetsTop.html#cb2380-17"></a>  }</span>
<span id="cb2380-18"><a href="dsi-allCells-allSetsTop.html#cb2380-18"></a>} <span class="cf">else</span> {  </span>
<span id="cb2380-19"><a href="dsi-allCells-allSetsTop.html#cb2380-19"></a>  nbCells &lt;-<span class="st"> </span><span class="dv">500</span></span>
<span id="cb2380-20"><a href="dsi-allCells-allSetsTop.html#cb2380-20"></a>  setSuf &lt;-<span class="st"> &quot;_5hCps&quot;</span> <span class="co"># &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb2380-21"><a href="dsi-allCells-allSetsTop.html#cb2380-21"></a></span>
<span id="cb2380-22"><a href="dsi-allCells-allSetsTop.html#cb2380-22"></a>  <span class="co">#nbCells &lt;- 1000</span></span>
<span id="cb2380-23"><a href="dsi-allCells-allSetsTop.html#cb2380-23"></a>  <span class="co">#setSuf &lt;- &quot;_1kCps&quot; # &quot;_1kCellPerSpl&quot;</span></span>
<span id="cb2380-24"><a href="dsi-allCells-allSetsTop.html#cb2380-24"></a></span>
<span id="cb2380-25"><a href="dsi-allCells-allSetsTop.html#cb2380-25"></a>  <span class="cf">for</span>(spx <span class="cf">in</span> splVec)</span>
<span id="cb2380-26"><a href="dsi-allCells-allSetsTop.html#cb2380-26"></a>  {</span>
<span id="cb2380-27"><a href="dsi-allCells-allSetsTop.html#cb2380-27"></a>    vec.bc &lt;-<span class="st"> </span><span class="kw">colData</span>(sce) <span class="op">%&gt;%</span></span>
<span id="cb2380-28"><a href="dsi-allCells-allSetsTop.html#cb2380-28"></a><span class="st">        </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2380-29"><a href="dsi-allCells-allSetsTop.html#cb2380-29"></a><span class="st">        </span><span class="kw">filter</span>(Sample.Name2 <span class="op">==</span><span class="st"> </span>spx) <span class="op">%&gt;%</span></span>
<span id="cb2380-30"><a href="dsi-allCells-allSetsTop.html#cb2380-30"></a><span class="st">        </span><span class="kw">slice_sample</span>(<span class="dt">n=</span>nbCells) <span class="op">%&gt;%</span></span>
<span id="cb2380-31"><a href="dsi-allCells-allSetsTop.html#cb2380-31"></a><span class="st">        </span><span class="kw">pull</span>(Barcode)</span>
<span id="cb2380-32"><a href="dsi-allCells-allSetsTop.html#cb2380-32"></a>    tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Barcode <span class="op">%in%</span><span class="st"> </span>vec.bc)</span>
<span id="cb2380-33"><a href="dsi-allCells-allSetsTop.html#cb2380-33"></a>    all.sce[[spx]] &lt;-<span class="st"> </span>sce[,tmpInd]</span>
<span id="cb2380-34"><a href="dsi-allCells-allSetsTop.html#cb2380-34"></a>  }</span>
<span id="cb2380-35"><a href="dsi-allCells-allSetsTop.html#cb2380-35"></a>}</span>
<span id="cb2380-36"><a href="dsi-allCells-allSetsTop.html#cb2380-36"></a></span>
<span id="cb2380-37"><a href="dsi-allCells-allSetsTop.html#cb2380-37"></a><span class="co"># show size of sets:</span></span>
<span id="cb2380-38"><a href="dsi-allCells-allSetsTop.html#cb2380-38"></a><span class="kw">lapply</span>(all.sce, dim)</span></code></pre></div>
<pre><code>## $`ETV6-RUNX1_1`
## [1] 18431  2853
## 
## $`ETV6-RUNX1_2`
## [1] 18431  6615
## 
## $`ETV6-RUNX1_3`
## [1] 18431  4727
## 
## $`ETV6-RUNX1_4`
## [1] 18431  5293
## 
## $HHD_1
## [1] 18431  4551
## 
## $HHD_2
## [1] 18431  5437
## 
## $`PRE-T_1`
## [1] 18431  3841
## 
## $`PRE-T_2`
## [1] 18431  3628
## 
## $PBMMC_1
## [1] 18431  2084
## 
## $PBMMC_2
## [1] 18431  4658
## 
## $PBMMC_3
## [1] 18431  4143</code></pre>
<div class="sourceCode" id="cb2382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2382-1"><a href="dsi-allCells-allSetsTop.html#cb2382-1"></a><span class="co"># set number of samples:</span></span>
<span id="cb2382-2"><a href="dsi-allCells-allSetsTop.html#cb2382-2"></a>nbSpl &lt;-<span class="st"> </span><span class="kw">length</span>(all.sce)</span></code></pre></div>
<p>We will analyse each sample separately, namely:</p>
<ul>
<li>normalise counts</li>
<li>model gene expression variance</li>
<li>identify highly variable genes</li>
<li>perform dimensionality reduction</li>
<li>cluster cells</li>
</ul>
<div class="sourceCode" id="cb2383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2383-1"><a href="dsi-allCells-allSetsTop.html#cb2383-1"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb2383-2"><a href="dsi-allCells-allSetsTop.html#cb2383-2"></a>all.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, logNormCounts)</span>
<span id="cb2383-3"><a href="dsi-allCells-allSetsTop.html#cb2383-3"></a></span>
<span id="cb2383-4"><a href="dsi-allCells-allSetsTop.html#cb2383-4"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb2383-5"><a href="dsi-allCells-allSetsTop.html#cb2383-5"></a><span class="kw">library</span>(scran)</span>
<span id="cb2383-6"><a href="dsi-allCells-allSetsTop.html#cb2383-6"></a>all.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, modelGeneVar)</span>
<span id="cb2383-7"><a href="dsi-allCells-allSetsTop.html#cb2383-7"></a>all.hvgs &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, getTopHVGs, <span class="dt">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb2383-8"><a href="dsi-allCells-allSetsTop.html#cb2383-8"></a></span>
<span id="cb2383-9"><a href="dsi-allCells-allSetsTop.html#cb2383-9"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb2383-10"><a href="dsi-allCells-allSetsTop.html#cb2383-10"></a><span class="kw">set.seed</span>(<span class="dv">10000</span>)</span>
<span id="cb2383-11"><a href="dsi-allCells-allSetsTop.html#cb2383-11"></a>all.sce &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="dt">FUN=</span>runPCA, <span class="dt">x=</span>all.sce, <span class="dt">subset_row=</span>all.hvgs, </span>
<span id="cb2383-12"><a href="dsi-allCells-allSetsTop.html#cb2383-12"></a>    <span class="dt">MoreArgs=</span><span class="kw">list</span>(<span class="dt">ncomponents=</span><span class="dv">25</span>, <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>()), </span>
<span id="cb2383-13"><a href="dsi-allCells-allSetsTop.html#cb2383-13"></a>    <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb2383-14"><a href="dsi-allCells-allSetsTop.html#cb2383-14"></a></span>
<span id="cb2383-15"><a href="dsi-allCells-allSetsTop.html#cb2383-15"></a><span class="co"># TSNE</span></span>
<span id="cb2383-16"><a href="dsi-allCells-allSetsTop.html#cb2383-16"></a><span class="co">#set.seed(100000)</span></span>
<span id="cb2383-17"><a href="dsi-allCells-allSetsTop.html#cb2383-17"></a><span class="co">#all.sce &lt;- lapply(all.sce, runTSNE, dimred=&quot;PCA&quot;)</span></span>
<span id="cb2383-18"><a href="dsi-allCells-allSetsTop.html#cb2383-18"></a></span>
<span id="cb2383-19"><a href="dsi-allCells-allSetsTop.html#cb2383-19"></a><span class="co"># UMAP</span></span>
<span id="cb2383-20"><a href="dsi-allCells-allSetsTop.html#cb2383-20"></a><span class="co">#set.seed(1000000)</span></span>
<span id="cb2383-21"><a href="dsi-allCells-allSetsTop.html#cb2383-21"></a><span class="co">#all.sce &lt;- lapply(all.sce, runUMAP, dimred=&quot;PCA&quot;)</span></span>
<span id="cb2383-22"><a href="dsi-allCells-allSetsTop.html#cb2383-22"></a></span>
<span id="cb2383-23"><a href="dsi-allCells-allSetsTop.html#cb2383-23"></a><span class="co">#--- clustering ---#</span></span>
<span id="cb2383-24"><a href="dsi-allCells-allSetsTop.html#cb2383-24"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="kw">names</span>(all.sce)) {</span>
<span id="cb2383-25"><a href="dsi-allCells-allSetsTop.html#cb2383-25"></a>    g &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(all.sce[[n]], <span class="dt">k=</span><span class="dv">10</span>, <span class="dt">use.dimred=</span><span class="st">&#39;PCA&#39;</span>, <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2383-26"><a href="dsi-allCells-allSetsTop.html#cb2383-26"></a>    clust &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_walktrap</span>(g)<span class="op">$</span>membership</span>
<span id="cb2383-27"><a href="dsi-allCells-allSetsTop.html#cb2383-27"></a>    <span class="kw">colLabels</span>(all.sce[[n]])  &lt;-<span class="st"> </span><span class="kw">factor</span>(clust)</span>
<span id="cb2383-28"><a href="dsi-allCells-allSetsTop.html#cb2383-28"></a>}</span></code></pre></div>
<p>To prepare for the batch correction, we need to:</p>
<ul>
<li>subset all batches to the common “universe” of features</li>
<li>rescale each batch to adjust for differences in sequencing depth between batches</li>
<li>perform feature selection by averaging the variance components across all batches</li>
</ul>
<p>We subset all batches to the common “universe” of features. In this case, it is straightforward as batches use Ensembl gene annotation.</p>
<div class="sourceCode" id="cb2384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2384-1"><a href="dsi-allCells-allSetsTop.html#cb2384-1"></a>allNames &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){<span class="kw">rownames</span>(x)}))</span>
<span id="cb2384-2"><a href="dsi-allCells-allSetsTop.html#cb2384-2"></a>allNamesNb &lt;-<span class="st"> </span><span class="kw">table</span>(allNames)</span>
<span id="cb2384-3"><a href="dsi-allCells-allSetsTop.html#cb2384-3"></a>universe &lt;-<span class="st"> </span><span class="kw">names</span>(allNamesNb)[allNamesNb<span class="op">==</span><span class="kw">length</span>(splVec)] </span>
<span id="cb2384-4"><a href="dsi-allCells-allSetsTop.html#cb2384-4"></a><span class="kw">length</span>(universe)</span></code></pre></div>
<pre><code>## [1] 18431</code></pre>
<div class="sourceCode" id="cb2386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2386-1"><a href="dsi-allCells-allSetsTop.html#cb2386-1"></a><span class="co"># Subsetting the SingleCellExperiment object.</span></span>
<span id="cb2386-2"><a href="dsi-allCells-allSetsTop.html#cb2386-2"></a>uni.sce &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.sce, <span class="cf">function</span>(x){x[universe,]})</span>
<span id="cb2386-3"><a href="dsi-allCells-allSetsTop.html#cb2386-3"></a><span class="co"># Also subsetting the variance modelling results, for convenience.</span></span>
<span id="cb2386-4"><a href="dsi-allCells-allSetsTop.html#cb2386-4"></a>uni.dec &lt;-<span class="st"> </span><span class="kw">lapply</span>(all.dec, <span class="cf">function</span>(x){x[universe,]})</span></code></pre></div>
<p>We rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.</p>
<div class="sourceCode" id="cb2387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2387-1"><a href="dsi-allCells-allSetsTop.html#cb2387-1"></a><span class="kw">library</span>(batchelor)</span>
<span id="cb2387-2"><a href="dsi-allCells-allSetsTop.html#cb2387-2"></a>rescaled.mbn &lt;-<span class="st"> </span><span class="kw">multiBatchNorm</span>(uni.sce, <span class="dt">batch =</span> <span class="st">&quot;Sample.Name2&quot;</span>)</span></code></pre></div>
<p>We perform feature selection by averaging the variance components across all batches with the combineVar() function. We compute the average as it is responsive to batch-specific HVGs while still preserving the within-batch ranking of genes. This allows us to use the same strategies described in Section 8.3 to select genes of interest. In contrast, approaches based on taking the intersection or union of HVGs across batches become increasingly conservative or liberal, respectively, with an increasing number of batches.</p>
<div class="sourceCode" id="cb2388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2388-1"><a href="dsi-allCells-allSetsTop.html#cb2388-1"></a><span class="kw">library</span>(scran)</span>
<span id="cb2388-2"><a href="dsi-allCells-allSetsTop.html#cb2388-2"></a>combined.dec &lt;-<span class="st"> </span><span class="kw">combineVar</span>(</span>
<span id="cb2388-3"><a href="dsi-allCells-allSetsTop.html#cb2388-3"></a>  uni.dec[[<span class="dv">1</span>]], uni.dec[[<span class="dv">2</span>]], uni.dec[[<span class="dv">3</span>]], uni.dec[[<span class="dv">4</span>]],</span>
<span id="cb2388-4"><a href="dsi-allCells-allSetsTop.html#cb2388-4"></a>  uni.dec[[<span class="dv">5</span>]], uni.dec[[<span class="dv">6</span>]],</span>
<span id="cb2388-5"><a href="dsi-allCells-allSetsTop.html#cb2388-5"></a>  uni.dec[[<span class="dv">7</span>]], uni.dec[[<span class="dv">8</span>]],</span>
<span id="cb2388-6"><a href="dsi-allCells-allSetsTop.html#cb2388-6"></a>  uni.dec[[<span class="dv">9</span>]], uni.dec[[<span class="dv">10</span>]], uni.dec[[<span class="dv">11</span>]]</span>
<span id="cb2388-7"><a href="dsi-allCells-allSetsTop.html#cb2388-7"></a>  )</span>
<span id="cb2388-8"><a href="dsi-allCells-allSetsTop.html#cb2388-8"></a>chosen.hvgs &lt;-<span class="st"> </span>combined.dec<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb2388-9"><a href="dsi-allCells-allSetsTop.html#cb2388-9"></a><span class="kw">sum</span>(chosen.hvgs)</span></code></pre></div>
<pre><code>## [1] 12466</code></pre>
<p>When integrating datasets of variable composition, it is generally safer to err on the side of including more genes than are used in a single dataset analysis, to ensure that markers are retained for any dataset-specific subpopulations that might be present. For a top X selection, this means using a larger X (say, ~5000), or in this case, we simply take all genes above the trend. That said, many of the signal-to-noise considerations described in Section 8.3 still apply here, so some experimentation may be necessary for best results.</p>
<p>Alternatively, a more forceful approach to feature selection can be used based on marker genes from within-batch comparisons.</p>
</div>
<div id="diagnosing-batch-effects-9" class="section level2">
<h2><span class="header-section-number">29.4</span> Diagnosing batch effects</h2>
<p>Before we actually perform any correction, it is worth examining whether there is any batch effect in this dataset. We combine the SingleCellExperiments and perform a PCA on the log-expression values for all genes with positive (average) biological components.</p>
<div class="sourceCode" id="cb2390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2390-1"><a href="dsi-allCells-allSetsTop.html#cb2390-1"></a><span class="co"># Synchronizing the metadata for cbind()ing.</span></span>
<span id="cb2390-2"><a href="dsi-allCells-allSetsTop.html#cb2390-2"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[2]]))</span></span>
<span id="cb2390-3"><a href="dsi-allCells-allSetsTop.html#cb2390-3"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[3]]))</span></span>
<span id="cb2390-4"><a href="dsi-allCells-allSetsTop.html#cb2390-4"></a><span class="co">#identical(rowData(rescaled.mbn[[1]]), rowData(rescaled.mbn[[4]]))</span></span>
<span id="cb2390-5"><a href="dsi-allCells-allSetsTop.html#cb2390-5"></a></span>
<span id="cb2390-6"><a href="dsi-allCells-allSetsTop.html#cb2390-6"></a>rescaled2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(rescaled.mbn, <span class="cf">function</span>(x){x<span class="op">$</span>batch &lt;-<span class="st"> </span>x<span class="op">$</span>Sample.Name2; x})</span>
<span id="cb2390-7"><a href="dsi-allCells-allSetsTop.html#cb2390-7"></a>rescaled.mbn &lt;-<span class="st"> </span>rescaled2</span>
<span id="cb2390-8"><a href="dsi-allCells-allSetsTop.html#cb2390-8"></a><span class="kw">rm</span>(rescaled2)</span>
<span id="cb2390-9"><a href="dsi-allCells-allSetsTop.html#cb2390-9"></a></span>
<span id="cb2390-10"><a href="dsi-allCells-allSetsTop.html#cb2390-10"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">do.call</span>(cbind, rescaled.mbn)</span>
<span id="cb2390-11"><a href="dsi-allCells-allSetsTop.html#cb2390-11"></a></span>
<span id="cb2390-12"><a href="dsi-allCells-allSetsTop.html#cb2390-12"></a><span class="co"># Using RandomParam() as it is more efficient for file-backed matrices.</span></span>
<span id="cb2390-13"><a href="dsi-allCells-allSetsTop.html#cb2390-13"></a><span class="kw">library</span>(scater)</span>
<span id="cb2390-14"><a href="dsi-allCells-allSetsTop.html#cb2390-14"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb2390-15"><a href="dsi-allCells-allSetsTop.html#cb2390-15"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runPCA</span>(</span>
<span id="cb2390-16"><a href="dsi-allCells-allSetsTop.html#cb2390-16"></a>  uncorrected,</span>
<span id="cb2390-17"><a href="dsi-allCells-allSetsTop.html#cb2390-17"></a>  <span class="dt">subset_row=</span>chosen.hvgs,</span>
<span id="cb2390-18"><a href="dsi-allCells-allSetsTop.html#cb2390-18"></a>  <span class="dt">BSPARAM=</span>BiocSingular<span class="op">::</span><span class="kw">RandomParam</span>())</span>
<span id="cb2390-19"><a href="dsi-allCells-allSetsTop.html#cb2390-19"></a></span>
<span id="cb2390-20"><a href="dsi-allCells-allSetsTop.html#cb2390-20"></a><span class="co"># adjust walk.trap number of steps to that of cells</span></span>
<span id="cb2390-21"><a href="dsi-allCells-allSetsTop.html#cb2390-21"></a>nSteps &lt;-<span class="st"> </span><span class="dv">4</span> <span class="co"># walktrap default</span></span>
<span id="cb2390-22"><a href="dsi-allCells-allSetsTop.html#cb2390-22"></a><span class="cf">if</span>(<span class="kw">ncol</span>(uncorrected) <span class="op">&gt;</span><span class="st"> </span><span class="dv">10000</span>){</span>
<span id="cb2390-23"><a href="dsi-allCells-allSetsTop.html#cb2390-23"></a>    nSteps &lt;-<span class="st"> </span><span class="dv">20</span> <span class="co"># or else get 63 clusters with caron allCells</span></span>
<span id="cb2390-24"><a href="dsi-allCells-allSetsTop.html#cb2390-24"></a>}</span></code></pre></div>
<p>We use graph-based clustering on the components to obtain a summary of the population structure. As our each sample group is represented by at least two replicates, each cluster should ideally consist of cells from several batches. This is the case for some but not all clusters. Some clusters comprise of cells from a single sample. This may indicate that cells of the same type are artificially separated due to technical differences between batches. They may also be cancer cell population private to samples.</p>
<div class="sourceCode" id="cb2391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2391-1"><a href="dsi-allCells-allSetsTop.html#cb2391-1"></a><span class="co"># 30+ min run</span></span>
<span id="cb2391-2"><a href="dsi-allCells-allSetsTop.html#cb2391-2"></a><span class="co"># see clusterRows below for faster run</span></span>
<span id="cb2391-3"><a href="dsi-allCells-allSetsTop.html#cb2391-3"></a></span>
<span id="cb2391-4"><a href="dsi-allCells-allSetsTop.html#cb2391-4"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2391-5"><a href="dsi-allCells-allSetsTop.html#cb2391-5"></a></span>
<span id="cb2391-6"><a href="dsi-allCells-allSetsTop.html#cb2391-6"></a><span class="kw">library</span>(scran)</span>
<span id="cb2391-7"><a href="dsi-allCells-allSetsTop.html#cb2391-7"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(uncorrected,</span>
<span id="cb2391-8"><a href="dsi-allCells-allSetsTop.html#cb2391-8"></a>            <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb2391-9"><a href="dsi-allCells-allSetsTop.html#cb2391-9"></a>            <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2391-10"><a href="dsi-allCells-allSetsTop.html#cb2391-10"></a><span class="co">#clusters &lt;- igraph::cluster_walktrap(snn.gr, steps=nSteps)$membership # slow</span></span>
<span id="cb2391-11"><a href="dsi-allCells-allSetsTop.html#cb2391-11"></a>clusters &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_fast_greedy</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb2391-12"><a href="dsi-allCells-allSetsTop.html#cb2391-12"></a></span>
<span id="cb2391-13"><a href="dsi-allCells-allSetsTop.html#cb2391-13"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
##  50.966   4.818  39.766</code></pre>
<div class="sourceCode" id="cb2393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2393-1"><a href="dsi-allCells-allSetsTop.html#cb2393-1"></a><span class="co">#tab &lt;- table(Cluster=clusters, Batch=uncorrected$batch)</span></span>
<span id="cb2393-2"><a href="dsi-allCells-allSetsTop.html#cb2393-2"></a><span class="co">#tab</span></span>
<span id="cb2393-3"><a href="dsi-allCells-allSetsTop.html#cb2393-3"></a></span>
<span id="cb2393-4"><a href="dsi-allCells-allSetsTop.html#cb2393-4"></a>df.uncorr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters, <span class="st">&quot;batch&quot;</span>=uncorrected<span class="op">$</span>batch)</span>
<span id="cb2393-5"><a href="dsi-allCells-allSetsTop.html#cb2393-5"></a>tab.uncorr &lt;-<span class="st"> </span><span class="kw">table</span>(df.uncorr)</span></code></pre></div>
<div class="sourceCode" id="cb2394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2394-1"><a href="dsi-allCells-allSetsTop.html#cb2394-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.uncorr,</span>
<span id="cb2394-2"><a href="dsi-allCells-allSetsTop.html#cb2394-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2394-3"><a href="dsi-allCells-allSetsTop.html#cb2394-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2394-4"><a href="dsi-allCells-allSetsTop.html#cb2394-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2394-5"><a href="dsi-allCells-allSetsTop.html#cb2394-5"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagBatch_heatmap_dataSetIntegration_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2395-1"><a href="dsi-allCells-allSetsTop.html#cb2395-1"></a><span class="co"># NNGraphParam k?</span></span>
<span id="cb2395-2"><a href="dsi-allCells-allSetsTop.html#cb2395-2"></a><span class="co"># k is for makeSNNGraph: nearest neighbors to consider during graph construction</span></span>
<span id="cb2395-3"><a href="dsi-allCells-allSetsTop.html#cb2395-3"></a></span>
<span id="cb2395-4"><a href="dsi-allCells-allSetsTop.html#cb2395-4"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2395-5"><a href="dsi-allCells-allSetsTop.html#cb2395-5"></a></span>
<span id="cb2395-6"><a href="dsi-allCells-allSetsTop.html#cb2395-6"></a><span class="kw">set.seed</span>(<span class="dv">1000</span>)</span>
<span id="cb2395-7"><a href="dsi-allCells-allSetsTop.html#cb2395-7"></a>nSteps &lt;-<span class="st"> </span><span class="dv">4</span> <span class="co"># walktrap default</span></span>
<span id="cb2395-8"><a href="dsi-allCells-allSetsTop.html#cb2395-8"></a>clusters &lt;-<span class="st"> </span><span class="kw">clusterRows</span>(<span class="kw">reducedDim</span>(uncorrected, <span class="st">&quot;PCA&quot;</span>),</span>
<span id="cb2395-9"><a href="dsi-allCells-allSetsTop.html#cb2395-9"></a>                      <span class="kw">TwoStepParam</span>(<span class="kw">KmeansParam</span>(<span class="dt">centers=</span><span class="dv">2000</span>,<span class="dt">iter.max=</span><span class="dv">30</span>),</span>
<span id="cb2395-10"><a href="dsi-allCells-allSetsTop.html#cb2395-10"></a>                           <span class="kw">NNGraphParam</span>(</span>
<span id="cb2395-11"><a href="dsi-allCells-allSetsTop.html#cb2395-11"></a>                                <span class="dt">shared =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2395-12"><a href="dsi-allCells-allSetsTop.html#cb2395-12"></a>                                <span class="dt">k=</span><span class="dv">5</span>,</span>
<span id="cb2395-13"><a href="dsi-allCells-allSetsTop.html#cb2395-13"></a>                                <span class="dt">cluster.fun =</span> <span class="st">&quot;walktrap&quot;</span>,</span>
<span id="cb2395-14"><a href="dsi-allCells-allSetsTop.html#cb2395-14"></a>                                <span class="dt">cluster.args =</span> <span class="kw">list</span>(<span class="dt">steps=</span>nSteps)</span>
<span id="cb2395-15"><a href="dsi-allCells-allSetsTop.html#cb2395-15"></a>                                )))</span>
<span id="cb2395-16"><a href="dsi-allCells-allSetsTop.html#cb2395-16"></a></span>
<span id="cb2395-17"><a href="dsi-allCells-allSetsTop.html#cb2395-17"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
##  13.999   0.051  14.068</code></pre>
<div class="sourceCode" id="cb2397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2397-1"><a href="dsi-allCells-allSetsTop.html#cb2397-1"></a><span class="co">##tab2 &lt;- table(Cluster=clusters, Batch=uncorrected$batch)</span></span>
<span id="cb2397-2"><a href="dsi-allCells-allSetsTop.html#cb2397-2"></a><span class="co">##tab2</span></span>
<span id="cb2397-3"><a href="dsi-allCells-allSetsTop.html#cb2397-3"></a></span>
<span id="cb2397-4"><a href="dsi-allCells-allSetsTop.html#cb2397-4"></a>df.uncorr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters, <span class="st">&quot;batch&quot;</span>=uncorrected<span class="op">$</span>batch)</span>
<span id="cb2397-5"><a href="dsi-allCells-allSetsTop.html#cb2397-5"></a>tab.uncorr &lt;-<span class="st"> </span><span class="kw">table</span>(df.uncorr)</span></code></pre></div>
<div class="sourceCode" id="cb2398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2398-1"><a href="dsi-allCells-allSetsTop.html#cb2398-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.uncorr,</span>
<span id="cb2398-2"><a href="dsi-allCells-allSetsTop.html#cb2398-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2398-3"><a href="dsi-allCells-allSetsTop.html#cb2398-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2398-4"><a href="dsi-allCells-allSetsTop.html#cb2398-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2398-5"><a href="dsi-allCells-allSetsTop.html#cb2398-5"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagBatch_blusterHeatmap_dataSetIntegration_allSets-1.png" width="672" /></p>
<p>Cluster size and cell contribution by sample:</p>
<div class="sourceCode" id="cb2399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2399-1"><a href="dsi-allCells-allSetsTop.html#cb2399-1"></a>tmpMat &lt;-<span class="st"> </span>df.uncorr</span>
<span id="cb2399-2"><a href="dsi-allCells-allSetsTop.html#cb2399-2"></a>tmpMatTab &lt;-<span class="st"> </span>tab.uncorr</span>
<span id="cb2399-3"><a href="dsi-allCells-allSetsTop.html#cb2399-3"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb2399-4"><a href="dsi-allCells-allSetsTop.html#cb2399-4"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2399-5"><a href="dsi-allCells-allSetsTop.html#cb2399-5"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2399-6"><a href="dsi-allCells-allSetsTop.html#cb2399-6"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2399-7"><a href="dsi-allCells-allSetsTop.html#cb2399-7"></a><span class="st">    </span><span class="kw">geom_col</span>() <span class="op">+</span></span>
<span id="cb2399-8"><a href="dsi-allCells-allSetsTop.html#cb2399-8"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2399-9"><a href="dsi-allCells-allSetsTop.html#cb2399-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;uncorrected, cell numbers&quot;</span>) <span class="op">+</span></span>
<span id="cb2399-10"><a href="dsi-allCells-allSetsTop.html#cb2399-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2399-11"><a href="dsi-allCells-allSetsTop.html#cb2399-11"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2399-12"><a href="dsi-allCells-allSetsTop.html#cb2399-12"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2399-13"><a href="dsi-allCells-allSetsTop.html#cb2399-13"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent) <span class="op">+</span></span>
<span id="cb2399-14"><a href="dsi-allCells-allSetsTop.html#cb2399-14"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2399-15"><a href="dsi-allCells-allSetsTop.html#cb2399-15"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;uncorrected, proportions&quot;</span>) <span class="op">+</span></span>
<span id="cb2399-16"><a href="dsi-allCells-allSetsTop.html#cb2399-16"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2399-17"><a href="dsi-allCells-allSetsTop.html#cb2399-17"></a><span class="kw">grid.arrange</span>(p1, p2)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagPlotDraw_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2400-1"><a href="dsi-allCells-allSetsTop.html#cb2400-1"></a><span class="co">#legend &lt;- get_legend(p1)</span></span>
<span id="cb2400-2"><a href="dsi-allCells-allSetsTop.html#cb2400-2"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>))</span>
<span id="cb2400-3"><a href="dsi-allCells-allSetsTop.html#cb2400-3"></a><span class="co">#p1 &lt;- p1 + theme(legend.position=&quot;none&quot;)</span></span>
<span id="cb2400-4"><a href="dsi-allCells-allSetsTop.html#cb2400-4"></a><span class="co">#p2 &lt;- p2 + theme(legend.position=&quot;none&quot;)</span></span>
<span id="cb2400-5"><a href="dsi-allCells-allSetsTop.html#cb2400-5"></a><span class="kw">grid.arrange</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2400-6"><a href="dsi-allCells-allSetsTop.html#cb2400-6"></a>                        p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2400-7"><a href="dsi-allCells-allSetsTop.html#cb2400-7"></a>                        legend,</span>
<span id="cb2400-8"><a href="dsi-allCells-allSetsTop.html#cb2400-8"></a>            <span class="dt">ncol=</span><span class="dv">1</span>,</span>
<span id="cb2400-9"><a href="dsi-allCells-allSetsTop.html#cb2400-9"></a>            <span class="dt">heights=</span><span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">2</span>)</span>
<span id="cb2400-10"><a href="dsi-allCells-allSetsTop.html#cb2400-10"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagPlotDraw_dsi_allCells_allSets-2.png" width="672" /></p>
<div class="sourceCode" id="cb2401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2401-1"><a href="dsi-allCells-allSetsTop.html#cb2401-1"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2401-2"><a href="dsi-allCells-allSetsTop.html#cb2401-2"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2401-3"><a href="dsi-allCells-allSetsTop.html#cb2401-3"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2401-4"><a href="dsi-allCells-allSetsTop.html#cb2401-4"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2401-5"><a href="dsi-allCells-allSetsTop.html#cb2401-5"></a>         legend,</span>
<span id="cb2401-6"><a href="dsi-allCells-allSetsTop.html#cb2401-6"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2401-7"><a href="dsi-allCells-allSetsTop.html#cb2401-7"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagPlotDraw_dsi_allCells_allSets-3.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The strong separation between cells from different batches is consistent with the clustering results.</p>
<div class="sourceCode" id="cb2402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2402-1"><a href="dsi-allCells-allSetsTop.html#cb2402-1"></a><span class="kw">set.seed</span>(<span class="dv">1111001</span>)</span>
<span id="cb2402-2"><a href="dsi-allCells-allSetsTop.html#cb2402-2"></a><span class="co">#uncorrected &lt;- runTSNE(uncorrected, dimred=&quot;PCA&quot;)</span></span>
<span id="cb2402-3"><a href="dsi-allCells-allSetsTop.html#cb2402-3"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(uncorrected, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb2402-4"><a href="dsi-allCells-allSetsTop.html#cb2402-4"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2402-5"><a href="dsi-allCells-allSetsTop.html#cb2402-5"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2402-6"><a href="dsi-allCells-allSetsTop.html#cb2402-6"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2402-7"><a href="dsi-allCells-allSetsTop.html#cb2402-7"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2402-8"><a href="dsi-allCells-allSetsTop.html#cb2402-8"></a><span class="co"># draw:</span></span>
<span id="cb2402-9"><a href="dsi-allCells-allSetsTop.html#cb2402-9"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(uncorrected,</span>
<span id="cb2402-10"><a href="dsi-allCells-allSetsTop.html#cb2402-10"></a>          <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>,</span>
<span id="cb2402-11"><a href="dsi-allCells-allSetsTop.html#cb2402-11"></a>          <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>) <span class="op">+</span></span>
<span id="cb2402-12"><a href="dsi-allCells-allSetsTop.html#cb2402-12"></a><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<div class="sourceCode" id="cb2403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2403-1"><a href="dsi-allCells-allSetsTop.html#cb2403-1"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagTsnePlot1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2404-1"><a href="dsi-allCells-allSetsTop.html#cb2404-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>uncorrected<span class="op">$</span>source_name, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagTsnePlotSplit_dsi_allCells_allSets-1.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a UMAP plot:</p>
<div class="sourceCode" id="cb2405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2405-1"><a href="dsi-allCells-allSetsTop.html#cb2405-1"></a><span class="kw">set.seed</span>(<span class="dv">1111001</span>)</span>
<span id="cb2405-2"><a href="dsi-allCells-allSetsTop.html#cb2405-2"></a><span class="co">#uncorrected &lt;- runUMAP(uncorrected, dimred=&quot;PCA&quot;)</span></span>
<span id="cb2405-3"><a href="dsi-allCells-allSetsTop.html#cb2405-3"></a>uncorrected &lt;-<span class="st"> </span><span class="kw">runUMAP</span>(uncorrected, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb2405-4"><a href="dsi-allCells-allSetsTop.html#cb2405-4"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2405-5"><a href="dsi-allCells-allSetsTop.html#cb2405-5"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2405-6"><a href="dsi-allCells-allSetsTop.html#cb2405-6"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2405-7"><a href="dsi-allCells-allSetsTop.html#cb2405-7"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2405-8"><a href="dsi-allCells-allSetsTop.html#cb2405-8"></a>p &lt;-<span class="st"> </span><span class="kw">plotUMAP</span>(uncorrected,</span>
<span id="cb2405-9"><a href="dsi-allCells-allSetsTop.html#cb2405-9"></a>          <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>,</span>
<span id="cb2405-10"><a href="dsi-allCells-allSetsTop.html#cb2405-10"></a>          <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2406-1"><a href="dsi-allCells-allSetsTop.html#cb2406-1"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagUmapPlot1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2407-1"><a href="dsi-allCells-allSetsTop.html#cb2407-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>uncorrected<span class="op">$</span>source_name, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagUmapPlotSplit_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2408-1"><a href="dsi-allCells-allSetsTop.html#cb2408-1"></a><span class="co">#grid.arrange(</span></span>
<span id="cb2408-2"><a href="dsi-allCells-allSetsTop.html#cb2408-2"></a><span class="co">#  p,</span></span>
<span id="cb2408-3"><a href="dsi-allCells-allSetsTop.html#cb2408-3"></a><span class="co">#  p + facet_wrap(~uncorrected$source_name, ncol=2),</span></span>
<span id="cb2408-4"><a href="dsi-allCells-allSetsTop.html#cb2408-4"></a><span class="co">#  ncol=2)</span></span>
<span id="cb2408-5"><a href="dsi-allCells-allSetsTop.html#cb2408-5"></a></span>
<span id="cb2408-6"><a href="dsi-allCells-allSetsTop.html#cb2408-6"></a></span>
<span id="cb2408-7"><a href="dsi-allCells-allSetsTop.html#cb2408-7"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p)</span>
<span id="cb2408-8"><a href="dsi-allCells-allSetsTop.html#cb2408-8"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb2408-9"><a href="dsi-allCells-allSetsTop.html#cb2408-9"></a></span>
<span id="cb2408-10"><a href="dsi-allCells-allSetsTop.html#cb2408-10"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p,</span>
<span id="cb2408-11"><a href="dsi-allCells-allSetsTop.html#cb2408-11"></a>             p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>uncorrected<span class="op">$</span>source_name, <span class="dt">ncol=</span><span class="dv">2</span>),</span>
<span id="cb2408-12"><a href="dsi-allCells-allSetsTop.html#cb2408-12"></a>                         <span class="dt">ncol=</span><span class="dv">2</span>),</span>
<span id="cb2408-13"><a href="dsi-allCells-allSetsTop.html#cb2408-13"></a>         legend,</span>
<span id="cb2408-14"><a href="dsi-allCells-allSetsTop.html#cb2408-14"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2408-15"><a href="dsi-allCells-allSetsTop.html#cb2408-15"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/diagUmapPlotSplit2_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Of course, the other explanation for batch-specific clusters is that there are cell types that are unique to each batch. The degree of intermingling of cells from different batches is not an effective diagnostic when the batches involved might actually contain unique cell subpopulations. If a cluster only contains cells from a single batch, one can always debate whether that is caused by a failure of the correction method or if there is truly a batch-specific subpopulation. For example, do batch-specific metabolic or differentiation states represent distinct subpopulations? Or should they be merged together? We will not attempt to answer this here, only noting that each batch correction algorithm will make different (and possibly inappropriate) decisions on what constitutes “shared” and “unique” populations.</p>
</div>
<div id="linear-regression-9" class="section level2">
<h2><span class="header-section-number">29.5</span> Linear regression</h2>
<p>Batch effects in bulk RNA sequencing studies are commonly removed with linear regression. This involves fitting a linear model to each gene’s expression profile, setting the undesirable batch term to zero and recomputing the observations sans the batch effect, yielding a set of corrected expression values for downstream analyses. Linear modelling is the basis of the <code>removeBatchEffect()</code> function from the limma package (Ritchie et al. 2015) as well the <code>comBat()</code> function from the <code>sva</code> package (Leek et al. 2012).</p>
<p>To use this approach in a scRNA-seq context, we assume that the composition of cell subpopulations is the same across batches. We also assume that the batch effect is additive, i.e., any batch-induced fold-change in expression is the same across different cell subpopulations for any given gene. These are strong assumptions as batches derived from different individuals will naturally exhibit variation in cell type abundances and expression. Nonetheless, they may be acceptable when dealing with batches that are technical replicates generated from the same population of cells. (In fact, when its assumptions hold, linear regression is the most statistically efficient as it uses information from all cells to compute the common batch vector.) Linear modelling can also accommodate situations where the composition is known a priori by including the cell type as a factor in the linear model, but this situation is even less common.</p>
<p>We use the <code>rescaleBatches()</code> function from the batchelor package to remove the batch effect. This is roughly equivalent to applying a linear regression to the log-expression values per gene, with some adjustments to improve performance and efficiency. For each gene, the mean expression in each batch is scaled down until it is equal to the lowest mean across all batches. We deliberately choose to scale all expression values down as this mitigates differences in variance when batches lie at different positions on the mean-variance trend. (Specifically, the shrinkage effect of the pseudo-count is greater for smaller counts, suppressing any differences in variance across batches.) An additional feature of <code>rescaleBatches()</code> is that it will preserve sparsity in the input matrix for greater efficiency, whereas other methods like <code>removeBatchEffect()</code> will always return a dense matrix.</p>
<div class="sourceCode" id="cb2409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2409-1"><a href="dsi-allCells-allSetsTop.html#cb2409-1"></a><span class="co">#library(batchelor)</span></span>
<span id="cb2409-2"><a href="dsi-allCells-allSetsTop.html#cb2409-2"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">rescaleBatches</span>(rescaled.mbn)</span>
<span id="cb2409-3"><a href="dsi-allCells-allSetsTop.html#cb2409-3"></a>rescaled.rb</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 18431 47830 
## metadata(0):
## assays(1): corrected
## rownames(18431): ENSG00000000003 ENSG00000000419 ... ENSG00000285486
##   ENSG00000285492
## rowData names(0):
## colnames: NULL
## colData names(1): batch
## reducedDimNames(0):
## altExpNames(0):</code></pre>
<p>After clustering, we observe fewer clusters and these consist of mixtures of cells from the several replicates, consistent with the removal of the batch effect. This conclusion is supported by the apparent mixing of cells from different batches on the TSNE plot below. However, at least one batch-specific cluster is still present, indicating that the correction is not entirely complete. This is attributable to violation of one of the aforementioned assumptions, even in this simple case involving replicated batches.</p>
<div class="sourceCode" id="cb2411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2411-1"><a href="dsi-allCells-allSetsTop.html#cb2411-1"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2411-2"><a href="dsi-allCells-allSetsTop.html#cb2411-2"></a></span>
<span id="cb2411-3"><a href="dsi-allCells-allSetsTop.html#cb2411-3"></a><span class="kw">set.seed</span>(<span class="dv">1010101010</span>) <span class="co"># To ensure reproducibility of IRLBA.</span></span>
<span id="cb2411-4"><a href="dsi-allCells-allSetsTop.html#cb2411-4"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">runPCA</span>(rescaled.rb,</span>
<span id="cb2411-5"><a href="dsi-allCells-allSetsTop.html#cb2411-5"></a>              <span class="dt">subset_row=</span>chosen.hvgs,</span>
<span id="cb2411-6"><a href="dsi-allCells-allSetsTop.html#cb2411-6"></a>              <span class="dt">exprs_values=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb2411-7"><a href="dsi-allCells-allSetsTop.html#cb2411-7"></a>              <span class="co">#BSPARAM=IrlbaParam(),</span></span>
<span id="cb2411-8"><a href="dsi-allCells-allSetsTop.html#cb2411-8"></a>              <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>(),</span>
<span id="cb2411-9"><a href="dsi-allCells-allSetsTop.html#cb2411-9"></a>              <span class="dt">BPPARAM=</span>bpp</span>
<span id="cb2411-10"><a href="dsi-allCells-allSetsTop.html#cb2411-10"></a>)</span>
<span id="cb2411-11"><a href="dsi-allCells-allSetsTop.html#cb2411-11"></a></span>
<span id="cb2411-12"><a href="dsi-allCells-allSetsTop.html#cb2411-12"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
##   6.635   8.312 110.580</code></pre>
<div class="sourceCode" id="cb2413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2413-1"><a href="dsi-allCells-allSetsTop.html#cb2413-1"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2413-2"><a href="dsi-allCells-allSetsTop.html#cb2413-2"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(rescaled.rb,</span>
<span id="cb2413-3"><a href="dsi-allCells-allSetsTop.html#cb2413-3"></a>            <span class="dt">use.dimred=</span><span class="st">&quot;PCA&quot;</span>, <span class="co"># or use BSPARAM</span></span>
<span id="cb2413-4"><a href="dsi-allCells-allSetsTop.html#cb2413-4"></a>            <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2413-5"><a href="dsi-allCells-allSetsTop.html#cb2413-5"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
## 607.361 136.503  20.507</code></pre>
<div class="sourceCode" id="cb2415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2415-1"><a href="dsi-allCells-allSetsTop.html#cb2415-1"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2415-2"><a href="dsi-allCells-allSetsTop.html#cb2415-2"></a><span class="co">#clusters.resc &lt;- igraph::cluster_walktrap(snn.gr, steps=nSteps)$membership # slow</span></span>
<span id="cb2415-3"><a href="dsi-allCells-allSetsTop.html#cb2415-3"></a>clusters.resc &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_louvain</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb2415-4"><a href="dsi-allCells-allSetsTop.html#cb2415-4"></a><span class="co">#clusters.resc &lt;- igraph::cluster_fast_greedy(snn.gr)$membership # coarse</span></span>
<span id="cb2415-5"><a href="dsi-allCells-allSetsTop.html#cb2415-5"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
##  33.209   0.839  34.125</code></pre>
<div class="sourceCode" id="cb2417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2417-1"><a href="dsi-allCells-allSetsTop.html#cb2417-1"></a>rescaled.rb<span class="op">$</span>clusters.resc &lt;-<span class="st"> </span><span class="kw">factor</span>(clusters.resc)</span>
<span id="cb2417-2"><a href="dsi-allCells-allSetsTop.html#cb2417-2"></a></span>
<span id="cb2417-3"><a href="dsi-allCells-allSetsTop.html#cb2417-3"></a><span class="co">##tab.resc &lt;- table(Cluster=clusters.resc, Batch=rescaled.rb$batch)</span></span>
<span id="cb2417-4"><a href="dsi-allCells-allSetsTop.html#cb2417-4"></a><span class="co">##tab.resc</span></span>
<span id="cb2417-5"><a href="dsi-allCells-allSetsTop.html#cb2417-5"></a></span>
<span id="cb2417-6"><a href="dsi-allCells-allSetsTop.html#cb2417-6"></a>df.resc &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.resc, <span class="st">&quot;batch&quot;</span>=rescaled.rb<span class="op">$</span>batch)</span>
<span id="cb2417-7"><a href="dsi-allCells-allSetsTop.html#cb2417-7"></a>tab.resc &lt;-<span class="st"> </span><span class="kw">table</span>(df.resc)</span></code></pre></div>
<div class="sourceCode" id="cb2418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2418-1"><a href="dsi-allCells-allSetsTop.html#cb2418-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.resc,</span>
<span id="cb2418-2"><a href="dsi-allCells-allSetsTop.html#cb2418-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2418-3"><a href="dsi-allCells-allSetsTop.html#cb2418-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2418-4"><a href="dsi-allCells-allSetsTop.html#cb2418-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2418-5"><a href="dsi-allCells-allSetsTop.html#cb2418-5"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/linReg_diagHeatmap_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Cluster size and cell contribution by sample, with clusters sorted by size:</p>
<div class="sourceCode" id="cb2419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2419-1"><a href="dsi-allCells-allSetsTop.html#cb2419-1"></a>tmpMat &lt;-<span class="st"> </span>df.resc</span>
<span id="cb2419-2"><a href="dsi-allCells-allSetsTop.html#cb2419-2"></a>tmpMatTab &lt;-<span class="st"> </span>tab.resc</span>
<span id="cb2419-3"><a href="dsi-allCells-allSetsTop.html#cb2419-3"></a></span>
<span id="cb2419-4"><a href="dsi-allCells-allSetsTop.html#cb2419-4"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb2419-5"><a href="dsi-allCells-allSetsTop.html#cb2419-5"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2419-6"><a href="dsi-allCells-allSetsTop.html#cb2419-6"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2419-7"><a href="dsi-allCells-allSetsTop.html#cb2419-7"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2419-8"><a href="dsi-allCells-allSetsTop.html#cb2419-8"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2419-9"><a href="dsi-allCells-allSetsTop.html#cb2419-9"></a><span class="st">    </span><span class="kw">geom_col</span>() <span class="op">+</span></span>
<span id="cb2419-10"><a href="dsi-allCells-allSetsTop.html#cb2419-10"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2419-11"><a href="dsi-allCells-allSetsTop.html#cb2419-11"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2419-12"><a href="dsi-allCells-allSetsTop.html#cb2419-12"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2419-13"><a href="dsi-allCells-allSetsTop.html#cb2419-13"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2419-14"><a href="dsi-allCells-allSetsTop.html#cb2419-14"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent) <span class="op">+</span></span>
<span id="cb2419-15"><a href="dsi-allCells-allSetsTop.html#cb2419-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2419-16"><a href="dsi-allCells-allSetsTop.html#cb2419-16"></a><span class="co">#grid.arrange(p1, p2)</span></span>
<span id="cb2419-17"><a href="dsi-allCells-allSetsTop.html#cb2419-17"></a></span>
<span id="cb2419-18"><a href="dsi-allCells-allSetsTop.html#cb2419-18"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2419-19"><a href="dsi-allCells-allSetsTop.html#cb2419-19"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2419-20"><a href="dsi-allCells-allSetsTop.html#cb2419-20"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2419-21"><a href="dsi-allCells-allSetsTop.html#cb2419-21"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2419-22"><a href="dsi-allCells-allSetsTop.html#cb2419-22"></a>         legend,</span>
<span id="cb2419-23"><a href="dsi-allCells-allSetsTop.html#cb2419-23"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2419-24"><a href="dsi-allCells-allSetsTop.html#cb2419-24"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/linReg_diagPlotDraw_dsi_allCells_allSets-1.png" width="672" /></p>
<!-- and/or barplots -->
<p>Compute and plot t-SNE:</p>
<div class="sourceCode" id="cb2420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2420-1"><a href="dsi-allCells-allSetsTop.html#cb2420-1"></a>rescaled.rb &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(rescaled.rb, <span class="dt">dimred=</span><span class="st">&quot;PCA&quot;</span>,</span>
<span id="cb2420-2"><a href="dsi-allCells-allSetsTop.html#cb2420-2"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2420-3"><a href="dsi-allCells-allSetsTop.html#cb2420-3"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2420-4"><a href="dsi-allCells-allSetsTop.html#cb2420-4"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2420-5"><a href="dsi-allCells-allSetsTop.html#cb2420-5"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2420-6"><a href="dsi-allCells-allSetsTop.html#cb2420-6"></a>rescaled.rb<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(rescaled.rb<span class="op">$</span>batch)</span>
<span id="cb2420-7"><a href="dsi-allCells-allSetsTop.html#cb2420-7"></a>rescaled.rb<span class="op">$</span>source_name &lt;-<span class="st"> </span>uncorrected<span class="op">$</span>source_name</span>
<span id="cb2420-8"><a href="dsi-allCells-allSetsTop.html#cb2420-8"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(rescaled.rb, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2421-1"><a href="dsi-allCells-allSetsTop.html#cb2421-1"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/linReg_diagTsnePlot1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2422-1"><a href="dsi-allCells-allSetsTop.html#cb2422-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>rescaled.rb<span class="op">$</span>source_name)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/linReg_diagTsnePlotSplit_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2423-1"><a href="dsi-allCells-allSetsTop.html#cb2423-1"></a>ptm &lt;-<span class="st"> </span><span class="kw">proc.time</span>()</span>
<span id="cb2423-2"><a href="dsi-allCells-allSetsTop.html#cb2423-2"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(rescaled.rb, <span class="dt">colour_by=</span><span class="st">&quot;clusters.resc&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2423-3"><a href="dsi-allCells-allSetsTop.html#cb2423-3"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(rescaled.rb, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2423-4"><a href="dsi-allCells-allSetsTop.html#cb2423-4"></a><span class="kw">grid.arrange</span>(p.clu, p.batch, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/linReg_diagTsnePlotSplit2_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2424-1"><a href="dsi-allCells-allSetsTop.html#cb2424-1"></a><span class="kw">proc.time</span>() <span class="op">-</span><span class="st"> </span>ptm</span></code></pre></div>
<pre><code>##    user  system elapsed 
##   0.539   0.072   0.612</code></pre>
</div>
<div id="mutual-nearest-neighbour-correction-5" class="section level2">
<h2><span class="header-section-number">29.6</span> Mutual Nearest Neighbour correction</h2>
<div id="algorithm-overview-9" class="section level3">
<h3><span class="header-section-number">29.6.1</span> Algorithm overview</h3>
<p>Consider a cell a in batch A, and identify the cells in batch B that are nearest neighbors to a in the expression space defined by the selected features. Repeat this for a cell b in batch B, identifying its nearest neighbors in A. Mutual nearest neighbors (MNNs) are pairs of cells from different batches that belong in each other’s set of nearest neighbors. The reasoning is that MNN pairs represent cells from the same biological state prior to the application of a batch effect - see Haghverdi et al. (2018) for full theoretical details. Thus, the difference between cells in MNN pairs can be used as an estimate of the batch effect, the subtraction of which yields batch-corrected values.</p>
<p>Compared to linear regression, MNN correction does not assume that the population composition is the same or known beforehand. This is because it learns the shared population structure via identification of MNN pairs and uses this information to obtain an appropriate estimate of the batch effect. Instead, the key assumption of MNN-based approaches is that the batch effect is orthogonal to the biology in high-dimensional expression space. Violations reduce the effectiveness and accuracy of the correction, with the most common case arising from variations in the direction of the batch effect between clusters. Nonetheless, the assumption is usually reasonable as a random vector is very likely to be orthogonal in high-dimensional space.</p>
</div>
<div id="application-to-the-data-9" class="section level3">
<h3><span class="header-section-number">29.6.2</span> Application to the data</h3>
<p>The batchelor package provides an implementation of the MNN approach via the fastMNN() function. Unlike the MNN method originally described by Haghverdi et al. (2018), the fastMNN() function performs PCA to reduce the dimensions beforehand and speed up the downstream neighbor detection steps.</p>
<p>We apply it to our two PBMC batches to remove the batch effect across the highly variable genes in chosen.hvgs. To reduce computational work and technical noise, all cells in all batches are projected into the low-dimensional space defined by the top d principal components. Identification of MNNs and calculation of correction vectors are then performed in this low-dimensional space.</p>
<div class="sourceCode" id="cb2426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2426-1"><a href="dsi-allCells-allSetsTop.html#cb2426-1"></a><span class="co"># Using randomized SVD here, as this is faster than </span></span>
<span id="cb2426-2"><a href="dsi-allCells-allSetsTop.html#cb2426-2"></a><span class="co"># irlba for file-backed matrices.</span></span>
<span id="cb2426-3"><a href="dsi-allCells-allSetsTop.html#cb2426-3"></a><span class="kw">set.seed</span>(<span class="dv">1000101001</span>)</span>
<span id="cb2426-4"><a href="dsi-allCells-allSetsTop.html#cb2426-4"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(</span>
<span id="cb2426-5"><a href="dsi-allCells-allSetsTop.html#cb2426-5"></a>  rescaled.mbn,</span>
<span id="cb2426-6"><a href="dsi-allCells-allSetsTop.html#cb2426-6"></a>  <span class="dt">auto.merge=</span><span class="ot">TRUE</span>,</span>
<span id="cb2426-7"><a href="dsi-allCells-allSetsTop.html#cb2426-7"></a>  <span class="dt">d=</span><span class="dv">50</span>,</span>
<span id="cb2426-8"><a href="dsi-allCells-allSetsTop.html#cb2426-8"></a>  <span class="dt">k=</span><span class="dv">20</span>,</span>
<span id="cb2426-9"><a href="dsi-allCells-allSetsTop.html#cb2426-9"></a>  <span class="dt">subset.row=</span>chosen.hvgs,</span>
<span id="cb2426-10"><a href="dsi-allCells-allSetsTop.html#cb2426-10"></a>  <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>),</span>
<span id="cb2426-11"><a href="dsi-allCells-allSetsTop.html#cb2426-11"></a>  <span class="dt">BPPARAM=</span>bpp</span>
<span id="cb2426-12"><a href="dsi-allCells-allSetsTop.html#cb2426-12"></a>  )</span>
<span id="cb2426-13"><a href="dsi-allCells-allSetsTop.html#cb2426-13"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 12466 47830 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(12466): ENSG00000000003 ENSG00000000457 ... ENSG00000285476
##   ENSG00000285492
## rowData names(1): rotation
## colnames: NULL
## colData names(1): batch
## reducedDimNames(1): corrected
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb2428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2428-1"><a href="dsi-allCells-allSetsTop.html#cb2428-1"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb2428-2"><a href="dsi-allCells-allSetsTop.html#cb2428-2"></a>mnn.out<span class="op">$</span>splType &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch)</span>
<span id="cb2428-3"><a href="dsi-allCells-allSetsTop.html#cb2428-3"></a>mnn.out.corre.dim &lt;-<span class="st"> </span><span class="kw">dim</span>(<span class="kw">reducedDim</span>(mnn.out, <span class="st">&quot;corrected&quot;</span>))</span>
<span id="cb2428-4"><a href="dsi-allCells-allSetsTop.html#cb2428-4"></a>mnn.out.corre.dim</span></code></pre></div>
<pre><code>## [1] 47830    50</code></pre>
<div class="sourceCode" id="cb2430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2430-1"><a href="dsi-allCells-allSetsTop.html#cb2430-1"></a>mnn.out.recon.dim &lt;-<span class="st"> </span><span class="kw">dim</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>))</span>
<span id="cb2430-2"><a href="dsi-allCells-allSetsTop.html#cb2430-2"></a>mnn.out.recon.dim</span></code></pre></div>
<pre><code>## [1] 12466 47830</code></pre>
<p>The function returns a SingleCellExperiment object containing corrected values for downstream analyses like clustering or visualization. Each column of mnn.out corresponds to a cell in one of the batches, while each row corresponds to an input gene in chosen.hvgs. The batch field in the column metadata contains a vector specifying the batch of origin of each cell.</p>
<p>The <code>corrected</code> matrix in the <code>reducedDims()</code> contains the low-dimensional corrected coordinates for all cells, which we will use in place of the PCs in our downstream analyses (47830 cells and 50 PCs).</p>
<p>A <code>reconstructed</code> matrix in the <code>assays()</code> contains the corrected expression values for each gene in each cell, obtained by projecting the low-dimensional coordinates in corrected back into gene expression space (12466 genes and 47830 cells). We do not recommend using this for anything other than visualization.</p>
<div class="sourceCode" id="cb2432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2432-1"><a href="dsi-allCells-allSetsTop.html#cb2432-1"></a><span class="kw">print</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</span></code></pre></div>
<pre><code>## &lt;5 x 3&gt; matrix of class LowRankMatrix and type &quot;double&quot;:
##                          [,1]          [,2]          [,3]
## ENSG00000000003 -8.970458e-05 -6.411674e-05 -6.896130e-05
## ENSG00000000457  6.004816e-05  5.014359e-04  2.004637e-04
## ENSG00000000938 -3.567196e-04  3.084119e-04  4.221600e-04
## ENSG00000001167  2.328682e-04  7.270576e-04  6.971589e-04
## ENSG00000001461 -8.717612e-04 -6.734327e-04 -7.530539e-04</code></pre>
<p>The most relevant parameter for tuning fastMNN() is k, which specifies the number of nearest neighbors to consider when defining MNN pairs. This can be interpreted as the minimum anticipated frequency of any shared cell type or state in each batch. Increasing k will generally result in more aggressive merging as the algorithm is more generous in matching subpopulations across batches. It can occasionally be desirable to increase k if one clearly sees that the same cell types are not being adequately merged across batches.</p>
<div class="sourceCode" id="cb2434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2434-1"><a href="dsi-allCells-allSetsTop.html#cb2434-1"></a>colDataList &lt;-<span class="st"> </span><span class="kw">lapply</span>(rescaled.mbn, <span class="cf">function</span>(x){<span class="kw">colData</span>(x)})</span>
<span id="cb2434-2"><a href="dsi-allCells-allSetsTop.html#cb2434-2"></a>colDataDf &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, colDataList)</span>
<span id="cb2434-3"><a href="dsi-allCells-allSetsTop.html#cb2434-3"></a><span class="co">#colData(mnn.out) &lt;- cbind(colDataDf, colData(mnn.out)$cluster)</span></span>
<span id="cb2434-4"><a href="dsi-allCells-allSetsTop.html#cb2434-4"></a><span class="kw">colData</span>(mnn.out) &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(<span class="kw">cbind</span>(<span class="kw">colData</span>(mnn.out), colDataDf)) <span class="co"># no rearranging of columns by mnncorrect</span></span></code></pre></div>
<!--
See Chapter 32 for an example of a more complex fastMNN() merge involving several human pancreas datasets generated by different authors on different patients with different technologies.
-->
</div>
</div>
<div id="correction-diagnostics-9" class="section level2">
<h2><span class="header-section-number">29.7</span> Correction diagnostics</h2>
<p>We cluster on the low-dimensional corrected coordinates to obtain a partitioning of the cells that serves as a proxy for the population structure. If the batch effect is successfully corrected, clusters corresponding to shared cell types or states should contain cells from multiple batches. We see that all clusters contain contributions from each batch after correction, consistent with our expectation that the two batches are replicates of each other.</p>
<div class="sourceCode" id="cb2435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2435-1"><a href="dsi-allCells-allSetsTop.html#cb2435-1"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(mnn.out,</span>
<span id="cb2435-2"><a href="dsi-allCells-allSetsTop.html#cb2435-2"></a>            <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>, <span class="co"># or use BSPARAM</span></span>
<span id="cb2435-3"><a href="dsi-allCells-allSetsTop.html#cb2435-3"></a>            <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2435-4"><a href="dsi-allCells-allSetsTop.html#cb2435-4"></a><span class="co">#clusters.mnn &lt;- igraph::cluster_walktrap(snn.gr, steps=nSteps)$membership</span></span>
<span id="cb2435-5"><a href="dsi-allCells-allSetsTop.html#cb2435-5"></a>clusters.mnn &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_louvain</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb2435-6"><a href="dsi-allCells-allSetsTop.html#cb2435-6"></a><span class="co">#clusters.mnn &lt;- igraph::cluster_fast_greedy(snn.gr)$membership</span></span>
<span id="cb2435-7"><a href="dsi-allCells-allSetsTop.html#cb2435-7"></a>mnn.out<span class="op">$</span>clusters.mnn &lt;-<span class="st"> </span><span class="kw">factor</span>(clusters.mnn)</span>
<span id="cb2435-8"><a href="dsi-allCells-allSetsTop.html#cb2435-8"></a></span>
<span id="cb2435-9"><a href="dsi-allCells-allSetsTop.html#cb2435-9"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>mnn.out<span class="op">$</span>clusters.mnn,</span>
<span id="cb2435-10"><a href="dsi-allCells-allSetsTop.html#cb2435-10"></a>         <span class="dt">Batch=</span>mnn.out<span class="op">$</span>batch)</span>
<span id="cb2435-11"><a href="dsi-allCells-allSetsTop.html#cb2435-11"></a>tab.mnn</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##      1          1155         1552          336         1566   470   349      37
##      2           366          578          113          568   251   491     234
##      3           112         2129         1246          291   188  1278      17
##      4           306          595          194          825   216   514      50
##      5             7          101          462          140   229    37      61
##      6           730          823          191          330  1517  2307      43
##      7             2          192            8            3   588    52       2
##      8           108          372           96          130   439   119      25
##      9            33            6           31           26   178    44     250
##      10            1            1            5            2    32    34      19
##      11            0            7           22           42    51    13     270
##      12            3            3          117          234     1    13      25
##      13            0            2            2            1    11     1     336
##      14           14           75          390          117   159    74      98
##      15           10          144         1306          219   213    86     595
##      16            0           21           10            5     2     6      15
##      17            6           14          198          794     6    19       7
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##      1       25      70       1       0
##      2       87     255     181      22
##      3       18      96       3       1
##      4       18      45      12       5
##      5      422     402      21     114
##      6       22      47       6      49
##      7        1       7       1       0
##      8       24      14     860     211
##      9      321     247     633     237
##      10      14      12    2058    2012
##      11     431     755       6     235
##      12     463      92       1      74
##      13     195     738       1       6
##      14     470     395      28     366
##      15    1050     875      24     251
##      16      33      33       1      25
##      17    1064      60       4      20</code></pre>
<div class="sourceCode" id="cb2437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2437-1"><a href="dsi-allCells-allSetsTop.html#cb2437-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.mnn,</span>
<span id="cb2437-2"><a href="dsi-allCells-allSetsTop.html#cb2437-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2437-3"><a href="dsi-allCells-allSetsTop.html#cb2437-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2437-4"><a href="dsi-allCells-allSetsTop.html#cb2437-4"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2437-5"><a href="dsi-allCells-allSetsTop.html#cb2437-5"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagPlotheatmap_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Cluster size and cell contribution by sample, with clusters sorted by size:</p>
<div class="sourceCode" id="cb2438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2438-1"><a href="dsi-allCells-allSetsTop.html#cb2438-1"></a><span class="co">#mnn.out$source_name &lt;- uncorrected$source_name # cell order is maintained by scran functions</span></span>
<span id="cb2438-2"><a href="dsi-allCells-allSetsTop.html#cb2438-2"></a>mnn.out<span class="op">$</span>Sample.Name &lt;-<span class="st"> </span>uncorrected<span class="op">$</span>Sample.Name <span class="co"># cell order is maintained by scran functions</span></span>
<span id="cb2438-3"><a href="dsi-allCells-allSetsTop.html#cb2438-3"></a></span>
<span id="cb2438-4"><a href="dsi-allCells-allSetsTop.html#cb2438-4"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.mnn, <span class="st">&quot;batch&quot;</span>=mnn.out<span class="op">$</span>Sample.Name2)</span>
<span id="cb2438-5"><a href="dsi-allCells-allSetsTop.html#cb2438-5"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2438-6"><a href="dsi-allCells-allSetsTop.html#cb2438-6"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb2438-7"><a href="dsi-allCells-allSetsTop.html#cb2438-7"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2438-8"><a href="dsi-allCells-allSetsTop.html#cb2438-8"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2438-9"><a href="dsi-allCells-allSetsTop.html#cb2438-9"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2438-10"><a href="dsi-allCells-allSetsTop.html#cb2438-10"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2438-11"><a href="dsi-allCells-allSetsTop.html#cb2438-11"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2438-12"><a href="dsi-allCells-allSetsTop.html#cb2438-12"></a><span class="st">    </span><span class="kw">geom_col</span>() <span class="op">+</span></span>
<span id="cb2438-13"><a href="dsi-allCells-allSetsTop.html#cb2438-13"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2438-14"><a href="dsi-allCells-allSetsTop.html#cb2438-14"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2438-15"><a href="dsi-allCells-allSetsTop.html#cb2438-15"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2438-16"><a href="dsi-allCells-allSetsTop.html#cb2438-16"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2438-17"><a href="dsi-allCells-allSetsTop.html#cb2438-17"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent) <span class="op">+</span></span>
<span id="cb2438-18"><a href="dsi-allCells-allSetsTop.html#cb2438-18"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span>
<span id="cb2438-19"><a href="dsi-allCells-allSetsTop.html#cb2438-19"></a></span>
<span id="cb2438-20"><a href="dsi-allCells-allSetsTop.html#cb2438-20"></a><span class="co">#grid.arrange(p1, p2)</span></span>
<span id="cb2438-21"><a href="dsi-allCells-allSetsTop.html#cb2438-21"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2438-22"><a href="dsi-allCells-allSetsTop.html#cb2438-22"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2438-23"><a href="dsi-allCells-allSetsTop.html#cb2438-23"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2438-24"><a href="dsi-allCells-allSetsTop.html#cb2438-24"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2438-25"><a href="dsi-allCells-allSetsTop.html#cb2438-25"></a>         legend,</span>
<span id="cb2438-26"><a href="dsi-allCells-allSetsTop.html#cb2438-26"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2438-27"><a href="dsi-allCells-allSetsTop.html#cb2438-27"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagPlotDrawSampleName_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Cluster size and cell contribution by sample type, with clusters sorted by size:</p>
<div class="sourceCode" id="cb2439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2439-1"><a href="dsi-allCells-allSetsTop.html#cb2439-1"></a>mnn.out<span class="op">$</span>source_name &lt;-<span class="st"> </span><span class="kw">factor</span>(uncorrected<span class="op">$</span>source_name) <span class="co"># cell order is maintained by scran functions</span></span>
<span id="cb2439-2"><a href="dsi-allCells-allSetsTop.html#cb2439-2"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.mnn, <span class="st">&quot;batch&quot;</span>=mnn.out<span class="op">$</span>source_name)</span>
<span id="cb2439-3"><a href="dsi-allCells-allSetsTop.html#cb2439-3"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2439-4"><a href="dsi-allCells-allSetsTop.html#cb2439-4"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb2439-5"><a href="dsi-allCells-allSetsTop.html#cb2439-5"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2439-6"><a href="dsi-allCells-allSetsTop.html#cb2439-6"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2439-7"><a href="dsi-allCells-allSetsTop.html#cb2439-7"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2439-8"><a href="dsi-allCells-allSetsTop.html#cb2439-8"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2439-9"><a href="dsi-allCells-allSetsTop.html#cb2439-9"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2439-10"><a href="dsi-allCells-allSetsTop.html#cb2439-10"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb2439-11"><a href="dsi-allCells-allSetsTop.html#cb2439-11"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2439-12"><a href="dsi-allCells-allSetsTop.html#cb2439-12"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2439-13"><a href="dsi-allCells-allSetsTop.html#cb2439-13"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2439-14"><a href="dsi-allCells-allSetsTop.html#cb2439-14"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span>
<span id="cb2439-15"><a href="dsi-allCells-allSetsTop.html#cb2439-15"></a></span>
<span id="cb2439-16"><a href="dsi-allCells-allSetsTop.html#cb2439-16"></a><span class="co">#grid.arrange(p1, p2)</span></span>
<span id="cb2439-17"><a href="dsi-allCells-allSetsTop.html#cb2439-17"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2439-18"><a href="dsi-allCells-allSetsTop.html#cb2439-18"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2439-19"><a href="dsi-allCells-allSetsTop.html#cb2439-19"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2439-20"><a href="dsi-allCells-allSetsTop.html#cb2439-20"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2439-21"><a href="dsi-allCells-allSetsTop.html#cb2439-21"></a>         legend,</span>
<span id="cb2439-22"><a href="dsi-allCells-allSetsTop.html#cb2439-22"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2439-23"><a href="dsi-allCells-allSetsTop.html#cb2439-23"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagPlotDrawSampleType_dsi_allCells_allSets-1.png" width="672" /></p>
<p>We can also visualize the corrected coordinates using a t-SNE plot. The presence of visual clusters containing cells from batches provides a comforting illusion that the correction was successful.</p>
<div class="sourceCode" id="cb2440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2440-1"><a href="dsi-allCells-allSetsTop.html#cb2440-1"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb2440-2"><a href="dsi-allCells-allSetsTop.html#cb2440-2"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb2440-3"><a href="dsi-allCells-allSetsTop.html#cb2440-3"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2440-4"><a href="dsi-allCells-allSetsTop.html#cb2440-4"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2440-5"><a href="dsi-allCells-allSetsTop.html#cb2440-5"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2440-6"><a href="dsi-allCells-allSetsTop.html#cb2440-6"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2440-7"><a href="dsi-allCells-allSetsTop.html#cb2440-7"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2441-1"><a href="dsi-allCells-allSetsTop.html#cb2441-1"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagTsnePlot1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2442-1"><a href="dsi-allCells-allSetsTop.html#cb2442-1"></a>mnn.out<span class="op">$</span>splType &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch) <span class="co"># TODO fix; shouldn&#39;t need to; done above already</span></span>
<span id="cb2442-2"><a href="dsi-allCells-allSetsTop.html#cb2442-2"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>splType, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagTsnePlotSplit_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2443-1"><a href="dsi-allCells-allSetsTop.html#cb2443-1"></a><span class="co"># show clusters</span></span>
<span id="cb2443-2"><a href="dsi-allCells-allSetsTop.html#cb2443-2"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;clusters.mnn&quot;</span>)</span>
<span id="cb2443-3"><a href="dsi-allCells-allSetsTop.html#cb2443-3"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>)</span>
<span id="cb2443-4"><a href="dsi-allCells-allSetsTop.html#cb2443-4"></a><span class="co">#grid.arrange(p.clu, p.batch, ncol=2)</span></span>
<span id="cb2443-5"><a href="dsi-allCells-allSetsTop.html#cb2443-5"></a><span class="kw">grid.arrange</span>(p.clu, p.batch<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>splType), <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p>We can also compute the variation in the log-abundances to rank the clusters with the greatest variability in their proportional abundances across batches. We can then focus on batch-specific clusters that may be indicative of incomplete batch correction. Obviously, though, this diagnostic is subject to interpretation as the same outcome can be caused by batch-specific populations; some prior knowledge about the biological context is necessary to distinguish between these two possibilities. The table below shows the number of cells for each cluster (row) and sample (column) together with the variance in cell number across these samples (‘var’ column).</p>
<div class="sourceCode" id="cb2444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2444-1"><a href="dsi-allCells-allSetsTop.html#cb2444-1"></a><span class="co"># Avoid minor difficulties with the &#39;table&#39; class.</span></span>
<span id="cb2444-2"><a href="dsi-allCells-allSetsTop.html#cb2444-2"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">unclass</span>(tab.mnn)</span>
<span id="cb2444-3"><a href="dsi-allCells-allSetsTop.html#cb2444-3"></a></span>
<span id="cb2444-4"><a href="dsi-allCells-allSetsTop.html#cb2444-4"></a><span class="co"># Using a large pseudo.count to avoid unnecessarily</span></span>
<span id="cb2444-5"><a href="dsi-allCells-allSetsTop.html#cb2444-5"></a><span class="co"># large variances when the counts are low.</span></span>
<span id="cb2444-6"><a href="dsi-allCells-allSetsTop.html#cb2444-6"></a>norm &lt;-<span class="st"> </span><span class="kw">normalizeCounts</span>(tab.mnn, <span class="dt">pseudo_count=</span><span class="dv">10</span>)</span>
<span id="cb2444-7"><a href="dsi-allCells-allSetsTop.html#cb2444-7"></a></span>
<span id="cb2444-8"><a href="dsi-allCells-allSetsTop.html#cb2444-8"></a><span class="co"># Ranking clusters by the largest variances.</span></span>
<span id="cb2444-9"><a href="dsi-allCells-allSetsTop.html#cb2444-9"></a>rv &lt;-<span class="st"> </span><span class="kw">rowVars</span>(norm) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">round</span>(<span class="dv">2</span>)</span>
<span id="cb2444-10"><a href="dsi-allCells-allSetsTop.html#cb2444-10"></a></span>
<span id="cb2444-11"><a href="dsi-allCells-allSetsTop.html#cb2444-11"></a><span class="co"># show</span></span>
<span id="cb2444-12"><a href="dsi-allCells-allSetsTop.html#cb2444-12"></a><span class="co">#DataFrame(Batch=tab.mnn, var=rv)[order(rv, decreasing=TRUE),]</span></span>
<span id="cb2444-13"><a href="dsi-allCells-allSetsTop.html#cb2444-13"></a><span class="kw">DataFrame</span>(tab.mnn, <span class="dt">var=</span>rv)[<span class="kw">order</span>(rv, <span class="dt">decreasing=</span><span class="ot">TRUE</span>),]</span></code></pre></div>
<pre><code>## DataFrame with 17 rows and 12 columns
##     ETV6.RUNX1_1 ETV6.RUNX1_2 ETV6.RUNX1_3 ETV6.RUNX1_4     HHD_1     HHD_2
##        &lt;integer&gt;    &lt;integer&gt;    &lt;integer&gt;    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 10             1            1            5            2        32        34
## 1           1155         1552          336         1566       470       349
## 13             0            2            2            1        11         1
## 3            112         2129         1246          291       188      1278
## 11             0            7           22           42        51        13
## ...          ...          ...          ...          ...       ...       ...
## 8            108          372           96          130       439       119
## 5              7          101          462          140       229        37
## 14            14           75          390          117       159        74
## 2            366          578          113          568       251       491
## 16             0           21           10            5         2         6
##       PBMMC_1   PBMMC_2   PBMMC_3   PRE.T_1   PRE.T_2       var
##     &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;numeric&gt;
## 10         19        14        12      2058      2012      8.20
## 1          37        25        70         1         0      6.99
## 13        336       195       738         1         6      6.35
## 3          17        18        96         3         1      6.29
## 11        270       431       755         6       235      5.47
## ...       ...       ...       ...       ...       ...       ...
## 8          25        24        14       860       211      2.48
## 5          61       422       402        21       114      2.37
## 14         98       470       395        28       366      2.10
## 2         234        87       255       181        22      1.53
## 16         15        33        33         1        25      0.72</code></pre>
<p>For fastMNN(), one useful diagnostic is the proportion of variance within each batch that is lost during MNN correction. Specifically, this refers to the within-batch variance that is removed during orthogonalization with respect to the average correction vector at each merge step. This is returned via the lost.var field in the metadata of mnn.out, which contains a matrix of the variance lost in each batch (column) at each merge step (row).</p>
<div class="sourceCode" id="cb2446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2446-1"><a href="dsi-allCells-allSetsTop.html#cb2446-1"></a><span class="kw">round</span>(<span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var,<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##       ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PRE-T_1
##  [1,]         0.00         0.00         0.00         0.00  0.00  0.00    0.00
##  [2,]         0.00         0.00         0.08         0.00  0.00  0.00    0.00
##  [3,]         0.00         0.00         0.01         0.10  0.00  0.00    0.00
##  [4,]         0.00         0.06         0.02         0.04  0.00  0.00    0.00
##  [5,]         0.00         0.01         0.04         0.07  0.05  0.00    0.00
##  [6,]         0.00         0.00         0.00         0.00  0.00  0.07    0.00
##  [7,]         0.00         0.01         0.02         0.03  0.02  0.02    0.00
##  [8,]         0.00         0.01         0.01         0.00  0.01  0.01    0.00
##  [9,]         0.08         0.00         0.01         0.01  0.00  0.00    0.00
## [10,]         0.01         0.01         0.01         0.01  0.01  0.01    0.07
##       PRE-T_2 PBMMC_1 PBMMC_2 PBMMC_3
##  [1,]    0.00    0.00    0.07    0.04
##  [2,]    0.00    0.00    0.00    0.01
##  [3,]    0.00    0.00    0.01    0.00
##  [4,]    0.00    0.00    0.04    0.01
##  [5,]    0.00    0.00    0.08    0.02
##  [6,]    0.00    0.00    0.00    0.00
##  [7,]    0.00    0.08    0.04    0.02
##  [8,]    0.11    0.01    0.00    0.01
##  [9,]    0.01    0.00    0.02    0.01
## [10,]    0.01    0.01    0.02    0.01</code></pre>
<div class="sourceCode" id="cb2448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2448-1"><a href="dsi-allCells-allSetsTop.html#cb2448-1"></a>tmpData &lt;-<span class="st"> </span><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span>
<span id="cb2448-2"><a href="dsi-allCells-allSetsTop.html#cb2448-2"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb2448-3"><a href="dsi-allCells-allSetsTop.html#cb2448-3"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2448-4"><a href="dsi-allCells-allSetsTop.html#cb2448-4"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2448-5"><a href="dsi-allCells-allSetsTop.html#cb2448-5"></a>           <span class="dt">cluster_rows      =</span> <span class="ot">FALSE</span>,</span>
<span id="cb2448-6"><a href="dsi-allCells-allSetsTop.html#cb2448-6"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2448-7"><a href="dsi-allCells-allSetsTop.html#cb2448-7"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagLostVarHeatmap_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2449-1"><a href="dsi-allCells-allSetsTop.html#cb2449-1"></a><span class="kw">rm</span>(tmpData)</span></code></pre></div>
<p>Large proportions of lost variance (&gt;10%) suggest that correction is removing genuine biological heterogeneity. This would occur due to violations of the assumption of orthogonality between the batch effect and the biological subspace (Haghverdi et al. 2018). In this case, the proportion of lost variance is small, indicating that non-orthogonality is not a major concern.</p>
<p>The following t-SNE shows the clusters identified:</p>
<div class="sourceCode" id="cb2450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2450-1"><a href="dsi-allCells-allSetsTop.html#cb2450-1"></a>mnn.out<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, clusters.mnn)</span>
<span id="cb2450-2"><a href="dsi-allCells-allSetsTop.html#cb2450-2"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb2450-3"><a href="dsi-allCells-allSetsTop.html#cb2450-3"></a>              <span class="dt">colour_by=</span><span class="st">&quot;cluster&quot;</span>,</span>
<span id="cb2450-4"><a href="dsi-allCells-allSetsTop.html#cb2450-4"></a>              <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>,</span>
<span id="cb2450-5"><a href="dsi-allCells-allSetsTop.html#cb2450-5"></a>              <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2450-6"><a href="dsi-allCells-allSetsTop.html#cb2450-6"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagTsneClu_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2451-1"><a href="dsi-allCells-allSetsTop.html#cb2451-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out)<span class="op">$</span>source_name, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagTsneCluBySampleType_dsi_allCells_allSets-1.png" width="672" /></p>
<p>The following t-SNE plots show expression levels of known cell type marker genes.</p>
<div class="sourceCode" id="cb2452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2452-1"><a href="dsi-allCells-allSetsTop.html#cb2452-1"></a>genesToShow &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb2452-2"><a href="dsi-allCells-allSetsTop.html#cb2452-2"></a>         <span class="st">&quot;CD79A&quot;</span>, <span class="co"># CD79A   B ***</span></span>
<span id="cb2452-3"><a href="dsi-allCells-allSetsTop.html#cb2452-3"></a>         <span class="st">&quot;CST3&quot;</span>, <span class="co"># CST3     monocytes ***</span></span>
<span id="cb2452-4"><a href="dsi-allCells-allSetsTop.html#cb2452-4"></a>         <span class="st">&quot;CD3D&quot;</span>, <span class="co"># CD3D      T cells ***</span></span>
<span id="cb2452-5"><a href="dsi-allCells-allSetsTop.html#cb2452-5"></a>         <span class="st">&quot;HBA1&quot;</span> <span class="co"># HBA1   erythrocytes ***</span></span>
<span id="cb2452-6"><a href="dsi-allCells-allSetsTop.html#cb2452-6"></a>        )</span>
<span id="cb2452-7"><a href="dsi-allCells-allSetsTop.html#cb2452-7"></a></span>
<span id="cb2452-8"><a href="dsi-allCells-allSetsTop.html#cb2452-8"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(uncorrected)<span class="op">$</span>Symbol <span class="op">%in%</span><span class="st"> </span>genesToShow)</span>
<span id="cb2452-9"><a href="dsi-allCells-allSetsTop.html#cb2452-9"></a>ensToShow &lt;-<span class="st"> </span><span class="kw">rowData</span>(uncorrected)<span class="op">$</span>ensembl_gene_id[tmpInd]</span>
<span id="cb2452-10"><a href="dsi-allCells-allSetsTop.html#cb2452-10"></a></span>
<span id="cb2452-11"><a href="dsi-allCells-allSetsTop.html#cb2452-11"></a><span class="co">#B cells:</span></span>
<span id="cb2452-12"><a href="dsi-allCells-allSetsTop.html#cb2452-12"></a></span>
<span id="cb2452-13"><a href="dsi-allCells-allSetsTop.html#cb2452-13"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">1</span>]</span>
<span id="cb2452-14"><a href="dsi-allCells-allSetsTop.html#cb2452-14"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2452-15"><a href="dsi-allCells-allSetsTop.html#cb2452-15"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2452-16"><a href="dsi-allCells-allSetsTop.html#cb2452-16"></a>        <span class="kw">paste</span>(<span class="st">&quot;B cells&quot;</span>, genex,</span>
<span id="cb2452-17"><a href="dsi-allCells-allSetsTop.html#cb2452-17"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2452-18"><a href="dsi-allCells-allSetsTop.html#cb2452-18"></a>    )</span>
<span id="cb2452-19"><a href="dsi-allCells-allSetsTop.html#cb2452-19"></a>pB &lt;-<span class="st"> </span>p</span>
<span id="cb2452-20"><a href="dsi-allCells-allSetsTop.html#cb2452-20"></a></span>
<span id="cb2452-21"><a href="dsi-allCells-allSetsTop.html#cb2452-21"></a><span class="co">#T cells:</span></span>
<span id="cb2452-22"><a href="dsi-allCells-allSetsTop.html#cb2452-22"></a></span>
<span id="cb2452-23"><a href="dsi-allCells-allSetsTop.html#cb2452-23"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">3</span>]</span>
<span id="cb2452-24"><a href="dsi-allCells-allSetsTop.html#cb2452-24"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2452-25"><a href="dsi-allCells-allSetsTop.html#cb2452-25"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2452-26"><a href="dsi-allCells-allSetsTop.html#cb2452-26"></a>        <span class="kw">paste</span>(<span class="st">&quot;T cells&quot;</span>, genex,</span>
<span id="cb2452-27"><a href="dsi-allCells-allSetsTop.html#cb2452-27"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2452-28"><a href="dsi-allCells-allSetsTop.html#cb2452-28"></a>    )</span>
<span id="cb2452-29"><a href="dsi-allCells-allSetsTop.html#cb2452-29"></a>pT &lt;-<span class="st"> </span>p</span>
<span id="cb2452-30"><a href="dsi-allCells-allSetsTop.html#cb2452-30"></a></span>
<span id="cb2452-31"><a href="dsi-allCells-allSetsTop.html#cb2452-31"></a><span class="co">#monocytes:</span></span>
<span id="cb2452-32"><a href="dsi-allCells-allSetsTop.html#cb2452-32"></a></span>
<span id="cb2452-33"><a href="dsi-allCells-allSetsTop.html#cb2452-33"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">2</span>]</span>
<span id="cb2452-34"><a href="dsi-allCells-allSetsTop.html#cb2452-34"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2452-35"><a href="dsi-allCells-allSetsTop.html#cb2452-35"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2452-36"><a href="dsi-allCells-allSetsTop.html#cb2452-36"></a>        <span class="kw">paste</span>(<span class="st">&quot;monocytes&quot;</span>, genex,</span>
<span id="cb2452-37"><a href="dsi-allCells-allSetsTop.html#cb2452-37"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2452-38"><a href="dsi-allCells-allSetsTop.html#cb2452-38"></a>    )</span>
<span id="cb2452-39"><a href="dsi-allCells-allSetsTop.html#cb2452-39"></a>pM &lt;-<span class="st"> </span>p</span>
<span id="cb2452-40"><a href="dsi-allCells-allSetsTop.html#cb2452-40"></a></span>
<span id="cb2452-41"><a href="dsi-allCells-allSetsTop.html#cb2452-41"></a><span class="co">#erythrocytes:</span></span>
<span id="cb2452-42"><a href="dsi-allCells-allSetsTop.html#cb2452-42"></a></span>
<span id="cb2452-43"><a href="dsi-allCells-allSetsTop.html#cb2452-43"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">4</span>]</span>
<span id="cb2452-44"><a href="dsi-allCells-allSetsTop.html#cb2452-44"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> genex, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2452-45"><a href="dsi-allCells-allSetsTop.html#cb2452-45"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2452-46"><a href="dsi-allCells-allSetsTop.html#cb2452-46"></a>        <span class="kw">paste</span>(<span class="st">&quot;erythrocytes&quot;</span>, genex,</span>
<span id="cb2452-47"><a href="dsi-allCells-allSetsTop.html#cb2452-47"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2452-48"><a href="dsi-allCells-allSetsTop.html#cb2452-48"></a>    )</span>
<span id="cb2452-49"><a href="dsi-allCells-allSetsTop.html#cb2452-49"></a>pE &lt;-<span class="st"> </span>p</span></code></pre></div>
<div class="sourceCode" id="cb2453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2453-1"><a href="dsi-allCells-allSetsTop.html#cb2453-1"></a><span class="kw">grid.arrange</span>(pB <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2453-2"><a href="dsi-allCells-allSetsTop.html#cb2453-2"></a>                        pT <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2453-3"><a href="dsi-allCells-allSetsTop.html#cb2453-3"></a>                        pM <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2453-4"><a href="dsi-allCells-allSetsTop.html#cb2453-4"></a>                        pE <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2453-5"><a href="dsi-allCells-allSetsTop.html#cb2453-5"></a>                        <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/fastmnn_diagTsneCellTypeMarkerShow_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Compare to the uncorrected values:</p>
<div class="sourceCode" id="cb2454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2454-1"><a href="dsi-allCells-allSetsTop.html#cb2454-1"></a><span class="co"># B cells</span></span>
<span id="cb2454-2"><a href="dsi-allCells-allSetsTop.html#cb2454-2"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">1</span>]</span>
<span id="cb2454-3"><a href="dsi-allCells-allSetsTop.html#cb2454-3"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by =</span> genex, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2454-4"><a href="dsi-allCells-allSetsTop.html#cb2454-4"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2454-5"><a href="dsi-allCells-allSetsTop.html#cb2454-5"></a>        <span class="kw">paste</span>(<span class="st">&quot;B cells&quot;</span>, genex,</span>
<span id="cb2454-6"><a href="dsi-allCells-allSetsTop.html#cb2454-6"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2454-7"><a href="dsi-allCells-allSetsTop.html#cb2454-7"></a>    )</span>
<span id="cb2454-8"><a href="dsi-allCells-allSetsTop.html#cb2454-8"></a>pBu &lt;-<span class="st"> </span>p</span>
<span id="cb2454-9"><a href="dsi-allCells-allSetsTop.html#cb2454-9"></a></span>
<span id="cb2454-10"><a href="dsi-allCells-allSetsTop.html#cb2454-10"></a><span class="co">#Compare to the uncorrected values, T cells:</span></span>
<span id="cb2454-11"><a href="dsi-allCells-allSetsTop.html#cb2454-11"></a></span>
<span id="cb2454-12"><a href="dsi-allCells-allSetsTop.html#cb2454-12"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">3</span>]</span>
<span id="cb2454-13"><a href="dsi-allCells-allSetsTop.html#cb2454-13"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by =</span> genex, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2454-14"><a href="dsi-allCells-allSetsTop.html#cb2454-14"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2454-15"><a href="dsi-allCells-allSetsTop.html#cb2454-15"></a>        <span class="kw">paste</span>(<span class="st">&quot;T cells&quot;</span>, genex,</span>
<span id="cb2454-16"><a href="dsi-allCells-allSetsTop.html#cb2454-16"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2454-17"><a href="dsi-allCells-allSetsTop.html#cb2454-17"></a>    )</span>
<span id="cb2454-18"><a href="dsi-allCells-allSetsTop.html#cb2454-18"></a>pTu &lt;-<span class="st"> </span>p</span>
<span id="cb2454-19"><a href="dsi-allCells-allSetsTop.html#cb2454-19"></a></span>
<span id="cb2454-20"><a href="dsi-allCells-allSetsTop.html#cb2454-20"></a><span class="co">#Compare to the uncorrected values, monocytes:</span></span>
<span id="cb2454-21"><a href="dsi-allCells-allSetsTop.html#cb2454-21"></a></span>
<span id="cb2454-22"><a href="dsi-allCells-allSetsTop.html#cb2454-22"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">2</span>]</span>
<span id="cb2454-23"><a href="dsi-allCells-allSetsTop.html#cb2454-23"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by =</span> genex, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2454-24"><a href="dsi-allCells-allSetsTop.html#cb2454-24"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2454-25"><a href="dsi-allCells-allSetsTop.html#cb2454-25"></a>        <span class="kw">paste</span>(<span class="st">&quot;monocytes&quot;</span>, genex,</span>
<span id="cb2454-26"><a href="dsi-allCells-allSetsTop.html#cb2454-26"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2454-27"><a href="dsi-allCells-allSetsTop.html#cb2454-27"></a>    )</span>
<span id="cb2454-28"><a href="dsi-allCells-allSetsTop.html#cb2454-28"></a>pMu &lt;-<span class="st"> </span>p</span>
<span id="cb2454-29"><a href="dsi-allCells-allSetsTop.html#cb2454-29"></a></span>
<span id="cb2454-30"><a href="dsi-allCells-allSetsTop.html#cb2454-30"></a><span class="co">#Compare to the uncorrected values, erythrocytes:</span></span>
<span id="cb2454-31"><a href="dsi-allCells-allSetsTop.html#cb2454-31"></a></span>
<span id="cb2454-32"><a href="dsi-allCells-allSetsTop.html#cb2454-32"></a>genex &lt;-<span class="st"> </span>ensToShow[<span class="dv">4</span>]</span>
<span id="cb2454-33"><a href="dsi-allCells-allSetsTop.html#cb2454-33"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(uncorrected, <span class="dt">colour_by =</span> genex, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2454-34"><a href="dsi-allCells-allSetsTop.html#cb2454-34"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(</span>
<span id="cb2454-35"><a href="dsi-allCells-allSetsTop.html#cb2454-35"></a>        <span class="kw">paste</span>(<span class="st">&quot;erythrocytes&quot;</span>, genex,</span>
<span id="cb2454-36"><a href="dsi-allCells-allSetsTop.html#cb2454-36"></a>        <span class="kw">rowData</span>(uncorrected)[genex,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2454-37"><a href="dsi-allCells-allSetsTop.html#cb2454-37"></a>    )</span>
<span id="cb2454-38"><a href="dsi-allCells-allSetsTop.html#cb2454-38"></a>pEu &lt;-<span class="st"> </span>p</span></code></pre></div>
<div class="sourceCode" id="cb2455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2455-1"><a href="dsi-allCells-allSetsTop.html#cb2455-1"></a><span class="kw">grid.arrange</span>(pBu <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2455-2"><a href="dsi-allCells-allSetsTop.html#cb2455-2"></a>                        pTu <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2455-3"><a href="dsi-allCells-allSetsTop.html#cb2455-3"></a>                        pMu <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2455-4"><a href="dsi-allCells-allSetsTop.html#cb2455-4"></a>                        pEu <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2455-5"><a href="dsi-allCells-allSetsTop.html#cb2455-5"></a>                        <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/uncorr_diagTsneCellTypeMarkerShow_dsi_allCells_allSets-1.png" width="672" /></p>
<div id="preserving-biological-heterogeneity-10" class="section level3">
<h3><span class="header-section-number">29.7.1</span> Preserving biological heterogeneity</h3>
<div id="comparison-to-within-batch-clusters-3" class="section level4">
<h4><span class="header-section-number">29.7.1.1</span> Comparison to within-batch clusters</h4>
<p>Another useful diagnostic check is to compare the clustering within each batch to the clustering of the merged data. Accurate data integration should preserve variance within each batch as there should be nothing to remove between cells in the same batch. This check complements the previously mentioned diagnostics that only focus on the removal of differences between batches. Specifically, it protects us against cases where the correction method simply aggregates all cells together, which would achieve perfect mixing but also discard the biological heterogeneity of interest.</p>
<p>Another useful diagnostic check is to compare the clustering within each batch to the clustering of the merged data. Accurate data integration should preserve variance within each batch as there should be nothing to remove between cells in the same batch. This check complements the previously mentioned diagnostics that only focus on the removal of differences between batches. Specifically, it protects us against cases where the correction method simply aggregates all cells together, which would achieve perfect mixing but also discard the biological heterogeneity of interest.</p>
<p>Ideally, we should see a many-to-1 mapping where the across-batch clustering is nested inside the within-batch clusterings. This indicates that any within-batch structure was preserved after correction while acknowledging that greater resolution is possible with more cells. In practice, more discrepancies can be expected even when the correction is perfect, due to the existence of closely related clusters that were arbitrarily separated in the within-batch clustering. As a general rule, we can be satisfied with the correction if the vast majority of entries are zero, though this may depend on whether specific clusters of interest are gained or lost.</p>
<p>One heatmap is generated for each dataset, where each entry is colored according to the number of cells with each pair of labels (before and after correction), on the log10 scale with pseudocounts (+10) for a smoother color transition (so a minimum value of log10(0+10) == 1).</p>
<div class="sourceCode" id="cb2456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2456-1"><a href="dsi-allCells-allSetsTop.html#cb2456-1"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2456-2"><a href="dsi-allCells-allSetsTop.html#cb2456-2"></a>treeList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2456-3"><a href="dsi-allCells-allSetsTop.html#cb2456-3"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2456-4"><a href="dsi-allCells-allSetsTop.html#cb2456-4"></a>  <span class="co"># heatmap</span></span>
<span id="cb2456-5"><a href="dsi-allCells-allSetsTop.html#cb2456-5"></a>  tab &lt;-<span class="st"> </span><span class="kw">table</span>(</span>
<span id="cb2456-6"><a href="dsi-allCells-allSetsTop.html#cb2456-6"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]), <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),</span>
<span id="cb2456-7"><a href="dsi-allCells-allSetsTop.html#cb2456-7"></a>    <span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]], <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb2456-8"><a href="dsi-allCells-allSetsTop.html#cb2456-8"></a>    )</span>
<span id="cb2456-9"><a href="dsi-allCells-allSetsTop.html#cb2456-9"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb2456-10"><a href="dsi-allCells-allSetsTop.html#cb2456-10"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2456-11"><a href="dsi-allCells-allSetsTop.html#cb2456-11"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2456-12"><a href="dsi-allCells-allSetsTop.html#cb2456-12"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2456-13"><a href="dsi-allCells-allSetsTop.html#cb2456-13"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s&quot;</span>,</span>
<span id="cb2456-14"><a href="dsi-allCells-allSetsTop.html#cb2456-14"></a>                                              splVec[splIdx]),</span>
<span id="cb2456-15"><a href="dsi-allCells-allSetsTop.html#cb2456-15"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>,</span>
<span id="cb2456-16"><a href="dsi-allCells-allSetsTop.html#cb2456-16"></a>                                 <span class="dt">fontsize=</span><span class="dv">7</span>)</span>
<span id="cb2456-17"><a href="dsi-allCells-allSetsTop.html#cb2456-17"></a>  <span class="co"># cluster tree:</span></span>
<span id="cb2456-18"><a href="dsi-allCells-allSetsTop.html#cb2456-18"></a>  combined &lt;-<span class="st"> </span><span class="kw">cbind</span>(</span>
<span id="cb2456-19"><a href="dsi-allCells-allSetsTop.html#cb2456-19"></a>    <span class="dt">cl.1=</span><span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]),</span>
<span id="cb2456-20"><a href="dsi-allCells-allSetsTop.html#cb2456-20"></a>    <span class="dt">cl.2=</span>clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb2456-21"><a href="dsi-allCells-allSetsTop.html#cb2456-21"></a>  treeList[[splIdx]]  &lt;-<span class="st"> </span><span class="kw">clustree</span>(combined, <span class="dt">prefix=</span><span class="st">&quot;cl.&quot;</span>, <span class="dt">edge_arrow=</span><span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2456-22"><a href="dsi-allCells-allSetsTop.html#cb2456-22"></a><span class="st">    </span><span class="kw">ggtitle</span>(splVec[splIdx]) <span class="op">+</span></span>
<span id="cb2456-23"><a href="dsi-allCells-allSetsTop.html#cb2456-23"></a><span class="st">    </span><span class="co">#theme(legend.background = element_rect(color = &quot;yellow&quot;)) +</span></span>
<span id="cb2456-24"><a href="dsi-allCells-allSetsTop.html#cb2456-24"></a><span class="st">    </span><span class="co">#theme(legend.position=&#39;bottom&#39;) +</span></span>
<span id="cb2456-25"><a href="dsi-allCells-allSetsTop.html#cb2456-25"></a><span class="st">    </span><span class="co">#theme(legend.box=&quot;vertical&quot;) +</span></span>
<span id="cb2456-26"><a href="dsi-allCells-allSetsTop.html#cb2456-26"></a><span class="st">    </span><span class="co">#theme(legend.box=&quot;horizontal&quot;) +</span></span>
<span id="cb2456-27"><a href="dsi-allCells-allSetsTop.html#cb2456-27"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.margin=</span><span class="kw">margin</span>()) <span class="co">#+</span></span>
<span id="cb2456-28"><a href="dsi-allCells-allSetsTop.html#cb2456-28"></a>    <span class="co">#guides(fill=guide_legend(nrow=2, byrow=FALSE))</span></span>
<span id="cb2456-29"><a href="dsi-allCells-allSetsTop.html#cb2456-29"></a>    <span class="co">#theme(legend.position = &quot;none&quot;)</span></span>
<span id="cb2456-30"><a href="dsi-allCells-allSetsTop.html#cb2456-30"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb2457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2457-1"><a href="dsi-allCells-allSetsTop.html#cb2457-1"></a><span class="kw">length</span>(treeList)</span></code></pre></div>
<pre><code>## [1] 11</code></pre>
<div class="sourceCode" id="cb2459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2459-1"><a href="dsi-allCells-allSetsTop.html#cb2459-1"></a><span class="kw">names</span>(treeList)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb2461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2461-1"><a href="dsi-allCells-allSetsTop.html#cb2461-1"></a><span class="kw">class</span>(treeList[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>## [1] &quot;ggraph&quot; &quot;gg&quot;     &quot;ggplot&quot;</code></pre>
<div class="sourceCode" id="cb2463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2463-1"><a href="dsi-allCells-allSetsTop.html#cb2463-1"></a>treeList[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/biolHet_plotDraw_dsi_allCells_allSets_debug-1.png" width="672" /></p>
<div class="sourceCode" id="cb2464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2464-1"><a href="dsi-allCells-allSetsTop.html#cb2464-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co"> debug</span></span>
<span id="cb2464-2"><a href="dsi-allCells-allSetsTop.html#cb2464-2"></a>g_legend&lt;-<span class="cf">function</span>(a.gplot){</span>
<span id="cb2464-3"><a href="dsi-allCells-allSetsTop.html#cb2464-3"></a>   tmp &lt;-<span class="st"> </span><span class="kw">ggplot_gtable</span>(<span class="kw">ggplot_build</span>(a.gplot))</span>
<span id="cb2464-4"><a href="dsi-allCells-allSetsTop.html#cb2464-4"></a>   leg &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">sapply</span>(tmp<span class="op">$</span>grobs, <span class="cf">function</span>(x) x<span class="op">$</span>name) <span class="op">==</span><span class="st"> &quot;guide-box&quot;</span>)</span>
<span id="cb2464-5"><a href="dsi-allCells-allSetsTop.html#cb2464-5"></a>   legend &lt;-<span class="st"> </span>tmp<span class="op">$</span>grobs[[leg]]</span>
<span id="cb2464-6"><a href="dsi-allCells-allSetsTop.html#cb2464-6"></a>   <span class="kw">return</span>(legend)</span>
<span id="cb2464-7"><a href="dsi-allCells-allSetsTop.html#cb2464-7"></a>}</span>
<span id="cb2464-8"><a href="dsi-allCells-allSetsTop.html#cb2464-8"></a></span>
<span id="cb2464-9"><a href="dsi-allCells-allSetsTop.html#cb2464-9"></a>redrawClutree &lt;-<span class="st"> </span><span class="cf">function</span>(p){</span>
<span id="cb2464-10"><a href="dsi-allCells-allSetsTop.html#cb2464-10"></a><span class="co">#p &lt;- treeList[[1]] + theme(legend.position=&#39;bottom&#39;)</span></span>
<span id="cb2464-11"><a href="dsi-allCells-allSetsTop.html#cb2464-11"></a><span class="co">#p &lt;- p + theme(legend.background = element_rect(color = &quot;yellow&quot;))</span></span>
<span id="cb2464-12"><a href="dsi-allCells-allSetsTop.html#cb2464-12"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.justification =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb2464-13"><a href="dsi-allCells-allSetsTop.html#cb2464-13"></a><span class="co">#p &lt;- p + theme(legend.justification = c(0,1))</span></span>
<span id="cb2464-14"><a href="dsi-allCells-allSetsTop.html#cb2464-14"></a><span class="co">#lemon::gtable_show_names(p)</span></span>
<span id="cb2464-15"><a href="dsi-allCells-allSetsTop.html#cb2464-15"></a>pNoLeg &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb2464-16"><a href="dsi-allCells-allSetsTop.html#cb2464-16"></a><span class="co"># edge colour:</span></span>
<span id="cb2464-17"><a href="dsi-allCells-allSetsTop.html#cb2464-17"></a>pEdgeCol &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2464-18"><a href="dsi-allCells-allSetsTop.html#cb2464-18"></a><span class="st">  </span><span class="co">#guides(edge_colour = FALSE) +</span></span>
<span id="cb2464-19"><a href="dsi-allCells-allSetsTop.html#cb2464-19"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-20"><a href="dsi-allCells-allSetsTop.html#cb2464-20"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-21"><a href="dsi-allCells-allSetsTop.html#cb2464-21"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2464-22"><a href="dsi-allCells-allSetsTop.html#cb2464-22"></a>pEdgeCol.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pEdgeCol)</span>
<span id="cb2464-23"><a href="dsi-allCells-allSetsTop.html#cb2464-23"></a><span class="co"># edge alpha:</span></span>
<span id="cb2464-24"><a href="dsi-allCells-allSetsTop.html#cb2464-24"></a>pEdgeAlpha &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2464-25"><a href="dsi-allCells-allSetsTop.html#cb2464-25"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-26"><a href="dsi-allCells-allSetsTop.html#cb2464-26"></a><span class="st">  </span><span class="co">#guides(edge_alpha = FALSE) +</span></span>
<span id="cb2464-27"><a href="dsi-allCells-allSetsTop.html#cb2464-27"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-28"><a href="dsi-allCells-allSetsTop.html#cb2464-28"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2464-29"><a href="dsi-allCells-allSetsTop.html#cb2464-29"></a>pEdgeAlpha.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pEdgeAlpha)</span>
<span id="cb2464-30"><a href="dsi-allCells-allSetsTop.html#cb2464-30"></a><span class="co"># size</span></span>
<span id="cb2464-31"><a href="dsi-allCells-allSetsTop.html#cb2464-31"></a>pSize &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2464-32"><a href="dsi-allCells-allSetsTop.html#cb2464-32"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-33"><a href="dsi-allCells-allSetsTop.html#cb2464-33"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-34"><a href="dsi-allCells-allSetsTop.html#cb2464-34"></a><span class="st">  </span><span class="co">#guides(size = FALSE) +</span></span>
<span id="cb2464-35"><a href="dsi-allCells-allSetsTop.html#cb2464-35"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2464-36"><a href="dsi-allCells-allSetsTop.html#cb2464-36"></a>pSize.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pSize)</span>
<span id="cb2464-37"><a href="dsi-allCells-allSetsTop.html#cb2464-37"></a><span class="co"># colour</span></span>
<span id="cb2464-38"><a href="dsi-allCells-allSetsTop.html#cb2464-38"></a>pColour &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2464-39"><a href="dsi-allCells-allSetsTop.html#cb2464-39"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-40"><a href="dsi-allCells-allSetsTop.html#cb2464-40"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2464-41"><a href="dsi-allCells-allSetsTop.html#cb2464-41"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="co">#+</span></span>
<span id="cb2464-42"><a href="dsi-allCells-allSetsTop.html#cb2464-42"></a>  <span class="co">#guides(colour = FALSE) </span></span>
<span id="cb2464-43"><a href="dsi-allCells-allSetsTop.html#cb2464-43"></a>pColour.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pColour)</span>
<span id="cb2464-44"><a href="dsi-allCells-allSetsTop.html#cb2464-44"></a></span>
<span id="cb2464-45"><a href="dsi-allCells-allSetsTop.html#cb2464-45"></a>grobx &lt;-<span class="st"> </span><span class="kw">arrangeGrob</span>(pNoLeg,</span>
<span id="cb2464-46"><a href="dsi-allCells-allSetsTop.html#cb2464-46"></a>                        pEdgeCol.leg,</span>
<span id="cb2464-47"><a href="dsi-allCells-allSetsTop.html#cb2464-47"></a>                        pEdgeAlpha.leg,</span>
<span id="cb2464-48"><a href="dsi-allCells-allSetsTop.html#cb2464-48"></a>                        <span class="co">#pColour.leg,</span></span>
<span id="cb2464-49"><a href="dsi-allCells-allSetsTop.html#cb2464-49"></a>                        pSize.leg,</span>
<span id="cb2464-50"><a href="dsi-allCells-allSetsTop.html#cb2464-50"></a>                        <span class="dt">nrow=</span><span class="dv">1</span>, <span class="dt">ncol=</span><span class="dv">4</span>,</span>
<span id="cb2464-51"><a href="dsi-allCells-allSetsTop.html#cb2464-51"></a>                        <span class="dt">layout_matrix=</span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="dt">ncol=</span><span class="dv">4</span>, <span class="dt">byrow=</span><span class="ot">TRUE</span>),</span>
<span id="cb2464-52"><a href="dsi-allCells-allSetsTop.html#cb2464-52"></a>                        <span class="dt">widths=</span><span class="kw">c</span>(<span class="kw">unit</span>(.<span class="dv">64</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2464-53"><a href="dsi-allCells-allSetsTop.html#cb2464-53"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2464-54"><a href="dsi-allCells-allSetsTop.html#cb2464-54"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2464-55"><a href="dsi-allCells-allSetsTop.html#cb2464-55"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>))</span>
<span id="cb2464-56"><a href="dsi-allCells-allSetsTop.html#cb2464-56"></a>                        )</span>
<span id="cb2464-57"><a href="dsi-allCells-allSetsTop.html#cb2464-57"></a>}</span>
<span id="cb2464-58"><a href="dsi-allCells-allSetsTop.html#cb2464-58"></a><span class="co">##gx &lt;- redrawClutree(treeList[[1]] + theme(legend.position=&#39;bottom&#39;))</span></span>
<span id="cb2464-59"><a href="dsi-allCells-allSetsTop.html#cb2464-59"></a><span class="co">##grid::grid.draw(gx)</span></span>
<span id="cb2464-60"><a href="dsi-allCells-allSetsTop.html#cb2464-60"></a><span class="co">## fine # gxList &lt;- lapply(treeList, function(x){redrawClutree(x+theme(legend.position=&#39;bottom&#39;))})</span></span>
<span id="cb2464-61"><a href="dsi-allCells-allSetsTop.html#cb2464-61"></a>gxList &lt;-<span class="st"> </span><span class="kw">lapply</span>(treeList, <span class="cf">function</span>(x){<span class="kw">redrawClutree</span>(x)})</span>
<span id="cb2464-62"><a href="dsi-allCells-allSetsTop.html#cb2464-62"></a><span class="co">##marrangeGrob(gxList, nrow=2, ncol=2)</span></span></code></pre></div>
<div class="sourceCode" id="cb2465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2465-1"><a href="dsi-allCells-allSetsTop.html#cb2465-1"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2465-2"><a href="dsi-allCells-allSetsTop.html#cb2465-2"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2465-3"><a href="dsi-allCells-allSetsTop.html#cb2465-3"></a>      <span class="dt">ncol=</span><span class="dv">3</span>,</span>
<span id="cb2465-4"><a href="dsi-allCells-allSetsTop.html#cb2465-4"></a>      <span class="dt">top =</span> grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&quot;clusterings concordance (number of cells, log10 scale)&quot;</span>,</span>
<span id="cb2465-5"><a href="dsi-allCells-allSetsTop.html#cb2465-5"></a>                     <span class="dt">gp=</span>grid<span class="op">::</span><span class="kw">gpar</span>(<span class="dt">fontsize=</span><span class="dv">12</span>,<span class="dt">font=</span><span class="dv">3</span>))</span>
<span id="cb2465-6"><a href="dsi-allCells-allSetsTop.html#cb2465-6"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/biolHet_plotShowHeatmap_dsi_allCells_allSets-1.png" width="672" /></p>
<p>The redistribution of cells from one set of clusters to another, here ‘within-batch before’ and ‘across-batch after’ correction, may also be visualized with a clustering tree <a href="https://cran.r-project.org/package=clustree">clustree</a>. Clusters are represented as filled circles colored by cluster set (‘before’ in pink, ‘after’ in blue) and sized by cell number. A pair of clusters from two sets are linked according to the number of cells they share with a link that informs on the number of cells shared (color) and the ‘incoming node’ proportion for the node it points to (transparency). Although these plots convey more information than heatmaps below, they may not be as easy to read.</p>
<div class="sourceCode" id="cb2466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2466-1"><a href="dsi-allCells-allSetsTop.html#cb2466-1"></a><span class="co">#```{r biolHet_plotShowTree1_dsi_allCells_allSets, fig.height=figSize*length(treeList)*2/3, fig.width=figSize}</span></span>
<span id="cb2466-2"><a href="dsi-allCells-allSetsTop.html#cb2466-2"></a><span class="co">#grid.arrange(grobs = treeList,</span></span>
<span id="cb2466-3"><a href="dsi-allCells-allSetsTop.html#cb2466-3"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],</span>
<span id="cb2466-4"><a href="dsi-allCells-allSetsTop.html#cb2466-4"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2466-5"><a href="dsi-allCells-allSetsTop.html#cb2466-5"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb2467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2467-1"><a href="dsi-allCells-allSetsTop.html#cb2467-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">5</span><span class="op">:</span><span class="dv">6</span>],</span>
<span id="cb2467-2"><a href="dsi-allCells-allSetsTop.html#cb2467-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2467-3"><a href="dsi-allCells-allSetsTop.html#cb2467-3"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb2468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2468-1"><a href="dsi-allCells-allSetsTop.html#cb2468-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">7</span><span class="op">:</span><span class="dv">8</span>],</span>
<span id="cb2468-2"><a href="dsi-allCells-allSetsTop.html#cb2468-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2468-3"><a href="dsi-allCells-allSetsTop.html#cb2468-3"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb2469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2469-1"><a href="dsi-allCells-allSetsTop.html#cb2469-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">9</span><span class="op">:</span><span class="dv">11</span>],</span>
<span id="cb2469-2"><a href="dsi-allCells-allSetsTop.html#cb2469-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2469-3"><a href="dsi-allCells-allSetsTop.html#cb2469-3"></a>)</span></code></pre></div>
<p>The same plots in more compact form with no legend:</p>
<!-- remove legend and have plots in two columns -->
<div class="sourceCode" id="cb2470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2470-1"><a href="dsi-allCells-allSetsTop.html#cb2470-1"></a>treeList &lt;-<span class="st"> </span><span class="kw">lapply</span>(treeList, <span class="cf">function</span>(p){</span>
<span id="cb2470-2"><a href="dsi-allCells-allSetsTop.html#cb2470-2"></a>  p <span class="op">+</span></span>
<span id="cb2470-3"><a href="dsi-allCells-allSetsTop.html#cb2470-3"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2470-4"><a href="dsi-allCells-allSetsTop.html#cb2470-4"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2470-5"><a href="dsi-allCells-allSetsTop.html#cb2470-5"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2470-6"><a href="dsi-allCells-allSetsTop.html#cb2470-6"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2470-7"><a href="dsi-allCells-allSetsTop.html#cb2470-7"></a>})</span></code></pre></div>
<div class="sourceCode" id="cb2471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2471-1"><a href="dsi-allCells-allSetsTop.html#cb2471-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> treeList,</span>
<span id="cb2471-2"><a href="dsi-allCells-allSetsTop.html#cb2471-2"></a>      <span class="dt">ncol=</span><span class="dv">2</span></span>
<span id="cb2471-3"><a href="dsi-allCells-allSetsTop.html#cb2471-3"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/biolHet_plotShowTree5_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
<div id="coassignment-probabilities-8" class="section level4">
<h4><span class="header-section-number">29.7.1.2</span> Coassignment probabilities</h4>
<p>Another evaluation approach is to compute the <strong>coassignment probabilities</strong>, i.e. the probability that cells from two within-batch clusters are clustered together in the across-batch clustering. High probabilities off the diagonal indicate that within-batch clusters are merged in the across-batch analysis. We would generally expect low off-diagonal probabilities for most pairs of clusters, though this may not be reasonably possible if the within-batch clusters were poorly separated in the first place.</p>
<p>The plots below display the coassignment probabilities for the within-batch clusters, based on coassignment of cells in the across-batch clusters obtained after MNN correction. One heatmap is generated for each sample, where each entry is colored according to the coassignment probability between each pair of within-batch clusters:</p>
<div class="sourceCode" id="cb2472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2472-1"><a href="dsi-allCells-allSetsTop.html#cb2472-1"></a><span class="co"># coassignProb manual: now deprecated for pairwiseRand. </span></span>
<span id="cb2472-2"><a href="dsi-allCells-allSetsTop.html#cb2472-2"></a><span class="co"># Note that the coassignment probability is closely related to the Rand index-based ratios broken down by cluster pair in pairwiseRand with mode=&quot;ratio&quot; and adjusted=FALSE. The off-diagonal coassignment probabilities are simply 1 minus the off-diagonal ratio, while the on-diagonal values differ only by the lack of consideration of pairs of the same cell in pairwiseRand. </span></span>
<span id="cb2472-3"><a href="dsi-allCells-allSetsTop.html#cb2472-3"></a></span>
<span id="cb2472-4"><a href="dsi-allCells-allSetsTop.html#cb2472-4"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2472-5"><a href="dsi-allCells-allSetsTop.html#cb2472-5"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2472-6"><a href="dsi-allCells-allSetsTop.html#cb2472-6"></a>  tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]),</span>
<span id="cb2472-7"><a href="dsi-allCells-allSetsTop.html#cb2472-7"></a>                      clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb2472-8"><a href="dsi-allCells-allSetsTop.html#cb2472-8"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab,</span>
<span id="cb2472-9"><a href="dsi-allCells-allSetsTop.html#cb2472-9"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2472-10"><a href="dsi-allCells-allSetsTop.html#cb2472-10"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2472-11"><a href="dsi-allCells-allSetsTop.html#cb2472-11"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2472-12"><a href="dsi-allCells-allSetsTop.html#cb2472-12"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[splIdx]),</span>
<span id="cb2472-13"><a href="dsi-allCells-allSetsTop.html#cb2472-13"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2472-14"><a href="dsi-allCells-allSetsTop.html#cb2472-14"></a>}</span>
<span id="cb2472-15"><a href="dsi-allCells-allSetsTop.html#cb2472-15"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2472-16"><a href="dsi-allCells-allSetsTop.html#cb2472-16"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2472-17"><a href="dsi-allCells-allSetsTop.html#cb2472-17"></a>      <span class="dt">ncol=</span><span class="dv">3</span></span>
<span id="cb2472-18"><a href="dsi-allCells-allSetsTop.html#cb2472-18"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/coasProb_plot_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Note that the coassignment probability is closely related to the Rand index-based ratios broken down by cluster pair (in <code>pairwiseRand()</code> with mode=“ratio” and adjusted=FALSE). The Rand index is introduced below.</p>
</div>
<div id="rand-index-9" class="section level4">
<h4><span class="header-section-number">29.7.1.3</span> Rand index</h4>
<p>Finally, we can summarize the agreement between clusterings by computing the Rand index. This provides a simple metric that we can use to assess the preservation of variation by different correction methods. Larger rand indices (i.e., closer to 1) are more desirable, though this must be balanced against the ability of each method to actually remove the batch effect.</p>
<div class="sourceCode" id="cb2473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2473-1"><a href="dsi-allCells-allSetsTop.html#cb2473-1"></a><span class="co"># OLD</span></span>
<span id="cb2473-2"><a href="dsi-allCells-allSetsTop.html#cb2473-2"></a></span>
<span id="cb2473-3"><a href="dsi-allCells-allSetsTop.html#cb2473-3"></a></span>
<span id="cb2473-4"><a href="dsi-allCells-allSetsTop.html#cb2473-4"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(fossil))</span>
<span id="cb2473-5"><a href="dsi-allCells-allSetsTop.html#cb2473-5"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb2473-6"><a href="dsi-allCells-allSetsTop.html#cb2473-6"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2473-7"><a href="dsi-allCells-allSetsTop.html#cb2473-7"></a>ri1 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb2473-8"><a href="dsi-allCells-allSetsTop.html#cb2473-8"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2473-9"><a href="dsi-allCells-allSetsTop.html#cb2473-9"></a>ri1</span>
<span id="cb2473-10"><a href="dsi-allCells-allSetsTop.html#cb2473-10"></a></span>
<span id="cb2473-11"><a href="dsi-allCells-allSetsTop.html#cb2473-11"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb2473-12"><a href="dsi-allCells-allSetsTop.html#cb2473-12"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2473-13"><a href="dsi-allCells-allSetsTop.html#cb2473-13"></a>ri2 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb2473-14"><a href="dsi-allCells-allSetsTop.html#cb2473-14"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2473-15"><a href="dsi-allCells-allSetsTop.html#cb2473-15"></a>ri2</span></code></pre></div>
<div class="sourceCode" id="cb2474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2474-1"><a href="dsi-allCells-allSetsTop.html#cb2474-1"></a><span class="co"># pairwiseRand(), index, adjusted</span></span>
<span id="cb2474-2"><a href="dsi-allCells-allSetsTop.html#cb2474-2"></a>ariVec &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;numeric&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2474-3"><a href="dsi-allCells-allSetsTop.html#cb2474-3"></a><span class="kw">names</span>(ariVec) &lt;-<span class="st"> </span>splVec</span>
<span id="cb2474-4"><a href="dsi-allCells-allSetsTop.html#cb2474-4"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2474-5"><a href="dsi-allCells-allSetsTop.html#cb2474-5"></a>  ariVec[splIdx] &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb2474-6"><a href="dsi-allCells-allSetsTop.html#cb2474-6"></a>    <span class="dt">ref=</span><span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]])),</span>
<span id="cb2474-7"><a href="dsi-allCells-allSetsTop.html#cb2474-7"></a>    <span class="dt">alt=</span><span class="kw">as.integer</span>(clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb2474-8"><a href="dsi-allCells-allSetsTop.html#cb2474-8"></a>    <span class="dt">mode=</span><span class="st">&quot;index&quot;</span>)</span>
<span id="cb2474-9"><a href="dsi-allCells-allSetsTop.html#cb2474-9"></a>}</span>
<span id="cb2474-10"><a href="dsi-allCells-allSetsTop.html#cb2474-10"></a>ariVec &lt;-<span class="st"> </span><span class="kw">round</span>(ariVec,<span class="dv">2</span>)</span>
<span id="cb2474-11"><a href="dsi-allCells-allSetsTop.html#cb2474-11"></a>ariVec</span></code></pre></div>
<pre><code>## ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4        HHD_1        HHD_2 
##         0.30         0.24         0.73         0.42         0.35         0.48 
##      PRE-T_1      PRE-T_2      PBMMC_1      PBMMC_2      PBMMC_3 
##         0.41         0.54         0.59         0.72         0.69</code></pre>
<p>A sample may show a low Rand index value if cells grouped together in a small cluster before correction are split into distinct clusters after correction because the latter comprise cell populations not observed in that sample but present in other samples.</p>
<p>This would be the case of GSM3872434 aka ETV6-RUNX1_1 with far fewer erythrocytes (grouped in a single cluster) than GSM3872443 aka PBMMC_2, in which subtypes can be distinguished.
<!--
--></p>
<p>We can also break down the <strong>adjusted Rand index (ARI)</strong> into per-cluster ratios for more detailed diagnostics. For example, we could see low ratios off the diagonal if distinct clusters in the within-batch clustering were incorrectly aggregated in the merged clustering. Conversely, we might see low ratios on the diagonal if the correction inflated or introduced spurious heterogeneity inside a within-batch cluster.</p>
<div class="sourceCode" id="cb2476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2476-1"><a href="dsi-allCells-allSetsTop.html#cb2476-1"></a><span class="co"># pairwiseRand(), ratio, adjusted</span></span>
<span id="cb2476-2"><a href="dsi-allCells-allSetsTop.html#cb2476-2"></a><span class="co"># square numeric matrix is returned with number of rows equal to the number of unique levels in ref.</span></span>
<span id="cb2476-3"><a href="dsi-allCells-allSetsTop.html#cb2476-3"></a></span>
<span id="cb2476-4"><a href="dsi-allCells-allSetsTop.html#cb2476-4"></a>tabList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2476-5"><a href="dsi-allCells-allSetsTop.html#cb2476-5"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2476-6"><a href="dsi-allCells-allSetsTop.html#cb2476-6"></a>  tabList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb2476-7"><a href="dsi-allCells-allSetsTop.html#cb2476-7"></a>      <span class="dt">ref=</span><span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]])),</span>
<span id="cb2476-8"><a href="dsi-allCells-allSetsTop.html#cb2476-8"></a>    <span class="dt">alt=</span><span class="kw">as.integer</span>(clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb2476-9"><a href="dsi-allCells-allSetsTop.html#cb2476-9"></a>    )</span>
<span id="cb2476-10"><a href="dsi-allCells-allSetsTop.html#cb2476-10"></a>}</span>
<span id="cb2476-11"><a href="dsi-allCells-allSetsTop.html#cb2476-11"></a>randVal &lt;-<span class="st"> </span><span class="kw">unlist</span>(tabList) </span>
<span id="cb2476-12"><a href="dsi-allCells-allSetsTop.html#cb2476-12"></a></span>
<span id="cb2476-13"><a href="dsi-allCells-allSetsTop.html#cb2476-13"></a><span class="co">## make breaks from combined range</span></span>
<span id="cb2476-14"><a href="dsi-allCells-allSetsTop.html#cb2476-14"></a>limits &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb2476-15"><a href="dsi-allCells-allSetsTop.html#cb2476-15"></a>  <span class="kw">min</span>(randVal, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb2476-16"><a href="dsi-allCells-allSetsTop.html#cb2476-16"></a>  <span class="kw">max</span>(randVal, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>))</span>
<span id="cb2476-17"><a href="dsi-allCells-allSetsTop.html#cb2476-17"></a>limits &lt;-<span class="st"> </span><span class="kw">quantile</span>(randVal, <span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</span>
<span id="cb2476-18"><a href="dsi-allCells-allSetsTop.html#cb2476-18"></a></span>
<span id="cb2476-19"><a href="dsi-allCells-allSetsTop.html#cb2476-19"></a>Breaks &lt;-<span class="st"> </span><span class="kw">seq</span>(limits[<span class="dv">1</span>], limits[<span class="dv">2</span>],</span>
<span id="cb2476-20"><a href="dsi-allCells-allSetsTop.html#cb2476-20"></a>              <span class="dt">length =</span> <span class="dv">100</span>)</span>
<span id="cb2476-21"><a href="dsi-allCells-allSetsTop.html#cb2476-21"></a></span>
<span id="cb2476-22"><a href="dsi-allCells-allSetsTop.html#cb2476-22"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2476-23"><a href="dsi-allCells-allSetsTop.html#cb2476-23"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2476-24"><a href="dsi-allCells-allSetsTop.html#cb2476-24"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tabList[[splIdx]],</span>
<span id="cb2476-25"><a href="dsi-allCells-allSetsTop.html#cb2476-25"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2476-26"><a href="dsi-allCells-allSetsTop.html#cb2476-26"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2476-27"><a href="dsi-allCells-allSetsTop.html#cb2476-27"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2476-28"><a href="dsi-allCells-allSetsTop.html#cb2476-28"></a>                                 <span class="dt">breaks=</span>Breaks,</span>
<span id="cb2476-29"><a href="dsi-allCells-allSetsTop.html#cb2476-29"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s ratio&quot;</span>, splVec[splIdx]),</span>
<span id="cb2476-30"><a href="dsi-allCells-allSetsTop.html#cb2476-30"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2476-31"><a href="dsi-allCells-allSetsTop.html#cb2476-31"></a>}</span>
<span id="cb2476-32"><a href="dsi-allCells-allSetsTop.html#cb2476-32"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2476-33"><a href="dsi-allCells-allSetsTop.html#cb2476-33"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2476-34"><a href="dsi-allCells-allSetsTop.html#cb2476-34"></a>      <span class="dt">ncol=</span><span class="dv">3</span></span>
<span id="cb2476-35"><a href="dsi-allCells-allSetsTop.html#cb2476-35"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/randIndex_plot_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
</div>
<div id="encouraging-consistency-with-marker-genes-9" class="section level3">
<h3><span class="header-section-number">29.7.2</span> Encouraging consistency with marker genes</h3>
<p>In some situations, we will already have performed within-batch analyses to characterize salient aspects of population heterogeneity. This is not uncommon when merging datasets from different sources where each dataset has already been analyzed, annotated and interpreted separately. It is subsequently desirable for the integration procedure to retain these “known interesting” aspects of each dataset in the merged dataset. We can encourage this outcome by using the marker genes within each dataset as our selected feature set for <code>fastMNN()</code> and related methods. This focuses on the relevant heterogeneity and represents a semi-supervised approach that is a natural extension of the strategy described in the feature selection section.</p>
<p>We identify the top marker genes from pairwise Wilcoxon ranked sum tests between every pair of clusters within each batch, analogous to the method used by <a href="https://www.bioconductor.org/packages/release/bioc/html/SingleR.html">SingleR</a>. In this case, we use the top 10 marker genes but any value can be used depending on the acceptable trade-off between signal and noise (and speed). We then take the union across all comparisons in all batches and use that in place of our HVG set in <code>fastMNN()</code>.</p>
<div class="sourceCode" id="cb2477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2477-1"><a href="dsi-allCells-allSetsTop.html#cb2477-1"></a><span class="co"># OLD</span></span>
<span id="cb2477-2"><a href="dsi-allCells-allSetsTop.html#cb2477-2"></a></span>
<span id="cb2477-3"><a href="dsi-allCells-allSetsTop.html#cb2477-3"></a><span class="co"># Recall that groups for marker detection</span></span>
<span id="cb2477-4"><a href="dsi-allCells-allSetsTop.html#cb2477-4"></a><span class="co"># are automatically defined from &#39;colLabels()&#39;. </span></span>
<span id="cb2477-5"><a href="dsi-allCells-allSetsTop.html#cb2477-5"></a>stats1 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">1</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb2477-6"><a href="dsi-allCells-allSetsTop.html#cb2477-6"></a>markers1 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats1[[<span class="dv">1</span>]], stats1[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb2477-7"><a href="dsi-allCells-allSetsTop.html#cb2477-7"></a></span>
<span id="cb2477-8"><a href="dsi-allCells-allSetsTop.html#cb2477-8"></a>stats2 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">2</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb2477-9"><a href="dsi-allCells-allSetsTop.html#cb2477-9"></a>markers2 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats2[[<span class="dv">1</span>]], stats2[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb2477-10"><a href="dsi-allCells-allSetsTop.html#cb2477-10"></a></span>
<span id="cb2477-11"><a href="dsi-allCells-allSetsTop.html#cb2477-11"></a>stats3 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">3</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb2477-12"><a href="dsi-allCells-allSetsTop.html#cb2477-12"></a>markers3 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats3[[<span class="dv">1</span>]], stats3[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb2477-13"><a href="dsi-allCells-allSetsTop.html#cb2477-13"></a></span>
<span id="cb2477-14"><a href="dsi-allCells-allSetsTop.html#cb2477-14"></a>stats4 &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(rescaled.mbn[[<span class="dv">4</span>]], <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>)</span>
<span id="cb2477-15"><a href="dsi-allCells-allSetsTop.html#cb2477-15"></a>markers4 &lt;-<span class="st"> </span><span class="kw">getTopMarkers</span>(stats4[[<span class="dv">1</span>]], stats4[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>)</span>
<span id="cb2477-16"><a href="dsi-allCells-allSetsTop.html#cb2477-16"></a></span>
<span id="cb2477-17"><a href="dsi-allCells-allSetsTop.html#cb2477-17"></a>marker.set &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">unlist</span>(<span class="kw">c</span>(<span class="kw">unlist</span>(markers1), <span class="kw">unlist</span>(markers2), <span class="kw">unlist</span>(markers3), <span class="kw">unlist</span>(markers4))))</span>
<span id="cb2477-18"><a href="dsi-allCells-allSetsTop.html#cb2477-18"></a><span class="kw">length</span>(marker.set) <span class="co"># getting the total number of genes selected in this manner.</span></span></code></pre></div>
<div class="sourceCode" id="cb2478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2478-1"><a href="dsi-allCells-allSetsTop.html#cb2478-1"></a><span class="co"># Recall that groups for marker detection</span></span>
<span id="cb2478-2"><a href="dsi-allCells-allSetsTop.html#cb2478-2"></a><span class="co"># are automatically defined from &#39;colLabels()&#39;. </span></span>
<span id="cb2478-3"><a href="dsi-allCells-allSetsTop.html#cb2478-3"></a>markerList &lt;-<span class="st"> </span><span class="kw">lapply</span>(rescaled.mbn, <span class="cf">function</span>(x){</span>
<span id="cb2478-4"><a href="dsi-allCells-allSetsTop.html#cb2478-4"></a>  y &lt;-<span class="st"> </span><span class="kw">pairwiseWilcox</span>(x, <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>, <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2478-5"><a href="dsi-allCells-allSetsTop.html#cb2478-5"></a>  <span class="kw">getTopMarkers</span>(y[[<span class="dv">1</span>]], y[[<span class="dv">2</span>]], <span class="dt">n=</span><span class="dv">10</span>) <span class="op">%&gt;%</span><span class="st"> </span>unlist <span class="op">%&gt;%</span><span class="st"> </span>unlist</span>
<span id="cb2478-6"><a href="dsi-allCells-allSetsTop.html#cb2478-6"></a>  })</span>
<span id="cb2478-7"><a href="dsi-allCells-allSetsTop.html#cb2478-7"></a>marker.set &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">unlist</span>(markerList))</span>
<span id="cb2478-8"><a href="dsi-allCells-allSetsTop.html#cb2478-8"></a><span class="co">#length(marker.set) # getting the total number of genes selected in this manner.</span></span></code></pre></div>
<!-- MIND marker.set is from thr RUN etc and used for all 11 samples -->
<div class="sourceCode" id="cb2479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2479-1"><a href="dsi-allCells-allSetsTop.html#cb2479-1"></a><span class="kw">set.seed</span>(<span class="dv">1000110</span>)</span>
<span id="cb2479-2"><a href="dsi-allCells-allSetsTop.html#cb2479-2"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(</span>
<span id="cb2479-3"><a href="dsi-allCells-allSetsTop.html#cb2479-3"></a>  rescaled.mbn[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],</span>
<span id="cb2479-4"><a href="dsi-allCells-allSetsTop.html#cb2479-4"></a>  <span class="dt">subset.row=</span>marker.set,</span>
<span id="cb2479-5"><a href="dsi-allCells-allSetsTop.html#cb2479-5"></a>  <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>),</span>
<span id="cb2479-6"><a href="dsi-allCells-allSetsTop.html#cb2479-6"></a>  <span class="dt">BPPARAM=</span>bpp)</span></code></pre></div>
<p>The total number of genes selected in this manner is: 1188.</p>
<div class="sourceCode" id="cb2480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2480-1"><a href="dsi-allCells-allSetsTop.html#cb2480-1"></a><span class="kw">set.seed</span>(<span class="dv">1000110</span>)</span>
<span id="cb2480-2"><a href="dsi-allCells-allSetsTop.html#cb2480-2"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(rescaled.mbn,</span>
<span id="cb2480-3"><a href="dsi-allCells-allSetsTop.html#cb2480-3"></a>                    <span class="dt">subset.row=</span>marker.set,</span>
<span id="cb2480-4"><a href="dsi-allCells-allSetsTop.html#cb2480-4"></a>                    <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>),</span>
<span id="cb2480-5"><a href="dsi-allCells-allSetsTop.html#cb2480-5"></a>                    <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2480-6"><a href="dsi-allCells-allSetsTop.html#cb2480-6"></a>mnn.out2<span class="op">$</span>source_name &lt;-<span class="st"> </span>uncorrected<span class="op">$</span>source_name <span class="co"># cell order is maintained by scran functions</span></span>
<span id="cb2480-7"><a href="dsi-allCells-allSetsTop.html#cb2480-7"></a></span>
<span id="cb2480-8"><a href="dsi-allCells-allSetsTop.html#cb2480-8"></a><span class="co"># compute t-SNE:</span></span>
<span id="cb2480-9"><a href="dsi-allCells-allSetsTop.html#cb2480-9"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out2,</span>
<span id="cb2480-10"><a href="dsi-allCells-allSetsTop.html#cb2480-10"></a>                    <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb2480-11"><a href="dsi-allCells-allSetsTop.html#cb2480-11"></a>                    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>,</span>
<span id="cb2480-12"><a href="dsi-allCells-allSetsTop.html#cb2480-12"></a>                    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2480-13"><a href="dsi-allCells-allSetsTop.html#cb2480-13"></a>                    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2480-14"><a href="dsi-allCells-allSetsTop.html#cb2480-14"></a>                    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span></code></pre></div>
<p>We can also visualize the corrected coordinates using a t-SNE plot:</p>
<div class="sourceCode" id="cb2481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2481-1"><a href="dsi-allCells-allSetsTop.html#cb2481-1"></a><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_plotTsne_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2482-1"><a href="dsi-allCells-allSetsTop.html#cb2482-1"></a><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by=</span><span class="st">&quot;source_name&quot;</span>) <span class="op">+</span></span>
<span id="cb2482-2"><a href="dsi-allCells-allSetsTop.html#cb2482-2"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out2)<span class="op">$</span>source_name, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_plotTsneBySampleType_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2483-1"><a href="dsi-allCells-allSetsTop.html#cb2483-1"></a><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">shape_by=</span><span class="st">&quot;source_name&quot;</span>) <span class="op">+</span></span>
<span id="cb2483-2"><a href="dsi-allCells-allSetsTop.html#cb2483-2"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out2)<span class="op">$</span>batch, <span class="dt">ncol=</span><span class="dv">4</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_plotTsneByBatch_dsi_allCells_allSets-1.png" width="672" /></p>
<p>A quick inspection indicates that the original within-batch structure is indeed preserved in the corrected data. This highlights the utility of a marker-based feature set for integrating datasets that have already been characterized separately in a manner that preserves existing interpretations of each dataset. We note that some within-batch clusters have merged, most likely due to the lack of robust separation in the first place, though this may also be treated as a diagnostic on the appropriateness of the integration depending on the context.</p>
<div class="sourceCode" id="cb2484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2484-1"><a href="dsi-allCells-allSetsTop.html#cb2484-1"></a>mnn.out2 &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out2, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb2484-2"><a href="dsi-allCells-allSetsTop.html#cb2484-2"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2484-3"><a href="dsi-allCells-allSetsTop.html#cb2484-3"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2484-4"><a href="dsi-allCells-allSetsTop.html#cb2484-4"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2484-5"><a href="dsi-allCells-allSetsTop.html#cb2484-5"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2484-6"><a href="dsi-allCells-allSetsTop.html#cb2484-6"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb2484-7"><a href="dsi-allCells-allSetsTop.html#cb2484-7"></a>batchVec &lt;-<span class="st"> </span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)</span></code></pre></div>
<div class="sourceCode" id="cb2485"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2485-1"><a href="dsi-allCells-allSetsTop.html#cb2485-1"></a><span class="co"># for sample type in batchVec[1]</span></span>
<span id="cb2485-2"><a href="dsi-allCells-allSetsTop.html#cb2485-2"></a><span class="kw">grid.arrange</span>(</span>
<span id="cb2485-3"><a href="dsi-allCells-allSetsTop.html#cb2485-3"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">1</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">1</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">1</span>]),</span>
<span id="cb2485-4"><a href="dsi-allCells-allSetsTop.html#cb2485-4"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">2</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">2</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">2</span>]),</span>
<span id="cb2485-5"><a href="dsi-allCells-allSetsTop.html#cb2485-5"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">3</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">3</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">3</span>]),</span>
<span id="cb2485-6"><a href="dsi-allCells-allSetsTop.html#cb2485-6"></a>    <span class="kw">plotTSNE</span>(mnn.out2[,mnn.out2<span class="op">$</span>batch<span class="op">==</span>batchVec[<span class="dv">4</span>]], <span class="dt">colour_by=</span><span class="kw">I</span>(<span class="kw">colLabels</span>(rescaled.mbn[[<span class="dv">4</span>]]))) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(batchVec[<span class="dv">4</span>]),</span>
<span id="cb2485-7"><a href="dsi-allCells-allSetsTop.html#cb2485-7"></a>    <span class="dt">ncol=</span><span class="dv">2</span></span>
<span id="cb2485-8"><a href="dsi-allCells-allSetsTop.html#cb2485-8"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_plotTsneCluBySample1_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
<div id="using-the-corrected-values-1" class="section level3">
<h3><span class="header-section-number">29.7.3</span> Using the corrected values</h3>
<p>it is preferable to perform DE analyses using the uncorrected expression values with blocking on the batch. This strategy is based on the expectation that any genuine DE between clusters should still be present in a within-batch comparison where batch effects are absent. It penalizes genes that exhibit inconsistent DE across batches, thus protecting against misleading conclusions when a population in one batch is aligned to a similar-but-not-identical population in another batch. We demonstrate this approach below using a blocked t-test to detect markers in the PBMC dataset, where the presence of the same pattern across clusters within each batch is reassuring. If integration is performed across multiple conditions, it is even more important to use the uncorrected expression values for downstream analyses - see Section 14.5.2 for a discussion.</p>
<div class="sourceCode" id="cb2486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2486-1"><a href="dsi-allCells-allSetsTop.html#cb2486-1"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(</span>
<span id="cb2486-2"><a href="dsi-allCells-allSetsTop.html#cb2486-2"></a>  uncorrected,</span>
<span id="cb2486-3"><a href="dsi-allCells-allSetsTop.html#cb2486-3"></a>  clusters.mnn,</span>
<span id="cb2486-4"><a href="dsi-allCells-allSetsTop.html#cb2486-4"></a>  <span class="dt">block=</span>uncorrected<span class="op">$</span>batch, <span class="co"># </span><span class="al">TODO</span><span class="co"> batch or type?</span></span>
<span id="cb2486-5"><a href="dsi-allCells-allSetsTop.html#cb2486-5"></a>  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb2486-6"><a href="dsi-allCells-allSetsTop.html#cb2486-6"></a>  <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb2486-7"><a href="dsi-allCells-allSetsTop.html#cb2486-7"></a>  <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,</span>
<span id="cb2486-8"><a href="dsi-allCells-allSetsTop.html#cb2486-8"></a>                                <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),</span>
<span id="cb2486-9"><a href="dsi-allCells-allSetsTop.html#cb2486-9"></a>                                <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb2487"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2487-1"><a href="dsi-allCells-allSetsTop.html#cb2487-1"></a>demo &lt;-<span class="st"> </span>m.out[[<span class="st">&quot;1&quot;</span>]]</span>
<span id="cb2487-2"><a href="dsi-allCells-allSetsTop.html#cb2487-2"></a><span class="kw">as.data.frame</span>(demo[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,<span class="kw">c</span>(<span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;Top&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]) </span>
<span id="cb2487-3"><a href="dsi-allCells-allSetsTop.html#cb2487-3"></a><span class="co">#as.data.frame(demo[1:20,c(&quot;external_gene_name&quot;, &quot;Top&quot;, &quot;p.value&quot;, &quot;FDR&quot;)]) </span></span></code></pre></div>
<p>Expression level for the top gene, :</p>
<div class="sourceCode" id="cb2488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2488-1"><a href="dsi-allCells-allSetsTop.html#cb2488-1"></a>geneEnsId &lt;-<span class="st"> </span><span class="kw">rownames</span>(demo)[<span class="dv">1</span>]</span>
<span id="cb2488-2"><a href="dsi-allCells-allSetsTop.html#cb2488-2"></a><span class="kw">plotExpression</span>(uncorrected, <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)), </span>
<span id="cb2488-3"><a href="dsi-allCells-allSetsTop.html#cb2488-3"></a>    <span class="dt">features=</span>geneEnsId, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by)</span></code></pre></div>
<div class="sourceCode" id="cb2489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2489-1"><a href="dsi-allCells-allSetsTop.html#cb2489-1"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(uncorrected,</span>
<span id="cb2489-2"><a href="dsi-allCells-allSetsTop.html#cb2489-2"></a>                     clusters.mnn,</span>
<span id="cb2489-3"><a href="dsi-allCells-allSetsTop.html#cb2489-3"></a>                     <span class="dt">block=</span>uncorrected<span class="op">$</span>batch,</span>
<span id="cb2489-4"><a href="dsi-allCells-allSetsTop.html#cb2489-4"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb2489-5"><a href="dsi-allCells-allSetsTop.html#cb2489-5"></a>                     <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb2489-6"><a href="dsi-allCells-allSetsTop.html#cb2489-6"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),<span class="dt">drop=</span><span class="ot">FALSE</span>],</span>
<span id="cb2489-7"><a href="dsi-allCells-allSetsTop.html#cb2489-7"></a>                     <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2489-8"><a href="dsi-allCells-allSetsTop.html#cb2489-8"></a></span>
<span id="cb2489-9"><a href="dsi-allCells-allSetsTop.html#cb2489-9"></a><span class="co">#lapply(m.out, function(x){head(x[,2:6])})</span></span>
<span id="cb2489-10"><a href="dsi-allCells-allSetsTop.html#cb2489-10"></a></span>
<span id="cb2489-11"><a href="dsi-allCells-allSetsTop.html#cb2489-11"></a>tl1 &lt;-<span class="st"> </span><span class="kw">lapply</span>(m.out, <span class="cf">function</span>(x){x[x<span class="op">$</span>Symbol<span class="op">==</span><span class="st">&quot;CD3D&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.10</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>]}) <span class="co"># T-cell</span></span>
<span id="cb2489-12"><a href="dsi-allCells-allSetsTop.html#cb2489-12"></a>tl2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(m.out, <span class="cf">function</span>(x){x[x<span class="op">$</span>Symbol<span class="op">==</span><span class="st">&quot;CD69&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.20</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>]}) <span class="co"># activation</span></span>
<span id="cb2489-13"><a href="dsi-allCells-allSetsTop.html#cb2489-13"></a></span>
<span id="cb2489-14"><a href="dsi-allCells-allSetsTop.html#cb2489-14"></a>tb1 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(tl1, nrow)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb2489-15"><a href="dsi-allCells-allSetsTop.html#cb2489-15"></a>tb2 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(tl2, nrow)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb2489-16"><a href="dsi-allCells-allSetsTop.html#cb2489-16"></a></span>
<span id="cb2489-17"><a href="dsi-allCells-allSetsTop.html#cb2489-17"></a>cluToGet &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">c</span>(<span class="kw">which</span>(tb1), <span class="kw">which</span>(tb2)))[<span class="dv">1</span>] <span class="co"># 3 # 19 # 4</span></span>
<span id="cb2489-18"><a href="dsi-allCells-allSetsTop.html#cb2489-18"></a>demo &lt;-<span class="st"> </span>m.out[[cluToGet]]</span>
<span id="cb2489-19"><a href="dsi-allCells-allSetsTop.html#cb2489-19"></a><span class="co">#as.data.frame(demo[1:20,c(&quot;Symbol&quot;, &quot;Top&quot;, &quot;p.value&quot;, &quot;FDR&quot;, &quot;summary.logFC&quot;)]) </span></span>
<span id="cb2489-20"><a href="dsi-allCells-allSetsTop.html#cb2489-20"></a>geneEnsId &lt;-<span class="st"> </span>demo <span class="op">%&gt;%</span><span class="st"> </span>data.frame <span class="op">%&gt;%</span></span>
<span id="cb2489-21"><a href="dsi-allCells-allSetsTop.html#cb2489-21"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">str_detect</span>(Symbol, <span class="st">&quot;^RP&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb2489-22"><a href="dsi-allCells-allSetsTop.html#cb2489-22"></a><span class="st">  </span><span class="kw">pull</span>(ensembl_gene_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>(<span class="dv">1</span>)</span>
<span id="cb2489-23"><a href="dsi-allCells-allSetsTop.html#cb2489-23"></a>geneSymbol &lt;-<span class="st"> </span><span class="kw">rowData</span>(uncorrected)[geneEnsId,<span class="st">&quot;Symbol&quot;</span>]</span></code></pre></div>
<!-- OLD Expression level for the top gene,  on violin plots: -->
<div class="sourceCode" id="cb2490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2490-1"><a href="dsi-allCells-allSetsTop.html#cb2490-1"></a><span class="kw">plotExpression</span>(uncorrected,</span>
<span id="cb2490-2"><a href="dsi-allCells-allSetsTop.html#cb2490-2"></a>               <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)),</span>
<span id="cb2490-3"><a href="dsi-allCells-allSetsTop.html#cb2490-3"></a>               <span class="dt">features=</span>geneEnsId, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span></span>
<span id="cb2490-4"><a href="dsi-allCells-allSetsTop.html#cb2490-4"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by) <span class="op">+</span></span>
<span id="cb2490-5"><a href="dsi-allCells-allSetsTop.html#cb2490-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2490-6"><a href="dsi-allCells-allSetsTop.html#cb2490-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s %s&quot;</span>,</span>
<span id="cb2490-7"><a href="dsi-allCells-allSetsTop.html#cb2490-7"></a>          geneEnsId,</span>
<span id="cb2490-8"><a href="dsi-allCells-allSetsTop.html#cb2490-8"></a>          <span class="kw">rowData</span>(uncorrected)[geneEnsId,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2490-9"><a href="dsi-allCells-allSetsTop.html#cb2490-9"></a>          )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_uncorr_topGene_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Expression level for the top gene, FTL on t-SNE plot:</p>
<p><strong>Not Encouraging consistency</strong> with marker genes</p>
<div class="sourceCode" id="cb2491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2491-1"><a href="dsi-allCells-allSetsTop.html#cb2491-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by =</span> geneEnsId, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2491-2"><a href="dsi-allCells-allSetsTop.html#cb2491-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>( <span class="kw">paste</span>(<span class="st">&quot;cluster&quot;</span>, cluToGet, geneEnsId, geneSymbol))</span>
<span id="cb2491-3"><a href="dsi-allCells-allSetsTop.html#cb2491-3"></a><span class="co">#print(p)</span></span>
<span id="cb2491-4"><a href="dsi-allCells-allSetsTop.html#cb2491-4"></a>p1 &lt;-<span class="st"> </span>p</span>
<span id="cb2491-5"><a href="dsi-allCells-allSetsTop.html#cb2491-5"></a>p2 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out)<span class="op">$</span>batch)</span>
<span id="cb2491-6"><a href="dsi-allCells-allSetsTop.html#cb2491-6"></a><span class="kw">grid.arrange</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2491-7"><a href="dsi-allCells-allSetsTop.html#cb2491-7"></a>                        p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2491-8"><a href="dsi-allCells-allSetsTop.html#cb2491-8"></a>                        <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_uncorr_tsneWoEncour_dsi_allCells_allSets-1.png" width="672" /></p>
<p><strong>Encouraging consistency</strong> with marker genes</p>
<div class="sourceCode" id="cb2492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2492-1"><a href="dsi-allCells-allSetsTop.html#cb2492-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out2, <span class="dt">colour_by =</span> geneEnsId, <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2492-2"><a href="dsi-allCells-allSetsTop.html#cb2492-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>( <span class="kw">paste</span>(<span class="st">&quot;cluster&quot;</span>, cluToGet, geneEnsId, geneSymbol))</span>
<span id="cb2492-3"><a href="dsi-allCells-allSetsTop.html#cb2492-3"></a><span class="co">#print(p)</span></span>
<span id="cb2492-4"><a href="dsi-allCells-allSetsTop.html#cb2492-4"></a>p1 &lt;-<span class="st"> </span>p</span>
<span id="cb2492-5"><a href="dsi-allCells-allSetsTop.html#cb2492-5"></a>p2 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out2)<span class="op">$</span>batch)</span>
<span id="cb2492-6"><a href="dsi-allCells-allSetsTop.html#cb2492-6"></a><span class="kw">grid.arrange</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2492-7"><a href="dsi-allCells-allSetsTop.html#cb2492-7"></a>                        p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2492-8"><a href="dsi-allCells-allSetsTop.html#cb2492-8"></a>                        <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/consis_uncorr_tsneWiEncour_dsi_allCells_allSets-1.png" width="672" /></p>
<p>We suggest limiting the use of per-gene corrected values to visualization, e.g., when coloring points on a t-SNE plot by per-cell expression. This can be more aesthetically pleasing than uncorrected expression values that may contain large shifts on the colour scale between cells in different batches. Use of the corrected values in any quantitative procedure should be treated with caution, and should be backed up by similar results from an analysis on the uncorrected values.</p>
<div class="sourceCode" id="cb2493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2493-1"><a href="dsi-allCells-allSetsTop.html#cb2493-1"></a><span class="co"># before we save the mnn.out object in a file,</span></span>
<span id="cb2493-2"><a href="dsi-allCells-allSetsTop.html#cb2493-2"></a><span class="co"># we should copy some of the cell meta data over,</span></span>
<span id="cb2493-3"><a href="dsi-allCells-allSetsTop.html#cb2493-3"></a><span class="co"># eg Barcode and lib size.</span></span>
<span id="cb2493-4"><a href="dsi-allCells-allSetsTop.html#cb2493-4"></a></span>
<span id="cb2493-5"><a href="dsi-allCells-allSetsTop.html#cb2493-5"></a><span class="co"># Mind sets may have been downsampled, eg with nbCells set to 1000.</span></span>
<span id="cb2493-6"><a href="dsi-allCells-allSetsTop.html#cb2493-6"></a><span class="co"># But that is not in the file name (yet?)</span></span>
<span id="cb2493-7"><a href="dsi-allCells-allSetsTop.html#cb2493-7"></a></span>
<span id="cb2493-8"><a href="dsi-allCells-allSetsTop.html#cb2493-8"></a><span class="co"># save object?</span></span>
<span id="cb2493-9"><a href="dsi-allCells-allSetsTop.html#cb2493-9"></a>fn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi_%s.Rds&quot;</span>,</span>
<span id="cb2493-10"><a href="dsi-allCells-allSetsTop.html#cb2493-10"></a><span class="co">#fn &lt;- sprintf(&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi2_%s.Rds&quot;,</span></span>
<span id="cb2493-11"><a href="dsi-allCells-allSetsTop.html#cb2493-11"></a>              projDir,</span>
<span id="cb2493-12"><a href="dsi-allCells-allSetsTop.html#cb2493-12"></a>              outDirBit,</span>
<span id="cb2493-13"><a href="dsi-allCells-allSetsTop.html#cb2493-13"></a>              setName,</span>
<span id="cb2493-14"><a href="dsi-allCells-allSetsTop.html#cb2493-14"></a>              setSuf,</span>
<span id="cb2493-15"><a href="dsi-allCells-allSetsTop.html#cb2493-15"></a>              splSetToGet2) <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb2493-16"><a href="dsi-allCells-allSetsTop.html#cb2493-16"></a><span class="kw">saveRDS</span>(mnn.out, <span class="dt">file=</span>fn)</span>
<span id="cb2493-17"><a href="dsi-allCells-allSetsTop.html#cb2493-17"></a><span class="co">#saveRDS(mnn.out2, file=fn)</span></span></code></pre></div>
</div>
</div>
<div id="challenge-same-but-with-an-ordered-merging-1" class="section level2">
<h2><span class="header-section-number">29.8</span> <strong>Challenge</strong> Same but with an ordered merging</h2>
<p>We will first merge replicates in each sample group separately, then sample groups, starting with the group with the larger number of ‘cell types’.</p>
<p>Hint: use the merge.order option in fastMNN ( … maybe with “list( list(1,2,3,4), list(9,10,11), list(5,6), list(7,8) )” )</p>
<div class="sourceCode" id="cb2494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2494-1"><a href="dsi-allCells-allSetsTop.html#cb2494-1"></a><span class="co"># Using randomized SVD here, as this is faster than </span></span>
<span id="cb2494-2"><a href="dsi-allCells-allSetsTop.html#cb2494-2"></a><span class="co"># irlba for file-backed matrices.</span></span>
<span id="cb2494-3"><a href="dsi-allCells-allSetsTop.html#cb2494-3"></a><span class="kw">set.seed</span>(<span class="dv">1000101001</span>)</span>
<span id="cb2494-4"><a href="dsi-allCells-allSetsTop.html#cb2494-4"></a><span class="kw">rm</span>(mnn.out)</span>
<span id="cb2494-5"><a href="dsi-allCells-allSetsTop.html#cb2494-5"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">fastMNN</span>(rescaled.mbn,</span>
<span id="cb2494-6"><a href="dsi-allCells-allSetsTop.html#cb2494-6"></a>           <span class="dt">merge.order=</span><span class="kw">list</span>( <span class="kw">list</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="kw">list</span>(<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">11</span>), <span class="kw">list</span>(<span class="dv">5</span>,<span class="dv">6</span>), <span class="kw">list</span>(<span class="dv">7</span>,<span class="dv">8</span>) ),</span>
<span id="cb2494-7"><a href="dsi-allCells-allSetsTop.html#cb2494-7"></a>           <span class="dt">d=</span><span class="dv">50</span>, <span class="dt">k=</span><span class="dv">20</span>, <span class="dt">subset.row=</span>chosen.hvgs,</span>
<span id="cb2494-8"><a href="dsi-allCells-allSetsTop.html#cb2494-8"></a>    <span class="dt">BSPARAM=</span><span class="kw">RandomParam</span>(<span class="dt">deferred=</span><span class="ot">TRUE</span>),</span>
<span id="cb2494-9"><a href="dsi-allCells-allSetsTop.html#cb2494-9"></a>    <span class="dt">BPPARAM=</span>bpp</span>
<span id="cb2494-10"><a href="dsi-allCells-allSetsTop.html#cb2494-10"></a>    )</span>
<span id="cb2494-11"><a href="dsi-allCells-allSetsTop.html#cb2494-11"></a>mnn.out</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 12466 47830 
## metadata(2): merge.info pca.info
## assays(1): reconstructed
## rownames(12466): ENSG00000000003 ENSG00000000457 ... ENSG00000285476
##   ENSG00000285492
## rowData names(1): rotation
## colnames: NULL
## colData names(1): batch
## reducedDimNames(1): corrected
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb2496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2496-1"><a href="dsi-allCells-allSetsTop.html#cb2496-1"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb2496-2"><a href="dsi-allCells-allSetsTop.html#cb2496-2"></a>mnn.out<span class="op">$</span>splType &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch)</span>
<span id="cb2496-3"><a href="dsi-allCells-allSetsTop.html#cb2496-3"></a>mnn.out<span class="op">$</span>splType &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>splType)</span></code></pre></div>
<div class="sourceCode" id="cb2497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2497-1"><a href="dsi-allCells-allSetsTop.html#cb2497-1"></a><span class="co">#class(mnn.out$batch)</span></span>
<span id="cb2497-2"><a href="dsi-allCells-allSetsTop.html#cb2497-2"></a><span class="co">#head(mnn.out$batch)</span></span>
<span id="cb2497-3"><a href="dsi-allCells-allSetsTop.html#cb2497-3"></a><span class="co">#dim(reducedDim(mnn.out, &quot;corrected&quot;))</span></span>
<span id="cb2497-4"><a href="dsi-allCells-allSetsTop.html#cb2497-4"></a><span class="co">#assay(mnn.out, &quot;reconstructed&quot;)</span></span>
<span id="cb2497-5"><a href="dsi-allCells-allSetsTop.html#cb2497-5"></a><span class="kw">print</span>(<span class="kw">dim</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)))</span></code></pre></div>
<pre><code>## [1] 12466 47830</code></pre>
<div class="sourceCode" id="cb2499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2499-1"><a href="dsi-allCells-allSetsTop.html#cb2499-1"></a><span class="kw">print</span>(<span class="kw">assay</span>(mnn.out, <span class="st">&quot;reconstructed&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</span></code></pre></div>
<pre><code>## &lt;5 x 3&gt; matrix of class LowRankMatrix and type &quot;double&quot;:
##                          [,1]          [,2]          [,3]
## ENSG00000000003 -1.272426e-04 -6.787507e-05 -5.702688e-05
## ENSG00000000457 -3.356405e-05  4.087098e-04  1.054453e-04
## ENSG00000000938 -2.351897e-03 -4.617906e-04 -3.513377e-04
## ENSG00000001167 -6.807745e-05  2.264625e-04  2.274302e-04
## ENSG00000001461 -2.447150e-04 -4.125638e-05 -2.362878e-04</code></pre>
<p>Diagnostic table and plots:</p>
<div class="sourceCode" id="cb2501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2501-1"><a href="dsi-allCells-allSetsTop.html#cb2501-1"></a>snn.gr &lt;-<span class="st"> </span><span class="kw">buildSNNGraph</span>(mnn.out, <span class="dt">use.dimred=</span><span class="st">&quot;corrected&quot;</span>, <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2501-2"><a href="dsi-allCells-allSetsTop.html#cb2501-2"></a><span class="co">#clusters.mnn &lt;- igraph::cluster_walktrap(snn.gr)$membership</span></span>
<span id="cb2501-3"><a href="dsi-allCells-allSetsTop.html#cb2501-3"></a>clusters.mnn &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">cluster_louvain</span>(snn.gr)<span class="op">$</span>membership</span>
<span id="cb2501-4"><a href="dsi-allCells-allSetsTop.html#cb2501-4"></a>mnn.out<span class="op">$</span>clusters.mnn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;c%s&quot;</span>, clusters.mnn)</span>
<span id="cb2501-5"><a href="dsi-allCells-allSetsTop.html#cb2501-5"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>mnn.out<span class="op">$</span>clusters.mnn, <span class="dt">Batch=</span>mnn.out<span class="op">$</span>batch)</span>
<span id="cb2501-6"><a href="dsi-allCells-allSetsTop.html#cb2501-6"></a>tab.mnn</span></code></pre></div>
<pre><code>##        Batch
## Cluster ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PBMMC_1
##     c1           756          696          298          279  1142  1804     166
##     c10            8            1            5            8    24    12      93
##     c11            7           11           39           10     9    14     128
##     c12            0           24            3            2   679   128     221
##     c13            0            1           16           11     1     1     403
##     c14            3            3          138          261     1    15      67
##     c15            6          101          458          140   228    37      62
##     c16            1            7           23           42    51    10     194
##     c17            0           21           10            5     2     7      36
##     c2           384          645          109          582   337   560     319
##     c3           101          374           92           85   353   108      22
##     c4           996          803          175         1433   303   392      27
##     c5            13          138         1350          218   192    89     210
##     c6             6           14          187          781     6    19       7
##     c7            11           80          327          107   175    70      79
##     c8           395          519          195          806   406   595      14
##     c9           166         3177         1302          523   642  1576      36
##        Batch
## Cluster PBMMC_2 PBMMC_3 PRE-T_1 PRE-T_2
##     c1      168     207       4      51
##     c10      44      38    1972    1104
##     c11      21      72     181    1057
##     c12     160     439       1       6
##     c13     186      26       2       6
##     c14     617     144       2      68
##     c15     420     459      23     114
##     c16     337     641       6     224
##     c17      67      60       2      28
##     c2      113     319     736     124
##     c3       31      12     849     189
##     c4       27      61       2       0
##     c5      871     884      22     324
##     c6     1066      59       4      44
##     c7      461     361      28     282
##     c8       24      54       5       0
##     c9       45     307       2       7</code></pre>
<div class="sourceCode" id="cb2503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2503-1"><a href="dsi-allCells-allSetsTop.html#cb2503-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tab.mnn,</span>
<span id="cb2503-2"><a href="dsi-allCells-allSetsTop.html#cb2503-2"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2503-3"><a href="dsi-allCells-allSetsTop.html#cb2503-3"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2503-4"><a href="dsi-allCells-allSetsTop.html#cb2503-4"></a>           <span class="co">#cluster_rows      = FALSE,</span></span>
<span id="cb2503-5"><a href="dsi-allCells-allSetsTop.html#cb2503-5"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2503-6"><a href="dsi-allCells-allSetsTop.html#cb2503-6"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb2504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2504-1"><a href="dsi-allCells-allSetsTop.html#cb2504-1"></a><span class="kw">set.seed</span>(<span class="dv">0010101010</span>)</span>
<span id="cb2504-2"><a href="dsi-allCells-allSetsTop.html#cb2504-2"></a>mnn.out &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(mnn.out, <span class="dt">dimred=</span><span class="st">&quot;corrected&quot;</span>,</span>
<span id="cb2504-3"><a href="dsi-allCells-allSetsTop.html#cb2504-3"></a>    <span class="dt">external_neighbors=</span><span class="ot">TRUE</span>, </span>
<span id="cb2504-4"><a href="dsi-allCells-allSetsTop.html#cb2504-4"></a>    <span class="dt">BNPARAM=</span><span class="kw">AnnoyParam</span>(),</span>
<span id="cb2504-5"><a href="dsi-allCells-allSetsTop.html#cb2504-5"></a>    <span class="dt">BPPARAM=</span>bpp,</span>
<span id="cb2504-6"><a href="dsi-allCells-allSetsTop.html#cb2504-6"></a>    <span class="dt">n_threads=</span><span class="kw">bpnworkers</span>(bpp))</span>
<span id="cb2504-7"><a href="dsi-allCells-allSetsTop.html#cb2504-7"></a></span>
<span id="cb2504-8"><a href="dsi-allCells-allSetsTop.html#cb2504-8"></a>p.batch &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2504-9"><a href="dsi-allCells-allSetsTop.html#cb2504-9"></a>p.clu &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out, <span class="dt">colour_by=</span><span class="st">&quot;clusters.mnn&quot;</span>, <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2504-10"><a href="dsi-allCells-allSetsTop.html#cb2504-10"></a><span class="co">#grid.arrange(p.clu, p.batch, ncol=2)</span></span>
<span id="cb2504-11"><a href="dsi-allCells-allSetsTop.html#cb2504-11"></a>mnn.out<span class="op">$</span>splType &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_[1-4]&quot;</span>,<span class="st">&quot;&quot;</span>,mnn.out<span class="op">$</span>batch) <span class="co"># TODO fix; shouldn&#39;t need to; done above already</span></span>
<span id="cb2504-12"><a href="dsi-allCells-allSetsTop.html#cb2504-12"></a><span class="kw">grid.arrange</span>(p.clu, p.batch<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>splType), <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/unnamed-chunk-13-1.png" width="806.4" /></p>
<p>Write mnn.out object to file:</p>
<div class="sourceCode" id="cb2505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2505-1"><a href="dsi-allCells-allSetsTop.html#cb2505-1"></a><span class="kw">colData</span>(mnn.out) &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">colData</span>(uncorrected),</span>
<span id="cb2505-2"><a href="dsi-allCells-allSetsTop.html#cb2505-2"></a>              <span class="kw">colData</span>(mnn.out)[,<span class="kw">c</span>(<span class="st">&quot;splType&quot;</span>, <span class="st">&quot;clusters.mnn&quot;</span>)])</span>
<span id="cb2505-3"><a href="dsi-allCells-allSetsTop.html#cb2505-3"></a><span class="co"># Write object to file</span></span>
<span id="cb2505-4"><a href="dsi-allCells-allSetsTop.html#cb2505-4"></a><span class="co"># fastMnnWholeByList -&gt; Fmwbl</span></span>
<span id="cb2505-5"><a href="dsi-allCells-allSetsTop.html#cb2505-5"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi_%s_Fmwbl.Rds&quot;</span>,</span>
<span id="cb2505-6"><a href="dsi-allCells-allSetsTop.html#cb2505-6"></a>         projDir, outDirBit, setName, setSuf, splSetToGet2)</span>
<span id="cb2505-7"><a href="dsi-allCells-allSetsTop.html#cb2505-7"></a><span class="kw">saveRDS</span>(mnn.out, tmpFn)</span>
<span id="cb2505-8"><a href="dsi-allCells-allSetsTop.html#cb2505-8"></a></span>
<span id="cb2505-9"><a href="dsi-allCells-allSetsTop.html#cb2505-9"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi_%s_Fmwbl2.Rds&quot;</span>,</span>
<span id="cb2505-10"><a href="dsi-allCells-allSetsTop.html#cb2505-10"></a>         projDir, outDirBit, setName, setSuf, splSetToGet2)</span>
<span id="cb2505-11"><a href="dsi-allCells-allSetsTop.html#cb2505-11"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;chosen.hvgs&quot;</span>=chosen.hvgs,</span>
<span id="cb2505-12"><a href="dsi-allCells-allSetsTop.html#cb2505-12"></a>         <span class="st">&quot;uncorrected&quot;</span>=uncorrected,</span>
<span id="cb2505-13"><a href="dsi-allCells-allSetsTop.html#cb2505-13"></a>         <span class="st">&quot;rescaled.mbn&quot;</span>=rescaled.mbn),</span>
<span id="cb2505-14"><a href="dsi-allCells-allSetsTop.html#cb2505-14"></a>    tmpFn)</span></code></pre></div>
<p>Proportions of lost variance:</p>
<div class="sourceCode" id="cb2506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2506-1"><a href="dsi-allCells-allSetsTop.html#cb2506-1"></a><span class="kw">round</span>(<span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var,<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##       ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4 HHD_1 HHD_2 PRE-T_1
##  [1,]         0.00         0.00         0.00         0.00  0.00  0.00    0.02
##  [2,]         0.00         0.00         0.00         0.00  0.01  0.03    0.00
##  [3,]         0.00         0.00         0.00         0.00  0.00  0.00    0.00
##  [4,]         0.00         0.00         0.00         0.00  0.00  0.00    0.00
##  [5,]         0.02         0.02         0.00         0.00  0.00  0.00    0.00
##  [6,]         0.01         0.01         0.04         0.00  0.00  0.00    0.00
##  [7,]         0.00         0.00         0.01         0.06  0.00  0.00    0.00
##  [8,]         0.04         0.04         0.11         0.16  0.00  0.00    0.00
##  [9,]         0.01         0.02         0.03         0.04  0.07  0.08    0.00
## [10,]         0.02         0.02         0.03         0.03  0.02  0.02    0.07
##       PRE-T_2 PBMMC_1 PBMMC_2 PBMMC_3
##  [1,]    0.01    0.00    0.00    0.00
##  [2,]    0.00    0.00    0.00    0.00
##  [3,]    0.00    0.04    0.18    0.00
##  [4,]    0.00    0.00    0.00    0.07
##  [5,]    0.00    0.00    0.00    0.00
##  [6,]    0.00    0.00    0.00    0.00
##  [7,]    0.00    0.00    0.00    0.00
##  [8,]    0.00    0.02    0.02    0.02
##  [9,]    0.00    0.02    0.05    0.02
## [10,]    0.12    0.02    0.02    0.02</code></pre>
<div class="sourceCode" id="cb2508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2508-1"><a href="dsi-allCells-allSetsTop.html#cb2508-1"></a>tmpData &lt;-<span class="st"> </span><span class="kw">metadata</span>(mnn.out)<span class="op">$</span>merge.info<span class="op">$</span>lost.var</span>
<span id="cb2508-2"><a href="dsi-allCells-allSetsTop.html#cb2508-2"></a></span>
<span id="cb2508-3"><a href="dsi-allCells-allSetsTop.html#cb2508-3"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(tmpData,</span>
<span id="cb2508-4"><a href="dsi-allCells-allSetsTop.html#cb2508-4"></a>           <span class="dt">border_color      =</span> <span class="ot">NA</span>,</span>
<span id="cb2508-5"><a href="dsi-allCells-allSetsTop.html#cb2508-5"></a>           <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,</span>
<span id="cb2508-6"><a href="dsi-allCells-allSetsTop.html#cb2508-6"></a>           <span class="dt">cluster_rows      =</span> <span class="ot">FALSE</span>,</span>
<span id="cb2508-7"><a href="dsi-allCells-allSetsTop.html#cb2508-7"></a>           <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span></span>
<span id="cb2508-8"><a href="dsi-allCells-allSetsTop.html#cb2508-8"></a>           )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div id="preserving-biological-heterogeneity-11" class="section level3">
<h3><span class="header-section-number">29.8.1</span> Preserving biological heterogeneity</h3>
<div id="comparison-to-within-batch-clusters-4" class="section level4">
<h4><span class="header-section-number">29.8.1.1</span> Comparison to within-batch clusters</h4>
<div class="sourceCode" id="cb2509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2509-1"><a href="dsi-allCells-allSetsTop.html#cb2509-1"></a><span class="co"># OLD</span></span>
<span id="cb2509-2"><a href="dsi-allCells-allSetsTop.html#cb2509-2"></a></span>
<span id="cb2509-3"><a href="dsi-allCells-allSetsTop.html#cb2509-3"></a>mnn.out<span class="op">$</span>batch &lt;-<span class="st"> </span><span class="kw">factor</span>(mnn.out<span class="op">$</span>batch) <span class="co"># somehow need to re-factor batch</span></span>
<span id="cb2509-4"><a href="dsi-allCells-allSetsTop.html#cb2509-4"></a><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)</span>
<span id="cb2509-5"><a href="dsi-allCells-allSetsTop.html#cb2509-5"></a></span>
<span id="cb2509-6"><a href="dsi-allCells-allSetsTop.html#cb2509-6"></a><span class="co"># For the first batch (adding +10 for a smoother color transition</span></span>
<span id="cb2509-7"><a href="dsi-allCells-allSetsTop.html#cb2509-7"></a><span class="co"># from zero to non-zero counts for any given matrix entry).</span></span>
<span id="cb2509-8"><a href="dsi-allCells-allSetsTop.html#cb2509-8"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb2509-9"><a href="dsi-allCells-allSetsTop.html#cb2509-9"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2509-10"><a href="dsi-allCells-allSetsTop.html#cb2509-10"></a><span class="kw">length</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]))</span>
<span id="cb2509-11"><a href="dsi-allCells-allSetsTop.html#cb2509-11"></a>rescaled.mbn[[batchPlace]]</span>
<span id="cb2509-12"><a href="dsi-allCells-allSetsTop.html#cb2509-12"></a>rescaled.mbn[[batchPlace]] <span class="op">%&gt;%</span><span class="st"> </span>colData <span class="op">%&gt;%</span><span class="st"> </span>head</span>
<span id="cb2509-13"><a href="dsi-allCells-allSetsTop.html#cb2509-13"></a></span>
<span id="cb2509-14"><a href="dsi-allCells-allSetsTop.html#cb2509-14"></a><span class="kw">length</span>(<span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2509-15"><a href="dsi-allCells-allSetsTop.html#cb2509-15"></a><span class="co">#save.image(&quot;dataSetIntegrationWhole.debug.Rdata&quot;)</span></span>
<span id="cb2509-16"><a href="dsi-allCells-allSetsTop.html#cb2509-16"></a></span>
<span id="cb2509-17"><a href="dsi-allCells-allSetsTop.html#cb2509-17"></a><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]))</span>
<span id="cb2509-18"><a href="dsi-allCells-allSetsTop.html#cb2509-18"></a><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2509-19"><a href="dsi-allCells-allSetsTop.html#cb2509-19"></a></span>
<span id="cb2509-20"><a href="dsi-allCells-allSetsTop.html#cb2509-20"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb2509-21"><a href="dsi-allCells-allSetsTop.html#cb2509-21"></a>             <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2509-22"><a href="dsi-allCells-allSetsTop.html#cb2509-22"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb2509-23"><a href="dsi-allCells-allSetsTop.html#cb2509-23"></a>  <span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb2509-24"><a href="dsi-allCells-allSetsTop.html#cb2509-24"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2509-25"><a href="dsi-allCells-allSetsTop.html#cb2509-25"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2509-26"><a href="dsi-allCells-allSetsTop.html#cb2509-26"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]),</span>
<span id="cb2509-27"><a href="dsi-allCells-allSetsTop.html#cb2509-27"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2509-28"><a href="dsi-allCells-allSetsTop.html#cb2509-28"></a></span>
<span id="cb2509-29"><a href="dsi-allCells-allSetsTop.html#cb2509-29"></a><span class="co"># For the second batch.</span></span>
<span id="cb2509-30"><a href="dsi-allCells-allSetsTop.html#cb2509-30"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb2509-31"><a href="dsi-allCells-allSetsTop.html#cb2509-31"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2509-32"><a href="dsi-allCells-allSetsTop.html#cb2509-32"></a>tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[tmpInd]),</span>
<span id="cb2509-33"><a href="dsi-allCells-allSetsTop.html#cb2509-33"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2509-34"><a href="dsi-allCells-allSetsTop.html#cb2509-34"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb2509-35"><a href="dsi-allCells-allSetsTop.html#cb2509-35"></a>  <span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb2509-36"><a href="dsi-allCells-allSetsTop.html#cb2509-36"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2509-37"><a href="dsi-allCells-allSetsTop.html#cb2509-37"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2509-38"><a href="dsi-allCells-allSetsTop.html#cb2509-38"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s comparison&quot;</span>, splVec[batchPlace]),</span>
<span id="cb2509-39"><a href="dsi-allCells-allSetsTop.html#cb2509-39"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2509-40"><a href="dsi-allCells-allSetsTop.html#cb2509-40"></a></span>
<span id="cb2509-41"><a href="dsi-allCells-allSetsTop.html#cb2509-41"></a><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<div class="sourceCode" id="cb2510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2510-1"><a href="dsi-allCells-allSetsTop.html#cb2510-1"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2510-2"><a href="dsi-allCells-allSetsTop.html#cb2510-2"></a>treeList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2510-3"><a href="dsi-allCells-allSetsTop.html#cb2510-3"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2510-4"><a href="dsi-allCells-allSetsTop.html#cb2510-4"></a>  <span class="co"># heatmap</span></span>
<span id="cb2510-5"><a href="dsi-allCells-allSetsTop.html#cb2510-5"></a>  tab &lt;-<span class="st"> </span><span class="kw">table</span>(</span>
<span id="cb2510-6"><a href="dsi-allCells-allSetsTop.html#cb2510-6"></a>    <span class="kw">paste</span>(<span class="st">&quot;before&quot;</span>, <span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]), <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),</span>
<span id="cb2510-7"><a href="dsi-allCells-allSetsTop.html#cb2510-7"></a>    <span class="kw">paste</span>(<span class="st">&quot;after&quot;</span>, clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]], <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb2510-8"><a href="dsi-allCells-allSetsTop.html#cb2510-8"></a>    )</span>
<span id="cb2510-9"><a href="dsi-allCells-allSetsTop.html#cb2510-9"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(<span class="kw">log10</span>(tab<span class="op">+</span><span class="dv">10</span>),</span>
<span id="cb2510-10"><a href="dsi-allCells-allSetsTop.html#cb2510-10"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2510-11"><a href="dsi-allCells-allSetsTop.html#cb2510-11"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2510-12"><a href="dsi-allCells-allSetsTop.html#cb2510-12"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2510-13"><a href="dsi-allCells-allSetsTop.html#cb2510-13"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s&quot;</span>,</span>
<span id="cb2510-14"><a href="dsi-allCells-allSetsTop.html#cb2510-14"></a>                                              splVec[splIdx]),</span>
<span id="cb2510-15"><a href="dsi-allCells-allSetsTop.html#cb2510-15"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>,</span>
<span id="cb2510-16"><a href="dsi-allCells-allSetsTop.html#cb2510-16"></a>                                 <span class="dt">fontsize=</span><span class="dv">7</span>)</span>
<span id="cb2510-17"><a href="dsi-allCells-allSetsTop.html#cb2510-17"></a>  <span class="co"># cluster tree:</span></span>
<span id="cb2510-18"><a href="dsi-allCells-allSetsTop.html#cb2510-18"></a>  combined &lt;-<span class="st"> </span><span class="kw">cbind</span>(</span>
<span id="cb2510-19"><a href="dsi-allCells-allSetsTop.html#cb2510-19"></a>    <span class="dt">cl.1=</span><span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]),</span>
<span id="cb2510-20"><a href="dsi-allCells-allSetsTop.html#cb2510-20"></a>    <span class="dt">cl.2=</span>clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb2510-21"><a href="dsi-allCells-allSetsTop.html#cb2510-21"></a>  treeList[[splIdx]]  &lt;-<span class="st"> </span><span class="kw">clustree</span>(combined, <span class="dt">prefix=</span><span class="st">&quot;cl.&quot;</span>, <span class="dt">edge_arrow=</span><span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2510-22"><a href="dsi-allCells-allSetsTop.html#cb2510-22"></a><span class="st">    </span><span class="kw">ggtitle</span>(splVec[splIdx]) <span class="op">+</span></span>
<span id="cb2510-23"><a href="dsi-allCells-allSetsTop.html#cb2510-23"></a><span class="st">    </span><span class="co">#theme(legend.background = element_rect(color = &quot;yellow&quot;)) +</span></span>
<span id="cb2510-24"><a href="dsi-allCells-allSetsTop.html#cb2510-24"></a><span class="st">    </span><span class="co">#theme(legend.position=&#39;bottom&#39;) +</span></span>
<span id="cb2510-25"><a href="dsi-allCells-allSetsTop.html#cb2510-25"></a><span class="st">    </span><span class="co">#theme(legend.box=&quot;vertical&quot;) +</span></span>
<span id="cb2510-26"><a href="dsi-allCells-allSetsTop.html#cb2510-26"></a><span class="st">    </span><span class="co">#theme(legend.box=&quot;horizontal&quot;) +</span></span>
<span id="cb2510-27"><a href="dsi-allCells-allSetsTop.html#cb2510-27"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.margin=</span><span class="kw">margin</span>()) <span class="co">#+</span></span>
<span id="cb2510-28"><a href="dsi-allCells-allSetsTop.html#cb2510-28"></a>    <span class="co">#guides(fill=guide_legend(nrow=2, byrow=FALSE))</span></span>
<span id="cb2510-29"><a href="dsi-allCells-allSetsTop.html#cb2510-29"></a>    <span class="co">#theme(legend.position = &quot;none&quot;)</span></span>
<span id="cb2510-30"><a href="dsi-allCells-allSetsTop.html#cb2510-30"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb2511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2511-1"><a href="dsi-allCells-allSetsTop.html#cb2511-1"></a>g_legend&lt;-<span class="cf">function</span>(a.gplot){</span>
<span id="cb2511-2"><a href="dsi-allCells-allSetsTop.html#cb2511-2"></a>   tmp &lt;-<span class="st"> </span><span class="kw">ggplot_gtable</span>(<span class="kw">ggplot_build</span>(a.gplot))</span>
<span id="cb2511-3"><a href="dsi-allCells-allSetsTop.html#cb2511-3"></a>   leg &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">sapply</span>(tmp<span class="op">$</span>grobs, <span class="cf">function</span>(x) x<span class="op">$</span>name) <span class="op">==</span><span class="st"> &quot;guide-box&quot;</span>)</span>
<span id="cb2511-4"><a href="dsi-allCells-allSetsTop.html#cb2511-4"></a>   legend &lt;-<span class="st"> </span>tmp<span class="op">$</span>grobs[[leg]]</span>
<span id="cb2511-5"><a href="dsi-allCells-allSetsTop.html#cb2511-5"></a>   <span class="kw">return</span>(legend)</span>
<span id="cb2511-6"><a href="dsi-allCells-allSetsTop.html#cb2511-6"></a>}</span>
<span id="cb2511-7"><a href="dsi-allCells-allSetsTop.html#cb2511-7"></a></span>
<span id="cb2511-8"><a href="dsi-allCells-allSetsTop.html#cb2511-8"></a>redrawClutree &lt;-<span class="st"> </span><span class="cf">function</span>(p){</span>
<span id="cb2511-9"><a href="dsi-allCells-allSetsTop.html#cb2511-9"></a><span class="co">#p &lt;- treeList[[1]] + theme(legend.position=&#39;bottom&#39;)</span></span>
<span id="cb2511-10"><a href="dsi-allCells-allSetsTop.html#cb2511-10"></a><span class="co">#p &lt;- p + theme(legend.background = element_rect(color = &quot;yellow&quot;))</span></span>
<span id="cb2511-11"><a href="dsi-allCells-allSetsTop.html#cb2511-11"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.justification =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb2511-12"><a href="dsi-allCells-allSetsTop.html#cb2511-12"></a><span class="co">#p &lt;- p + theme(legend.justification = c(0,1))</span></span>
<span id="cb2511-13"><a href="dsi-allCells-allSetsTop.html#cb2511-13"></a><span class="co">#lemon::gtable_show_names(p)</span></span>
<span id="cb2511-14"><a href="dsi-allCells-allSetsTop.html#cb2511-14"></a>pNoLeg &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb2511-15"><a href="dsi-allCells-allSetsTop.html#cb2511-15"></a><span class="co"># edge colour:</span></span>
<span id="cb2511-16"><a href="dsi-allCells-allSetsTop.html#cb2511-16"></a>pEdgeCol &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2511-17"><a href="dsi-allCells-allSetsTop.html#cb2511-17"></a><span class="st">  </span><span class="co">#guides(edge_colour = FALSE) +</span></span>
<span id="cb2511-18"><a href="dsi-allCells-allSetsTop.html#cb2511-18"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-19"><a href="dsi-allCells-allSetsTop.html#cb2511-19"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-20"><a href="dsi-allCells-allSetsTop.html#cb2511-20"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2511-21"><a href="dsi-allCells-allSetsTop.html#cb2511-21"></a>pEdgeCol.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pEdgeCol)</span>
<span id="cb2511-22"><a href="dsi-allCells-allSetsTop.html#cb2511-22"></a><span class="co"># edge alpha:</span></span>
<span id="cb2511-23"><a href="dsi-allCells-allSetsTop.html#cb2511-23"></a>pEdgeAlpha &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2511-24"><a href="dsi-allCells-allSetsTop.html#cb2511-24"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-25"><a href="dsi-allCells-allSetsTop.html#cb2511-25"></a><span class="st">  </span><span class="co">#guides(edge_alpha = FALSE) +</span></span>
<span id="cb2511-26"><a href="dsi-allCells-allSetsTop.html#cb2511-26"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-27"><a href="dsi-allCells-allSetsTop.html#cb2511-27"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2511-28"><a href="dsi-allCells-allSetsTop.html#cb2511-28"></a>pEdgeAlpha.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pEdgeAlpha)</span>
<span id="cb2511-29"><a href="dsi-allCells-allSetsTop.html#cb2511-29"></a><span class="co"># size</span></span>
<span id="cb2511-30"><a href="dsi-allCells-allSetsTop.html#cb2511-30"></a>pSize &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2511-31"><a href="dsi-allCells-allSetsTop.html#cb2511-31"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-32"><a href="dsi-allCells-allSetsTop.html#cb2511-32"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-33"><a href="dsi-allCells-allSetsTop.html#cb2511-33"></a><span class="st">  </span><span class="co">#guides(size = FALSE) +</span></span>
<span id="cb2511-34"><a href="dsi-allCells-allSetsTop.html#cb2511-34"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2511-35"><a href="dsi-allCells-allSetsTop.html#cb2511-35"></a>pSize.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pSize)</span>
<span id="cb2511-36"><a href="dsi-allCells-allSetsTop.html#cb2511-36"></a><span class="co"># colour</span></span>
<span id="cb2511-37"><a href="dsi-allCells-allSetsTop.html#cb2511-37"></a>pColour &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb2511-38"><a href="dsi-allCells-allSetsTop.html#cb2511-38"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-39"><a href="dsi-allCells-allSetsTop.html#cb2511-39"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2511-40"><a href="dsi-allCells-allSetsTop.html#cb2511-40"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="co">#+</span></span>
<span id="cb2511-41"><a href="dsi-allCells-allSetsTop.html#cb2511-41"></a>  <span class="co">#guides(colour = FALSE) </span></span>
<span id="cb2511-42"><a href="dsi-allCells-allSetsTop.html#cb2511-42"></a>pColour.leg &lt;-<span class="st"> </span><span class="kw">g_legend</span>(pColour)</span>
<span id="cb2511-43"><a href="dsi-allCells-allSetsTop.html#cb2511-43"></a></span>
<span id="cb2511-44"><a href="dsi-allCells-allSetsTop.html#cb2511-44"></a>grobx &lt;-<span class="st"> </span><span class="kw">arrangeGrob</span>(pNoLeg,</span>
<span id="cb2511-45"><a href="dsi-allCells-allSetsTop.html#cb2511-45"></a>                        pEdgeCol.leg,</span>
<span id="cb2511-46"><a href="dsi-allCells-allSetsTop.html#cb2511-46"></a>                        pEdgeAlpha.leg,</span>
<span id="cb2511-47"><a href="dsi-allCells-allSetsTop.html#cb2511-47"></a>                        <span class="co">#pColour.leg,</span></span>
<span id="cb2511-48"><a href="dsi-allCells-allSetsTop.html#cb2511-48"></a>                        pSize.leg,</span>
<span id="cb2511-49"><a href="dsi-allCells-allSetsTop.html#cb2511-49"></a>                        <span class="dt">nrow=</span><span class="dv">1</span>, <span class="dt">ncol=</span><span class="dv">4</span>,</span>
<span id="cb2511-50"><a href="dsi-allCells-allSetsTop.html#cb2511-50"></a>                        <span class="dt">layout_matrix=</span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="dt">ncol=</span><span class="dv">4</span>, <span class="dt">byrow=</span><span class="ot">TRUE</span>),</span>
<span id="cb2511-51"><a href="dsi-allCells-allSetsTop.html#cb2511-51"></a>                        <span class="dt">widths=</span><span class="kw">c</span>(<span class="kw">unit</span>(.<span class="dv">64</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2511-52"><a href="dsi-allCells-allSetsTop.html#cb2511-52"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2511-53"><a href="dsi-allCells-allSetsTop.html#cb2511-53"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb2511-54"><a href="dsi-allCells-allSetsTop.html#cb2511-54"></a>                                  <span class="kw">unit</span>(.<span class="dv">12</span>, <span class="st">&quot;npc&quot;</span>))</span>
<span id="cb2511-55"><a href="dsi-allCells-allSetsTop.html#cb2511-55"></a>                        )</span>
<span id="cb2511-56"><a href="dsi-allCells-allSetsTop.html#cb2511-56"></a>}</span>
<span id="cb2511-57"><a href="dsi-allCells-allSetsTop.html#cb2511-57"></a><span class="co">##gx &lt;- redrawClutree(treeList[[1]] + theme(legend.position=&#39;bottom&#39;))</span></span>
<span id="cb2511-58"><a href="dsi-allCells-allSetsTop.html#cb2511-58"></a><span class="co">##grid::grid.draw(gx)</span></span>
<span id="cb2511-59"><a href="dsi-allCells-allSetsTop.html#cb2511-59"></a><span class="co">## fine # gxList &lt;- lapply(treeList, function(x){redrawClutree(x+theme(legend.position=&#39;bottom&#39;))})</span></span>
<span id="cb2511-60"><a href="dsi-allCells-allSetsTop.html#cb2511-60"></a>gxList &lt;-<span class="st"> </span><span class="kw">lapply</span>(treeList, <span class="cf">function</span>(x){<span class="kw">redrawClutree</span>(x)})</span>
<span id="cb2511-61"><a href="dsi-allCells-allSetsTop.html#cb2511-61"></a><span class="co">##marrangeGrob(gxList, nrow=2, ncol=2)</span></span></code></pre></div>
<div class="sourceCode" id="cb2512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2512-1"><a href="dsi-allCells-allSetsTop.html#cb2512-1"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2512-2"><a href="dsi-allCells-allSetsTop.html#cb2512-2"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2512-3"><a href="dsi-allCells-allSetsTop.html#cb2512-3"></a>      <span class="dt">ncol=</span><span class="dv">3</span>,</span>
<span id="cb2512-4"><a href="dsi-allCells-allSetsTop.html#cb2512-4"></a>      <span class="dt">top =</span> grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&quot;clusterings concordance (number of cells, log10 scale)&quot;</span>,</span>
<span id="cb2512-5"><a href="dsi-allCells-allSetsTop.html#cb2512-5"></a>                     <span class="dt">gp=</span>grid<span class="op">::</span><span class="kw">gpar</span>(<span class="dt">fontsize=</span><span class="dv">12</span>,<span class="dt">font=</span><span class="dv">3</span>))</span>
<span id="cb2512-6"><a href="dsi-allCells-allSetsTop.html#cb2512-6"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowHeatmap_dsi_allCells_allSets-1.png" width="672" /></p>
<p>The redistribution of cells from one set of clusters to another, here ‘within-batch before’ and ‘across-batch after’ correction, may also be visualized with a clustering tree <a href="https://cran.r-project.org/package=clustree">clustree</a>. Clusters are represented as filled circles colored by cluster set (‘before’ in pink, ‘after’ in blue) and sized by cell number. A pair of clusters from two sets are linked according to the number of cells they share with a link that informs on the number of cells shared (color) and the ‘incoming node’ proportion for the node it points to (transparency). Although these plots convey more information than heatmaps below, they may not be as easy to read.</p>
<div class="sourceCode" id="cb2513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2513-1"><a href="dsi-allCells-allSetsTop.html#cb2513-1"></a><span class="co">#```{r, fig.height=figSize*length(treeList)/2, fig.width=figSize}</span></span>
<span id="cb2513-2"><a href="dsi-allCells-allSetsTop.html#cb2513-2"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> treeList,</span>
<span id="cb2513-3"><a href="dsi-allCells-allSetsTop.html#cb2513-3"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2513-4"><a href="dsi-allCells-allSetsTop.html#cb2513-4"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb2514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2514-1"><a href="dsi-allCells-allSetsTop.html#cb2514-1"></a><span class="co">#```{r, fig.height=figSize*length(treeList)/2, fig.width=figSize}</span></span>
<span id="cb2514-2"><a href="dsi-allCells-allSetsTop.html#cb2514-2"></a><span class="co">#grid.arrange(grobs = treeList,</span></span>
<span id="cb2514-3"><a href="dsi-allCells-allSetsTop.html#cb2514-3"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],</span>
<span id="cb2514-4"><a href="dsi-allCells-allSetsTop.html#cb2514-4"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2514-5"><a href="dsi-allCells-allSetsTop.html#cb2514-5"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowTree1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2515-1"><a href="dsi-allCells-allSetsTop.html#cb2515-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">5</span><span class="op">:</span><span class="dv">6</span>],</span>
<span id="cb2515-2"><a href="dsi-allCells-allSetsTop.html#cb2515-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2515-3"><a href="dsi-allCells-allSetsTop.html#cb2515-3"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowTree2_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2516-1"><a href="dsi-allCells-allSetsTop.html#cb2516-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">7</span><span class="op">:</span><span class="dv">8</span>],</span>
<span id="cb2516-2"><a href="dsi-allCells-allSetsTop.html#cb2516-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2516-3"><a href="dsi-allCells-allSetsTop.html#cb2516-3"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowTree3_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2517-1"><a href="dsi-allCells-allSetsTop.html#cb2517-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> gxList[<span class="dv">9</span><span class="op">:</span><span class="dv">11</span>],</span>
<span id="cb2517-2"><a href="dsi-allCells-allSetsTop.html#cb2517-2"></a>      <span class="dt">ncol=</span><span class="dv">1</span></span>
<span id="cb2517-3"><a href="dsi-allCells-allSetsTop.html#cb2517-3"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowTree4_dsi_allCells_allSets-1.png" width="672" /></p>
<p>The same plots in more compact form with no legend:</p>
<!-- remove legend and have plots in two columns -->
<div class="sourceCode" id="cb2518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2518-1"><a href="dsi-allCells-allSetsTop.html#cb2518-1"></a>treeList &lt;-<span class="st"> </span><span class="kw">lapply</span>(treeList, <span class="cf">function</span>(p){</span>
<span id="cb2518-2"><a href="dsi-allCells-allSetsTop.html#cb2518-2"></a>  p <span class="op">+</span></span>
<span id="cb2518-3"><a href="dsi-allCells-allSetsTop.html#cb2518-3"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">edge_colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2518-4"><a href="dsi-allCells-allSetsTop.html#cb2518-4"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">edge_alpha =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2518-5"><a href="dsi-allCells-allSetsTop.html#cb2518-5"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb2518-6"><a href="dsi-allCells-allSetsTop.html#cb2518-6"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) </span>
<span id="cb2518-7"><a href="dsi-allCells-allSetsTop.html#cb2518-7"></a>})</span></code></pre></div>
<div class="sourceCode" id="cb2519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2519-1"><a href="dsi-allCells-allSetsTop.html#cb2519-1"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> treeList,</span>
<span id="cb2519-2"><a href="dsi-allCells-allSetsTop.html#cb2519-2"></a>      <span class="dt">ncol=</span><span class="dv">3</span></span>
<span id="cb2519-3"><a href="dsi-allCells-allSetsTop.html#cb2519-3"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_biolHet_plotShowTree5_dsi_allCells_allSets-1.png" width="672" />
Co-assignment probabilities</p>
<div class="sourceCode" id="cb2520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2520-1"><a href="dsi-allCells-allSetsTop.html#cb2520-1"></a><span class="co"># For the first batch.</span></span>
<span id="cb2520-2"><a href="dsi-allCells-allSetsTop.html#cb2520-2"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb2520-3"><a href="dsi-allCells-allSetsTop.html#cb2520-3"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2520-4"><a href="dsi-allCells-allSetsTop.html#cb2520-4"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb2520-5"><a href="dsi-allCells-allSetsTop.html#cb2520-5"></a>heat1 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb2520-6"><a href="dsi-allCells-allSetsTop.html#cb2520-6"></a>  tab,</span>
<span id="cb2520-7"><a href="dsi-allCells-allSetsTop.html#cb2520-7"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2520-8"><a href="dsi-allCells-allSetsTop.html#cb2520-8"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2520-9"><a href="dsi-allCells-allSetsTop.html#cb2520-9"></a>  <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2520-10"><a href="dsi-allCells-allSetsTop.html#cb2520-10"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]),</span>
<span id="cb2520-11"><a href="dsi-allCells-allSetsTop.html#cb2520-11"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2520-12"><a href="dsi-allCells-allSetsTop.html#cb2520-12"></a></span>
<span id="cb2520-13"><a href="dsi-allCells-allSetsTop.html#cb2520-13"></a><span class="co"># For the second batch.</span></span>
<span id="cb2520-14"><a href="dsi-allCells-allSetsTop.html#cb2520-14"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb2520-15"><a href="dsi-allCells-allSetsTop.html#cb2520-15"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2520-16"><a href="dsi-allCells-allSetsTop.html#cb2520-16"></a>tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]]), clusters.mnn[tmpInd])</span>
<span id="cb2520-17"><a href="dsi-allCells-allSetsTop.html#cb2520-17"></a>heat2 &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(</span>
<span id="cb2520-18"><a href="dsi-allCells-allSetsTop.html#cb2520-18"></a>  tab,</span>
<span id="cb2520-19"><a href="dsi-allCells-allSetsTop.html#cb2520-19"></a>  <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2520-20"><a href="dsi-allCells-allSetsTop.html#cb2520-20"></a>  <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2520-21"><a href="dsi-allCells-allSetsTop.html#cb2520-21"></a>  <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2520-22"><a href="dsi-allCells-allSetsTop.html#cb2520-22"></a>  <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s probabilities&quot;</span>, splVec[batchPlace]),</span>
<span id="cb2520-23"><a href="dsi-allCells-allSetsTop.html#cb2520-23"></a>  <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2520-24"><a href="dsi-allCells-allSetsTop.html#cb2520-24"></a></span>
<span id="cb2520-25"><a href="dsi-allCells-allSetsTop.html#cb2520-25"></a><span class="kw">grid.arrange</span>(heat1[[<span class="dv">4</span>]], heat2[[<span class="dv">4</span>]])</span></code></pre></div>
<div class="sourceCode" id="cb2521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2521-1"><a href="dsi-allCells-allSetsTop.html#cb2521-1"></a><span class="co"># coassignProb manual: now deprecated for pairwiseRand. </span></span>
<span id="cb2521-2"><a href="dsi-allCells-allSetsTop.html#cb2521-2"></a><span class="co"># Note that the coassignment probability is closely related to the Rand index-based ratios broken down by cluster pair in pairwiseRand with mode=&quot;ratio&quot; and adjusted=FALSE. The off-diagonal coassignment probabilities are simply 1 minus the off-diagonal ratio, while the on-diagonal values differ only by the lack of consideration of pairs of the same cell in pairwiseRand. </span></span>
<span id="cb2521-3"><a href="dsi-allCells-allSetsTop.html#cb2521-3"></a></span>
<span id="cb2521-4"><a href="dsi-allCells-allSetsTop.html#cb2521-4"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2521-5"><a href="dsi-allCells-allSetsTop.html#cb2521-5"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2521-6"><a href="dsi-allCells-allSetsTop.html#cb2521-6"></a>  tab &lt;-<span class="st"> </span><span class="kw">coassignProb</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]),</span>
<span id="cb2521-7"><a href="dsi-allCells-allSetsTop.html#cb2521-7"></a>                      clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]])</span>
<span id="cb2521-8"><a href="dsi-allCells-allSetsTop.html#cb2521-8"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tab,</span>
<span id="cb2521-9"><a href="dsi-allCells-allSetsTop.html#cb2521-9"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2521-10"><a href="dsi-allCells-allSetsTop.html#cb2521-10"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2521-11"><a href="dsi-allCells-allSetsTop.html#cb2521-11"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2521-12"><a href="dsi-allCells-allSetsTop.html#cb2521-12"></a>                                 <span class="co">#main=sprintf(&quot;%s probabilities&quot;, splVec[splIdx]),</span></span>
<span id="cb2521-13"><a href="dsi-allCells-allSetsTop.html#cb2521-13"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s&quot;</span>, splVec[splIdx]),</span>
<span id="cb2521-14"><a href="dsi-allCells-allSetsTop.html#cb2521-14"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2521-15"><a href="dsi-allCells-allSetsTop.html#cb2521-15"></a>}</span>
<span id="cb2521-16"><a href="dsi-allCells-allSetsTop.html#cb2521-16"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2521-17"><a href="dsi-allCells-allSetsTop.html#cb2521-17"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2521-18"><a href="dsi-allCells-allSetsTop.html#cb2521-18"></a>      <span class="dt">ncol=</span><span class="dv">3</span></span>
<span id="cb2521-19"><a href="dsi-allCells-allSetsTop.html#cb2521-19"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_coasProb_plot_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
<div id="rand-index-10" class="section level4">
<h4><span class="header-section-number">29.8.1.2</span> Rand index:</h4>
<div class="sourceCode" id="cb2522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2522-1"><a href="dsi-allCells-allSetsTop.html#cb2522-1"></a><span class="co"># OLD</span></span>
<span id="cb2522-2"><a href="dsi-allCells-allSetsTop.html#cb2522-2"></a></span>
<span id="cb2522-3"><a href="dsi-allCells-allSetsTop.html#cb2522-3"></a><span class="kw">library</span>(fossil)</span>
<span id="cb2522-4"><a href="dsi-allCells-allSetsTop.html#cb2522-4"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb2522-5"><a href="dsi-allCells-allSetsTop.html#cb2522-5"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2522-6"><a href="dsi-allCells-allSetsTop.html#cb2522-6"></a>ri1 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb2522-7"><a href="dsi-allCells-allSetsTop.html#cb2522-7"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2522-8"><a href="dsi-allCells-allSetsTop.html#cb2522-8"></a>ri1</span>
<span id="cb2522-9"><a href="dsi-allCells-allSetsTop.html#cb2522-9"></a></span>
<span id="cb2522-10"><a href="dsi-allCells-allSetsTop.html#cb2522-10"></a>batchPlace &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb2522-11"><a href="dsi-allCells-allSetsTop.html#cb2522-11"></a>tmpInd &lt;-<span class="st"> </span>mnn.out<span class="op">$</span>batch<span class="op">==</span><span class="kw">levels</span>(mnn.out<span class="op">$</span>batch)[batchPlace]</span>
<span id="cb2522-12"><a href="dsi-allCells-allSetsTop.html#cb2522-12"></a>ri2 &lt;-<span class="st"> </span><span class="kw">rand.index</span>(<span class="kw">as.integer</span>(clusters.mnn[tmpInd]),</span>
<span id="cb2522-13"><a href="dsi-allCells-allSetsTop.html#cb2522-13"></a>    <span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[batchPlace]])))</span>
<span id="cb2522-14"><a href="dsi-allCells-allSetsTop.html#cb2522-14"></a>ri2</span></code></pre></div>
<div class="sourceCode" id="cb2523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2523-1"><a href="dsi-allCells-allSetsTop.html#cb2523-1"></a><span class="co"># pairwiseRand(), index, adjusted</span></span>
<span id="cb2523-2"><a href="dsi-allCells-allSetsTop.html#cb2523-2"></a>ariVec &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;numeric&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2523-3"><a href="dsi-allCells-allSetsTop.html#cb2523-3"></a><span class="kw">names</span>(ariVec) &lt;-<span class="st"> </span>splVec</span>
<span id="cb2523-4"><a href="dsi-allCells-allSetsTop.html#cb2523-4"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2523-5"><a href="dsi-allCells-allSetsTop.html#cb2523-5"></a>  ariVec[splIdx] &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb2523-6"><a href="dsi-allCells-allSetsTop.html#cb2523-6"></a>    <span class="dt">ref=</span><span class="kw">as.integer</span>(clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb2523-7"><a href="dsi-allCells-allSetsTop.html#cb2523-7"></a>    <span class="dt">alt=</span><span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]])),</span>
<span id="cb2523-8"><a href="dsi-allCells-allSetsTop.html#cb2523-8"></a>    <span class="dt">mode=</span><span class="st">&quot;index&quot;</span>)</span>
<span id="cb2523-9"><a href="dsi-allCells-allSetsTop.html#cb2523-9"></a>}</span>
<span id="cb2523-10"><a href="dsi-allCells-allSetsTop.html#cb2523-10"></a>ariVec &lt;-<span class="st"> </span><span class="kw">round</span>(ariVec,<span class="dv">2</span>)</span>
<span id="cb2523-11"><a href="dsi-allCells-allSetsTop.html#cb2523-11"></a>ariVec</span></code></pre></div>
<pre><code>## ETV6-RUNX1_1 ETV6-RUNX1_2 ETV6-RUNX1_3 ETV6-RUNX1_4        HHD_1        HHD_2 
##         0.35         0.25         0.72         0.41         0.34         0.54 
##      PRE-T_1      PRE-T_2      PBMMC_1      PBMMC_2      PBMMC_3 
##         0.36         0.66         0.75         0.64         0.59</code></pre>
<div class="sourceCode" id="cb2525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2525-1"><a href="dsi-allCells-allSetsTop.html#cb2525-1"></a><span class="co"># pairwiseRand(), ratio, adjusted</span></span>
<span id="cb2525-2"><a href="dsi-allCells-allSetsTop.html#cb2525-2"></a><span class="co"># square numeric matrix is returned with number of rows equal to the number of unique levels in ref.</span></span>
<span id="cb2525-3"><a href="dsi-allCells-allSetsTop.html#cb2525-3"></a></span>
<span id="cb2525-4"><a href="dsi-allCells-allSetsTop.html#cb2525-4"></a>tabList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2525-5"><a href="dsi-allCells-allSetsTop.html#cb2525-5"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2525-6"><a href="dsi-allCells-allSetsTop.html#cb2525-6"></a>  tabList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pairwiseRand</span>(</span>
<span id="cb2525-7"><a href="dsi-allCells-allSetsTop.html#cb2525-7"></a>    <span class="dt">ref=</span><span class="kw">as.integer</span>(clusters.mnn[rescaled.rb<span class="op">$</span>batch<span class="op">==</span>splVec[splIdx]]),</span>
<span id="cb2525-8"><a href="dsi-allCells-allSetsTop.html#cb2525-8"></a>      <span class="dt">alt=</span><span class="kw">as.integer</span>(<span class="kw">colLabels</span>(rescaled.mbn[[splIdx]]))</span>
<span id="cb2525-9"><a href="dsi-allCells-allSetsTop.html#cb2525-9"></a>    )</span>
<span id="cb2525-10"><a href="dsi-allCells-allSetsTop.html#cb2525-10"></a>}</span>
<span id="cb2525-11"><a href="dsi-allCells-allSetsTop.html#cb2525-11"></a>randVal &lt;-<span class="st"> </span><span class="kw">unlist</span>(tabList) </span>
<span id="cb2525-12"><a href="dsi-allCells-allSetsTop.html#cb2525-12"></a></span>
<span id="cb2525-13"><a href="dsi-allCells-allSetsTop.html#cb2525-13"></a><span class="co">## make breaks from combined range</span></span>
<span id="cb2525-14"><a href="dsi-allCells-allSetsTop.html#cb2525-14"></a>limits &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb2525-15"><a href="dsi-allCells-allSetsTop.html#cb2525-15"></a>  <span class="kw">min</span>(randVal, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</span>
<span id="cb2525-16"><a href="dsi-allCells-allSetsTop.html#cb2525-16"></a>  <span class="kw">max</span>(randVal, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>))</span>
<span id="cb2525-17"><a href="dsi-allCells-allSetsTop.html#cb2525-17"></a>limits &lt;-<span class="st"> </span><span class="kw">quantile</span>(randVal, <span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</span>
<span id="cb2525-18"><a href="dsi-allCells-allSetsTop.html#cb2525-18"></a></span>
<span id="cb2525-19"><a href="dsi-allCells-allSetsTop.html#cb2525-19"></a>Breaks &lt;-<span class="st"> </span><span class="kw">seq</span>(limits[<span class="dv">1</span>], limits[<span class="dv">2</span>],</span>
<span id="cb2525-20"><a href="dsi-allCells-allSetsTop.html#cb2525-20"></a>              <span class="dt">length =</span> <span class="dv">100</span>)</span>
<span id="cb2525-21"><a href="dsi-allCells-allSetsTop.html#cb2525-21"></a></span>
<span id="cb2525-22"><a href="dsi-allCells-allSetsTop.html#cb2525-22"></a>plotList &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">mode =</span> <span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> <span class="kw">length</span>(splVec))</span>
<span id="cb2525-23"><a href="dsi-allCells-allSetsTop.html#cb2525-23"></a><span class="cf">for</span> (splIdx <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(splVec)) {</span>
<span id="cb2525-24"><a href="dsi-allCells-allSetsTop.html#cb2525-24"></a>  plotList[[splIdx]] &lt;-<span class="st"> </span><span class="kw">pheatmap</span>(tabList[[splIdx]],</span>
<span id="cb2525-25"><a href="dsi-allCells-allSetsTop.html#cb2525-25"></a>                                 <span class="dt">cluster_row=</span><span class="ot">FALSE</span>,</span>
<span id="cb2525-26"><a href="dsi-allCells-allSetsTop.html#cb2525-26"></a>                                 <span class="dt">cluster_col=</span><span class="ot">FALSE</span>,</span>
<span id="cb2525-27"><a href="dsi-allCells-allSetsTop.html#cb2525-27"></a>                                 <span class="dt">col=</span><span class="kw">rev</span>(viridis<span class="op">::</span><span class="kw">magma</span>(<span class="dv">100</span>)),</span>
<span id="cb2525-28"><a href="dsi-allCells-allSetsTop.html#cb2525-28"></a>                                 <span class="dt">breaks=</span>Breaks,</span>
<span id="cb2525-29"><a href="dsi-allCells-allSetsTop.html#cb2525-29"></a>                                 <span class="co">#main=sprintf(&quot;%s ratio&quot;, splVec[splIdx]),</span></span>
<span id="cb2525-30"><a href="dsi-allCells-allSetsTop.html#cb2525-30"></a>                                 <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s&quot;</span>, splVec[splIdx]),</span>
<span id="cb2525-31"><a href="dsi-allCells-allSetsTop.html#cb2525-31"></a>                                 <span class="dt">silent=</span><span class="ot">TRUE</span>)</span>
<span id="cb2525-32"><a href="dsi-allCells-allSetsTop.html#cb2525-32"></a>}</span>
<span id="cb2525-33"><a href="dsi-allCells-allSetsTop.html#cb2525-33"></a>grobList &lt;-<span class="st"> </span><span class="kw">lapply</span>(plotList, <span class="cf">function</span>(x){x[[<span class="dv">4</span>]]})</span>
<span id="cb2525-34"><a href="dsi-allCells-allSetsTop.html#cb2525-34"></a><span class="kw">grid.arrange</span>(<span class="dt">grobs =</span> grobList,</span>
<span id="cb2525-35"><a href="dsi-allCells-allSetsTop.html#cb2525-35"></a>      <span class="dt">ncol=</span><span class="dv">3</span></span>
<span id="cb2525-36"><a href="dsi-allCells-allSetsTop.html#cb2525-36"></a>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_randIndex_plot_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
</div>
<div id="cluster-markers-1" class="section level3">
<h3><span class="header-section-number">29.8.2</span> Cluster markers:</h3>
<div class="sourceCode" id="cb2526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2526-1"><a href="dsi-allCells-allSetsTop.html#cb2526-1"></a><span class="co"># OLD</span></span>
<span id="cb2526-2"><a href="dsi-allCells-allSetsTop.html#cb2526-2"></a></span>
<span id="cb2526-3"><a href="dsi-allCells-allSetsTop.html#cb2526-3"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(</span>
<span id="cb2526-4"><a href="dsi-allCells-allSetsTop.html#cb2526-4"></a>  uncorrected,</span>
<span id="cb2526-5"><a href="dsi-allCells-allSetsTop.html#cb2526-5"></a>  clusters.mnn,</span>
<span id="cb2526-6"><a href="dsi-allCells-allSetsTop.html#cb2526-6"></a>  <span class="dt">block=</span>uncorrected<span class="op">$</span>batch,</span>
<span id="cb2526-7"><a href="dsi-allCells-allSetsTop.html#cb2526-7"></a>  <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb2526-8"><a href="dsi-allCells-allSetsTop.html#cb2526-8"></a>  <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb2526-9"><a href="dsi-allCells-allSetsTop.html#cb2526-9"></a>  <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[,</span>
<span id="cb2526-10"><a href="dsi-allCells-allSetsTop.html#cb2526-10"></a>                                <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),</span>
<span id="cb2526-11"><a href="dsi-allCells-allSetsTop.html#cb2526-11"></a>                                <span class="dt">drop=</span><span class="ot">FALSE</span>])</span></code></pre></div>
<div class="sourceCode" id="cb2527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2527-1"><a href="dsi-allCells-allSetsTop.html#cb2527-1"></a><span class="co">#m.out &lt;- findMarkers(uncorrected,</span></span>
<span id="cb2527-2"><a href="dsi-allCells-allSetsTop.html#cb2527-2"></a>m.out &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(uncorrected[<span class="kw">rownames</span>(mnn.out),],</span>
<span id="cb2527-3"><a href="dsi-allCells-allSetsTop.html#cb2527-3"></a>                     clusters.mnn,</span>
<span id="cb2527-4"><a href="dsi-allCells-allSetsTop.html#cb2527-4"></a>                     <span class="dt">block=</span>uncorrected<span class="op">$</span>batch,</span>
<span id="cb2527-5"><a href="dsi-allCells-allSetsTop.html#cb2527-5"></a>                     <span class="dt">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb2527-6"><a href="dsi-allCells-allSetsTop.html#cb2527-6"></a>                     <span class="dt">lfc=</span><span class="dv">1</span>,</span>
<span id="cb2527-7"><a href="dsi-allCells-allSetsTop.html#cb2527-7"></a>                     <span class="co">#row.data=rowData(uncorrected)[,c(&quot;ensembl_gene_id&quot;,&quot;Symbol&quot;),drop=FALSE],</span></span>
<span id="cb2527-8"><a href="dsi-allCells-allSetsTop.html#cb2527-8"></a>                     <span class="dt">row.data=</span><span class="kw">rowData</span>(uncorrected)[<span class="kw">rownames</span>(mnn.out),<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;Symbol&quot;</span>),<span class="dt">drop=</span><span class="ot">FALSE</span>],</span>
<span id="cb2527-9"><a href="dsi-allCells-allSetsTop.html#cb2527-9"></a>                     <span class="dt">BPPARAM=</span>bpp)</span>
<span id="cb2527-10"><a href="dsi-allCells-allSetsTop.html#cb2527-10"></a></span>
<span id="cb2527-11"><a href="dsi-allCells-allSetsTop.html#cb2527-11"></a><span class="co">#lapply(m.out, function(x){head(x[,2:6])})</span></span>
<span id="cb2527-12"><a href="dsi-allCells-allSetsTop.html#cb2527-12"></a></span>
<span id="cb2527-13"><a href="dsi-allCells-allSetsTop.html#cb2527-13"></a>tl1 &lt;-<span class="st"> </span><span class="kw">lapply</span>(m.out, <span class="cf">function</span>(x){x[x<span class="op">$</span>Symbol<span class="op">==</span><span class="st">&quot;CD3D&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.10</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>]}) <span class="co"># T-cell</span></span>
<span id="cb2527-14"><a href="dsi-allCells-allSetsTop.html#cb2527-14"></a>tl2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(m.out, <span class="cf">function</span>(x){x[x<span class="op">$</span>Symbol<span class="op">==</span><span class="st">&quot;CD69&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.20</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>]}) <span class="co"># activation</span></span>
<span id="cb2527-15"><a href="dsi-allCells-allSetsTop.html#cb2527-15"></a></span>
<span id="cb2527-16"><a href="dsi-allCells-allSetsTop.html#cb2527-16"></a>tb1 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(tl1, nrow)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb2527-17"><a href="dsi-allCells-allSetsTop.html#cb2527-17"></a>tb2 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(tl2, nrow)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb2527-18"><a href="dsi-allCells-allSetsTop.html#cb2527-18"></a></span>
<span id="cb2527-19"><a href="dsi-allCells-allSetsTop.html#cb2527-19"></a>cluToGet &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">c</span>(<span class="kw">which</span>(tb1), <span class="kw">which</span>(tb2)))[<span class="dv">1</span>] <span class="co"># 3 # 19 # 4</span></span>
<span id="cb2527-20"><a href="dsi-allCells-allSetsTop.html#cb2527-20"></a>demo &lt;-<span class="st"> </span>m.out[[cluToGet]]</span>
<span id="cb2527-21"><a href="dsi-allCells-allSetsTop.html#cb2527-21"></a><span class="co">#as.data.frame(demo[1:20,c(&quot;Symbol&quot;, &quot;Top&quot;, &quot;p.value&quot;, &quot;FDR&quot;, &quot;summary.logFC&quot;)]) </span></span>
<span id="cb2527-22"><a href="dsi-allCells-allSetsTop.html#cb2527-22"></a></span>
<span id="cb2527-23"><a href="dsi-allCells-allSetsTop.html#cb2527-23"></a>geneEnsId &lt;-<span class="st"> </span>demo <span class="op">%&gt;%</span><span class="st"> </span>data.frame <span class="op">%&gt;%</span></span>
<span id="cb2527-24"><a href="dsi-allCells-allSetsTop.html#cb2527-24"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">str_detect</span>(Symbol, <span class="st">&quot;^RP&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb2527-25"><a href="dsi-allCells-allSetsTop.html#cb2527-25"></a><span class="st">  </span><span class="kw">pull</span>(ensembl_gene_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>(<span class="dv">1</span>)</span>
<span id="cb2527-26"><a href="dsi-allCells-allSetsTop.html#cb2527-26"></a>geneSymbol &lt;-<span class="st"> </span><span class="kw">rowData</span>(uncorrected)[geneEnsId,<span class="st">&quot;Symbol&quot;</span>]</span></code></pre></div>
<p>Expression level for the top gene in cluster 5, CD3D, (ENSG00000167286):</p>
<div class="sourceCode" id="cb2528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2528-1"><a href="dsi-allCells-allSetsTop.html#cb2528-1"></a><span class="co"># OLD</span></span>
<span id="cb2528-2"><a href="dsi-allCells-allSetsTop.html#cb2528-2"></a></span>
<span id="cb2528-3"><a href="dsi-allCells-allSetsTop.html#cb2528-3"></a>geneEnsId &lt;-<span class="st"> </span><span class="kw">rownames</span>(demo)[<span class="dv">1</span>]</span>
<span id="cb2528-4"><a href="dsi-allCells-allSetsTop.html#cb2528-4"></a></span>
<span id="cb2528-5"><a href="dsi-allCells-allSetsTop.html#cb2528-5"></a><span class="kw">plotExpression</span>(</span>
<span id="cb2528-6"><a href="dsi-allCells-allSetsTop.html#cb2528-6"></a>  uncorrected,</span>
<span id="cb2528-7"><a href="dsi-allCells-allSetsTop.html#cb2528-7"></a>  <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)),</span>
<span id="cb2528-8"><a href="dsi-allCells-allSetsTop.html#cb2528-8"></a>  <span class="dt">features=</span>geneEnsId,</span>
<span id="cb2528-9"><a href="dsi-allCells-allSetsTop.html#cb2528-9"></a>  <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span></span>
<span id="cb2528-10"><a href="dsi-allCells-allSetsTop.html#cb2528-10"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by)</span></code></pre></div>
<div class="sourceCode" id="cb2529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2529-1"><a href="dsi-allCells-allSetsTop.html#cb2529-1"></a><span class="co">#geneEnsId &lt;- rownames(demo)[1]</span></span>
<span id="cb2529-2"><a href="dsi-allCells-allSetsTop.html#cb2529-2"></a></span>
<span id="cb2529-3"><a href="dsi-allCells-allSetsTop.html#cb2529-3"></a><span class="kw">plotExpression</span>(uncorrected,</span>
<span id="cb2529-4"><a href="dsi-allCells-allSetsTop.html#cb2529-4"></a>               <span class="dt">x=</span><span class="kw">I</span>(<span class="kw">factor</span>(clusters.mnn)),</span>
<span id="cb2529-5"><a href="dsi-allCells-allSetsTop.html#cb2529-5"></a>               <span class="dt">features=</span>geneEnsId,</span>
<span id="cb2529-6"><a href="dsi-allCells-allSetsTop.html#cb2529-6"></a>           <span class="dt">colour_by=</span><span class="st">&quot;batch&quot;</span>) <span class="op">+</span></span>
<span id="cb2529-7"><a href="dsi-allCells-allSetsTop.html#cb2529-7"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>colour_by) <span class="op">+</span></span>
<span id="cb2529-8"><a href="dsi-allCells-allSetsTop.html#cb2529-8"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2529-9"><a href="dsi-allCells-allSetsTop.html#cb2529-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s %s&quot;</span>,</span>
<span id="cb2529-10"><a href="dsi-allCells-allSetsTop.html#cb2529-10"></a>          geneEnsId,</span>
<span id="cb2529-11"><a href="dsi-allCells-allSetsTop.html#cb2529-11"></a>          <span class="kw">rowData</span>(uncorrected)[geneEnsId,<span class="st">&quot;Symbol&quot;</span>])</span>
<span id="cb2529-12"><a href="dsi-allCells-allSetsTop.html#cb2529-12"></a>          )</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_consis_uncorr_topGene_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Expression level for the top gene, CD3D on t-SNE plot:</p>
<p><strong>Not Encouraging consistency</strong> with marker genes</p>
<div class="sourceCode" id="cb2530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2530-1"><a href="dsi-allCells-allSetsTop.html#cb2530-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb2530-2"><a href="dsi-allCells-allSetsTop.html#cb2530-2"></a>          <span class="dt">colour_by =</span> geneEnsId,</span>
<span id="cb2530-3"><a href="dsi-allCells-allSetsTop.html#cb2530-3"></a>          <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb2530-4"><a href="dsi-allCells-allSetsTop.html#cb2530-4"></a>          <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2530-5"><a href="dsi-allCells-allSetsTop.html#cb2530-5"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;cluster&quot;</span>, cluToGet, geneEnsId,geneSymbol))</span>
<span id="cb2530-6"><a href="dsi-allCells-allSetsTop.html#cb2530-6"></a><span class="co">#print(p)</span></span>
<span id="cb2530-7"><a href="dsi-allCells-allSetsTop.html#cb2530-7"></a>p1 &lt;-<span class="st"> </span>p</span>
<span id="cb2530-8"><a href="dsi-allCells-allSetsTop.html#cb2530-8"></a>p2 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">colData</span>(mnn.out)<span class="op">$</span>batch)</span>
<span id="cb2530-9"><a href="dsi-allCells-allSetsTop.html#cb2530-9"></a><span class="kw">grid.arrange</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2530-10"><a href="dsi-allCells-allSetsTop.html#cb2530-10"></a>                        p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>),</span>
<span id="cb2530-11"><a href="dsi-allCells-allSetsTop.html#cb2530-11"></a>                        <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_consis_uncorr_tsneWoEncour_dsi_allCells_allSets-1.png" width="672" /></p>
</div>
</div>
<div id="identify-clusters-with-pbmmc-cells-5" class="section level2">
<h2><span class="header-section-number">29.9</span> Identify clusters with PBMMC cells</h2>
<!-- Mind clustering now comes after data set integration
so maybe have the session in a separate file
or point to the clustering chapter -->
<p>Cluster size and cell contribution by sample type, with clusters sorted by size:</p>
<div class="sourceCode" id="cb2531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2531-1"><a href="dsi-allCells-allSetsTop.html#cb2531-1"></a>mnn.out<span class="op">$</span>source_name &lt;-<span class="st"> </span>uncorrected<span class="op">$</span>source_name <span class="co"># cell order is maintained by scran functions</span></span>
<span id="cb2531-2"><a href="dsi-allCells-allSetsTop.html#cb2531-2"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.mnn, <span class="st">&quot;batch&quot;</span>=mnn.out<span class="op">$</span>source_name)</span>
<span id="cb2531-3"><a href="dsi-allCells-allSetsTop.html#cb2531-3"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2531-4"><a href="dsi-allCells-allSetsTop.html#cb2531-4"></a>sortVecNames &lt;-<span class="st"> </span>tmpMatTab <span class="op">%&gt;%</span><span class="st"> </span>rowSums <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>names</span>
<span id="cb2531-5"><a href="dsi-allCells-allSetsTop.html#cb2531-5"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2531-6"><a href="dsi-allCells-allSetsTop.html#cb2531-6"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(tmpMat)</span>
<span id="cb2531-7"><a href="dsi-allCells-allSetsTop.html#cb2531-7"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[sortVecNames,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2531-8"><a href="dsi-allCells-allSetsTop.html#cb2531-8"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2531-9"><a href="dsi-allCells-allSetsTop.html#cb2531-9"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2531-10"><a href="dsi-allCells-allSetsTop.html#cb2531-10"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb2531-11"><a href="dsi-allCells-allSetsTop.html#cb2531-11"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2531-12"><a href="dsi-allCells-allSetsTop.html#cb2531-12"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2531-13"><a href="dsi-allCells-allSetsTop.html#cb2531-13"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2531-14"><a href="dsi-allCells-allSetsTop.html#cb2531-14"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span>
<span id="cb2531-15"><a href="dsi-allCells-allSetsTop.html#cb2531-15"></a></span>
<span id="cb2531-16"><a href="dsi-allCells-allSetsTop.html#cb2531-16"></a><span class="co">#grid.arrange(p1, p2)</span></span>
<span id="cb2531-17"><a href="dsi-allCells-allSetsTop.html#cb2531-17"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2531-18"><a href="dsi-allCells-allSetsTop.html#cb2531-18"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2531-19"><a href="dsi-allCells-allSetsTop.html#cb2531-19"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2531-20"><a href="dsi-allCells-allSetsTop.html#cb2531-20"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2531-21"><a href="dsi-allCells-allSetsTop.html#cb2531-21"></a>         legend,</span>
<span id="cb2531-22"><a href="dsi-allCells-allSetsTop.html#cb2531-22"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2531-23"><a href="dsi-allCells-allSetsTop.html#cb2531-23"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotDescSize_dsi_allCells_allSets-1.png" width="672" /></p>
<p>Cluster size and cell contribution by sample type, with clusters sorted by
decreasing proportion of PBMMC:</p>
<div class="sourceCode" id="cb2532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2532-1"><a href="dsi-allCells-allSetsTop.html#cb2532-1"></a>tmpMat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;clusters&quot;</span>=clusters.mnn,</span>
<span id="cb2532-2"><a href="dsi-allCells-allSetsTop.html#cb2532-2"></a>                     <span class="st">&quot;batch&quot;</span>=<span class="kw">factor</span>(mnn.out<span class="op">$</span>source_name),</span>
<span id="cb2532-3"><a href="dsi-allCells-allSetsTop.html#cb2532-3"></a>                     <span class="st">&quot;Sample.Name&quot;</span>=mnn.out<span class="op">$</span>batch</span>
<span id="cb2532-4"><a href="dsi-allCells-allSetsTop.html#cb2532-4"></a>                     )</span>
<span id="cb2532-5"><a href="dsi-allCells-allSetsTop.html#cb2532-5"></a>sortVecNames &lt;-<span class="st"> </span><span class="kw">round</span>(tmpMatTab<span class="op">/</span><span class="kw">rowSums</span>(tmpMatTab),<span class="dv">2</span>) <span class="op">%&gt;%</span></span>
<span id="cb2532-6"><a href="dsi-allCells-allSetsTop.html#cb2532-6"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2532-7"><a href="dsi-allCells-allSetsTop.html#cb2532-7"></a><span class="st">  </span><span class="kw">filter</span>(batch<span class="op">==</span><span class="st">&quot;PBMMC&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb2532-8"><a href="dsi-allCells-allSetsTop.html#cb2532-8"></a><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(Freq)) <span class="op">%&gt;%</span></span>
<span id="cb2532-9"><a href="dsi-allCells-allSetsTop.html#cb2532-9"></a><span class="st">  </span><span class="kw">pull</span>(clusters)</span>
<span id="cb2532-10"><a href="dsi-allCells-allSetsTop.html#cb2532-10"></a>tmpMat<span class="op">$</span>clusters &lt;-<span class="st"> </span><span class="kw">factor</span>(tmpMat<span class="op">$</span>clusters, <span class="dt">levels=</span>sortVecNames)</span>
<span id="cb2532-11"><a href="dsi-allCells-allSetsTop.html#cb2532-11"></a>tmpMatTab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="st">&quot;clusters&quot;</span>=tmpMat<span class="op">$</span>clusters, <span class="st">&quot;batch&quot;</span>=tmpMat<span class="op">$</span>batch)</span>
<span id="cb2532-12"><a href="dsi-allCells-allSetsTop.html#cb2532-12"></a><span class="co">#tmpMatDf &lt;- tmpMatTab[sortVecNames,] %&gt;% data.frame()</span></span>
<span id="cb2532-13"><a href="dsi-allCells-allSetsTop.html#cb2532-13"></a>tmpMatDf &lt;-<span class="st"> </span>tmpMatTab[,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>()</span>
<span id="cb2532-14"><a href="dsi-allCells-allSetsTop.html#cb2532-14"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMatDf, <span class="kw">aes</span>(<span class="dt">x=</span>clusters,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2532-15"><a href="dsi-allCells-allSetsTop.html#cb2532-15"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2532-16"><a href="dsi-allCells-allSetsTop.html#cb2532-16"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb2532-17"><a href="dsi-allCells-allSetsTop.html#cb2532-17"></a></span>
<span id="cb2532-18"><a href="dsi-allCells-allSetsTop.html#cb2532-18"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>batch)) <span class="op">+</span></span>
<span id="cb2532-19"><a href="dsi-allCells-allSetsTop.html#cb2532-19"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2532-20"><a href="dsi-allCells-allSetsTop.html#cb2532-20"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2532-21"><a href="dsi-allCells-allSetsTop.html#cb2532-21"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span>
<span id="cb2532-22"><a href="dsi-allCells-allSetsTop.html#cb2532-22"></a><span class="co">#grid.arrange(p1, p2)</span></span>
<span id="cb2532-23"><a href="dsi-allCells-allSetsTop.html#cb2532-23"></a></span>
<span id="cb2532-24"><a href="dsi-allCells-allSetsTop.html#cb2532-24"></a>p3 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb2532-25"><a href="dsi-allCells-allSetsTop.html#cb2532-25"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb2532-26"><a href="dsi-allCells-allSetsTop.html#cb2532-26"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) </span>
<span id="cb2532-27"><a href="dsi-allCells-allSetsTop.html#cb2532-27"></a></span>
<span id="cb2532-28"><a href="dsi-allCells-allSetsTop.html#cb2532-28"></a>p4 &lt;-<span class="st"> </span>p3 <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent)</span>
<span id="cb2532-29"><a href="dsi-allCells-allSetsTop.html#cb2532-29"></a></span>
<span id="cb2532-30"><a href="dsi-allCells-allSetsTop.html#cb2532-30"></a>p1 &lt;-<span class="st"> </span>p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">5</span>))</span>
<span id="cb2532-31"><a href="dsi-allCells-allSetsTop.html#cb2532-31"></a>p2 &lt;-<span class="st"> </span>p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">5</span>))</span>
<span id="cb2532-32"><a href="dsi-allCells-allSetsTop.html#cb2532-32"></a>p3 &lt;-<span class="st"> </span>p3 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">5</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>tmpMat<span class="op">$</span>batch)</span>
<span id="cb2532-33"><a href="dsi-allCells-allSetsTop.html#cb2532-33"></a>p4 &lt;-<span class="st"> </span>p4 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">5</span>))</span>
<span id="cb2532-34"><a href="dsi-allCells-allSetsTop.html#cb2532-34"></a><span class="co">#grid.arrange(p1, p2, p3)</span></span></code></pre></div>
<div class="sourceCode" id="cb2533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2533-1"><a href="dsi-allCells-allSetsTop.html#cb2533-1"></a><span class="co">#grid.arrange(p1, p2, p4, p3, ncol=1)</span></span>
<span id="cb2533-2"><a href="dsi-allCells-allSetsTop.html#cb2533-2"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p1)</span>
<span id="cb2533-3"><a href="dsi-allCells-allSetsTop.html#cb2533-3"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2533-4"><a href="dsi-allCells-allSetsTop.html#cb2533-4"></a>                         p2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2533-5"><a href="dsi-allCells-allSetsTop.html#cb2533-5"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2533-6"><a href="dsi-allCells-allSetsTop.html#cb2533-6"></a>         legend,</span>
<span id="cb2533-7"><a href="dsi-allCells-allSetsTop.html#cb2533-7"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2533-8"><a href="dsi-allCells-allSetsTop.html#cb2533-8"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotDescPropPbmmc1_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2534-1"><a href="dsi-allCells-allSetsTop.html#cb2534-1"></a><span class="kw">ggplot</span>(<span class="dt">data=</span>tmpMat, <span class="kw">aes</span>(<span class="dt">x=</span>clusters, <span class="dt">fill=</span>Sample.Name)) <span class="op">+</span></span>
<span id="cb2534-2"><a href="dsi-allCells-allSetsTop.html#cb2534-2"></a><span class="st">  </span><span class="kw">geom_bar</span>() <span class="op">+</span></span>
<span id="cb2534-3"><a href="dsi-allCells-allSetsTop.html#cb2534-3"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>tmpMat<span class="op">$</span>batch)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotDescPropPbmmc2_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2535-1"><a href="dsi-allCells-allSetsTop.html#cb2535-1"></a>legend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(p3)</span>
<span id="cb2535-2"><a href="dsi-allCells-allSetsTop.html#cb2535-2"></a><span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p4 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2535-3"><a href="dsi-allCells-allSetsTop.html#cb2535-3"></a>                         p3 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>),</span>
<span id="cb2535-4"><a href="dsi-allCells-allSetsTop.html#cb2535-4"></a>                         <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb2535-5"><a href="dsi-allCells-allSetsTop.html#cb2535-5"></a>         legend,</span>
<span id="cb2535-6"><a href="dsi-allCells-allSetsTop.html#cb2535-6"></a>         <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">5</span><span class="op">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">6</span>),</span>
<span id="cb2535-7"><a href="dsi-allCells-allSetsTop.html#cb2535-7"></a>         <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotDescPropPbmmc3_dsi_allCells_allSets%7D-1.png" width="672" /></p>
<div class="sourceCode" id="cb2536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2536-1"><a href="dsi-allCells-allSetsTop.html#cb2536-1"></a><span class="co">#rm(p1, p2, p3, p4)</span></span></code></pre></div>
<!-- Split by sample type: -->
<div class="sourceCode" id="cb2537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2537-1"><a href="dsi-allCells-allSetsTop.html#cb2537-1"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.mnn,</span>
<span id="cb2537-2"><a href="dsi-allCells-allSetsTop.html#cb2537-2"></a>                 <span class="dt">Batch=</span><span class="kw">as.character</span>(mnn.out<span class="op">$</span>batch))</span>
<span id="cb2537-3"><a href="dsi-allCells-allSetsTop.html#cb2537-3"></a>                 <span class="co">#Batch=as.character(mnn.out$source_name))</span></span>
<span id="cb2537-4"><a href="dsi-allCells-allSetsTop.html#cb2537-4"></a><span class="co">#tab.mnn &lt;- as.data.frame(tab.mnn, stringsAsFactors=FALSE)</span></span>
<span id="cb2537-5"><a href="dsi-allCells-allSetsTop.html#cb2537-5"></a><span class="co">##tab.mnn</span></span>
<span id="cb2537-6"><a href="dsi-allCells-allSetsTop.html#cb2537-6"></a></span>
<span id="cb2537-7"><a href="dsi-allCells-allSetsTop.html#cb2537-7"></a><span class="co"># Using a large pseudo.count to avoid unnecessarily</span></span>
<span id="cb2537-8"><a href="dsi-allCells-allSetsTop.html#cb2537-8"></a><span class="co"># large variances when the counts are low.</span></span>
<span id="cb2537-9"><a href="dsi-allCells-allSetsTop.html#cb2537-9"></a>norm &lt;-<span class="st"> </span><span class="kw">normalizeCounts</span>(tab.mnn, <span class="dt">pseudo_count=</span><span class="dv">10</span>)</span>
<span id="cb2537-10"><a href="dsi-allCells-allSetsTop.html#cb2537-10"></a></span>
<span id="cb2537-11"><a href="dsi-allCells-allSetsTop.html#cb2537-11"></a>normNoLog &lt;-<span class="st"> </span><span class="kw">normalizeCounts</span>(tab.mnn, <span class="dt">pseudo_count=</span><span class="dv">10</span>, <span class="dt">log=</span><span class="ot">FALSE</span>)</span>
<span id="cb2537-12"><a href="dsi-allCells-allSetsTop.html#cb2537-12"></a></span>
<span id="cb2537-13"><a href="dsi-allCells-allSetsTop.html#cb2537-13"></a>sortVecNames &lt;-<span class="st"> </span><span class="kw">rowSums</span>(normNoLog) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">round</span>(<span class="dv">2</span>) <span class="op">%&gt;%</span></span>
<span id="cb2537-14"><a href="dsi-allCells-allSetsTop.html#cb2537-14"></a><span class="st">  </span><span class="kw">sort</span>(<span class="dt">decreasing=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb2537-15"><a href="dsi-allCells-allSetsTop.html#cb2537-15"></a><span class="st">  </span>names</span></code></pre></div>
<div class="sourceCode" id="cb2538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2538-1"><a href="dsi-allCells-allSetsTop.html#cb2538-1"></a>tab.mnn &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">Cluster=</span>clusters.mnn,</span>
<span id="cb2538-2"><a href="dsi-allCells-allSetsTop.html#cb2538-2"></a>         <span class="dt">Batch=</span><span class="kw">as.character</span>(mnn.out<span class="op">$</span>source_name))</span>
<span id="cb2538-3"><a href="dsi-allCells-allSetsTop.html#cb2538-3"></a><span class="co">##tab.mnn</span></span>
<span id="cb2538-4"><a href="dsi-allCells-allSetsTop.html#cb2538-4"></a></span>
<span id="cb2538-5"><a href="dsi-allCells-allSetsTop.html#cb2538-5"></a><span class="co"># Using a large pseudo.count to avoid unnecessarily</span></span>
<span id="cb2538-6"><a href="dsi-allCells-allSetsTop.html#cb2538-6"></a><span class="co"># large variances when the counts are low.</span></span>
<span id="cb2538-7"><a href="dsi-allCells-allSetsTop.html#cb2538-7"></a><span class="co">#norm &lt;- normalizeCounts(tab.mnn, pseudo_count=10)</span></span>
<span id="cb2538-8"><a href="dsi-allCells-allSetsTop.html#cb2538-8"></a></span>
<span id="cb2538-9"><a href="dsi-allCells-allSetsTop.html#cb2538-9"></a>normNoLog &lt;-<span class="st"> </span><span class="kw">normalizeCounts</span>(tab.mnn, <span class="dt">pseudo_count=</span><span class="dv">10</span>, <span class="dt">log=</span><span class="ot">FALSE</span>)</span>
<span id="cb2538-10"><a href="dsi-allCells-allSetsTop.html#cb2538-10"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame.matrix</span>()</span>
<span id="cb2538-11"><a href="dsi-allCells-allSetsTop.html#cb2538-11"></a></span>
<span id="cb2538-12"><a href="dsi-allCells-allSetsTop.html#cb2538-12"></a><span class="co"># sort by PBMMC proportion:</span></span>
<span id="cb2538-13"><a href="dsi-allCells-allSetsTop.html#cb2538-13"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sum=</span><span class="kw">rowSums</span>(.))</span>
<span id="cb2538-14"><a href="dsi-allCells-allSetsTop.html#cb2538-14"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">prop=</span>PBMMC<span class="op">/</span>sum)</span>
<span id="cb2538-15"><a href="dsi-allCells-allSetsTop.html#cb2538-15"></a></span>
<span id="cb2538-16"><a href="dsi-allCells-allSetsTop.html#cb2538-16"></a>sortVecNames &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span></span>
<span id="cb2538-17"><a href="dsi-allCells-allSetsTop.html#cb2538-17"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;clusters&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2538-18"><a href="dsi-allCells-allSetsTop.html#cb2538-18"></a><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(prop)) <span class="op">%&gt;%</span></span>
<span id="cb2538-19"><a href="dsi-allCells-allSetsTop.html#cb2538-19"></a><span class="st">  </span><span class="kw">pull</span>(clusters)</span></code></pre></div>
<p>Some clusters with a high proportion of PBMMC cells also comprise a large number
of cancer cells. To select clusters to keep, we could use the following inclusion
criteria:</p>
<ul>
<li>proportion of PBMMC cells in cluster is lower than the threshold for the
proportion of PBMMC cells in a cluster, eg 50%</li>
<li>proportion of cancer cells in cluster higher than 5% of cells of that sample
type</li>
</ul>
<p>The bar plots below show the clusters ordered by decreasing proportion of PBMMC
and also split by selection outcome (where ‘TRUE’ means inclusion).</p>
<div class="sourceCode" id="cb2539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2539-1"><a href="dsi-allCells-allSetsTop.html#cb2539-1"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;cluster&quot;</span>)</span>
<span id="cb2539-2"><a href="dsi-allCells-allSetsTop.html#cb2539-2"></a>normNoLog<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, normNoLog<span class="op">$</span>cluster)</span>
<span id="cb2539-3"><a href="dsi-allCells-allSetsTop.html#cb2539-3"></a></span>
<span id="cb2539-4"><a href="dsi-allCells-allSetsTop.html#cb2539-4"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">colnames</span>(tab.mnn)</span>
<span id="cb2539-5"><a href="dsi-allCells-allSetsTop.html#cb2539-5"></a>otherSplType &lt;-<span class="st"> </span><span class="kw">setdiff</span>(<span class="kw">colnames</span>(tab.mnn), <span class="st">&quot;PBMMC&quot;</span>) <span class="co"># ok for pairs of sample types</span></span>
<span id="cb2539-6"><a href="dsi-allCells-allSetsTop.html#cb2539-6"></a>thdSize &lt;-<span class="st"> </span><span class="kw">sum</span>(normNoLog[,otherSplType])<span class="op">*</span><span class="fl">0.02</span></span>
<span id="cb2539-7"><a href="dsi-allCells-allSetsTop.html#cb2539-7"></a><span class="co">#thdSize &lt;- sum(normNoLog[,otherSplType])*0.05</span></span>
<span id="cb2539-8"><a href="dsi-allCells-allSetsTop.html#cb2539-8"></a>thdPropPbmmc &lt;-<span class="st"> </span><span class="fl">0.5</span></span>
<span id="cb2539-9"><a href="dsi-allCells-allSetsTop.html#cb2539-9"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span></span>
<span id="cb2539-10"><a href="dsi-allCells-allSetsTop.html#cb2539-10"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Non_PBMMC=</span><span class="kw">rowSums</span>(normNoLog[,otherSplType]))</span>
<span id="cb2539-11"><a href="dsi-allCells-allSetsTop.html#cb2539-11"></a>otherSplType &lt;-<span class="st"> &quot;Non_PBMMC&quot;</span></span>
<span id="cb2539-12"><a href="dsi-allCells-allSetsTop.html#cb2539-12"></a>  </span>
<span id="cb2539-13"><a href="dsi-allCells-allSetsTop.html#cb2539-13"></a><span class="co">#propLtDf &lt;- norm2 %&gt;% select(clusters,propLt050) %&gt;% unique()</span></span>
<span id="cb2539-14"><a href="dsi-allCells-allSetsTop.html#cb2539-14"></a><span class="co">#propLtDf$cluster &lt;- paste0(&quot;c&quot;, propLtDf$clusters)</span></span>
<span id="cb2539-15"><a href="dsi-allCells-allSetsTop.html#cb2539-15"></a></span>
<span id="cb2539-16"><a href="dsi-allCells-allSetsTop.html#cb2539-16"></a>propLtDf &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span></span>
<span id="cb2539-17"><a href="dsi-allCells-allSetsTop.html#cb2539-17"></a><span class="st">  </span><span class="kw">filter</span>(prop <span class="op">&lt;</span><span class="st"> </span>thdPropPbmmc <span class="op">|</span><span class="st"> </span><span class="op">!!</span><span class="kw">sym</span>(otherSplType) <span class="op">&gt;</span><span class="st"> </span>thdSize) <span class="co"># ok for pairs of sample types</span></span>
<span id="cb2539-18"><a href="dsi-allCells-allSetsTop.html#cb2539-18"></a></span>
<span id="cb2539-19"><a href="dsi-allCells-allSetsTop.html#cb2539-19"></a>normNoLog &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span></span>
<span id="cb2539-20"><a href="dsi-allCells-allSetsTop.html#cb2539-20"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tmpCluBool=</span> <span class="kw">ifelse</span>((prop <span class="op">&lt;</span><span class="st"> </span>thdPropPbmmc <span class="op">|</span><span class="st"> </span><span class="op">!!</span><span class="kw">sym</span>(otherSplType) <span class="op">&gt;</span><span class="st"> </span>thdSize), <span class="ot">TRUE</span>, <span class="ot">FALSE</span>))</span>
<span id="cb2539-21"><a href="dsi-allCells-allSetsTop.html#cb2539-21"></a></span>
<span id="cb2539-22"><a href="dsi-allCells-allSetsTop.html#cb2539-22"></a><span class="kw">colData</span>(mnn.out) &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out) <span class="op">%&gt;%</span></span>
<span id="cb2539-23"><a href="dsi-allCells-allSetsTop.html#cb2539-23"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2539-24"><a href="dsi-allCells-allSetsTop.html#cb2539-24"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">cluster=</span>clusters.mnn) <span class="op">%&gt;%</span></span>
<span id="cb2539-25"><a href="dsi-allCells-allSetsTop.html#cb2539-25"></a><span class="st">  </span><span class="co">#select(-tmpCluBool) %&gt;%</span></span>
<span id="cb2539-26"><a href="dsi-allCells-allSetsTop.html#cb2539-26"></a><span class="st">  </span><span class="kw">left_join</span>(normNoLog[,<span class="kw">c</span>(<span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;tmpCluBool&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;cluster&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb2539-27"><a href="dsi-allCells-allSetsTop.html#cb2539-27"></a><span class="st">  </span><span class="kw">DataFrame</span>()</span>
<span id="cb2539-28"><a href="dsi-allCells-allSetsTop.html#cb2539-28"></a></span>
<span id="cb2539-29"><a href="dsi-allCells-allSetsTop.html#cb2539-29"></a>norm2 &lt;-<span class="st"> </span>normNoLog <span class="op">%&gt;%</span></span>
<span id="cb2539-30"><a href="dsi-allCells-allSetsTop.html#cb2539-30"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2539-31"><a href="dsi-allCells-allSetsTop.html#cb2539-31"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>sum, <span class="op">-</span>prop) <span class="op">%&gt;%</span></span>
<span id="cb2539-32"><a href="dsi-allCells-allSetsTop.html#cb2539-32"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span><span class="kw">c</span>(<span class="kw">grep</span>(<span class="st">&quot;propOut&quot;</span>, <span class="kw">colnames</span>(normNoLog), <span class="dt">value=</span><span class="ot">TRUE</span>))) <span class="op">%&gt;%</span></span>
<span id="cb2539-33"><a href="dsi-allCells-allSetsTop.html#cb2539-33"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span><span class="kw">c</span>(<span class="kw">grep</span>(<span class="st">&quot;propLt&quot;</span>, <span class="kw">colnames</span>(normNoLog), <span class="dt">value=</span><span class="ot">TRUE</span>))) <span class="op">%&gt;%</span></span>
<span id="cb2539-34"><a href="dsi-allCells-allSetsTop.html#cb2539-34"></a><span class="st">  </span><span class="co">#tibble::rownames_to_column(&quot;clusters&quot;) %&gt;% </span></span>
<span id="cb2539-35"><a href="dsi-allCells-allSetsTop.html#cb2539-35"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_longer</span>(<span class="op">!</span><span class="kw">c</span>(cluster,</span>
<span id="cb2539-36"><a href="dsi-allCells-allSetsTop.html#cb2539-36"></a>                         <span class="kw">grep</span>(<span class="st">&quot;tmpCluBool&quot;</span>, <span class="kw">colnames</span>(normNoLog), <span class="dt">value=</span><span class="ot">TRUE</span>)</span>
<span id="cb2539-37"><a href="dsi-allCells-allSetsTop.html#cb2539-37"></a>                         ),</span>
<span id="cb2539-38"><a href="dsi-allCells-allSetsTop.html#cb2539-38"></a>                      <span class="dt">names_to=</span><span class="st">&quot;source_name&quot;</span>,</span>
<span id="cb2539-39"><a href="dsi-allCells-allSetsTop.html#cb2539-39"></a>                      <span class="dt">values_to=</span><span class="st">&quot;Freq&quot;</span>)</span>
<span id="cb2539-40"><a href="dsi-allCells-allSetsTop.html#cb2539-40"></a></span>
<span id="cb2539-41"><a href="dsi-allCells-allSetsTop.html#cb2539-41"></a>norm2<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">factor</span>(norm2<span class="op">$</span>cluster,</span>
<span id="cb2539-42"><a href="dsi-allCells-allSetsTop.html#cb2539-42"></a>                        <span class="dt">levels=</span><span class="kw">paste0</span>(<span class="st">&quot;c&quot;</span>, sortVecNames))</span>
<span id="cb2539-43"><a href="dsi-allCells-allSetsTop.html#cb2539-43"></a></span>
<span id="cb2539-44"><a href="dsi-allCells-allSetsTop.html#cb2539-44"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data=</span>norm2, <span class="kw">aes</span>(<span class="dt">x=</span>cluster,<span class="dt">y=</span>Freq, <span class="dt">fill=</span>source_name)) <span class="op">+</span></span>
<span id="cb2539-45"><a href="dsi-allCells-allSetsTop.html#cb2539-45"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb2539-46"><a href="dsi-allCells-allSetsTop.html#cb2539-46"></a><span class="st">    </span><span class="kw">geom_col</span>()</span>
<span id="cb2539-47"><a href="dsi-allCells-allSetsTop.html#cb2539-47"></a><span class="kw">grid.arrange</span>(p, p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(norm2<span class="op">$</span>tmpCluBool))</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/ordMerge_idPbmmcClu_select_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2540-1"><a href="dsi-allCells-allSetsTop.html#cb2540-1"></a><span class="kw">rm</span>(p)</span></code></pre></div>
<div class="sourceCode" id="cb2541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2541-1"><a href="dsi-allCells-allSetsTop.html#cb2541-1"></a><span class="kw">colData</span>(mnn.out) &lt;-<span class="st"> </span><span class="kw">colData</span>(mnn.out) <span class="op">%&gt;%</span></span>
<span id="cb2541-2"><a href="dsi-allCells-allSetsTop.html#cb2541-2"></a><span class="st">  </span><span class="kw">data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb2541-3"><a href="dsi-allCells-allSetsTop.html#cb2541-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">cluster=</span>clusters.mnn) <span class="op">%&gt;%</span></span>
<span id="cb2541-4"><a href="dsi-allCells-allSetsTop.html#cb2541-4"></a><span class="st">  </span><span class="co">#select(-tmpCluBool) %&gt;%</span></span>
<span id="cb2541-5"><a href="dsi-allCells-allSetsTop.html#cb2541-5"></a><span class="st">  </span><span class="kw">left_join</span>(normNoLog[,<span class="kw">c</span>(<span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;prop&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;cluster&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb2541-6"><a href="dsi-allCells-allSetsTop.html#cb2541-6"></a><span class="st">  </span><span class="kw">DataFrame</span>()</span>
<span id="cb2541-7"><a href="dsi-allCells-allSetsTop.html#cb2541-7"></a></span>
<span id="cb2541-8"><a href="dsi-allCells-allSetsTop.html#cb2541-8"></a><span class="co"># cluster:</span></span>
<span id="cb2541-9"><a href="dsi-allCells-allSetsTop.html#cb2541-9"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb2541-10"><a href="dsi-allCells-allSetsTop.html#cb2541-10"></a>              <span class="dt">colour_by=</span><span class="st">&quot;prop&quot;</span>,</span>
<span id="cb2541-11"><a href="dsi-allCells-allSetsTop.html#cb2541-11"></a>              <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb2541-12"><a href="dsi-allCells-allSetsTop.html#cb2541-12"></a>              <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2541-13"><a href="dsi-allCells-allSetsTop.html#cb2541-13"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;PBMMC cell proportion&quot;</span>)</span>
<span id="cb2541-14"><a href="dsi-allCells-allSetsTop.html#cb2541-14"></a>p1 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>tmpCluBool) <span class="op">+</span></span>
<span id="cb2541-15"><a href="dsi-allCells-allSetsTop.html#cb2541-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&#39;bottom&#39;</span>)</span>
<span id="cb2541-16"><a href="dsi-allCells-allSetsTop.html#cb2541-16"></a></span>
<span id="cb2541-17"><a href="dsi-allCells-allSetsTop.html#cb2541-17"></a><span class="co"># show</span></span>
<span id="cb2541-18"><a href="dsi-allCells-allSetsTop.html#cb2541-18"></a><span class="co">#p1</span></span>
<span id="cb2541-19"><a href="dsi-allCells-allSetsTop.html#cb2541-19"></a>p</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotPropPbmmcTsneProp_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2542-1"><a href="dsi-allCells-allSetsTop.html#cb2542-1"></a><span class="kw">rm</span>(p, p1)</span>
<span id="cb2542-2"><a href="dsi-allCells-allSetsTop.html#cb2542-2"></a><span class="co">#mnn.out$tmpCluBool &lt;- NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb2543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2543-1"><a href="dsi-allCells-allSetsTop.html#cb2543-1"></a><span class="co"># cluster:</span></span>
<span id="cb2543-2"><a href="dsi-allCells-allSetsTop.html#cb2543-2"></a>p &lt;-<span class="st"> </span><span class="kw">plotTSNE</span>(mnn.out,</span>
<span id="cb2543-3"><a href="dsi-allCells-allSetsTop.html#cb2543-3"></a>              <span class="dt">colour_by=</span><span class="st">&quot;cluster&quot;</span>,</span>
<span id="cb2543-4"><a href="dsi-allCells-allSetsTop.html#cb2543-4"></a>              <span class="dt">by_exprs_values=</span><span class="st">&quot;reconstructed&quot;</span>,</span>
<span id="cb2543-5"><a href="dsi-allCells-allSetsTop.html#cb2543-5"></a>              <span class="dt">point_size=</span><span class="fl">0.3</span>)</span>
<span id="cb2543-6"><a href="dsi-allCells-allSetsTop.html#cb2543-6"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;clusters&quot;</span>)</span>
<span id="cb2543-7"><a href="dsi-allCells-allSetsTop.html#cb2543-7"></a>p1 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>mnn.out<span class="op">$</span>tmpCluBool) <span class="op">+</span></span>
<span id="cb2543-8"><a href="dsi-allCells-allSetsTop.html#cb2543-8"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&#39;bottom&#39;</span>)</span>
<span id="cb2543-9"><a href="dsi-allCells-allSetsTop.html#cb2543-9"></a></span>
<span id="cb2543-10"><a href="dsi-allCells-allSetsTop.html#cb2543-10"></a><span class="co"># show</span></span>
<span id="cb2543-11"><a href="dsi-allCells-allSetsTop.html#cb2543-11"></a>p1</span></code></pre></div>
<p><img src="dataSetIntegration_expand_allSets_files/figure-html/idPbmmcClu_barPlotPropPbmmcTsneSelec_dsi_allCells_allSets-1.png" width="672" /></p>
<div class="sourceCode" id="cb2544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2544-1"><a href="dsi-allCells-allSetsTop.html#cb2544-1"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb2544-2"><a href="dsi-allCells-allSetsTop.html#cb2544-2"></a><span class="co"># save object?</span></span>
<span id="cb2544-3"><a href="dsi-allCells-allSetsTop.html#cb2544-3"></a><span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb2544-4"><a href="dsi-allCells-allSetsTop.html#cb2544-4"></a>fn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dsi_%s_normNoLog.Rds&quot;</span>,</span>
<span id="cb2544-5"><a href="dsi-allCells-allSetsTop.html#cb2544-5"></a>              projDir,</span>
<span id="cb2544-6"><a href="dsi-allCells-allSetsTop.html#cb2544-6"></a>              outDirBit,</span>
<span id="cb2544-7"><a href="dsi-allCells-allSetsTop.html#cb2544-7"></a>              setName,</span>
<span id="cb2544-8"><a href="dsi-allCells-allSetsTop.html#cb2544-8"></a>              setSuf,</span>
<span id="cb2544-9"><a href="dsi-allCells-allSetsTop.html#cb2544-9"></a>              splSetToGet2)</span>
<span id="cb2544-10"><a href="dsi-allCells-allSetsTop.html#cb2544-10"></a><span class="kw">saveRDS</span>(normNoLog, <span class="dt">file=</span>fn)</span></code></pre></div>
</div>
<div id="session-information-24" class="section level2">
<h2><span class="header-section-number">29.10</span> Session information</h2>
<details>
<div class="sourceCode" id="cb2545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2545-1"><a href="dsi-allCells-allSetsTop.html#cb2545-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] gridExtra_2.3               cowplot_1.1.1              
##  [3] Cairo_1.5-12.2              clustree_0.4.3             
##  [5] ggraph_2.0.5                pheatmap_1.0.12            
##  [7] forcats_0.5.1               stringr_1.4.0              
##  [9] dplyr_1.0.6                 purrr_0.3.4                
## [11] readr_1.4.0                 tidyr_1.1.3                
## [13] tibble_3.1.2                tidyverse_1.3.1            
## [15] bluster_1.0.0               batchelor_1.6.3            
## [17] BiocNeighbors_1.8.2         BiocSingular_1.6.0         
## [19] scran_1.18.7                scater_1.18.6              
## [21] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
## [23] Biobase_2.50.0              GenomicRanges_1.42.0       
## [25] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [27] S4Vectors_0.28.1            BiocGenerics_0.36.1        
## [29] MatrixGenerics_1.2.1        matrixStats_0.58.0         
## [31] ggplot2_3.3.3               BiocParallel_1.24.1        
## [33] knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##  [1] Rtsne_0.15                ggbeeswarm_0.6.0         
##  [3] colorspace_2.0-1          ellipsis_0.3.2           
##  [5] scuttle_1.0.4             XVector_0.30.0           
##  [7] fs_1.5.0                  rstudioapi_0.13          
##  [9] farver_2.1.0              graphlayouts_0.7.1       
## [11] ggrepel_0.9.1             RSpectra_0.16-0          
## [13] fansi_0.4.2               lubridate_1.7.10         
## [15] xml2_1.3.2                codetools_0.2-18         
## [17] sparseMatrixStats_1.2.1   polyclip_1.10-0          
## [19] jsonlite_1.7.2            ResidualMatrix_1.0.0     
## [21] broom_0.7.6               dbplyr_2.1.1             
## [23] uwot_0.1.10               ggforce_0.3.3            
## [25] compiler_4.0.3            httr_1.4.2               
## [27] dqrng_0.3.0               backports_1.2.1          
## [29] assertthat_0.2.1          Matrix_1.3-3             
## [31] limma_3.46.0              cli_2.5.0                
## [33] tweenr_1.0.2              htmltools_0.5.1.1        
## [35] tools_4.0.3               rsvd_1.0.5               
## [37] igraph_1.2.6              gtable_0.3.0             
## [39] glue_1.4.2                GenomeInfoDbData_1.2.4   
## [41] Rcpp_1.0.6                cellranger_1.1.0         
## [43] jquerylib_0.1.4           vctrs_0.3.8              
## [45] DelayedMatrixStats_1.12.3 xfun_0.23                
## [47] ps_1.6.0                  beachmat_2.6.4           
## [49] rvest_1.0.0               lifecycle_1.0.0          
## [51] irlba_2.3.3               statmod_1.4.36           
## [53] edgeR_3.32.1              zlibbioc_1.36.0          
## [55] MASS_7.3-54               scales_1.1.1             
## [57] tidygraph_1.2.0           hms_1.0.0                
## [59] RColorBrewer_1.1-2        yaml_2.2.1               
## [61] sass_0.4.0                stringi_1.6.1            
## [63] highr_0.9                 checkmate_2.0.0          
## [65] rlang_0.4.11              pkgconfig_2.0.3          
## [67] bitops_1.0-7              evaluate_0.14            
## [69] lattice_0.20-44           labeling_0.4.2           
## [71] tidyselect_1.1.1          magrittr_2.0.1           
## [73] bookdown_0.22             R6_2.5.0                 
## [75] generics_0.1.0            DelayedArray_0.16.3      
## [77] DBI_1.1.1                 pillar_1.6.1             
## [79] haven_2.4.1               withr_2.4.2              
## [81] RCurl_1.98-1.3            modelr_0.1.8             
## [83] crayon_1.4.1              utf8_1.2.1               
## [85] rmarkdown_2.8             viridis_0.6.1            
## [87] locfit_1.5-9.4            grid_4.0.3               
## [89] readxl_1.3.1              reprex_2.0.0             
## [91] digest_0.6.27             munsell_0.5.0            
## [93] beeswarm_0.3.1            viridisLite_0.4.0        
## [95] vipor_0.4.5               bslib_0.2.5</code></pre>
</details>

<div class="sourceCode" id="cb2547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2547-1"><a href="dsi-allCells-allSetsTop.html#cb2547-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb2547-2"><a href="dsi-allCells-allSetsTop.html#cb2547-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb2547-3"><a href="dsi-allCells-allSetsTop.html#cb2547-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb2547-4"><a href="dsi-allCells-allSetsTop.html#cb2547-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb2547-5"><a href="dsi-allCells-allSetsTop.html#cb2547-5"></a>splSetToGet &lt;-<span class="st"> </span>params<span class="op">$</span>splSetToGet</span>
<span id="cb2547-6"><a href="dsi-allCells-allSetsTop.html#cb2547-6"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb2547-7"><a href="dsi-allCells-allSetsTop.html#cb2547-7"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb2547-8"><a href="dsi-allCells-allSetsTop.html#cb2547-8"></a>dsiSuf &lt;-<span class="st"> </span>params<span class="op">$</span>dsiSuf <span class="co"># &#39;dsi&#39; for data set integration</span></span>
<span id="cb2547-9"><a href="dsi-allCells-allSetsTop.html#cb2547-9"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb2547-10"><a href="dsi-allCells-allSetsTop.html#cb2547-10"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb2547-11"><a href="dsi-allCells-allSetsTop.html#cb2547-11"></a>    splSetToGet &lt;-<span class="st"> &quot;PBMMC,ETV6-RUNX1&quot;</span></span>
<span id="cb2547-12"><a href="dsi-allCells-allSetsTop.html#cb2547-12"></a>    setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb2547-13"><a href="dsi-allCells-allSetsTop.html#cb2547-13"></a>    dsiSuf &lt;-<span class="st"> &#39;_dsi&#39;</span></span>
<span id="cb2547-14"><a href="dsi-allCells-allSetsTop.html#cb2547-14"></a>}</span>
<span id="cb2547-15"><a href="dsi-allCells-allSetsTop.html#cb2547-15"></a></span>
<span id="cb2547-16"><a href="dsi-allCells-allSetsTop.html#cb2547-16"></a>splSetVec &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>(splSetToGet, <span class="st">&quot;,&quot;</span>)) <span class="co"># params may not be read in if knitting book.</span></span>
<span id="cb2547-17"><a href="dsi-allCells-allSetsTop.html#cb2547-17"></a>splSetToGet2 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;_&quot;</span>, splSetToGet)</span>
<span id="cb2547-18"><a href="dsi-allCells-allSetsTop.html#cb2547-18"></a>nbPcToComp &lt;-<span class="st"> </span><span class="dv">50</span></span>
<span id="cb2547-19"><a href="dsi-allCells-allSetsTop.html#cb2547-19"></a>figSize &lt;-<span class="st"> </span><span class="dv">7</span></span></code></pre></div>
<div class="sourceCode" id="cb2548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2548-1"><a href="dsi-allCells-allSetsTop.html#cb2548-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb2548-2"><a href="dsi-allCells-allSetsTop.html#cb2548-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb2548-3"><a href="dsi-allCells-allSetsTop.html#cb2548-3"></a><span class="kw">library</span>(scran)</span>
<span id="cb2548-4"><a href="dsi-allCells-allSetsTop.html#cb2548-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb2548-5"><a href="dsi-allCells-allSetsTop.html#cb2548-5"></a><span class="kw">library</span>(Cairo)</span>
<span id="cb2548-6"><a href="dsi-allCells-allSetsTop.html#cb2548-6"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>), <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dsi-allCells-PBMMC-ETV6-RUNX1Top.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cellCyclePhasesTop.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/dataSetIntegration_expand_allSets.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["KmWiC2.pdf", "KmWiC2.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
