<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Feature selection | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</title>
  <meta name="description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Feature selection | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A report for all analyses (bookdown::gitbook)." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Feature selection | CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis" />
  
  <meta name="twitter:description" content="A report for all analyses (bookdown::gitbook)." />
  

<meta name="author" content="Stephane Ballereau, Zeynep Kalender Atak" />


<meta name="date" content="2021-04-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dimensionality-reduction-for-visualisation.html"/>
<link rel="next" href="batch-correction-gsm3872442-set.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CRUK Functional Genomics Summer School 2020</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-study"><i class="fa fa-check"></i><b>1.1</b> The study</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-plan"><i class="fa fa-check"></i><b>1.2</b> The plan</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#the-analyses"><i class="fa fa-check"></i><b>1.3</b> The analyses</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#overall-summary"><i class="fa fa-check"></i><b>1.4</b> Overall summary</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i><b>1.6</b> Abbreviations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="SeqQualTop.html"><a href="SeqQualTop.html"><i class="fa fa-check"></i><b>2</b> Sequence Quality</a><ul>
<li class="chapter" data-level="2.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---fastqc"><i class="fa fa-check"></i><b>2.2</b> CaronBourque2020 - fastqc</a></li>
<li class="chapter" data-level="2.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#caronbourque2020---multiqc"><i class="fa fa-check"></i><b>2.3</b> CaronBourque2020 - MultiQC</a><ul>
<li class="chapter" data-level="2.3.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1"><i class="fa fa-check"></i><b>2.3.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.3.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1"><i class="fa fa-check"></i><b>2.3.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.3.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2"><i class="fa fa-check"></i><b>2.3.3</b> insert: R2</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---fastqc"><i class="fa fa-check"></i><b>2.4</b> HCA adult BMMC - fastqc</a></li>
<li class="chapter" data-level="2.5" data-path="SeqQualTop.html"><a href="SeqQualTop.html#hca-adult-bmmc---multiqc"><i class="fa fa-check"></i><b>2.5</b> HCA adult BMMC - MultiQC</a><ul>
<li class="chapter" data-level="2.5.1" data-path="SeqQualTop.html"><a href="SeqQualTop.html#sample-index-i1-1"><i class="fa fa-check"></i><b>2.5.1</b> sample index: I1</a></li>
<li class="chapter" data-level="2.5.2" data-path="SeqQualTop.html"><a href="SeqQualTop.html#cell-barcode-umi-r1-1"><i class="fa fa-check"></i><b>2.5.2</b> cell barcode + UMI: R1</a></li>
<li class="chapter" data-level="2.5.3" data-path="SeqQualTop.html"><a href="SeqQualTop.html#insert-r2-1"><i class="fa fa-check"></i><b>2.5.3</b> insert: R2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html"><i class="fa fa-check"></i><b>3</b> Alignment and feature counting</a><ul>
<li class="chapter" data-level="3.1" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-pipeline-in-brief"><i class="fa fa-check"></i><b>3.2</b> 10X cellranger pipeline in brief</a></li>
<li class="chapter" data-level="3.3" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#sample-sheet"><i class="fa fa-check"></i><b>3.3</b> sample sheet</a></li>
<li class="chapter" data-level="3.4" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-caronbourque2020"><i class="fa fa-check"></i><b>3.4</b> 10X cellranger reports for CaronBourque2020</a></li>
<li class="chapter" data-level="3.5" data-path="AliFeatCountTop.html"><a href="AliFeatCountTop.html#x-cellranger-reports-for-hcas-adult-bmmcs"><i class="fa fa-check"></i><b>3.5</b> 10X cellranger reports for HCA’s adult BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#load-packages"><i class="fa fa-check"></i><b>4.2</b> Load packages</a></li>
<li class="chapter" data-level="4.3" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#sample-sheet-1"><i class="fa fa-check"></i><b>4.3</b> Sample sheet</a></li>
<li class="chapter" data-level="4.4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#data-representation"><i class="fa fa-check"></i><b>4.4</b> Data representation</a></li>
<li class="chapter" data-level="4.5" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#example"><i class="fa fa-check"></i><b>4.5</b> Example</a></li>
<li class="chapter" data-level="4.6" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#mapping-qc"><i class="fa fa-check"></i><b>4.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="4.6.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#gene-body-coverage"><i class="fa fa-check"></i><b>4.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="4.6.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#amplification-rate"><i class="fa fa-check"></i><b>4.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#cell-calling-for-droplet-data"><i class="fa fa-check"></i><b>4.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="4.7.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#mixture-model"><i class="fa fa-check"></i><b>4.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="4.7.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#barcode-rank-plot"><i class="fa fa-check"></i><b>4.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="4.7.3" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#inflection-point"><i class="fa fa-check"></i><b>4.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="4.7.4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#cellranger-v1-and-v2"><i class="fa fa-check"></i><b>4.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="4.7.5" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#dropletutils-and-emptydrops"><i class="fa fa-check"></i><b>4.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#load-filtered-matrices"><i class="fa fa-check"></i><b>4.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="4.9" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#properties-of-scrna-seq-data"><i class="fa fa-check"></i><b>4.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="4.10" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#quality-control"><i class="fa fa-check"></i><b>4.10</b> Quality control</a><ul>
<li class="chapter" data-level="4.10.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#qc-metric-distribution"><i class="fa fa-check"></i><b>4.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#identification-of-low-quality-cells-with-adaptive-thresholds"><i class="fa fa-check"></i><b>4.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="4.11.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#library-size"><i class="fa fa-check"></i><b>4.11.1</b> Library size</a></li>
<li class="chapter" data-level="4.11.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#number-of-genes"><i class="fa fa-check"></i><b>4.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="4.11.3" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.11.4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#summary"><i class="fa fa-check"></i><b>4.11.4</b> Summary</a></li>
<li class="chapter" data-level="4.11.5" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#all-steps-at-once"><i class="fa fa-check"></i><b>4.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="4.11.6" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#assumptions"><i class="fa fa-check"></i><b>4.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="4.12" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#experimental-factors"><i class="fa fa-check"></i><b>4.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="4.12.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#identify-poor-quality-batches"><i class="fa fa-check"></i><b>4.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="4.12.1.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#number-of-genes-detected"><i class="fa fa-check"></i><b>4.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="4.12.1.1.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#mitochondrial-content-1"><i class="fa fa-check"></i><b>4.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="4.12.1.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#samples-to-check"><i class="fa fa-check"></i><b>4.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="4.12.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#qc-metrics-space"><i class="fa fa-check"></i><b>4.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="4.12.3" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#qc-pca"><i class="fa fa-check"></i><b>4.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="4.12.4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#other-diagnostic-plots"><i class="fa fa-check"></i><b>4.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="4.12.5" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#filter-low-quality-cells-out"><i class="fa fa-check"></i><b>4.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#novelty"><i class="fa fa-check"></i><b>4.13</b> Novelty</a></li>
<li class="chapter" data-level="4.14" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#qc-based-on-sparsity"><i class="fa fa-check"></i><b>4.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="4.14.1" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#remove-genes-that-are-not-expressed-at-all"><i class="fa fa-check"></i><b>4.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="4.14.2" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#sparsity-plots"><i class="fa fa-check"></i><b>4.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="4.14.3" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#filters"><i class="fa fa-check"></i><b>4.14.3</b> Filters</a></li>
<li class="chapter" data-level="4.14.4" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#separate-caron-and-hca-batches"><i class="fa fa-check"></i><b>4.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="4.14.5" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#caron-only"><i class="fa fa-check"></i><b>4.14.5</b> Caron only</a></li>
<li class="chapter" data-level="4.14.6" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#hca-only"><i class="fa fa-check"></i><b>4.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="4.15" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#subsample-hca-set"><i class="fa fa-check"></i><b>4.15</b> Subsample Hca set</a></li>
<li class="chapter" data-level="4.16" data-path="QcTopAllCells.html"><a href="QcTopAllCells.html#session-information"><i class="fa fa-check"></i><b>4.16</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="QcTop.html"><a href="QcTop.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a><ul>
<li class="chapter" data-level="5.1" data-path="QcTop.html"><a href="QcTop.html#introduction-3"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="QcTop.html"><a href="QcTop.html#load-packages-1"><i class="fa fa-check"></i><b>5.2</b> Load packages</a></li>
<li class="chapter" data-level="5.3" data-path="QcTop.html"><a href="QcTop.html#sample-sheet-2"><i class="fa fa-check"></i><b>5.3</b> Sample sheet</a></li>
<li class="chapter" data-level="5.4" data-path="QcTop.html"><a href="QcTop.html#data-representation-1"><i class="fa fa-check"></i><b>5.4</b> Data representation</a></li>
<li class="chapter" data-level="5.5" data-path="QcTop.html"><a href="QcTop.html#example-1"><i class="fa fa-check"></i><b>5.5</b> Example</a></li>
<li class="chapter" data-level="5.6" data-path="QcTop.html"><a href="QcTop.html#mapping-qc-1"><i class="fa fa-check"></i><b>5.6</b> Mapping QC</a><ul>
<li class="chapter" data-level="5.6.1" data-path="QcTop.html"><a href="QcTop.html#gene-body-coverage-1"><i class="fa fa-check"></i><b>5.6.1</b> Gene body coverage</a></li>
<li class="chapter" data-level="5.6.2" data-path="QcTop.html"><a href="QcTop.html#amplification-rate-1"><i class="fa fa-check"></i><b>5.6.2</b> Amplification rate</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="QcTop.html"><a href="QcTop.html#cell-calling-for-droplet-data-1"><i class="fa fa-check"></i><b>5.7</b> Cell calling for droplet data</a><ul>
<li class="chapter" data-level="5.7.1" data-path="QcTop.html"><a href="QcTop.html#mixture-model-1"><i class="fa fa-check"></i><b>5.7.1</b> Mixture model</a></li>
<li class="chapter" data-level="5.7.2" data-path="QcTop.html"><a href="QcTop.html#barcode-rank-plot-1"><i class="fa fa-check"></i><b>5.7.2</b> Barcode rank plot</a></li>
<li class="chapter" data-level="5.7.3" data-path="QcTop.html"><a href="QcTop.html#inflection-point-1"><i class="fa fa-check"></i><b>5.7.3</b> Inflection point</a></li>
<li class="chapter" data-level="5.7.4" data-path="QcTop.html"><a href="QcTop.html#cellranger-v1-and-v2-1"><i class="fa fa-check"></i><b>5.7.4</b> Cellranger v1 and v2</a></li>
<li class="chapter" data-level="5.7.5" data-path="QcTop.html"><a href="QcTop.html#dropletutils-and-emptydrops-1"><i class="fa fa-check"></i><b>5.7.5</b> DropletUtils and EmptyDrops</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="QcTop.html"><a href="QcTop.html#load-filtered-matrices-1"><i class="fa fa-check"></i><b>5.8</b> Load filtered matrices</a></li>
<li class="chapter" data-level="5.9" data-path="QcTop.html"><a href="QcTop.html#properties-of-scrna-seq-data-1"><i class="fa fa-check"></i><b>5.9</b> Properties of scRNA-seq data</a></li>
<li class="chapter" data-level="5.10" data-path="QcTop.html"><a href="QcTop.html#quality-control-1"><i class="fa fa-check"></i><b>5.10</b> Quality control</a><ul>
<li class="chapter" data-level="5.10.1" data-path="QcTop.html"><a href="QcTop.html#qc-metric-distribution-1"><i class="fa fa-check"></i><b>5.10.1</b> QC metric distribution</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="QcTop.html"><a href="QcTop.html#identification-of-low-quality-cells-with-adaptive-thresholds-1"><i class="fa fa-check"></i><b>5.11</b> Identification of low-quality cells with adaptive thresholds</a><ul>
<li class="chapter" data-level="5.11.1" data-path="QcTop.html"><a href="QcTop.html#library-size-1"><i class="fa fa-check"></i><b>5.11.1</b> Library size</a></li>
<li class="chapter" data-level="5.11.2" data-path="QcTop.html"><a href="QcTop.html#number-of-genes-1"><i class="fa fa-check"></i><b>5.11.2</b> Number of genes</a></li>
<li class="chapter" data-level="5.11.3" data-path="QcTop.html"><a href="QcTop.html#mitochondrial-content-2"><i class="fa fa-check"></i><b>5.11.3</b> Mitochondrial content</a></li>
<li class="chapter" data-level="5.11.4" data-path="QcTop.html"><a href="QcTop.html#summary-1"><i class="fa fa-check"></i><b>5.11.4</b> Summary</a></li>
<li class="chapter" data-level="5.11.5" data-path="QcTop.html"><a href="QcTop.html#all-steps-at-once-1"><i class="fa fa-check"></i><b>5.11.5</b> All steps at once</a></li>
<li class="chapter" data-level="5.11.6" data-path="QcTop.html"><a href="QcTop.html#assumptions-1"><i class="fa fa-check"></i><b>5.11.6</b> Assumptions</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="QcTop.html"><a href="QcTop.html#experimental-factors-1"><i class="fa fa-check"></i><b>5.12</b> Experimental factors</a><ul>
<li class="chapter" data-level="5.12.1" data-path="QcTop.html"><a href="QcTop.html#identify-poor-quality-batches-1"><i class="fa fa-check"></i><b>5.12.1</b> Identify poor-quality batches</a><ul>
<li class="chapter" data-level="5.12.1.1" data-path="QcTop.html"><a href="QcTop.html#number-of-genes-detected-1"><i class="fa fa-check"></i><b>5.12.1.1</b> Number of genes detected</a><ul>
<li class="chapter" data-level="5.12.1.1.1" data-path="QcTop.html"><a href="QcTop.html#mitochondrial-content-3"><i class="fa fa-check"></i><b>5.12.1.1.1</b> Mitochondrial content</a></li>
</ul></li>
<li class="chapter" data-level="5.12.1.2" data-path="QcTop.html"><a href="QcTop.html#samples-to-check-1"><i class="fa fa-check"></i><b>5.12.1.2</b> Samples to check</a></li>
</ul></li>
<li class="chapter" data-level="5.12.2" data-path="QcTop.html"><a href="QcTop.html#qc-metrics-space-1"><i class="fa fa-check"></i><b>5.12.2</b> QC metrics space</a></li>
<li class="chapter" data-level="5.12.3" data-path="QcTop.html"><a href="QcTop.html#qc-pca-1"><i class="fa fa-check"></i><b>5.12.3</b> QC PCA</a></li>
<li class="chapter" data-level="5.12.4" data-path="QcTop.html"><a href="QcTop.html#other-diagnostic-plots-1"><i class="fa fa-check"></i><b>5.12.4</b> Other diagnostic plots</a></li>
<li class="chapter" data-level="5.12.5" data-path="QcTop.html"><a href="QcTop.html#filter-low-quality-cells-out-1"><i class="fa fa-check"></i><b>5.12.5</b> Filter low-quality cells out</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="QcTop.html"><a href="QcTop.html#novelty-1"><i class="fa fa-check"></i><b>5.13</b> Novelty</a></li>
<li class="chapter" data-level="5.14" data-path="QcTop.html"><a href="QcTop.html#qc-based-on-sparsity-1"><i class="fa fa-check"></i><b>5.14</b> QC based on sparsity</a><ul>
<li class="chapter" data-level="5.14.1" data-path="QcTop.html"><a href="QcTop.html#remove-genes-that-are-not-expressed-at-all-1"><i class="fa fa-check"></i><b>5.14.1</b> Remove genes that are not expressed at all</a></li>
<li class="chapter" data-level="5.14.2" data-path="QcTop.html"><a href="QcTop.html#sparsity-plots-1"><i class="fa fa-check"></i><b>5.14.2</b> Sparsity plots</a></li>
<li class="chapter" data-level="5.14.3" data-path="QcTop.html"><a href="QcTop.html#filters-1"><i class="fa fa-check"></i><b>5.14.3</b> Filters</a></li>
<li class="chapter" data-level="5.14.4" data-path="QcTop.html"><a href="QcTop.html#separate-caron-and-hca-batches-1"><i class="fa fa-check"></i><b>5.14.4</b> Separate Caron and Hca batches</a></li>
<li class="chapter" data-level="5.14.5" data-path="QcTop.html"><a href="QcTop.html#caron-only-1"><i class="fa fa-check"></i><b>5.14.5</b> Caron only</a></li>
<li class="chapter" data-level="5.14.6" data-path="QcTop.html"><a href="QcTop.html#hca-only-1"><i class="fa fa-check"></i><b>5.14.6</b> Hca only</a></li>
</ul></li>
<li class="chapter" data-level="5.15" data-path="QcTop.html"><a href="QcTop.html#session-information-1"><i class="fa fa-check"></i><b>5.15</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html"><i class="fa fa-check"></i><b>6</b> Normalisation</a><ul>
<li class="chapter" data-level="6.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#caron"><i class="fa fa-check"></i><b>6.1</b> Caron</a><ul>
<li class="chapter" data-level="6.1.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#library-size-normalization"><i class="fa fa-check"></i><b>6.1.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.1.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#deconvolution"><i class="fa fa-check"></i><b>6.1.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.1.3" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#compute-size-factors"><i class="fa fa-check"></i><b>6.1.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.1.3.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#apply-size-factors"><i class="fa fa-check"></i><b>6.1.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.1.3.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#save-object"><i class="fa fa-check"></i><b>6.1.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#hca"><i class="fa fa-check"></i><b>6.2</b> Hca</a><ul>
<li class="chapter" data-level="6.2.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#library-size-normalization-1"><i class="fa fa-check"></i><b>6.2.1</b> Library size normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#deconvolution-1"><i class="fa fa-check"></i><b>6.2.2</b> Deconvolution</a></li>
<li class="chapter" data-level="6.2.3" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#compute-size-factors-1"><i class="fa fa-check"></i><b>6.2.3</b> Compute size factors</a><ul>
<li class="chapter" data-level="6.2.3.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#apply-size-factors-1"><i class="fa fa-check"></i><b>6.2.3.1</b> Apply size factors</a></li>
<li class="chapter" data-level="6.2.3.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#save-object-1"><i class="fa fa-check"></i><b>6.2.3.2</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#sctransform"><i class="fa fa-check"></i><b>6.3</b> SCTransform</a><ul>
<li class="chapter" data-level="6.3.1" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#caron-1"><i class="fa fa-check"></i><b>6.3.1</b> Caron</a></li>
<li class="chapter" data-level="6.3.2" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#hca-1"><i class="fa fa-check"></i><b>6.3.2</b> Hca</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="NormalisationAllCells.html"><a href="NormalisationAllCells.html#session-information-2"><i class="fa fa-check"></i><b>6.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html"><i class="fa fa-check"></i><b>7</b> Normalisation - GSM3872434 set</a><ul>
<li class="chapter" data-level="7.1" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#scaling-normalization"><i class="fa fa-check"></i><b>7.1</b> Scaling normalization</a><ul>
<li class="chapter" data-level="7.1.1" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#cpm"><i class="fa fa-check"></i><b>7.1.1</b> CPM</a></li>
<li class="chapter" data-level="7.1.2" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#deseqs-size-factor"><i class="fa fa-check"></i><b>7.1.2</b> DESeq’s size factor</a></li>
<li class="chapter" data-level="7.1.3" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#weighted-trimmed-mean-of-m-values"><i class="fa fa-check"></i><b>7.1.3</b> Weighted Trimmed mean of M-values</a></li>
<li class="chapter" data-level="7.1.4" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#library-size-factor-distribution"><i class="fa fa-check"></i><b>7.1.4</b> Library size factor distribution</a></li>
<li class="chapter" data-level="7.1.5" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#deconvolution-2"><i class="fa fa-check"></i><b>7.1.5</b> Deconvolution</a><ul>
<li class="chapter" data-level="7.1.5.1" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#cluster-cells"><i class="fa fa-check"></i><b>7.1.5.1</b> Cluster cells</a></li>
<li class="chapter" data-level="7.1.5.2" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#compute-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.2</b> Compute size factors</a></li>
<li class="chapter" data-level="7.1.5.3" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#apply-size-factors-2"><i class="fa fa-check"></i><b>7.1.5.3</b> Apply size factors</a></li>
<li class="chapter" data-level="7.1.5.4" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#save-object-2"><i class="fa fa-check"></i><b>7.1.5.4</b> Save object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#sctransform-1"><i class="fa fa-check"></i><b>7.2</b> SCTransform</a><ul>
<li class="chapter" data-level="7.2.1" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#inspect-data"><i class="fa fa-check"></i><b>7.2.1</b> Inspect data</a></li>
<li class="chapter" data-level="7.2.2" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#transformation"><i class="fa fa-check"></i><b>7.2.2</b> Transformation</a></li>
<li class="chapter" data-level="7.2.3" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#save-sce-object"><i class="fa fa-check"></i><b>7.2.3</b> Save SCE object</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#visualisation"><i class="fa fa-check"></i><b>7.3</b> Visualisation</a><ul>
<li class="chapter" data-level="7.3.1" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#log-raw-counts"><i class="fa fa-check"></i><b>7.3.1</b> log raw counts</a></li>
<li class="chapter" data-level="7.3.2" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#log-cpm"><i class="fa fa-check"></i><b>7.3.2</b> log CPM</a></li>
<li class="chapter" data-level="7.3.3" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#scran"><i class="fa fa-check"></i><b>7.3.3</b> scran</a></li>
<li class="chapter" data-level="7.3.4" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#sctransform-2"><i class="fa fa-check"></i><b>7.3.4</b> SCTransform</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="normalisation-gsm3872434-set.html"><a href="normalisation-gsm3872434-set.html#session-information-3"><i class="fa fa-check"></i><b>7.4</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html"><i class="fa fa-check"></i><b>8</b> Dimensionality reduction for visualisation</a><ul>
<li class="chapter" data-level="8.1" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a><ul>
<li class="chapter" data-level="8.1.1" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#description"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#example-2"><i class="fa fa-check"></i><b>8.1.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#load-packages-2"><i class="fa fa-check"></i><b>8.2</b> Load packages</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#load-data"><i class="fa fa-check"></i><b>8.3</b> Load data</a></li>
<li class="chapter" data-level="8.4" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#pca"><i class="fa fa-check"></i><b>8.4</b> PCA</a><ul>
<li class="chapter" data-level="8.4.1" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#correlation-between-pcs-and-the-total-number-of-features-detected"><i class="fa fa-check"></i><b>8.4.1</b> Correlation between PCs and the total number of features detected</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#t-sne-t-distributed-stochastic-neighbor-embedding"><i class="fa fa-check"></i><b>8.5</b> t-SNE: t-Distributed Stochastic Neighbor Embedding</a><ul>
<li class="chapter" data-level="8.5.1" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#perplexity"><i class="fa fa-check"></i><b>8.5.1</b> Perplexity</a></li>
<li class="chapter" data-level="8.5.2" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#stochasticity"><i class="fa fa-check"></i><b>8.5.2</b> Stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#umap"><i class="fa fa-check"></i><b>8.6</b> UMAP</a></li>
<li class="chapter" data-level="8.7" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#session-information-4"><i class="fa fa-check"></i><b>8.7</b> Session information</a></li>
<li class="chapter" data-level="8.8" data-path="dimensionality-reduction-for-visualisation.html"><a href="dimensionality-reduction-for-visualisation.html#session-information-5"><i class="fa fa-check"></i><b>8.8</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="feature-selection.html"><a href="feature-selection.html"><i class="fa fa-check"></i><b>9</b> Feature selection</a><ul>
<li class="chapter" data-level="9.1" data-path="feature-selection.html"><a href="feature-selection.html#load-data-1"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="feature-selection.html"><a href="feature-selection.html#feature-selection-with-scran"><i class="fa fa-check"></i><b>9.2</b> Feature selection with scran</a><ul>
<li class="chapter" data-level="9.2.1" data-path="feature-selection.html"><a href="feature-selection.html#modelling-and-removing-technical-noise"><i class="fa fa-check"></i><b>9.2.1</b> Modelling and removing technical noise</a></li>
<li class="chapter" data-level="9.2.2" data-path="feature-selection.html"><a href="feature-selection.html#choosing-some-hvgs"><i class="fa fa-check"></i><b>9.2.2</b> Choosing some HVGs:</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="feature-selection.html"><a href="feature-selection.html#session-information-6"><i class="fa fa-check"></i><b>9.3</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html"><i class="fa fa-check"></i><b>10</b> batch correction - GSM3872442 set</a><ul>
<li class="chapter" data-level="10.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#prepare-data"><i class="fa fa-check"></i><b>10.1</b> Prepare data</a></li>
<li class="chapter" data-level="10.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-each-separately-and-re-pool"><i class="fa fa-check"></i><b>10.2</b> Normalise each separately and re-pool</a></li>
<li class="chapter" data-level="10.3" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#normalise-batches-together"><i class="fa fa-check"></i><b>10.3</b> Normalise batches together</a></li>
<li class="chapter" data-level="10.4" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-correction"><i class="fa fa-check"></i><b>10.4</b> Batch correction</a><ul>
<li class="chapter" data-level="10.4.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#gaussian-normal-linear-models"><i class="fa fa-check"></i><b>10.4.1</b> Gaussian (normal) linear models</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#sctransform-3"><i class="fa fa-check"></i><b>10.5</b> SCTransform</a><ul>
<li class="chapter" data-level="10.5.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#batch-only"><i class="fa fa-check"></i><b>10.5.1</b> Batch only</a></li>
<li class="chapter" data-level="10.5.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#both-library-size-and-batch"><i class="fa fa-check"></i><b>10.5.2</b> Both library size and batch</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#mnncorrect"><i class="fa fa-check"></i><b>10.6</b> mnnCorrect</a><ul>
<li class="chapter" data-level="10.6.1" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#check-presence-of-batch-effect"><i class="fa fa-check"></i><b>10.6.1</b> Check presence of batch effect</a></li>
<li class="chapter" data-level="10.6.2" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#correct-batch-effect-with-mnncorrect"><i class="fa fa-check"></i><b>10.6.2</b> Correct batch effect with mnnCorrect</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#fastmnn"><i class="fa fa-check"></i><b>10.7</b> fastMNN</a></li>
<li class="chapter" data-level="10.8" data-path="batch-correction-gsm3872442-set.html"><a href="batch-correction-gsm3872442-set.html#harmony"><i class="fa fa-check"></i><b>10.8</b> Harmony</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html"><i class="fa fa-check"></i><b>11</b> Dimensionality reduction for analysis</a><ul>
<li class="chapter" data-level="11.1" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#load-data-2"><i class="fa fa-check"></i><b>11.1</b> Load data</a></li>
<li class="chapter" data-level="11.2" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#denoising-expression-values-using-pca"><i class="fa fa-check"></i><b>11.2</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="11.3" data-path="dimensionality-reduction-for-analysis.html"><a href="dimensionality-reduction-for-analysis.html#visualise-expression-patterns-of-some-hvgs"><i class="fa fa-check"></i><b>11.3</b> Visualise expression patterns of some HVGs</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html"><i class="fa fa-check"></i><b>12</b> Data integration</a><ul>
<li class="chapter" data-level="12.1" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#abbreviations-1"><i class="fa fa-check"></i><b>12.1</b> Abbreviations</a></li>
<li class="chapter" data-level="12.2" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#motivation"><i class="fa fa-check"></i><b>12.2</b> Motivation</a></li>
<li class="chapter" data-level="12.3" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#loading-the-data"><i class="fa fa-check"></i><b>12.3</b> Loading the data</a></li>
<li class="chapter" data-level="12.4" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#diagnosing-batch-effects"><i class="fa fa-check"></i><b>12.4</b> Diagnosing batch effects</a></li>
<li class="chapter" data-level="12.5" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#linear-regression"><i class="fa fa-check"></i><b>12.5</b> Linear regression</a></li>
<li class="chapter" data-level="12.6" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#performing-mnn-correction"><i class="fa fa-check"></i><b>12.6</b> Performing MNN correction</a><ul>
<li class="chapter" data-level="12.6.1" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#algorithm-overview"><i class="fa fa-check"></i><b>12.6.1</b> Algorithm overview</a></li>
<li class="chapter" data-level="12.6.2" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#application-to-the-data"><i class="fa fa-check"></i><b>12.6.2</b> Application to the data</a></li>
<li class="chapter" data-level="12.6.3" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#correction-diagnostics"><i class="fa fa-check"></i><b>12.6.3</b> Correction diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#preserving-biological-heterogeneity"><i class="fa fa-check"></i><b>12.7</b> Preserving biological heterogeneity</a><ul>
<li class="chapter" data-level="12.7.1" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#comparison-to-within-batch-clusters"><i class="fa fa-check"></i><b>12.7.1</b> Comparison to within-batch clusters</a></li>
<li class="chapter" data-level="12.7.2" data-path="DataIntegrationPbmmc.html"><a href="DataIntegrationPbmmc.html#encouraging-consistency-with-marker-genes"><i class="fa fa-check"></i><b>12.7.2</b> Encouraging consistency with marker genes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CRUK Bioinformatics Summer School 2020 - single-cell RNA-seq analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="feature-selection" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Feature selection</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="feature-selection.html#cb1-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="feature-selection.html#cb1-2"></a><span class="kw">library</span>(scater)</span>
<span id="cb1-3"><a href="feature-selection.html#cb1-3"></a><span class="kw">library</span>(scran)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="feature-selection.html#cb2-1"></a>projDir &lt;-<span class="st"> </span>params<span class="op">$</span>projDir</span>
<span id="cb2-2"><a href="feature-selection.html#cb2-2"></a>dirRel &lt;-<span class="st"> </span>params<span class="op">$</span>dirRel</span>
<span id="cb2-3"><a href="feature-selection.html#cb2-3"></a>outDirBit &lt;-<span class="st"> </span>params<span class="op">$</span>outDirBit</span>
<span id="cb2-4"><a href="feature-selection.html#cb2-4"></a>cacheBool &lt;-<span class="st"> </span>params<span class="op">$</span>cacheBool</span>
<span id="cb2-5"><a href="feature-selection.html#cb2-5"></a>setName &lt;-<span class="st"> </span>params<span class="op">$</span>setName</span>
<span id="cb2-6"><a href="feature-selection.html#cb2-6"></a>setSuf &lt;-<span class="st"> </span>params<span class="op">$</span>setSuf</span>
<span id="cb2-7"><a href="feature-selection.html#cb2-7"></a><span class="cf">if</span>(params<span class="op">$</span>bookType <span class="op">==</span><span class="st"> &quot;mk&quot;</span>){</span>
<span id="cb2-8"><a href="feature-selection.html#cb2-8"></a>    dirRel &lt;-<span class="st"> &quot;..&quot;</span></span>
<span id="cb2-9"><a href="feature-selection.html#cb2-9"></a>    setName &lt;-<span class="st"> &quot;caron&quot;</span></span>
<span id="cb2-10"><a href="feature-selection.html#cb2-10"></a>    setSuf &lt;-<span class="st"> &quot;_5hCellPerSpl&quot;</span></span>
<span id="cb2-11"><a href="feature-selection.html#cb2-11"></a>}</span>
<span id="cb2-12"><a href="feature-selection.html#cb2-12"></a></span>
<span id="cb2-13"><a href="feature-selection.html#cb2-13"></a>fontsize &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>), <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</span></code></pre></div>
<div id="load-data-1" class="section level2">
<h2><span class="header-section-number">9.1</span> Load data</h2>
<p>We will load the R file keeping the SCE object with the normalised counts for 500 cells per sample.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="feature-selection.html#cb3-1"></a><span class="co"># Read object in:</span></span>
<span id="cb3-2"><a href="feature-selection.html#cb3-2"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_dimRed.Rds&quot;</span>,</span>
<span id="cb3-3"><a href="feature-selection.html#cb3-3"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb3-4"><a href="feature-selection.html#cb3-4"></a><span class="kw">print</span>(tmpFn)</span></code></pre></div>
<pre><code>## [1] &quot;/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020/AnaWiSce/Ana1/Robjects/caron_sce_nz_postDeconv_5hCellPerSpl_dimRed.Rds&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="feature-selection.html#cb5-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(tmpFn))</span>
<span id="cb5-2"><a href="feature-selection.html#cb5-2"></a>{</span>
<span id="cb5-3"><a href="feature-selection.html#cb5-3"></a>    knitr<span class="op">::</span><span class="kw">knit_exit</span>()</span>
<span id="cb5-4"><a href="feature-selection.html#cb5-4"></a>}</span>
<span id="cb5-5"><a href="feature-selection.html#cb5-5"></a>sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>(tmpFn)</span>
<span id="cb5-6"><a href="feature-selection.html#cb5-6"></a>sce</span></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 16629 5500 
## metadata(0):
## assays(2): counts logcounts
## rownames(16629): ENSG00000237491 ENSG00000225880 ... ENSG00000275063
##   ENSG00000271254
## rowData names(11): ensembl_gene_id external_gene_name ... detected
##   gene_sparsity
## colnames: NULL
## colData names(16): Barcode Run ... cell_sparsity sizeFactor
## reducedDimNames(3): PCA TSNE UMAP
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="feature-selection.html#cb7-1"></a><span class="kw">head</span>(<span class="kw">rowData</span>(sce))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 11 columns
##                 ensembl_gene_id external_gene_name chromosome_name
##                     &lt;character&gt;        &lt;character&gt;     &lt;character&gt;
## ENSG00000237491 ENSG00000237491          LINC01409               1
## ENSG00000225880 ENSG00000225880          LINC00115               1
## ENSG00000230368 ENSG00000230368             FAM41C               1
## ENSG00000230699 ENSG00000230699         AL645608.2               1
## ENSG00000188976 ENSG00000188976              NOC2L               1
## ENSG00000187961 ENSG00000187961             KLHL17               1
##                 start_position end_position strandNum      Symbol
##                      &lt;integer&gt;    &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## ENSG00000237491         778747       810065         1  AL669831.5
## ENSG00000225880         826206       827522        -1   LINC00115
## ENSG00000230368         868071       876903        -1      FAM41C
## ENSG00000230699         911435       914948         1  AL645608.3
## ENSG00000188976         944203       959309        -1       NOC2L
## ENSG00000187961         960584       965719         1      KLHL17
##                            Type       mean  detected gene_sparsity
##                     &lt;character&gt;  &lt;numeric&gt; &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000237491 Gene Expression 0.02785355  2.706672      0.977951
## ENSG00000225880 Gene Expression 0.01376941  1.340222      0.985699
## ENSG00000230368 Gene Expression 0.02027381  1.946076      0.980821
## ENSG00000230699 Gene Expression 0.00144251  0.144251      0.997704
## ENSG00000188976 Gene Expression 0.17711393 14.511645      0.835565
## ENSG00000187961 Gene Expression 0.00354070  0.348825      0.995935</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="feature-selection.html#cb9-1"></a><span class="co">#any(duplicated(rowData(sce)$ensembl_gene_id))</span></span>
<span id="cb9-2"><a href="feature-selection.html#cb9-2"></a><span class="co"># some function(s) used below complain about &#39;strand&#39; already being used in row data,</span></span>
<span id="cb9-3"><a href="feature-selection.html#cb9-3"></a><span class="co"># so rename that column now:</span></span>
<span id="cb9-4"><a href="feature-selection.html#cb9-4"></a><span class="kw">colnames</span>(<span class="kw">rowData</span>(sce))[<span class="kw">colnames</span>(<span class="kw">rowData</span>(sce)) <span class="op">==</span><span class="st"> &quot;strand&quot;</span>] &lt;-<span class="st"> &quot;strandNum&quot;</span></span>
<span id="cb9-5"><a href="feature-selection.html#cb9-5"></a><span class="kw">assayNames</span>(sce)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
</div>
<div id="feature-selection-with-scran" class="section level2">
<h2><span class="header-section-number">9.2</span> Feature selection with scran</h2>
<p>scRNASeq measures the expression of thousands of genes in each cell. The biological question asked in a study will most often relate to a fraction of these genes only, linked for example to differences between cell types, drivers of differentiation, or response to perturbation.</p>
<p>Most high-throughput molecular data include variation created by the assay itself, not biology, i.e. technical noise, for example caused by sampling during RNA capture and library preparation. In scRNASeq, this technical noise will result in most genes being detected at different levels. This noise may hinder the detection of the biological signal.</p>
<p>Let’s identify Highly Variables Genes (HVGs) with the aim to find those underlying the heterogeneity observed across cells.</p>
<div id="modelling-and-removing-technical-noise" class="section level3">
<h3><span class="header-section-number">9.2.1</span> Modelling and removing technical noise</h3>
<p>Some assays allow the inclusion of known molecules in a known amount covering a wide range, from low to high abundance: spike-ins. The technical noise is assessed based on the amount of spike-ins used, the corresponding read counts obtained and their variation across cells. The variance in expression can then be decomposed into the biolgical and technical components.</p>
<p>UMI-based assays do not (yet?) allow spike-ins. But one can still identify HVGs, that is genes with the highest biological component. Assuming that expression does not vary across cells for most genes, the total variance for these genes mainly reflects technical noise. The latter can thus be assessed by fitting a trend to the variance in expression. The fitted value will be the estimate of the technical component.</p>
<p>Let’s fit a trend to the variance, using modelGeneVar().</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="feature-selection.html#cb11-1"></a>dec.sce &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(sce)</span></code></pre></div>
<p>Let’s plot variance against mean of expression (log scale) and the mean-dependent trend fitted to the variance:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="feature-selection.html#cb12-1"></a><span class="co"># Visualizing the fit:</span></span>
<span id="cb12-2"><a href="feature-selection.html#cb12-2"></a>var.fit &lt;-<span class="st"> </span><span class="kw">metadata</span>(dec.sce)</span>
<span id="cb12-3"><a href="feature-selection.html#cb12-3"></a><span class="kw">plot</span>(var.fit<span class="op">$</span>mean, var.fit<span class="op">$</span>var, <span class="dt">xlab=</span><span class="st">&quot;Mean of log-expression&quot;</span>,</span>
<span id="cb12-4"><a href="feature-selection.html#cb12-4"></a>    <span class="dt">ylab=</span><span class="st">&quot;Variance of log-expression&quot;</span>)</span>
<span id="cb12-5"><a href="feature-selection.html#cb12-5"></a><span class="kw">curve</span>(var.fit<span class="op">$</span><span class="kw">trend</span>(x), <span class="dt">col=</span><span class="st">&quot;dodgerblue&quot;</span>, <span class="dt">add=</span><span class="ot">TRUE</span>, <span class="dt">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_var_trend-1.png" width="672" /></p>
<p>In the absence of spike-in data, one can attempt to create a trend by making some distributional assumptions about the noise. For example, UMI counts typically exhibit near-Poisson variation if we only consider technical noise from library preparation and sequencing. This can be used to construct a mean-variance trend in the log-counts with the modelGeneVarByPoisson() function. Note the increased residuals of the high-abundance genes, which can be interpreted as the amount of biological variation that was assumed to be “uninteresting” when fitting the gene-based trend above.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="feature-selection.html#cb13-1"></a><span class="kw">set.seed</span>(<span class="dv">0010101</span>)</span>
<span id="cb13-2"><a href="feature-selection.html#cb13-2"></a>dec.pois.sce &lt;-<span class="st"> </span><span class="kw">modelGeneVarByPoisson</span>(sce)</span>
<span id="cb13-3"><a href="feature-selection.html#cb13-3"></a>dec.pois.sce &lt;-<span class="st"> </span>dec.pois.sce[<span class="kw">order</span>(dec.pois.sce<span class="op">$</span>bio, <span class="dt">decreasing=</span><span class="ot">TRUE</span>),]</span>
<span id="cb13-4"><a href="feature-selection.html#cb13-4"></a><span class="kw">head</span>(dec.pois.sce)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##                      mean     total      tech       bio   p.value       FDR
##                 &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## ENSG00000244734  2.509485  12.14273  0.944558  11.19817         0         0
## ENSG00000188536  1.869015   9.81279  1.046947   8.76585         0         0
## ENSG00000206172  1.624877   8.95495  1.031745   7.92321         0         0
## ENSG00000223609  0.785968   4.36010  0.707898   3.65221         0         0
## ENSG00000019582  2.842386   4.37079  0.848232   3.52256         0         0
## ENSG00000204287  2.329591   3.66889  0.987724   2.68117         0         0</code></pre>
<p>Plot:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="feature-selection.html#cb15-1"></a><span class="kw">plot</span>(dec.pois.sce<span class="op">$</span>mean,</span>
<span id="cb15-2"><a href="feature-selection.html#cb15-2"></a>     dec.pois.sce<span class="op">$</span>total,</span>
<span id="cb15-3"><a href="feature-selection.html#cb15-3"></a>     <span class="dt">pch=</span><span class="dv">16</span>,</span>
<span id="cb15-4"><a href="feature-selection.html#cb15-4"></a>     <span class="dt">xlab=</span><span class="st">&quot;Mean of log-expression&quot;</span>,</span>
<span id="cb15-5"><a href="feature-selection.html#cb15-5"></a>     <span class="dt">ylab=</span><span class="st">&quot;Variance of log-expression&quot;</span>)</span>
<span id="cb15-6"><a href="feature-selection.html#cb15-6"></a><span class="kw">curve</span>(<span class="kw">metadata</span>(dec.pois.sce)<span class="op">$</span><span class="kw">trend</span>(x), <span class="dt">col=</span><span class="st">&quot;dodgerblue&quot;</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/modelGeneVarByPoisson_plot-1.png" width="672" /></p>
<p>Interestingly, trends based purely on technical noise tend to yield large biological components for highly-expressed genes. This often includes so-called “house-keeping” genes coding for essential cellular components such as ribosomal proteins, which are considered uninteresting for characterizing cellular heterogeneity. These observations suggest that a more accurate noise model does not necessarily yield a better ranking of HVGs, though one should keep an open mind - house-keeping genes are regularly DE in a variety of conditions (Glare et al. 2002; Nazari, Parham, and Maleki 2015; Guimaraes and Zavolan 2016), and the fact that they have large biological components indicates that there is strong variation across cells that may not be completely irrelevant.</p>
</div>
<div id="choosing-some-hvgs" class="section level3">
<h3><span class="header-section-number">9.2.2</span> Choosing some HVGs:</h3>
<p>Identify the top 20 HVGs by sorting genes in decreasing order of biological component.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="feature-selection.html#cb16-1"></a><span class="co"># order genes by decreasing order of biological component</span></span>
<span id="cb16-2"><a href="feature-selection.html#cb16-2"></a>o &lt;-<span class="st"> </span><span class="kw">order</span>(dec.sce<span class="op">$</span>bio, <span class="dt">decreasing=</span><span class="ot">TRUE</span>)</span>
<span id="cb16-3"><a href="feature-selection.html#cb16-3"></a><span class="co"># check top and bottom of sorted table</span></span>
<span id="cb16-4"><a href="feature-selection.html#cb16-4"></a><span class="kw">head</span>(dec.sce[o,])</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##                      mean     total      tech       bio      p.value
##                 &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;
## ENSG00000244734  2.509485  12.14273  1.501176  10.64155  0.00000e+00
## ENSG00000188536  1.869015   9.81279  1.407077   8.40572  0.00000e+00
## ENSG00000206172  1.624877   8.95495  1.328093   7.62686  0.00000e+00
## ENSG00000223609  0.785968   4.36010  0.814664   3.54544 1.76443e-226
## ENSG00000019582  2.842386   4.37079  1.491606   2.87918  2.57120e-46
## ENSG00000206177  0.580477   3.24218  0.632917   2.60926 1.77838e-203
##                          FDR
##                    &lt;numeric&gt;
## ENSG00000244734  0.00000e+00
## ENSG00000188536  0.00000e+00
## ENSG00000206172  0.00000e+00
## ENSG00000223609 5.86744e-223
## ENSG00000019582  1.01789e-43
## ENSG00000206177 4.92819e-200</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="feature-selection.html#cb18-1"></a><span class="kw">tail</span>(dec.sce[o,])</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##                      mean     total      tech       bio   p.value       FDR
##                 &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## ENSG00000108654   2.35282   1.23874   1.50084 -0.262102  0.901192  0.963563
## ENSG00000063177   3.03164   1.20580   1.48289 -0.277097  0.915987  0.972156
## ENSG00000174748   3.76111   1.14211   1.43802 -0.295910  0.935507  0.983329
## ENSG00000115268   3.81719   1.11875   1.42987 -0.311116  0.945777  0.988399
## ENSG00000149806   2.86380   1.14650   1.49128 -0.344780  0.955963  0.994046
## ENSG00000173812   3.05559   1.13496   1.48016 -0.345203  0.957336  0.994976</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="feature-selection.html#cb20-1"></a><span class="co"># choose the top 20 genes with the highest biological component</span></span>
<span id="cb20-2"><a href="feature-selection.html#cb20-2"></a>chosen.genes.index &lt;-<span class="st"> </span>o[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]</span></code></pre></div>
<p>Show the top 20 HVGs on the plot displaying the variance against the mean expression:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="feature-selection.html#cb21-1"></a><span class="kw">plot</span>(var.fit<span class="op">$</span>mean, var.fit<span class="op">$</span>var)</span>
<span id="cb21-2"><a href="feature-selection.html#cb21-2"></a><span class="kw">curve</span>(var.fit<span class="op">$</span><span class="kw">trend</span>(x), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</span>
<span id="cb21-3"><a href="feature-selection.html#cb21-3"></a><span class="kw">points</span>(var.fit<span class="op">$</span>mean[chosen.genes.index], var.fit<span class="op">$</span>var[chosen.genes.index], <span class="dt">col=</span><span class="st">&quot;orange&quot;</span>)</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_var_trend_HVGtop20-1.png" width="672" /></p>
<p>Rather than choosing a fixed number of top genes, one may define ‘HVGs’ as genes with a positive biological component, ie whose variance is higher than the fitted value for the corresponding mean expression.</p>
<p>Select and show these ‘HVGs’ on the plot displaying the variance against the mean expression:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="feature-selection.html#cb22-1"></a>hvgBool &lt;-<span class="st"> </span>dec.sce<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb22-2"><a href="feature-selection.html#cb22-2"></a><span class="kw">table</span>(hvgBool)</span></code></pre></div>
<pre><code>## hvgBool
## FALSE  TRUE 
##  7301  9328</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="feature-selection.html#cb24-1"></a>hvg.index &lt;-<span class="st"> </span><span class="kw">which</span>(hvgBool)</span>
<span id="cb24-2"><a href="feature-selection.html#cb24-2"></a><span class="kw">plot</span>(var.fit<span class="op">$</span>mean, var.fit<span class="op">$</span>var)</span>
<span id="cb24-3"><a href="feature-selection.html#cb24-3"></a><span class="kw">curve</span>(var.fit<span class="op">$</span><span class="kw">trend</span>(x), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</span>
<span id="cb24-4"><a href="feature-selection.html#cb24-4"></a><span class="kw">points</span>(var.fit<span class="op">$</span>mean[hvg.index], var.fit<span class="op">$</span>var[hvg.index], <span class="dt">col=</span><span class="st">&quot;orange&quot;</span>)</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_var_trend_HVGbioCompPos-1.png" width="672" /></p>
<p>Save objects to file</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="feature-selection.html#cb25-1"></a>tmpFn &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%s/%s/Robjects/%s_sce_nz_postDeconv%s_featSel.Rds&quot;</span>,</span>
<span id="cb25-2"><a href="feature-selection.html#cb25-2"></a>         projDir, outDirBit, setName, setSuf)</span>
<span id="cb25-3"><a href="feature-selection.html#cb25-3"></a><span class="kw">saveRDS</span>(<span class="kw">list</span>(<span class="st">&quot;dec.sce&quot;</span>=dec.sce,<span class="st">&quot;hvg.index&quot;</span>=hvg.index), tmpFn)</span></code></pre></div>
<p>HVGs may be driven by outlier cells. So let’s plot the distribution of expression values for the genes with the largest biological components.</p>
<p>First, get gene names to replace ensembl IDs on plot.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="feature-selection.html#cb26-1"></a><span class="co"># the count matrix rows are named with ensembl gene IDs. Let&#39;s label gene with their name instead:</span></span>
<span id="cb26-2"><a href="feature-selection.html#cb26-2"></a><span class="co"># row indices of genes in rowData(sce)</span></span>
<span id="cb26-3"><a href="feature-selection.html#cb26-3"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index])</span>
<span id="cb26-4"><a href="feature-selection.html#cb26-4"></a><span class="co"># check:</span></span>
<span id="cb26-5"><a href="feature-selection.html#cb26-5"></a><span class="kw">rowData</span>(sce)[tmpInd,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;external_gene_name&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 20 rows and 2 columns
##                 ensembl_gene_id external_gene_name
##                     &lt;character&gt;        &lt;character&gt;
## ENSG00000143546 ENSG00000143546             S100A8
## ENSG00000211592 ENSG00000211592               IGKC
## ENSG00000145335 ENSG00000145335               SNCA
## ENSG00000170180 ENSG00000170180               GYPA
## ENSG00000019582 ENSG00000019582               CD74
## ...                         ...                ...
## ENSG00000206177 ENSG00000206177                HBM
## ENSG00000188536 ENSG00000188536               HBA2
## ENSG00000206172 ENSG00000206172               HBA1
## ENSG00000169877 ENSG00000169877               AHSP
## ENSG00000090013 ENSG00000090013              BLVRB</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="feature-selection.html#cb28-1"></a><span class="co"># store names:</span></span>
<span id="cb28-2"><a href="feature-selection.html#cb28-2"></a>tmpName &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;external_gene_name&quot;</span>]</span>
<span id="cb28-3"><a href="feature-selection.html#cb28-3"></a><span class="co"># the gene name may not be known, so keep the ensembl gene ID in that case:</span></span>
<span id="cb28-4"><a href="feature-selection.html#cb28-4"></a>tmpName[tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>]</span>
<span id="cb28-5"><a href="feature-selection.html#cb28-5"></a>tmpName[<span class="kw">is.na</span>(tmpName)] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][<span class="kw">is.na</span>(tmpName)]</span>
<span id="cb28-6"><a href="feature-selection.html#cb28-6"></a><span class="kw">rm</span>(tmpInd)</span></code></pre></div>
<p>Now show a violin plot for each gene, using plotExpression() and label genes with their name:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="feature-selection.html#cb29-1"></a>g &lt;-<span class="st"> </span><span class="kw">plotExpression</span>(sce, <span class="kw">rownames</span>(dec.sce)[chosen.genes.index], </span>
<span id="cb29-2"><a href="feature-selection.html#cb29-2"></a>    <span class="dt">point_alpha=</span><span class="fl">0.05</span>, <span class="dt">jitter=</span><span class="st">&quot;jitter&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb29-3"><a href="feature-selection.html#cb29-3"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks=</span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index],</span>
<span id="cb29-4"><a href="feature-selection.html#cb29-4"></a>        <span class="dt">labels=</span>tmpName)</span>
<span id="cb29-5"><a href="feature-selection.html#cb29-5"></a>g</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_count_HVGtop20-1.png" width="672" /></p>
<p>Another few genes:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="feature-selection.html#cb30-1"></a>chosen.genes.index &lt;-<span class="st"> </span>o[<span class="dv">21</span><span class="op">:</span><span class="dv">40</span>]</span>
<span id="cb30-2"><a href="feature-selection.html#cb30-2"></a></span>
<span id="cb30-3"><a href="feature-selection.html#cb30-3"></a>tmpInd &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index])</span>
<span id="cb30-4"><a href="feature-selection.html#cb30-4"></a><span class="co"># check:</span></span>
<span id="cb30-5"><a href="feature-selection.html#cb30-5"></a><span class="kw">rowData</span>(sce)[tmpInd,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;external_gene_name&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 20 rows and 2 columns
##                 ensembl_gene_id external_gene_name
##                     &lt;character&gt;        &lt;character&gt;
## ENSG00000117632 ENSG00000117632              STMN1
## ENSG00000163220 ENSG00000163220             S100A9
## ENSG00000196154 ENSG00000196154             S100A4
## ENSG00000136732 ENSG00000136732               GYPC
## ENSG00000250361 ENSG00000250361               GYPB
## ...                         ...                ...
## ENSG00000013306 ENSG00000013306           SLC25A39
## ENSG00000171223 ENSG00000171223               JUNB
## ENSG00000167815 ENSG00000167815              PRDX2
## ENSG00000198692 ENSG00000198692             EIF1AY
## ENSG00000169575 ENSG00000169575             VPREB1</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="feature-selection.html#cb32-1"></a><span class="co"># store names:</span></span>
<span id="cb32-2"><a href="feature-selection.html#cb32-2"></a>tmpName &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;external_gene_name&quot;</span>]</span>
<span id="cb32-3"><a href="feature-selection.html#cb32-3"></a><span class="co"># the gene name may not be known, so keep the ensembl gene ID in that case:</span></span>
<span id="cb32-4"><a href="feature-selection.html#cb32-4"></a>tmpName[tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>]</span>
<span id="cb32-5"><a href="feature-selection.html#cb32-5"></a>tmpName[<span class="kw">is.na</span>(tmpName)] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][<span class="kw">is.na</span>(tmpName)]</span>
<span id="cb32-6"><a href="feature-selection.html#cb32-6"></a><span class="kw">rm</span>(tmpInd)</span>
<span id="cb32-7"><a href="feature-selection.html#cb32-7"></a></span>
<span id="cb32-8"><a href="feature-selection.html#cb32-8"></a>g &lt;-<span class="st"> </span><span class="kw">plotExpression</span>(sce, <span class="kw">rownames</span>(dec.sce)[chosen.genes.index], </span>
<span id="cb32-9"><a href="feature-selection.html#cb32-9"></a>    <span class="dt">point_alpha=</span><span class="fl">0.05</span>, <span class="dt">jitter=</span><span class="st">&quot;jitter&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb32-10"><a href="feature-selection.html#cb32-10"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks=</span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index],</span>
<span id="cb32-11"><a href="feature-selection.html#cb32-11"></a>        <span class="dt">labels=</span>tmpName)</span>
<span id="cb32-12"><a href="feature-selection.html#cb32-12"></a>g</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_count_HVGtop20_2-1.png" width="672" /></p>
<p><strong>Challenge</strong>: Show violin plots for the 20 genes with the lowest biological component. How do they compare to the those for HVGs chosen above?</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="feature-selection.html#cb33-1"></a>chosen.genes.index.tmp &lt;-<span class="st"> </span><span class="kw">order</span>(dec.sce<span class="op">$</span>bio, <span class="dt">decreasing=</span><span class="ot">FALSE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]</span>
<span id="cb33-2"><a href="feature-selection.html#cb33-2"></a>tmpInd &lt;-<span class="st"> </span>(<span class="kw">which</span>(<span class="kw">rowData</span>(sce)<span class="op">$</span>ensembl_gene_id <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index.tmp]))</span>
<span id="cb33-3"><a href="feature-selection.html#cb33-3"></a><span class="co"># check:</span></span>
<span id="cb33-4"><a href="feature-selection.html#cb33-4"></a><span class="kw">rowData</span>(sce)[tmpInd,<span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;external_gene_name&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 20 rows and 2 columns
##                 ensembl_gene_id external_gene_name
##                     &lt;character&gt;        &lt;character&gt;
## ENSG00000142676 ENSG00000142676              RPL11
## ENSG00000187514 ENSG00000187514               PTMA
## ENSG00000174748 ENSG00000174748              RPL15
## ENSG00000232112 ENSG00000232112               TMA7
## ENSG00000104529 ENSG00000104529              EEF1D
## ...                         ...                ...
## ENSG00000105640 ENSG00000105640             RPL18A
## ENSG00000105372 ENSG00000105372              RPS19
## ENSG00000063177 ENSG00000063177              RPL18
## ENSG00000142534 ENSG00000142534              RPS11
## ENSG00000170889 ENSG00000170889               RPS9</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="feature-selection.html#cb35-1"></a><span class="co"># store names:</span></span>
<span id="cb35-2"><a href="feature-selection.html#cb35-2"></a>tmpName &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;external_gene_name&quot;</span>]</span>
<span id="cb35-3"><a href="feature-selection.html#cb35-3"></a><span class="co"># the gene name may not be known, so keep the ensembl gene ID in that case:</span></span>
<span id="cb35-4"><a href="feature-selection.html#cb35-4"></a>tmpName[tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][tmpName<span class="op">==</span><span class="st">&quot;&quot;</span>]</span>
<span id="cb35-5"><a href="feature-selection.html#cb35-5"></a>tmpName[<span class="kw">is.na</span>(tmpName)] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)[tmpInd,<span class="st">&quot;ensembl_gene_id&quot;</span>][<span class="kw">is.na</span>(tmpName)]</span>
<span id="cb35-6"><a href="feature-selection.html#cb35-6"></a><span class="kw">rm</span>(tmpInd)</span>
<span id="cb35-7"><a href="feature-selection.html#cb35-7"></a>g &lt;-<span class="st"> </span><span class="kw">plotExpression</span>(sce, <span class="kw">rownames</span>(dec.sce)[chosen.genes.index.tmp], </span>
<span id="cb35-8"><a href="feature-selection.html#cb35-8"></a>            <span class="dt">point_alpha=</span><span class="fl">0.05</span>, <span class="dt">jitter=</span><span class="st">&quot;jitter&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</span>
<span id="cb35-9"><a href="feature-selection.html#cb35-9"></a>g &lt;-<span class="st"> </span>g <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks=</span><span class="kw">rownames</span>(dec.sce)[chosen.genes.index.tmp],</span>
<span id="cb35-10"><a href="feature-selection.html#cb35-10"></a>        <span class="dt">labels=</span>tmpName)</span>
<span id="cb35-11"><a href="feature-selection.html#cb35-11"></a>g</span></code></pre></div>
<p><img src="MkWoC_files/figure-html/plot_count_violoin_HVGbot20-1.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="feature-selection.html#cb36-1"></a><span class="kw">rm</span>(chosen.genes.index.tmp)</span></code></pre></div>
</div>
</div>
<div id="session-information-6" class="section level2">
<h2><span class="header-section-number">9.3</span> Session information</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="feature-selection.html#cb37-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 8 (Core)
## 
## Matrix products: default
## BLAS:   /opt/R/R-4.0.3/lib64/R/lib/libRblas.so
## LAPACK: /opt/R/R-4.0.3/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] knitr_1.32                  scran_1.18.7               
##  [3] scater_1.18.6               SingleCellExperiment_1.12.0
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
##  [9] IRanges_2.24.1              S4Vectors_0.28.1           
## [11] BiocGenerics_0.36.1         MatrixGenerics_1.2.1       
## [13] matrixStats_0.58.0          ggplot2_3.3.3              
## 
## loaded via a namespace (and not attached):
##  [1] viridis_0.5.1             edgeR_3.32.1             
##  [3] sass_0.3.1                BiocSingular_1.6.0       
##  [5] jsonlite_1.7.2            viridisLite_0.3.0        
##  [7] DelayedMatrixStats_1.12.3 scuttle_1.0.4            
##  [9] bslib_0.2.4               assertthat_0.2.1         
## [11] statmod_1.4.35            highr_0.8                
## [13] dqrng_0.2.1               GenomeInfoDbData_1.2.4   
## [15] vipor_0.4.5               yaml_2.2.1               
## [17] pillar_1.6.0              lattice_0.20-41          
## [19] limma_3.46.0              glue_1.4.2               
## [21] beachmat_2.6.4            digest_0.6.27            
## [23] XVector_0.30.0            colorspace_2.0-0         
## [25] cowplot_1.1.1             htmltools_0.5.1.1        
## [27] Matrix_1.2-18             pkgconfig_2.0.3          
## [29] bookdown_0.21             zlibbioc_1.36.0          
## [31] purrr_0.3.4               scales_1.1.1             
## [33] BiocParallel_1.24.1       tibble_3.1.1             
## [35] farver_2.1.0              generics_0.1.0           
## [37] ellipsis_0.3.1            withr_2.4.2              
## [39] magrittr_2.0.1            crayon_1.4.1             
## [41] evaluate_0.14             fansi_0.4.2              
## [43] bluster_1.0.0             beeswarm_0.2.3           
## [45] tools_4.0.3               lifecycle_1.0.0          
## [47] stringr_1.4.0             locfit_1.5-9.4           
## [49] munsell_0.5.0             DelayedArray_0.16.3      
## [51] irlba_2.3.3               compiler_4.0.3           
## [53] jquerylib_0.1.3           rsvd_1.0.3               
## [55] rlang_0.4.10              grid_4.0.3               
## [57] RCurl_1.98-1.3            BiocNeighbors_1.8.2      
## [59] rstudioapi_0.13           igraph_1.2.6             
## [61] labeling_0.4.2            bitops_1.0-6             
## [63] rmarkdown_2.7             gtable_0.3.0             
## [65] DBI_1.1.1                 R6_2.5.0                 
## [67] gridExtra_2.3             dplyr_1.0.5              
## [69] utf8_1.2.1                stringi_1.5.3            
## [71] ggbeeswarm_0.6.0          Rcpp_1.0.6               
## [73] vctrs_0.3.7               tidyselect_1.1.0         
## [75] xfun_0.22                 sparseMatrixStats_1.2.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dimensionality-reduction-for-visualisation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="batch-correction-gsm3872442-set.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/featSelec.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MkWoC.pdf", "MkWoC.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"code_folding": "hide"
});
});
</script>

</body>

</html>
