---
title: "CRUK CI Summer School 2020 - introduction to single-cell RNA-seq analysis"
subtitle: 'Data integration'
author: "Stephane Ballereau, Zeynep Kalender Atak, Katarzyna Kania"
output:
  html_document:
    df_print: paged
    toc: yes
    number_sections: true
    code_folding: hide
  html_notebook:
    code_folding: hide
    toc: yes
    toc_float: yes
    number_sections: true
  html_book:
    code_folding: hide
params:
  projDir: "/ssd/personal/baller01/20200511_FernandesM_ME_crukBiSs2020"
  dirRel: ".."
  inpDirBit: "AnaWiSce/Ana1"
  outDirBit: "AnaWiSce/Ana1"
  bookType: "mk"
  cacheBool: FALSE  
---

# Data integration - PBMMC {#DataIntegrationPbmmcTop}

```{r}
projDir <- params$projDir
dirRel <- params$dirRel
outDirBit <- params$outDirBit
nbPcToComp <- 50

if(params$bookType == "mk"){
	setName <- "caron"
	splSetToGet <- "PBMMC"
	setSuf <- "_allCells"
}

```

```{r dataSetIntegration_PBMMC_setup, include=FALSE, echo=FALSE, dev="CairoPNG"}
cacheBool <- params$cacheBool
require(knitr)
opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE, cache=cacheBool)
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(eval = TRUE) 
options(stringsAsFactors = FALSE)
opts_chunk$set(fig.width=7, fig.height=7)
#opts_chunk$set(dev = c("png"))
options(bitmapType='cairo')
knitr::opts_chunk$set(dev="CairoPNG")
set.seed(123) # for reproducibility 
```

```{r, include=FALSE}
library(ggplot2)
library(scater)
library(scran)
library(dplyr)
fontsize <- theme(axis.text=element_text(size=12), axis.title=element_text(size=16))
```

Source: ['Integrating Datasets'](https://osca.bioconductor.org/integrating-datasets.html) chapter in the OSCA book. Its text is reproduced below with few modifications to adapt it to the data set under scrutiny here.

## Abbreviations

* HVG: highly variable genes
* MNN: mutual nearest neighbors
* PBMMC: peripheral blood mononuclear cell
* SCE: SingleCellExperiment

## Motivation

Large single-cell RNA sequencing (scRNA-seq) projects usually need to generate data across multiple batches due to logistical constraints. However, the processing of different batches is often subject to uncontrollable differences, e.g., changes in operator, differences in reagent quality. This results in systematic differences in the observed expression in cells from different batches, which we refer to as “batch effects”. Batch effects are problematic as they can be major drivers of heterogeneity in the data, masking the relevant biological differences and complicating interpretation of the results.

Computational correction of these effects is critical for eliminating batch-to-batch variation, allowing data across multiple batches to be combined for common downstream analysis. However, existing methods based on linear models (Ritchie et al. 2015; Leek et al. 2012) assume that the composition of cell populations are either known or the same across batches. To overcome these limitations, bespoke methods have been developed for batch correction of single-cell data (Haghverdi et al. 2018; Butler et al. 2018; Lin et al. 2019) that do not require a priori knowledge about the composition of the population. This allows them to be used in workflows for exploratory analyses of scRNA-seq data where such knowledge is usually unavailable.

## Loading the data

We will load the R file keeping the SCE object with the normalised counts, and subset 1000 cells per sample.

```{r}
tmpFn <- sprintf("%s/%s/Robjects/%s_sce_nz_postDeconv%s.Rds", projDir, outDirBit, setName, setSuf)

print(tmpFn)

if(!file.exists(tmpFn))
{
	knitr::knit_exit()
}
sce <- readRDS(tmpFn)
sce
colnames(rowData(sce))[colnames(rowData(sce)) == "strand"] <- "strandNum"
```

We next subset the data for the `r splSetToGet` sample group: 

```{r}
# CaronBourque2020
cb_sampleSheetFn <- file.path(projDir, "Data/CaronBourque2020/SraRunTable.txt")
cb_sampleSheet <- read.table(cb_sampleSheetFn, header=T, sep=",")
splVec <- cb_sampleSheet %>% filter(source_name == splSetToGet) %>%
	pull(Sample.Name) %>% unique()

sourceNames <- unique(colData(sce)$source_name)
sceOrig <- sce
sce <- sceOrig[,sce$source_name == splSetToGet ]
nbCells <- 1000
all.sce <- list()
for(spx in splVec)
{
	vec.bc <- colData(sce) %>%
		data.frame() %>%
		filter(Sample.Name == spx) %>%
		sample_n(nbCells) %>%
		pull(Barcode)
	tmpInd <- which(colData(sce)$Barcode %in% vec.bc)
	
	all.sce[[spx]] <- sce[,tmpInd]
}
nbSpl <- length(all.sce)
```

We then apply the standard workflow to each sample separately:
* normalisation,
* variance modelling
* dimensionality reduction
* clustering

```{r}
#--- normalization ---#
# use logNormCounts()
all.sce <- lapply(all.sce, logNormCounts)

#--- variance-modelling ---#
# model varaince with modelGeneVar()
# find highly variable genes (HVGs) with getTopHVGs()
library(scran)
all.dec <- lapply(all.sce, modelGeneVar)
all.hvgs <- lapply(all.dec, getTopHVGs, prop=0.1)

#--- dimensionality-reduction ---#
# use runPCA()
# then compute embeddings with runTSNE() and runUMAP()
library(BiocSingular)
set.seed(10000)
all.sce <- mapply(FUN=runPCA, x=all.sce, subset_row=all.hvgs, 
    MoreArgs=list(ncomponents=25, BSPARAM=RandomParam()), 
    SIMPLIFY=FALSE)

set.seed(100000)
all.sce <- lapply(all.sce, runTSNE, dimred="PCA")

set.seed(1000000)
all.sce <- lapply(all.sce, runUMAP, dimred="PCA")

#--- clustering ---#
# cluster each sample separately
for (n in names(all.sce)) {
    g <- buildSNNGraph(all.sce[[n]], k=10, use.dimred='PCA')
    clust <- igraph::cluster_walktrap(g)$membership
    colLabels(all.sce[[n]])  <- factor(clust)
}
```

To prepare for the batch correction:

* We subset all batches to the common “universe” of features. In this case, it is straightforward as both batches use Ensembl gene annotation.

```{r}
allNames <- unlist(lapply(all.sce, function(x){rownames(x)}))
allNamesNb <- table(allNames)
universe <- names(allNamesNb)[allNamesNb==nbSpl] 
#length(universe)
```

* The size of this common “universe” of features here is the number of features shared by all `r nbSpl` samples is: `r length(universe)`.

```{r}
# Subsetting the SingleCellExperiment object.
uni.sce <- lapply(all.sce, function(x){x[universe,]})
# Also subsetting the variance modelling results, for convenience.
uni.dec <- lapply(all.dec, function(x){x[universe,]})
```

* We rescale each batch to adjust for differences in sequencing depth between batches. The `multiBatchNorm()` function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment (SCE) objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.

```{r}
# rescale each batch to adjust for differences in sequencing depth between batches
library(batchelor)
rescaled <- multiBatchNorm(uni.sce, batch = "Sample.Name")
```

* We perform feature selection by averaging the variance components across all batches with the `combineVar()` function. We compute the average as it is responsive to batch-specific highly-variable genes (HVGs) while still preserving the within-batch ranking of genes.

```{r}
library(scran)
# compute average variance components across samples
combined.dec <- combineVar(uni.dec[[1]], uni.dec[[2]], uni.dec[[3]])
# identify highly variables genes
# here as those with a positive biological component
chosen.hvgs <- combined.dec$bio > 0
# sum(chosen.hvgs)
```

Number of HVGs: `r sum(chosen.hvgs)`.

When integrating datasets of variable composition, it is generally safer to err on the side of including more genes than are used in a single dataset analysis, to ensure that markers are retained for any dataset-specific subpopulations that might be present. For a top X selection, this means using a larger X (say, ~5000), or in this case, we simply take all genes above the trend.

Alternatively, a more forceful approach to feature selection can be used based on marker genes from within-batch comparisons.

## Diagnosing batch effects

Before we actually perform any correction, it is worth examining whether there is any batch effect in this dataset. We combine the three SCE  objects and perform a PCA on the log-expression values for all genes with positive (average) biological components.

```{r}
# Synchronizing the metadata for cbind()ing.

# before concatenating matrices,
# check that features are identical in all matrices: 
identical(rowData(rescaled[[1]]), rowData(rescaled[[2]]))
identical(rowData(rescaled[[1]]), rowData(rescaled[[3]]))

# have a 'batch' entry in cell meta data:
# (not strikly necessary as any column can be used)
rescaled2 <- lapply(rescaled, function(x){x$batch <- x$Sample.Name; x})
rescaled <- rescaled2
rm(rescaled2)

# concat matrices:
uncorrected <- cbind(rescaled[[1]], rescaled[[2]], rescaled[[3]])

# Perform PCA
# Using RandomParam() as it is more efficient for file-backed matrices.
library(scater)
set.seed(0010101010)
uncorrected <- runPCA(uncorrected,
		      subset_row=chosen.hvgs,
		      BSPARAM=BiocSingular::RandomParam())
```

We use graph-based clustering on the components to obtain a summary of the population structure.

As the samples should be replicates, each cluster should ideally consist of cells from each batch. However, we instead see clusters that are comprised of cells from a single batch. This indicates that cells of the same type are artificially separated due to technical differences between batches.

```{r}
library(scran)
# build shared nearest-neighbour graph
snn.gr <- buildSNNGraph(uncorrected, use.dimred="PCA")
# identify cluster with the walk trap method
clusters <- igraph::cluster_walktrap(snn.gr)$membership
# get number of cells for each {cluster, batch} pair
tab <- table(Cluster=clusters, Batch=uncorrected$batch)
tab
```

```{r}
tmpMat <- data.frame("clusters"=clusters, "batch"=uncorrected$batch)
```

```{r, eval=FALSE, include=FALSE}
# clusters ordered by name
p <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
	geom_bar() +
	theme(axis.text.x=element_text(angle = 90, hjust = 0))
p1 <- p
p <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
  geom_bar(position = "fill") +
  scale_y_continuous(labels = scales::percent)
p2 <- p
```

```{r, eval=FALSE, include=FALSE}
# clusters ordered by name

gridExtra::grid.arrange(p1, p2)
```

Cluster size and cell contribution by sample:

```{r}
tmpMatTab <- table(tmpMat)
sortVecNames <- tmpMatTab %>% rowSums %>% sort(decreasing=TRUE) %>% names
tmpMat$clusters <- factor(tmpMat$clusters, levels=sortVecNames)
tmpMatDf <- tmpMatTab[sortVecNames,] %>% data.frame()
p1 <- ggplot(data=tmpMatDf, aes(x=clusters,y=Freq, fill=batch)) +
	geom_col() +
  ggtitle("uncorrected, cell numbers")
p2 <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
  geom_bar(position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  ggtitle("uncorrected, proportions")
```
```{r}
gridExtra::grid.arrange(p1, p2)
```

We can also visualize the corrected coordinates using a t-SNE plot. The strong separation between cells from different batches is consistent with the clustering results.

```{r}
set.seed(1111001)
uncorrected <- runTSNE(uncorrected, dimred="PCA")
plotTSNE(uncorrected, colour_by="batch")
```

Of course, the other explanation for batch-specific clusters is that there are cell types that are unique to each batch. The degree of intermingling of cells from different batches is not an effective diagnostic when the batches involved might actually contain unique cell subpopulations (which is not a consideration in the PBMMC dataset, but the same cannot be said in general). If a cluster only contains cells from a single batch, one can always debate whether that is caused by a failure of the correction method or if there is truly a batch-specific subpopulation. For example, do batch-specific metabolic or differentiation states represent distinct subpopulations? Or should they be merged together? We will not attempt to answer this here, only noting that each batch correction algorithm will make different (and possibly inappropriate) decisions on what constitutes “shared” and “unique” populations.

## Linear regression

Batch effects in bulk RNA sequencing studies are commonly removed with linear regression. This involves fitting a linear model to each gene’s expression profile, setting the undesirable batch term to zero and recomputing the observations sans the batch effect, yielding a set of corrected expression values for downstream analyses. Linear modelling is the basis of the `removeBatchEffect()` function from the `limma` package (Ritchie et al. 2015) as well the `comBat()` function from the `sva` package (Leek et al. 2012).

To use this approach in a scRNA-seq context, we assume that the composition of cell subpopulations is the same across batches. We also assume that the batch effect is additive, i.e., any batch-induced fold-change in expression is the same across different cell subpopulations for any given gene. These are strong assumptions as batches derived from different individuals will naturally exhibit variation in cell type abundances and expression. Nonetheless, they may be acceptable when dealing with batches that are technical replicates generated from the same population of cells. (In fact, when its assumptions hold, linear regression is the most statistically efficient as it uses information from all cells to compute the common batch vector.) Linear modelling can also accommodate situations where the composition is known a priori by including the cell type as a factor in the linear model, but this situation is even less common.

We use the `rescaleBatches()` function from the batchelor package to remove the batch effect. This is roughly equivalent to applying a linear regression to the log-expression values per gene, with some adjustments to improve performance and efficiency. For each gene, the mean expression in each batch is scaled down until it is equal to the lowest mean across all batches. We deliberately choose to scale all expression values down as this mitigates differences in variance when batches lie at different positions on the mean-variance trend. (Specifically, the shrinkage effect of the pseudo-count is greater for smaller counts, suppressing any differences in variance across batches.) An additional feature of `rescaleBatches()` is that it will preserve sparsity in the input matrix for greater efficiency, whereas other methods like `removeBatchEffect()` will always return a dense matrix.

```{r}
library(batchelor)
rescaled2 <- rescaleBatches(rescaled)
rescaled2
```

After clustering, we should observe that most clusters consist of mixtures of cells from the replicate batches, consistent with the removal of the batch effect.

```{r}
set.seed(1010101010) # To ensure reproducibility of IRLBA.
rescaled2 <- runPCA(rescaled2, subset_row=chosen.hvgs, exprs_values="corrected")

snn.gr <- buildSNNGraph(rescaled2, use.dimred="PCA")
clusters.resc <- igraph::cluster_walktrap(snn.gr)$membership
tab.resc <- table(Cluster=clusters.resc, Batch=rescaled2$batch)
tab.resc
```

<!-- Cluster size and cell contribution by sample: -->

```{r}
tmpMat <- data.frame("clusters"=clusters.resc, "batch"=rescaled2$batch)
```

```{r, eval=FALSE, include=FALSE}
# clusters ordered by name
p <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
	geom_bar() +
	theme(axis.text.x=element_text(angle = 90, hjust = 0))
p1 <- p
p <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
  geom_bar(position = "fill") +
  scale_y_continuous(labels = scales::percent)
p2 <- p
```

```{r, eval=FALSE, include=FALSE}
gridExtra::grid.arrange(p1, p2)
```

Cluster size and cell contribution by sample, with clusters sorted by size:

```{r}
tmpMatTab <- table(tmpMat)
sortVecNames <- tmpMatTab %>% rowSums %>% sort(decreasing=TRUE) %>% names
tmpMat$clusters <- factor(tmpMat$clusters, levels=sortVecNames)
tmpMatDf <- tmpMatTab[sortVecNames,] %>% data.frame()
p1 <- ggplot(data=tmpMatDf, aes(x=clusters,y=Freq, fill=batch)) +
	geom_col()
p2 <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
  geom_bar(position = "fill") +
  scale_y_continuous(labels = scales::percent)
```
```{r}
gridExtra::grid.arrange(p1, p2)
```

```{r}
rescaled2 <- runTSNE(rescaled2, dimred="PCA")
rescaled2$batch <- factor(rescaled2$batch)
plotTSNE(rescaled2, colour_by="batch")
```

## Performing MNN correction

### Algorithm overview

Consider a cell a in batch A, and identify the cells in batch B that are nearest neighbors to a in the expression space defined by the selected features. Repeat this for a cell b in batch B, identifying its nearest neighbors in A. Mutual nearest neighbors (MNN) are pairs of cells from different batches that belong in each other’s set of nearest neighbors. The reasoning is that MNN pairs represent cells from the same biological state prior to the application of a batch effect - see Haghverdi et al. (2018) for full theoretical details. Thus, the difference between cells in MNN pairs can be used as an estimate of the batch effect, the subtraction of which yields batch-corrected values.

Compared to linear regression, MNN correction does not assume that the population composition is the same or known beforehand. This is because it learns the shared population structure via identification of MNN pairs and uses this information to obtain an appropriate estimate of the batch effect. Instead, the key assumption of MNN-based approaches is that the batch effect is orthogonal to the biology in high-dimensional expression space. Violations reduce the effectiveness and accuracy of the correction, with the most common case arising from variations in the direction of the batch effect between clusters. Nonetheless, the assumption is usually reasonable as a random vector is very likely to be orthogonal in high-dimensional space.

### Application to the data

The `batchelor` package provides an implementation of the MNN approach via the `fastMNN()` function. (Unlike the MNN method originally described by Haghverdi et al. (2018), the `fastMNN()` function performs PCA to reduce the dimensions beforehand and speed up the downstream neighbor detection steps.) We apply it to our batches to remove the batch effect across the highly variable genes in `chosen.hvgs`. To reduce computational work and technical noise, all cells in all batches are projected into the low-dimensional space defined by the top d principal components. Identification of MNNs and calculation of correction vectors are then performed in this low-dimensional space.

```{r}
# Using randomized SVD here, as this is faster than 
# irlba for file-backed matrices.
set.seed(1000101001)
mnn.out <- fastMNN(rescaled,
		   auto.merge=TRUE,
		   d=50,
		   k=20,
		   subset.row=chosen.hvgs,
		   BSPARAM=BiocSingular::RandomParam(deferred=TRUE))
mnn.out
```

The function returns a SCE object containing corrected values for downstream analyses like clustering or visualization. Each column of `mnn.out` corresponds to a cell in one of the batches, while each row corresponds to an input gene in chosen.hvgs. The batch field in the column metadata contains a vector specifying the batch of origin of each cell.

```{r}
print(head(mnn.out$batch))
```

The corrected matrix in the `reducedDims()` contains the low-dimensional corrected coordinates for all cells, which we will use in place of the principal components (PCs) in our downstream analyses.

```{r}
dim(reducedDim(mnn.out, "corrected"))
```

A reconstructed matrix in the `assays()` contains the corrected expression values for each gene in each cell, obtained by projecting the low-dimensional coordinates in corrected back into gene expression space. We do not recommend using this for anything other than visualization.

```{r}
print(assay(mnn.out, "reconstructed"))
```

The most relevant parameter for tuning `fastMNN()` is `k`, which specifies the number of nearest neighbors to consider when defining MNN pairs. This can be interpreted as the minimum anticipated frequency of any shared cell type or state in each batch. Increasing `k` will generally result in more aggressive merging as the algorithm is more generous in matching subpopulations across batches. It can occasionally be desirable to increase `k` if one clearly sees that the same cell types are not being adequately merged across batches.

<!--
See Chapter 32 for an example of a more complex fastMNN() merge involving several human pancreas datasets generated by different authors on different patients with different technologies.
-->

### Correction diagnostics

We cluster on the low-dimensional corrected coordinates to obtain a partitioning of the cells that serves as a proxy for the population structure. If the batch effect is successfully corrected, clusters corresponding to shared cell types or states should contain cells from multiple batches. We see that most clusters contain contributions from each batch after correction, consistent with our expectation that the batches are replicates of each other.

```{r}
library(scran)
snn.gr <- buildSNNGraph(mnn.out, use.dimred="corrected")
clusters.mnn <- igraph::cluster_walktrap(snn.gr)$membership
tab.mnn <- table(Cluster=clusters.mnn, Batch=mnn.out$batch)
tab.mnn
```

Cluster size and cell contribution by sample, with clusters sorted by size:

```{r}
tmpMat <- data.frame("clusters"=clusters.mnn, "batch"=mnn.out$batch)
tmpMatTab <- table(tmpMat)
sortVecNames <- tmpMatTab %>% rowSums %>% sort(decreasing=TRUE) %>% names
tmpMat$clusters <- factor(tmpMat$clusters, levels=sortVecNames)
tmpMatTab <- table(tmpMat)
tmpMatDf <- tmpMatTab[sortVecNames,] %>% data.frame()
p1 <- ggplot(data=tmpMatDf, aes(x=clusters,y=Freq, fill=batch)) +
	geom_col()
p2 <- ggplot(data=tmpMat, aes(x=clusters, fill=batch)) +
  geom_bar(position = "fill") +
  scale_y_continuous(labels = scales::percent)
```
```{r}
gridExtra::grid.arrange(p1, p2)
```

We can also compute the variation in the log-abundances to rank the clusters with the greatest variability in their proportional abundances across batches. We can then focus on batch-specific clusters that may be indicative of incomplete batch correction. Obviously, though, this diagnostic is subject to interpretation as the same outcome can be caused by batch-specific populations; some prior knowledge about the biological context is necessary to distinguish between these two possibilities.

```{r}
# Avoid minor difficulties with the 'table' class.
tab.mnn <- unclass(tab.mnn)

# Using a large pseudo.count to avoid unnecessarily
# large variances when the counts are low.
norm <- normalizeCounts(tab.mnn, pseudo_count=10)

# Ranking clusters by the largest variances.
rv <- rowVars(norm) %>% round(2)
DataFrame(Batch=tab.mnn, var=rv)[order(rv, decreasing=TRUE),]
```

We can also visualize the corrected coordinates using a t-SNE plot. The presence of visual clusters containing cells from all batches provides a comforting illusion that the correction was successful.

```{r}
library(scater)
set.seed(0010101010)
mnn.out <- runTSNE(mnn.out, dimred="corrected")

mnn.out$batch <- factor(mnn.out$batch)
plotTSNE(mnn.out, colour_by="batch")
#mnn.out$type <- gsub("_[1-4]","",mnn.out$batch)
#p <- plotTSNE(mnn.out, colour_by="batch", shape_by="type")
#p + facet_wrap(. ~ mnn.out$type)
```

For `fastMNN()`, one useful diagnostic is the proportion of variance within each batch that is lost during MNN correction. Specifically, this refers to the within-batch variance that is removed during orthogonalization with respect to the average correction vector at each merge step. This is returned via the `lost.var` field in the metadata of `mnn.out`, which contains a matrix of the variance lost in each batch (column) at each merge step (row).

```{r}
round(metadata(mnn.out)$merge.info$lost.var,2)
```

Large proportions of lost variance (>10%) suggest that correction is removing genuine biological heterogeneity. This would occur due to violations of the assumption of orthogonality between the batch effect and the biological subspace (Haghverdi et al. 2018). In this case, the proportion of lost variance is small, indicating that non-orthogonality is not a major concern.

The following t-SNE shows clusters identified:

```{r}
mnn.out$cluster <- paste0("c", clusters.mnn)
plotTSNE(mnn.out, colour_by="cluster")
```

The following t-SNE plots show expression levels of known cell type marker genes.

```{r}
genesToShow <- c(
		 "CD79A", # CD79A 	B ***
		 "CST3", # CST3 	monocytes ***
		 "CD3D", # CD3D 	 T cells ***
		 "HBA1" # HBA1 	 erythrocytes ***
	 	)

tmpInd <- which(rowData(uncorrected)$Symbol %in% genesToShow)
ensToShow <- rowData(uncorrected)$ensembl_gene_id[tmpInd]
```

B cells:

```{r}
genex <- ensToShow[1]
	p <- plotTSNE(mnn.out, colour_by = genex, by_exprs_values="reconstructed")
	p <- p + ggtitle(
			paste("B cells", genex,
			rowData(uncorrected)[genex,"Symbol"])
		)
	print(p)
```

T cells:

```{r}
genex <- ensToShow[3]
	p <- plotTSNE(mnn.out, colour_by = genex, by_exprs_values="reconstructed")
	p <- p + ggtitle(
			paste("T cells", genex,
			rowData(uncorrected)[genex,"Symbol"])
		)
	print(p)
```

monocytes:

```{r}
genex <- ensToShow[2]
	p <- plotTSNE(mnn.out, colour_by = genex, by_exprs_values="reconstructed")
	p <- p + ggtitle(
			paste("monocytes", genex,
			rowData(uncorrected)[genex,"Symbol"])
		)
	print(p)
```

erythrocytes:

```{r}
genex <- ensToShow[4]
	p <- plotTSNE(mnn.out, colour_by = genex, by_exprs_values="reconstructed")
	p <- p + ggtitle(
			paste("erythrocytes", genex,
			rowData(uncorrected)[genex,"Symbol"])
		)
	print(p)
```

Other genes (exercise)

```{r, eval=FALSE}
genesToShow2 <- c(
		 "IL7R", # IL7R, CCR7 	Naive CD4+ T
		 "CCR7", # IL7R, CCR7 	Naive CD4+ T
		 "S100A4", # IL7R, S100A4 	Memory CD4+
		 "CD14", # CD14, LYZ 	CD14+ Mono
		 "LYZ", # CD14, LYZ 	CD14+ Mono
		 "MS4A1", # MS4A1 	B
		 "CD8A", # CD8A 	CD8+ T
		 "FCGR3A", # FCGR3A, MS4A7 	FCGR3A+ Mono
		 "MS4A7", # FCGR3A, MS4A7 	FCGR3A+ Mono
		 "GNLY", # GNLY, NKG7 	NK
		 "NKG7", # GNLY, NKG7 	NK
		 "FCER1A", # DC
		 "CST3", # DC
		 "PPBP" # Platelet
		)
```

```{r, eval=FALSE}
tmpInd <- which(rowData(uncorrected)$Symbol %in% genesToShow2)
ensToShow <- rowData(uncorrected)$ensembl_gene_id[tmpInd]
```

```{r, eval=FALSE}
for (genex in ensToShow)
{
	p <- plotTSNE(mnn.out, colour_by = genex, by_exprs_values="reconstructed")
	p <- p + ggtitle(paste(genex, rowData(uncorrected)[genex,"Symbol"]))
	print(p)
}
```

## Preserving biological heterogeneity

### Comparison to within-batch clusters

Another useful diagnostic check is to compare the clustering within each batch to the clustering of the merged data. Accurate data integration should preserve variance within each batch as there should be nothing to remove between cells in the same batch. This check complements the previously mentioned diagnostics that only focus on the removal of differences between batches. Specifically, it protects us against cases where the correction method simply aggregates all cells together, which would achieve perfect mixing but also discard the biological heterogeneity of interest.

Ideally, we should see a many-to-1 mapping where the across-batch clustering is nested inside the within-batch clusterings. This indicates that any within-batch structure was preserved after correction while acknowledging that greater resolution is possible with more cells. In practice, more discrepancies can be expected even when the correction is perfect, due to the existence of closely related clusters that were arbitrarily separated in the within-batch clustering. As a general rule, we can be satisfied with the correction if the vast majority of entries are zero, though this may depend on whether specific clusters of interest are gained or lost.

One heatmap is generated for each dataset, where each entry is colored according to the number of cells with each pair of labels (before and after correction), on the log10 scale with a pseudocounts +10 for a smoother color transition (so a minimum value of log10(0+10) == 1). 

```{r}
library(pheatmap)

# have common color scale:
# for the number of cells on the log10 scale
breakVec = seq(0, 3, by = 0.2)
colPal <- colorRampPalette(rev(RColorBrewer::brewer.pal(n = 7, name = "RdYlBu")))(length(breakVec))
# Defines the vector of colors for the legend (it has to be of the same length as breakVec)


# For the first batch (adding +10 for a smoother color transition
# from zero to non-zero counts for any given matrix entry).
splIdx <- 1
tab <- table(paste("after", clusters.mnn[rescaled2$batch==splVec[splIdx]]),
    paste("before", colLabels(rescaled[[splIdx]])))
heat1 <- pheatmap(log10(tab+10), cluster_row=FALSE, cluster_col=FALSE,
		main=sprintf("%s comparison (number of cells, log10 scale)", splVec[splIdx]),
		silent=TRUE,
		col=colPal)

library(clustree)
combined <- cbind(cl.1=colLabels(rescaled[[splIdx]]),
		  cl.2=clusters.mnn[rescaled2$batch==splVec[splIdx]])
clustree(combined, prefix="cl.", edge_arrow=FALSE)
```

```{r}
# For the second batch.
splIdx <- 2
tab <- table(paste("after", clusters.mnn[rescaled2$batch==splVec[splIdx]]),
    paste("before", colLabels(rescaled[[splIdx]])))
heat2 <- pheatmap(log10(tab+10), cluster_row=FALSE, cluster_col=FALSE,
		main=sprintf("%s comparison (number of cells, log10 scale)", splVec[splIdx]),
		silent=TRUE)

library(clustree)
combined <- cbind(cl.1=colLabels(rescaled[[splIdx]]),
		  cl.2=clusters.mnn[rescaled2$batch==splVec[splIdx]])
clustree(combined, prefix="cl.", edge_arrow=FALSE)
```

```{r}
gridExtra::grid.arrange(heat1[[4]], heat2[[4]])
```

Another evaluation approach is to compute the **coassignment probabilities**, i.e. the probability that cells from two within-batch clusters are clustered together in the across-batch clustering. High probabilities off the diagonal indicate that within-batch clusters are merged in the across-batch analysis. We would generally expect low off-diagonal probabilities for most pairs of clusters, though this may not be reasonably possible if the within-batch clusters were poorly separated in the first place.

Coassignment probabilities for the within-batch clusters, based on coassignment of cells in the across-batch clusters obtained after MNN correction. One heatmap is generated for each sample, where each entry is colored according to the coassignment probability between each pair of within-batch clusters:

```{r}
# For the first batch.
splIdx <- 1
tab <- coassignProb(colLabels(rescaled[[splIdx]]), clusters.mnn[rescaled2$batch==splVec[splIdx]])
heat1 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    main=sprintf("%s probabilities", splVec[splIdx]),
    silent=TRUE)

# For the second batch.
splIdx <- 2
tab <- coassignProb(colLabels(rescaled[[splIdx]]), clusters.mnn[rescaled2$batch==splVec[splIdx]])
heat2 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    main=sprintf("%s probabilities", splVec[splIdx]),
    silent=TRUE)

gridExtra::grid.arrange(heat1[[4]], heat2[[4]])
```

Finally, we can summarize the agreement between clusterings by computing the **Rand index**. This provides a simple metric that we can use to assess the preservation of variation by different correction methods. Larger rand indices (i.e., closer to 1) are more desirable, though this must be balanced against the ability of each method to actually remove the batch effect.

```{r, warning=FALSE}
suppressMessages(library(fossil))
ri1 <- rand.index(as.integer(clusters.mnn[rescaled2$batch=="GSM3872443"]),
    as.integer(colLabels(rescaled[[1]])))
#ri1
```

```{r}
# same as above but with `bluster`
splIdx <- 1

library(bluster)
ri1 <- pairwiseRand(as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]]),
		     as.integer(colLabels(rescaled[[splIdx]])),
		     mode="index")
```

Rand index for `r splVec[splIdx]`: `r round(ri1,2)`.

```{r}
splIdx <- 2

ri2 <- pairwiseRand(as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]]),
		     as.integer(colLabels(rescaled[[splIdx]])),
		     mode="index")
#ri2
```

Rand index for `r splVec[splIdx]`: `r round(ri2,2)`

We can also break down the adjusted Rand index (ARI) into per-cluster ratios for more detailed diagnostics. For example, we could see low ratios off the diagonal if distinct clusters in the within-batch clustering were incorrectly aggregated in the merged clustering. Conversely, we might see low ratios on the diagonal if the correction inflated or introduced spurious heterogeneity inside a within-batch cluster.

```{r, eval=FALSE}
# For the first batch.
splIdx <- 1

tab <- pairwiseRand(
	as.integer(colLabels(rescaled[[splIdx]])),
  as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]])
)

heat1 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
    silent=TRUE)

# For the second batch.
splIdx <- 2

tab <- pairwiseRand(
	as.integer(colLabels(rescaled[[splIdx]])),
  as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]])
)

heat2 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
    silent=TRUE)

gridExtra::grid.arrange(heat1[[4]], heat2[[4]])
```

```{r}
# For the first batch.
splIdx <- 1

tab <- pairwiseRand(
	as.integer(colLabels(rescaled[[splIdx]])),
  as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]])
)
tab1 <- tab

#heat1 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
#    col=rev(viridis::magma(100)),
#    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
#    silent=TRUE)

# For the second batch.
splIdx <- 2

tab <- pairwiseRand(
	as.integer(colLabels(rescaled[[splIdx]])),
  as.integer(clusters.mnn[rescaled2$batch==splVec[splIdx]])
)
tab2 <- tab
rm(tab)

#heat2 <- pheatmap(tab, cluster_row=FALSE, cluster_col=FALSE,
#    col=rev(viridis::magma(100)),
#    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
#    silent=TRUE)

#gridExtra::grid.arrange(heat1[[4]], heat2[[4]])

  ## make breaks from combined range
Breaks <- seq(min(c(tab1, tab2), na.rm = TRUE),
              max(c(tab1, tab2), na.rm = TRUE),
              length = 100)

splIdx <- 1
heat1 <- pheatmap(tab1, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    breaks=Breaks,
    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
    silent=TRUE)

splIdx <- 2
heat2 <- pheatmap(tab2, cluster_row=FALSE, cluster_col=FALSE,
    col=rev(viridis::magma(100)),
    breaks=Breaks,
    main=sprintf("%s adjusted Rand index", splVec[splIdx]),
    silent=TRUE)

gridExtra::grid.arrange(heat1[[4]], heat2[[4]])
```

### Encouraging consistency with marker genes

In some situations, we will already have performed within-batch analyses to characterize salient aspects of population heterogeneity. This is not uncommon when merging datasets from different sources where each dataset has already been analyzed, annotated and interpreted separately. It is subsequently desirable for the integration procedure to retain these “known interesting” aspects of each dataset in the merged dataset. We can encourage this outcome by using the marker genes within each dataset as our selected feature set for `fastMNN()` and related methods. This focuses on the relevant heterogeneity and represents a semi-supervised approach.

To illustrate, we apply this strategy to our datasets. We identify the top marker genes from pairwise Wilcoxon ranked sum tests between every pair of clusters within each batch, analogous to the method used by `[SingleR](https://www.bioconductor.org/packages/release/bioc/html/SingleR.html)`. In this case, we use the top 10 marker genes but any value can be used depending on the acceptable trade-off between signal and noise (and speed). We then take the union across all comparisons in all batches and use that in place of our HVG set in `fastMNN()`.

```{r}
# Recall that groups for marker detection
# are automatically defined from 'colLabels()'. 
stats1 <- pairwiseWilcox(rescaled[[1]], direction="up")
markers1 <- getTopMarkers(stats1[[1]], stats1[[2]], n=10)

stats2 <- pairwiseWilcox(rescaled[[2]], direction="up")
markers2 <- getTopMarkers(stats2[[1]], stats2[[2]], n=10)

stats3 <- pairwiseWilcox(rescaled[[3]], direction="up")
markers3 <- getTopMarkers(stats3[[1]], stats3[[2]], n=10)

marker.set <- unique(unlist(c(unlist(markers1), unlist(markers2), unlist(markers3))))
length(marker.set) # getting the total number of genes selected in this manner.
```

Total number of genes selected in this manner: `r length(marker.set)`.

```{r}
set.seed(1000110)
mnn.out2 <- fastMNN(rescaled, subset.row=marker.set,
    BSPARAM=BiocSingular::RandomParam(deferred=TRUE))
# compute t-SNE:
mnn.out2 <- runTSNE(mnn.out2, dimred="corrected")
```

We can also visualize the corrected coordinates using a t-SNE plot:

```{r}
plotTSNE(mnn.out2, colour_by="batch")
```

A quick inspection indicates that the original within-batch structure is indeed preserved in the corrected data. This highlights the utility of a marker-based feature set for integrating datasets that have already been characterized separately in a manner that preserves existing interpretations of each dataset. We note that some within-batch clusters have merged, most likely due to the lack of robust separation in the first place, though this may also be treated as a diagnostic on the appropriateness of the integration depending on the context.

```{r}
gridExtra::grid.arrange(
    plotTSNE(mnn.out2[,mnn.out2$batch=="GSM3872442"], colour_by=I(colLabels(rescaled[[1]]))),
    plotTSNE(mnn.out2[,mnn.out2$batch=="GSM3872443"], colour_by=I(colLabels(rescaled[[2]]))),
    plotTSNE(mnn.out2[,mnn.out2$batch=="GSM3872444"], colour_by=I(colLabels(rescaled[[3]]))),
    ncol=2
)
```

```{r}
# B cells
genex <- ensToShow[1]
p <- plotTSNE(mnn.out2, colour_by = genex, by_exprs_values="reconstructed")
p <- p + ggtitle(
  		paste("B cells", genex,
			rowData(uncorrected)[genex,"Symbol"])
			)
print(p)
```

```{r, eval=FALSE}
# cannot find 'ENSG00000115523'
# TODO: use another gene?

# erythrocytes:
genex <- ensToShow[4]
p <- plotTSNE(mnn.out2, colour_by = genex, by_exprs_values="reconstructed")
p <- p + ggtitle(
			paste("erythrocytes", genex,
			rowData(uncorrected)[genex,"Symbol"])
			)
print(p)
```

We suggest limiting the use of per-gene corrected values to visualization, e.g., when coloring points on a t-SNE plot by per-cell expression. This can be more aesthetically pleasing than uncorrected expression values that may contain large shifts on the colour scale between cells in different batches. Use of the corrected values in any quantitative procedure should be treated with caution, and should be backed up by similar results from an analysis on the uncorrected values.

```{r, eval=FALSE, include=FALSE}
# B cells
plotExpression(uncorrected, ensToShow[1],
               x = "label",
               colour_by = "label")
```

B cells:

```{r}
m.out <- findMarkers(uncorrected,
                     clusters.mnn,
                     block=uncorrected$batch,
                     direction="up",
                     lfc=1,
                     row.data=rowData(uncorrected)[,c("ensembl_gene_id","Symbol"),
                                                   drop=FALSE])
```

<!-- B cells: -->

```{r, eval=FALSE, include=FALSE}
# A B cell subtype of some sort:
demo <- m.out[["1"]] %>% data.frame %>% arrange(FDR)
as.data.frame(demo[1:20,c("Symbol", "Top", "p.value", "FDR")]) 
demo[ensToShow[1],]

demo <- m.out[["6"]] %>% data.frame %>% arrange(FDR)
as.data.frame(demo[1:20,c("Symbol", "Top", "p.value", "FDR")]) 
demo[ensToShow[1],]

demo <- m.out[["10"]] %>% data.frame %>% arrange(FDR)
as.data.frame(demo[1:20,c("Symbol", "Top", "p.value", "FDR")]) 
demo[ensToShow[1],]
```

Erythrocytes:

```{r}
demo <- m.out[["11"]] %>% data.frame %>% arrange(FDR)
demo[ensToShow[4],]
as.data.frame(demo[1:20,c("Symbol", "Top", "p.value", "FDR")]) 
```

Expression level for the top gene, `r rownames(demo)[1]`:

```{r}
geneEnsId <- rownames(demo)[1]
plotExpression(uncorrected, x=I(factor(clusters.mnn)), 
    features=geneEnsId, colour_by="batch") + facet_wrap(~colour_by)
```

## Session information

```{r sessionInfo_dataSetIntegration_PBMMC}
sessionInfo()
```
